{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate data and calculate similarity\n",
    "\n",
    "The goal of this notebook is to determine how much of the structure in the original dataset (single experiment) is retained after adding some number of experiments.\n",
    "\n",
    "The approach is to,\n",
    "1. Generates simulated data by sampling from a trained VAE model.  Simulate ```num_simulated_samples```\n",
    "\n",
    "    Here we are trying to simulate data from a single experiment type.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/batch_effects/lib/python3.5/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n",
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "from plotnine import ggplot, ggtitle, xlab, ylab, geom_point, geom_line, aes\n",
    "from keras.models import load_model\n",
    "from sklearn.metrics.pairwise import euclidean_distances\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(action='ignore')\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import generate_data\n",
    "from functions import similarity_metric\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# User parameters\n",
    "NN_architecture = 'NN_2500_30'\n",
    "analysis_name = 'analysis_1'\n",
    "num_simulated_samples = 100\n",
    "experiment_id = 'E-GEOD-24036'\n",
    "color_by_field = 'genotype'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Input files\n",
    "\n",
    "# base dir on repo\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../..\")) \n",
    "\n",
    "# base dir on local machine for data storage\n",
    "# os.makedirs doesn't recognize `~`\n",
    "local_dir = local_dir = os.path.abspath(os.path.join(os.getcwd(), \"../../../..\")) \n",
    "\n",
    "NN_dir = base_dir + \"/models/\" + NN_architecture\n",
    "latent_dim = NN_architecture.split('_')[-1]\n",
    "\n",
    "mapping_file = os.path.join(\n",
    "    base_dir,\n",
    "    \"data\",\n",
    "    \"metadata\",\n",
    "    \"sample_annotations.tsv\")\n",
    "\n",
    "normalized_data_file = os.path.join(\n",
    "    base_dir,\n",
    "    \"data\",\n",
    "    \"input\",\n",
    "    \"train_set_normalized.pcl\")\n",
    "\n",
    "model_encoder_file = glob.glob(os.path.join(\n",
    "    NN_dir,\n",
    "    \"tybalt_2layer_{}latent_encoder_model.h5\".format(latent_dim)))[0]\n",
    "\n",
    "weights_encoder_file = glob.glob(os.path.join(\n",
    "    NN_dir,\n",
    "    \"tybalt_2layer_{}latent_encoder_weights.h5\".format(latent_dim)))[0]\n",
    "\n",
    "model_decoder_file = glob.glob(os.path.join(\n",
    "    NN_dir,\n",
    "    \"tybalt_2layer_{}latent_decoder_model.h5\".format(latent_dim)))[0]\n",
    "\n",
    "weights_decoder_file = glob.glob(os.path.join(\n",
    "    NN_dir,\n",
    "    \"tybalt_2layer_{}latent_decoder_weights.h5\".format(latent_dim)))[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory already exists: \n",
      " /home/alexandra/Documents/Data/Batch_effects/simulated/analysis_1\n"
     ]
    }
   ],
   "source": [
    "# Output file\n",
    "new_dir = os.path.join(local_dir, \"Data\", \"Batch_effects\", \"simulated\")\n",
    "\n",
    "analysis_dir = os.path.join(new_dir, analysis_name)\n",
    "\n",
    "if os.path.exists(analysis_dir):\n",
    "    print('Directory already exists: \\n {}'.format(analysis_dir))\n",
    "else:\n",
    "    print('Creating new directory: \\n {}'.format(analysis_dir))\n",
    "os.makedirs(analysis_dir, exist_ok=True)\n",
    "\n",
    "    \n",
    "simulated_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"Data\",\n",
    "    \"Batch_effects\",\n",
    "    \"simulated\",\n",
    "    analysis_name,\n",
    "    \"simulated_data.txt.xz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load saved models\n",
    "loaded_model = load_model(model_encoder_file)\n",
    "loaded_decode_model = load_model(model_decoder_file)\n",
    "\n",
    "loaded_model.load_weights(weights_encoder_file)\n",
    "loaded_decode_model.load_weights(weights_decoder_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get experiment ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>ml_data_source</th>\n",
       "      <th>description</th>\n",
       "      <th>nucleic_acid</th>\n",
       "      <th>medium</th>\n",
       "      <th>genotype</th>\n",
       "      <th>od</th>\n",
       "      <th>growth_setting_1</th>\n",
       "      <th>growth_setting_2</th>\n",
       "      <th>strain</th>\n",
       "      <th>temperature</th>\n",
       "      <th>treatment</th>\n",
       "      <th>additional_notes</th>\n",
       "      <th>variant_phenotype</th>\n",
       "      <th>abx_marker</th>\n",
       "      <th>biotic_int_lv_2</th>\n",
       "      <th>biotic_int_lv_1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experiment</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>E-GEOD-46947</th>\n",
       "      <td>GSM1141730 1</td>\n",
       "      <td>GSM1141730_PA01_ZnO_PZO_.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PAO1 LB aerated  5 h wi...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>planktonic</td>\n",
       "      <td>aerated</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>37.0</td>\n",
       "      <td>1 mM ZnO nanoparticles</td>\n",
       "      <td>Grown for 5h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-46947</th>\n",
       "      <td>GSM1141729 1</td>\n",
       "      <td>GSM1141729_PA01_none_PC_.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PAO1 LB aerated  5 h</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>planktonic</td>\n",
       "      <td>aerated</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Grown for 5h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608059 1</td>\n",
       "      <td>GSM1608059_Planktonic_1.CEL</td>\n",
       "      <td>PAO1 WT. Planktonic. Rep1</td>\n",
       "      <td>RNA</td>\n",
       "      <td>PBM plus 1 g / L glucose.</td>\n",
       "      <td>WT</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Aerated</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Grown shaking at 200rpm</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608060 1</td>\n",
       "      <td>GSM1608060_Planktonic_2.CEL</td>\n",
       "      <td>PAO1 WT. Planktonic. Rep2</td>\n",
       "      <td>RNA</td>\n",
       "      <td>PBM plus 1 g / L glucose.</td>\n",
       "      <td>WT</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Aerated</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Grown shaking at 200rpm</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608061 1</td>\n",
       "      <td>GSM1608061_Planktonic_3.CEL</td>\n",
       "      <td>PAO1 WT. Planktonic. Rep3</td>\n",
       "      <td>RNA</td>\n",
       "      <td>PBM plus 1 g / L glucose.</td>\n",
       "      <td>WT</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Aerated</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Grown shaking at 200rpm</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name                ml_data_source  \\\n",
       "experiment                                                 \n",
       "E-GEOD-46947  GSM1141730 1  GSM1141730_PA01_ZnO_PZO_.CEL   \n",
       "E-GEOD-46947  GSM1141729 1  GSM1141729_PA01_none_PC_.CEL   \n",
       "E-GEOD-65882  GSM1608059 1   GSM1608059_Planktonic_1.CEL   \n",
       "E-GEOD-65882  GSM1608060 1   GSM1608060_Planktonic_2.CEL   \n",
       "E-GEOD-65882  GSM1608061 1   GSM1608061_Planktonic_3.CEL   \n",
       "\n",
       "                                                    description nucleic_acid  \\\n",
       "experiment                                                                     \n",
       "E-GEOD-46947  Pseudomonas aeruginosa PAO1 LB aerated  5 h wi...          RNA   \n",
       "E-GEOD-46947        Pseudomonas aeruginosa PAO1 LB aerated  5 h          RNA   \n",
       "E-GEOD-65882                          PAO1 WT. Planktonic. Rep1          RNA   \n",
       "E-GEOD-65882                          PAO1 WT. Planktonic. Rep2          RNA   \n",
       "E-GEOD-65882                          PAO1 WT. Planktonic. Rep3          RNA   \n",
       "\n",
       "                                  medium genotype    od growth_setting_1  \\\n",
       "experiment                                                                 \n",
       "E-GEOD-46947                          LB      NaN   NaN       planktonic   \n",
       "E-GEOD-46947                          LB      NaN   NaN       planktonic   \n",
       "E-GEOD-65882  PBM plus 1 g / L glucose.        WT  0.26      Planktonic    \n",
       "E-GEOD-65882  PBM plus 1 g / L glucose.        WT  0.26      Planktonic    \n",
       "E-GEOD-65882  PBM plus 1 g / L glucose.        WT  0.26      Planktonic    \n",
       "\n",
       "             growth_setting_2 strain  temperature               treatment  \\\n",
       "experiment                                                                  \n",
       "E-GEOD-46947          aerated   PAO1         37.0  1 mM ZnO nanoparticles   \n",
       "E-GEOD-46947          aerated   PAO1         37.0                     NaN   \n",
       "E-GEOD-65882          Aerated   PAO1         37.0                     NaN   \n",
       "E-GEOD-65882          Aerated   PAO1         37.0                     NaN   \n",
       "E-GEOD-65882          Aerated   PAO1         37.0                     NaN   \n",
       "\n",
       "                     additional_notes variant_phenotype abx_marker  \\\n",
       "experiment                                                           \n",
       "E-GEOD-46947             Grown for 5h               NaN        NaN   \n",
       "E-GEOD-46947             Grown for 5h               NaN        NaN   \n",
       "E-GEOD-65882  Grown shaking at 200rpm               NaN        NaN   \n",
       "E-GEOD-65882  Grown shaking at 200rpm               NaN        NaN   \n",
       "E-GEOD-65882  Grown shaking at 200rpm               NaN        NaN   \n",
       "\n",
       "             biotic_int_lv_2 biotic_int_lv_1  \n",
       "experiment                                    \n",
       "E-GEOD-46947             NaN             NaN  \n",
       "E-GEOD-46947             NaN             NaN  \n",
       "E-GEOD-65882             NaN             NaN  \n",
       "E-GEOD-65882             NaN             NaN  \n",
       "E-GEOD-65882             NaN             NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in metadata\n",
    "metadata = pd.read_table(\n",
    "    mapping_file, \n",
    "    header=0, \n",
    "    sep='\\t', \n",
    "    index_col=0)\n",
    "\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>ml_data_source</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experiment</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>E-GEOD-46947</th>\n",
       "      <td>GSM1141730 1</td>\n",
       "      <td>GSM1141730_PA01_ZnO_PZO_.CEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-46947</th>\n",
       "      <td>GSM1141729 1</td>\n",
       "      <td>GSM1141729_PA01_none_PC_.CEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608059 1</td>\n",
       "      <td>GSM1608059_Planktonic_1.CEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608060 1</td>\n",
       "      <td>GSM1608060_Planktonic_2.CEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-65882</th>\n",
       "      <td>GSM1608061 1</td>\n",
       "      <td>GSM1608061_Planktonic_3.CEL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name                ml_data_source\n",
       "experiment                                              \n",
       "E-GEOD-46947  GSM1141730 1  GSM1141730_PA01_ZnO_PZO_.CEL\n",
       "E-GEOD-46947  GSM1141729 1  GSM1141729_PA01_none_PC_.CEL\n",
       "E-GEOD-65882  GSM1608059 1   GSM1608059_Planktonic_1.CEL\n",
       "E-GEOD-65882  GSM1608060 1   GSM1608060_Planktonic_2.CEL\n",
       "E-GEOD-65882  GSM1608061 1   GSM1608061_Planktonic_3.CEL"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "map_experiment_sample = metadata[['sample_name', 'ml_data_source']]\n",
    "map_experiment_sample.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 135 experiments in the compendium\n"
     ]
    }
   ],
   "source": [
    "experiment_ids = np.unique(np.array(map_experiment_sample.index))\n",
    "print(\"There are {} experiments in the compendium\".format(len(experiment_ids)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set user selected experiment id or randomly select experiment id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not experiment_id:\n",
    "    selected_experiment_id = np.random.RandomState(randomState).choice(experiment_ids)\n",
    "else:\n",
    "    selected_experiment_id = experiment_id"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get samples belonging to selected experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>ml_data_source</th>\n",
       "      <th>description</th>\n",
       "      <th>nucleic_acid</th>\n",
       "      <th>medium</th>\n",
       "      <th>genotype</th>\n",
       "      <th>od</th>\n",
       "      <th>growth_setting_1</th>\n",
       "      <th>growth_setting_2</th>\n",
       "      <th>strain</th>\n",
       "      <th>temperature</th>\n",
       "      <th>treatment</th>\n",
       "      <th>additional_notes</th>\n",
       "      <th>variant_phenotype</th>\n",
       "      <th>abx_marker</th>\n",
       "      <th>biotic_int_lv_2</th>\n",
       "      <th>biotic_int_lv_1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experiment</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>E-GEOD-24036</th>\n",
       "      <td>GSM591496 1</td>\n",
       "      <td>GSM591496.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PA14 WT, grown in LB, n...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>WT</td>\n",
       "      <td>2</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Shaking</td>\n",
       "      <td>PA14</td>\n",
       "      <td>37.0</td>\n",
       "      <td>no treatment</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-24036</th>\n",
       "      <td>GSM591601 1</td>\n",
       "      <td>GSM591601.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PA14 WT, grown in LB, n...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>WT</td>\n",
       "      <td>2</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Shaking</td>\n",
       "      <td>PA14</td>\n",
       "      <td>37.0</td>\n",
       "      <td>no treatment</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-24036</th>\n",
       "      <td>GSM591602 1</td>\n",
       "      <td>GSM591602.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PA14 WT, grown in LB, n...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>WT</td>\n",
       "      <td>2</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Shaking</td>\n",
       "      <td>PA14</td>\n",
       "      <td>37.0</td>\n",
       "      <td>no treatment</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-24036</th>\n",
       "      <td>GSM591603 1</td>\n",
       "      <td>GSM591603.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PA14 WT,  grown in LB w...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>WT</td>\n",
       "      <td>2</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Shaking</td>\n",
       "      <td>PA14</td>\n",
       "      <td>37.0</td>\n",
       "      <td>2-AA (3mM)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E-GEOD-24036</th>\n",
       "      <td>GSM591622 1</td>\n",
       "      <td>GSM591622.CEL</td>\n",
       "      <td>Pseudomonas aeruginosa PA14 WT,  grown in LB w...</td>\n",
       "      <td>RNA</td>\n",
       "      <td>LB</td>\n",
       "      <td>WT</td>\n",
       "      <td>2</td>\n",
       "      <td>Planktonic</td>\n",
       "      <td>Shaking</td>\n",
       "      <td>PA14</td>\n",
       "      <td>37.0</td>\n",
       "      <td>2-AA (3mM)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_name ml_data_source  \\\n",
       "experiment                                 \n",
       "E-GEOD-24036  GSM591496 1  GSM591496.CEL   \n",
       "E-GEOD-24036  GSM591601 1  GSM591601.CEL   \n",
       "E-GEOD-24036  GSM591602 1  GSM591602.CEL   \n",
       "E-GEOD-24036  GSM591603 1  GSM591603.CEL   \n",
       "E-GEOD-24036  GSM591622 1  GSM591622.CEL   \n",
       "\n",
       "                                                    description nucleic_acid  \\\n",
       "experiment                                                                     \n",
       "E-GEOD-24036  Pseudomonas aeruginosa PA14 WT, grown in LB, n...          RNA   \n",
       "E-GEOD-24036  Pseudomonas aeruginosa PA14 WT, grown in LB, n...          RNA   \n",
       "E-GEOD-24036  Pseudomonas aeruginosa PA14 WT, grown in LB, n...          RNA   \n",
       "E-GEOD-24036  Pseudomonas aeruginosa PA14 WT,  grown in LB w...          RNA   \n",
       "E-GEOD-24036  Pseudomonas aeruginosa PA14 WT,  grown in LB w...          RNA   \n",
       "\n",
       "             medium genotype od growth_setting_1 growth_setting_2 strain  \\\n",
       "experiment                                                                 \n",
       "E-GEOD-24036     LB       WT  2       Planktonic          Shaking   PA14   \n",
       "E-GEOD-24036     LB       WT  2       Planktonic          Shaking   PA14   \n",
       "E-GEOD-24036     LB       WT  2       Planktonic          Shaking   PA14   \n",
       "E-GEOD-24036     LB       WT  2       Planktonic          Shaking   PA14   \n",
       "E-GEOD-24036     LB       WT  2       Planktonic          Shaking   PA14   \n",
       "\n",
       "              temperature     treatment additional_notes variant_phenotype  \\\n",
       "experiment                                                                   \n",
       "E-GEOD-24036         37.0  no treatment              NaN               NaN   \n",
       "E-GEOD-24036         37.0  no treatment              NaN               NaN   \n",
       "E-GEOD-24036         37.0  no treatment              NaN               NaN   \n",
       "E-GEOD-24036         37.0    2-AA (3mM)              NaN               NaN   \n",
       "E-GEOD-24036         37.0    2-AA (3mM)              NaN               NaN   \n",
       "\n",
       "             abx_marker biotic_int_lv_2 biotic_int_lv_1  \n",
       "experiment                                               \n",
       "E-GEOD-24036        NaN             NaN             NaN  \n",
       "E-GEOD-24036        NaN             NaN             NaN  \n",
       "E-GEOD-24036        NaN             NaN             NaN  \n",
       "E-GEOD-24036        NaN             NaN             NaN  \n",
       "E-GEOD-24036        NaN             NaN             NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_metadata = metadata.loc[selected_experiment_id]\n",
    "selected_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 18 samples in experiment E-GEOD-24036\n"
     ]
    }
   ],
   "source": [
    "print(\"There are {} samples in experiment {}\".format(selected_metadata.shape[0], selected_experiment_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GSM591496.CEL',\n",
       " 'GSM591601.CEL',\n",
       " 'GSM591602.CEL',\n",
       " 'GSM591603.CEL',\n",
       " 'GSM591622.CEL',\n",
       " 'GSM591623.CEL',\n",
       " 'GSM591624.CEL',\n",
       " 'GSM591625.CEL',\n",
       " 'GSM591627.CEL',\n",
       " 'GSM597216.CEL',\n",
       " 'GSM597217.CEL',\n",
       " 'GSM597218.CEL',\n",
       " 'GSM597219.CEL',\n",
       " 'GSM597220.CEL',\n",
       " 'GSM597221.CEL',\n",
       " 'GSM597329.CEL',\n",
       " 'GSM597330.CEL',\n",
       " 'GSM597331.CEL']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_ids = list(selected_metadata['ml_data_source'])\n",
    "sample_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>PA0001</th>\n",
       "      <th>PA0002</th>\n",
       "      <th>PA0003</th>\n",
       "      <th>PA0004</th>\n",
       "      <th>PA0005</th>\n",
       "      <th>PA0006</th>\n",
       "      <th>PA0007</th>\n",
       "      <th>PA0008</th>\n",
       "      <th>PA0009</th>\n",
       "      <th>PA0010</th>\n",
       "      <th>...</th>\n",
       "      <th>PA5561</th>\n",
       "      <th>PA5562</th>\n",
       "      <th>PA5563</th>\n",
       "      <th>PA5564</th>\n",
       "      <th>PA5565</th>\n",
       "      <th>PA5566</th>\n",
       "      <th>PA5567</th>\n",
       "      <th>PA5568</th>\n",
       "      <th>PA5569</th>\n",
       "      <th>PA5570</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>05_PA14000-4-2_5-10-07_S2.CEL</th>\n",
       "      <td>0.853357</td>\n",
       "      <td>0.725280</td>\n",
       "      <td>0.640617</td>\n",
       "      <td>0.811465</td>\n",
       "      <td>0.694460</td>\n",
       "      <td>0.533958</td>\n",
       "      <td>0.158865</td>\n",
       "      <td>0.889579</td>\n",
       "      <td>0.884945</td>\n",
       "      <td>0.176558</td>\n",
       "      <td>...</td>\n",
       "      <td>0.466871</td>\n",
       "      <td>0.702785</td>\n",
       "      <td>0.790965</td>\n",
       "      <td>0.893249</td>\n",
       "      <td>0.789939</td>\n",
       "      <td>0.164157</td>\n",
       "      <td>0.970470</td>\n",
       "      <td>0.887472</td>\n",
       "      <td>0.900484</td>\n",
       "      <td>0.880012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54375-4-05.CEL</th>\n",
       "      <td>0.778790</td>\n",
       "      <td>0.767873</td>\n",
       "      <td>0.614859</td>\n",
       "      <td>0.907865</td>\n",
       "      <td>0.398800</td>\n",
       "      <td>0.460849</td>\n",
       "      <td>0.113876</td>\n",
       "      <td>0.761351</td>\n",
       "      <td>0.801740</td>\n",
       "      <td>0.222709</td>\n",
       "      <td>...</td>\n",
       "      <td>0.352020</td>\n",
       "      <td>0.694387</td>\n",
       "      <td>0.733186</td>\n",
       "      <td>0.639074</td>\n",
       "      <td>0.681204</td>\n",
       "      <td>0.110301</td>\n",
       "      <td>0.619554</td>\n",
       "      <td>0.747656</td>\n",
       "      <td>0.749893</td>\n",
       "      <td>0.805374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKGlu_plus_nt_7-8-09_s1.CEL</th>\n",
       "      <td>0.789155</td>\n",
       "      <td>0.729508</td>\n",
       "      <td>0.725913</td>\n",
       "      <td>0.718989</td>\n",
       "      <td>0.530160</td>\n",
       "      <td>0.466327</td>\n",
       "      <td>0.079507</td>\n",
       "      <td>0.731643</td>\n",
       "      <td>0.827707</td>\n",
       "      <td>0.241847</td>\n",
       "      <td>...</td>\n",
       "      <td>0.392405</td>\n",
       "      <td>0.700352</td>\n",
       "      <td>0.773422</td>\n",
       "      <td>0.791118</td>\n",
       "      <td>0.931585</td>\n",
       "      <td>0.172570</td>\n",
       "      <td>0.797148</td>\n",
       "      <td>0.753785</td>\n",
       "      <td>0.856253</td>\n",
       "      <td>0.811099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>anaerobic_NO3_1.CEL</th>\n",
       "      <td>0.716320</td>\n",
       "      <td>0.585079</td>\n",
       "      <td>0.390211</td>\n",
       "      <td>0.193248</td>\n",
       "      <td>0.279456</td>\n",
       "      <td>0.301781</td>\n",
       "      <td>0.513547</td>\n",
       "      <td>0.342051</td>\n",
       "      <td>0.415668</td>\n",
       "      <td>0.125914</td>\n",
       "      <td>...</td>\n",
       "      <td>0.398308</td>\n",
       "      <td>0.419574</td>\n",
       "      <td>0.593955</td>\n",
       "      <td>0.527203</td>\n",
       "      <td>0.706524</td>\n",
       "      <td>0.205510</td>\n",
       "      <td>0.504767</td>\n",
       "      <td>0.105662</td>\n",
       "      <td>0.363409</td>\n",
       "      <td>0.544780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>anaerobic_NO3_2.CEL</th>\n",
       "      <td>0.658015</td>\n",
       "      <td>0.592172</td>\n",
       "      <td>0.410331</td>\n",
       "      <td>0.245504</td>\n",
       "      <td>0.312028</td>\n",
       "      <td>0.305852</td>\n",
       "      <td>0.513499</td>\n",
       "      <td>0.336723</td>\n",
       "      <td>0.334226</td>\n",
       "      <td>0.162965</td>\n",
       "      <td>...</td>\n",
       "      <td>0.407801</td>\n",
       "      <td>0.478697</td>\n",
       "      <td>0.571460</td>\n",
       "      <td>0.473054</td>\n",
       "      <td>0.669643</td>\n",
       "      <td>0.155548</td>\n",
       "      <td>0.562927</td>\n",
       "      <td>0.049738</td>\n",
       "      <td>0.388931</td>\n",
       "      <td>0.548814</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 5549 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_symbol                      PA0001    PA0002    PA0003    PA0004  \\\n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.853357  0.725280  0.640617  0.811465   \n",
       "54375-4-05.CEL                 0.778790  0.767873  0.614859  0.907865   \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.789155  0.729508  0.725913  0.718989   \n",
       "anaerobic_NO3_1.CEL            0.716320  0.585079  0.390211  0.193248   \n",
       "anaerobic_NO3_2.CEL            0.658015  0.592172  0.410331  0.245504   \n",
       "\n",
       "Gene_symbol                      PA0005    PA0006    PA0007    PA0008  \\\n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.694460  0.533958  0.158865  0.889579   \n",
       "54375-4-05.CEL                 0.398800  0.460849  0.113876  0.761351   \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.530160  0.466327  0.079507  0.731643   \n",
       "anaerobic_NO3_1.CEL            0.279456  0.301781  0.513547  0.342051   \n",
       "anaerobic_NO3_2.CEL            0.312028  0.305852  0.513499  0.336723   \n",
       "\n",
       "Gene_symbol                      PA0009    PA0010    ...       PA5561  \\\n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.884945  0.176558    ...     0.466871   \n",
       "54375-4-05.CEL                 0.801740  0.222709    ...     0.352020   \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.827707  0.241847    ...     0.392405   \n",
       "anaerobic_NO3_1.CEL            0.415668  0.125914    ...     0.398308   \n",
       "anaerobic_NO3_2.CEL            0.334226  0.162965    ...     0.407801   \n",
       "\n",
       "Gene_symbol                      PA5562    PA5563    PA5564    PA5565  \\\n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.702785  0.790965  0.893249  0.789939   \n",
       "54375-4-05.CEL                 0.694387  0.733186  0.639074  0.681204   \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.700352  0.773422  0.791118  0.931585   \n",
       "anaerobic_NO3_1.CEL            0.419574  0.593955  0.527203  0.706524   \n",
       "anaerobic_NO3_2.CEL            0.478697  0.571460  0.473054  0.669643   \n",
       "\n",
       "Gene_symbol                      PA5566    PA5567    PA5568    PA5569  \\\n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.164157  0.970470  0.887472  0.900484   \n",
       "54375-4-05.CEL                 0.110301  0.619554  0.747656  0.749893   \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.172570  0.797148  0.753785  0.856253   \n",
       "anaerobic_NO3_1.CEL            0.205510  0.504767  0.105662  0.363409   \n",
       "anaerobic_NO3_2.CEL            0.155548  0.562927  0.049738  0.388931   \n",
       "\n",
       "Gene_symbol                      PA5570  \n",
       "05_PA14000-4-2_5-10-07_S2.CEL  0.880012  \n",
       "54375-4-05.CEL                 0.805374  \n",
       "AKGlu_plus_nt_7-8-09_s1.CEL    0.811099  \n",
       "anaerobic_NO3_1.CEL            0.544780  \n",
       "anaerobic_NO3_2.CEL            0.548814  \n",
       "\n",
       "[5 rows x 5549 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normalized_data = pd.read_table(\n",
    "    normalized_data_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0).T\n",
    "\n",
    "normalized_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found\n"
     ]
    }
   ],
   "source": [
    "# Check that samples are in gene expression dataset\n",
    "if any(x in list(normalized_data.index) for x in sample_ids):\n",
    "    print('found')\n",
    "else:\n",
    "    print('not found')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>PA0001</th>\n",
       "      <th>PA0002</th>\n",
       "      <th>PA0003</th>\n",
       "      <th>PA0004</th>\n",
       "      <th>PA0005</th>\n",
       "      <th>PA0006</th>\n",
       "      <th>PA0007</th>\n",
       "      <th>PA0008</th>\n",
       "      <th>PA0009</th>\n",
       "      <th>PA0010</th>\n",
       "      <th>...</th>\n",
       "      <th>PA5561</th>\n",
       "      <th>PA5562</th>\n",
       "      <th>PA5563</th>\n",
       "      <th>PA5564</th>\n",
       "      <th>PA5565</th>\n",
       "      <th>PA5566</th>\n",
       "      <th>PA5567</th>\n",
       "      <th>PA5568</th>\n",
       "      <th>PA5569</th>\n",
       "      <th>PA5570</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM591496.CEL</th>\n",
       "      <td>0.758438</td>\n",
       "      <td>0.808715</td>\n",
       "      <td>0.626747</td>\n",
       "      <td>0.774033</td>\n",
       "      <td>0.514039</td>\n",
       "      <td>0.383739</td>\n",
       "      <td>0.343325</td>\n",
       "      <td>0.719577</td>\n",
       "      <td>0.541627</td>\n",
       "      <td>0.216560</td>\n",
       "      <td>...</td>\n",
       "      <td>0.245720</td>\n",
       "      <td>0.671244</td>\n",
       "      <td>0.670357</td>\n",
       "      <td>0.767884</td>\n",
       "      <td>0.640626</td>\n",
       "      <td>0.151623</td>\n",
       "      <td>0.659674</td>\n",
       "      <td>0.643497</td>\n",
       "      <td>0.746590</td>\n",
       "      <td>0.707505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591601.CEL</th>\n",
       "      <td>0.702824</td>\n",
       "      <td>0.755786</td>\n",
       "      <td>0.537629</td>\n",
       "      <td>0.755608</td>\n",
       "      <td>0.462175</td>\n",
       "      <td>0.316667</td>\n",
       "      <td>0.308082</td>\n",
       "      <td>0.704596</td>\n",
       "      <td>0.558551</td>\n",
       "      <td>0.160684</td>\n",
       "      <td>...</td>\n",
       "      <td>0.214021</td>\n",
       "      <td>0.653341</td>\n",
       "      <td>0.668257</td>\n",
       "      <td>0.686663</td>\n",
       "      <td>0.630709</td>\n",
       "      <td>0.182788</td>\n",
       "      <td>0.703150</td>\n",
       "      <td>0.649179</td>\n",
       "      <td>0.739723</td>\n",
       "      <td>0.728428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591602.CEL</th>\n",
       "      <td>0.776721</td>\n",
       "      <td>0.736631</td>\n",
       "      <td>0.582667</td>\n",
       "      <td>0.749297</td>\n",
       "      <td>0.486610</td>\n",
       "      <td>0.302209</td>\n",
       "      <td>0.323932</td>\n",
       "      <td>0.782628</td>\n",
       "      <td>0.807336</td>\n",
       "      <td>0.225348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.171650</td>\n",
       "      <td>0.560667</td>\n",
       "      <td>0.688716</td>\n",
       "      <td>0.699709</td>\n",
       "      <td>0.647261</td>\n",
       "      <td>0.163239</td>\n",
       "      <td>0.589900</td>\n",
       "      <td>0.649514</td>\n",
       "      <td>0.724675</td>\n",
       "      <td>0.696667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591603.CEL</th>\n",
       "      <td>0.650666</td>\n",
       "      <td>0.733063</td>\n",
       "      <td>0.484848</td>\n",
       "      <td>0.682237</td>\n",
       "      <td>0.404460</td>\n",
       "      <td>0.314539</td>\n",
       "      <td>0.461867</td>\n",
       "      <td>0.569701</td>\n",
       "      <td>0.396897</td>\n",
       "      <td>0.201233</td>\n",
       "      <td>...</td>\n",
       "      <td>0.248430</td>\n",
       "      <td>0.533048</td>\n",
       "      <td>0.536321</td>\n",
       "      <td>0.565483</td>\n",
       "      <td>0.565437</td>\n",
       "      <td>0.290283</td>\n",
       "      <td>0.518061</td>\n",
       "      <td>0.411037</td>\n",
       "      <td>0.675788</td>\n",
       "      <td>0.625928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591622.CEL</th>\n",
       "      <td>0.667370</td>\n",
       "      <td>0.710359</td>\n",
       "      <td>0.466625</td>\n",
       "      <td>0.627510</td>\n",
       "      <td>0.389397</td>\n",
       "      <td>0.330307</td>\n",
       "      <td>0.557265</td>\n",
       "      <td>0.504433</td>\n",
       "      <td>0.340656</td>\n",
       "      <td>0.158090</td>\n",
       "      <td>...</td>\n",
       "      <td>0.247211</td>\n",
       "      <td>0.480189</td>\n",
       "      <td>0.524795</td>\n",
       "      <td>0.570522</td>\n",
       "      <td>0.514418</td>\n",
       "      <td>0.238918</td>\n",
       "      <td>0.437085</td>\n",
       "      <td>0.393830</td>\n",
       "      <td>0.658838</td>\n",
       "      <td>0.614567</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 5549 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_symbol      PA0001    PA0002    PA0003    PA0004    PA0005    PA0006  \\\n",
       "GSM591496.CEL  0.758438  0.808715  0.626747  0.774033  0.514039  0.383739   \n",
       "GSM591601.CEL  0.702824  0.755786  0.537629  0.755608  0.462175  0.316667   \n",
       "GSM591602.CEL  0.776721  0.736631  0.582667  0.749297  0.486610  0.302209   \n",
       "GSM591603.CEL  0.650666  0.733063  0.484848  0.682237  0.404460  0.314539   \n",
       "GSM591622.CEL  0.667370  0.710359  0.466625  0.627510  0.389397  0.330307   \n",
       "\n",
       "Gene_symbol      PA0007    PA0008    PA0009    PA0010    ...       PA5561  \\\n",
       "GSM591496.CEL  0.343325  0.719577  0.541627  0.216560    ...     0.245720   \n",
       "GSM591601.CEL  0.308082  0.704596  0.558551  0.160684    ...     0.214021   \n",
       "GSM591602.CEL  0.323932  0.782628  0.807336  0.225348    ...     0.171650   \n",
       "GSM591603.CEL  0.461867  0.569701  0.396897  0.201233    ...     0.248430   \n",
       "GSM591622.CEL  0.557265  0.504433  0.340656  0.158090    ...     0.247211   \n",
       "\n",
       "Gene_symbol      PA5562    PA5563    PA5564    PA5565    PA5566    PA5567  \\\n",
       "GSM591496.CEL  0.671244  0.670357  0.767884  0.640626  0.151623  0.659674   \n",
       "GSM591601.CEL  0.653341  0.668257  0.686663  0.630709  0.182788  0.703150   \n",
       "GSM591602.CEL  0.560667  0.688716  0.699709  0.647261  0.163239  0.589900   \n",
       "GSM591603.CEL  0.533048  0.536321  0.565483  0.565437  0.290283  0.518061   \n",
       "GSM591622.CEL  0.480189  0.524795  0.570522  0.514418  0.238918  0.437085   \n",
       "\n",
       "Gene_symbol      PA5568    PA5569    PA5570  \n",
       "GSM591496.CEL  0.643497  0.746590  0.707505  \n",
       "GSM591601.CEL  0.649179  0.739723  0.728428  \n",
       "GSM591602.CEL  0.649514  0.724675  0.696667  \n",
       "GSM591603.CEL  0.411037  0.675788  0.625928  \n",
       "GSM591622.CEL  0.393830  0.658838  0.614567  \n",
       "\n",
       "[5 rows x 5549 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_data_df = normalized_data.loc[sample_ids]\n",
    "\n",
    "selected_data_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Embed samples into latent space using trained VAE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>23</th>\n",
       "      <th>24</th>\n",
       "      <th>25</th>\n",
       "      <th>26</th>\n",
       "      <th>27</th>\n",
       "      <th>28</th>\n",
       "      <th>29</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM591496.CEL</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.319957</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.248557</td>\n",
       "      <td>4.816005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.494926</td>\n",
       "      <td>5.298022</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.513655</td>\n",
       "      <td>4.027642</td>\n",
       "      <td>2.254136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.111289</td>\n",
       "      <td>2.762950</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591601.CEL</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.105274</td>\n",
       "      <td>2.417805</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.623933</td>\n",
       "      <td>4.440513</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.778685</td>\n",
       "      <td>4.065911</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.891980</td>\n",
       "      <td>3.340123</td>\n",
       "      <td>1.900610</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.077000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591602.CEL</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.081273</td>\n",
       "      <td>2.659945</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.612093</td>\n",
       "      <td>4.991161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.569157</td>\n",
       "      <td>4.860400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.914119</td>\n",
       "      <td>1.246660</td>\n",
       "      <td>3.097568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.688240</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591603.CEL</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.320578</td>\n",
       "      <td>3.445772</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.248308</td>\n",
       "      <td>6.596386</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.226627</td>\n",
       "      <td>1.418302</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.664482</td>\n",
       "      <td>2.135759</td>\n",
       "      <td>2.746596</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.410463</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.295563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591622.CEL</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.321950</td>\n",
       "      <td>3.624446</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.290001</td>\n",
       "      <td>6.581908</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.033994</td>\n",
       "      <td>...</td>\n",
       "      <td>0.358296</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.796658</td>\n",
       "      <td>2.299481</td>\n",
       "      <td>2.677809</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.601242</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                0         1         2    3         4         5    6    7   \\\n",
       "GSM591496.CEL  0.0  0.000000  2.319957  0.0  0.248557  4.816005  0.0  0.0   \n",
       "GSM591601.CEL  0.0  0.105274  2.417805  0.0  0.623933  4.440513  0.0  0.0   \n",
       "GSM591602.CEL  0.0  0.081273  2.659945  0.0  0.612093  4.991161  0.0  0.0   \n",
       "GSM591603.CEL  0.0  1.320578  3.445772  0.0  0.248308  6.596386  0.0  0.0   \n",
       "GSM591622.CEL  0.0  1.321950  3.624446  0.0  0.290001  6.581908  0.0  0.0   \n",
       "\n",
       "                     8         9     ...           20   21        22  \\\n",
       "GSM591496.CEL  3.494926  5.298022    ...     0.000000  0.0  0.513655   \n",
       "GSM591601.CEL  2.778685  4.065911    ...     0.000000  0.0  0.891980   \n",
       "GSM591602.CEL  3.569157  4.860400    ...     0.000000  0.0  0.914119   \n",
       "GSM591603.CEL  0.226627  1.418302    ...     0.000000  0.0  0.664482   \n",
       "GSM591622.CEL  0.000000  1.033994    ...     0.358296  0.0  0.796658   \n",
       "\n",
       "                     23        24   25        26        27   28        29  \n",
       "GSM591496.CEL  4.027642  2.254136  0.0  0.111289  2.762950  0.0  0.000000  \n",
       "GSM591601.CEL  3.340123  1.900610  0.0  0.000000  3.077000  0.0  0.000000  \n",
       "GSM591602.CEL  1.246660  3.097568  0.0  0.000000  2.688240  0.0  0.000000  \n",
       "GSM591603.CEL  2.135759  2.746596  0.0  0.000000  0.410463  0.0  0.295563  \n",
       "GSM591622.CEL  2.299481  2.677809  0.0  0.000000  0.601242  0.0  0.000000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Encode into latent space\n",
    "data_encoded = loaded_model.predict_on_batch(selected_data_df)\n",
    "data_encoded_df = pd.DataFrame(data_encoded, index=selected_data_df.index)\n",
    "\n",
    "data_encoded_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate simulated gene expression data \n",
    "\n",
    "Shift samples in a different direction while maintaining the distance between samples.\n",
    "\n",
    "The below implementation is considering each pair of closest points simulating a new sample by stepping in a random direction but preserving the distance between those two point.  Thus, by design, this implementation is preserving local structure of the experiment but not necessarily the global strucuture."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>23</th>\n",
       "      <th>24</th>\n",
       "      <th>25</th>\n",
       "      <th>26</th>\n",
       "      <th>27</th>\n",
       "      <th>28</th>\n",
       "      <th>29</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM591496.CEL</th>\n",
       "      <td>-1.085631</td>\n",
       "      <td>0.997345</td>\n",
       "      <td>2.602936</td>\n",
       "      <td>-1.506295</td>\n",
       "      <td>-0.330044</td>\n",
       "      <td>6.467442</td>\n",
       "      <td>-2.426679</td>\n",
       "      <td>-0.428913</td>\n",
       "      <td>4.760863</td>\n",
       "      <td>4.431282</td>\n",
       "      <td>...</td>\n",
       "      <td>0.737369</td>\n",
       "      <td>1.490732</td>\n",
       "      <td>-0.422179</td>\n",
       "      <td>5.203471</td>\n",
       "      <td>1.000255</td>\n",
       "      <td>-0.637752</td>\n",
       "      <td>1.018394</td>\n",
       "      <td>1.334269</td>\n",
       "      <td>-0.140069</td>\n",
       "      <td>-0.861755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591601.CEL</th>\n",
       "      <td>-1.085631</td>\n",
       "      <td>1.102619</td>\n",
       "      <td>2.700784</td>\n",
       "      <td>-1.506295</td>\n",
       "      <td>0.045333</td>\n",
       "      <td>6.091949</td>\n",
       "      <td>-2.426679</td>\n",
       "      <td>-0.428913</td>\n",
       "      <td>4.044621</td>\n",
       "      <td>3.199171</td>\n",
       "      <td>...</td>\n",
       "      <td>0.737369</td>\n",
       "      <td>1.490732</td>\n",
       "      <td>-0.043853</td>\n",
       "      <td>4.515953</td>\n",
       "      <td>0.646730</td>\n",
       "      <td>-0.637752</td>\n",
       "      <td>0.907105</td>\n",
       "      <td>1.648319</td>\n",
       "      <td>-0.140069</td>\n",
       "      <td>-0.861755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591602.CEL</th>\n",
       "      <td>-1.085631</td>\n",
       "      <td>1.078619</td>\n",
       "      <td>2.942924</td>\n",
       "      <td>-1.506295</td>\n",
       "      <td>0.033493</td>\n",
       "      <td>6.642598</td>\n",
       "      <td>-2.426679</td>\n",
       "      <td>-0.428913</td>\n",
       "      <td>4.835093</td>\n",
       "      <td>3.993660</td>\n",
       "      <td>...</td>\n",
       "      <td>0.737369</td>\n",
       "      <td>1.490732</td>\n",
       "      <td>-0.021715</td>\n",
       "      <td>2.422489</td>\n",
       "      <td>1.843688</td>\n",
       "      <td>-0.637752</td>\n",
       "      <td>0.907105</td>\n",
       "      <td>1.259559</td>\n",
       "      <td>-0.140069</td>\n",
       "      <td>-0.861755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591603.CEL</th>\n",
       "      <td>-1.085631</td>\n",
       "      <td>2.317923</td>\n",
       "      <td>3.728750</td>\n",
       "      <td>-1.506295</td>\n",
       "      <td>-0.330292</td>\n",
       "      <td>8.247823</td>\n",
       "      <td>-2.426679</td>\n",
       "      <td>-0.428913</td>\n",
       "      <td>1.492563</td>\n",
       "      <td>0.551561</td>\n",
       "      <td>...</td>\n",
       "      <td>0.737369</td>\n",
       "      <td>1.490732</td>\n",
       "      <td>-0.271352</td>\n",
       "      <td>3.311588</td>\n",
       "      <td>1.492715</td>\n",
       "      <td>-0.637752</td>\n",
       "      <td>0.907105</td>\n",
       "      <td>-1.018218</td>\n",
       "      <td>-0.140069</td>\n",
       "      <td>-0.566192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591622.CEL</th>\n",
       "      <td>-1.085631</td>\n",
       "      <td>2.319295</td>\n",
       "      <td>3.907425</td>\n",
       "      <td>-1.506295</td>\n",
       "      <td>-0.288599</td>\n",
       "      <td>8.233345</td>\n",
       "      <td>-2.426679</td>\n",
       "      <td>-0.428913</td>\n",
       "      <td>1.265936</td>\n",
       "      <td>0.167254</td>\n",
       "      <td>...</td>\n",
       "      <td>1.095664</td>\n",
       "      <td>1.490732</td>\n",
       "      <td>-0.139176</td>\n",
       "      <td>3.475310</td>\n",
       "      <td>1.423928</td>\n",
       "      <td>-0.637752</td>\n",
       "      <td>0.907105</td>\n",
       "      <td>-0.827439</td>\n",
       "      <td>-0.140069</td>\n",
       "      <td>-0.861755</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     0         1         2         3         4         5   \\\n",
       "GSM591496.CEL -1.085631  0.997345  2.602936 -1.506295 -0.330044  6.467442   \n",
       "GSM591601.CEL -1.085631  1.102619  2.700784 -1.506295  0.045333  6.091949   \n",
       "GSM591602.CEL -1.085631  1.078619  2.942924 -1.506295  0.033493  6.642598   \n",
       "GSM591603.CEL -1.085631  2.317923  3.728750 -1.506295 -0.330292  8.247823   \n",
       "GSM591622.CEL -1.085631  2.319295  3.907425 -1.506295 -0.288599  8.233345   \n",
       "\n",
       "                     6         7         8         9     ...           20  \\\n",
       "GSM591496.CEL -2.426679 -0.428913  4.760863  4.431282    ...     0.737369   \n",
       "GSM591601.CEL -2.426679 -0.428913  4.044621  3.199171    ...     0.737369   \n",
       "GSM591602.CEL -2.426679 -0.428913  4.835093  3.993660    ...     0.737369   \n",
       "GSM591603.CEL -2.426679 -0.428913  1.492563  0.551561    ...     0.737369   \n",
       "GSM591622.CEL -2.426679 -0.428913  1.265936  0.167254    ...     1.095664   \n",
       "\n",
       "                     21        22        23        24        25        26  \\\n",
       "GSM591496.CEL  1.490732 -0.422179  5.203471  1.000255 -0.637752  1.018394   \n",
       "GSM591601.CEL  1.490732 -0.043853  4.515953  0.646730 -0.637752  0.907105   \n",
       "GSM591602.CEL  1.490732 -0.021715  2.422489  1.843688 -0.637752  0.907105   \n",
       "GSM591603.CEL  1.490732 -0.271352  3.311588  1.492715 -0.637752  0.907105   \n",
       "GSM591622.CEL  1.490732 -0.139176  3.475310  1.423928 -0.637752  0.907105   \n",
       "\n",
       "                     27        28        29  \n",
       "GSM591496.CEL  1.334269 -0.140069 -0.861755  \n",
       "GSM591601.CEL  1.648319 -0.140069 -0.861755  \n",
       "GSM591602.CEL  1.259559 -0.140069 -0.861755  \n",
       "GSM591603.CEL -1.018218 -0.140069 -0.566192  \n",
       "GSM591622.CEL -0.827439 -0.140069 -0.861755  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "randomState = 123\n",
    "seed(randomState)\n",
    "simulated_data_encoded_df = data_encoded_df + np.random.RandomState(randomState).normal(0, 1.0, int(latent_dim))\n",
    "\n",
    "simulated_data_encoded_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Decode simulated data into gene space"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>5540</th>\n",
       "      <th>5541</th>\n",
       "      <th>5542</th>\n",
       "      <th>5543</th>\n",
       "      <th>5544</th>\n",
       "      <th>5545</th>\n",
       "      <th>5546</th>\n",
       "      <th>5547</th>\n",
       "      <th>5548</th>\n",
       "      <th>color_by</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM591496.CEL</th>\n",
       "      <td>0.797644</td>\n",
       "      <td>0.799561</td>\n",
       "      <td>0.573416</td>\n",
       "      <td>0.827819</td>\n",
       "      <td>0.398233</td>\n",
       "      <td>0.306977</td>\n",
       "      <td>0.293209</td>\n",
       "      <td>0.673541</td>\n",
       "      <td>0.506386</td>\n",
       "      <td>0.189901</td>\n",
       "      <td>...</td>\n",
       "      <td>0.698717</td>\n",
       "      <td>0.720014</td>\n",
       "      <td>0.702985</td>\n",
       "      <td>0.622383</td>\n",
       "      <td>0.189934</td>\n",
       "      <td>0.688720</td>\n",
       "      <td>0.569247</td>\n",
       "      <td>0.673973</td>\n",
       "      <td>0.691933</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591601.CEL</th>\n",
       "      <td>0.765701</td>\n",
       "      <td>0.777498</td>\n",
       "      <td>0.544344</td>\n",
       "      <td>0.795009</td>\n",
       "      <td>0.396641</td>\n",
       "      <td>0.310865</td>\n",
       "      <td>0.301406</td>\n",
       "      <td>0.651905</td>\n",
       "      <td>0.492264</td>\n",
       "      <td>0.197113</td>\n",
       "      <td>...</td>\n",
       "      <td>0.677271</td>\n",
       "      <td>0.689790</td>\n",
       "      <td>0.667960</td>\n",
       "      <td>0.611020</td>\n",
       "      <td>0.214126</td>\n",
       "      <td>0.659484</td>\n",
       "      <td>0.548962</td>\n",
       "      <td>0.658613</td>\n",
       "      <td>0.679130</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591602.CEL</th>\n",
       "      <td>0.763721</td>\n",
       "      <td>0.780755</td>\n",
       "      <td>0.511801</td>\n",
       "      <td>0.777404</td>\n",
       "      <td>0.407298</td>\n",
       "      <td>0.283069</td>\n",
       "      <td>0.332274</td>\n",
       "      <td>0.683959</td>\n",
       "      <td>0.531228</td>\n",
       "      <td>0.184785</td>\n",
       "      <td>...</td>\n",
       "      <td>0.649350</td>\n",
       "      <td>0.660366</td>\n",
       "      <td>0.643134</td>\n",
       "      <td>0.628420</td>\n",
       "      <td>0.203199</td>\n",
       "      <td>0.646247</td>\n",
       "      <td>0.535502</td>\n",
       "      <td>0.659145</td>\n",
       "      <td>0.686068</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591603.CEL</th>\n",
       "      <td>0.679670</td>\n",
       "      <td>0.731089</td>\n",
       "      <td>0.412020</td>\n",
       "      <td>0.687244</td>\n",
       "      <td>0.342096</td>\n",
       "      <td>0.261917</td>\n",
       "      <td>0.457032</td>\n",
       "      <td>0.492887</td>\n",
       "      <td>0.324505</td>\n",
       "      <td>0.199644</td>\n",
       "      <td>...</td>\n",
       "      <td>0.563030</td>\n",
       "      <td>0.578363</td>\n",
       "      <td>0.457818</td>\n",
       "      <td>0.496192</td>\n",
       "      <td>0.237614</td>\n",
       "      <td>0.566808</td>\n",
       "      <td>0.327465</td>\n",
       "      <td>0.524913</td>\n",
       "      <td>0.567808</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM591622.CEL</th>\n",
       "      <td>0.667093</td>\n",
       "      <td>0.726802</td>\n",
       "      <td>0.409756</td>\n",
       "      <td>0.690103</td>\n",
       "      <td>0.330283</td>\n",
       "      <td>0.266757</td>\n",
       "      <td>0.454745</td>\n",
       "      <td>0.488113</td>\n",
       "      <td>0.318980</td>\n",
       "      <td>0.199653</td>\n",
       "      <td>...</td>\n",
       "      <td>0.545345</td>\n",
       "      <td>0.572971</td>\n",
       "      <td>0.444497</td>\n",
       "      <td>0.476583</td>\n",
       "      <td>0.241327</td>\n",
       "      <td>0.554977</td>\n",
       "      <td>0.313927</td>\n",
       "      <td>0.518031</td>\n",
       "      <td>0.565803</td>\n",
       "      <td>WT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 5550 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      0         1         2         3         4         5  \\\n",
       "GSM591496.CEL  0.797644  0.799561  0.573416  0.827819  0.398233  0.306977   \n",
       "GSM591601.CEL  0.765701  0.777498  0.544344  0.795009  0.396641  0.310865   \n",
       "GSM591602.CEL  0.763721  0.780755  0.511801  0.777404  0.407298  0.283069   \n",
       "GSM591603.CEL  0.679670  0.731089  0.412020  0.687244  0.342096  0.261917   \n",
       "GSM591622.CEL  0.667093  0.726802  0.409756  0.690103  0.330283  0.266757   \n",
       "\n",
       "                      6         7         8         9    ...         5540  \\\n",
       "GSM591496.CEL  0.293209  0.673541  0.506386  0.189901    ...     0.698717   \n",
       "GSM591601.CEL  0.301406  0.651905  0.492264  0.197113    ...     0.677271   \n",
       "GSM591602.CEL  0.332274  0.683959  0.531228  0.184785    ...     0.649350   \n",
       "GSM591603.CEL  0.457032  0.492887  0.324505  0.199644    ...     0.563030   \n",
       "GSM591622.CEL  0.454745  0.488113  0.318980  0.199653    ...     0.545345   \n",
       "\n",
       "                   5541      5542      5543      5544      5545      5546  \\\n",
       "GSM591496.CEL  0.720014  0.702985  0.622383  0.189934  0.688720  0.569247   \n",
       "GSM591601.CEL  0.689790  0.667960  0.611020  0.214126  0.659484  0.548962   \n",
       "GSM591602.CEL  0.660366  0.643134  0.628420  0.203199  0.646247  0.535502   \n",
       "GSM591603.CEL  0.578363  0.457818  0.496192  0.237614  0.566808  0.327465   \n",
       "GSM591622.CEL  0.572971  0.444497  0.476583  0.241327  0.554977  0.313927   \n",
       "\n",
       "                   5547      5548  color_by  \n",
       "GSM591496.CEL  0.673973  0.691933        WT  \n",
       "GSM591601.CEL  0.658613  0.679130        WT  \n",
       "GSM591602.CEL  0.659145  0.686068        WT  \n",
       "GSM591603.CEL  0.524913  0.567808        WT  \n",
       "GSM591622.CEL  0.518031  0.565803        WT  \n",
       "\n",
       "[5 rows x 5550 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Decode simulated data into raw gene space\n",
    "\n",
    "simulated_data_decoded = loaded_decode_model.predict_on_batch(simulated_data_encoded_df)\n",
    "simulated_data_decoded_df = pd.DataFrame(simulated_data_decoded, index=simulated_data_encoded_df.index)\n",
    "\n",
    "simulated_data_decoded_df['color_by'] = list(selected_metadata[color_by_field])\n",
    "simulated_data_decoded_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save simulated data\n",
    "simulated_data_decoded_df.to_csv(simulated_data_file, float_format='%.3f', sep='\\t', compression='xz')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:batch_effects]",
   "language": "python",
   "name": "conda-env-batch_effects-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
