{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Process data\n",
    "This notebook does the following:\n",
    "\n",
    "1. Selects template experiment\n",
    "2. Downloads subset of recount2 data, including the template experiment (50 random experiments + 1 template experiment)\n",
    "3. Train VAE on subset of recount2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: \n",
      "examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.\n",
      "  \"found relative to the 'datapath' directory.\".format(key))\n",
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%load_ext rpy2.ipython\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import rpy2\n",
    "import seaborn as sns\n",
    "from sklearn import preprocessing\n",
    "import pickle\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import generate_labeled_data, utils, pipeline\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in config variables\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../\"))\n",
    "\n",
    "config_file = os.path.abspath(os.path.join(base_dir,\n",
    "                                           \"Rank_pathways\",\n",
    "                                           \"init_config.tsv\"))\n",
    "params = utils.read_config(config_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select template experiment\n",
    "\n",
    "We manually selected bioproject [SRP012656](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37764), which contains primary non-small cell lung adenocarcinoma tumors and adjacent normal tissues of 6 never-smoker Korean female patients with 2 replicates each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load params\n",
    "local_dir = params[\"local_dir\"]\n",
    "dataset_name = params['dataset_name']\n",
    "NN_architecture = params['NN_architecture']\n",
    "project_id = params['project_id']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download subset of recount2 to use as a compendium\n",
    "The compendium will be composed of random experiments + the selected template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Select 59\n",
    "# Run one time\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(\"recount\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library('recount')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#%%R -i project_id -i local_dir\n",
    "\n",
    "#source('../functions/download_recount2_data.R')\n",
    "\n",
    "#get_recount2_compendium(project_id, local_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download expression data for selected project id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Loading objects:\n",
       "  rse_gene\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i project_id -i local_dir\n",
    "\n",
    "source('../functions/download_recount2_data.R')\n",
    "\n",
    "get_recount2_template_experiment(project_id, local_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subset genes\n",
    "For our downstream analysis we will be comparing our set of differentially expression genes against the set found in [Crow et. al. publication](https://www.pnas.org/content/pnas/116/13/6491.full.pdf), therefore we will limit our genes to include only those genes shared between our starting set of genes and those in publication. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_Order</th>\n",
       "      <th>Gene_EntrezID</th>\n",
       "      <th>N_HitLists</th>\n",
       "      <th>DE_Prior_Rank</th>\n",
       "      <th>Gene_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7503</td>\n",
       "      <td>79</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>XIST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>8653</td>\n",
       "      <td>64</td>\n",
       "      <td>0.999948</td>\n",
       "      <td>DDX3Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>9086</td>\n",
       "      <td>62</td>\n",
       "      <td>0.999870</td>\n",
       "      <td>EIF1AY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>8284</td>\n",
       "      <td>52</td>\n",
       "      <td>0.999870</td>\n",
       "      <td>KDM5D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>8287</td>\n",
       "      <td>46</td>\n",
       "      <td>0.999791</td>\n",
       "      <td>USP9Y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene_Order  Gene_EntrezID  N_HitLists  DE_Prior_Rank Gene_Name\n",
       "0           1           7503          79       1.000000      XIST\n",
       "1           2           8653          64       0.999948     DDX3Y\n",
       "2           3           9086          62       0.999870    EIF1AY\n",
       "3           4           8284          52       0.999870     KDM5D\n",
       "4           5           8287          46       0.999791     USP9Y"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get generic genes identified by Crow et. al.\n",
    "DE_prior_file = \"https://raw.githubusercontent.com/maggiecrow/DEprior/master/DE_Prior.txt\"\n",
    "\n",
    "DE_prior = pd.read_csv(DE_prior_file,\n",
    "                       header=0,\n",
    "                       sep=\"\\t\")\n",
    "\n",
    "DE_prior.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get list of published generic genes\n",
    "published_generic_genes = list(DE_prior['Gene_Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get list of our genes\n",
    "# Load real template experiment\n",
    "template_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"recount2_template_data.tsv\")\n",
    "\n",
    "# Read template data\n",
    "template_data = pd.read_csv(\n",
    "    template_data_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "\n",
    "our_gene_ids = list(template_data.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# File mapping ensembl ids to hgnc symbols\n",
    "gene_id_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"ensembl_hgnc_mapping.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "suppressWarnings(library(\"biomaRt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i template_data_file -i gene_id_file\n",
    "\n",
    "# Get mapping between ensembl gene ids (ours) to HGNC gene symbols (published)\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "if (file.exists(gene_id_file) == FALSE){\n",
    "    gene_id_mapping <- get_ensembl_symbol_mapping(template_data_file, gene_id_file)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(57210, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000002330</td>\n",
       "      <td>BAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000003137</td>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000003249</td>\n",
       "      <td>DBNDD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000004799</td>\n",
       "      <td>PDK4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000006062</td>\n",
       "      <td>MAP3K14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id hgnc_symbol\n",
       "1  ENSG00000002330         BAD\n",
       "2  ENSG00000003137     CYP26B1\n",
       "3  ENSG00000003249      DBNDD1\n",
       "4  ENSG00000004799        PDK4\n",
       "5  ENSG00000006062     MAP3K14"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read gene id mapping\n",
    "gene_id_mapping = pd.read_csv(\n",
    "        gene_id_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "\n",
    "print(gene_id_mapping.shape)\n",
    "gene_id_mapping.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58037, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>ensembl_parsed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.14</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.12</td>\n",
       "      <td>ENSG00000000419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.13</td>\n",
       "      <td>ENSG00000000457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.16</td>\n",
       "      <td>ENSG00000000460</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      ensembl_version   ensembl_parsed\n",
       "0  ENSG00000000003.14  ENSG00000000003\n",
       "1   ENSG00000000005.5  ENSG00000000005\n",
       "2  ENSG00000000419.12  ENSG00000000419\n",
       "3  ENSG00000000457.13  ENSG00000000457\n",
       "4  ENSG00000000460.16  ENSG00000000460"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get mapping between ensembl ids with and without version numbers\n",
    "# Expressiond data uses ensembl gene ids with version number \n",
    "ensembl_gene_ids = pd.DataFrame(data={'ensembl_version': our_gene_ids,\n",
    "                                      'ensembl_parsed': [gene_id.split('.')[0] for gene_id in our_gene_ids]})\n",
    "\n",
    "print(ensembl_gene_ids.shape)\n",
    "ensembl_gene_ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58129, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>ensembl_parsed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl_version</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000002330.13</th>\n",
       "      <td>ENSG00000002330</td>\n",
       "      <td>BAD</td>\n",
       "      <td>ENSG00000002330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000003137.8</th>\n",
       "      <td>ENSG00000003137</td>\n",
       "      <td>CYP26B1</td>\n",
       "      <td>ENSG00000003137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000003249.13</th>\n",
       "      <td>ENSG00000003249</td>\n",
       "      <td>DBNDD1</td>\n",
       "      <td>ENSG00000003249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000004799.7</th>\n",
       "      <td>ENSG00000004799</td>\n",
       "      <td>PDK4</td>\n",
       "      <td>ENSG00000004799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000006062.13</th>\n",
       "      <td>ENSG00000006062</td>\n",
       "      <td>MAP3K14</td>\n",
       "      <td>ENSG00000006062</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensembl_gene_id hgnc_symbol   ensembl_parsed\n",
       "ensembl_version                                                 \n",
       "ENSG00000002330.13  ENSG00000002330         BAD  ENSG00000002330\n",
       "ENSG00000003137.8   ENSG00000003137     CYP26B1  ENSG00000003137\n",
       "ENSG00000003249.13  ENSG00000003249      DBNDD1  ENSG00000003249\n",
       "ENSG00000004799.7   ENSG00000004799        PDK4  ENSG00000004799\n",
       "ENSG00000006062.13  ENSG00000006062     MAP3K14  ENSG00000006062"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Map ensembl ids with version number to gene_id_mapping_df\n",
    "gene_id_mapping = pd.merge(gene_id_mapping, \n",
    "                           ensembl_gene_ids, \n",
    "                           left_on='ensembl_gene_id',\n",
    "                           right_on='ensembl_parsed', \n",
    "                           how='outer')\n",
    "\n",
    "print(gene_id_mapping.shape)\n",
    "gene_id_mapping.set_index('ensembl_version', inplace=True)\n",
    "gene_id_mapping.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17788\n"
     ]
    }
   ],
   "source": [
    "# Get intersection of gene lists\n",
    "our_gene_ids_hgnc = gene_id_mapping.loc[our_gene_ids]['hgnc_symbol']\n",
    "shared_genes_hgnc = set(our_gene_ids_hgnc).intersection(published_generic_genes)\n",
    "print(len(shared_genes_hgnc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17851\n"
     ]
    }
   ],
   "source": [
    "# Convert shared hgnc gene ids back to ensembl ids with version\n",
    "shared_genes = list(gene_id_mapping[gene_id_mapping['hgnc_symbol'].isin(shared_genes_hgnc)].index)\n",
    "print(len(shared_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save shared genes\n",
    "shared_genes_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"shared_gene_ids.pickle\")\n",
    "\n",
    "outfile = open(shared_genes_file,'wb')\n",
    "pickle.dump(shared_genes,outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since this experiment contains both RNA-seq and smRNA-seq samples which are in different ranges so we will drop smRNA samples so that samples are within the same range. The analysis identifying these two subsets of samples can be found in this [notebook](0_explore_input_data.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Drop smRNA samples so that samples are within the same range\n",
    "smRNA_samples = [\"SRR493961\",\n",
    "                 \"SRR493962\",\n",
    "                 \"SRR493963\",\n",
    "                 \"SRR493964\",\n",
    "                 \"SRR493965\",\n",
    "                 \"SRR493966\",\n",
    "                 \"SRR493967\",\n",
    "                 \"SRR493968\",\n",
    "                 \"SRR493969\",\n",
    "                 \"SRR493970\",\n",
    "                 \"SRR493971\",\n",
    "                 \"SRR493972\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Drop samples\n",
    "template_data = template_data.drop(smRNA_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(24, 17851)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG00000002330.13</th>\n",
       "      <th>ENSG00000003137.8</th>\n",
       "      <th>ENSG00000003249.13</th>\n",
       "      <th>ENSG00000004799.7</th>\n",
       "      <th>ENSG00000006062.13</th>\n",
       "      <th>ENSG00000006282.20</th>\n",
       "      <th>ENSG00000006283.17</th>\n",
       "      <th>ENSG00000006327.13</th>\n",
       "      <th>ENSG00000007384.15</th>\n",
       "      <th>ENSG00000007968.6</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000203965.12</th>\n",
       "      <th>ENSG00000205352.10</th>\n",
       "      <th>ENSG00000206538.8</th>\n",
       "      <th>ENSG00000211450.9</th>\n",
       "      <th>ENSG00000213246.6</th>\n",
       "      <th>ENSG00000224470.7</th>\n",
       "      <th>ENSG00000244879.5</th>\n",
       "      <th>ENSG00000264364.2</th>\n",
       "      <th>ENSG00000226978.1</th>\n",
       "      <th>ENSG00000230292.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR493937</th>\n",
       "      <td>12.990968</td>\n",
       "      <td>0.848728</td>\n",
       "      <td>0.809381</td>\n",
       "      <td>34.353087</td>\n",
       "      <td>12.810135</td>\n",
       "      <td>12.114271</td>\n",
       "      <td>0.280827</td>\n",
       "      <td>36.630775</td>\n",
       "      <td>7.228667</td>\n",
       "      <td>0.276567</td>\n",
       "      <td>...</td>\n",
       "      <td>0.634714</td>\n",
       "      <td>19.870212</td>\n",
       "      <td>1.752787</td>\n",
       "      <td>13.408707</td>\n",
       "      <td>16.808396</td>\n",
       "      <td>8.976874</td>\n",
       "      <td>0.976822</td>\n",
       "      <td>11.274722</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR493938</th>\n",
       "      <td>12.513636</td>\n",
       "      <td>1.016285</td>\n",
       "      <td>0.907216</td>\n",
       "      <td>34.784134</td>\n",
       "      <td>12.080994</td>\n",
       "      <td>12.518965</td>\n",
       "      <td>0.283395</td>\n",
       "      <td>34.576562</td>\n",
       "      <td>7.338301</td>\n",
       "      <td>0.361873</td>\n",
       "      <td>...</td>\n",
       "      <td>0.538282</td>\n",
       "      <td>19.251368</td>\n",
       "      <td>1.746381</td>\n",
       "      <td>13.165898</td>\n",
       "      <td>16.929152</td>\n",
       "      <td>8.891573</td>\n",
       "      <td>0.912389</td>\n",
       "      <td>11.632596</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR493939</th>\n",
       "      <td>25.954618</td>\n",
       "      <td>0.494410</td>\n",
       "      <td>1.870051</td>\n",
       "      <td>9.915847</td>\n",
       "      <td>6.725095</td>\n",
       "      <td>21.012343</td>\n",
       "      <td>0.223001</td>\n",
       "      <td>32.285838</td>\n",
       "      <td>17.261978</td>\n",
       "      <td>0.705615</td>\n",
       "      <td>...</td>\n",
       "      <td>1.356601</td>\n",
       "      <td>34.834635</td>\n",
       "      <td>0.962179</td>\n",
       "      <td>22.371422</td>\n",
       "      <td>23.873134</td>\n",
       "      <td>5.430164</td>\n",
       "      <td>1.493810</td>\n",
       "      <td>14.609568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR493940</th>\n",
       "      <td>25.763659</td>\n",
       "      <td>0.624823</td>\n",
       "      <td>1.736342</td>\n",
       "      <td>10.573514</td>\n",
       "      <td>6.583890</td>\n",
       "      <td>21.389096</td>\n",
       "      <td>0.174230</td>\n",
       "      <td>32.901649</td>\n",
       "      <td>17.409587</td>\n",
       "      <td>0.633884</td>\n",
       "      <td>...</td>\n",
       "      <td>1.452227</td>\n",
       "      <td>35.175937</td>\n",
       "      <td>0.971337</td>\n",
       "      <td>22.598672</td>\n",
       "      <td>24.089967</td>\n",
       "      <td>5.504614</td>\n",
       "      <td>1.512516</td>\n",
       "      <td>14.139525</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR493941</th>\n",
       "      <td>18.379733</td>\n",
       "      <td>3.738637</td>\n",
       "      <td>1.443360</td>\n",
       "      <td>32.690667</td>\n",
       "      <td>5.856697</td>\n",
       "      <td>13.002362</td>\n",
       "      <td>0.189597</td>\n",
       "      <td>8.971186</td>\n",
       "      <td>13.199591</td>\n",
       "      <td>0.145056</td>\n",
       "      <td>...</td>\n",
       "      <td>0.932711</td>\n",
       "      <td>18.635415</td>\n",
       "      <td>4.780810</td>\n",
       "      <td>9.416897</td>\n",
       "      <td>19.599759</td>\n",
       "      <td>9.417604</td>\n",
       "      <td>0.960957</td>\n",
       "      <td>12.767854</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 17851 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ENSG00000002330.13  ENSG00000003137.8  ENSG00000003249.13  \\\n",
       "SRR493937           12.990968           0.848728            0.809381   \n",
       "SRR493938           12.513636           1.016285            0.907216   \n",
       "SRR493939           25.954618           0.494410            1.870051   \n",
       "SRR493940           25.763659           0.624823            1.736342   \n",
       "SRR493941           18.379733           3.738637            1.443360   \n",
       "\n",
       "           ENSG00000004799.7  ENSG00000006062.13  ENSG00000006282.20  \\\n",
       "SRR493937          34.353087           12.810135           12.114271   \n",
       "SRR493938          34.784134           12.080994           12.518965   \n",
       "SRR493939           9.915847            6.725095           21.012343   \n",
       "SRR493940          10.573514            6.583890           21.389096   \n",
       "SRR493941          32.690667            5.856697           13.002362   \n",
       "\n",
       "           ENSG00000006283.17  ENSG00000006327.13  ENSG00000007384.15  \\\n",
       "SRR493937            0.280827           36.630775            7.228667   \n",
       "SRR493938            0.283395           34.576562            7.338301   \n",
       "SRR493939            0.223001           32.285838           17.261978   \n",
       "SRR493940            0.174230           32.901649           17.409587   \n",
       "SRR493941            0.189597            8.971186           13.199591   \n",
       "\n",
       "           ENSG00000007968.6        ...          ENSG00000203965.12  \\\n",
       "SRR493937           0.276567        ...                    0.634714   \n",
       "SRR493938           0.361873        ...                    0.538282   \n",
       "SRR493939           0.705615        ...                    1.356601   \n",
       "SRR493940           0.633884        ...                    1.452227   \n",
       "SRR493941           0.145056        ...                    0.932711   \n",
       "\n",
       "           ENSG00000205352.10  ENSG00000206538.8  ENSG00000211450.9  \\\n",
       "SRR493937           19.870212           1.752787          13.408707   \n",
       "SRR493938           19.251368           1.746381          13.165898   \n",
       "SRR493939           34.834635           0.962179          22.371422   \n",
       "SRR493940           35.175937           0.971337          22.598672   \n",
       "SRR493941           18.635415           4.780810           9.416897   \n",
       "\n",
       "           ENSG00000213246.6  ENSG00000224470.7  ENSG00000244879.5  \\\n",
       "SRR493937          16.808396           8.976874           0.976822   \n",
       "SRR493938          16.929152           8.891573           0.912389   \n",
       "SRR493939          23.873134           5.430164           1.493810   \n",
       "SRR493940          24.089967           5.504614           1.512516   \n",
       "SRR493941          19.599759           9.417604           0.960957   \n",
       "\n",
       "           ENSG00000264364.2  ENSG00000226978.1  ENSG00000230292.5  \n",
       "SRR493937          11.274722                0.0                0.0  \n",
       "SRR493938          11.632596                0.0                0.0  \n",
       "SRR493939          14.609568                0.0                0.0  \n",
       "SRR493940          14.139525                0.0                0.0  \n",
       "SRR493941          12.767854                0.0                0.0  \n",
       "\n",
       "[5 rows x 17851 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop genes\n",
    "template_data = template_data[shared_genes]\n",
    "\n",
    "print(template_data.shape)\n",
    "template_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save \n",
    "template_data.to_csv(template_data_file, float_format='%.5f', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Normalize compendium "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load real gene expression data\n",
    "original_compendium_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"recount2_compedium_data.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1552, 17851)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG00000002330.13</th>\n",
       "      <th>ENSG00000003137.8</th>\n",
       "      <th>ENSG00000003249.13</th>\n",
       "      <th>ENSG00000004799.7</th>\n",
       "      <th>ENSG00000006062.13</th>\n",
       "      <th>ENSG00000006282.20</th>\n",
       "      <th>ENSG00000006283.17</th>\n",
       "      <th>ENSG00000006327.13</th>\n",
       "      <th>ENSG00000007384.15</th>\n",
       "      <th>ENSG00000007968.6</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000203965.12</th>\n",
       "      <th>ENSG00000205352.10</th>\n",
       "      <th>ENSG00000206538.8</th>\n",
       "      <th>ENSG00000211450.9</th>\n",
       "      <th>ENSG00000213246.6</th>\n",
       "      <th>ENSG00000224470.7</th>\n",
       "      <th>ENSG00000244879.5</th>\n",
       "      <th>ENSG00000264364.2</th>\n",
       "      <th>ENSG00000226978.1</th>\n",
       "      <th>ENSG00000230292.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR1604987</th>\n",
       "      <td>13.982</td>\n",
       "      <td>0.053</td>\n",
       "      <td>9.477</td>\n",
       "      <td>0.895</td>\n",
       "      <td>2.042</td>\n",
       "      <td>15.303</td>\n",
       "      <td>0.012</td>\n",
       "      <td>118.426</td>\n",
       "      <td>5.025</td>\n",
       "      <td>0.490</td>\n",
       "      <td>...</td>\n",
       "      <td>2.291</td>\n",
       "      <td>52.515</td>\n",
       "      <td>0.758</td>\n",
       "      <td>29.027</td>\n",
       "      <td>34.794</td>\n",
       "      <td>7.110</td>\n",
       "      <td>1.314</td>\n",
       "      <td>28.624</td>\n",
       "      <td>0.007</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604988</th>\n",
       "      <td>13.665</td>\n",
       "      <td>0.074</td>\n",
       "      <td>9.050</td>\n",
       "      <td>1.025</td>\n",
       "      <td>1.951</td>\n",
       "      <td>16.045</td>\n",
       "      <td>0.016</td>\n",
       "      <td>116.722</td>\n",
       "      <td>5.386</td>\n",
       "      <td>0.468</td>\n",
       "      <td>...</td>\n",
       "      <td>2.362</td>\n",
       "      <td>53.092</td>\n",
       "      <td>0.655</td>\n",
       "      <td>29.024</td>\n",
       "      <td>35.853</td>\n",
       "      <td>7.036</td>\n",
       "      <td>1.251</td>\n",
       "      <td>28.456</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604989</th>\n",
       "      <td>7.415</td>\n",
       "      <td>0.108</td>\n",
       "      <td>8.961</td>\n",
       "      <td>3.269</td>\n",
       "      <td>3.297</td>\n",
       "      <td>19.080</td>\n",
       "      <td>0.020</td>\n",
       "      <td>114.528</td>\n",
       "      <td>4.698</td>\n",
       "      <td>0.313</td>\n",
       "      <td>...</td>\n",
       "      <td>1.752</td>\n",
       "      <td>36.212</td>\n",
       "      <td>0.236</td>\n",
       "      <td>27.901</td>\n",
       "      <td>21.716</td>\n",
       "      <td>9.401</td>\n",
       "      <td>3.564</td>\n",
       "      <td>23.767</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604990</th>\n",
       "      <td>7.149</td>\n",
       "      <td>0.094</td>\n",
       "      <td>8.979</td>\n",
       "      <td>3.733</td>\n",
       "      <td>3.227</td>\n",
       "      <td>18.335</td>\n",
       "      <td>0.022</td>\n",
       "      <td>115.940</td>\n",
       "      <td>5.013</td>\n",
       "      <td>0.262</td>\n",
       "      <td>...</td>\n",
       "      <td>1.676</td>\n",
       "      <td>35.664</td>\n",
       "      <td>0.296</td>\n",
       "      <td>27.573</td>\n",
       "      <td>23.040</td>\n",
       "      <td>9.456</td>\n",
       "      <td>3.490</td>\n",
       "      <td>24.948</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604991</th>\n",
       "      <td>11.509</td>\n",
       "      <td>0.656</td>\n",
       "      <td>8.244</td>\n",
       "      <td>2.268</td>\n",
       "      <td>4.801</td>\n",
       "      <td>19.253</td>\n",
       "      <td>0.027</td>\n",
       "      <td>61.891</td>\n",
       "      <td>3.208</td>\n",
       "      <td>0.326</td>\n",
       "      <td>...</td>\n",
       "      <td>1.366</td>\n",
       "      <td>32.978</td>\n",
       "      <td>3.788</td>\n",
       "      <td>19.781</td>\n",
       "      <td>27.818</td>\n",
       "      <td>7.313</td>\n",
       "      <td>1.215</td>\n",
       "      <td>16.875</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 17851 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ENSG00000002330.13  ENSG00000003137.8  ENSG00000003249.13  \\\n",
       "SRR1604987              13.982              0.053               9.477   \n",
       "SRR1604988              13.665              0.074               9.050   \n",
       "SRR1604989               7.415              0.108               8.961   \n",
       "SRR1604990               7.149              0.094               8.979   \n",
       "SRR1604991              11.509              0.656               8.244   \n",
       "\n",
       "            ENSG00000004799.7  ENSG00000006062.13  ENSG00000006282.20  \\\n",
       "SRR1604987              0.895               2.042              15.303   \n",
       "SRR1604988              1.025               1.951              16.045   \n",
       "SRR1604989              3.269               3.297              19.080   \n",
       "SRR1604990              3.733               3.227              18.335   \n",
       "SRR1604991              2.268               4.801              19.253   \n",
       "\n",
       "            ENSG00000006283.17  ENSG00000006327.13  ENSG00000007384.15  \\\n",
       "SRR1604987               0.012             118.426               5.025   \n",
       "SRR1604988               0.016             116.722               5.386   \n",
       "SRR1604989               0.020             114.528               4.698   \n",
       "SRR1604990               0.022             115.940               5.013   \n",
       "SRR1604991               0.027              61.891               3.208   \n",
       "\n",
       "            ENSG00000007968.6        ...          ENSG00000203965.12  \\\n",
       "SRR1604987              0.490        ...                       2.291   \n",
       "SRR1604988              0.468        ...                       2.362   \n",
       "SRR1604989              0.313        ...                       1.752   \n",
       "SRR1604990              0.262        ...                       1.676   \n",
       "SRR1604991              0.326        ...                       1.366   \n",
       "\n",
       "            ENSG00000205352.10  ENSG00000206538.8  ENSG00000211450.9  \\\n",
       "SRR1604987              52.515              0.758             29.027   \n",
       "SRR1604988              53.092              0.655             29.024   \n",
       "SRR1604989              36.212              0.236             27.901   \n",
       "SRR1604990              35.664              0.296             27.573   \n",
       "SRR1604991              32.978              3.788             19.781   \n",
       "\n",
       "            ENSG00000213246.6  ENSG00000224470.7  ENSG00000244879.5  \\\n",
       "SRR1604987             34.794              7.110              1.314   \n",
       "SRR1604988             35.853              7.036              1.251   \n",
       "SRR1604989             21.716              9.401              3.564   \n",
       "SRR1604990             23.040              9.456              3.490   \n",
       "SRR1604991             27.818              7.313              1.215   \n",
       "\n",
       "            ENSG00000264364.2  ENSG00000226978.1  ENSG00000230292.5  \n",
       "SRR1604987             28.624              0.007                0.0  \n",
       "SRR1604988             28.456              0.000                0.0  \n",
       "SRR1604989             23.767              0.000                0.0  \n",
       "SRR1604990             24.948              0.000                0.0  \n",
       "SRR1604991             16.875              0.000                0.0  \n",
       "\n",
       "[5 rows x 17851 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read data\n",
    "original_compendium = pd.read_table(\n",
    "    original_compendium_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "\n",
    "# Drop genes\n",
    "original_compendium = original_compendium[shared_genes]\n",
    "\n",
    "print(original_compendium.shape)\n",
    "original_compendium.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1552, 17851)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG00000002330.13</th>\n",
       "      <th>ENSG00000003137.8</th>\n",
       "      <th>ENSG00000003249.13</th>\n",
       "      <th>ENSG00000004799.7</th>\n",
       "      <th>ENSG00000006062.13</th>\n",
       "      <th>ENSG00000006282.20</th>\n",
       "      <th>ENSG00000006283.17</th>\n",
       "      <th>ENSG00000006327.13</th>\n",
       "      <th>ENSG00000007384.15</th>\n",
       "      <th>ENSG00000007968.6</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000203965.12</th>\n",
       "      <th>ENSG00000205352.10</th>\n",
       "      <th>ENSG00000206538.8</th>\n",
       "      <th>ENSG00000211450.9</th>\n",
       "      <th>ENSG00000213246.6</th>\n",
       "      <th>ENSG00000224470.7</th>\n",
       "      <th>ENSG00000244879.5</th>\n",
       "      <th>ENSG00000264364.2</th>\n",
       "      <th>ENSG00000226978.1</th>\n",
       "      <th>ENSG00000230292.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR1604987</th>\n",
       "      <td>0.030346</td>\n",
       "      <td>0.001769</td>\n",
       "      <td>0.183531</td>\n",
       "      <td>0.010066</td>\n",
       "      <td>0.071795</td>\n",
       "      <td>0.168457</td>\n",
       "      <td>0.000242</td>\n",
       "      <td>0.346751</td>\n",
       "      <td>0.135361</td>\n",
       "      <td>0.010723</td>\n",
       "      <td>...</td>\n",
       "      <td>0.203789</td>\n",
       "      <td>0.174118</td>\n",
       "      <td>0.010270</td>\n",
       "      <td>0.082616</td>\n",
       "      <td>0.186744</td>\n",
       "      <td>0.210119</td>\n",
       "      <td>0.026650</td>\n",
       "      <td>0.260230</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604988</th>\n",
       "      <td>0.029658</td>\n",
       "      <td>0.002470</td>\n",
       "      <td>0.175262</td>\n",
       "      <td>0.011528</td>\n",
       "      <td>0.068596</td>\n",
       "      <td>0.176625</td>\n",
       "      <td>0.000323</td>\n",
       "      <td>0.341762</td>\n",
       "      <td>0.145085</td>\n",
       "      <td>0.010242</td>\n",
       "      <td>...</td>\n",
       "      <td>0.210105</td>\n",
       "      <td>0.176031</td>\n",
       "      <td>0.008875</td>\n",
       "      <td>0.082608</td>\n",
       "      <td>0.192428</td>\n",
       "      <td>0.207932</td>\n",
       "      <td>0.025372</td>\n",
       "      <td>0.258703</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604989</th>\n",
       "      <td>0.016093</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.173538</td>\n",
       "      <td>0.036767</td>\n",
       "      <td>0.115920</td>\n",
       "      <td>0.210035</td>\n",
       "      <td>0.000404</td>\n",
       "      <td>0.335338</td>\n",
       "      <td>0.126552</td>\n",
       "      <td>0.006850</td>\n",
       "      <td>...</td>\n",
       "      <td>0.155844</td>\n",
       "      <td>0.120064</td>\n",
       "      <td>0.003198</td>\n",
       "      <td>0.079411</td>\n",
       "      <td>0.116553</td>\n",
       "      <td>0.277824</td>\n",
       "      <td>0.072283</td>\n",
       "      <td>0.216073</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604990</th>\n",
       "      <td>0.015516</td>\n",
       "      <td>0.003138</td>\n",
       "      <td>0.173887</td>\n",
       "      <td>0.041985</td>\n",
       "      <td>0.113459</td>\n",
       "      <td>0.201834</td>\n",
       "      <td>0.000444</td>\n",
       "      <td>0.339472</td>\n",
       "      <td>0.135038</td>\n",
       "      <td>0.005734</td>\n",
       "      <td>...</td>\n",
       "      <td>0.149084</td>\n",
       "      <td>0.118247</td>\n",
       "      <td>0.004011</td>\n",
       "      <td>0.078478</td>\n",
       "      <td>0.123659</td>\n",
       "      <td>0.279449</td>\n",
       "      <td>0.070782</td>\n",
       "      <td>0.226810</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1604991</th>\n",
       "      <td>0.024979</td>\n",
       "      <td>0.021898</td>\n",
       "      <td>0.159653</td>\n",
       "      <td>0.025508</td>\n",
       "      <td>0.168800</td>\n",
       "      <td>0.211939</td>\n",
       "      <td>0.000545</td>\n",
       "      <td>0.181217</td>\n",
       "      <td>0.086415</td>\n",
       "      <td>0.007134</td>\n",
       "      <td>...</td>\n",
       "      <td>0.121509</td>\n",
       "      <td>0.109341</td>\n",
       "      <td>0.051324</td>\n",
       "      <td>0.056300</td>\n",
       "      <td>0.149303</td>\n",
       "      <td>0.216118</td>\n",
       "      <td>0.024642</td>\n",
       "      <td>0.153416</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 17851 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ENSG00000002330.13  ENSG00000003137.8  ENSG00000003249.13  \\\n",
       "SRR1604987            0.030346           0.001769            0.183531   \n",
       "SRR1604988            0.029658           0.002470            0.175262   \n",
       "SRR1604989            0.016093           0.003605            0.173538   \n",
       "SRR1604990            0.015516           0.003138            0.173887   \n",
       "SRR1604991            0.024979           0.021898            0.159653   \n",
       "\n",
       "            ENSG00000004799.7  ENSG00000006062.13  ENSG00000006282.20  \\\n",
       "SRR1604987           0.010066            0.071795            0.168457   \n",
       "SRR1604988           0.011528            0.068596            0.176625   \n",
       "SRR1604989           0.036767            0.115920            0.210035   \n",
       "SRR1604990           0.041985            0.113459            0.201834   \n",
       "SRR1604991           0.025508            0.168800            0.211939   \n",
       "\n",
       "            ENSG00000006283.17  ENSG00000006327.13  ENSG00000007384.15  \\\n",
       "SRR1604987            0.000242            0.346751            0.135361   \n",
       "SRR1604988            0.000323            0.341762            0.145085   \n",
       "SRR1604989            0.000404            0.335338            0.126552   \n",
       "SRR1604990            0.000444            0.339472            0.135038   \n",
       "SRR1604991            0.000545            0.181217            0.086415   \n",
       "\n",
       "            ENSG00000007968.6        ...          ENSG00000203965.12  \\\n",
       "SRR1604987           0.010723        ...                    0.203789   \n",
       "SRR1604988           0.010242        ...                    0.210105   \n",
       "SRR1604989           0.006850        ...                    0.155844   \n",
       "SRR1604990           0.005734        ...                    0.149084   \n",
       "SRR1604991           0.007134        ...                    0.121509   \n",
       "\n",
       "            ENSG00000205352.10  ENSG00000206538.8  ENSG00000211450.9  \\\n",
       "SRR1604987            0.174118           0.010270           0.082616   \n",
       "SRR1604988            0.176031           0.008875           0.082608   \n",
       "SRR1604989            0.120064           0.003198           0.079411   \n",
       "SRR1604990            0.118247           0.004011           0.078478   \n",
       "SRR1604991            0.109341           0.051324           0.056300   \n",
       "\n",
       "            ENSG00000213246.6  ENSG00000224470.7  ENSG00000244879.5  \\\n",
       "SRR1604987           0.186744           0.210119           0.026650   \n",
       "SRR1604988           0.192428           0.207932           0.025372   \n",
       "SRR1604989           0.116553           0.277824           0.072283   \n",
       "SRR1604990           0.123659           0.279449           0.070782   \n",
       "SRR1604991           0.149303           0.216118           0.024642   \n",
       "\n",
       "            ENSG00000264364.2  ENSG00000226978.1  ENSG00000230292.5  \n",
       "SRR1604987           0.260230           0.000075                0.0  \n",
       "SRR1604988           0.258703           0.000000                0.0  \n",
       "SRR1604989           0.216073           0.000000                0.0  \n",
       "SRR1604990           0.226810           0.000000                0.0  \n",
       "SRR1604991           0.153416           0.000000                0.0  \n",
       "\n",
       "[5 rows x 17851 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 0-1 normalize per gene\n",
    "scaler = preprocessing.MinMaxScaler()\n",
    "original_data_scaled = scaler.fit_transform(original_compendium)\n",
    "original_data_scaled_df = pd.DataFrame(original_data_scaled,\n",
    "                                columns=original_compendium.columns,\n",
    "                                index=original_compendium.index)\n",
    "\n",
    "print(original_data_scaled_df.shape)\n",
    "original_data_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save data\n",
    "normalized_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"normalized_recount2_compendium_data.tsv\")\n",
    "\n",
    "original_data_scaled_df.to_csv(\n",
    "    normalized_data_file, float_format='%.3f', sep='\\t')\n",
    "\n",
    "original_compendium.to_csv(\n",
    "    original_compendium_file, float_format='%.3f', sep='\\t')\n",
    "\n",
    "# Save scaler transform\n",
    "scaler_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"scaler_transform.pickle\")\n",
    "\n",
    "outfile = open(scaler_file,'wb')\n",
    "pickle.dump(scaler,outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Train VAE "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'# Setup directories\\n# Create VAE directories\\noutput_dirs = [os.path.join(base_dir, dataset_name, \"models\"),\\n               os.path.join(base_dir, dataset_name, \"logs\")]\\n\\n# Check if analysis output directory exist otherwise create\\nfor each_dir in output_dirs:\\n    if os.path.exists(each_dir) == False:\\n        print(\\'creating new directory: {}\\'.format(each_dir))\\n        os.makedirs(each_dir, exist_ok=True)\\n\\n# Check if NN architecture directory exist otherwise create\\nfor each_dir in output_dirs:\\n    new_dir = os.path.join(each_dir, NN_architecture)\\n    if os.path.exists(new_dir) == False:\\n        print(\\'creating new directory: {}\\'.format(new_dir))\\n        os.makedirs(new_dir, exist_ok=True)'"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"# Setup directories\n",
    "# Create VAE directories\n",
    "output_dirs = [os.path.join(base_dir, dataset_name, \"models\"),\n",
    "               os.path.join(base_dir, dataset_name, \"logs\")]\n",
    "\n",
    "# Check if analysis output directory exist otherwise create\n",
    "for each_dir in output_dirs:\n",
    "    if os.path.exists(each_dir) == False:\n",
    "        print('creating new directory: {}'.format(each_dir))\n",
    "        os.makedirs(each_dir, exist_ok=True)\n",
    "\n",
    "# Check if NN architecture directory exist otherwise create\n",
    "for each_dir in output_dirs:\n",
    "    new_dir = os.path.join(each_dir, NN_architecture)\n",
    "    if os.path.exists(new_dir) == False:\n",
    "        print('creating new directory: {}'.format(new_dir))\n",
    "        os.makedirs(new_dir, exist_ok=True)\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'# Train VAE on new compendium data\\n# Write out model to rank_pathways directory\\npipeline.train_vae(config_file,\\n                   normalized_data_file)'"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"# Train VAE on new compendium data\n",
    "# Write out model to rank_pathways directory\n",
    "pipeline.train_vae(config_file,\n",
    "                   normalized_data_file)\"\"\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:ranked_pathways] *",
   "language": "python",
   "name": "conda-env-ranked_pathways-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
