{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Statistical analysis\n",
    "This notebook performs differential expression analysis using the real template experiment and simulated experiments, as a null set. Then the set of differentially expressed genes (DEGs) obtained from this analysis are used to perform gene set enrichment analysis (GSEA) to identify pathways enriched in these set of DEGs.\n",
    "\n",
    "*Note:* To run datatables, need to refresh the window and then run all cells (DO NOT restart and run all, only run all works)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: \n",
      "examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.\n",
      "  \"found relative to the 'datapath' directory.\".format(key))\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%load_ext rpy2.ipython\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import seaborn as sns\n",
    "import rpy2.robjects\n",
    "from rpy2.robjects.packages import importr\n",
    "from rpy2.robjects import pandas2ri\n",
    "pandas2ri.activate()\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "from jupyter_datatables import init_datatables_mode\n",
    "\n",
    "\n",
    "def fxn():\n",
    "    warnings.warn(\"deprecated\", DeprecationWarning)\n",
    "\n",
    "with warnings.catch_warnings():\n",
    "    warnings.simplefilter(\"ignore\")\n",
    "    fxn()\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import utils\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in config variables\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../\"))\n",
    "\n",
    "config_file = os.path.abspath(os.path.join(base_dir,\n",
    "                                           \"Rank_pathways\",\n",
    "                                           \"init_config.tsv\"))\n",
    "params = utils.read_config(config_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# User selected experiment id\n",
    "project_id = \"SRP000762\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load params\n",
    "local_dir = params[\"local_dir\"]\n",
    "dataset_name = params['dataset_name']\n",
    "num_runs=25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load real template experiment\n",
    "template_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"recount2_template_data.tsv\")\n",
    "\n",
    "# Load metadata file with grouping assignments for samples\n",
    "metadata_file = os.path.join(\n",
    "    base_dir,\n",
    "    \"Rank_pathways\",\n",
    "    \"data\",\n",
    "    \"metadata\",\n",
    "    \"SRP000762_groups.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Select 59\n",
    "# Run one time\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(\"limma\")\n",
    "#BiocManager::install('EnhancedVolcano')\n",
    "#devtools::install_github('kevinblighe/EnhancedVolcano')\n",
    "#BiocManager::install('clusterProfiler')\n",
    "#BiocManager::install(\"org.Hs.eg.db\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library('limma')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get differentially expressed genes from template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i metadata_file -i project_id -i template_data_file -i local_dir\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "out_file <- paste(local_dir,\n",
    "                  \"DE_stats/DE_stats_template_data_\",\n",
    "                  project_id,\n",
    "                  \"_real.txt\",\n",
    "                  sep=\"\")\n",
    "\n",
    "if (file.exists(out_file) == FALSE){\n",
    "    get_DE_stats(metadata_file,\n",
    "                 project_id, \n",
    "                 template_data_file,\n",
    "                 \"template\",\n",
    "                 local_dir,\n",
    "                 \"real\")\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get differentially expressed genes from each simulated experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i metadata_file -i project_id -i base_dir -i local_dir -i num_runs -o num_sign_DEGs_simulated\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "num_sign_DEGs_simulated <- c()\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_data_file <- paste(local_dir, \n",
    "                                 \"pseudo_experiment/selected_simulated_data_\",\n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    out_file <- paste(local_dir, \n",
    "                      \"DE_stats/DE_stats_simulated_data_\",\n",
    "                      project_id,\n",
    "                      \"_\",\n",
    "                      i,\n",
    "                      \".txt\", \n",
    "                      sep=\"\")\n",
    "    \n",
    "    if (file.exists(out_file) == FALSE){\n",
    "        run_output <- get_DE_stats(metadata_file,\n",
    "                                   project_id, \n",
    "                                   simulated_data_file,\n",
    "                                   \"simulated\",\n",
    "                                   local_dir,\n",
    "                                   i)\n",
    "        num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, run_output)\n",
    "    } else {\n",
    "        # Read in DE stats data\n",
    "        DE_stats_data <- as.data.frame(read.table(out_file, sep=\"\\t\", header=TRUE, row.names=1))\n",
    "        \n",
    "        # Get number of genes that exceed threshold\n",
    "        threshold <- 0.05\n",
    "        sign_DEGs <- DE_stats_data[DE_stats_data[,'adj.P.Val']<threshold,]\n",
    "        \n",
    "        num_sign_DEGs <- nrow(sign_DEGs)\n",
    "        \n",
    "        num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, num_sign_DEGs)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-05-20 17:31:24,790 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/scipy/stats/stats.py:1713: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.\n",
      "  return np.add.reduce(sorted[indexer] * weights, axis=axis) / sumval\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7fd0d850d310>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvqOYd8AAADCJJREFUeJzt3F2MXHUdxvHn0YoXglrspjYIrlFigkYLbqrG94AGvQCMRkXEGknWREkw4gXRC43c4AvghcZYhVANvoKEJtYXrBhCAo1bbbCl0aIBLZZ2EaMYo6bweLGnZiWzzJmZMzPdX7+fZDMzZ87u+f3b9LvDYeY4iQAAq99Tpj0AAKAbBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBFrJnmwdevWZXZ2dpKHBIBVb9euXQ8nmem330SDPjs7q4WFhUkeEgBWPdsPtNmPUy4AUARBB4AiCDoAFEHQAaAIgg4ARfQNuu1Tbd9u+17be21f1mz/tO0Hbe9uvt42/nEBACtp87bFI5IuT/Ir2ydJ2mX7tua5a5N8YXzjAQDa6hv0JAclHWzuP2p7n6RTxj0YAGAwA51Dtz0r6UxJO5tNl9q+x/b1ttd2PBsAYACtPylq+0RJN0v6aJK/2/6KpCslpbm9WtIHe3zfvKR5STrttNO6mBmFfWvnH6c9wsS995X8u0A3Wr1Ct/00LcX8xiQ/kKQkh5I8luRxSV+TtKnX9ybZkmQuydzMTN9LEQAAhtTmXS6WdJ2kfUmuWbZ9w7Ld3i5pT/fjAQDaanPK5TWSLpb0G9u7m22fkHSh7Y1aOuVyv6QPjWVCAEArbd7lcqck93hqe/fjAACGxSdFAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAU0Tfotk+1fbvte23vtX1Zs/1k27fZ3t/crh3/uACAlbR5hX5E0uVJzpD0KkkfsX2GpCsk7UhyuqQdzWMAwJT0DXqSg0l+1dx/VNI+SadIOl/S1ma3rZIuGNeQAID+BjqHbntW0pmSdkpan+Rg89RDktZ3OhkAYCCtg277REk3S/pokr8vfy5JJGWF75u3vWB7YXFxcaRhAQAraxV020/TUsxvTPKDZvMh2xua5zdIOtzre5NsSTKXZG5mZqaLmQEAPbR5l4slXSdpX5Jrlj21TdLm5v5mSbd2Px4AoK01LfZ5jaSLJf3G9u5m2yckXSXpe7YvkfSApHeNZ0QAQBt9g57kTkle4emzux0HADAsPikKAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARfQNuu3rbR+2vWfZtk/bftD27ubrbeMdEwDQT5tX6DdIOrfH9muTbGy+tnc7FgBgUH2DnuQOSY9MYBYAwAhGOYd+qe17mlMyazubCAAwlGGD/hVJL5S0UdJBSVevtKPtedsLthcWFxeHPBwAoJ+hgp7kUJLHkjwu6WuSNj3JvluSzCWZm5mZGXZOAEAfQwXd9oZlD98uac9K+wIAJmNNvx1sf1vSGyWts31A0qckvdH2RkmRdL+kD41xRgBAC32DnuTCHpuvG8MsAIAR8ElRACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBF9A267ettH7a9Z9m2k23fZnt/c7t2vGMCAPpp8wr9BknnPmHbFZJ2JDld0o7mMQBgivoGPckdkh55wubzJW1t7m+VdEHHcwEABjTsOfT1SQ429x+StL6jeQAAQxr5f4omiaSs9LztedsLthcWFxdHPRwAYAXDBv2Q7Q2S1NweXmnHJFuSzCWZm5mZGfJwAIB+hg36Nkmbm/ubJd3azTgAgGG1edvityXdJenFtg/YvkTSVZLebHu/pHOaxwCAKVrTb4ckF67w1NkdzwIAGAGfFAWAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4Ai1ozyzbbvl/SopMckHUky18VQAIDBjRT0xpuSPNzBzwEAjIBTLgBQxKhBj6Sf2t5le77XDrbnbS/YXlhcXBzxcACAlYwa9NcmOUvSWyV9xPbrn7hDki1J5pLMzczMjHg4AMBKRgp6kgeb28OSbpG0qYuhAACDGzrotp9h+6Sj9yW9RdKergYDAAxmlHe5rJd0i+2jP+dbSX7cyVQAgIENHfQkf5D08g5nAQCMgLctAkARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAoYqSg2z7X9m9t32f7iq6GAgAMbuig236qpC9LequkMyRdaPuMrgYDAAxmlFfomyTdl+QPSf4j6TuSzu9mLADAoEYJ+imS/rTs8YFmGwBgCtaM+wC25yXNNw//Yfu34z7mGKyT9PC0h5ig42290hTXfNE0DrqEv+fV4/ltdhol6A9KOnXZ4+c12/5Pki2StoxwnKmzvZBkbtpzTMrxtl6JNR8vqq95lFMuv5R0uu0X2D5B0nskbetmLADAoIZ+hZ7kiO1LJf1E0lMlXZ9kb2eTAQAGMtI59CTbJW3vaJZj2ao+ZTSE4229Ems+XpRes5NMewYAQAf46D8AFEHQe7B9su3bbO9vbtc+yb7PtH3A9pcmOWOX2qzX9kbbd9nea/se2++exqyj6ne5CttPt/3d5vmdtmcnP2W3Wqz5Y7bvbf5ed9hu9Ra5Y1nby5LYfoft2C7xzheC3tsVknYkOV3SjubxSq6UdMdEphqfNuv9p6T3J3mJpHMlfdH2syc448haXq7iEkl/TfIiSddK+uxkp+xWyzX/WtJckpdJuknS5yY7ZbfaXpbE9kmSLpO0c7ITjg9B7+18SVub+1slXdBrJ9uvkLRe0k8nNNe49F1vkt8l2d/c/7Okw5JmJjZhN9pcrmL5n8VNks627QnO2LW+a05ye5J/Ng/v1tJnSlaztpcluVJLv7D/Ncnhxomg97Y+ycHm/kNaivb/sf0USVdL+vgkBxuTvutdzvYmSSdI+v24B+tYm8tV/G+fJEck/U3ScyYy3XgMeomOSyT9aKwTjV/fNds+S9KpSX44ycHGbewf/T9W2f6ZpOf2eOqTyx8kie1ebwX6sKTtSQ6shhdwHaz36M/ZIOmbkjYnebzbKTFNtt8naU7SG6Y9yzg1L8aukfSBKY/SueM26EnOWek524dsb0hysAnY4R67vVrS62x/WNKJkk6w/Y8kx+R14TtYr2w/U9IPJX0yyd1jGnWc2lyu4ug+B2yvkfQsSX+ZzHhj0eoSHbbP0dIv9zck+feEZhuXfms+SdJLJf2ieTH2XEnbbJ+XZGFiU44Bp1x62yZpc3N/s6Rbn7hDkouSnJZkVkunXb5xrMa8hb7rbS7vcIuW1nnTBGfrUpvLVSz/s3inpJ9ndX9Yo++abZ8p6auSzkvS85f5KvOka07ytyTrksw2/37v1tLaV3XMJYK+kqskvdn2fknnNI9le87216c62Xi0We+7JL1e0gds726+Nk5n3OE058SPXq5in6TvJdlr+zO2z2t2u07Sc2zfJ+ljevJ3OB3zWq7581r6r8zvN3+vq/qaTC3XXBKfFAWAIniFDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgiP8C9ML+LndAfc4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Plot distribution of differentially expressed genes for simulated experiments\n",
    "sns.distplot(num_sign_DEGs_simulated,\n",
    "            kde=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Observation:** All simulated experiments found 0 DEGs using adjusted p-value cutoff of <5%"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "As a check, we compared the number of DEGs identified here versus what was reported in the [Reddy et. al. publication](https://www.ncbi.nlm.nih.gov//pubmed/19801529), which found:\n",
    "* 234 genes with a significant (FDR < 5%) change in expression in response to DEX treatment. \n",
    "* After removing pseudogenes (listed in Supplemental Table S3), 209 differentially expressed genes remained \n",
    "* Of the DEX-responsive genes, more showed increases in transcript levels 123 (59%) than showed decreases 86 (41%), and the up-regulation was slightly but significantly stronger than the down-regulation\n",
    "\n",
    "By comparison:\n",
    "* Our study found 60 DEGs instead of 234. \n",
    "* Spot checking the genes identified with their list of DEX-responsive genes (Supplementary Dataset 2), we found the same genes and FC direction was consistent though magnitudes of theirs was lower compared to ours. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load association statistics for template experiment\n",
    "template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_real.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000067082.14</th>\n",
       "      <td>19.385246</td>\n",
       "      <td>18.673139</td>\n",
       "      <td>14.055682</td>\n",
       "      <td>2.342010e-12</td>\n",
       "      <td>1.359233e-07</td>\n",
       "      <td>9.739709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000177606.6</th>\n",
       "      <td>-8.236487</td>\n",
       "      <td>9.185753</td>\n",
       "      <td>-13.126898</td>\n",
       "      <td>8.859555e-12</td>\n",
       "      <td>2.570910e-07</td>\n",
       "      <td>9.292456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000114315.3</th>\n",
       "      <td>-6.285228</td>\n",
       "      <td>6.042663</td>\n",
       "      <td>-12.840532</td>\n",
       "      <td>1.355507e-11</td>\n",
       "      <td>2.622318e-07</td>\n",
       "      <td>9.141666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185950.8</th>\n",
       "      <td>8.790781</td>\n",
       "      <td>17.705828</td>\n",
       "      <td>12.594113</td>\n",
       "      <td>1.966140e-11</td>\n",
       "      <td>2.852721e-07</td>\n",
       "      <td>9.006626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000095752.6</th>\n",
       "      <td>-7.505790</td>\n",
       "      <td>7.299177</td>\n",
       "      <td>-12.204218</td>\n",
       "      <td>3.582671e-11</td>\n",
       "      <td>4.158550e-07</td>\n",
       "      <td>8.782431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        logFC    AveExpr          t       P.Value  \\\n",
       "ENSG00000067082.14  19.385246  18.673139  14.055682  2.342010e-12   \n",
       "ENSG00000177606.6   -8.236487   9.185753 -13.126898  8.859555e-12   \n",
       "ENSG00000114315.3   -6.285228   6.042663 -12.840532  1.355507e-11   \n",
       "ENSG00000185950.8    8.790781  17.705828  12.594113  1.966140e-11   \n",
       "ENSG00000095752.6   -7.505790   7.299177 -12.204218  3.582671e-11   \n",
       "\n",
       "                       adj.P.Val         B  \n",
       "ENSG00000067082.14  1.359233e-07  9.739709  \n",
       "ENSG00000177606.6   2.570910e-07  9.292456  \n",
       "ENSG00000114315.3   2.622318e-07  9.141666  \n",
       "ENSG00000185950.8   2.852721e-07  9.006626  \n",
       "ENSG00000095752.6   4.158550e-07  8.782431  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "template_DE_stats = pd.read_csv(\n",
    "    template_DE_stats_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "\n",
    "template_DEGs = template_DE_stats[template_DE_stats['adj.P.Val']<0.05]\n",
    "print(template_DEGs.shape)\n",
    "template_DEGs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R\n",
    "#library(EnhancedVolcano)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R -i project_id -i template_DE_stats_file -i local_dir\n",
    "\n",
    "#source('../functions/DE_analysis.R')\n",
    "\n",
    "#create_volcano(template_DE_stats_file,\n",
    "#               project_id,\n",
    "#               \"adj.P.Val\",\n",
    "#               local_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene set enrichment analysis\n",
    "\n",
    "**Goal:** To detect modest but coordinated changes in prespecified sets of related genes (i.e. those genes in the same pathway or share the same GO term).\n",
    "\n",
    "1. Ranks all genes based using DE association statistics. In this case we used the p-value scores to rank genes. logFC returned error -- need to look into this.\n",
    "2. An enrichment score (ES) is defined as the maximum distance from the middle of the ranked list. Thus, the enrichment score indicates whether the genes contained in a gene set are clustered towards the beginning or the end of the ranked list (indicating a correlation with change in expression). \n",
    "3. Estimate the statistical significance of the ES by a phenotypic-based permutation test in order to produce a null distribution for the ES( i.e. scores based on permuted phenotype)\n",
    "\n",
    "**Note:** Since there were 0 differentially expressed genes using simulated experiments, we used gene set enrichement analysis instead of over-representation analysis to get ranking of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-05-20 17:31:26,932 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,211 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Registered S3 method overwritten by 'enrichplot':\n",
      "  method               from\n",
      "  fortify.enrichResult DOSE\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,246 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: clusterProfiler v3.14.3  For help: https://guangchuangyu.github.io/software/clusterProfiler\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,251 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: AnnotationDbi\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,253 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: stats4\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,259 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: BiocGenerics\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,261 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: parallel\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,273 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,275 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,275 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,276 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,277 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,278 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: Biobase\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,283 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,287 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: IRanges\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,290 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: S4Vectors\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,298 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,299 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n",
      "2020-05-20 17:31:28,383 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: DOSE v3.12.0  For help: https://guangchuangyu.github.io/software/DOSE\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "suppressWarnings(library(clusterProfiler))\n",
    "suppressWarnings(library(org.Hs.eg.db))\n",
    "suppressWarnings(library(DOSE))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get pathway enrichment for template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-05-20 17:32:17,664 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/robjects/pandas2ri.py:191: FutureWarning: from_items is deprecated. Please use DataFrame.from_dict(dict(items), ...) instead. DataFrame.from_dict(OrderedDict(items)) may be used to preserve the key order.\n",
      "  res = PandasDataFrame.from_items(items)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file  -o template_enriched_pathways\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "template_enriched_pathways <- find_enriched_pathways(template_DE_stats_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(914, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000070</td>\n",
       "      <td>mitotic sister chromatid segregation</td>\n",
       "      <td>152</td>\n",
       "      <td>0.515293</td>\n",
       "      <td>1.604690</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.01635</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>16866</td>\n",
       "      <td>tags=52%, list=29%, signal=37%</td>\n",
       "      <td>ENSG00000154473/ENSG00000121152/ENSG0000016784...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000075</td>\n",
       "      <td>cell cycle checkpoint</td>\n",
       "      <td>215</td>\n",
       "      <td>0.486637</td>\n",
       "      <td>1.538099</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.01635</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>16017</td>\n",
       "      <td>tags=47%, list=28%, signal=34%</td>\n",
       "      <td>ENSG00000182973/ENSG00000154473/ENSG0000016841...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000077</td>\n",
       "      <td>DNA damage checkpoint</td>\n",
       "      <td>144</td>\n",
       "      <td>0.487972</td>\n",
       "      <td>1.516285</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.01635</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>17564</td>\n",
       "      <td>tags=49%, list=30%, signal=34%</td>\n",
       "      <td>ENSG00000182973/ENSG00000168411/ENSG0000013984...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000082</td>\n",
       "      <td>G1/S transition of mitotic cell cycle</td>\n",
       "      <td>267</td>\n",
       "      <td>0.439629</td>\n",
       "      <td>1.398898</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.01635</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>14089</td>\n",
       "      <td>tags=37%, list=24%, signal=28%</td>\n",
       "      <td>ENSG00000198987/ENSG00000199157/ENSG0000020759...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000086</td>\n",
       "      <td>G2/M transition of mitotic cell cycle</td>\n",
       "      <td>244</td>\n",
       "      <td>0.508493</td>\n",
       "      <td>1.614165</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.01635</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>14763</td>\n",
       "      <td>tags=48%, list=25%, signal=36%</td>\n",
       "      <td>ENSG00000196230/ENSG00000139289/ENSG0000017520...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ONTOLOGY          ID                            Description  setSize  \\\n",
       "0       BP  GO:0000070   mitotic sister chromatid segregation      152   \n",
       "1       BP  GO:0000075                  cell cycle checkpoint      215   \n",
       "2       BP  GO:0000077                  DNA damage checkpoint      144   \n",
       "3       BP  GO:0000082  G1/S transition of mitotic cell cycle      267   \n",
       "4       BP  GO:0000086  G2/M transition of mitotic cell cycle      244   \n",
       "\n",
       "   enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "0         0.515293  1.604690  0.000999   0.01635  0.013268  16866   \n",
       "1         0.486637  1.538099  0.000999   0.01635  0.013268  16017   \n",
       "2         0.487972  1.516285  0.000999   0.01635  0.013268  17564   \n",
       "3         0.439629  1.398898  0.000999   0.01635  0.013268  14089   \n",
       "4         0.508493  1.614165  0.000999   0.01635  0.013268  14763   \n",
       "\n",
       "                     leading_edge  \\\n",
       "0  tags=52%, list=29%, signal=37%   \n",
       "1  tags=47%, list=28%, signal=34%   \n",
       "2  tags=49%, list=30%, signal=34%   \n",
       "3  tags=37%, list=24%, signal=28%   \n",
       "4  tags=48%, list=25%, signal=36%   \n",
       "\n",
       "                                     core_enrichment  \n",
       "0  ENSG00000154473/ENSG00000121152/ENSG0000016784...  \n",
       "1  ENSG00000182973/ENSG00000154473/ENSG0000016841...  \n",
       "2  ENSG00000182973/ENSG00000168411/ENSG0000013984...  \n",
       "3  ENSG00000198987/ENSG00000199157/ENSG0000020759...  \n",
       "4  ENSG00000196230/ENSG00000139289/ENSG0000017520...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get pathway enrichment for simulated experiments**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i project_id -i local_dir -i num_runs \n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_DE_stats_file <- paste(local_dir, \n",
    "                                 \"DE_stats/DE_stats_simulated_data_\", \n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    \n",
    "    out_file = paste(local_dir, \n",
    "                     \"GSEA_stats/GSEA_simulated_data_\",\n",
    "                     project_id,\n",
    "                     \"_\",\n",
    "                     i,\n",
    "                     \".txt\", \n",
    "                     sep=\"\")\n",
    "    \n",
    "    if (file.exists(out_file) == FALSE){\n",
    "        enriched_pathways <- find_enriched_pathways(simulated_DE_stats_file) \n",
    "    \n",
    "        write.table(enriched_pathways, file = out_file, row.names = T, sep = \"\\t\", quote = F)\n",
    "        }\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_to_rank = 'enrichmentScore'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(914, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0017166</th>\n",
       "      <td>MF</td>\n",
       "      <td>vinculin binding</td>\n",
       "      <td>11</td>\n",
       "      <td>0.764407</td>\n",
       "      <td>1.823127</td>\n",
       "      <td>0.001058</td>\n",
       "      <td>0.016639</td>\n",
       "      <td>0.013503</td>\n",
       "      <td>8431</td>\n",
       "      <td>tags=64%, list=15%, signal=54%</td>\n",
       "      <td>ENSG00000137076/ENSG00000173402/ENSG0000008915...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0019774</th>\n",
       "      <td>CC</td>\n",
       "      <td>proteasome core complex, beta-subunit complex</td>\n",
       "      <td>11</td>\n",
       "      <td>0.756475</td>\n",
       "      <td>1.804210</td>\n",
       "      <td>0.001058</td>\n",
       "      <td>0.016639</td>\n",
       "      <td>0.013503</td>\n",
       "      <td>10764</td>\n",
       "      <td>tags=73%, list=19%, signal=59%</td>\n",
       "      <td>ENSG00000126067/ENSG00000159377/ENSG0000024006...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0005763</th>\n",
       "      <td>CC</td>\n",
       "      <td>mitochondrial small ribosomal subunit</td>\n",
       "      <td>28</td>\n",
       "      <td>0.744290</td>\n",
       "      <td>2.052372</td>\n",
       "      <td>0.001001</td>\n",
       "      <td>0.016350</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>12851</td>\n",
       "      <td>tags=75%, list=22%, signal=58%</td>\n",
       "      <td>ENSG00000135972/ENSG00000132676/ENSG0000006258...</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000314</th>\n",
       "      <td>CC</td>\n",
       "      <td>organellar small ribosomal subunit</td>\n",
       "      <td>28</td>\n",
       "      <td>0.744290</td>\n",
       "      <td>2.052372</td>\n",
       "      <td>0.001001</td>\n",
       "      <td>0.016350</td>\n",
       "      <td>0.013268</td>\n",
       "      <td>12851</td>\n",
       "      <td>tags=75%, list=22%, signal=58%</td>\n",
       "      <td>ENSG00000135972/ENSG00000132676/ENSG0000006258...</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000805</th>\n",
       "      <td>CC</td>\n",
       "      <td>X chromosome</td>\n",
       "      <td>10</td>\n",
       "      <td>0.741553</td>\n",
       "      <td>1.721130</td>\n",
       "      <td>0.002160</td>\n",
       "      <td>0.029290</td>\n",
       "      <td>0.023769</td>\n",
       "      <td>10394</td>\n",
       "      <td>tags=60%, list=18%, signal=49%</td>\n",
       "      <td>ENSG00000121931/ENSG00000132475/ENSG0000012337...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ONTOLOGY                                    Description  setSize  \\\n",
       "ID                                                                            \n",
       "GO:0017166       MF                               vinculin binding       11   \n",
       "GO:0019774       CC  proteasome core complex, beta-subunit complex       11   \n",
       "GO:0005763       CC          mitochondrial small ribosomal subunit       28   \n",
       "GO:0000314       CC             organellar small ribosomal subunit       28   \n",
       "GO:0000805       CC                                   X chromosome       10   \n",
       "\n",
       "            enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "ID                                                                           \n",
       "GO:0017166         0.764407  1.823127  0.001058  0.016639  0.013503   8431   \n",
       "GO:0019774         0.756475  1.804210  0.001058  0.016639  0.013503  10764   \n",
       "GO:0005763         0.744290  2.052372  0.001001  0.016350  0.013268  12851   \n",
       "GO:0000314         0.744290  2.052372  0.001001  0.016350  0.013268  12851   \n",
       "GO:0000805         0.741553  1.721130  0.002160  0.029290  0.023769  10394   \n",
       "\n",
       "                              leading_edge  \\\n",
       "ID                                           \n",
       "GO:0017166  tags=64%, list=15%, signal=54%   \n",
       "GO:0019774  tags=73%, list=19%, signal=59%   \n",
       "GO:0005763  tags=75%, list=22%, signal=58%   \n",
       "GO:0000314  tags=75%, list=22%, signal=58%   \n",
       "GO:0000805  tags=60%, list=18%, signal=49%   \n",
       "\n",
       "                                              core_enrichment  ranking  \n",
       "ID                                                                      \n",
       "GO:0017166  ENSG00000137076/ENSG00000173402/ENSG0000008915...      1.0  \n",
       "GO:0019774  ENSG00000126067/ENSG00000159377/ENSG0000024006...      2.0  \n",
       "GO:0005763  ENSG00000135972/ENSG00000132676/ENSG0000006258...      3.5  \n",
       "GO:0000314  ENSG00000135972/ENSG00000132676/ENSG0000006258...      3.5  \n",
       "GO:0000805  ENSG00000121931/ENSG00000132475/ENSG0000012337...      5.0  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get ranks of template experiment\n",
    "# Rank pathways by highest enrichment score\n",
    "template_enriched_pathways['ranking'] = template_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "template_enriched_pathways = template_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "\n",
    "# Set index to GO ID\n",
    "template_enriched_pathways.set_index(\"ID\", inplace=True)\n",
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check that GO IDs are unique\n",
    "template_enriched_pathways.index.nunique() == len(template_enriched_pathways)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "Again, we want to compare our ranked pathways found against what was reported in the original publication.\n",
    "\n",
    "*The DEX-responsive genes that we identified are primarily implicated in two broad classes: stress response and development (Table 1; Reimand et al. 2007). Comparison to Gene Ontology (GO) categories (Ashburner et al. 2000) revealed that the identified genes are involved in stress response (P = 6 × 10−11), organ development (P = 5 × 10−15), cell differentiation (P = 1 × 10−11), hormone secretion (P = 4 × 10−7), and apoptosis (P = 5 × 10−12).*\n",
    "\n",
    "We found pathways that are consistent with what publication found: pathways related to anatomical structure (i.e. vinculin, ) and cell differentiation (i.e. centromeric sister chromatid cohesion), hormone secretion (i.e. somatic hypermutation of immunoglobulin genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Simulated experiments**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(42136, 13)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0000353</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000353</td>\n",
       "      <td>formation of quadruple SL/U4/U5/U6 snRNP</td>\n",
       "      <td>12</td>\n",
       "      <td>0.765343</td>\n",
       "      <td>1.817596</td>\n",
       "      <td>0.001047</td>\n",
       "      <td>0.005979</td>\n",
       "      <td>0.003670</td>\n",
       "      <td>13628</td>\n",
       "      <td>tags=100%, list=23%, signal=77%</td>\n",
       "      <td>ENSG00000199377/ENSG00000200156/ENSG0000020735...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0045291</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0045291</td>\n",
       "      <td>mRNA trans splicing, SL addition</td>\n",
       "      <td>12</td>\n",
       "      <td>0.765343</td>\n",
       "      <td>1.817596</td>\n",
       "      <td>0.001047</td>\n",
       "      <td>0.005979</td>\n",
       "      <td>0.003670</td>\n",
       "      <td>13628</td>\n",
       "      <td>tags=100%, list=23%, signal=77%</td>\n",
       "      <td>ENSG00000199377/ENSG00000200156/ENSG0000020735...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000365</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000365</td>\n",
       "      <td>mRNA trans splicing, via spliceosome</td>\n",
       "      <td>12</td>\n",
       "      <td>0.765343</td>\n",
       "      <td>1.817596</td>\n",
       "      <td>0.001047</td>\n",
       "      <td>0.005979</td>\n",
       "      <td>0.003670</td>\n",
       "      <td>13628</td>\n",
       "      <td>tags=100%, list=23%, signal=77%</td>\n",
       "      <td>ENSG00000199377/ENSG00000200156/ENSG0000020735...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:1904851</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:1904851</td>\n",
       "      <td>positive regulation of establishment of protei...</td>\n",
       "      <td>10</td>\n",
       "      <td>0.762148</td>\n",
       "      <td>1.749047</td>\n",
       "      <td>0.002128</td>\n",
       "      <td>0.010340</td>\n",
       "      <td>0.006347</td>\n",
       "      <td>12577</td>\n",
       "      <td>tags=90%, list=22%, signal=71%</td>\n",
       "      <td>ENSG00000120438/ENSG00000130826/ENSG0000015075...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:1903405</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:1903405</td>\n",
       "      <td>protein localization to nuclear body</td>\n",
       "      <td>10</td>\n",
       "      <td>0.762148</td>\n",
       "      <td>1.749047</td>\n",
       "      <td>0.002128</td>\n",
       "      <td>0.010340</td>\n",
       "      <td>0.006347</td>\n",
       "      <td>12577</td>\n",
       "      <td>tags=90%, list=22%, signal=71%</td>\n",
       "      <td>ENSG00000120438/ENSG00000130826/ENSG0000015075...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ONTOLOGY          ID  \\\n",
       "GO:0000353       BP  GO:0000353   \n",
       "GO:0045291       BP  GO:0045291   \n",
       "GO:0000365       BP  GO:0000365   \n",
       "GO:1904851       BP  GO:1904851   \n",
       "GO:1903405       BP  GO:1903405   \n",
       "\n",
       "                                                  Description  setSize  \\\n",
       "GO:0000353           formation of quadruple SL/U4/U5/U6 snRNP       12   \n",
       "GO:0045291                   mRNA trans splicing, SL addition       12   \n",
       "GO:0000365               mRNA trans splicing, via spliceosome       12   \n",
       "GO:1904851  positive regulation of establishment of protei...       10   \n",
       "GO:1903405               protein localization to nuclear body       10   \n",
       "\n",
       "            enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "GO:0000353         0.765343  1.817596  0.001047  0.005979  0.003670  13628   \n",
       "GO:0045291         0.765343  1.817596  0.001047  0.005979  0.003670  13628   \n",
       "GO:0000365         0.765343  1.817596  0.001047  0.005979  0.003670  13628   \n",
       "GO:1904851         0.762148  1.749047  0.002128  0.010340  0.006347  12577   \n",
       "GO:1903405         0.762148  1.749047  0.002128  0.010340  0.006347  12577   \n",
       "\n",
       "                               leading_edge  \\\n",
       "GO:0000353  tags=100%, list=23%, signal=77%   \n",
       "GO:0045291  tags=100%, list=23%, signal=77%   \n",
       "GO:0000365  tags=100%, list=23%, signal=77%   \n",
       "GO:1904851   tags=90%, list=22%, signal=71%   \n",
       "GO:1903405   tags=90%, list=22%, signal=71%   \n",
       "\n",
       "                                              core_enrichment  ranking  \n",
       "GO:0000353  ENSG00000199377/ENSG00000200156/ENSG0000020735...      2.0  \n",
       "GO:0045291  ENSG00000199377/ENSG00000200156/ENSG0000020735...      2.0  \n",
       "GO:0000365  ENSG00000199377/ENSG00000200156/ENSG0000020735...      2.0  \n",
       "GO:1904851  ENSG00000120438/ENSG00000130826/ENSG0000015075...      5.0  \n",
       "GO:1903405  ENSG00000120438/ENSG00000130826/ENSG0000015075...      5.0  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concatenate simulated experiments\n",
    "simulated_enriched_pathways_all = pd.DataFrame()\n",
    "for i in range(num_runs):\n",
    "    simulated_GSEA_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"GSEA_stats\",\n",
    "        \"GSEA_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "    \n",
    "    #Read results\n",
    "    simulated_enriched_pathways = pd.read_csv(\n",
    "        simulated_GSEA_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "    \n",
    "    # Add ranks of simulated experiment\n",
    "    simulated_enriched_pathways['ranking'] = simulated_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "    simulated_enriched_pathways = simulated_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "    \n",
    "    # Concatenate df\n",
    "    simulated_enriched_pathways_all = pd.concat([simulated_enriched_pathways_all,\n",
    "                                       simulated_enriched_pathways])\n",
    "    \n",
    "print(simulated_enriched_pathways_all.shape)\n",
    "simulated_enriched_pathways_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">enrichmentScore</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>count</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0000018</th>\n",
       "      <td>0.441868</td>\n",
       "      <td>0.016700</td>\n",
       "      <td>4</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>1919.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000027</th>\n",
       "      <td>0.548594</td>\n",
       "      <td>0.021737</td>\n",
       "      <td>4</td>\n",
       "      <td>0.006503</td>\n",
       "      <td>328.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000028</th>\n",
       "      <td>0.573867</td>\n",
       "      <td>0.056410</td>\n",
       "      <td>5</td>\n",
       "      <td>0.007121</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000038</th>\n",
       "      <td>0.535535</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0.009009</td>\n",
       "      <td>415.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000041</th>\n",
       "      <td>0.449833</td>\n",
       "      <td>0.019520</td>\n",
       "      <td>11</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>1167.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           enrichmentScore                    pvalue ranking\n",
       "                      mean       std count    median  median\n",
       "ID                                                          \n",
       "GO:0000018        0.441868  0.016700     4  0.005495  1919.0\n",
       "GO:0000027        0.548594  0.021737     4  0.006503   328.5\n",
       "GO:0000028        0.573867  0.056410     5  0.007121   113.0\n",
       "GO:0000038        0.535535       NaN     1  0.009009   415.0\n",
       "GO:0000041        0.449833  0.019520    11  0.001998  1167.0"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simulated_enriched_pathways_stats = simulated_enriched_pathways_all.groupby(['ID'])[['enrichmentScore', 'pvalue', 'ranking']].agg({\n",
    "    col_to_rank:['mean', 'std','count'],\n",
    "    'pvalue':['median'],\n",
    "    'ranking':['median']\n",
    "})\n",
    "\n",
    "simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-05-20 17:32:18,780 [461] WARNING  py.warnings:110: [JupyterRequire] /home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/reshape/merge.py:544: UserWarning: merging between different levels can give an unintended result (1 levels on the left, 2 on the right)\n",
      "  warnings.warn(msg, UserWarning)\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "      <th>(enrichmentScore, mean)</th>\n",
       "      <th>(enrichmentScore, std)</th>\n",
       "      <th>(enrichmentScore, count)</th>\n",
       "      <th>(pvalue, median)</th>\n",
       "      <th>(ranking, median)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0000018</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.441868</td>\n",
       "      <td>0.016700</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>1919.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000027</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.548594</td>\n",
       "      <td>0.021737</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.006503</td>\n",
       "      <td>328.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000028</th>\n",
       "      <td>BP</td>\n",
       "      <td>ribosomal small subunit assembly</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.628994</td>\n",
       "      <td>1.622578</td>\n",
       "      <td>0.005097</td>\n",
       "      <td>0.047097</td>\n",
       "      <td>0.038218</td>\n",
       "      <td>14890.0</td>\n",
       "      <td>tags=56%, list=26%, signal=41%</td>\n",
       "      <td>ENSG00000083845/ENSG00000115268/ENSG0000013259...</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.573867</td>\n",
       "      <td>0.056410</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.007121</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000038</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.535535</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.009009</td>\n",
       "      <td>415.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000041</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.449833</td>\n",
       "      <td>0.019520</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>1167.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ONTOLOGY                       Description  setSize  \\\n",
       "ID                                                               \n",
       "GO:0000018      NaN                               NaN      NaN   \n",
       "GO:0000027      NaN                               NaN      NaN   \n",
       "GO:0000028       BP  ribosomal small subunit assembly     18.0   \n",
       "GO:0000038      NaN                               NaN      NaN   \n",
       "GO:0000041      NaN                               NaN      NaN   \n",
       "\n",
       "            enrichmentScore       NES    pvalue  p.adjust   qvalues     rank  \\\n",
       "ID                                                                             \n",
       "GO:0000018              NaN       NaN       NaN       NaN       NaN      NaN   \n",
       "GO:0000027              NaN       NaN       NaN       NaN       NaN      NaN   \n",
       "GO:0000028         0.628994  1.622578  0.005097  0.047097  0.038218  14890.0   \n",
       "GO:0000038              NaN       NaN       NaN       NaN       NaN      NaN   \n",
       "GO:0000041              NaN       NaN       NaN       NaN       NaN      NaN   \n",
       "\n",
       "                              leading_edge  \\\n",
       "ID                                           \n",
       "GO:0000018                             NaN   \n",
       "GO:0000027                             NaN   \n",
       "GO:0000028  tags=56%, list=26%, signal=41%   \n",
       "GO:0000038                             NaN   \n",
       "GO:0000041                             NaN   \n",
       "\n",
       "                                              core_enrichment  ranking  \\\n",
       "ID                                                                       \n",
       "GO:0000018                                                NaN      NaN   \n",
       "GO:0000027                                                NaN      NaN   \n",
       "GO:0000028  ENSG00000083845/ENSG00000115268/ENSG0000013259...     84.0   \n",
       "GO:0000038                                                NaN      NaN   \n",
       "GO:0000041                                                NaN      NaN   \n",
       "\n",
       "            (enrichmentScore, mean)  (enrichmentScore, std)  \\\n",
       "ID                                                            \n",
       "GO:0000018                 0.441868                0.016700   \n",
       "GO:0000027                 0.548594                0.021737   \n",
       "GO:0000028                 0.573867                0.056410   \n",
       "GO:0000038                 0.535535                     NaN   \n",
       "GO:0000041                 0.449833                0.019520   \n",
       "\n",
       "            (enrichmentScore, count)  (pvalue, median)  (ranking, median)  \n",
       "ID                                                                         \n",
       "GO:0000018                       4.0          0.005495             1919.0  \n",
       "GO:0000027                       4.0          0.006503              328.5  \n",
       "GO:0000028                       5.0          0.007121              113.0  \n",
       "GO:0000038                       1.0          0.009009              415.0  \n",
       "GO:0000041                      11.0          0.001998             1167.0  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge template statistics with simulated statistics\n",
    "template_simulated_enriched_pathways_stats = template_enriched_pathways.merge(simulated_enriched_pathways_stats, \n",
    "                                                                              how='outer',\n",
    "                                                                              left_index=True,\n",
    "                                                                              right_index=True)\n",
    "template_simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse columns\n",
    "median_pval_simulated = template_simulated_enriched_pathways_stats[('pvalue','median')]\n",
    "median_rank_simulated = template_simulated_enriched_pathways_stats[('ranking','median')]\n",
    "mean_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'mean')]\n",
    "std_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'std')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Examine enriched pathways\n",
    "Examine the enriched pathways identified from template experiment -- How are these enriched pathways ranked in the simulated experiments?\n",
    "\n",
    "Recall that there were 25 simulated experiments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculations for summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "{ \n",
       "        'use strict';\n",
       "        \n",
       "        const href = \"https://cdn.datatables.net/v/dt/dt-1.10.18/af-2.3.3/b-1.5.6/b-colvis-1.5.6/b-flash-1.5.6/b-html5-1.5.6/b-print-1.5.6/cr-1.5.0/fc-3.2.5/fh-3.1.4/kt-2.5.0/r-2.2.2/rg-1.1.0/rr-1.2.4/sc-2.0.0/sl-1.3.0/datatables.min.css\";\n",
       "        const attributes = {'id': 'datatables.min.css'} || {};\n",
       "\n",
       "        if ( !$(`link[href*=\"${href}\"]`).length ) {\n",
       "            let link = document.createElement(\"link\");\n",
       "\n",
       "            link.rel = \"stylesheet\";\n",
       "            link.type = \"text/css\";\n",
       "            try {\n",
       "                link.href = requirejs.toUrl(href, 'css');\n",
       "            } catch (error) {\n",
       "                link.href = href;\n",
       "            }\n",
       "            \n",
       "            Object.entries(attributes)\n",
       "                .forEach( ([attr, val]) => $(link).attr(attr, val) );\n",
       "            \n",
       "            document.head.appendChild(link);\n",
       "        }\n",
       "     }"
      ],
      "text/plain": [
       "<JupyterRequire.display.SafeScript object>"
      ]
     },
     "metadata": {
      "finalized": true
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "{ \n",
       "        'use strict'\n",
       "        \n",
       "        const style = `.dt-button.is-disabled{background-color:#fff;border-color:#dbdbdb;box-shadow:none;opacity:.5;cursor:not-allowed;pointer-events:none}.dt-chart-toolbar{text-align:left}.dt-chart-settings-container{background-color:#fff;border-bottom-left-radius:6px;border-bottom-right-radius:6px;border-top:2px solid #dbdbdb;box-shadow:0 8px 8px rgba(10,10,10,.1);font-size:.875rem;position:absolute;z-index:20}.dt-chart-settings{font-family:BlinkMacSystemFont,-apple-system,\"Segoe UI\",Roboto,Oxygen,Ubuntu,Cantarell,\"Fira Sans\",\"Droid Sans\",\"Helvetica Neue\",Helvetica,Arial,sans-serif;font-size:1.10rem}.dt-chart-menu-label:first-child{margin-top:5px;margin-left:5px;font-size:.85em}table.dataTable{border-collapse:separate !important;border-spacing:0;clear:both;table-layout:auto !important}table.dataTable thead th,table.dataTable thead td{padding:10px 18px;border-bottom:thin solid #ddd}table.dataTable td.column-data-preview{padding:11px 7px}.dataTables_scrollBody{z-index:auto}.dataTables_scrollBody thead:not(.data-preview){visibility:collapse}.dataTables_wrapper.no-footer .dataTables_scrollBody{padding-bottom:20px;border-bottom:0}.dataTables_wrapper.no-footer div.dataTables_scrollBody>table{border-top:thin solid #ddd;border-bottom:thin solid #ddd}.dataTables_wrapper input{display:inline-block;padding:.375rem .75rem;height:calc(2.75rem + 5px);line-height:1.5;color:#495057;background-color:#fff;background-clip:padding-box;border:thin solid #ced4da;border-radius:.25rem;transition:border-color .15s ease-in-out,box-shadow .15s ease-in-out}.dataTables_wrapper label{margin-right:0 !important;margin-bottom:11px;box-sizing:border-box;white-space:nowrap}.dataTables_wrapper select{display:inline-block;padding:.375rem 2.75rem .375rem 1.75rem;height:calc(2.75rem + 5px);line-height:1.5;vertical-align:middle;background:#fff url(\"data:image/svg+xml;charset=utf8,%3Csvgxmlns='http://www.w3.org/2000/svg'viewBox='0045'%3E%3Cpathfill='%23343a40'd='M20L02h4zm05L03h4z'/%3E%3C/svg%3E\") no-repeat right .75rem center;background-size:auto 11px;border:thin solid #ced4da;border-radius:.25rem;-webkit-appearance:none;-moz-appearance:none}.dataTables_wrapper select.dtype{padding:0 1.75rem;font-size:9px;height:calc(1.25rem + 5px);background:#ffdd57;color:rgba(0,0,0,.7);border:0;border-radius:.25rem;text-transform:uppercase}`;\n",
       "        const attributes = {'id': 'jupyter-datatables-css'} || {};\n",
       "        \n",
       "        let id = attributes.id;\n",
       "        let elem_exists = id ? $(`style#${id}`).length > 0 : false;\n",
       "        \n",
       "        let e = elem_exists ? document.querySelector(`style#${id}`)\n",
       "                            : document.createElement(\"style\");\n",
       "        \n",
       "        $(e).text(`${style}`).attr('type', 'text/css');\n",
       "        \n",
       "        Object.entries(attributes)\n",
       "            .forEach( ([attr, val]) => $(e).attr(attr, val) );\n",
       "\n",
       "        if (!elem_exists) document.head.appendChild(e);\n",
       "     }"
      ],
      "text/plain": [
       "<JupyterRequire.display.SafeScript object>"
      ]
     },
     "metadata": {
      "finalized": true
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "{ \n",
       "        'use strict';\n",
       "    \n",
       "        const script = `define('dt-config', ['underscore', 'moment'], function (_, moment) {\n",
       "\n",
       "  let mapDType = (dtypes, target) => _.object(_.zip(dtypes, new Array(dtypes.length).fill(target)))\n",
       "\n",
       "  $.fn.dataTable.defaults.dateDisplayFormat = 'YYYYMMDD'\n",
       "  $.fn.dataTable.defaults.formatDate = (t, format) => moment(t, format || $.fn.dataTable.defaults.dateDisplayFormat)\n",
       "\n",
       "  $.fn.dataTable.defaults.dTypeMap = {\n",
       "    ...mapDType(['int8', 'int16', 'int32', 'int64', 'float8', 'float16', 'float32', 'float64'], 'num'),\n",
       "    ...mapDType(['datetime8[ns]', 'datetime16[ns]', 'datetime32[ns]', 'datetime64[ns]'], 'date'),\n",
       "    ...mapDType(['timedelta8[ns]', 'timedelta16[ns]', 'timedelta32[ns]', 'timedelta64[ns]'], 'string'), // TODO: Custom type \\`timedelta\\`\n",
       "    ...mapDType(['object', 'string'], 'string'),\n",
       "    ...mapDType(['bool'], 'boolean'),\n",
       "    ...mapDType(['default'], 'num')\n",
       "  }\n",
       "\n",
       "  $.fn.dataTable.defaults.chartMap = {\n",
       "    boolean: ['CategoricalBar', 'Histogram'],\n",
       "    date: ['CategoricalBar', 'Histogram'],\n",
       "    num: ['Histogram', 'CategoricalBar', 'Bar', 'Line'],\n",
       "    string: ['CategoricalBar'],\n",
       "\n",
       "    undefined: ['Bar']\n",
       "  }\n",
       "\n",
       "  return $.fn.dataTable.defaults\n",
       "})\n",
       "`;\n",
       "        const attributes = {'id': 'dt-config-js'} || {};\n",
       "        \n",
       "        let id = attributes.id;\n",
       "        let elem_exists = id ? $(`script#${id}`).length > 0 : false;\n",
       "        \n",
       "        let e = elem_exists ? document.querySelector(`script#${id}`)\n",
       "                            : document.createElement(\"script\");\n",
       "        \n",
       "        $(e).text(`${script}`).attr('type', 'text/javascript');\n",
       "        \n",
       "        Object.entries(attributes)\n",
       "            .forEach( ([attr, val]) => $(e).attr(attr, val) );\n",
       "\n",
       "        if (!elem_exists) document.head.appendChild(e);\n",
       "     }"
      ],
      "text/plain": [
       "<JupyterRequire.display.SafeScript object>"
      ]
     },
     "metadata": {
      "finalized": true
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "init_datatables_mode()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "data": {},
     "metadata": {
      "display": {
       "element": {
        "0": {},
        "length": 1
       }
      },
      "finalized": false,
      "frozen": true,
      "frozen_output": {
       "text/html": "<div id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_wrapper\" class=\"dataTables_wrapper no-footer\"><div class=\"dt-buttons\">   <button class=\"dt-button buttons-print\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" type=\"button\"><span>Print</span></button> <button class=\"dt-button buttons-csv buttons-html5\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" type=\"button\"><span>CSV</span></button> <button class=\"dt-button buttons-pdf buttons-html5\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" type=\"button\"><span>PDF</span></button> </div><div class=\"dataTables_length\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_length\"><label>Show <select name=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_length\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" class=\"\"><option value=\"10\">10</option><option value=\"25\">25</option><option value=\"50\">50</option><option value=\"100\">100</option></select> entries</label></div><div id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_filter\" class=\"dataTables_filter\"><label>Search:<input type=\"search\" class=\"\" placeholder=\"\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\"></label></div><div class=\"dataTables_scroll\"><div class=\"dataTables_scrollHead\" style=\"overflow: hidden; position: relative; border: 0px; width: 100%;\"><div class=\"dataTables_scrollHeadInner\" style=\"box-sizing: content-box; width: 2052.8px; padding-right: 0px;\"><table border=\"1\" class=\"dataframe table cell-border nowrap dataTable no-footer\" role=\"grid\" style=\"margin-left: 0px; width: 2052.8px;\"><thead>\n    <tr style=\"text-align: right;\" role=\"row\"><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 56px;\" aria-label=\": activate to sort column ascending\"></th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148px;\" aria-label=\"GO ID: activate to sort column ascending\">GO ID</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 224px;\" aria-label=\"Pathway: activate to sort column ascending\">Pathway</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px;\" aria-label=\"P-value (Real): activate to sort column ascending\">P-value (Real)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px;\" aria-label=\"Rank (Real): activate to sort column ascending\">Rank (Real)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px;\" aria-label=\"Test statistic (Real): activate to sort column ascending\">Test statistic (Real)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 157.6px;\" aria-label=\"Median p-value (simulated): activate to sort column ascending\">Median p-value (simulated)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px;\" aria-label=\"Median rank (simulated): activate to sort column ascending\">Median rank (simulated)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 176px;\" aria-label=\"Mean test statistic (simulated): activate to sort column ascending\">Mean test statistic (simulated)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px;\" aria-label=\"Std deviation (simulated): activate to sort column ascending\">Std deviation (simulated)</th><th class=\"dt-body-center dt-head-center sorting\" tabindex=\"0\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px;\" aria-label=\"Z score: activate to sort column ascending\">Z score</th></tr>\n  <tr role=\"row\" class=\"dtype-preview\">\n      <th class=\" dt-body-center dt-head-center\"></th>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 1\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"object\">object</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 2\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"object\">object</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 3\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 4\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 5\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 6\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 7\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 8\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 9\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 10\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></td>\n    </tr><tr role=\"row\" class=\"column-data-preview\">\n      <th class=\" dt-body-center dt-head-center\"></th>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 1\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 2\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 3\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 4\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 5\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 6\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 7\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 8\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 9\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 10\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></td>\n    </tr></thead></table></div></div><div class=\"dataTables_scrollBody\" style=\"position: relative; overflow: auto; width: 100%;\"><table border=\"1\" class=\"dataframe table cell-border nowrap dataTable no-footer\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" role=\"grid\" aria-describedby=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_info\" style=\"width: 2053px;\"><thead>\n    <tr style=\"text-align: right; height: 12px;\" role=\"row\"><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 56px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\": activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"></div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"GO ID: activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">GO ID</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 224px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Pathway: activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Pathway</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"P-value (Real): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">P-value (Real)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Rank (Real): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Rank (Real)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 148.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Test statistic (Real): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Test statistic (Real)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 157.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Median p-value (simulated): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Median p-value (simulated)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Median rank (simulated): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Median rank (simulated)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 176px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Mean test statistic (simulated): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Mean test statistic (simulated)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Std deviation (simulated): activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Std deviation (simulated)</div></th><th class=\"dt-body-center dt-head-center sorting\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" rowspan=\"1\" colspan=\"1\" style=\"width: 149.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\" aria-label=\"Z score: activate to sort column ascending\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\">Z score</div></th></tr>\n  <tr role=\"row\" class=\"dtype-preview\" style=\"height: 12px;\">\n      <th class=\" dt-body-center dt-head-center\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 56px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"></div></th>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 1\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 148px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"object\">object</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 2\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 224px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"object\">object</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 3\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 148.8px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 4\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 148.8px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 5\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 148.8px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 6\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 157.6px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 7\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 149.6px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 8\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 176px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 9\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 149.6px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n      <td class=\"column-dtype-preview dt-head-center\" aria-label=\"dtype preview for column 10\" style=\"padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px; width: 149.6px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dtype-container\"><select role=\"option\" class=\"dtype\"><option value=\"float64\">float64</option></select></div></div></td>\n    </tr><tr role=\"row\" class=\"column-data-preview\" style=\"height: 12px;\">\n      <th class=\" dt-body-center dt-head-center\" style=\"width: 56px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"></div></th>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 1\" style=\"width: 170px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 2\" style=\"width: 246px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 3\" style=\"width: 170.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 4\" style=\"width: 170.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 5\" style=\"width: 170.8px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 6\" style=\"width: 179.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 7\" style=\"width: 171.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 8\" style=\"width: 198px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 9\" style=\"width: 171.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n      <td class=\"column-data-preview\" role=\"figure\" aria-label=\"data preview for column 10\" style=\"width: 171.6px; padding-top: 0px; padding-bottom: 0px; border-top-width: 0px; border-bottom-width: 0px; height: 0px;\"><div class=\"dataTables_sizing\" style=\"height: 0px; overflow: hidden;\"><div class=\"dt-chart-container\"><div is=\"dt-container\" class=\"dt-container dt-chart-toolbar\"></div><canvas width=\"208\" height=\"103\" style=\"margin: auto; display: block; height: 83px; width: 167px;\"></canvas></div></div></td>\n    </tr></thead>\n  \n  <tbody>\n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n    \n  <tr role=\"row\" class=\"odd\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000075</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000075</td>\n      <td class=\" dt-body-center dt-head-center\">cell cycle checkpoint</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">489.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.486637</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">1041.5</td>\n      <td class=\" dt-body-center dt-head-center\">0.443276</td>\n      <td class=\" dt-body-center dt-head-center\">0.037489</td>\n      <td class=\" dt-body-center dt-head-center\">1.156647</td>\n    </tr><tr role=\"row\" class=\"even\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000077</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000077</td>\n      <td class=\" dt-body-center dt-head-center\">DNA damage checkpoint</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">472.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.487972</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">1009.5</td>\n      <td class=\" dt-body-center dt-head-center\">0.442027</td>\n      <td class=\" dt-body-center dt-head-center\">0.039051</td>\n      <td class=\" dt-body-center dt-head-center\">1.176552</td>\n    </tr><tr role=\"row\" class=\"odd\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000082</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000082</td>\n      <td class=\" dt-body-center dt-head-center\">G1/S transition of mitotic cell cycle</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">763.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.439629</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">1003.5</td>\n      <td class=\" dt-body-center dt-head-center\">0.445661</td>\n      <td class=\" dt-body-center dt-head-center\">0.039140</td>\n      <td class=\" dt-body-center dt-head-center\">-0.154111</td>\n    </tr><tr role=\"row\" class=\"even\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000083</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000083</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">0.006008</td>\n      <td class=\" dt-body-center dt-head-center\">293.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.540150</td>\n      <td class=\" dt-body-center dt-head-center\">0.002715</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n    </tr><tr role=\"row\" class=\"odd\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000086</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000086</td>\n      <td class=\" dt-body-center dt-head-center\">G2/M transition of mitotic cell cycle</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">373.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.508493</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">1095.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.442299</td>\n      <td class=\" dt-body-center dt-head-center\">0.042893</td>\n      <td class=\" dt-body-center dt-head-center\">1.543225</td>\n    </tr><tr role=\"row\" class=\"even\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000123</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000123</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">0.006993</td>\n      <td class=\" dt-body-center dt-head-center\">1392.5</td>\n      <td class=\" dt-body-center dt-head-center\">0.446779</td>\n      <td class=\" dt-body-center dt-head-center\">0.015836</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n    </tr><tr role=\"row\" class=\"odd\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000151</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000151</td>\n      <td class=\" dt-body-center dt-head-center\">ubiquitin ligase complex</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">768.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.438137</td>\n      <td class=\" dt-body-center dt-head-center\">0.000999</td>\n      <td class=\" dt-body-center dt-head-center\">1727.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.407017</td>\n      <td class=\" dt-body-center dt-head-center\">0.040695</td>\n      <td class=\" dt-body-center dt-head-center\">0.764728</td>\n    </tr><tr role=\"row\" class=\"even\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000152</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000152</td>\n      <td class=\" dt-body-center dt-head-center\">nuclear ubiquitin ligase complex</td>\n      <td class=\" dt-body-center dt-head-center\">0.004995</td>\n      <td class=\" dt-body-center dt-head-center\">283.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.531832</td>\n      <td class=\" dt-body-center dt-head-center\">0.003497</td>\n      <td class=\" dt-body-center dt-head-center\">396.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.513999</td>\n      <td class=\" dt-body-center dt-head-center\">0.009909</td>\n      <td class=\" dt-body-center dt-head-center\">1.799736</td>\n    </tr><tr role=\"row\" class=\"odd\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000175</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000175</td>\n      <td class=\" dt-body-center dt-head-center\">3'-5'-exoribonuclease activity</td>\n      <td class=\" dt-body-center dt-head-center\">0.004995</td>\n      <td class=\" dt-body-center dt-head-center\">212.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.560727</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n    </tr><tr role=\"row\" class=\"even\">\n      <th class=\" dt-body-center dt-head-center\">GO:0000177</th>\n      <td class=\" dt-body-center dt-head-center\">GO:0000177</td>\n      <td class=\" dt-body-center dt-head-center\">cytoplasmic exosome (RNase complex)</td>\n      <td class=\" dt-body-center dt-head-center\">0.003096</td>\n      <td class=\" dt-body-center dt-head-center\">50.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.661216</td>\n      <td class=\" dt-body-center dt-head-center\">0.009346</td>\n      <td class=\" dt-body-center dt-head-center\">71.0</td>\n      <td class=\" dt-body-center dt-head-center\">0.637138</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n      <td class=\" dt-body-center dt-head-center\">NaN</td>\n    </tr></tbody>\n</table></div></div><div class=\"dataTables_info\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_info\" role=\"status\" aria-live=\"polite\">Showing 11 to 20 of 2,000 entries</div><div class=\"dataTables_paginate paging_simple_numbers\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_paginate\"><a class=\"paginate_button previous\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"0\" tabindex=\"0\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_previous\">Previous</a><span><a class=\"paginate_button \" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"1\" tabindex=\"0\">1</a><a class=\"paginate_button current\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"2\" tabindex=\"0\">2</a><a class=\"paginate_button \" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"3\" tabindex=\"0\">3</a><a class=\"paginate_button \" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"4\" tabindex=\"0\">4</a><a class=\"paginate_button \" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"5\" tabindex=\"0\">5</a><span class=\"ellipsis\">…</span><a class=\"paginate_button \" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"6\" tabindex=\"0\">200</a></span><a class=\"paginate_button next\" aria-controls=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e\" data-dt-idx=\"7\" tabindex=\"0\" id=\"366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e_next\">Next</a></div></div>",
       "text/plain": "<JupyterRequire.display.FrozenOutput object>"
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "{ \n",
       "    setTimeout(() => {\n",
       "        const table_id = '366c39993a088da74fedb4dd632fa1e3b4f26469103fbf3e85b077929ff8335e';\n",
       "        const table    = $(`#${table_id}_wrapper`);\n",
       "        \n",
       "        let scrollHead = table.find('div.dataTables_scrollHead');\n",
       "        let scrollBody = table.find('div.dataTables_scrollBody');\n",
       "        \n",
       "        $(scrollBody).on(\n",
       "            'scroll',\n",
       "            (e) => {\n",
       "                scrollHead.scrollLeft(scrollBody.scrollLeft());\n",
       "            },\n",
       "        );\n",
       "    }, 200);\n",
       "     }"
      ],
      "text/plain": [
       "<JupyterRequire.display.SafeScript object>"
      ]
     },
     "metadata": {
      "finalized": true
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "    (function() {\n",
       "        const sample_size = Number(2000).toLocaleString();\n",
       "        const adjusted = Boolean('False' == 'True')\n",
       "\n",
       "        const total = Number(4489).toLocaleString();\n",
       "\n",
       "        element.append($('<p>').text(\n",
       "            `Sample size: ${sample_size} out of ${total} ${ adjusted ? \"(adjusted)\" : \"\" }`));\n",
       "    }());\n",
       "    "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GO ID</th>\n",
       "      <th>Pathway</th>\n",
       "      <th>P-value (Real)</th>\n",
       "      <th>Rank (Real)</th>\n",
       "      <th>Test statistic (Real)</th>\n",
       "      <th>Median p-value (simulated)</th>\n",
       "      <th>Median rank (simulated)</th>\n",
       "      <th>Mean test statistic (simulated)</th>\n",
       "      <th>Std deviation (simulated)</th>\n",
       "      <th>Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0000018</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>1919.0</td>\n",
       "      <td>0.441868</td>\n",
       "      <td>0.016700</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000027</th>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006503</td>\n",
       "      <td>328.5</td>\n",
       "      <td>0.548594</td>\n",
       "      <td>0.021737</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000028</th>\n",
       "      <td>GO:0000028</td>\n",
       "      <td>ribosomal small subunit assembly</td>\n",
       "      <td>0.005097</td>\n",
       "      <td>84.0</td>\n",
       "      <td>0.628994</td>\n",
       "      <td>0.007121</td>\n",
       "      <td>113.0</td>\n",
       "      <td>0.573867</td>\n",
       "      <td>0.056410</td>\n",
       "      <td>0.977254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000038</th>\n",
       "      <td>GO:0000038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009009</td>\n",
       "      <td>415.0</td>\n",
       "      <td>0.535535</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000041</th>\n",
       "      <td>GO:0000041</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>1167.0</td>\n",
       "      <td>0.449833</td>\n",
       "      <td>0.019520</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000045</th>\n",
       "      <td>GO:0000045</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007503</td>\n",
       "      <td>1877.5</td>\n",
       "      <td>0.415553</td>\n",
       "      <td>0.074887</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000049</th>\n",
       "      <td>GO:0000049</td>\n",
       "      <td>tRNA binding</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>308.0</td>\n",
       "      <td>0.525192</td>\n",
       "      <td>0.009491</td>\n",
       "      <td>1201.0</td>\n",
       "      <td>0.472127</td>\n",
       "      <td>0.004882</td>\n",
       "      <td>10.870066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000050</th>\n",
       "      <td>GO:0000050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002092</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.698310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000054</th>\n",
       "      <td>GO:0000054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010373</td>\n",
       "      <td>109.5</td>\n",
       "      <td>0.632219</td>\n",
       "      <td>0.004196</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000070</th>\n",
       "      <td>GO:0000070</td>\n",
       "      <td>mitotic sister chromatid segregation</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>341.0</td>\n",
       "      <td>0.515293</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>809.0</td>\n",
       "      <td>0.464635</td>\n",
       "      <td>0.043823</td>\n",
       "      <td>1.155955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000075</th>\n",
       "      <td>GO:0000075</td>\n",
       "      <td>cell cycle checkpoint</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>489.0</td>\n",
       "      <td>0.486637</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1041.5</td>\n",
       "      <td>0.443276</td>\n",
       "      <td>0.037489</td>\n",
       "      <td>1.156647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000077</th>\n",
       "      <td>GO:0000077</td>\n",
       "      <td>DNA damage checkpoint</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>472.0</td>\n",
       "      <td>0.487972</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1009.5</td>\n",
       "      <td>0.442027</td>\n",
       "      <td>0.039051</td>\n",
       "      <td>1.176552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000079</th>\n",
       "      <td>GO:0000079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003497</td>\n",
       "      <td>1302.0</td>\n",
       "      <td>0.446131</td>\n",
       "      <td>0.010134</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000082</th>\n",
       "      <td>GO:0000082</td>\n",
       "      <td>G1/S transition of mitotic cell cycle</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>763.0</td>\n",
       "      <td>0.439629</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1003.5</td>\n",
       "      <td>0.445661</td>\n",
       "      <td>0.039140</td>\n",
       "      <td>-0.154111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000083</th>\n",
       "      <td>GO:0000083</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006008</td>\n",
       "      <td>293.0</td>\n",
       "      <td>0.540150</td>\n",
       "      <td>0.002715</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000086</th>\n",
       "      <td>GO:0000086</td>\n",
       "      <td>G2/M transition of mitotic cell cycle</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>373.0</td>\n",
       "      <td>0.508493</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1095.0</td>\n",
       "      <td>0.442299</td>\n",
       "      <td>0.042893</td>\n",
       "      <td>1.543225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000096</th>\n",
       "      <td>GO:0000096</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004498</td>\n",
       "      <td>351.0</td>\n",
       "      <td>0.523622</td>\n",
       "      <td>0.016675</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000097</th>\n",
       "      <td>GO:0000097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009633</td>\n",
       "      <td>212.0</td>\n",
       "      <td>0.598447</td>\n",
       "      <td>0.036846</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000118</th>\n",
       "      <td>GO:0000118</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>573.0</td>\n",
       "      <td>0.496446</td>\n",
       "      <td>0.024355</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000123</th>\n",
       "      <td>GO:0000123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006993</td>\n",
       "      <td>1392.5</td>\n",
       "      <td>0.446779</td>\n",
       "      <td>0.015836</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000146</th>\n",
       "      <td>GO:0000146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007049</td>\n",
       "      <td>146.0</td>\n",
       "      <td>0.588601</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000149</th>\n",
       "      <td>GO:0000149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>0.432978</td>\n",
       "      <td>0.011591</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000151</th>\n",
       "      <td>GO:0000151</td>\n",
       "      <td>ubiquitin ligase complex</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>768.0</td>\n",
       "      <td>0.438137</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1727.0</td>\n",
       "      <td>0.407017</td>\n",
       "      <td>0.040695</td>\n",
       "      <td>0.764728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000152</th>\n",
       "      <td>GO:0000152</td>\n",
       "      <td>nuclear ubiquitin ligase complex</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>283.0</td>\n",
       "      <td>0.531832</td>\n",
       "      <td>0.003497</td>\n",
       "      <td>396.0</td>\n",
       "      <td>0.513999</td>\n",
       "      <td>0.009909</td>\n",
       "      <td>1.799736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000154</th>\n",
       "      <td>GO:0000154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008000</td>\n",
       "      <td>755.0</td>\n",
       "      <td>0.528346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000175</th>\n",
       "      <td>GO:0000175</td>\n",
       "      <td>3'-5'-exoribonuclease activity</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>212.0</td>\n",
       "      <td>0.560727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000177</th>\n",
       "      <td>GO:0000177</td>\n",
       "      <td>cytoplasmic exosome (RNase complex)</td>\n",
       "      <td>0.003096</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.661216</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.637138</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000183</th>\n",
       "      <td>GO:0000183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>271.0</td>\n",
       "      <td>0.538470</td>\n",
       "      <td>0.034442</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000184</th>\n",
       "      <td>GO:0000184</td>\n",
       "      <td>nuclear-transcribed mRNA catabolic process, no...</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>307.0</td>\n",
       "      <td>0.525601</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>487.0</td>\n",
       "      <td>0.497044</td>\n",
       "      <td>0.043644</td>\n",
       "      <td>0.654332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000187</th>\n",
       "      <td>GO:0000187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>1782.0</td>\n",
       "      <td>0.418994</td>\n",
       "      <td>0.044654</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000779</th>\n",
       "      <td>GO:2000779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>0.455877</td>\n",
       "      <td>0.014529</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000785</th>\n",
       "      <td>GO:2000785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007493</td>\n",
       "      <td>543.5</td>\n",
       "      <td>0.534986</td>\n",
       "      <td>0.040063</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000786</th>\n",
       "      <td>GO:2000786</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010449</td>\n",
       "      <td>103.0</td>\n",
       "      <td>0.655661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000816</th>\n",
       "      <td>GO:2000816</td>\n",
       "      <td>negative regulation of mitotic sister chromati...</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>104.0</td>\n",
       "      <td>0.614064</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>365.0</td>\n",
       "      <td>0.513636</td>\n",
       "      <td>0.009022</td>\n",
       "      <td>11.131195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001020</th>\n",
       "      <td>GO:2001020</td>\n",
       "      <td>regulation of response to DNA damage stimulus</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>866.0</td>\n",
       "      <td>0.407856</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1520.0</td>\n",
       "      <td>0.437003</td>\n",
       "      <td>0.016712</td>\n",
       "      <td>-1.744132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001021</th>\n",
       "      <td>GO:2001021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>1082.5</td>\n",
       "      <td>0.467728</td>\n",
       "      <td>0.017132</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001022</th>\n",
       "      <td>GO:2001022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>0.448394</td>\n",
       "      <td>0.013161</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001023</th>\n",
       "      <td>GO:2001023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>1203.0</td>\n",
       "      <td>0.443684</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001024</th>\n",
       "      <td>GO:2001024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005020</td>\n",
       "      <td>325.0</td>\n",
       "      <td>0.550256</td>\n",
       "      <td>0.012091</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001026</th>\n",
       "      <td>GO:2001026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006059</td>\n",
       "      <td>162.5</td>\n",
       "      <td>0.582120</td>\n",
       "      <td>0.029513</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001028</th>\n",
       "      <td>GO:2001028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004615</td>\n",
       "      <td>79.5</td>\n",
       "      <td>0.630664</td>\n",
       "      <td>0.008811</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001038</th>\n",
       "      <td>GO:2001038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011000</td>\n",
       "      <td>220.0</td>\n",
       "      <td>0.548057</td>\n",
       "      <td>0.021367</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001044</th>\n",
       "      <td>GO:2001044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005478</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.585139</td>\n",
       "      <td>0.090283</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001056</th>\n",
       "      <td>GO:2001056</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007992</td>\n",
       "      <td>2023.0</td>\n",
       "      <td>0.404573</td>\n",
       "      <td>0.008552</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001057</th>\n",
       "      <td>GO:2001057</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>910.0</td>\n",
       "      <td>0.460226</td>\n",
       "      <td>0.047645</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001135</th>\n",
       "      <td>GO:2001135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010638</td>\n",
       "      <td>44.0</td>\n",
       "      <td>0.666135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001169</th>\n",
       "      <td>GO:2001169</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>1639.0</td>\n",
       "      <td>0.450547</td>\n",
       "      <td>0.015852</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001198</th>\n",
       "      <td>GO:2001198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005198</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.645315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001233</th>\n",
       "      <td>GO:2001233</td>\n",
       "      <td>regulation of apoptotic signaling pathway</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>865.0</td>\n",
       "      <td>0.408466</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1157.0</td>\n",
       "      <td>0.435219</td>\n",
       "      <td>0.039549</td>\n",
       "      <td>-0.676443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001234</th>\n",
       "      <td>GO:2001234</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1095.0</td>\n",
       "      <td>0.443975</td>\n",
       "      <td>0.038904</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001235</th>\n",
       "      <td>GO:2001235</td>\n",
       "      <td>positive regulation of apoptotic signaling pat...</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>853.0</td>\n",
       "      <td>0.415543</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1141.5</td>\n",
       "      <td>0.444190</td>\n",
       "      <td>0.043013</td>\n",
       "      <td>-0.666014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001236</th>\n",
       "      <td>GO:2001236</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1163.0</td>\n",
       "      <td>0.447423</td>\n",
       "      <td>0.044880</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001237</th>\n",
       "      <td>GO:2001237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>1149.0</td>\n",
       "      <td>0.465523</td>\n",
       "      <td>0.027825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001238</th>\n",
       "      <td>GO:2001238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004496</td>\n",
       "      <td>726.5</td>\n",
       "      <td>0.510045</td>\n",
       "      <td>0.022396</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001239</th>\n",
       "      <td>GO:2001239</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>736.0</td>\n",
       "      <td>0.501766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001242</th>\n",
       "      <td>GO:2001242</td>\n",
       "      <td>regulation of intrinsic apoptotic signaling pa...</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>465.0</td>\n",
       "      <td>0.489879</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>941.0</td>\n",
       "      <td>0.454232</td>\n",
       "      <td>0.039344</td>\n",
       "      <td>0.906047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001243</th>\n",
       "      <td>GO:2001243</td>\n",
       "      <td>negative regulation of intrinsic apoptotic sig...</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>281.0</td>\n",
       "      <td>0.532725</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>608.0</td>\n",
       "      <td>0.471632</td>\n",
       "      <td>0.035523</td>\n",
       "      <td>1.719794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001244</th>\n",
       "      <td>GO:2001244</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>994.0</td>\n",
       "      <td>0.481203</td>\n",
       "      <td>0.024956</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001251</th>\n",
       "      <td>GO:2001251</td>\n",
       "      <td>negative regulation of chromosome organization</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>678.0</td>\n",
       "      <td>0.456322</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1136.0</td>\n",
       "      <td>0.438954</td>\n",
       "      <td>0.041697</td>\n",
       "      <td>0.416552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001252</th>\n",
       "      <td>GO:2001252</td>\n",
       "      <td>positive regulation of chromosome organization</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>421.0</td>\n",
       "      <td>0.499840</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1100.0</td>\n",
       "      <td>0.441648</td>\n",
       "      <td>0.048746</td>\n",
       "      <td>1.193795</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4489 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 GO ID                                            Pathway  \\\n",
       "GO:0000018  GO:0000018                                                NaN   \n",
       "GO:0000027  GO:0000027                                                NaN   \n",
       "GO:0000028  GO:0000028                   ribosomal small subunit assembly   \n",
       "GO:0000038  GO:0000038                                                NaN   \n",
       "GO:0000041  GO:0000041                                                NaN   \n",
       "GO:0000045  GO:0000045                                                NaN   \n",
       "GO:0000049  GO:0000049                                       tRNA binding   \n",
       "GO:0000050  GO:0000050                                                NaN   \n",
       "GO:0000054  GO:0000054                                                NaN   \n",
       "GO:0000070  GO:0000070               mitotic sister chromatid segregation   \n",
       "GO:0000075  GO:0000075                              cell cycle checkpoint   \n",
       "GO:0000077  GO:0000077                              DNA damage checkpoint   \n",
       "GO:0000079  GO:0000079                                                NaN   \n",
       "GO:0000082  GO:0000082              G1/S transition of mitotic cell cycle   \n",
       "GO:0000083  GO:0000083                                                NaN   \n",
       "GO:0000086  GO:0000086              G2/M transition of mitotic cell cycle   \n",
       "GO:0000096  GO:0000096                                                NaN   \n",
       "GO:0000097  GO:0000097                                                NaN   \n",
       "GO:0000118  GO:0000118                                                NaN   \n",
       "GO:0000123  GO:0000123                                                NaN   \n",
       "GO:0000146  GO:0000146                                                NaN   \n",
       "GO:0000149  GO:0000149                                                NaN   \n",
       "GO:0000151  GO:0000151                           ubiquitin ligase complex   \n",
       "GO:0000152  GO:0000152                   nuclear ubiquitin ligase complex   \n",
       "GO:0000154  GO:0000154                                                NaN   \n",
       "GO:0000175  GO:0000175                     3'-5'-exoribonuclease activity   \n",
       "GO:0000177  GO:0000177                cytoplasmic exosome (RNase complex)   \n",
       "GO:0000183  GO:0000183                                                NaN   \n",
       "GO:0000184  GO:0000184  nuclear-transcribed mRNA catabolic process, no...   \n",
       "GO:0000187  GO:0000187                                                NaN   \n",
       "...                ...                                                ...   \n",
       "GO:2000779  GO:2000779                                                NaN   \n",
       "GO:2000785  GO:2000785                                                NaN   \n",
       "GO:2000786  GO:2000786                                                NaN   \n",
       "GO:2000816  GO:2000816  negative regulation of mitotic sister chromati...   \n",
       "GO:2001020  GO:2001020      regulation of response to DNA damage stimulus   \n",
       "GO:2001021  GO:2001021                                                NaN   \n",
       "GO:2001022  GO:2001022                                                NaN   \n",
       "GO:2001023  GO:2001023                                                NaN   \n",
       "GO:2001024  GO:2001024                                                NaN   \n",
       "GO:2001026  GO:2001026                                                NaN   \n",
       "GO:2001028  GO:2001028                                                NaN   \n",
       "GO:2001038  GO:2001038                                                NaN   \n",
       "GO:2001044  GO:2001044                                                NaN   \n",
       "GO:2001056  GO:2001056                                                NaN   \n",
       "GO:2001057  GO:2001057                                                NaN   \n",
       "GO:2001135  GO:2001135                                                NaN   \n",
       "GO:2001169  GO:2001169                                                NaN   \n",
       "GO:2001198  GO:2001198                                                NaN   \n",
       "GO:2001233  GO:2001233          regulation of apoptotic signaling pathway   \n",
       "GO:2001234  GO:2001234                                                NaN   \n",
       "GO:2001235  GO:2001235  positive regulation of apoptotic signaling pat...   \n",
       "GO:2001236  GO:2001236                                                NaN   \n",
       "GO:2001237  GO:2001237                                                NaN   \n",
       "GO:2001238  GO:2001238                                                NaN   \n",
       "GO:2001239  GO:2001239                                                NaN   \n",
       "GO:2001242  GO:2001242  regulation of intrinsic apoptotic signaling pa...   \n",
       "GO:2001243  GO:2001243  negative regulation of intrinsic apoptotic sig...   \n",
       "GO:2001244  GO:2001244                                                NaN   \n",
       "GO:2001251  GO:2001251     negative regulation of chromosome organization   \n",
       "GO:2001252  GO:2001252     positive regulation of chromosome organization   \n",
       "\n",
       "            P-value (Real)  Rank (Real)  Test statistic (Real)  \\\n",
       "GO:0000018             NaN          NaN                    NaN   \n",
       "GO:0000027             NaN          NaN                    NaN   \n",
       "GO:0000028        0.005097         84.0               0.628994   \n",
       "GO:0000038             NaN          NaN                    NaN   \n",
       "GO:0000041             NaN          NaN                    NaN   \n",
       "GO:0000045             NaN          NaN                    NaN   \n",
       "GO:0000049        0.002997        308.0               0.525192   \n",
       "GO:0000050             NaN          NaN                    NaN   \n",
       "GO:0000054             NaN          NaN                    NaN   \n",
       "GO:0000070        0.000999        341.0               0.515293   \n",
       "GO:0000075        0.000999        489.0               0.486637   \n",
       "GO:0000077        0.000999        472.0               0.487972   \n",
       "GO:0000079             NaN          NaN                    NaN   \n",
       "GO:0000082        0.000999        763.0               0.439629   \n",
       "GO:0000083             NaN          NaN                    NaN   \n",
       "GO:0000086        0.000999        373.0               0.508493   \n",
       "GO:0000096             NaN          NaN                    NaN   \n",
       "GO:0000097             NaN          NaN                    NaN   \n",
       "GO:0000118             NaN          NaN                    NaN   \n",
       "GO:0000123             NaN          NaN                    NaN   \n",
       "GO:0000146             NaN          NaN                    NaN   \n",
       "GO:0000149             NaN          NaN                    NaN   \n",
       "GO:0000151        0.000999        768.0               0.438137   \n",
       "GO:0000152        0.004995        283.0               0.531832   \n",
       "GO:0000154             NaN          NaN                    NaN   \n",
       "GO:0000175        0.004995        212.0               0.560727   \n",
       "GO:0000177        0.003096         50.0               0.661216   \n",
       "GO:0000183             NaN          NaN                    NaN   \n",
       "GO:0000184        0.000999        307.0               0.525601   \n",
       "GO:0000187             NaN          NaN                    NaN   \n",
       "...                    ...          ...                    ...   \n",
       "GO:2000779             NaN          NaN                    NaN   \n",
       "GO:2000785             NaN          NaN                    NaN   \n",
       "GO:2000786             NaN          NaN                    NaN   \n",
       "GO:2000816        0.001000        104.0               0.614064   \n",
       "GO:2001020        0.003996        866.0               0.407856   \n",
       "GO:2001021             NaN          NaN                    NaN   \n",
       "GO:2001022             NaN          NaN                    NaN   \n",
       "GO:2001023             NaN          NaN                    NaN   \n",
       "GO:2001024             NaN          NaN                    NaN   \n",
       "GO:2001026             NaN          NaN                    NaN   \n",
       "GO:2001028             NaN          NaN                    NaN   \n",
       "GO:2001038             NaN          NaN                    NaN   \n",
       "GO:2001044             NaN          NaN                    NaN   \n",
       "GO:2001056             NaN          NaN                    NaN   \n",
       "GO:2001057             NaN          NaN                    NaN   \n",
       "GO:2001135             NaN          NaN                    NaN   \n",
       "GO:2001169             NaN          NaN                    NaN   \n",
       "GO:2001198             NaN          NaN                    NaN   \n",
       "GO:2001233        0.000999        865.0               0.408466   \n",
       "GO:2001234             NaN          NaN                    NaN   \n",
       "GO:2001235        0.003996        853.0               0.415543   \n",
       "GO:2001236             NaN          NaN                    NaN   \n",
       "GO:2001237             NaN          NaN                    NaN   \n",
       "GO:2001238             NaN          NaN                    NaN   \n",
       "GO:2001239             NaN          NaN                    NaN   \n",
       "GO:2001242        0.000999        465.0               0.489879   \n",
       "GO:2001243        0.000999        281.0               0.532725   \n",
       "GO:2001244             NaN          NaN                    NaN   \n",
       "GO:2001251        0.000999        678.0               0.456322   \n",
       "GO:2001252        0.000999        421.0               0.499840   \n",
       "\n",
       "            Median p-value (simulated)  Median rank (simulated)  \\\n",
       "GO:0000018                    0.005495                   1919.0   \n",
       "GO:0000027                    0.006503                    328.5   \n",
       "GO:0000028                    0.007121                    113.0   \n",
       "GO:0000038                    0.009009                    415.0   \n",
       "GO:0000041                    0.001998                   1167.0   \n",
       "GO:0000045                    0.007503                   1877.5   \n",
       "GO:0000049                    0.009491                   1201.0   \n",
       "GO:0000050                    0.002092                     28.0   \n",
       "GO:0000054                    0.010373                    109.5   \n",
       "GO:0000070                    0.000999                    809.0   \n",
       "GO:0000075                    0.000999                   1041.5   \n",
       "GO:0000077                    0.000999                   1009.5   \n",
       "GO:0000079                    0.003497                   1302.0   \n",
       "GO:0000082                    0.000999                   1003.5   \n",
       "GO:0000083                    0.006008                    293.0   \n",
       "GO:0000086                    0.000999                   1095.0   \n",
       "GO:0000096                    0.004498                    351.0   \n",
       "GO:0000097                    0.009633                    212.0   \n",
       "GO:0000118                    0.001998                    573.0   \n",
       "GO:0000123                    0.006993                   1392.5   \n",
       "GO:0000146                    0.007049                    146.0   \n",
       "GO:0000149                    0.006494                   1979.0   \n",
       "GO:0000151                    0.000999                   1727.0   \n",
       "GO:0000152                    0.003497                    396.0   \n",
       "GO:0000154                    0.008000                    755.0   \n",
       "GO:0000175                         NaN                      NaN   \n",
       "GO:0000177                    0.009346                     71.0   \n",
       "GO:0000183                    0.003996                    271.0   \n",
       "GO:0000184                    0.000999                    487.0   \n",
       "GO:0000187                    0.002997                   1782.0   \n",
       "...                                ...                      ...   \n",
       "GO:2000779                    0.004995                   1355.0   \n",
       "GO:2000785                    0.007493                    543.5   \n",
       "GO:2000786                    0.010449                    103.0   \n",
       "GO:2000816                    0.004995                    365.0   \n",
       "GO:2001020                    0.000999                   1520.0   \n",
       "GO:2001021                    0.001998                   1082.5   \n",
       "GO:2001022                    0.004995                   1355.0   \n",
       "GO:2001023                    0.004995                   1203.0   \n",
       "GO:2001024                    0.005020                    325.0   \n",
       "GO:2001026                    0.006059                    162.5   \n",
       "GO:2001028                    0.004615                     79.5   \n",
       "GO:2001038                    0.011000                    220.0   \n",
       "GO:2001044                    0.005478                     90.0   \n",
       "GO:2001056                    0.007992                   2023.0   \n",
       "GO:2001057                    0.003996                    910.0   \n",
       "GO:2001135                    0.010638                     44.0   \n",
       "GO:2001169                    0.003996                   1639.0   \n",
       "GO:2001198                    0.005198                     48.0   \n",
       "GO:2001233                    0.000999                   1157.0   \n",
       "GO:2001234                    0.000999                   1095.0   \n",
       "GO:2001235                    0.000999                   1141.5   \n",
       "GO:2001236                    0.000999                   1163.0   \n",
       "GO:2001237                    0.002997                   1149.0   \n",
       "GO:2001238                    0.004496                    726.5   \n",
       "GO:2001239                    0.009990                    736.0   \n",
       "GO:2001242                    0.000999                    941.0   \n",
       "GO:2001243                    0.000999                    608.0   \n",
       "GO:2001244                    0.004995                    994.0   \n",
       "GO:2001251                    0.000999                   1136.0   \n",
       "GO:2001252                    0.000999                   1100.0   \n",
       "\n",
       "            Mean test statistic (simulated)  Std deviation (simulated)  \\\n",
       "GO:0000018                         0.441868                   0.016700   \n",
       "GO:0000027                         0.548594                   0.021737   \n",
       "GO:0000028                         0.573867                   0.056410   \n",
       "GO:0000038                         0.535535                        NaN   \n",
       "GO:0000041                         0.449833                   0.019520   \n",
       "GO:0000045                         0.415553                   0.074887   \n",
       "GO:0000049                         0.472127                   0.004882   \n",
       "GO:0000050                         0.698310                        NaN   \n",
       "GO:0000054                         0.632219                   0.004196   \n",
       "GO:0000070                         0.464635                   0.043823   \n",
       "GO:0000075                         0.443276                   0.037489   \n",
       "GO:0000077                         0.442027                   0.039051   \n",
       "GO:0000079                         0.446131                   0.010134   \n",
       "GO:0000082                         0.445661                   0.039140   \n",
       "GO:0000083                         0.540150                   0.002715   \n",
       "GO:0000086                         0.442299                   0.042893   \n",
       "GO:0000096                         0.523622                   0.016675   \n",
       "GO:0000097                         0.598447                   0.036846   \n",
       "GO:0000118                         0.496446                   0.024355   \n",
       "GO:0000123                         0.446779                   0.015836   \n",
       "GO:0000146                         0.588601                        NaN   \n",
       "GO:0000149                         0.432978                   0.011591   \n",
       "GO:0000151                         0.407017                   0.040695   \n",
       "GO:0000152                         0.513999                   0.009909   \n",
       "GO:0000154                         0.528346                        NaN   \n",
       "GO:0000175                              NaN                        NaN   \n",
       "GO:0000177                         0.637138                        NaN   \n",
       "GO:0000183                         0.538470                   0.034442   \n",
       "GO:0000184                         0.497044                   0.043644   \n",
       "GO:0000187                         0.418994                   0.044654   \n",
       "...                                     ...                        ...   \n",
       "GO:2000779                         0.455877                   0.014529   \n",
       "GO:2000785                         0.534986                   0.040063   \n",
       "GO:2000786                         0.655661                        NaN   \n",
       "GO:2000816                         0.513636                   0.009022   \n",
       "GO:2001020                         0.437003                   0.016712   \n",
       "GO:2001021                         0.467728                   0.017132   \n",
       "GO:2001022                         0.448394                   0.013161   \n",
       "GO:2001023                         0.443684                        NaN   \n",
       "GO:2001024                         0.550256                   0.012091   \n",
       "GO:2001026                         0.582120                   0.029513   \n",
       "GO:2001028                         0.630664                   0.008811   \n",
       "GO:2001038                         0.548057                   0.021367   \n",
       "GO:2001044                         0.585139                   0.090283   \n",
       "GO:2001056                         0.404573                   0.008552   \n",
       "GO:2001057                         0.460226                   0.047645   \n",
       "GO:2001135                         0.666135                        NaN   \n",
       "GO:2001169                         0.450547                   0.015852   \n",
       "GO:2001198                         0.645315                        NaN   \n",
       "GO:2001233                         0.435219                   0.039549   \n",
       "GO:2001234                         0.443975                   0.038904   \n",
       "GO:2001235                         0.444190                   0.043013   \n",
       "GO:2001236                         0.447423                   0.044880   \n",
       "GO:2001237                         0.465523                   0.027825   \n",
       "GO:2001238                         0.510045                   0.022396   \n",
       "GO:2001239                         0.501766                        NaN   \n",
       "GO:2001242                         0.454232                   0.039344   \n",
       "GO:2001243                         0.471632                   0.035523   \n",
       "GO:2001244                         0.481203                   0.024956   \n",
       "GO:2001251                         0.438954                   0.041697   \n",
       "GO:2001252                         0.441648                   0.048746   \n",
       "\n",
       "              Z score  \n",
       "GO:0000018        NaN  \n",
       "GO:0000027        NaN  \n",
       "GO:0000028   0.977254  \n",
       "GO:0000038        NaN  \n",
       "GO:0000041        NaN  \n",
       "GO:0000045        NaN  \n",
       "GO:0000049  10.870066  \n",
       "GO:0000050        NaN  \n",
       "GO:0000054        NaN  \n",
       "GO:0000070   1.155955  \n",
       "GO:0000075   1.156647  \n",
       "GO:0000077   1.176552  \n",
       "GO:0000079        NaN  \n",
       "GO:0000082  -0.154111  \n",
       "GO:0000083        NaN  \n",
       "GO:0000086   1.543225  \n",
       "GO:0000096        NaN  \n",
       "GO:0000097        NaN  \n",
       "GO:0000118        NaN  \n",
       "GO:0000123        NaN  \n",
       "GO:0000146        NaN  \n",
       "GO:0000149        NaN  \n",
       "GO:0000151   0.764728  \n",
       "GO:0000152   1.799736  \n",
       "GO:0000154        NaN  \n",
       "GO:0000175        NaN  \n",
       "GO:0000177        NaN  \n",
       "GO:0000183        NaN  \n",
       "GO:0000184   0.654332  \n",
       "GO:0000187        NaN  \n",
       "...               ...  \n",
       "GO:2000779        NaN  \n",
       "GO:2000785        NaN  \n",
       "GO:2000786        NaN  \n",
       "GO:2000816  11.131195  \n",
       "GO:2001020  -1.744132  \n",
       "GO:2001021        NaN  \n",
       "GO:2001022        NaN  \n",
       "GO:2001023        NaN  \n",
       "GO:2001024        NaN  \n",
       "GO:2001026        NaN  \n",
       "GO:2001028        NaN  \n",
       "GO:2001038        NaN  \n",
       "GO:2001044        NaN  \n",
       "GO:2001056        NaN  \n",
       "GO:2001057        NaN  \n",
       "GO:2001135        NaN  \n",
       "GO:2001169        NaN  \n",
       "GO:2001198        NaN  \n",
       "GO:2001233  -0.676443  \n",
       "GO:2001234        NaN  \n",
       "GO:2001235  -0.666014  \n",
       "GO:2001236        NaN  \n",
       "GO:2001237        NaN  \n",
       "GO:2001238        NaN  \n",
       "GO:2001239        NaN  \n",
       "GO:2001242   0.906047  \n",
       "GO:2001243   1.719794  \n",
       "GO:2001244        NaN  \n",
       "GO:2001251   0.416552  \n",
       "GO:2001252   1.193795  \n",
       "\n",
       "[4489 rows x 10 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary = pd.DataFrame(data={'GO ID': template_simulated_enriched_pathways_stats.index,\n",
    "                             'Pathway': template_simulated_enriched_pathways_stats['Description'],\n",
    "                             'P-value (Real)': template_simulated_enriched_pathways_stats['pvalue'],\n",
    "                             'Rank (Real)': template_simulated_enriched_pathways_stats['ranking'],\n",
    "                             'Test statistic (Real)': template_enriched_pathways[col_to_rank],\n",
    "                             'Median p-value (simulated)': median_pval_simulated ,\n",
    "                             'Median rank (simulated)': median_rank_simulated ,\n",
    "                             'Mean test statistic (simulated)': mean_test_simulated ,\n",
    "                             'Std deviation (simulated)': std_test_simulated\n",
    "                            }\n",
    "                      )\n",
    "summary['Z score'] = (summary['Test statistic (Real)'] - summary['Mean test statistic (simulated)'])/summary['Std deviation (simulated)']\n",
    "summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save file\n",
    "summary_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"summary_table.tsv\")\n",
    "\n",
    "summary.to_csv(summary_file, float_format='%.5f', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are a few immune pathways that are template-enriched but there are others that look to be involved in cellular maintenancem, which I wouldn't expect to be specific to these experiment.\n",
    "\n",
    "For GSEA I am currently using the p-values from the DE analysis to rank genes, should I try using avg gene expression instead..? Maybe p-values don't cluster genes as much..?\n",
    "\n",
    "Note: this notebook takes ~1hr to run\n",
    "\n",
    "**Check:**\n",
    "* Are there template-specific pathways (i.e. pathways specific to Dexamethasone treatment) that we can use as a positive?\n",
    "* Are there agnostic pathways that we can use as a negative control?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:ranked_pathways] *",
   "language": "python",
   "name": "conda-env-ranked_pathways-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "require": {
   "paths": {
    "buttons.colvis": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.colVis.min",
    "buttons.flash": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.flash.min",
    "buttons.html5": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.html5.min",
    "buttons.print": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.print.min",
    "chartjs": "https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart",
    "d3": "https://d3js.org/d3.v5.min",
    "d3-array": "https://d3js.org/d3-array.v2.min",
    "datatables.net": "https://cdn.datatables.net/1.10.18/js/jquery.dataTables",
    "datatables.net-buttons": "https://cdn.datatables.net/buttons/1.5.6/js/dataTables.buttons.min",
    "datatables.responsive": "https://cdn.datatables.net/responsive/2.2.2/js/dataTables.responsive.min",
    "datatables.scroller": "https://cdn.datatables.net/scroller/2.0.0/js/dataTables.scroller.min",
    "datatables.select": "https://cdn.datatables.net/select/1.3.0/js/dataTables.select.min",
    "jszip": "https://cdnjs.cloudflare.com/ajax/libs/jszip/2.5.0/jszip.min",
    "moment": "https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.8.0/moment",
    "pdfmake": "https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/pdfmake.min",
    "vfsfonts": "https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/vfs_fonts"
   },
   "shim": {
    "buttons.colvis": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.flash": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.html5": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.print": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "chartjs": {
     "deps": [
      "moment"
     ]
    },
    "datatables.net": {
     "exports": "$.fn.dataTable"
    },
    "datatables.net-buttons": {
     "deps": [
      "datatables.net"
     ]
    },
    "pdfmake": {
     "deps": [
      "datatables.net"
     ]
    },
    "vfsfonts": {
     "deps": [
      "datatables.net"
     ]
    }
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
