{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Statistical analysis\n",
    "This notebook performs differential expression analysis using the real template experiment and simulated experiments, as a null set. Then the set of differentially expressed genes (DEGs) obtained from this analysis are used to perform gene set enrichment analysis (GSEA) to identify pathways enriched in these set of DEGs.\n",
    "\n",
    "*Note:* To run datatables, need to refresh the window and then run all cells (DO NOT restart and run all, only run all works)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: \n",
      "examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.\n",
      "  \"found relative to the 'datapath' directory.\".format(key))\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%load_ext rpy2.ipython\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import seaborn as sns\n",
    "import rpy2.robjects\n",
    "from rpy2.robjects.packages import importr\n",
    "from rpy2.robjects import pandas2ri\n",
    "pandas2ri.activate()\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "#from jupyter_datatables import init_datatables_mode\n",
    "\n",
    "\n",
    "def fxn():\n",
    "    warnings.warn(\"deprecated\", DeprecationWarning)\n",
    "\n",
    "with warnings.catch_warnings():\n",
    "    warnings.simplefilter(\"ignore\")\n",
    "    fxn()\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import utils\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in config variables\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../\"))\n",
    "\n",
    "config_file = os.path.abspath(os.path.join(base_dir,\n",
    "                                           \"Rank_pathways\",\n",
    "                                           \"init_config.tsv\"))\n",
    "params = utils.read_config(config_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# User selected experiment id\n",
    "project_id = \"SRP000762\"\n",
    "rerun_template = True\n",
    "rerun_simulated = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load params\n",
    "local_dir = params[\"local_dir\"]\n",
    "dataset_name = params['dataset_name']\n",
    "num_runs=25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load real template experiment\n",
    "template_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"recount2_template_data.tsv\")\n",
    "\n",
    "# Load metadata file with grouping assignments for samples\n",
    "metadata_file = os.path.join(\n",
    "    base_dir,\n",
    "    \"Rank_pathways\",\n",
    "    \"data\",\n",
    "    \"metadata\",\n",
    "    \"SRP000762_groups.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Select 59\n",
    "# Run one time\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(\"limma\")\n",
    "#BiocManager::install('EnhancedVolcano')\n",
    "#devtools::install_github('kevinblighe/EnhancedVolcano')\n",
    "#BiocManager::install('clusterProfiler')\n",
    "#BiocManager::install(\"org.Hs.eg.db\")\n",
    "#BiocManager::install(\"biomaRt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library('limma')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get differentially expressed genes from template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] 60\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i metadata_file -i project_id -i template_data_file -i local_dir -i rerun_template\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "out_file <- paste(local_dir,\n",
    "                  \"DE_stats/DE_stats_template_data_\",\n",
    "                  project_id,\n",
    "                  \"_real.txt\",\n",
    "                  sep=\"\")\n",
    "\n",
    "if (rerun_template){\n",
    "    get_DE_stats(metadata_file,\n",
    "                 project_id, \n",
    "                 template_data_file,\n",
    "                 \"template\",\n",
    "                 local_dir,\n",
    "                 \"real\")\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check signal strength"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load association statistics for template experiment\n",
    "template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_real.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000067082.14</th>\n",
       "      <td>19.385246</td>\n",
       "      <td>18.673139</td>\n",
       "      <td>14.055682</td>\n",
       "      <td>2.342010e-12</td>\n",
       "      <td>1.359233e-07</td>\n",
       "      <td>9.739709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000177606.6</th>\n",
       "      <td>-8.236487</td>\n",
       "      <td>9.185753</td>\n",
       "      <td>-13.126898</td>\n",
       "      <td>8.859555e-12</td>\n",
       "      <td>2.570910e-07</td>\n",
       "      <td>9.292456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000114315.3</th>\n",
       "      <td>-6.285228</td>\n",
       "      <td>6.042663</td>\n",
       "      <td>-12.840532</td>\n",
       "      <td>1.355507e-11</td>\n",
       "      <td>2.622318e-07</td>\n",
       "      <td>9.141666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185950.8</th>\n",
       "      <td>8.790781</td>\n",
       "      <td>17.705828</td>\n",
       "      <td>12.594113</td>\n",
       "      <td>1.966140e-11</td>\n",
       "      <td>2.852721e-07</td>\n",
       "      <td>9.006626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000095752.6</th>\n",
       "      <td>-7.505790</td>\n",
       "      <td>7.299177</td>\n",
       "      <td>-12.204218</td>\n",
       "      <td>3.582671e-11</td>\n",
       "      <td>4.158550e-07</td>\n",
       "      <td>8.782431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        logFC    AveExpr          t       P.Value  \\\n",
       "ENSG00000067082.14  19.385246  18.673139  14.055682  2.342010e-12   \n",
       "ENSG00000177606.6   -8.236487   9.185753 -13.126898  8.859555e-12   \n",
       "ENSG00000114315.3   -6.285228   6.042663 -12.840532  1.355507e-11   \n",
       "ENSG00000185950.8    8.790781  17.705828  12.594113  1.966140e-11   \n",
       "ENSG00000095752.6   -7.505790   7.299177 -12.204218  3.582671e-11   \n",
       "\n",
       "                       adj.P.Val         B  \n",
       "ENSG00000067082.14  1.359233e-07  9.739709  \n",
       "ENSG00000177606.6   2.570910e-07  9.292456  \n",
       "ENSG00000114315.3   2.622318e-07  9.141666  \n",
       "ENSG00000185950.8   2.852721e-07  9.006626  \n",
       "ENSG00000095752.6   4.158550e-07  8.782431  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "template_DE_stats = pd.read_csv(\n",
    "    template_DE_stats_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "\n",
    "template_DEGs = template_DE_stats[template_DE_stats['adj.P.Val']<0.05]\n",
    "print(template_DEGs.shape)\n",
    "template_DEGs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R\n",
    "#library(EnhancedVolcano)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R -i project_id -i template_DE_stats_file -i local_dir\n",
    "\n",
    "#source('../functions/DE_analysis.R')\n",
    "\n",
    "#create_volcano(template_DE_stats_file,\n",
    "#               project_id,\n",
    "#               \"adj.P.Val\",\n",
    "#               local_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get differentially expressed genes from each simulated experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i metadata_file -i project_id -i base_dir -i local_dir -i num_runs -i rerun_simulated -o num_sign_DEGs_simulated\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "num_sign_DEGs_simulated <- c()\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_data_file <- paste(local_dir, \n",
    "                                 \"pseudo_experiment/selected_simulated_data_\",\n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    out_file <- paste(local_dir, \n",
    "                      \"DE_stats/DE_stats_simulated_data_\",\n",
    "                      project_id,\n",
    "                      \"_\",\n",
    "                      i,\n",
    "                      \".txt\", \n",
    "                      sep=\"\")\n",
    "    \n",
    "    if (rerun_simulated){\n",
    "        run_output <- get_DE_stats(metadata_file,\n",
    "                                   project_id, \n",
    "                                   simulated_data_file,\n",
    "                                   \"simulated\",\n",
    "                                   local_dir,\n",
    "                                   i)\n",
    "        num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, run_output)\n",
    "    } else {\n",
    "        # Read in DE stats data\n",
    "        DE_stats_data <- as.data.frame(read.table(out_file, sep=\"\\t\", header=TRUE, row.names=1))\n",
    "        \n",
    "        # Get number of genes that exceed threshold\n",
    "        threshold <- 0.05\n",
    "        sign_DEGs <- DE_stats_data[DE_stats_data[,'adj.P.Val']<threshold,]\n",
    "        \n",
    "        num_sign_DEGs <- nrow(sign_DEGs)\n",
    "        \n",
    "        num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, num_sign_DEGs)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/scipy/stats/stats.py:1713: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.\n",
      "  return np.add.reduce(sorted[indexer] * weights, axis=axis) / sumval\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7fccc7634290>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvqOYd8AAADCJJREFUeJzt3F2MXHUdxvHn0YoXglrspjYIrlFigkYLbqrG94AGvQCMRkXEGknWREkw4gXRC43c4AvghcZYhVANvoKEJtYXrBhCAo1bbbCl0aIBLZZ2EaMYo6bweLGnZiWzzJmZMzPdX7+fZDMzZ87u+f3b9LvDYeY4iQAAq99Tpj0AAKAbBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBFrJnmwdevWZXZ2dpKHBIBVb9euXQ8nmem330SDPjs7q4WFhUkeEgBWPdsPtNmPUy4AUARBB4AiCDoAFEHQAaAIgg4ARfQNuu1Tbd9u+17be21f1mz/tO0Hbe9uvt42/nEBACtp87bFI5IuT/Ir2ydJ2mX7tua5a5N8YXzjAQDa6hv0JAclHWzuP2p7n6RTxj0YAGAwA51Dtz0r6UxJO5tNl9q+x/b1ttd2PBsAYACtPylq+0RJN0v6aJK/2/6KpCslpbm9WtIHe3zfvKR5STrttNO6mBmFfWvnH6c9wsS995X8u0A3Wr1Ct/00LcX8xiQ/kKQkh5I8luRxSV+TtKnX9ybZkmQuydzMTN9LEQAAhtTmXS6WdJ2kfUmuWbZ9w7Ld3i5pT/fjAQDaanPK5TWSLpb0G9u7m22fkHSh7Y1aOuVyv6QPjWVCAEArbd7lcqck93hqe/fjAACGxSdFAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAU0Tfotk+1fbvte23vtX1Zs/1k27fZ3t/crh3/uACAlbR5hX5E0uVJzpD0KkkfsX2GpCsk7UhyuqQdzWMAwJT0DXqSg0l+1dx/VNI+SadIOl/S1ma3rZIuGNeQAID+BjqHbntW0pmSdkpan+Rg89RDktZ3OhkAYCCtg277REk3S/pokr8vfy5JJGWF75u3vWB7YXFxcaRhAQAraxV020/TUsxvTPKDZvMh2xua5zdIOtzre5NsSTKXZG5mZqaLmQEAPbR5l4slXSdpX5Jrlj21TdLm5v5mSbd2Px4AoK01LfZ5jaSLJf3G9u5m2yckXSXpe7YvkfSApHeNZ0QAQBt9g57kTkle4emzux0HADAsPikKAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARfQNuu3rbR+2vWfZtk/bftD27ubrbeMdEwDQT5tX6DdIOrfH9muTbGy+tnc7FgBgUH2DnuQOSY9MYBYAwAhGOYd+qe17mlMyazubCAAwlGGD/hVJL5S0UdJBSVevtKPtedsLthcWFxeHPBwAoJ+hgp7kUJLHkjwu6WuSNj3JvluSzCWZm5mZGXZOAEAfQwXd9oZlD98uac9K+wIAJmNNvx1sf1vSGyWts31A0qckvdH2RkmRdL+kD41xRgBAC32DnuTCHpuvG8MsAIAR8ElRACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBF9A267ettH7a9Z9m2k23fZnt/c7t2vGMCAPpp8wr9BknnPmHbFZJ2JDld0o7mMQBgivoGPckdkh55wubzJW1t7m+VdEHHcwEABjTsOfT1SQ429x+StL6jeQAAQxr5f4omiaSs9LztedsLthcWFxdHPRwAYAXDBv2Q7Q2S1NweXmnHJFuSzCWZm5mZGfJwAIB+hg36Nkmbm/ubJd3azTgAgGG1edvityXdJenFtg/YvkTSVZLebHu/pHOaxwCAKVrTb4ckF67w1NkdzwIAGAGfFAWAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4Ai1ozyzbbvl/SopMckHUky18VQAIDBjRT0xpuSPNzBzwEAjIBTLgBQxKhBj6Sf2t5le77XDrbnbS/YXlhcXBzxcACAlYwa9NcmOUvSWyV9xPbrn7hDki1J5pLMzczMjHg4AMBKRgp6kgeb28OSbpG0qYuhAACDGzrotp9h+6Sj9yW9RdKergYDAAxmlHe5rJd0i+2jP+dbSX7cyVQAgIENHfQkf5D08g5nAQCMgLctAkARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAoYqSg2z7X9m9t32f7iq6GAgAMbuig236qpC9LequkMyRdaPuMrgYDAAxmlFfomyTdl+QPSf4j6TuSzu9mLADAoEYJ+imS/rTs8YFmGwBgCtaM+wC25yXNNw//Yfu34z7mGKyT9PC0h5ig42290hTXfNE0DrqEv+fV4/ltdhol6A9KOnXZ4+c12/5Pki2StoxwnKmzvZBkbtpzTMrxtl6JNR8vqq95lFMuv5R0uu0X2D5B0nskbetmLADAoIZ+hZ7kiO1LJf1E0lMlXZ9kb2eTAQAGMtI59CTbJW3vaJZj2ao+ZTSE4229Ems+XpRes5NMewYAQAf46D8AFEHQe7B9su3bbO9vbtc+yb7PtH3A9pcmOWOX2qzX9kbbd9nea/se2++exqyj6ne5CttPt/3d5vmdtmcnP2W3Wqz5Y7bvbf5ed9hu9Ra5Y1nby5LYfoft2C7xzheC3tsVknYkOV3SjubxSq6UdMdEphqfNuv9p6T3J3mJpHMlfdH2syc448haXq7iEkl/TfIiSddK+uxkp+xWyzX/WtJckpdJuknS5yY7ZbfaXpbE9kmSLpO0c7ITjg9B7+18SVub+1slXdBrJ9uvkLRe0k8nNNe49F1vkt8l2d/c/7Okw5JmJjZhN9pcrmL5n8VNks627QnO2LW+a05ye5J/Ng/v1tJnSlaztpcluVJLv7D/Ncnhxomg97Y+ycHm/kNaivb/sf0USVdL+vgkBxuTvutdzvYmSSdI+v24B+tYm8tV/G+fJEck/U3ScyYy3XgMeomOSyT9aKwTjV/fNds+S9KpSX44ycHGbewf/T9W2f6ZpOf2eOqTyx8kie1ebwX6sKTtSQ6shhdwHaz36M/ZIOmbkjYnebzbKTFNtt8naU7SG6Y9yzg1L8aukfSBKY/SueM26EnOWek524dsb0hysAnY4R67vVrS62x/WNKJkk6w/Y8kx+R14TtYr2w/U9IPJX0yyd1jGnWc2lyu4ug+B2yvkfQsSX+ZzHhj0eoSHbbP0dIv9zck+feEZhuXfms+SdJLJf2ieTH2XEnbbJ+XZGFiU44Bp1x62yZpc3N/s6Rbn7hDkouSnJZkVkunXb5xrMa8hb7rbS7vcIuW1nnTBGfrUpvLVSz/s3inpJ9ndX9Yo++abZ8p6auSzkvS85f5KvOka07ytyTrksw2/37v1tLaV3XMJYK+kqskvdn2fknnNI9le87216c62Xi0We+7JL1e0gds726+Nk5n3OE058SPXq5in6TvJdlr+zO2z2t2u07Sc2zfJ+ljevJ3OB3zWq7581r6r8zvN3+vq/qaTC3XXBKfFAWAIniFDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgiP8C9ML+LndAfc4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Plot distribution of differentially expressed genes for simulated experiments\n",
    "sns.distplot(num_sign_DEGs_simulated,\n",
    "            kde=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Observation:** All simulated experiments found 0 DEGs using adjusted p-value cutoff of <5%"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "As a check, we compared the number of DEGs identified here versus what was reported in the [Reddy et. al. publication](https://www.ncbi.nlm.nih.gov//pubmed/19801529), which found:\n",
    "* 234 genes with a significant (FDR < 5%) change in expression in response to DEX treatment. \n",
    "* After removing pseudogenes (listed in Supplemental Table S3), 209 differentially expressed genes remained \n",
    "* Of the DEX-responsive genes, more showed increases in transcript levels 123 (59%) than showed decreases 86 (41%), and the up-regulation was slightly but significantly stronger than the down-regulation\n",
    "\n",
    "By comparison:\n",
    "* Our study found 60 DEGs instead of 234. \n",
    "* Spot checking the genes identified with their list of DEX-responsive genes (Supplementary Dataset 2), we found the same genes and FC direction was consistent though magnitudes of theirs was lower compared to ours. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get statistics for differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_to_rank = 'logFC'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000264125.1</th>\n",
       "      <td>1678.911403</td>\n",
       "      <td>808.151173</td>\n",
       "      <td>1.059870</td>\n",
       "      <td>0.300891</td>\n",
       "      <td>0.531495</td>\n",
       "      <td>-4.984019</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229276.1</th>\n",
       "      <td>1402.821520</td>\n",
       "      <td>671.971224</td>\n",
       "      <td>1.063376</td>\n",
       "      <td>0.299335</td>\n",
       "      <td>0.531357</td>\n",
       "      <td>-4.981014</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000210082.2</th>\n",
       "      <td>821.464218</td>\n",
       "      <td>10147.594571</td>\n",
       "      <td>0.547916</td>\n",
       "      <td>0.589357</td>\n",
       "      <td>0.724887</td>\n",
       "      <td>-5.323706</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000200732.1</th>\n",
       "      <td>777.003071</td>\n",
       "      <td>374.050328</td>\n",
       "      <td>1.060896</td>\n",
       "      <td>0.300435</td>\n",
       "      <td>0.531426</td>\n",
       "      <td>-4.983141</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228253.1</th>\n",
       "      <td>775.857799</td>\n",
       "      <td>5944.350932</td>\n",
       "      <td>0.885889</td>\n",
       "      <td>0.385417</td>\n",
       "      <td>0.552448</td>\n",
       "      <td>-5.121933</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         logFC       AveExpr         t   P.Value  adj.P.Val  \\\n",
       "ENSG00000264125.1  1678.911403    808.151173  1.059870  0.300891   0.531495   \n",
       "ENSG00000229276.1  1402.821520    671.971224  1.063376  0.299335   0.531357   \n",
       "ENSG00000210082.2   821.464218  10147.594571  0.547916  0.589357   0.724887   \n",
       "ENSG00000200732.1   777.003071    374.050328  1.060896  0.300435   0.531426   \n",
       "ENSG00000228253.1   775.857799   5944.350932  0.885889  0.385417   0.552448   \n",
       "\n",
       "                          B  ranking  \n",
       "ENSG00000264125.1 -4.984019      1.0  \n",
       "ENSG00000229276.1 -4.981014      2.0  \n",
       "ENSG00000210082.2 -5.323706      3.0  \n",
       "ENSG00000200732.1 -4.983141      4.0  \n",
       "ENSG00000228253.1 -5.121933      5.0  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get ranks of template experiment\n",
    "# Rank pathways by highest enrichment score\n",
    "template_DE_stats['ranking'] = template_DE_stats[col_to_rank].rank(ascending = 0) \n",
    "template_DE_stats = template_DE_stats.sort_values(by=col_to_rank, ascending=False)\n",
    "\n",
    "template_DE_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1450925, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>29841</th>\n",
       "      <td>ENSG00000281591.1</td>\n",
       "      <td>4597.867091</td>\n",
       "      <td>2199.015913</td>\n",
       "      <td>1.057327</td>\n",
       "      <td>0.302138</td>\n",
       "      <td>0.584780</td>\n",
       "      <td>-5.114569</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29843</th>\n",
       "      <td>ENSG00000229276.1</td>\n",
       "      <td>3835.325295</td>\n",
       "      <td>1834.311261</td>\n",
       "      <td>1.057327</td>\n",
       "      <td>0.302138</td>\n",
       "      <td>0.584780</td>\n",
       "      <td>-5.114569</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28254</th>\n",
       "      <td>ENSG00000075624.13</td>\n",
       "      <td>2234.895167</td>\n",
       "      <td>2431.335739</td>\n",
       "      <td>1.144976</td>\n",
       "      <td>0.264848</td>\n",
       "      <td>0.544002</td>\n",
       "      <td>-5.032193</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29866</th>\n",
       "      <td>ENSG00000276007.1</td>\n",
       "      <td>1279.786894</td>\n",
       "      <td>612.100826</td>\n",
       "      <td>1.057324</td>\n",
       "      <td>0.302139</td>\n",
       "      <td>0.584780</td>\n",
       "      <td>-5.114572</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29845</th>\n",
       "      <td>ENSG00000276980.1</td>\n",
       "      <td>1130.097432</td>\n",
       "      <td>540.498130</td>\n",
       "      <td>1.057327</td>\n",
       "      <td>0.302138</td>\n",
       "      <td>0.584780</td>\n",
       "      <td>-5.114569</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    index        logFC      AveExpr         t   P.Value  \\\n",
       "29841   ENSG00000281591.1  4597.867091  2199.015913  1.057327  0.302138   \n",
       "29843   ENSG00000229276.1  3835.325295  1834.311261  1.057327  0.302138   \n",
       "28254  ENSG00000075624.13  2234.895167  2431.335739  1.144976  0.264848   \n",
       "29866   ENSG00000276007.1  1279.786894   612.100826  1.057324  0.302139   \n",
       "29845   ENSG00000276980.1  1130.097432   540.498130  1.057327  0.302138   \n",
       "\n",
       "       adj.P.Val         B  ranking  \n",
       "29841   0.584780 -5.114569      1.0  \n",
       "29843   0.584780 -5.114569      2.0  \n",
       "28254   0.544002 -5.032193      3.0  \n",
       "29866   0.584780 -5.114572      4.0  \n",
       "29845   0.584780 -5.114569      5.0  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concatenate simulated experiments\n",
    "simulated_DE_stats_all = pd.DataFrame()\n",
    "for i in range(num_runs):\n",
    "    simulated_DE_stats_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"DE_stats\",\n",
    "        \"DE_stats_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "    \n",
    "    #Read results\n",
    "    simulated_DE_stats = pd.read_csv(\n",
    "        simulated_DE_stats_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "    \n",
    "    simulated_DE_stats.reset_index(inplace=True)\n",
    "    \n",
    "    # Add ranks of simulated experiment\n",
    "    simulated_DE_stats['ranking'] = simulated_DE_stats[col_to_rank].rank(ascending = 0) \n",
    "    simulated_DE_stats = simulated_DE_stats.sort_values(by=col_to_rank, ascending=False)\n",
    "    \n",
    "    # Concatenate df\n",
    "    simulated_DE_stats_all = pd.concat([simulated_DE_stats_all,\n",
    "                                       simulated_DE_stats])\n",
    "    \n",
    "print(simulated_DE_stats_all.shape)\n",
    "simulated_DE_stats_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">logFC</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>count</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003.14</th>\n",
       "      <td>0.216765</td>\n",
       "      <td>0.303148</td>\n",
       "      <td>25</td>\n",
       "      <td>0.879047</td>\n",
       "      <td>8904.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005.5</th>\n",
       "      <td>0.001933</td>\n",
       "      <td>0.006770</td>\n",
       "      <td>25</td>\n",
       "      <td>0.977763</td>\n",
       "      <td>19576.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419.12</th>\n",
       "      <td>2.241911</td>\n",
       "      <td>1.478394</td>\n",
       "      <td>25</td>\n",
       "      <td>0.725082</td>\n",
       "      <td>1609.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.13</th>\n",
       "      <td>0.216039</td>\n",
       "      <td>0.115236</td>\n",
       "      <td>25</td>\n",
       "      <td>0.681224</td>\n",
       "      <td>6998.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460.16</th>\n",
       "      <td>0.123367</td>\n",
       "      <td>0.169765</td>\n",
       "      <td>25</td>\n",
       "      <td>0.924282</td>\n",
       "      <td>9519.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       logFC                 adj.P.Val  ranking\n",
       "                        mean       std count    median   median\n",
       "index                                                          \n",
       "ENSG00000000003.14  0.216765  0.303148    25  0.879047   8904.0\n",
       "ENSG00000000005.5   0.001933  0.006770    25  0.977763  19576.5\n",
       "ENSG00000000419.12  2.241911  1.478394    25  0.725082   1609.0\n",
       "ENSG00000000457.13  0.216039  0.115236    25  0.681224   6998.0\n",
       "ENSG00000000460.16  0.123367  0.169765    25  0.924282   9519.0"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simulated_DE_summary_stats = simulated_DE_stats_all.groupby(['index'])[[col_to_rank, 'adj.P.Val', 'ranking']].agg({\n",
    "    col_to_rank:['mean', 'std','count'],\n",
    "    'adj.P.Val':['median'],\n",
    "    'ranking':['median']\n",
    "})\n",
    "\n",
    "simulated_DE_summary_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58037, 12)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/reshape/merge.py:544: UserWarning: merging between different levels can give an unintended result (1 levels on the left, 2 on the right)\n",
      "  warnings.warn(msg, UserWarning)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "      <th>ranking</th>\n",
       "      <th>(logFC, mean)</th>\n",
       "      <th>(logFC, std)</th>\n",
       "      <th>(logFC, count)</th>\n",
       "      <th>(adj.P.Val, median)</th>\n",
       "      <th>(ranking, median)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000264125.1</th>\n",
       "      <td>1678.911403</td>\n",
       "      <td>808.151173</td>\n",
       "      <td>1.059870</td>\n",
       "      <td>0.300891</td>\n",
       "      <td>0.531495</td>\n",
       "      <td>-4.984019</td>\n",
       "      <td>1.0</td>\n",
       "      <td>50.264600</td>\n",
       "      <td>80.288095</td>\n",
       "      <td>25</td>\n",
       "      <td>0.584247</td>\n",
       "      <td>475.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229276.1</th>\n",
       "      <td>1402.821520</td>\n",
       "      <td>671.971224</td>\n",
       "      <td>1.063376</td>\n",
       "      <td>0.299335</td>\n",
       "      <td>0.531357</td>\n",
       "      <td>-4.981014</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1024.128898</td>\n",
       "      <td>1290.523085</td>\n",
       "      <td>25</td>\n",
       "      <td>0.580464</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000210082.2</th>\n",
       "      <td>821.464218</td>\n",
       "      <td>10147.594571</td>\n",
       "      <td>0.547916</td>\n",
       "      <td>0.589357</td>\n",
       "      <td>0.724887</td>\n",
       "      <td>-5.323706</td>\n",
       "      <td>3.0</td>\n",
       "      <td>53.927394</td>\n",
       "      <td>661.780706</td>\n",
       "      <td>25</td>\n",
       "      <td>0.971493</td>\n",
       "      <td>58000.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000200732.1</th>\n",
       "      <td>777.003071</td>\n",
       "      <td>374.050328</td>\n",
       "      <td>1.060896</td>\n",
       "      <td>0.300435</td>\n",
       "      <td>0.531426</td>\n",
       "      <td>-4.983141</td>\n",
       "      <td>4.0</td>\n",
       "      <td>26.514304</td>\n",
       "      <td>41.965338</td>\n",
       "      <td>25</td>\n",
       "      <td>0.584247</td>\n",
       "      <td>571.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228253.1</th>\n",
       "      <td>775.857799</td>\n",
       "      <td>5944.350932</td>\n",
       "      <td>0.885889</td>\n",
       "      <td>0.385417</td>\n",
       "      <td>0.552448</td>\n",
       "      <td>-5.121933</td>\n",
       "      <td>5.0</td>\n",
       "      <td>95.819618</td>\n",
       "      <td>265.108963</td>\n",
       "      <td>25</td>\n",
       "      <td>0.963800</td>\n",
       "      <td>92.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         logFC       AveExpr         t   P.Value  adj.P.Val  \\\n",
       "ENSG00000264125.1  1678.911403    808.151173  1.059870  0.300891   0.531495   \n",
       "ENSG00000229276.1  1402.821520    671.971224  1.063376  0.299335   0.531357   \n",
       "ENSG00000210082.2   821.464218  10147.594571  0.547916  0.589357   0.724887   \n",
       "ENSG00000200732.1   777.003071    374.050328  1.060896  0.300435   0.531426   \n",
       "ENSG00000228253.1   775.857799   5944.350932  0.885889  0.385417   0.552448   \n",
       "\n",
       "                          B  ranking  (logFC, mean)  (logFC, std)  \\\n",
       "ENSG00000264125.1 -4.984019      1.0      50.264600     80.288095   \n",
       "ENSG00000229276.1 -4.981014      2.0    1024.128898   1290.523085   \n",
       "ENSG00000210082.2 -5.323706      3.0      53.927394    661.780706   \n",
       "ENSG00000200732.1 -4.983141      4.0      26.514304     41.965338   \n",
       "ENSG00000228253.1 -5.121933      5.0      95.819618    265.108963   \n",
       "\n",
       "                   (logFC, count)  (adj.P.Val, median)  (ranking, median)  \n",
       "ENSG00000264125.1              25             0.584247              475.0  \n",
       "ENSG00000229276.1              25             0.580464                5.0  \n",
       "ENSG00000210082.2              25             0.971493            58000.0  \n",
       "ENSG00000200732.1              25             0.584247              571.0  \n",
       "ENSG00000228253.1              25             0.963800               92.0  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge template statistics with simulated statistics\n",
    "template_simulated_DE_stats = template_DE_stats.merge(simulated_DE_summary_stats, \n",
    "                                                     left_index=True,\n",
    "                                                     right_index=True)\n",
    "print(template_simulated_DE_stats.shape)\n",
    "template_simulated_DE_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse columns\n",
    "median_pval_simulated = template_simulated_DE_stats[('adj.P.Val','median')]\n",
    "median_rank_simulated = template_simulated_DE_stats[('ranking','median')]\n",
    "mean_test_simulated = template_simulated_DE_stats[(col_to_rank,'mean')]\n",
    "std_test_simulated = template_simulated_DE_stats[(col_to_rank,'std')]\n",
    "count_simulated = template_simulated_DE_stats[(col_to_rank,'count')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Adj P-value (Real)</th>\n",
       "      <th>Rank (Real)</th>\n",
       "      <th>Test statistic (Real)</th>\n",
       "      <th>Median adj p-value (simulated)</th>\n",
       "      <th>Median rank (simulated)</th>\n",
       "      <th>Mean test statistic (simulated)</th>\n",
       "      <th>Std deviation (simulated)</th>\n",
       "      <th>Number of experiments (simulated)</th>\n",
       "      <th>Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000264125.1</th>\n",
       "      <td>ENSG00000264125.1</td>\n",
       "      <td>0.531495</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1678.911403</td>\n",
       "      <td>0.584247</td>\n",
       "      <td>475.0</td>\n",
       "      <td>50.264600</td>\n",
       "      <td>80.288095</td>\n",
       "      <td>25</td>\n",
       "      <td>20.285035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229276.1</th>\n",
       "      <td>ENSG00000229276.1</td>\n",
       "      <td>0.531357</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1402.821520</td>\n",
       "      <td>0.580464</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1024.128898</td>\n",
       "      <td>1290.523085</td>\n",
       "      <td>25</td>\n",
       "      <td>0.293441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000210082.2</th>\n",
       "      <td>ENSG00000210082.2</td>\n",
       "      <td>0.724887</td>\n",
       "      <td>3.0</td>\n",
       "      <td>821.464218</td>\n",
       "      <td>0.971493</td>\n",
       "      <td>58000.0</td>\n",
       "      <td>53.927394</td>\n",
       "      <td>661.780706</td>\n",
       "      <td>25</td>\n",
       "      <td>1.159805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000200732.1</th>\n",
       "      <td>ENSG00000200732.1</td>\n",
       "      <td>0.531426</td>\n",
       "      <td>4.0</td>\n",
       "      <td>777.003071</td>\n",
       "      <td>0.584247</td>\n",
       "      <td>571.0</td>\n",
       "      <td>26.514304</td>\n",
       "      <td>41.965338</td>\n",
       "      <td>25</td>\n",
       "      <td>17.883539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228253.1</th>\n",
       "      <td>ENSG00000228253.1</td>\n",
       "      <td>0.552448</td>\n",
       "      <td>5.0</td>\n",
       "      <td>775.857799</td>\n",
       "      <td>0.963800</td>\n",
       "      <td>92.0</td>\n",
       "      <td>95.819618</td>\n",
       "      <td>265.108963</td>\n",
       "      <td>25</td>\n",
       "      <td>2.565127</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene ID  Adj P-value (Real)  Rank (Real)  \\\n",
       "ENSG00000264125.1  ENSG00000264125.1            0.531495          1.0   \n",
       "ENSG00000229276.1  ENSG00000229276.1            0.531357          2.0   \n",
       "ENSG00000210082.2  ENSG00000210082.2            0.724887          3.0   \n",
       "ENSG00000200732.1  ENSG00000200732.1            0.531426          4.0   \n",
       "ENSG00000228253.1  ENSG00000228253.1            0.552448          5.0   \n",
       "\n",
       "                   Test statistic (Real)  Median adj p-value (simulated)  \\\n",
       "ENSG00000264125.1            1678.911403                        0.584247   \n",
       "ENSG00000229276.1            1402.821520                        0.580464   \n",
       "ENSG00000210082.2             821.464218                        0.971493   \n",
       "ENSG00000200732.1             777.003071                        0.584247   \n",
       "ENSG00000228253.1             775.857799                        0.963800   \n",
       "\n",
       "                   Median rank (simulated)  Mean test statistic (simulated)  \\\n",
       "ENSG00000264125.1                    475.0                        50.264600   \n",
       "ENSG00000229276.1                      5.0                      1024.128898   \n",
       "ENSG00000210082.2                  58000.0                        53.927394   \n",
       "ENSG00000200732.1                    571.0                        26.514304   \n",
       "ENSG00000228253.1                     92.0                        95.819618   \n",
       "\n",
       "                   Std deviation (simulated)  \\\n",
       "ENSG00000264125.1                  80.288095   \n",
       "ENSG00000229276.1                1290.523085   \n",
       "ENSG00000210082.2                 661.780706   \n",
       "ENSG00000200732.1                  41.965338   \n",
       "ENSG00000228253.1                 265.108963   \n",
       "\n",
       "                   Number of experiments (simulated)    Z score  \n",
       "ENSG00000264125.1                                 25  20.285035  \n",
       "ENSG00000229276.1                                 25   0.293441  \n",
       "ENSG00000210082.2                                 25   1.159805  \n",
       "ENSG00000200732.1                                 25  17.883539  \n",
       "ENSG00000228253.1                                 25   2.565127  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary = pd.DataFrame(data={'Gene ID': template_simulated_DE_stats.index,\n",
    "                             'Adj P-value (Real)': template_simulated_DE_stats['adj.P.Val'],\n",
    "                             'Rank (Real)': template_simulated_DE_stats['ranking'],\n",
    "                             'Test statistic (Real)': template_simulated_DE_stats[col_to_rank],\n",
    "                             'Median adj p-value (simulated)': median_pval_simulated ,\n",
    "                             'Median rank (simulated)': median_rank_simulated ,\n",
    "                             'Mean test statistic (simulated)': mean_test_simulated ,\n",
    "                             'Std deviation (simulated)': std_test_simulated,\n",
    "                             'Number of experiments (simulated)': count_simulated\n",
    "                            }\n",
    "                      )\n",
    "summary['Z score'] = (summary['Test statistic (Real)'] - summary['Mean test statistic (simulated)'])/summary['Std deviation (simulated)']\n",
    "summary.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Adj P-value (Real)</th>\n",
       "      <th>Rank (Real)</th>\n",
       "      <th>Test statistic (Real)</th>\n",
       "      <th>Median adj p-value (simulated)</th>\n",
       "      <th>Median rank (simulated)</th>\n",
       "      <th>Mean test statistic (simulated)</th>\n",
       "      <th>Std deviation (simulated)</th>\n",
       "      <th>Number of experiments (simulated)</th>\n",
       "      <th>Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000231053.1</th>\n",
       "      <td>ENSG00000231053.1</td>\n",
       "      <td>0.515574</td>\n",
       "      <td>48376.0</td>\n",
       "      <td>-1.809574</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>54630.0</td>\n",
       "      <td>-3.117008</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>25</td>\n",
       "      <td>1803.750894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000207135.1</th>\n",
       "      <td>ENSG00000207135.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>38434.0</td>\n",
       "      <td>-1.016753</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>50595.0</td>\n",
       "      <td>-2.033245</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>25</td>\n",
       "      <td>1693.138607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277828.1</th>\n",
       "      <td>ENSG00000277828.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>7689.0</td>\n",
       "      <td>0.187870</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>-0.000034</td>\n",
       "      <td>0.000117</td>\n",
       "      <td>25</td>\n",
       "      <td>1601.296648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238124.1</th>\n",
       "      <td>ENSG00000238124.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>28946.5</td>\n",
       "      <td>-0.589662</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>44267.0</td>\n",
       "      <td>-1.179318</td>\n",
       "      <td>0.000373</td>\n",
       "      <td>25</td>\n",
       "      <td>1579.645423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000225911.1</th>\n",
       "      <td>ENSG00000225911.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>25548.0</td>\n",
       "      <td>-0.469049</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>42634.0</td>\n",
       "      <td>-1.009997</td>\n",
       "      <td>0.000387</td>\n",
       "      <td>25</td>\n",
       "      <td>1396.811969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000232529.1</th>\n",
       "      <td>ENSG00000232529.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>27694.0</td>\n",
       "      <td>-0.545429</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>40352.0</td>\n",
       "      <td>-0.785548</td>\n",
       "      <td>0.000183</td>\n",
       "      <td>25</td>\n",
       "      <td>1311.243896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000270829.1</th>\n",
       "      <td>ENSG00000270829.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>54432.0</td>\n",
       "      <td>-3.032371</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>55470.0</td>\n",
       "      <td>-3.546424</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>25</td>\n",
       "      <td>1306.920278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000266416.1</th>\n",
       "      <td>ENSG00000266416.1</td>\n",
       "      <td>0.531357</td>\n",
       "      <td>3326.0</td>\n",
       "      <td>1.672872</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>-0.000522</td>\n",
       "      <td>0.001313</td>\n",
       "      <td>25</td>\n",
       "      <td>1274.033372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000200444.1</th>\n",
       "      <td>ENSG00000200444.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>3456.0</td>\n",
       "      <td>1.564541</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>17680.0</td>\n",
       "      <td>-0.000042</td>\n",
       "      <td>0.001248</td>\n",
       "      <td>25</td>\n",
       "      <td>1253.896058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000256717.1</th>\n",
       "      <td>ENSG00000256717.1</td>\n",
       "      <td>0.556127</td>\n",
       "      <td>16746.0</td>\n",
       "      <td>-0.174638</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>36018.0</td>\n",
       "      <td>-0.379028</td>\n",
       "      <td>0.000167</td>\n",
       "      <td>25</td>\n",
       "      <td>1225.788027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000252068.1</th>\n",
       "      <td>ENSG00000252068.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>28864.0</td>\n",
       "      <td>-0.586799</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>44210.5</td>\n",
       "      <td>-1.173456</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>25</td>\n",
       "      <td>1195.774083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000258853.1</th>\n",
       "      <td>ENSG00000258853.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>30877.0</td>\n",
       "      <td>-0.665133</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>45660.0</td>\n",
       "      <td>-1.330060</td>\n",
       "      <td>0.000565</td>\n",
       "      <td>25</td>\n",
       "      <td>1176.608104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000200738.1</th>\n",
       "      <td>ENSG00000200738.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>57002.0</td>\n",
       "      <td>-5.167299</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>57651.0</td>\n",
       "      <td>-7.313679</td>\n",
       "      <td>0.001826</td>\n",
       "      <td>25</td>\n",
       "      <td>1175.179030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000252951.1</th>\n",
       "      <td>ENSG00000252951.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>55970.0</td>\n",
       "      <td>-3.880029</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>57280.0</td>\n",
       "      <td>-5.676128</td>\n",
       "      <td>0.001541</td>\n",
       "      <td>25</td>\n",
       "      <td>1165.887982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261815.1</th>\n",
       "      <td>ENSG00000261815.1</td>\n",
       "      <td>0.507178</td>\n",
       "      <td>24174.0</td>\n",
       "      <td>-0.423866</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>39480.0</td>\n",
       "      <td>-0.704755</td>\n",
       "      <td>0.000241</td>\n",
       "      <td>25</td>\n",
       "      <td>1165.376390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176567.1</th>\n",
       "      <td>ENSG00000176567.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>33593.0</td>\n",
       "      <td>-0.778529</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>46261.0</td>\n",
       "      <td>-1.397074</td>\n",
       "      <td>0.000542</td>\n",
       "      <td>25</td>\n",
       "      <td>1142.115932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261725.1</th>\n",
       "      <td>ENSG00000261725.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>21332.0</td>\n",
       "      <td>-0.330533</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>41714.0</td>\n",
       "      <td>-0.919491</td>\n",
       "      <td>0.000517</td>\n",
       "      <td>25</td>\n",
       "      <td>1139.597159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000265291.1</th>\n",
       "      <td>ENSG00000265291.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>57830.0</td>\n",
       "      <td>-11.063479</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>57968.0</td>\n",
       "      <td>-13.923416</td>\n",
       "      <td>0.002511</td>\n",
       "      <td>25</td>\n",
       "      <td>1139.139154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259691.2</th>\n",
       "      <td>ENSG00000259691.2</td>\n",
       "      <td>0.497393</td>\n",
       "      <td>45429.0</td>\n",
       "      <td>-1.497116</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>52133.0</td>\n",
       "      <td>-2.349556</td>\n",
       "      <td>0.000750</td>\n",
       "      <td>25</td>\n",
       "      <td>1136.364192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254749.1</th>\n",
       "      <td>ENSG00000254749.1</td>\n",
       "      <td>0.560640</td>\n",
       "      <td>42061.0</td>\n",
       "      <td>-1.232012</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>51133.0</td>\n",
       "      <td>-2.130578</td>\n",
       "      <td>0.000798</td>\n",
       "      <td>25</td>\n",
       "      <td>1126.099895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000202195.1</th>\n",
       "      <td>ENSG00000202195.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>7476.5</td>\n",
       "      <td>0.215643</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>-0.000062</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>25</td>\n",
       "      <td>1079.641496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273966.1</th>\n",
       "      <td>ENSG00000273966.1</td>\n",
       "      <td>0.531518</td>\n",
       "      <td>6051.0</td>\n",
       "      <td>0.462543</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>17673.0</td>\n",
       "      <td>-0.000200</td>\n",
       "      <td>0.000429</td>\n",
       "      <td>25</td>\n",
       "      <td>1078.525877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234763.1</th>\n",
       "      <td>ENSG00000234763.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>32253.0</td>\n",
       "      <td>-0.722552</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>46483.0</td>\n",
       "      <td>-1.425659</td>\n",
       "      <td>0.000661</td>\n",
       "      <td>25</td>\n",
       "      <td>1063.829721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224447.1</th>\n",
       "      <td>ENSG00000224447.1</td>\n",
       "      <td>0.490147</td>\n",
       "      <td>50040.0</td>\n",
       "      <td>-2.032519</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>54165.0</td>\n",
       "      <td>-2.926662</td>\n",
       "      <td>0.000841</td>\n",
       "      <td>25</td>\n",
       "      <td>1063.132505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000275776.1</th>\n",
       "      <td>ENSG00000275776.1</td>\n",
       "      <td>0.494916</td>\n",
       "      <td>50990.0</td>\n",
       "      <td>-2.178132</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>55334.0</td>\n",
       "      <td>-3.470714</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>25</td>\n",
       "      <td>1056.332978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000260004.1</th>\n",
       "      <td>ENSG00000260004.1</td>\n",
       "      <td>0.545192</td>\n",
       "      <td>17956.0</td>\n",
       "      <td>-0.217975</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>36891.0</td>\n",
       "      <td>-0.461073</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>25</td>\n",
       "      <td>1047.512553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000207027.1</th>\n",
       "      <td>ENSG00000207027.1</td>\n",
       "      <td>0.532513</td>\n",
       "      <td>7518.0</td>\n",
       "      <td>0.209541</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16315.0</td>\n",
       "      <td>-0.000067</td>\n",
       "      <td>0.000201</td>\n",
       "      <td>25</td>\n",
       "      <td>1043.424778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000257837.1</th>\n",
       "      <td>ENSG00000257837.1</td>\n",
       "      <td>0.531800</td>\n",
       "      <td>25334.0</td>\n",
       "      <td>-0.462149</td>\n",
       "      <td>0.469077</td>\n",
       "      <td>41012.0</td>\n",
       "      <td>-0.850762</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>25</td>\n",
       "      <td>1023.938992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000252977.1</th>\n",
       "      <td>ENSG00000252977.1</td>\n",
       "      <td>0.531642</td>\n",
       "      <td>6203.0</td>\n",
       "      <td>0.428354</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16875.5</td>\n",
       "      <td>-0.000177</td>\n",
       "      <td>0.000419</td>\n",
       "      <td>25</td>\n",
       "      <td>1021.548536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000252411.1</th>\n",
       "      <td>ENSG00000252411.1</td>\n",
       "      <td>0.490358</td>\n",
       "      <td>6268.0</td>\n",
       "      <td>0.411005</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16780.5</td>\n",
       "      <td>-0.000157</td>\n",
       "      <td>0.000409</td>\n",
       "      <td>25</td>\n",
       "      <td>1005.832161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000124334.17_PAR_Y</th>\n",
       "      <td>ENSG00000124334.17_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168939.11_PAR_Y</th>\n",
       "      <td>ENSG00000168939.11_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000167393.17_PAR_Y</th>\n",
       "      <td>ENSG00000167393.17_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000124333.15_PAR_Y</th>\n",
       "      <td>ENSG00000124333.15_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000002586.18_PAR_Y</th>\n",
       "      <td>ENSG00000002586.18_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000169084.13_PAR_Y</th>\n",
       "      <td>ENSG00000169084.13_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000169093.15_PAR_Y</th>\n",
       "      <td>ENSG00000169093.15_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000169100.13_PAR_Y</th>\n",
       "      <td>ENSG00000169100.13_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178605.13_PAR_Y</th>\n",
       "      <td>ENSG00000178605.13_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223484.7_PAR_Y</th>\n",
       "      <td>ENSG00000223484.7_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223511.6_PAR_Y</th>\n",
       "      <td>ENSG00000223511.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223571.6_PAR_Y</th>\n",
       "      <td>ENSG00000223571.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223773.7_PAR_Y</th>\n",
       "      <td>ENSG00000223773.7_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000225661.7_PAR_Y</th>\n",
       "      <td>ENSG00000225661.7_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000226179.6_PAR_Y</th>\n",
       "      <td>ENSG00000226179.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223274.6_PAR_Y</th>\n",
       "      <td>ENSG00000223274.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214717.10_PAR_Y</th>\n",
       "      <td>ENSG00000214717.10_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000230542.6_PAR_Y</th>\n",
       "      <td>ENSG00000230542.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228410.6_PAR_Y</th>\n",
       "      <td>ENSG00000228410.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227159.8_PAR_Y</th>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000205755.11_PAR_Y</th>\n",
       "      <td>ENSG00000205755.11_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229232.6_PAR_Y</th>\n",
       "      <td>ENSG00000229232.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228572.7_PAR_Y</th>\n",
       "      <td>ENSG00000228572.7_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237801.6_PAR_Y</th>\n",
       "      <td>ENSG00000237801.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234622.6_PAR_Y</th>\n",
       "      <td>ENSG00000234622.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234958.6_PAR_Y</th>\n",
       "      <td>ENSG00000234958.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236871.7_PAR_Y</th>\n",
       "      <td>ENSG00000236871.7_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237040.6_PAR_Y</th>\n",
       "      <td>ENSG00000237040.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237531.6_PAR_Y</th>\n",
       "      <td>ENSG00000237531.6_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236017.8_PAR_Y</th>\n",
       "      <td>ENSG00000236017.8_PAR_Y</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11483.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58037 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Gene ID  Adj P-value (Real)  \\\n",
       "ENSG00000231053.1                ENSG00000231053.1            0.515574   \n",
       "ENSG00000207135.1                ENSG00000207135.1            0.532513   \n",
       "ENSG00000277828.1                ENSG00000277828.1            0.532513   \n",
       "ENSG00000238124.1                ENSG00000238124.1            0.532513   \n",
       "ENSG00000225911.1                ENSG00000225911.1            0.532513   \n",
       "ENSG00000232529.1                ENSG00000232529.1            0.490147   \n",
       "ENSG00000270829.1                ENSG00000270829.1            0.490147   \n",
       "ENSG00000266416.1                ENSG00000266416.1            0.531357   \n",
       "ENSG00000200444.1                ENSG00000200444.1            0.490147   \n",
       "ENSG00000256717.1                ENSG00000256717.1            0.556127   \n",
       "ENSG00000252068.1                ENSG00000252068.1            0.532513   \n",
       "ENSG00000258853.1                ENSG00000258853.1            0.532513   \n",
       "ENSG00000200738.1                ENSG00000200738.1            0.490147   \n",
       "ENSG00000252951.1                ENSG00000252951.1            0.490147   \n",
       "ENSG00000261815.1                ENSG00000261815.1            0.507178   \n",
       "ENSG00000176567.1                ENSG00000176567.1            0.490147   \n",
       "ENSG00000261725.1                ENSG00000261725.1            0.532513   \n",
       "ENSG00000265291.1                ENSG00000265291.1            0.490147   \n",
       "ENSG00000259691.2                ENSG00000259691.2            0.497393   \n",
       "ENSG00000254749.1                ENSG00000254749.1            0.560640   \n",
       "ENSG00000202195.1                ENSG00000202195.1            0.532513   \n",
       "ENSG00000273966.1                ENSG00000273966.1            0.531518   \n",
       "ENSG00000234763.1                ENSG00000234763.1            0.532513   \n",
       "ENSG00000224447.1                ENSG00000224447.1            0.490147   \n",
       "ENSG00000275776.1                ENSG00000275776.1            0.494916   \n",
       "ENSG00000260004.1                ENSG00000260004.1            0.545192   \n",
       "ENSG00000207027.1                ENSG00000207027.1            0.532513   \n",
       "ENSG00000257837.1                ENSG00000257837.1            0.531800   \n",
       "ENSG00000252977.1                ENSG00000252977.1            0.531642   \n",
       "ENSG00000252411.1                ENSG00000252411.1            0.490358   \n",
       "...                                            ...                 ...   \n",
       "ENSG00000124334.17_PAR_Y  ENSG00000124334.17_PAR_Y            1.000000   \n",
       "ENSG00000168939.11_PAR_Y  ENSG00000168939.11_PAR_Y            1.000000   \n",
       "ENSG00000167393.17_PAR_Y  ENSG00000167393.17_PAR_Y            1.000000   \n",
       "ENSG00000124333.15_PAR_Y  ENSG00000124333.15_PAR_Y            1.000000   \n",
       "ENSG00000002586.18_PAR_Y  ENSG00000002586.18_PAR_Y            1.000000   \n",
       "ENSG00000169084.13_PAR_Y  ENSG00000169084.13_PAR_Y            1.000000   \n",
       "ENSG00000169093.15_PAR_Y  ENSG00000169093.15_PAR_Y            1.000000   \n",
       "ENSG00000169100.13_PAR_Y  ENSG00000169100.13_PAR_Y            1.000000   \n",
       "ENSG00000178605.13_PAR_Y  ENSG00000178605.13_PAR_Y            1.000000   \n",
       "ENSG00000223484.7_PAR_Y    ENSG00000223484.7_PAR_Y            1.000000   \n",
       "ENSG00000223511.6_PAR_Y    ENSG00000223511.6_PAR_Y            1.000000   \n",
       "ENSG00000223571.6_PAR_Y    ENSG00000223571.6_PAR_Y            1.000000   \n",
       "ENSG00000223773.7_PAR_Y    ENSG00000223773.7_PAR_Y            1.000000   \n",
       "ENSG00000225661.7_PAR_Y    ENSG00000225661.7_PAR_Y            1.000000   \n",
       "ENSG00000226179.6_PAR_Y    ENSG00000226179.6_PAR_Y            1.000000   \n",
       "ENSG00000223274.6_PAR_Y    ENSG00000223274.6_PAR_Y            1.000000   \n",
       "ENSG00000214717.10_PAR_Y  ENSG00000214717.10_PAR_Y            1.000000   \n",
       "ENSG00000230542.6_PAR_Y    ENSG00000230542.6_PAR_Y            1.000000   \n",
       "ENSG00000228410.6_PAR_Y    ENSG00000228410.6_PAR_Y            1.000000   \n",
       "ENSG00000227159.8_PAR_Y    ENSG00000227159.8_PAR_Y            1.000000   \n",
       "ENSG00000205755.11_PAR_Y  ENSG00000205755.11_PAR_Y            1.000000   \n",
       "ENSG00000229232.6_PAR_Y    ENSG00000229232.6_PAR_Y            1.000000   \n",
       "ENSG00000228572.7_PAR_Y    ENSG00000228572.7_PAR_Y            1.000000   \n",
       "ENSG00000237801.6_PAR_Y    ENSG00000237801.6_PAR_Y            1.000000   \n",
       "ENSG00000234622.6_PAR_Y    ENSG00000234622.6_PAR_Y            1.000000   \n",
       "ENSG00000234958.6_PAR_Y    ENSG00000234958.6_PAR_Y            1.000000   \n",
       "ENSG00000236871.7_PAR_Y    ENSG00000236871.7_PAR_Y            1.000000   \n",
       "ENSG00000237040.6_PAR_Y    ENSG00000237040.6_PAR_Y            1.000000   \n",
       "ENSG00000237531.6_PAR_Y    ENSG00000237531.6_PAR_Y            1.000000   \n",
       "ENSG00000236017.8_PAR_Y    ENSG00000236017.8_PAR_Y            1.000000   \n",
       "\n",
       "                          Rank (Real)  Test statistic (Real)  \\\n",
       "ENSG00000231053.1             48376.0              -1.809574   \n",
       "ENSG00000207135.1             38434.0              -1.016753   \n",
       "ENSG00000277828.1              7689.0               0.187870   \n",
       "ENSG00000238124.1             28946.5              -0.589662   \n",
       "ENSG00000225911.1             25548.0              -0.469049   \n",
       "ENSG00000232529.1             27694.0              -0.545429   \n",
       "ENSG00000270829.1             54432.0              -3.032371   \n",
       "ENSG00000266416.1              3326.0               1.672872   \n",
       "ENSG00000200444.1              3456.0               1.564541   \n",
       "ENSG00000256717.1             16746.0              -0.174638   \n",
       "ENSG00000252068.1             28864.0              -0.586799   \n",
       "ENSG00000258853.1             30877.0              -0.665133   \n",
       "ENSG00000200738.1             57002.0              -5.167299   \n",
       "ENSG00000252951.1             55970.0              -3.880029   \n",
       "ENSG00000261815.1             24174.0              -0.423866   \n",
       "ENSG00000176567.1             33593.0              -0.778529   \n",
       "ENSG00000261725.1             21332.0              -0.330533   \n",
       "ENSG00000265291.1             57830.0             -11.063479   \n",
       "ENSG00000259691.2             45429.0              -1.497116   \n",
       "ENSG00000254749.1             42061.0              -1.232012   \n",
       "ENSG00000202195.1              7476.5               0.215643   \n",
       "ENSG00000273966.1              6051.0               0.462543   \n",
       "ENSG00000234763.1             32253.0              -0.722552   \n",
       "ENSG00000224447.1             50040.0              -2.032519   \n",
       "ENSG00000275776.1             50990.0              -2.178132   \n",
       "ENSG00000260004.1             17956.0              -0.217975   \n",
       "ENSG00000207027.1              7518.0               0.209541   \n",
       "ENSG00000257837.1             25334.0              -0.462149   \n",
       "ENSG00000252977.1              6203.0               0.428354   \n",
       "ENSG00000252411.1              6268.0               0.411005   \n",
       "...                               ...                    ...   \n",
       "ENSG00000124334.17_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000168939.11_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000167393.17_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000124333.15_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000002586.18_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000169084.13_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000169093.15_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000169100.13_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000178605.13_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000223484.7_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000223511.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000223571.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000223773.7_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000225661.7_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000226179.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000223274.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000214717.10_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000230542.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000228410.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000227159.8_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000205755.11_PAR_Y      11483.0               0.000000   \n",
       "ENSG00000229232.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000228572.7_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000237801.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000234622.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000234958.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000236871.7_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000237040.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000237531.6_PAR_Y       11483.0               0.000000   \n",
       "ENSG00000236017.8_PAR_Y       11483.0               0.000000   \n",
       "\n",
       "                          Median adj p-value (simulated)  \\\n",
       "ENSG00000231053.1                               0.469077   \n",
       "ENSG00000207135.1                               0.469077   \n",
       "ENSG00000277828.1                               1.000000   \n",
       "ENSG00000238124.1                               0.469077   \n",
       "ENSG00000225911.1                               0.469077   \n",
       "ENSG00000232529.1                               0.469077   \n",
       "ENSG00000270829.1                               0.469077   \n",
       "ENSG00000266416.1                               1.000000   \n",
       "ENSG00000200444.1                               1.000000   \n",
       "ENSG00000256717.1                               0.469077   \n",
       "ENSG00000252068.1                               0.469077   \n",
       "ENSG00000258853.1                               0.469077   \n",
       "ENSG00000200738.1                               0.469077   \n",
       "ENSG00000252951.1                               0.469077   \n",
       "ENSG00000261815.1                               0.469077   \n",
       "ENSG00000176567.1                               0.469077   \n",
       "ENSG00000261725.1                               0.469077   \n",
       "ENSG00000265291.1                               0.469077   \n",
       "ENSG00000259691.2                               0.469077   \n",
       "ENSG00000254749.1                               0.469077   \n",
       "ENSG00000202195.1                               1.000000   \n",
       "ENSG00000273966.1                               1.000000   \n",
       "ENSG00000234763.1                               0.469077   \n",
       "ENSG00000224447.1                               0.469077   \n",
       "ENSG00000275776.1                               0.469077   \n",
       "ENSG00000260004.1                               0.469077   \n",
       "ENSG00000207027.1                               1.000000   \n",
       "ENSG00000257837.1                               0.469077   \n",
       "ENSG00000252977.1                               1.000000   \n",
       "ENSG00000252411.1                               1.000000   \n",
       "...                                                  ...   \n",
       "ENSG00000124334.17_PAR_Y                        1.000000   \n",
       "ENSG00000168939.11_PAR_Y                        1.000000   \n",
       "ENSG00000167393.17_PAR_Y                        1.000000   \n",
       "ENSG00000124333.15_PAR_Y                        1.000000   \n",
       "ENSG00000002586.18_PAR_Y                        1.000000   \n",
       "ENSG00000169084.13_PAR_Y                        1.000000   \n",
       "ENSG00000169093.15_PAR_Y                        1.000000   \n",
       "ENSG00000169100.13_PAR_Y                        1.000000   \n",
       "ENSG00000178605.13_PAR_Y                        1.000000   \n",
       "ENSG00000223484.7_PAR_Y                         1.000000   \n",
       "ENSG00000223511.6_PAR_Y                         1.000000   \n",
       "ENSG00000223571.6_PAR_Y                         1.000000   \n",
       "ENSG00000223773.7_PAR_Y                         1.000000   \n",
       "ENSG00000225661.7_PAR_Y                         1.000000   \n",
       "ENSG00000226179.6_PAR_Y                         1.000000   \n",
       "ENSG00000223274.6_PAR_Y                         1.000000   \n",
       "ENSG00000214717.10_PAR_Y                        1.000000   \n",
       "ENSG00000230542.6_PAR_Y                         1.000000   \n",
       "ENSG00000228410.6_PAR_Y                         1.000000   \n",
       "ENSG00000227159.8_PAR_Y                         1.000000   \n",
       "ENSG00000205755.11_PAR_Y                        1.000000   \n",
       "ENSG00000229232.6_PAR_Y                         1.000000   \n",
       "ENSG00000228572.7_PAR_Y                         1.000000   \n",
       "ENSG00000237801.6_PAR_Y                         1.000000   \n",
       "ENSG00000234622.6_PAR_Y                         1.000000   \n",
       "ENSG00000234958.6_PAR_Y                         1.000000   \n",
       "ENSG00000236871.7_PAR_Y                         1.000000   \n",
       "ENSG00000237040.6_PAR_Y                         1.000000   \n",
       "ENSG00000237531.6_PAR_Y                         1.000000   \n",
       "ENSG00000236017.8_PAR_Y                         1.000000   \n",
       "\n",
       "                          Median rank (simulated)  \\\n",
       "ENSG00000231053.1                         54630.0   \n",
       "ENSG00000207135.1                         50595.0   \n",
       "ENSG00000277828.1                         16387.0   \n",
       "ENSG00000238124.1                         44267.0   \n",
       "ENSG00000225911.1                         42634.0   \n",
       "ENSG00000232529.1                         40352.0   \n",
       "ENSG00000270829.1                         55470.0   \n",
       "ENSG00000266416.1                         16387.0   \n",
       "ENSG00000200444.1                         17680.0   \n",
       "ENSG00000256717.1                         36018.0   \n",
       "ENSG00000252068.1                         44210.5   \n",
       "ENSG00000258853.1                         45660.0   \n",
       "ENSG00000200738.1                         57651.0   \n",
       "ENSG00000252951.1                         57280.0   \n",
       "ENSG00000261815.1                         39480.0   \n",
       "ENSG00000176567.1                         46261.0   \n",
       "ENSG00000261725.1                         41714.0   \n",
       "ENSG00000265291.1                         57968.0   \n",
       "ENSG00000259691.2                         52133.0   \n",
       "ENSG00000254749.1                         51133.0   \n",
       "ENSG00000202195.1                         16387.0   \n",
       "ENSG00000273966.1                         17673.0   \n",
       "ENSG00000234763.1                         46483.0   \n",
       "ENSG00000224447.1                         54165.0   \n",
       "ENSG00000275776.1                         55334.0   \n",
       "ENSG00000260004.1                         36891.0   \n",
       "ENSG00000207027.1                         16315.0   \n",
       "ENSG00000257837.1                         41012.0   \n",
       "ENSG00000252977.1                         16875.5   \n",
       "ENSG00000252411.1                         16780.5   \n",
       "...                                           ...   \n",
       "ENSG00000124334.17_PAR_Y                  16387.0   \n",
       "ENSG00000168939.11_PAR_Y                  16387.0   \n",
       "ENSG00000167393.17_PAR_Y                  16387.0   \n",
       "ENSG00000124333.15_PAR_Y                  16387.0   \n",
       "ENSG00000002586.18_PAR_Y                  16387.0   \n",
       "ENSG00000169084.13_PAR_Y                  16387.0   \n",
       "ENSG00000169093.15_PAR_Y                  16387.0   \n",
       "ENSG00000169100.13_PAR_Y                  16387.0   \n",
       "ENSG00000178605.13_PAR_Y                  16387.0   \n",
       "ENSG00000223484.7_PAR_Y                   16387.0   \n",
       "ENSG00000223511.6_PAR_Y                   16387.0   \n",
       "ENSG00000223571.6_PAR_Y                   16387.0   \n",
       "ENSG00000223773.7_PAR_Y                   16387.0   \n",
       "ENSG00000225661.7_PAR_Y                   16387.0   \n",
       "ENSG00000226179.6_PAR_Y                   16387.0   \n",
       "ENSG00000223274.6_PAR_Y                   16387.0   \n",
       "ENSG00000214717.10_PAR_Y                  16387.0   \n",
       "ENSG00000230542.6_PAR_Y                   16387.0   \n",
       "ENSG00000228410.6_PAR_Y                   16387.0   \n",
       "ENSG00000227159.8_PAR_Y                   16387.0   \n",
       "ENSG00000205755.11_PAR_Y                  16387.0   \n",
       "ENSG00000229232.6_PAR_Y                   16387.0   \n",
       "ENSG00000228572.7_PAR_Y                   16387.0   \n",
       "ENSG00000237801.6_PAR_Y                   16387.0   \n",
       "ENSG00000234622.6_PAR_Y                   16387.0   \n",
       "ENSG00000234958.6_PAR_Y                   16387.0   \n",
       "ENSG00000236871.7_PAR_Y                   16387.0   \n",
       "ENSG00000237040.6_PAR_Y                   16387.0   \n",
       "ENSG00000237531.6_PAR_Y                   16387.0   \n",
       "ENSG00000236017.8_PAR_Y                   16387.0   \n",
       "\n",
       "                          Mean test statistic (simulated)  \\\n",
       "ENSG00000231053.1                               -3.117008   \n",
       "ENSG00000207135.1                               -2.033245   \n",
       "ENSG00000277828.1                               -0.000034   \n",
       "ENSG00000238124.1                               -1.179318   \n",
       "ENSG00000225911.1                               -1.009997   \n",
       "ENSG00000232529.1                               -0.785548   \n",
       "ENSG00000270829.1                               -3.546424   \n",
       "ENSG00000266416.1                               -0.000522   \n",
       "ENSG00000200444.1                               -0.000042   \n",
       "ENSG00000256717.1                               -0.379028   \n",
       "ENSG00000252068.1                               -1.173456   \n",
       "ENSG00000258853.1                               -1.330060   \n",
       "ENSG00000200738.1                               -7.313679   \n",
       "ENSG00000252951.1                               -5.676128   \n",
       "ENSG00000261815.1                               -0.704755   \n",
       "ENSG00000176567.1                               -1.397074   \n",
       "ENSG00000261725.1                               -0.919491   \n",
       "ENSG00000265291.1                              -13.923416   \n",
       "ENSG00000259691.2                               -2.349556   \n",
       "ENSG00000254749.1                               -2.130578   \n",
       "ENSG00000202195.1                               -0.000062   \n",
       "ENSG00000273966.1                               -0.000200   \n",
       "ENSG00000234763.1                               -1.425659   \n",
       "ENSG00000224447.1                               -2.926662   \n",
       "ENSG00000275776.1                               -3.470714   \n",
       "ENSG00000260004.1                               -0.461073   \n",
       "ENSG00000207027.1                               -0.000067   \n",
       "ENSG00000257837.1                               -0.850762   \n",
       "ENSG00000252977.1                               -0.000177   \n",
       "ENSG00000252411.1                               -0.000157   \n",
       "...                                                   ...   \n",
       "ENSG00000124334.17_PAR_Y                         0.000000   \n",
       "ENSG00000168939.11_PAR_Y                         0.000000   \n",
       "ENSG00000167393.17_PAR_Y                         0.000000   \n",
       "ENSG00000124333.15_PAR_Y                         0.000000   \n",
       "ENSG00000002586.18_PAR_Y                         0.000000   \n",
       "ENSG00000169084.13_PAR_Y                         0.000000   \n",
       "ENSG00000169093.15_PAR_Y                         0.000000   \n",
       "ENSG00000169100.13_PAR_Y                         0.000000   \n",
       "ENSG00000178605.13_PAR_Y                         0.000000   \n",
       "ENSG00000223484.7_PAR_Y                          0.000000   \n",
       "ENSG00000223511.6_PAR_Y                          0.000000   \n",
       "ENSG00000223571.6_PAR_Y                          0.000000   \n",
       "ENSG00000223773.7_PAR_Y                          0.000000   \n",
       "ENSG00000225661.7_PAR_Y                          0.000000   \n",
       "ENSG00000226179.6_PAR_Y                          0.000000   \n",
       "ENSG00000223274.6_PAR_Y                          0.000000   \n",
       "ENSG00000214717.10_PAR_Y                         0.000000   \n",
       "ENSG00000230542.6_PAR_Y                          0.000000   \n",
       "ENSG00000228410.6_PAR_Y                          0.000000   \n",
       "ENSG00000227159.8_PAR_Y                          0.000000   \n",
       "ENSG00000205755.11_PAR_Y                         0.000000   \n",
       "ENSG00000229232.6_PAR_Y                          0.000000   \n",
       "ENSG00000228572.7_PAR_Y                          0.000000   \n",
       "ENSG00000237801.6_PAR_Y                          0.000000   \n",
       "ENSG00000234622.6_PAR_Y                          0.000000   \n",
       "ENSG00000234958.6_PAR_Y                          0.000000   \n",
       "ENSG00000236871.7_PAR_Y                          0.000000   \n",
       "ENSG00000237040.6_PAR_Y                          0.000000   \n",
       "ENSG00000237531.6_PAR_Y                          0.000000   \n",
       "ENSG00000236017.8_PAR_Y                          0.000000   \n",
       "\n",
       "                          Std deviation (simulated)  \\\n",
       "ENSG00000231053.1                          0.000725   \n",
       "ENSG00000207135.1                          0.000600   \n",
       "ENSG00000277828.1                          0.000117   \n",
       "ENSG00000238124.1                          0.000373   \n",
       "ENSG00000225911.1                          0.000387   \n",
       "ENSG00000232529.1                          0.000183   \n",
       "ENSG00000270829.1                          0.000393   \n",
       "ENSG00000266416.1                          0.001313   \n",
       "ENSG00000200444.1                          0.001248   \n",
       "ENSG00000256717.1                          0.000167   \n",
       "ENSG00000252068.1                          0.000491   \n",
       "ENSG00000258853.1                          0.000565   \n",
       "ENSG00000200738.1                          0.001826   \n",
       "ENSG00000252951.1                          0.001541   \n",
       "ENSG00000261815.1                          0.000241   \n",
       "ENSG00000176567.1                          0.000542   \n",
       "ENSG00000261725.1                          0.000517   \n",
       "ENSG00000265291.1                          0.002511   \n",
       "ENSG00000259691.2                          0.000750   \n",
       "ENSG00000254749.1                          0.000798   \n",
       "ENSG00000202195.1                          0.000200   \n",
       "ENSG00000273966.1                          0.000429   \n",
       "ENSG00000234763.1                          0.000661   \n",
       "ENSG00000224447.1                          0.000841   \n",
       "ENSG00000275776.1                          0.001224   \n",
       "ENSG00000260004.1                          0.000232   \n",
       "ENSG00000207027.1                          0.000201   \n",
       "ENSG00000257837.1                          0.000380   \n",
       "ENSG00000252977.1                          0.000419   \n",
       "ENSG00000252411.1                          0.000409   \n",
       "...                                             ...   \n",
       "ENSG00000124334.17_PAR_Y                   0.000000   \n",
       "ENSG00000168939.11_PAR_Y                   0.000000   \n",
       "ENSG00000167393.17_PAR_Y                   0.000000   \n",
       "ENSG00000124333.15_PAR_Y                   0.000000   \n",
       "ENSG00000002586.18_PAR_Y                   0.000000   \n",
       "ENSG00000169084.13_PAR_Y                   0.000000   \n",
       "ENSG00000169093.15_PAR_Y                   0.000000   \n",
       "ENSG00000169100.13_PAR_Y                   0.000000   \n",
       "ENSG00000178605.13_PAR_Y                   0.000000   \n",
       "ENSG00000223484.7_PAR_Y                    0.000000   \n",
       "ENSG00000223511.6_PAR_Y                    0.000000   \n",
       "ENSG00000223571.6_PAR_Y                    0.000000   \n",
       "ENSG00000223773.7_PAR_Y                    0.000000   \n",
       "ENSG00000225661.7_PAR_Y                    0.000000   \n",
       "ENSG00000226179.6_PAR_Y                    0.000000   \n",
       "ENSG00000223274.6_PAR_Y                    0.000000   \n",
       "ENSG00000214717.10_PAR_Y                   0.000000   \n",
       "ENSG00000230542.6_PAR_Y                    0.000000   \n",
       "ENSG00000228410.6_PAR_Y                    0.000000   \n",
       "ENSG00000227159.8_PAR_Y                    0.000000   \n",
       "ENSG00000205755.11_PAR_Y                   0.000000   \n",
       "ENSG00000229232.6_PAR_Y                    0.000000   \n",
       "ENSG00000228572.7_PAR_Y                    0.000000   \n",
       "ENSG00000237801.6_PAR_Y                    0.000000   \n",
       "ENSG00000234622.6_PAR_Y                    0.000000   \n",
       "ENSG00000234958.6_PAR_Y                    0.000000   \n",
       "ENSG00000236871.7_PAR_Y                    0.000000   \n",
       "ENSG00000237040.6_PAR_Y                    0.000000   \n",
       "ENSG00000237531.6_PAR_Y                    0.000000   \n",
       "ENSG00000236017.8_PAR_Y                    0.000000   \n",
       "\n",
       "                          Number of experiments (simulated)      Z score  \n",
       "ENSG00000231053.1                                        25  1803.750894  \n",
       "ENSG00000207135.1                                        25  1693.138607  \n",
       "ENSG00000277828.1                                        25  1601.296648  \n",
       "ENSG00000238124.1                                        25  1579.645423  \n",
       "ENSG00000225911.1                                        25  1396.811969  \n",
       "ENSG00000232529.1                                        25  1311.243896  \n",
       "ENSG00000270829.1                                        25  1306.920278  \n",
       "ENSG00000266416.1                                        25  1274.033372  \n",
       "ENSG00000200444.1                                        25  1253.896058  \n",
       "ENSG00000256717.1                                        25  1225.788027  \n",
       "ENSG00000252068.1                                        25  1195.774083  \n",
       "ENSG00000258853.1                                        25  1176.608104  \n",
       "ENSG00000200738.1                                        25  1175.179030  \n",
       "ENSG00000252951.1                                        25  1165.887982  \n",
       "ENSG00000261815.1                                        25  1165.376390  \n",
       "ENSG00000176567.1                                        25  1142.115932  \n",
       "ENSG00000261725.1                                        25  1139.597159  \n",
       "ENSG00000265291.1                                        25  1139.139154  \n",
       "ENSG00000259691.2                                        25  1136.364192  \n",
       "ENSG00000254749.1                                        25  1126.099895  \n",
       "ENSG00000202195.1                                        25  1079.641496  \n",
       "ENSG00000273966.1                                        25  1078.525877  \n",
       "ENSG00000234763.1                                        25  1063.829721  \n",
       "ENSG00000224447.1                                        25  1063.132505  \n",
       "ENSG00000275776.1                                        25  1056.332978  \n",
       "ENSG00000260004.1                                        25  1047.512553  \n",
       "ENSG00000207027.1                                        25  1043.424778  \n",
       "ENSG00000257837.1                                        25  1023.938992  \n",
       "ENSG00000252977.1                                        25  1021.548536  \n",
       "ENSG00000252411.1                                        25  1005.832161  \n",
       "...                                                     ...          ...  \n",
       "ENSG00000124334.17_PAR_Y                                 25          NaN  \n",
       "ENSG00000168939.11_PAR_Y                                 25          NaN  \n",
       "ENSG00000167393.17_PAR_Y                                 25          NaN  \n",
       "ENSG00000124333.15_PAR_Y                                 25          NaN  \n",
       "ENSG00000002586.18_PAR_Y                                 25          NaN  \n",
       "ENSG00000169084.13_PAR_Y                                 25          NaN  \n",
       "ENSG00000169093.15_PAR_Y                                 25          NaN  \n",
       "ENSG00000169100.13_PAR_Y                                 25          NaN  \n",
       "ENSG00000178605.13_PAR_Y                                 25          NaN  \n",
       "ENSG00000223484.7_PAR_Y                                  25          NaN  \n",
       "ENSG00000223511.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000223571.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000223773.7_PAR_Y                                  25          NaN  \n",
       "ENSG00000225661.7_PAR_Y                                  25          NaN  \n",
       "ENSG00000226179.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000223274.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000214717.10_PAR_Y                                 25          NaN  \n",
       "ENSG00000230542.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000228410.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000227159.8_PAR_Y                                  25          NaN  \n",
       "ENSG00000205755.11_PAR_Y                                 25          NaN  \n",
       "ENSG00000229232.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000228572.7_PAR_Y                                  25          NaN  \n",
       "ENSG00000237801.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000234622.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000234958.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000236871.7_PAR_Y                                  25          NaN  \n",
       "ENSG00000237040.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000237531.6_PAR_Y                                  25          NaN  \n",
       "ENSG00000236017.8_PAR_Y                                  25          NaN  \n",
       "\n",
       "[58037 rows x 10 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary.sort_values(by=\"Z score\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save file\n",
    "summary_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"gene_summary_table.tsv\")\n",
    "\n",
    "summary.to_csv(summary_file, float_format='%.5f', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene set enrichment analysis\n",
    "\n",
    "**Goal:** To detect modest but coordinated changes in prespecified sets of related genes (i.e. those genes in the same pathway or share the same GO term).\n",
    "\n",
    "1. Ranks all genes based using DE association statistics. In this case we used the p-value scores to rank genes. logFC returned error -- need to look into this.\n",
    "2. An enrichment score (ES) is defined as the maximum distance from the middle of the ranked list. Thus, the enrichment score indicates whether the genes contained in a gene set are clustered towards the beginning or the end of the ranked list (indicating a correlation with change in expression). \n",
    "3. Estimate the statistical significance of the ES by a phenotypic-based permutation test in order to produce a null distribution for the ES( i.e. scores based on permuted phenotype)\n",
    "\n",
    "**Note:** Since there were 0 differentially expressed genes using simulated experiments, we used gene set enrichement analysis instead of over-representation analysis to get ranking of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load Hallmark pathways database used by Powers et. al.\n",
    "# https://github.com/CostelloLab/GSEA-InContext/blob/master/data/gene_sets/hallmarks.gmt\n",
    "hallmark_DB_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"hallmark_DB.gmt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Registered S3 method overwritten by 'enrichplot':\n",
      "  method               from\n",
      "  fortify.enrichResult DOSE\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: clusterProfiler v3.14.3  For help: https://guangchuangyu.github.io/software/clusterProfiler\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: AnnotationDbi\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: stats4\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: BiocGenerics\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: parallel\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: Biobase\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: IRanges\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: S4Vectors\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: DOSE v3.12.0  For help: https://guangchuangyu.github.io/software/DOSE\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "suppressWarnings(library(\"clusterProfiler\"))\n",
    "suppressWarnings(library(\"org.Hs.eg.db\"))\n",
    "suppressWarnings(library(\"DOSE\"))\n",
    "suppressWarnings(library(\"biomaRt\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get pathway enrichment for template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Cache found\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/robjects/pandas2ri.py:191: FutureWarning: from_items is deprecated. Please use DataFrame.from_dict(dict(items), ...) instead. DataFrame.from_dict(OrderedDict(items)) may be used to preserve the key order.\n",
      "  res = PandasDataFrame.from_items(items)\n"
     ]
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file -o gene_id_mapping\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "gene_id_mapping <- get_ensembl_symbol_mapping(template_DE_stats_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(57210, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000002330</th>\n",
       "      <td>BAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000003137</th>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000003249</th>\n",
       "      <td>DBNDD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000004799</th>\n",
       "      <td>PDK4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000006062</th>\n",
       "      <td>MAP3K14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                hgnc_symbol\n",
       "ensembl_gene_id            \n",
       "ENSG00000002330         BAD\n",
       "ENSG00000003137     CYP26B1\n",
       "ENSG00000003249      DBNDD1\n",
       "ENSG00000004799        PDK4\n",
       "ENSG00000006062     MAP3K14"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Set ensembl id as index\n",
    "gene_id_mapping.set_index(\"ensembl_gene_id\", inplace=True)\n",
    "print(gene_id_mapping.shape)\n",
    "gene_id_mapping.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace ensembl ids with gene symbols\n",
    "utils.replace_ensembl_ids(template_DE_stats_file,\n",
    "                          gene_id_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i template_DE_stats_file -i hallmark_DB_file -o template_enriched_pathways\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "template_enriched_pathways <- find_enriched_pathways(template_DE_stats_file, hallmark_DB_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>189</td>\n",
       "      <td>0.449242</td>\n",
       "      <td>1.376190</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>13232</td>\n",
       "      <td>tags=47%, list=34%, signal=31%</td>\n",
       "      <td>AK2/UQCR10/CCNG2/ECHS1/MDH2/CD151/ANGPT1/TALDO...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HALLMARK_DNA_REPAIR</td>\n",
       "      <td>HALLMARK_DNA_REPAIR</td>\n",
       "      <td>141</td>\n",
       "      <td>0.477451</td>\n",
       "      <td>1.443748</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>10751</td>\n",
       "      <td>tags=43%, list=28%, signal=31%</td>\n",
       "      <td>SUPT4H1/UMPS/POLR3C/RFC5/SEC61A1/DGUOK/IMPDH2/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>194</td>\n",
       "      <td>0.542046</td>\n",
       "      <td>1.663152</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>12057</td>\n",
       "      <td>tags=62%, list=31%, signal=43%</td>\n",
       "      <td>AK2/TUBB/SRSF2/MCM2/RPA1/LYAR/SLBP/DCK/KIF2C/H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>190</td>\n",
       "      <td>0.523181</td>\n",
       "      <td>1.602536</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>13134</td>\n",
       "      <td>tags=62%, list=34%, signal=41%</td>\n",
       "      <td>BUB3/PRMT5/SRSF2/CUL4A/MCM2/MNAT1/UCK2/CUL1/SM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HALLMARK_MITOTIC_SPINDLE</td>\n",
       "      <td>HALLMARK_MITOTIC_SPINDLE</td>\n",
       "      <td>197</td>\n",
       "      <td>0.466652</td>\n",
       "      <td>1.431851</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>12750</td>\n",
       "      <td>tags=48%, list=33%, signal=32%</td>\n",
       "      <td>DYNLL2/ARHGEF7/UXT/PLEKHG2/SPTAN1/KIF15/APC/KI...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         ID               Description  setSize  \\\n",
       "0     HALLMARK_ADIPOGENESIS     HALLMARK_ADIPOGENESIS      189   \n",
       "1       HALLMARK_DNA_REPAIR       HALLMARK_DNA_REPAIR      141   \n",
       "2      HALLMARK_E2F_TARGETS      HALLMARK_E2F_TARGETS      194   \n",
       "3   HALLMARK_G2M_CHECKPOINT   HALLMARK_G2M_CHECKPOINT      190   \n",
       "4  HALLMARK_MITOTIC_SPINDLE  HALLMARK_MITOTIC_SPINDLE      197   \n",
       "\n",
       "   enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "0         0.449242  1.376190  0.000999  0.004163  0.002366  13232   \n",
       "1         0.477451  1.443748  0.000999  0.004163  0.002366  10751   \n",
       "2         0.542046  1.663152  0.000999  0.004163  0.002366  12057   \n",
       "3         0.523181  1.602536  0.000999  0.004163  0.002366  13134   \n",
       "4         0.466652  1.431851  0.000999  0.004163  0.002366  12750   \n",
       "\n",
       "                     leading_edge  \\\n",
       "0  tags=47%, list=34%, signal=31%   \n",
       "1  tags=43%, list=28%, signal=31%   \n",
       "2  tags=62%, list=31%, signal=43%   \n",
       "3  tags=62%, list=34%, signal=41%   \n",
       "4  tags=48%, list=33%, signal=32%   \n",
       "\n",
       "                                     core_enrichment  \n",
       "0  AK2/UQCR10/CCNG2/ECHS1/MDH2/CD151/ANGPT1/TALDO...  \n",
       "1  SUPT4H1/UMPS/POLR3C/RFC5/SEC61A1/DGUOK/IMPDH2/...  \n",
       "2  AK2/TUBB/SRSF2/MCM2/RPA1/LYAR/SLBP/DCK/KIF2C/H...  \n",
       "3  BUB3/PRMT5/SRSF2/CUL4A/MCM2/MNAT1/UCK2/CUL1/SM...  \n",
       "4  DYNLL2/ARHGEF7/UXT/PLEKHG2/SPTAN1/KIF15/APC/KI...  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get pathway enrichment for simulated experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace ensembl ids with gene symbols\n",
    "if rerun_simulated:\n",
    "    for i in range(num_runs):\n",
    "        simulated_DE_stats_file = os.path.join(\n",
    "            local_dir, \n",
    "            \"DE_stats\",\n",
    "            \"DE_stats_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "\n",
    "        utils.replace_ensembl_ids(simulated_DE_stats_file,\n",
    "                                  gene_id_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R -i project_id -i local_dir -i hallmark_DB_file -i num_runs -i rerun_simulated\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_DE_stats_file <- paste(local_dir, \n",
    "                                 \"DE_stats/DE_stats_simulated_data_\", \n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    \n",
    "    out_file = paste(local_dir, \n",
    "                     \"GSEA_stats/GSEA_simulated_data_\",\n",
    "                     project_id,\n",
    "                     \"_\",\n",
    "                     i,\n",
    "                     \".txt\", \n",
    "                     sep=\"\")\n",
    "    \n",
    "    if (rerun_simulated){\n",
    "        enriched_pathways <- find_enriched_pathways(simulated_DE_stats_file, hallmark_DB_file) \n",
    "    \n",
    "        write.table(enriched_pathways, file = out_file, row.names = T, sep = \"\\t\", quote = F)\n",
    "        }\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "Again, we want to compare our ranked pathways found against what was reported in the original publication.\n",
    "\n",
    "*The DEX-responsive genes that we identified are primarily implicated in two broad classes: stress response and development (Table 1; Reimand et al. 2007). Comparison to Gene Ontology (GO) categories (Ashburner et al. 2000) revealed that the identified genes are involved in stress response (P = 6 × 10−11), organ development (P = 5 × 10−15), cell differentiation (P = 1 × 10−11), hormone secretion (P = 4 × 10−7), and apoptosis (P = 5 × 10−12).*\n",
    "\n",
    "We found pathways that are consistent with what publication found: pathways related to anatomical structure (i.e. vinculin, ) and cell differentiation (i.e. centromeric sister chromatid cohesion), hormone secretion (i.e. somatic hypermutation of immunoglobulin genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get statistics for enriched pathways\n",
    "Examine the enriched pathways identified from template experiment -- How are these enriched pathways ranked in the simulated experiments?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_to_rank = 'enrichmentScore'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</th>\n",
       "      <td>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</td>\n",
       "      <td>106</td>\n",
       "      <td>0.572412</td>\n",
       "      <td>1.706215</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>9573</td>\n",
       "      <td>tags=52%, list=25%, signal=39%</td>\n",
       "      <td>NFYA/NFYB/ALDH18A1/ATF4/NOP14/NHP2/CNOT2/PARN/...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYC_TARGETS_V1</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V1</td>\n",
       "      <td>193</td>\n",
       "      <td>0.548231</td>\n",
       "      <td>1.681245</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>12807</td>\n",
       "      <td>tags=71%, list=33%, signal=48%</td>\n",
       "      <td>PSMC4/BUB3/PWP1/SRSF2/APEX1/RPS2/MCM2/SRSF7/NH...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</th>\n",
       "      <td>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</td>\n",
       "      <td>46</td>\n",
       "      <td>0.542762</td>\n",
       "      <td>1.541286</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>11423</td>\n",
       "      <td>tags=52%, list=29%, signal=37%</td>\n",
       "      <td>FTL/G6PD/NDUFS2/ABCC1/SRXN1/NDUFA6/NQO1/PFKP/I...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_E2F_TARGETS</th>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>194</td>\n",
       "      <td>0.542046</td>\n",
       "      <td>1.663152</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>12057</td>\n",
       "      <td>tags=62%, list=31%, signal=43%</td>\n",
       "      <td>AK2/TUBB/SRSF2/MCM2/RPA1/LYAR/SLBP/DCK/KIF2C/H...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_PROTEIN_SECRETION</th>\n",
       "      <td>HALLMARK_PROTEIN_SECRETION</td>\n",
       "      <td>95</td>\n",
       "      <td>0.526593</td>\n",
       "      <td>1.561999</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>12685</td>\n",
       "      <td>tags=62%, list=33%, signal=42%</td>\n",
       "      <td>VAMP7/GNAS/AP2M1/MAPK1/STAM/AP2S1/SGMS1/CAV2/S...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                       Description  \\\n",
       "ID                                                                                   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE              HALLMARK_UNFOLDED_PROTEIN_RESPONSE   \n",
       "HALLMARK_MYC_TARGETS_V1                                    HALLMARK_MYC_TARGETS_V1   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY   \n",
       "HALLMARK_E2F_TARGETS                                          HALLMARK_E2F_TARGETS   \n",
       "HALLMARK_PROTEIN_SECRETION                              HALLMARK_PROTEIN_SECRETION   \n",
       "\n",
       "                                          setSize  enrichmentScore       NES  \\\n",
       "ID                                                                             \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE            106         0.572412  1.706215   \n",
       "HALLMARK_MYC_TARGETS_V1                       193         0.548231  1.681245   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY       46         0.542762  1.541286   \n",
       "HALLMARK_E2F_TARGETS                          194         0.542046  1.663152   \n",
       "HALLMARK_PROTEIN_SECRETION                     95         0.526593  1.561999   \n",
       "\n",
       "                                            pvalue  p.adjust   qvalues   rank  \\\n",
       "ID                                                                              \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE        0.000999  0.004163  0.002366   9573   \n",
       "HALLMARK_MYC_TARGETS_V1                   0.000999  0.004163  0.002366  12807   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  0.000999  0.004163  0.002366  11423   \n",
       "HALLMARK_E2F_TARGETS                      0.000999  0.004163  0.002366  12057   \n",
       "HALLMARK_PROTEIN_SECRETION                0.000999  0.004163  0.002366  12685   \n",
       "\n",
       "                                                            leading_edge  \\\n",
       "ID                                                                         \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE        tags=52%, list=25%, signal=39%   \n",
       "HALLMARK_MYC_TARGETS_V1                   tags=71%, list=33%, signal=48%   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  tags=52%, list=29%, signal=37%   \n",
       "HALLMARK_E2F_TARGETS                      tags=62%, list=31%, signal=43%   \n",
       "HALLMARK_PROTEIN_SECRETION                tags=62%, list=33%, signal=42%   \n",
       "\n",
       "                                                                            core_enrichment  \\\n",
       "ID                                                                                            \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE        NFYA/NFYB/ALDH18A1/ATF4/NOP14/NHP2/CNOT2/PARN/...   \n",
       "HALLMARK_MYC_TARGETS_V1                   PSMC4/BUB3/PWP1/SRSF2/APEX1/RPS2/MCM2/SRSF7/NH...   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  FTL/G6PD/NDUFS2/ABCC1/SRXN1/NDUFA6/NQO1/PFKP/I...   \n",
       "HALLMARK_E2F_TARGETS                      AK2/TUBB/SRSF2/MCM2/RPA1/LYAR/SLBP/DCK/KIF2C/H...   \n",
       "HALLMARK_PROTEIN_SECRETION                VAMP7/GNAS/AP2M1/MAPK1/STAM/AP2S1/SGMS1/CAV2/S...   \n",
       "\n",
       "                                          ranking  \n",
       "ID                                                 \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE            1.0  \n",
       "HALLMARK_MYC_TARGETS_V1                       2.0  \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY      3.0  \n",
       "HALLMARK_E2F_TARGETS                          4.0  \n",
       "HALLMARK_PROTEIN_SECRETION                    5.0  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get ranks of template experiment\n",
    "# Rank pathways by highest enrichment score\n",
    "template_enriched_pathways['ranking'] = template_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "template_enriched_pathways = template_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "\n",
    "# Set index to GO ID\n",
    "template_enriched_pathways.set_index(\"ID\", inplace=True)\n",
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check that GO IDs are unique\n",
    "template_enriched_pathways.index.nunique() == len(template_enriched_pathways)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simulated experiments "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(857, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYC_TARGETS_V1</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V1</td>\n",
       "      <td>HALLMARK_MYC_TARGETS_V1</td>\n",
       "      <td>193</td>\n",
       "      <td>0.556160</td>\n",
       "      <td>1.724459</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>0.000263</td>\n",
       "      <td>14297</td>\n",
       "      <td>tags=85%, list=37%, signal=54%</td>\n",
       "      <td>PPM1G/BUB3/TCP1/RPL22/RPS2/PSMA1/SNRPA1/TFDP1/...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_E2F_TARGETS</th>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>194</td>\n",
       "      <td>0.524376</td>\n",
       "      <td>1.626481</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>0.000263</td>\n",
       "      <td>13584</td>\n",
       "      <td>tags=75%, list=35%, signal=49%</td>\n",
       "      <td>AURKA/RACGAP1/CDKN1A/CDKN3/ATAD2/TIPIN/PPM1D/T...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYC_TARGETS_V2</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V2</td>\n",
       "      <td>HALLMARK_MYC_TARGETS_V2</td>\n",
       "      <td>58</td>\n",
       "      <td>0.517846</td>\n",
       "      <td>1.512182</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>0.000263</td>\n",
       "      <td>13264</td>\n",
       "      <td>tags=71%, list=34%, signal=47%</td>\n",
       "      <td>SLC19A1/NIP7/MRTO4/LAS1L/HSPE1/WDR74/MYBBP1A/M...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</th>\n",
       "      <td>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</td>\n",
       "      <td>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</td>\n",
       "      <td>46</td>\n",
       "      <td>0.516290</td>\n",
       "      <td>1.482771</td>\n",
       "      <td>0.002000</td>\n",
       "      <td>0.003125</td>\n",
       "      <td>0.000329</td>\n",
       "      <td>16437</td>\n",
       "      <td>tags=85%, list=42%, signal=49%</td>\n",
       "      <td>PRDX1/SBNO2/JUNB/MBP/CDKN2D/GPX3/ABCC1/SOD1/PR...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_G2M_CHECKPOINT</th>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>190</td>\n",
       "      <td>0.506406</td>\n",
       "      <td>1.570335</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>0.000263</td>\n",
       "      <td>15491</td>\n",
       "      <td>tags=77%, list=40%, signal=47%</td>\n",
       "      <td>AURKA/ODF2/RACGAP1/NCL/BUB3/CDKN3/E2F1/TNPO2/S...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                ID  \\\n",
       "HALLMARK_MYC_TARGETS_V1                                    HALLMARK_MYC_TARGETS_V1   \n",
       "HALLMARK_E2F_TARGETS                                          HALLMARK_E2F_TARGETS   \n",
       "HALLMARK_MYC_TARGETS_V2                                    HALLMARK_MYC_TARGETS_V2   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY   \n",
       "HALLMARK_G2M_CHECKPOINT                                    HALLMARK_G2M_CHECKPOINT   \n",
       "\n",
       "                                                                       Description  \\\n",
       "HALLMARK_MYC_TARGETS_V1                                    HALLMARK_MYC_TARGETS_V1   \n",
       "HALLMARK_E2F_TARGETS                                          HALLMARK_E2F_TARGETS   \n",
       "HALLMARK_MYC_TARGETS_V2                                    HALLMARK_MYC_TARGETS_V2   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY   \n",
       "HALLMARK_G2M_CHECKPOINT                                    HALLMARK_G2M_CHECKPOINT   \n",
       "\n",
       "                                          setSize  enrichmentScore       NES  \\\n",
       "HALLMARK_MYC_TARGETS_V1                       193         0.556160  1.724459   \n",
       "HALLMARK_E2F_TARGETS                          194         0.524376  1.626481   \n",
       "HALLMARK_MYC_TARGETS_V2                        58         0.517846  1.512182   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY       46         0.516290  1.482771   \n",
       "HALLMARK_G2M_CHECKPOINT                       190         0.506406  1.570335   \n",
       "\n",
       "                                            pvalue  p.adjust   qvalues   rank  \\\n",
       "HALLMARK_MYC_TARGETS_V1                   0.000999  0.002498  0.000263  14297   \n",
       "HALLMARK_E2F_TARGETS                      0.000999  0.002498  0.000263  13584   \n",
       "HALLMARK_MYC_TARGETS_V2                   0.000999  0.002498  0.000263  13264   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  0.002000  0.003125  0.000329  16437   \n",
       "HALLMARK_G2M_CHECKPOINT                   0.000999  0.002498  0.000263  15491   \n",
       "\n",
       "                                                            leading_edge  \\\n",
       "HALLMARK_MYC_TARGETS_V1                   tags=85%, list=37%, signal=54%   \n",
       "HALLMARK_E2F_TARGETS                      tags=75%, list=35%, signal=49%   \n",
       "HALLMARK_MYC_TARGETS_V2                   tags=71%, list=34%, signal=47%   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  tags=85%, list=42%, signal=49%   \n",
       "HALLMARK_G2M_CHECKPOINT                   tags=77%, list=40%, signal=47%   \n",
       "\n",
       "                                                                            core_enrichment  \\\n",
       "HALLMARK_MYC_TARGETS_V1                   PPM1G/BUB3/TCP1/RPL22/RPS2/PSMA1/SNRPA1/TFDP1/...   \n",
       "HALLMARK_E2F_TARGETS                      AURKA/RACGAP1/CDKN1A/CDKN3/ATAD2/TIPIN/PPM1D/T...   \n",
       "HALLMARK_MYC_TARGETS_V2                   SLC19A1/NIP7/MRTO4/LAS1L/HSPE1/WDR74/MYBBP1A/M...   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY  PRDX1/SBNO2/JUNB/MBP/CDKN2D/GPX3/ABCC1/SOD1/PR...   \n",
       "HALLMARK_G2M_CHECKPOINT                   AURKA/ODF2/RACGAP1/NCL/BUB3/CDKN3/E2F1/TNPO2/S...   \n",
       "\n",
       "                                          ranking  \n",
       "HALLMARK_MYC_TARGETS_V1                       1.0  \n",
       "HALLMARK_E2F_TARGETS                          2.0  \n",
       "HALLMARK_MYC_TARGETS_V2                       3.0  \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY      4.0  \n",
       "HALLMARK_G2M_CHECKPOINT                       5.0  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concatenate simulated experiments\n",
    "simulated_enriched_pathways_all = pd.DataFrame()\n",
    "for i in range(num_runs):\n",
    "    simulated_GSEA_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"GSEA_stats\",\n",
    "        \"GSEA_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "    \n",
    "    #Read results\n",
    "    simulated_enriched_pathways = pd.read_csv(\n",
    "        simulated_GSEA_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "    \n",
    "    # Add ranks of simulated experiment\n",
    "    simulated_enriched_pathways['ranking'] = simulated_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "    simulated_enriched_pathways = simulated_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "    \n",
    "    # Concatenate df\n",
    "    simulated_enriched_pathways_all = pd.concat([simulated_enriched_pathways_all,\n",
    "                                       simulated_enriched_pathways])\n",
    "    \n",
    "print(simulated_enriched_pathways_all.shape)\n",
    "simulated_enriched_pathways_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">enrichmentScore</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>count</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ADIPOGENESIS</th>\n",
       "      <td>0.442502</td>\n",
       "      <td>0.026254</td>\n",
       "      <td>25</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ALLOGRAFT_REJECTION</th>\n",
       "      <td>0.424805</td>\n",
       "      <td>0.035230</td>\n",
       "      <td>19</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANDROGEN_RESPONSE</th>\n",
       "      <td>0.452496</td>\n",
       "      <td>0.025508</td>\n",
       "      <td>18</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANGIOGENESIS</th>\n",
       "      <td>0.512233</td>\n",
       "      <td>0.019648</td>\n",
       "      <td>5</td>\n",
       "      <td>0.012987</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_APICAL_JUNCTION</th>\n",
       "      <td>0.425509</td>\n",
       "      <td>0.028067</td>\n",
       "      <td>20</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>25.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             enrichmentScore                    pvalue ranking\n",
       "                                        mean       std count    median  median\n",
       "ID                                                                            \n",
       "HALLMARK_ADIPOGENESIS               0.442502  0.026254    25  0.000999    16.0\n",
       "HALLMARK_ALLOGRAFT_REJECTION        0.424805  0.035230    19  0.000999    27.0\n",
       "HALLMARK_ANDROGEN_RESPONSE          0.452496  0.025508    18  0.002498    17.0\n",
       "HALLMARK_ANGIOGENESIS               0.512233  0.019648     5  0.012987     6.0\n",
       "HALLMARK_APICAL_JUNCTION            0.425509  0.028067    20  0.000999    25.5"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simulated_enriched_pathways_stats = simulated_enriched_pathways_all.groupby(['ID'])[['enrichmentScore', 'pvalue', 'ranking']].agg({\n",
    "    col_to_rank:['mean', 'std','count'],\n",
    "    'pvalue':['median'],\n",
    "    'ranking':['median']\n",
    "})\n",
    "\n",
    "simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ranking</th>\n",
       "      <th>(enrichmentScore, mean)</th>\n",
       "      <th>(enrichmentScore, std)</th>\n",
       "      <th>(enrichmentScore, count)</th>\n",
       "      <th>(pvalue, median)</th>\n",
       "      <th>(ranking, median)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ADIPOGENESIS</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>189.0</td>\n",
       "      <td>0.449242</td>\n",
       "      <td>1.376190</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.004163</td>\n",
       "      <td>0.002366</td>\n",
       "      <td>13232.0</td>\n",
       "      <td>tags=47%, list=34%, signal=31%</td>\n",
       "      <td>AK2/UQCR10/CCNG2/ECHS1/MDH2/CD151/ANGPT1/TALDO...</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.442502</td>\n",
       "      <td>0.026254</td>\n",
       "      <td>25</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ALLOGRAFT_REJECTION</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.424805</td>\n",
       "      <td>0.035230</td>\n",
       "      <td>19</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANDROGEN_RESPONSE</th>\n",
       "      <td>HALLMARK_ANDROGEN_RESPONSE</td>\n",
       "      <td>96.0</td>\n",
       "      <td>0.444627</td>\n",
       "      <td>1.319452</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>0.029382</td>\n",
       "      <td>0.016702</td>\n",
       "      <td>12826.0</td>\n",
       "      <td>tags=46%, list=33%, signal=31%</td>\n",
       "      <td>UAP1/UBE2J1/HERC3/RAB4A/GNAI3/ELK4/TPD52/APPBP...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.452496</td>\n",
       "      <td>0.025508</td>\n",
       "      <td>18</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANGIOGENESIS</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.512233</td>\n",
       "      <td>0.019648</td>\n",
       "      <td>5</td>\n",
       "      <td>0.012987</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_APICAL_JUNCTION</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.425509</td>\n",
       "      <td>0.028067</td>\n",
       "      <td>20</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>25.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Description  setSize  \\\n",
       "ID                                                                  \n",
       "HALLMARK_ADIPOGENESIS              HALLMARK_ADIPOGENESIS    189.0   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                         NaN      NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE    HALLMARK_ANDROGEN_RESPONSE     96.0   \n",
       "HALLMARK_ANGIOGENESIS                                NaN      NaN   \n",
       "HALLMARK_APICAL_JUNCTION                             NaN      NaN   \n",
       "\n",
       "                              enrichmentScore       NES    pvalue  p.adjust  \\\n",
       "ID                                                                            \n",
       "HALLMARK_ADIPOGENESIS                0.449242  1.376190  0.000999  0.004163   \n",
       "HALLMARK_ALLOGRAFT_REJECTION              NaN       NaN       NaN       NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE           0.444627  1.319452  0.009990  0.029382   \n",
       "HALLMARK_ANGIOGENESIS                     NaN       NaN       NaN       NaN   \n",
       "HALLMARK_APICAL_JUNCTION                  NaN       NaN       NaN       NaN   \n",
       "\n",
       "                               qvalues     rank  \\\n",
       "ID                                                \n",
       "HALLMARK_ADIPOGENESIS         0.002366  13232.0   \n",
       "HALLMARK_ALLOGRAFT_REJECTION       NaN      NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE    0.016702  12826.0   \n",
       "HALLMARK_ANGIOGENESIS              NaN      NaN   \n",
       "HALLMARK_APICAL_JUNCTION           NaN      NaN   \n",
       "\n",
       "                                                leading_edge  \\\n",
       "ID                                                             \n",
       "HALLMARK_ADIPOGENESIS         tags=47%, list=34%, signal=31%   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                             NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE    tags=46%, list=33%, signal=31%   \n",
       "HALLMARK_ANGIOGENESIS                                    NaN   \n",
       "HALLMARK_APICAL_JUNCTION                                 NaN   \n",
       "\n",
       "                                                                core_enrichment  \\\n",
       "ID                                                                                \n",
       "HALLMARK_ADIPOGENESIS         AK2/UQCR10/CCNG2/ECHS1/MDH2/CD151/ANGPT1/TALDO...   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                                NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE    UAP1/UBE2J1/HERC3/RAB4A/GNAI3/ELK4/TPD52/APPBP...   \n",
       "HALLMARK_ANGIOGENESIS                                                       NaN   \n",
       "HALLMARK_APICAL_JUNCTION                                                    NaN   \n",
       "\n",
       "                              ranking  (enrichmentScore, mean)  \\\n",
       "ID                                                               \n",
       "HALLMARK_ADIPOGENESIS            14.0                 0.442502   \n",
       "HALLMARK_ALLOGRAFT_REJECTION      NaN                 0.424805   \n",
       "HALLMARK_ANDROGEN_RESPONSE       15.0                 0.452496   \n",
       "HALLMARK_ANGIOGENESIS             NaN                 0.512233   \n",
       "HALLMARK_APICAL_JUNCTION          NaN                 0.425509   \n",
       "\n",
       "                              (enrichmentScore, std)  \\\n",
       "ID                                                     \n",
       "HALLMARK_ADIPOGENESIS                       0.026254   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                0.035230   \n",
       "HALLMARK_ANDROGEN_RESPONSE                  0.025508   \n",
       "HALLMARK_ANGIOGENESIS                       0.019648   \n",
       "HALLMARK_APICAL_JUNCTION                    0.028067   \n",
       "\n",
       "                              (enrichmentScore, count)  (pvalue, median)  \\\n",
       "ID                                                                         \n",
       "HALLMARK_ADIPOGENESIS                               25          0.000999   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                        19          0.000999   \n",
       "HALLMARK_ANDROGEN_RESPONSE                          18          0.002498   \n",
       "HALLMARK_ANGIOGENESIS                                5          0.012987   \n",
       "HALLMARK_APICAL_JUNCTION                            20          0.000999   \n",
       "\n",
       "                              (ranking, median)  \n",
       "ID                                               \n",
       "HALLMARK_ADIPOGENESIS                      16.0  \n",
       "HALLMARK_ALLOGRAFT_REJECTION               27.0  \n",
       "HALLMARK_ANDROGEN_RESPONSE                 17.0  \n",
       "HALLMARK_ANGIOGENESIS                       6.0  \n",
       "HALLMARK_APICAL_JUNCTION                   25.5  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge template statistics with simulated statistics\n",
    "template_simulated_enriched_pathways_stats = template_enriched_pathways.merge(simulated_enriched_pathways_stats, \n",
    "                                                                              how='outer',\n",
    "                                                                              left_index=True,\n",
    "                                                                              right_index=True)\n",
    "template_simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse columns\n",
    "median_pval_simulated = template_simulated_enriched_pathways_stats[('pvalue','median')]\n",
    "median_rank_simulated = template_simulated_enriched_pathways_stats[('ranking','median')]\n",
    "mean_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'mean')]\n",
    "std_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'std')]\n",
    "count_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'count')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculations for summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#init_datatables_mode()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "require": [
     "base/js/events",
     "datatables.net",
     "d3",
     "chartjs",
     "dt-config",
     "dt-components",
     "dt-graph-objects",
     "dt-toolbar",
     "dt-tooltips",
     "jupyter-datatables"
    ],
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pathway</th>\n",
       "      <th>P-value (Real)</th>\n",
       "      <th>Rank (Real)</th>\n",
       "      <th>Test statistic (Real)</th>\n",
       "      <th>Median p-value (simulated)</th>\n",
       "      <th>Median rank (simulated)</th>\n",
       "      <th>Mean test statistic (simulated)</th>\n",
       "      <th>Std deviation (simulated)</th>\n",
       "      <th>Number of experiments (simulated)</th>\n",
       "      <th>Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ADIPOGENESIS</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.449242</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.442502</td>\n",
       "      <td>0.026254</td>\n",
       "      <td>25</td>\n",
       "      <td>0.256710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ALLOGRAFT_REJECTION</th>\n",
       "      <td>HALLMARK_ALLOGRAFT_REJECTION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.424805</td>\n",
       "      <td>0.035230</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANDROGEN_RESPONSE</th>\n",
       "      <td>HALLMARK_ANDROGEN_RESPONSE</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.444627</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.452496</td>\n",
       "      <td>0.025508</td>\n",
       "      <td>18</td>\n",
       "      <td>-0.308466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ANGIOGENESIS</th>\n",
       "      <td>HALLMARK_ANGIOGENESIS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012987</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.512233</td>\n",
       "      <td>0.019648</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_APICAL_JUNCTION</th>\n",
       "      <td>HALLMARK_APICAL_JUNCTION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>25.5</td>\n",
       "      <td>0.425509</td>\n",
       "      <td>0.028067</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_APICAL_SURFACE</th>\n",
       "      <td>HALLMARK_APICAL_SURFACE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.034466</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.471780</td>\n",
       "      <td>0.022846</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_APOPTOSIS</th>\n",
       "      <td>HALLMARK_APOPTOSIS</td>\n",
       "      <td>0.012987</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.419026</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.434619</td>\n",
       "      <td>0.032389</td>\n",
       "      <td>21</td>\n",
       "      <td>-0.481440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_BILE_ACID_METABOLISM</th>\n",
       "      <td>HALLMARK_BILE_ACID_METABOLISM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011489</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.408767</td>\n",
       "      <td>0.013336</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_CHOLESTEROL_HOMEOSTASIS</th>\n",
       "      <td>HALLMARK_CHOLESTEROL_HOMEOSTASIS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.458548</td>\n",
       "      <td>0.026444</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_COAGULATION</th>\n",
       "      <td>HALLMARK_COAGULATION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.474545</td>\n",
       "      <td>0.049298</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_COMPLEMENT</th>\n",
       "      <td>HALLMARK_COMPLEMENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.429617</td>\n",
       "      <td>0.031434</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_DNA_REPAIR</th>\n",
       "      <td>HALLMARK_DNA_REPAIR</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.477451</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.446613</td>\n",
       "      <td>0.025418</td>\n",
       "      <td>24</td>\n",
       "      <td>1.213217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_E2F_TARGETS</th>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.542046</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.483154</td>\n",
       "      <td>0.044841</td>\n",
       "      <td>23</td>\n",
       "      <td>1.313331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</th>\n",
       "      <td>HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</td>\n",
       "      <td>0.006993</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.416906</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>6.5</td>\n",
       "      <td>0.474510</td>\n",
       "      <td>0.070747</td>\n",
       "      <td>20</td>\n",
       "      <td>-0.814228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ESTROGEN_RESPONSE_EARLY</th>\n",
       "      <td>HALLMARK_ESTROGEN_RESPONSE_EARLY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.430088</td>\n",
       "      <td>0.026459</td>\n",
       "      <td>24</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_ESTROGEN_RESPONSE_LATE</th>\n",
       "      <td>HALLMARK_ESTROGEN_RESPONSE_LATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.451628</td>\n",
       "      <td>0.028833</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_FATTY_ACID_METABOLISM</th>\n",
       "      <td>HALLMARK_FATTY_ACID_METABOLISM</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.426170</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.442034</td>\n",
       "      <td>0.029270</td>\n",
       "      <td>25</td>\n",
       "      <td>-0.542000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_G2M_CHECKPOINT</th>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.523181</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.465199</td>\n",
       "      <td>0.038199</td>\n",
       "      <td>23</td>\n",
       "      <td>1.517898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_GLYCOLYSIS</th>\n",
       "      <td>HALLMARK_GLYCOLYSIS</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.418027</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.422732</td>\n",
       "      <td>0.026385</td>\n",
       "      <td>22</td>\n",
       "      <td>-0.178326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_HEDGEHOG_SIGNALING</th>\n",
       "      <td>HALLMARK_HEDGEHOG_SIGNALING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.504954</td>\n",
       "      <td>0.037089</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_HEME_METABOLISM</th>\n",
       "      <td>HALLMARK_HEME_METABOLISM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.415679</td>\n",
       "      <td>0.019842</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_HYPOXIA</th>\n",
       "      <td>HALLMARK_HYPOXIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.427111</td>\n",
       "      <td>0.030178</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_IL2_STAT5_SIGNALING</th>\n",
       "      <td>HALLMARK_IL2_STAT5_SIGNALING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.430826</td>\n",
       "      <td>0.026054</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_IL6_JAK_STAT3_SIGNALING</th>\n",
       "      <td>HALLMARK_IL6_JAK_STAT3_SIGNALING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003497</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.452033</td>\n",
       "      <td>0.027665</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_INFLAMMATORY_RESPONSE</th>\n",
       "      <td>HALLMARK_INFLAMMATORY_RESPONSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>38.5</td>\n",
       "      <td>0.401231</td>\n",
       "      <td>0.022297</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_INTERFERON_ALPHA_RESPONSE</th>\n",
       "      <td>HALLMARK_INTERFERON_ALPHA_RESPONSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.464518</td>\n",
       "      <td>0.038661</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_INTERFERON_GAMMA_RESPONSE</th>\n",
       "      <td>HALLMARK_INTERFERON_GAMMA_RESPONSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.435360</td>\n",
       "      <td>0.041070</td>\n",
       "      <td>18</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_KRAS_SIGNALING_DN</th>\n",
       "      <td>HALLMARK_KRAS_SIGNALING_DN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>39.5</td>\n",
       "      <td>0.398906</td>\n",
       "      <td>0.001861</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_KRAS_SIGNALING_UP</th>\n",
       "      <td>HALLMARK_KRAS_SIGNALING_UP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>23.5</td>\n",
       "      <td>0.432399</td>\n",
       "      <td>0.028063</td>\n",
       "      <td>18</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MITOTIC_SPINDLE</th>\n",
       "      <td>HALLMARK_MITOTIC_SPINDLE</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.466652</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.406211</td>\n",
       "      <td>0.023926</td>\n",
       "      <td>19</td>\n",
       "      <td>2.526144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MTORC1_SIGNALING</th>\n",
       "      <td>HALLMARK_MTORC1_SIGNALING</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.524668</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.476843</td>\n",
       "      <td>0.032947</td>\n",
       "      <td>25</td>\n",
       "      <td>1.451578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYC_TARGETS_V1</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V1</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.548231</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.495167</td>\n",
       "      <td>0.046218</td>\n",
       "      <td>24</td>\n",
       "      <td>1.148117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYC_TARGETS_V2</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V2</td>\n",
       "      <td>0.014985</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.474126</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>4.5</td>\n",
       "      <td>0.501614</td>\n",
       "      <td>0.032265</td>\n",
       "      <td>18</td>\n",
       "      <td>-0.851921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_MYOGENESIS</th>\n",
       "      <td>HALLMARK_MYOGENESIS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.427330</td>\n",
       "      <td>0.039173</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_NOTCH_SIGNALING</th>\n",
       "      <td>HALLMARK_NOTCH_SIGNALING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.018981</td>\n",
       "      <td>8.5</td>\n",
       "      <td>0.481027</td>\n",
       "      <td>0.015753</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_OXIDATIVE_PHOSPHORYLATION</th>\n",
       "      <td>HALLMARK_OXIDATIVE_PHOSPHORYLATION</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.517292</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.472505</td>\n",
       "      <td>0.035356</td>\n",
       "      <td>25</td>\n",
       "      <td>1.266709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_P53_PATHWAY</th>\n",
       "      <td>HALLMARK_P53_PATHWAY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.438616</td>\n",
       "      <td>0.018184</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_PEROXISOME</th>\n",
       "      <td>HALLMARK_PEROXISOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006993</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.422071</td>\n",
       "      <td>0.015912</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_PI3K_AKT_MTOR_SIGNALING</th>\n",
       "      <td>HALLMARK_PI3K_AKT_MTOR_SIGNALING</td>\n",
       "      <td>0.011988</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.444463</td>\n",
       "      <td>0.013986</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.417357</td>\n",
       "      <td>0.018943</td>\n",
       "      <td>11</td>\n",
       "      <td>1.430888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_PROTEIN_SECRETION</th>\n",
       "      <td>HALLMARK_PROTEIN_SECRETION</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.526593</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.444882</td>\n",
       "      <td>0.017201</td>\n",
       "      <td>20</td>\n",
       "      <td>4.750219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</th>\n",
       "      <td>HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.542762</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.512115</td>\n",
       "      <td>0.032012</td>\n",
       "      <td>25</td>\n",
       "      <td>0.957375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_SPERMATOGENESIS</th>\n",
       "      <td>HALLMARK_SPERMATOGENESIS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011988</td>\n",
       "      <td>34.0</td>\n",
       "      <td>0.406095</td>\n",
       "      <td>0.022867</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_TGF_BETA_SIGNALING</th>\n",
       "      <td>HALLMARK_TGF_BETA_SIGNALING</td>\n",
       "      <td>0.006993</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.501049</td>\n",
       "      <td>0.019980</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.448995</td>\n",
       "      <td>0.013217</td>\n",
       "      <td>3</td>\n",
       "      <td>3.938435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_TNFA_SIGNALING_VIA_NFKB</th>\n",
       "      <td>HALLMARK_TNFA_SIGNALING_VIA_NFKB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.426720</td>\n",
       "      <td>0.030842</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</th>\n",
       "      <td>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.572412</td>\n",
       "      <td>0.002498</td>\n",
       "      <td>16.5</td>\n",
       "      <td>0.445914</td>\n",
       "      <td>0.028407</td>\n",
       "      <td>22</td>\n",
       "      <td>4.453009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_UV_RESPONSE_DN</th>\n",
       "      <td>HALLMARK_UV_RESPONSE_DN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.469651</td>\n",
       "      <td>0.012987</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.414658</td>\n",
       "      <td>0.025621</td>\n",
       "      <td>11</td>\n",
       "      <td>2.146359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_UV_RESPONSE_UP</th>\n",
       "      <td>HALLMARK_UV_RESPONSE_UP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.440968</td>\n",
       "      <td>0.025531</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HALLMARK_XENOBIOTIC_METABOLISM</th>\n",
       "      <td>HALLMARK_XENOBIOTIC_METABOLISM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.453611</td>\n",
       "      <td>0.033706</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                               Pathway  \\\n",
       "HALLMARK_ADIPOGENESIS                                            HALLMARK_ADIPOGENESIS   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                              HALLMARK_ALLOGRAFT_REJECTION   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                  HALLMARK_ANDROGEN_RESPONSE   \n",
       "HALLMARK_ANGIOGENESIS                                            HALLMARK_ANGIOGENESIS   \n",
       "HALLMARK_APICAL_JUNCTION                                      HALLMARK_APICAL_JUNCTION   \n",
       "HALLMARK_APICAL_SURFACE                                        HALLMARK_APICAL_SURFACE   \n",
       "HALLMARK_APOPTOSIS                                                  HALLMARK_APOPTOSIS   \n",
       "HALLMARK_BILE_ACID_METABOLISM                            HALLMARK_BILE_ACID_METABOLISM   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                      HALLMARK_CHOLESTEROL_HOMEOSTASIS   \n",
       "HALLMARK_COAGULATION                                              HALLMARK_COAGULATION   \n",
       "HALLMARK_COMPLEMENT                                                HALLMARK_COMPLEMENT   \n",
       "HALLMARK_DNA_REPAIR                                                HALLMARK_DNA_REPAIR   \n",
       "HALLMARK_E2F_TARGETS                                              HALLMARK_E2F_TARGETS   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION  HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                      HALLMARK_ESTROGEN_RESPONSE_EARLY   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                        HALLMARK_ESTROGEN_RESPONSE_LATE   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                          HALLMARK_FATTY_ACID_METABOLISM   \n",
       "HALLMARK_G2M_CHECKPOINT                                        HALLMARK_G2M_CHECKPOINT   \n",
       "HALLMARK_GLYCOLYSIS                                                HALLMARK_GLYCOLYSIS   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                HALLMARK_HEDGEHOG_SIGNALING   \n",
       "HALLMARK_HEME_METABOLISM                                      HALLMARK_HEME_METABOLISM   \n",
       "HALLMARK_HYPOXIA                                                      HALLMARK_HYPOXIA   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                              HALLMARK_IL2_STAT5_SIGNALING   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                      HALLMARK_IL6_JAK_STAT3_SIGNALING   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                          HALLMARK_INFLAMMATORY_RESPONSE   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                  HALLMARK_INTERFERON_ALPHA_RESPONSE   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                  HALLMARK_INTERFERON_GAMMA_RESPONSE   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                  HALLMARK_KRAS_SIGNALING_DN   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                  HALLMARK_KRAS_SIGNALING_UP   \n",
       "HALLMARK_MITOTIC_SPINDLE                                      HALLMARK_MITOTIC_SPINDLE   \n",
       "HALLMARK_MTORC1_SIGNALING                                    HALLMARK_MTORC1_SIGNALING   \n",
       "HALLMARK_MYC_TARGETS_V1                                        HALLMARK_MYC_TARGETS_V1   \n",
       "HALLMARK_MYC_TARGETS_V2                                        HALLMARK_MYC_TARGETS_V2   \n",
       "HALLMARK_MYOGENESIS                                                HALLMARK_MYOGENESIS   \n",
       "HALLMARK_NOTCH_SIGNALING                                      HALLMARK_NOTCH_SIGNALING   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                  HALLMARK_OXIDATIVE_PHOSPHORYLATION   \n",
       "HALLMARK_P53_PATHWAY                                              HALLMARK_P53_PATHWAY   \n",
       "HALLMARK_PEROXISOME                                                HALLMARK_PEROXISOME   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                      HALLMARK_PI3K_AKT_MTOR_SIGNALING   \n",
       "HALLMARK_PROTEIN_SECRETION                                  HALLMARK_PROTEIN_SECRETION   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY      HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY   \n",
       "HALLMARK_SPERMATOGENESIS                                      HALLMARK_SPERMATOGENESIS   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                HALLMARK_TGF_BETA_SIGNALING   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                      HALLMARK_TNFA_SIGNALING_VIA_NFKB   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                  HALLMARK_UNFOLDED_PROTEIN_RESPONSE   \n",
       "HALLMARK_UV_RESPONSE_DN                                        HALLMARK_UV_RESPONSE_DN   \n",
       "HALLMARK_UV_RESPONSE_UP                                        HALLMARK_UV_RESPONSE_UP   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                          HALLMARK_XENOBIOTIC_METABOLISM   \n",
       "\n",
       "                                            P-value (Real)  Rank (Real)  \\\n",
       "HALLMARK_ADIPOGENESIS                             0.000999         14.0   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                           NaN          NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE                        0.009990         15.0   \n",
       "HALLMARK_ANGIOGENESIS                                  NaN          NaN   \n",
       "HALLMARK_APICAL_JUNCTION                               NaN          NaN   \n",
       "HALLMARK_APICAL_SURFACE                                NaN          NaN   \n",
       "HALLMARK_APOPTOSIS                                0.012987         18.0   \n",
       "HALLMARK_BILE_ACID_METABOLISM                          NaN          NaN   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                       NaN          NaN   \n",
       "HALLMARK_COAGULATION                                   NaN          NaN   \n",
       "HALLMARK_COMPLEMENT                                    NaN          NaN   \n",
       "HALLMARK_DNA_REPAIR                               0.000999         10.0   \n",
       "HALLMARK_E2F_TARGETS                              0.000999          4.0   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION        0.006993         20.0   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                       NaN          NaN   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                        NaN          NaN   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                    0.005994         17.0   \n",
       "HALLMARK_G2M_CHECKPOINT                           0.000999          7.0   \n",
       "HALLMARK_GLYCOLYSIS                               0.005994         19.0   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                            NaN          NaN   \n",
       "HALLMARK_HEME_METABOLISM                               NaN          NaN   \n",
       "HALLMARK_HYPOXIA                                       NaN          NaN   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                           NaN          NaN   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                       NaN          NaN   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                         NaN          NaN   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                     NaN          NaN   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                     NaN          NaN   \n",
       "HALLMARK_KRAS_SIGNALING_DN                             NaN          NaN   \n",
       "HALLMARK_KRAS_SIGNALING_UP                             NaN          NaN   \n",
       "HALLMARK_MITOTIC_SPINDLE                          0.000999         13.0   \n",
       "HALLMARK_MTORC1_SIGNALING                         0.000999          6.0   \n",
       "HALLMARK_MYC_TARGETS_V1                           0.000999          2.0   \n",
       "HALLMARK_MYC_TARGETS_V2                           0.014985         11.0   \n",
       "HALLMARK_MYOGENESIS                                    NaN          NaN   \n",
       "HALLMARK_NOTCH_SIGNALING                               NaN          NaN   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                0.000999          8.0   \n",
       "HALLMARK_P53_PATHWAY                                   NaN          NaN   \n",
       "HALLMARK_PEROXISOME                                    NaN          NaN   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                  0.011988         16.0   \n",
       "HALLMARK_PROTEIN_SECRETION                        0.000999          5.0   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY          0.000999          3.0   \n",
       "HALLMARK_SPERMATOGENESIS                               NaN          NaN   \n",
       "HALLMARK_TGF_BETA_SIGNALING                       0.006993          9.0   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                       NaN          NaN   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                0.000999          1.0   \n",
       "HALLMARK_UV_RESPONSE_DN                           0.000999         12.0   \n",
       "HALLMARK_UV_RESPONSE_UP                                NaN          NaN   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                         NaN          NaN   \n",
       "\n",
       "                                            Test statistic (Real)  \\\n",
       "HALLMARK_ADIPOGENESIS                                    0.449242   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                  NaN   \n",
       "HALLMARK_ANDROGEN_RESPONSE                               0.444627   \n",
       "HALLMARK_ANGIOGENESIS                                         NaN   \n",
       "HALLMARK_APICAL_JUNCTION                                      NaN   \n",
       "HALLMARK_APICAL_SURFACE                                       NaN   \n",
       "HALLMARK_APOPTOSIS                                       0.419026   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                 NaN   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                              NaN   \n",
       "HALLMARK_COAGULATION                                          NaN   \n",
       "HALLMARK_COMPLEMENT                                           NaN   \n",
       "HALLMARK_DNA_REPAIR                                      0.477451   \n",
       "HALLMARK_E2F_TARGETS                                     0.542046   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION               0.416906   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                              NaN   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                               NaN   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                           0.426170   \n",
       "HALLMARK_G2M_CHECKPOINT                                  0.523181   \n",
       "HALLMARK_GLYCOLYSIS                                      0.418027   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                   NaN   \n",
       "HALLMARK_HEME_METABOLISM                                      NaN   \n",
       "HALLMARK_HYPOXIA                                              NaN   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                  NaN   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                              NaN   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                                NaN   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                            NaN   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                            NaN   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                    NaN   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                    NaN   \n",
       "HALLMARK_MITOTIC_SPINDLE                                 0.466652   \n",
       "HALLMARK_MTORC1_SIGNALING                                0.524668   \n",
       "HALLMARK_MYC_TARGETS_V1                                  0.548231   \n",
       "HALLMARK_MYC_TARGETS_V2                                  0.474126   \n",
       "HALLMARK_MYOGENESIS                                           NaN   \n",
       "HALLMARK_NOTCH_SIGNALING                                      NaN   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                       0.517292   \n",
       "HALLMARK_P53_PATHWAY                                          NaN   \n",
       "HALLMARK_PEROXISOME                                           NaN   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                         0.444463   \n",
       "HALLMARK_PROTEIN_SECRETION                               0.526593   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                 0.542762   \n",
       "HALLMARK_SPERMATOGENESIS                                      NaN   \n",
       "HALLMARK_TGF_BETA_SIGNALING                              0.501049   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                              NaN   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                       0.572412   \n",
       "HALLMARK_UV_RESPONSE_DN                                  0.469651   \n",
       "HALLMARK_UV_RESPONSE_UP                                       NaN   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                                NaN   \n",
       "\n",
       "                                            Median p-value (simulated)  \\\n",
       "HALLMARK_ADIPOGENESIS                                         0.000999   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                  0.000999   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                    0.002498   \n",
       "HALLMARK_ANGIOGENESIS                                         0.012987   \n",
       "HALLMARK_APICAL_JUNCTION                                      0.000999   \n",
       "HALLMARK_APICAL_SURFACE                                       0.034466   \n",
       "HALLMARK_APOPTOSIS                                            0.001998   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                 0.011489   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                              0.004995   \n",
       "HALLMARK_COAGULATION                                          0.000999   \n",
       "HALLMARK_COMPLEMENT                                           0.000999   \n",
       "HALLMARK_DNA_REPAIR                                           0.000999   \n",
       "HALLMARK_E2F_TARGETS                                          0.000999   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION                    0.000999   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                              0.000999   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                               0.000999   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                                0.000999   \n",
       "HALLMARK_G2M_CHECKPOINT                                       0.000999   \n",
       "HALLMARK_GLYCOLYSIS                                           0.000999   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                   0.010000   \n",
       "HALLMARK_HEME_METABOLISM                                      0.001998   \n",
       "HALLMARK_HYPOXIA                                              0.000999   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                  0.000999   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                              0.003497   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                                0.004995   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                            0.001998   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                            0.000999   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                    0.009990   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                    0.000999   \n",
       "HALLMARK_MITOTIC_SPINDLE                                      0.001998   \n",
       "HALLMARK_MTORC1_SIGNALING                                     0.000999   \n",
       "HALLMARK_MYC_TARGETS_V1                                       0.000999   \n",
       "HALLMARK_MYC_TARGETS_V2                                       0.000999   \n",
       "HALLMARK_MYOGENESIS                                           0.000999   \n",
       "HALLMARK_NOTCH_SIGNALING                                      0.018981   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                            0.000999   \n",
       "HALLMARK_P53_PATHWAY                                          0.000999   \n",
       "HALLMARK_PEROXISOME                                           0.006993   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                              0.013986   \n",
       "HALLMARK_PROTEIN_SECRETION                                    0.003996   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                      0.003996   \n",
       "HALLMARK_SPERMATOGENESIS                                      0.011988   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                   0.019980   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                              0.000999   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                            0.002498   \n",
       "HALLMARK_UV_RESPONSE_DN                                       0.012987   \n",
       "HALLMARK_UV_RESPONSE_UP                                       0.000999   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                                0.000999   \n",
       "\n",
       "                                            Median rank (simulated)  \\\n",
       "HALLMARK_ADIPOGENESIS                                          16.0   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                   27.0   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                     17.0   \n",
       "HALLMARK_ANGIOGENESIS                                           6.0   \n",
       "HALLMARK_APICAL_JUNCTION                                       25.5   \n",
       "HALLMARK_APICAL_SURFACE                                        19.0   \n",
       "HALLMARK_APOPTOSIS                                             19.0   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                  31.0   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                               13.0   \n",
       "HALLMARK_COAGULATION                                            5.0   \n",
       "HALLMARK_COMPLEMENT                                            25.0   \n",
       "HALLMARK_DNA_REPAIR                                            12.0   \n",
       "HALLMARK_E2F_TARGETS                                            4.0   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION                      6.5   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                               21.0   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                                11.0   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                                 15.0   \n",
       "HALLMARK_G2M_CHECKPOINT                                        11.0   \n",
       "HALLMARK_GLYCOLYSIS                                            26.0   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                     6.0   \n",
       "HALLMARK_HEME_METABOLISM                                       31.0   \n",
       "HALLMARK_HYPOXIA                                               27.0   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                   28.0   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                               16.0   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                                 38.5   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                             12.0   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                             24.0   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                     39.5   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                     23.5   \n",
       "HALLMARK_MITOTIC_SPINDLE                                       35.0   \n",
       "HALLMARK_MTORC1_SIGNALING                                       6.0   \n",
       "HALLMARK_MYC_TARGETS_V1                                         3.0   \n",
       "HALLMARK_MYC_TARGETS_V2                                         4.5   \n",
       "HALLMARK_MYOGENESIS                                            29.0   \n",
       "HALLMARK_NOTCH_SIGNALING                                        8.5   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                              8.0   \n",
       "HALLMARK_P53_PATHWAY                                           19.0   \n",
       "HALLMARK_PEROXISOME                                            25.0   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                               31.0   \n",
       "HALLMARK_PROTEIN_SECRETION                                     17.0   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                        2.0   \n",
       "HALLMARK_SPERMATOGENESIS                                       34.0   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                    13.0   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                               29.0   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                             16.5   \n",
       "HALLMARK_UV_RESPONSE_DN                                        35.0   \n",
       "HALLMARK_UV_RESPONSE_UP                                        18.0   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                                 13.0   \n",
       "\n",
       "                                            Mean test statistic (simulated)  \\\n",
       "HALLMARK_ADIPOGENESIS                                              0.442502   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                       0.424805   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                         0.452496   \n",
       "HALLMARK_ANGIOGENESIS                                              0.512233   \n",
       "HALLMARK_APICAL_JUNCTION                                           0.425509   \n",
       "HALLMARK_APICAL_SURFACE                                            0.471780   \n",
       "HALLMARK_APOPTOSIS                                                 0.434619   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                      0.408767   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                                   0.458548   \n",
       "HALLMARK_COAGULATION                                               0.474545   \n",
       "HALLMARK_COMPLEMENT                                                0.429617   \n",
       "HALLMARK_DNA_REPAIR                                                0.446613   \n",
       "HALLMARK_E2F_TARGETS                                               0.483154   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION                         0.474510   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                                   0.430088   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                                    0.451628   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                                     0.442034   \n",
       "HALLMARK_G2M_CHECKPOINT                                            0.465199   \n",
       "HALLMARK_GLYCOLYSIS                                                0.422732   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                        0.504954   \n",
       "HALLMARK_HEME_METABOLISM                                           0.415679   \n",
       "HALLMARK_HYPOXIA                                                   0.427111   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                       0.430826   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                                   0.452033   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                                     0.401231   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                                 0.464518   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                                 0.435360   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                         0.398906   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                         0.432399   \n",
       "HALLMARK_MITOTIC_SPINDLE                                           0.406211   \n",
       "HALLMARK_MTORC1_SIGNALING                                          0.476843   \n",
       "HALLMARK_MYC_TARGETS_V1                                            0.495167   \n",
       "HALLMARK_MYC_TARGETS_V2                                            0.501614   \n",
       "HALLMARK_MYOGENESIS                                                0.427330   \n",
       "HALLMARK_NOTCH_SIGNALING                                           0.481027   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                                 0.472505   \n",
       "HALLMARK_P53_PATHWAY                                               0.438616   \n",
       "HALLMARK_PEROXISOME                                                0.422071   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                                   0.417357   \n",
       "HALLMARK_PROTEIN_SECRETION                                         0.444882   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                           0.512115   \n",
       "HALLMARK_SPERMATOGENESIS                                           0.406095   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                        0.448995   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                                   0.426720   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                                 0.445914   \n",
       "HALLMARK_UV_RESPONSE_DN                                            0.414658   \n",
       "HALLMARK_UV_RESPONSE_UP                                            0.440968   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                                     0.453611   \n",
       "\n",
       "                                            Std deviation (simulated)  \\\n",
       "HALLMARK_ADIPOGENESIS                                        0.026254   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                 0.035230   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                   0.025508   \n",
       "HALLMARK_ANGIOGENESIS                                        0.019648   \n",
       "HALLMARK_APICAL_JUNCTION                                     0.028067   \n",
       "HALLMARK_APICAL_SURFACE                                      0.022846   \n",
       "HALLMARK_APOPTOSIS                                           0.032389   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                0.013336   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                             0.026444   \n",
       "HALLMARK_COAGULATION                                         0.049298   \n",
       "HALLMARK_COMPLEMENT                                          0.031434   \n",
       "HALLMARK_DNA_REPAIR                                          0.025418   \n",
       "HALLMARK_E2F_TARGETS                                         0.044841   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION                   0.070747   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                             0.026459   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                              0.028833   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                               0.029270   \n",
       "HALLMARK_G2M_CHECKPOINT                                      0.038199   \n",
       "HALLMARK_GLYCOLYSIS                                          0.026385   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                  0.037089   \n",
       "HALLMARK_HEME_METABOLISM                                     0.019842   \n",
       "HALLMARK_HYPOXIA                                             0.030178   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                 0.026054   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                             0.027665   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                               0.022297   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                           0.038661   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                           0.041070   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                   0.001861   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                   0.028063   \n",
       "HALLMARK_MITOTIC_SPINDLE                                     0.023926   \n",
       "HALLMARK_MTORC1_SIGNALING                                    0.032947   \n",
       "HALLMARK_MYC_TARGETS_V1                                      0.046218   \n",
       "HALLMARK_MYC_TARGETS_V2                                      0.032265   \n",
       "HALLMARK_MYOGENESIS                                          0.039173   \n",
       "HALLMARK_NOTCH_SIGNALING                                     0.015753   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                           0.035356   \n",
       "HALLMARK_P53_PATHWAY                                         0.018184   \n",
       "HALLMARK_PEROXISOME                                          0.015912   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                             0.018943   \n",
       "HALLMARK_PROTEIN_SECRETION                                   0.017201   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                     0.032012   \n",
       "HALLMARK_SPERMATOGENESIS                                     0.022867   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                  0.013217   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                             0.030842   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                           0.028407   \n",
       "HALLMARK_UV_RESPONSE_DN                                      0.025621   \n",
       "HALLMARK_UV_RESPONSE_UP                                      0.025531   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                               0.033706   \n",
       "\n",
       "                                            Number of experiments (simulated)  \\\n",
       "HALLMARK_ADIPOGENESIS                                                      25   \n",
       "HALLMARK_ALLOGRAFT_REJECTION                                               19   \n",
       "HALLMARK_ANDROGEN_RESPONSE                                                 18   \n",
       "HALLMARK_ANGIOGENESIS                                                       5   \n",
       "HALLMARK_APICAL_JUNCTION                                                   20   \n",
       "HALLMARK_APICAL_SURFACE                                                     2   \n",
       "HALLMARK_APOPTOSIS                                                         21   \n",
       "HALLMARK_BILE_ACID_METABOLISM                                              10   \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                                           21   \n",
       "HALLMARK_COAGULATION                                                       22   \n",
       "HALLMARK_COMPLEMENT                                                        22   \n",
       "HALLMARK_DNA_REPAIR                                                        24   \n",
       "HALLMARK_E2F_TARGETS                                                       23   \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION                                 20   \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                                           24   \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                                            25   \n",
       "HALLMARK_FATTY_ACID_METABOLISM                                             25   \n",
       "HALLMARK_G2M_CHECKPOINT                                                    23   \n",
       "HALLMARK_GLYCOLYSIS                                                        22   \n",
       "HALLMARK_HEDGEHOG_SIGNALING                                                 3   \n",
       "HALLMARK_HEME_METABOLISM                                                   21   \n",
       "HALLMARK_HYPOXIA                                                           21   \n",
       "HALLMARK_IL2_STAT5_SIGNALING                                               19   \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                                           16   \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                                             14   \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE                                         14   \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE                                         18   \n",
       "HALLMARK_KRAS_SIGNALING_DN                                                  2   \n",
       "HALLMARK_KRAS_SIGNALING_UP                                                 18   \n",
       "HALLMARK_MITOTIC_SPINDLE                                                   19   \n",
       "HALLMARK_MTORC1_SIGNALING                                                  25   \n",
       "HALLMARK_MYC_TARGETS_V1                                                    24   \n",
       "HALLMARK_MYC_TARGETS_V2                                                    18   \n",
       "HALLMARK_MYOGENESIS                                                        19   \n",
       "HALLMARK_NOTCH_SIGNALING                                                    6   \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION                                         25   \n",
       "HALLMARK_P53_PATHWAY                                                       22   \n",
       "HALLMARK_PEROXISOME                                                        15   \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING                                           11   \n",
       "HALLMARK_PROTEIN_SECRETION                                                 20   \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY                                   25   \n",
       "HALLMARK_SPERMATOGENESIS                                                   10   \n",
       "HALLMARK_TGF_BETA_SIGNALING                                                 3   \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                                           19   \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE                                         22   \n",
       "HALLMARK_UV_RESPONSE_DN                                                    11   \n",
       "HALLMARK_UV_RESPONSE_UP                                                    21   \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                                             25   \n",
       "\n",
       "                                             Z score  \n",
       "HALLMARK_ADIPOGENESIS                       0.256710  \n",
       "HALLMARK_ALLOGRAFT_REJECTION                     NaN  \n",
       "HALLMARK_ANDROGEN_RESPONSE                 -0.308466  \n",
       "HALLMARK_ANGIOGENESIS                            NaN  \n",
       "HALLMARK_APICAL_JUNCTION                         NaN  \n",
       "HALLMARK_APICAL_SURFACE                          NaN  \n",
       "HALLMARK_APOPTOSIS                         -0.481440  \n",
       "HALLMARK_BILE_ACID_METABOLISM                    NaN  \n",
       "HALLMARK_CHOLESTEROL_HOMEOSTASIS                 NaN  \n",
       "HALLMARK_COAGULATION                             NaN  \n",
       "HALLMARK_COMPLEMENT                              NaN  \n",
       "HALLMARK_DNA_REPAIR                         1.213217  \n",
       "HALLMARK_E2F_TARGETS                        1.313331  \n",
       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION -0.814228  \n",
       "HALLMARK_ESTROGEN_RESPONSE_EARLY                 NaN  \n",
       "HALLMARK_ESTROGEN_RESPONSE_LATE                  NaN  \n",
       "HALLMARK_FATTY_ACID_METABOLISM             -0.542000  \n",
       "HALLMARK_G2M_CHECKPOINT                     1.517898  \n",
       "HALLMARK_GLYCOLYSIS                        -0.178326  \n",
       "HALLMARK_HEDGEHOG_SIGNALING                      NaN  \n",
       "HALLMARK_HEME_METABOLISM                         NaN  \n",
       "HALLMARK_HYPOXIA                                 NaN  \n",
       "HALLMARK_IL2_STAT5_SIGNALING                     NaN  \n",
       "HALLMARK_IL6_JAK_STAT3_SIGNALING                 NaN  \n",
       "HALLMARK_INFLAMMATORY_RESPONSE                   NaN  \n",
       "HALLMARK_INTERFERON_ALPHA_RESPONSE               NaN  \n",
       "HALLMARK_INTERFERON_GAMMA_RESPONSE               NaN  \n",
       "HALLMARK_KRAS_SIGNALING_DN                       NaN  \n",
       "HALLMARK_KRAS_SIGNALING_UP                       NaN  \n",
       "HALLMARK_MITOTIC_SPINDLE                    2.526144  \n",
       "HALLMARK_MTORC1_SIGNALING                   1.451578  \n",
       "HALLMARK_MYC_TARGETS_V1                     1.148117  \n",
       "HALLMARK_MYC_TARGETS_V2                    -0.851921  \n",
       "HALLMARK_MYOGENESIS                              NaN  \n",
       "HALLMARK_NOTCH_SIGNALING                         NaN  \n",
       "HALLMARK_OXIDATIVE_PHOSPHORYLATION          1.266709  \n",
       "HALLMARK_P53_PATHWAY                             NaN  \n",
       "HALLMARK_PEROXISOME                              NaN  \n",
       "HALLMARK_PI3K_AKT_MTOR_SIGNALING            1.430888  \n",
       "HALLMARK_PROTEIN_SECRETION                  4.750219  \n",
       "HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY    0.957375  \n",
       "HALLMARK_SPERMATOGENESIS                         NaN  \n",
       "HALLMARK_TGF_BETA_SIGNALING                 3.938435  \n",
       "HALLMARK_TNFA_SIGNALING_VIA_NFKB                 NaN  \n",
       "HALLMARK_UNFOLDED_PROTEIN_RESPONSE          4.453009  \n",
       "HALLMARK_UV_RESPONSE_DN                     2.146359  \n",
       "HALLMARK_UV_RESPONSE_UP                          NaN  \n",
       "HALLMARK_XENOBIOTIC_METABOLISM                   NaN  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary = pd.DataFrame(data={'Pathway': template_simulated_enriched_pathways_stats.index,\n",
    "                             'P-value (Real)': template_simulated_enriched_pathways_stats['pvalue'],\n",
    "                             'Rank (Real)': template_simulated_enriched_pathways_stats['ranking'],\n",
    "                             'Test statistic (Real)': template_enriched_pathways[col_to_rank],\n",
    "                             'Median p-value (simulated)': median_pval_simulated ,\n",
    "                             'Median rank (simulated)': median_rank_simulated ,\n",
    "                             'Mean test statistic (simulated)': mean_test_simulated ,\n",
    "                             'Std deviation (simulated)': std_test_simulated,\n",
    "                             'Number of experiments (simulated)': count_simulated\n",
    "                            }\n",
    "                      )\n",
    "summary['Z score'] = (summary['Test statistic (Real)'] - summary['Mean test statistic (simulated)'])/summary['Std deviation (simulated)']\n",
    "summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save file\n",
    "summary_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"pathway_summary_table.tsv\")\n",
    "\n",
    "summary.to_csv(summary_file, float_format='%.5f', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are a few immune pathways that are template-enriched but there are others that look to be involved in cellular maintenancem, which I wouldn't expect to be specific to these experiment.\n",
    "\n",
    "For GSEA I am currently using the p-values from the DE analysis to rank genes, should I try using avg gene expression instead..? Maybe p-values don't cluster genes as much..?\n",
    "\n",
    "Note: this notebook takes ~1hr to run\n",
    "\n",
    "**Check:**\n",
    "* Are there template-specific pathways (i.e. pathways specific to Dexamethasone treatment) that we can use as a positive?\n",
    "* Are there agnostic pathways that we can use as a negative control?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:ranked_pathways] *",
   "language": "python",
   "name": "conda-env-ranked_pathways-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "require": {
   "paths": {
    "buttons.colvis": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.colVis.min",
    "buttons.flash": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.flash.min",
    "buttons.html5": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.html5.min",
    "buttons.print": "https://cdn.datatables.net/buttons/1.5.6/js/buttons.print.min",
    "chartjs": "https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart",
    "d3": "https://d3js.org/d3.v5.min",
    "d3-array": "https://d3js.org/d3-array.v2.min",
    "datatables.net": "https://cdn.datatables.net/1.10.18/js/jquery.dataTables",
    "datatables.net-buttons": "https://cdn.datatables.net/buttons/1.5.6/js/dataTables.buttons.min",
    "datatables.responsive": "https://cdn.datatables.net/responsive/2.2.2/js/dataTables.responsive.min",
    "datatables.scroller": "https://cdn.datatables.net/scroller/2.0.0/js/dataTables.scroller.min",
    "datatables.select": "https://cdn.datatables.net/select/1.3.0/js/dataTables.select.min",
    "jszip": "https://cdnjs.cloudflare.com/ajax/libs/jszip/2.5.0/jszip.min",
    "moment": "https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.8.0/moment",
    "pdfmake": "https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/pdfmake.min",
    "vfsfonts": "https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/vfs_fonts"
   },
   "shim": {
    "buttons.colvis": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.flash": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.html5": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "buttons.print": {
     "deps": [
      "jszip",
      "datatables.net-buttons"
     ]
    },
    "chartjs": {
     "deps": [
      "moment"
     ]
    },
    "datatables.net": {
     "exports": "$.fn.dataTable"
    },
    "datatables.net-buttons": {
     "deps": [
      "datatables.net"
     ]
    },
    "pdfmake": {
     "deps": [
      "datatables.net"
     ]
    },
    "vfsfonts": {
     "deps": [
      "datatables.net"
     ]
    }
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
