{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Statistical analysis\n",
    "This notebook performs differential expression analysis using the real template experiment and simulated experiments, as a null set. Then the set of differentially expressed genes (DEGs) obtained from this analysis are used to perform gene set enrichment analysis (GSEA) to identify pathways enriched in these set of DEGs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: \n",
      "examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.\n",
      "  \"found relative to the 'datapath' directory.\".format(key))\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%load_ext rpy2.ipython\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import seaborn as sns\n",
    "import rpy2.robjects\n",
    "from rpy2.robjects.packages import importr\n",
    "from rpy2.robjects import pandas2ri\n",
    "pandas2ri.activate()\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import utils\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in config variables\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../\"))\n",
    "\n",
    "config_file = os.path.abspath(os.path.join(base_dir,\n",
    "                                           \"Rank_pathways\",\n",
    "                                           \"init_config.tsv\"))\n",
    "params = utils.read_config(config_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# User selected experiment id\n",
    "project_id = \"SRP000762\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load params\n",
    "local_dir = params[\"local_dir\"]\n",
    "dataset_name = params['dataset_name']\n",
    "num_runs=25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load real template experiment\n",
    "template_data_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"recount2_template_data.tsv\")\n",
    "\n",
    "# Load metadata file with grouping assignments for samples\n",
    "metadata_file = os.path.join(\n",
    "    base_dir,\n",
    "    \"Rank_pathways\",\n",
    "    \"data\",\n",
    "    \"metadata\",\n",
    "    \"SRP000762_groups.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Select 59\n",
    "# Run one time\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install(\"limma\")\n",
    "#BiocManager::install('EnhancedVolcano')\n",
    "#devtools::install_github('kevinblighe/EnhancedVolcano')\n",
    "#BiocManager::install('clusterProfiler')\n",
    "#BiocManager::install(\"org.Hs.eg.db\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential expression analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library('limma')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get differentially expressed genes from template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] 60\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i metadata_file -i project_id -i template_data_file -i local_dir\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "get_DE_stats(metadata_file,\n",
    "             project_id, \n",
    "             template_data_file,\n",
    "             \"template\",\n",
    "             local_dir,\n",
    "             \"real\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get differentially expressed genes from each simulated experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_0.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_1.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_2.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_3.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_4.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_5.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_6.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_7.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_8.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_9.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_10.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_11.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_12.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_13.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_14.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_15.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_16.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_17.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_18.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_19.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_20.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_21.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_22.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_23.txt...\n",
       "running file: /home/alexandra/Documents/Data/Rank_pathways/pseudo_experiment/selected_simulated_data_SRP000762_24.txt...\n",
       "[1] 0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i metadata_file -i project_id -i base_dir -i local_dir -i num_runs -o num_sign_DEGs_simulated\n",
    "\n",
    "source('../functions/DE_analysis.R')\n",
    "\n",
    "num_sign_DEGs_simulated <- c()\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "  simulated_data_file <- paste(local_dir, \"pseudo_experiment/selected_simulated_data_\", project_id, \"_\", i, \".txt\", sep=\"\")\n",
    "  cat(paste(\"running file: \", simulated_data_file, \"...\\n\", sep=\"\"))\n",
    "  \n",
    "  run_output <- get_DE_stats(metadata_file,\n",
    "                             project_id, \n",
    "                             simulated_data_file,\n",
    "                             \"simulated\",\n",
    "                             local_dir,\n",
    "                             i)\n",
    "  \n",
    "  num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, run_output)\n",
    "}\n",
    "median(num_sign_DEGs_simulated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/scipy/stats/stats.py:1713: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.\n",
      "  return np.add.reduce(sorted[indexer] * weights, axis=axis) / sumval\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7ff4e5c99690>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvqOYd8AAADCJJREFUeJzt3F2MXHUdxvHn0YoXglrspjYIrlFigkYLbqrG94AGvQCMRkXEGknWREkw4gXRC43c4AvghcZYhVANvoKEJtYXrBhCAo1bbbCl0aIBLZZ2EaMYo6bweLGnZiWzzJmZMzPdX7+fZDMzZ87u+f3b9LvDYeY4iQAAq99Tpj0AAKAbBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBFrJnmwdevWZXZ2dpKHBIBVb9euXQ8nmem330SDPjs7q4WFhUkeEgBWPdsPtNmPUy4AUARBB4AiCDoAFEHQAaAIgg4ARfQNuu1Tbd9u+17be21f1mz/tO0Hbe9uvt42/nEBACtp87bFI5IuT/Ir2ydJ2mX7tua5a5N8YXzjAQDa6hv0JAclHWzuP2p7n6RTxj0YAGAwA51Dtz0r6UxJO5tNl9q+x/b1ttd2PBsAYACtPylq+0RJN0v6aJK/2/6KpCslpbm9WtIHe3zfvKR5STrttNO6mBmFfWvnH6c9wsS995X8u0A3Wr1Ct/00LcX8xiQ/kKQkh5I8luRxSV+TtKnX9ybZkmQuydzMTN9LEQAAhtTmXS6WdJ2kfUmuWbZ9w7Ld3i5pT/fjAQDaanPK5TWSLpb0G9u7m22fkHSh7Y1aOuVyv6QPjWVCAEArbd7lcqck93hqe/fjAACGxSdFAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAU0Tfotk+1fbvte23vtX1Zs/1k27fZ3t/crh3/uACAlbR5hX5E0uVJzpD0KkkfsX2GpCsk7UhyuqQdzWMAwJT0DXqSg0l+1dx/VNI+SadIOl/S1ma3rZIuGNeQAID+BjqHbntW0pmSdkpan+Rg89RDktZ3OhkAYCCtg277REk3S/pokr8vfy5JJGWF75u3vWB7YXFxcaRhAQAraxV020/TUsxvTPKDZvMh2xua5zdIOtzre5NsSTKXZG5mZqaLmQEAPbR5l4slXSdpX5Jrlj21TdLm5v5mSbd2Px4AoK01LfZ5jaSLJf3G9u5m2yckXSXpe7YvkfSApHeNZ0QAQBt9g57kTkle4emzux0HADAsPikKAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARfQNuu3rbR+2vWfZtk/bftD27ubrbeMdEwDQT5tX6DdIOrfH9muTbGy+tnc7FgBgUH2DnuQOSY9MYBYAwAhGOYd+qe17mlMyazubCAAwlGGD/hVJL5S0UdJBSVevtKPtedsLthcWFxeHPBwAoJ+hgp7kUJLHkjwu6WuSNj3JvluSzCWZm5mZGXZOAEAfQwXd9oZlD98uac9K+wIAJmNNvx1sf1vSGyWts31A0qckvdH2RkmRdL+kD41xRgBAC32DnuTCHpuvG8MsAIAR8ElRACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBF9A267ettH7a9Z9m2k23fZnt/c7t2vGMCAPpp8wr9BknnPmHbFZJ2JDld0o7mMQBgivoGPckdkh55wubzJW1t7m+VdEHHcwEABjTsOfT1SQ429x+StL6jeQAAQxr5f4omiaSs9LztedsLthcWFxdHPRwAYAXDBv2Q7Q2S1NweXmnHJFuSzCWZm5mZGfJwAIB+hg36Nkmbm/ubJd3azTgAgGG1edvityXdJenFtg/YvkTSVZLebHu/pHOaxwCAKVrTb4ckF67w1NkdzwIAGAGfFAWAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4Ai1ozyzbbvl/SopMckHUky18VQAIDBjRT0xpuSPNzBzwEAjIBTLgBQxKhBj6Sf2t5le77XDrbnbS/YXlhcXBzxcACAlYwa9NcmOUvSWyV9xPbrn7hDki1J5pLMzczMjHg4AMBKRgp6kgeb28OSbpG0qYuhAACDGzrotp9h+6Sj9yW9RdKergYDAAxmlHe5rJd0i+2jP+dbSX7cyVQAgIENHfQkf5D08g5nAQCMgLctAkARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAogqADQBEEHQCKIOgAUARBB4AiCDoAFEHQAaAIgg4ARRB0ACiCoANAEQQdAIog6ABQBEEHgCIIOgAUQdABoAiCDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgCIIOAEUQdAAoYqSg2z7X9m9t32f7iq6GAgAMbuig236qpC9LequkMyRdaPuMrgYDAAxmlFfomyTdl+QPSf4j6TuSzu9mLADAoEYJ+imS/rTs8YFmGwBgCtaM+wC25yXNNw//Yfu34z7mGKyT9PC0h5ig42290hTXfNE0DrqEv+fV4/ltdhol6A9KOnXZ4+c12/5Pki2StoxwnKmzvZBkbtpzTMrxtl6JNR8vqq95lFMuv5R0uu0X2D5B0nskbetmLADAoIZ+hZ7kiO1LJf1E0lMlXZ9kb2eTAQAGMtI59CTbJW3vaJZj2ao+ZTSE4229Ems+XpRes5NMewYAQAf46D8AFEHQe7B9su3bbO9vbtc+yb7PtH3A9pcmOWOX2qzX9kbbd9nea/se2++exqyj6ne5CttPt/3d5vmdtmcnP2W3Wqz5Y7bvbf5ed9hu9Ra5Y1nby5LYfoft2C7xzheC3tsVknYkOV3SjubxSq6UdMdEphqfNuv9p6T3J3mJpHMlfdH2syc448haXq7iEkl/TfIiSddK+uxkp+xWyzX/WtJckpdJuknS5yY7ZbfaXpbE9kmSLpO0c7ITjg9B7+18SVub+1slXdBrJ9uvkLRe0k8nNNe49F1vkt8l2d/c/7Okw5JmJjZhN9pcrmL5n8VNks627QnO2LW+a05ye5J/Ng/v1tJnSlaztpcluVJLv7D/Ncnhxomg97Y+ycHm/kNaivb/sf0USVdL+vgkBxuTvutdzvYmSSdI+v24B+tYm8tV/G+fJEck/U3ScyYy3XgMeomOSyT9aKwTjV/fNds+S9KpSX44ycHGbewf/T9W2f6ZpOf2eOqTyx8kie1ebwX6sKTtSQ6shhdwHaz36M/ZIOmbkjYnebzbKTFNtt8naU7SG6Y9yzg1L8aukfSBKY/SueM26EnOWek524dsb0hysAnY4R67vVrS62x/WNKJkk6w/Y8kx+R14TtYr2w/U9IPJX0yyd1jGnWc2lyu4ug+B2yvkfQsSX+ZzHhj0eoSHbbP0dIv9zck+feEZhuXfms+SdJLJf2ieTH2XEnbbJ+XZGFiU44Bp1x62yZpc3N/s6Rbn7hDkouSnJZkVkunXb5xrMa8hb7rbS7vcIuW1nnTBGfrUpvLVSz/s3inpJ9ndX9Yo++abZ8p6auSzkvS85f5KvOka07ytyTrksw2/37v1tLaV3XMJYK+kqskvdn2fknnNI9le87216c62Xi0We+7JL1e0gds726+Nk5n3OE058SPXq5in6TvJdlr+zO2z2t2u07Sc2zfJ+ljevJ3OB3zWq7581r6r8zvN3+vq/qaTC3XXBKfFAWAIniFDgBFEHQAKIKgA0ARBB0AiiDoAFAEQQeAIgg6ABRB0AGgiP8C9ML+LndAfc4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Plot distribution of differentially expressed genes for simulated experiments\n",
    "sns.distplot(num_sign_DEGs_simulated,\n",
    "            kde=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Observation:** All simulated experiments found 0 DEGs using adjusted p-value cutoff of <5%"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "As a check, we compared the number of DEGs identified here versus what was reported in the [Reddy et. al. publication](https://www.ncbi.nlm.nih.gov//pubmed/19801529), which found:\n",
    "* 234 genes with a significant (FDR < 5%) change in expression in response to DEX treatment. \n",
    "* After removing pseudogenes (listed in Supplemental Table S3), 209 differentially expressed genes remained \n",
    "* Of the DEX-responsive genes, more showed increases in transcript levels 123 (59%) than showed decreases 86 (41%), and the up-regulation was slightly but significantly stronger than the down-regulation\n",
    "\n",
    "By comparison:\n",
    "* Our study found 60 DEGs instead of 234. \n",
    "* Spot checking the genes identified with their list of DEX-responsive genes (Supplementary Dataset 2), we found the same genes and FC direction was consistent though magnitudes of theirs was lower compared to ours. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load association statistics for template experiment\n",
    "template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_real.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>t</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000067082.14</th>\n",
       "      <td>19.385246</td>\n",
       "      <td>18.673139</td>\n",
       "      <td>14.055682</td>\n",
       "      <td>2.342010e-12</td>\n",
       "      <td>1.359233e-07</td>\n",
       "      <td>9.739709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000177606.6</th>\n",
       "      <td>-8.236487</td>\n",
       "      <td>9.185753</td>\n",
       "      <td>-13.126898</td>\n",
       "      <td>8.859555e-12</td>\n",
       "      <td>2.570910e-07</td>\n",
       "      <td>9.292456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000114315.3</th>\n",
       "      <td>-6.285228</td>\n",
       "      <td>6.042663</td>\n",
       "      <td>-12.840532</td>\n",
       "      <td>1.355507e-11</td>\n",
       "      <td>2.622318e-07</td>\n",
       "      <td>9.141666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185950.8</th>\n",
       "      <td>8.790781</td>\n",
       "      <td>17.705828</td>\n",
       "      <td>12.594113</td>\n",
       "      <td>1.966140e-11</td>\n",
       "      <td>2.852721e-07</td>\n",
       "      <td>9.006626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000095752.6</th>\n",
       "      <td>-7.505790</td>\n",
       "      <td>7.299177</td>\n",
       "      <td>-12.204218</td>\n",
       "      <td>3.582671e-11</td>\n",
       "      <td>4.158550e-07</td>\n",
       "      <td>8.782431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        logFC    AveExpr          t       P.Value  \\\n",
       "ENSG00000067082.14  19.385246  18.673139  14.055682  2.342010e-12   \n",
       "ENSG00000177606.6   -8.236487   9.185753 -13.126898  8.859555e-12   \n",
       "ENSG00000114315.3   -6.285228   6.042663 -12.840532  1.355507e-11   \n",
       "ENSG00000185950.8    8.790781  17.705828  12.594113  1.966140e-11   \n",
       "ENSG00000095752.6   -7.505790   7.299177 -12.204218  3.582671e-11   \n",
       "\n",
       "                       adj.P.Val         B  \n",
       "ENSG00000067082.14  1.359233e-07  9.739709  \n",
       "ENSG00000177606.6   2.570910e-07  9.292456  \n",
       "ENSG00000114315.3   2.622318e-07  9.141666  \n",
       "ENSG00000185950.8   2.852721e-07  9.006626  \n",
       "ENSG00000095752.6   4.158550e-07  8.782431  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "template_DE_stats = pd.read_csv(\n",
    "    template_DE_stats_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "\n",
    "template_DEGs = template_DE_stats[template_DE_stats['adj.P.Val']<0.05]\n",
    "print(template_DEGs.shape)\n",
    "template_DEGs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R\n",
    "#library(EnhancedVolcano)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R -i project_id -i template_DE_stats_file -i local_dir\n",
    "\n",
    "#source('../functions/DE_analysis.R')\n",
    "\n",
    "#create_volcano(template_DE_stats_file,\n",
    "#               project_id,\n",
    "#               \"adj.P.Val\",\n",
    "#               local_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene set enrichment analysis\n",
    "\n",
    "**Goal:** To detect modest but coordinated changes in prespecified sets of related genes (i.e. those genes in the same pathway or share the same GO term).\n",
    "\n",
    "1. Ranks all genes based using DE association statistics. In this case we used the p-value scores to rank genes. logFC returned error -- need to look into this.\n",
    "2. An enrichment score (ES) is defined as the maximum distance from the middle of the ranked list. Thus, the enrichment score indicates whether the genes contained in a gene set are clustered towards the beginning or the end of the ranked list (indicating a correlation with change in expression). \n",
    "3. Estimate the statistical significance of the ES by a phenotypic-based permutation test in order to produce a null distribution for the ES( i.e. scores based on permuted phenotype)\n",
    "\n",
    "**Note:** Since there were 0 differentially expressed genes using simulated experiments, we used gene set enrichement analysis instead of over-representation analysis to get ranking of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Registered S3 method overwritten by 'enrichplot':\n",
      "  method               from\n",
      "  fortify.enrichResult DOSE\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: clusterProfiler v3.14.3  For help: https://guangchuangyu.github.io/software/clusterProfiler\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: AnnotationDbi\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: stats4\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: BiocGenerics\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: parallel\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: Biobase\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: IRanges\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: S4Vectors\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: DOSE v3.12.0  For help: https://guangchuangyu.github.io/software/DOSE\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "library(clusterProfiler)\n",
    "library(org.Hs.eg.db)\n",
    "library(DOSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get pathway enrichment for template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/robjects/pandas2ri.py:191: FutureWarning: from_items is deprecated. Please use DataFrame.from_dict(dict(items), ...) instead. DataFrame.from_dict(OrderedDict(items)) may be used to preserve the key order.\n",
      "  res = PandasDataFrame.from_items(items)\n"
     ]
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file  -o enriched_pathways\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "enriched_pathways <- find_enriched_pathways(template_DE_stats_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1028, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MF</td>\n",
       "      <td>GO:0000049</td>\n",
       "      <td>tRNA binding</td>\n",
       "      <td>60</td>\n",
       "      <td>0.525192</td>\n",
       "      <td>1.546860</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>17488</td>\n",
       "      <td>tags=52%, list=30%, signal=36%</td>\n",
       "      <td>ENSG00000124571/ENSG00000129480/ENSG0000007250...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000070</td>\n",
       "      <td>mitotic sister chromatid segregation</td>\n",
       "      <td>152</td>\n",
       "      <td>0.515293</td>\n",
       "      <td>1.604110</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>16866</td>\n",
       "      <td>tags=52%, list=29%, signal=37%</td>\n",
       "      <td>ENSG00000154473/ENSG00000121152/ENSG0000016784...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000075</td>\n",
       "      <td>cell cycle checkpoint</td>\n",
       "      <td>215</td>\n",
       "      <td>0.486637</td>\n",
       "      <td>1.537440</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>16017</td>\n",
       "      <td>tags=47%, list=28%, signal=34%</td>\n",
       "      <td>ENSG00000182973/ENSG00000154473/ENSG0000016841...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000077</td>\n",
       "      <td>DNA damage checkpoint</td>\n",
       "      <td>144</td>\n",
       "      <td>0.487972</td>\n",
       "      <td>1.516464</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>17564</td>\n",
       "      <td>tags=49%, list=30%, signal=34%</td>\n",
       "      <td>ENSG00000182973/ENSG00000168411/ENSG0000013984...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0000082</td>\n",
       "      <td>G1/S transition of mitotic cell cycle</td>\n",
       "      <td>267</td>\n",
       "      <td>0.439629</td>\n",
       "      <td>1.399748</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>14089</td>\n",
       "      <td>tags=37%, list=24%, signal=28%</td>\n",
       "      <td>ENSG00000198987/ENSG00000199157/ENSG0000020759...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ONTOLOGY          ID                            Description  setSize  \\\n",
       "0       MF  GO:0000049                           tRNA binding       60   \n",
       "1       BP  GO:0000070   mitotic sister chromatid segregation      152   \n",
       "2       BP  GO:0000075                  cell cycle checkpoint      215   \n",
       "3       BP  GO:0000077                  DNA damage checkpoint      144   \n",
       "4       BP  GO:0000082  G1/S transition of mitotic cell cycle      267   \n",
       "\n",
       "   enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "0         0.525192  1.546860  0.000999  0.013768  0.011039  17488   \n",
       "1         0.515293  1.604110  0.000999  0.013768  0.011039  16866   \n",
       "2         0.486637  1.537440  0.000999  0.013768  0.011039  16017   \n",
       "3         0.487972  1.516464  0.000999  0.013768  0.011039  17564   \n",
       "4         0.439629  1.399748  0.000999  0.013768  0.011039  14089   \n",
       "\n",
       "                     leading_edge  \\\n",
       "0  tags=52%, list=30%, signal=36%   \n",
       "1  tags=52%, list=29%, signal=37%   \n",
       "2  tags=47%, list=28%, signal=34%   \n",
       "3  tags=49%, list=30%, signal=34%   \n",
       "4  tags=37%, list=24%, signal=28%   \n",
       "\n",
       "                                     core_enrichment  \n",
       "0  ENSG00000124571/ENSG00000129480/ENSG0000007250...  \n",
       "1  ENSG00000154473/ENSG00000121152/ENSG0000016784...  \n",
       "2  ENSG00000182973/ENSG00000154473/ENSG0000016841...  \n",
       "3  ENSG00000182973/ENSG00000168411/ENSG0000013984...  \n",
       "4  ENSG00000198987/ENSG00000199157/ENSG0000020759...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(enriched_pathways.shape)\n",
    "enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get pathway enrichment for simulated experiments**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_0.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_1.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_2.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_3.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_4.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_5.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_6.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_7.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_8.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_9.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_10.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_11.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_12.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_13.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_14.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_15.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_16.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_17.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_18.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_19.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_20.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_21.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_22.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_23.txt...\n",
      "running file: /home/alexandra/Documents/Data/Rank_pathways/DE_stats/DE_stats_simulated_data_SRP000762_24.txt...\n",
      "Error in str_count(res$core_enrichment, \"/\") + 1 : \n",
      "  non-numeric argument to binary operator\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: no term enriched under specific pvalueCutoff...\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Error in str_count(res$core_enrichment, \"/\") + 1 : \n",
      "  non-numeric argument to binary operator\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: In addition: \n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: There were 50 or more warnings (use warnings() to see the first 50)\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "%%R -i project_id -i local_dir -i num_runs \n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_DE_stats_file <- paste(local_dir, \n",
    "                                 \"DE_stats/DE_stats_simulated_data_\", \n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    cat(paste(\"running file: \", simulated_DE_stats_file, \"...\\n\", sep=\"\"))\n",
    "    \n",
    "    enriched_pathways <- find_enriched_pathways(simulated_DE_stats_file)\n",
    "    \n",
    "    out_file = paste(local_dir, \"GSEA_stats/GSEA_simulated_data_\", project_id,\"_\", i, \".txt\", sep=\"\")\n",
    "    write.table(enriched_pathways, file = out_file, row.names = T, sep = \"\\t\", quote = F)\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Template experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_to_rank = 'enrichmentScore'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# zip GO ID and description to get unique key\n",
    "enriched_pathways[\"ID_description\"] = enriched_pathways[\"ID\"] +\"-\"+enriched_pathways[\"Description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ONTOLOGY</th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>setSize</th>\n",
       "      <th>enrichmentScore</th>\n",
       "      <th>NES</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>p.adjust</th>\n",
       "      <th>qvalues</th>\n",
       "      <th>rank</th>\n",
       "      <th>leading_edge</th>\n",
       "      <th>core_enrichment</th>\n",
       "      <th>ID_description</th>\n",
       "      <th>ranking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>MF</td>\n",
       "      <td>GO:0017166</td>\n",
       "      <td>vinculin binding</td>\n",
       "      <td>11</td>\n",
       "      <td>0.764407</td>\n",
       "      <td>1.851843</td>\n",
       "      <td>0.001083</td>\n",
       "      <td>0.014215</td>\n",
       "      <td>0.011397</td>\n",
       "      <td>8431</td>\n",
       "      <td>tags=64%, list=15%, signal=54%</td>\n",
       "      <td>ENSG00000137076/ENSG00000173402/ENSG0000008915...</td>\n",
       "      <td>GO:0017166-vinculin binding</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0019774</td>\n",
       "      <td>proteasome core complex, beta-subunit complex</td>\n",
       "      <td>11</td>\n",
       "      <td>0.756475</td>\n",
       "      <td>1.832628</td>\n",
       "      <td>0.001083</td>\n",
       "      <td>0.014215</td>\n",
       "      <td>0.011397</td>\n",
       "      <td>10764</td>\n",
       "      <td>tags=73%, list=19%, signal=59%</td>\n",
       "      <td>ENSG00000126067/ENSG00000159377/ENSG0000024006...</td>\n",
       "      <td>GO:0019774-proteasome core complex, beta-subun...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0000314</td>\n",
       "      <td>organellar small ribosomal subunit</td>\n",
       "      <td>28</td>\n",
       "      <td>0.744290</td>\n",
       "      <td>2.067121</td>\n",
       "      <td>0.001003</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>12851</td>\n",
       "      <td>tags=75%, list=22%, signal=58%</td>\n",
       "      <td>ENSG00000135972/ENSG00000132676/ENSG0000006258...</td>\n",
       "      <td>GO:0000314-organellar small ribosomal subunit</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005763</td>\n",
       "      <td>mitochondrial small ribosomal subunit</td>\n",
       "      <td>28</td>\n",
       "      <td>0.744290</td>\n",
       "      <td>2.067121</td>\n",
       "      <td>0.001003</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>12851</td>\n",
       "      <td>tags=75%, list=22%, signal=58%</td>\n",
       "      <td>ENSG00000135972/ENSG00000132676/ENSG0000006258...</td>\n",
       "      <td>GO:0005763-mitochondrial small ribosomal subunit</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0000805</td>\n",
       "      <td>X chromosome</td>\n",
       "      <td>10</td>\n",
       "      <td>0.741553</td>\n",
       "      <td>1.759370</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.014389</td>\n",
       "      <td>0.011537</td>\n",
       "      <td>10394</td>\n",
       "      <td>tags=60%, list=18%, signal=49%</td>\n",
       "      <td>ENSG00000121931/ENSG00000132475/ENSG0000012337...</td>\n",
       "      <td>GO:0000805-X chromosome</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0016282</td>\n",
       "      <td>eukaryotic 43S preinitiation complex</td>\n",
       "      <td>17</td>\n",
       "      <td>0.723414</td>\n",
       "      <td>1.900784</td>\n",
       "      <td>0.001017</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>14095</td>\n",
       "      <td>tags=88%, list=24%, signal=67%</td>\n",
       "      <td>ENSG00000184110/ENSG00000100353/ENSG0000020560...</td>\n",
       "      <td>GO:0016282-eukaryotic 43S preinitiation complex</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>861</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:1903441</td>\n",
       "      <td>protein localization to ciliary membrane</td>\n",
       "      <td>11</td>\n",
       "      <td>0.721912</td>\n",
       "      <td>1.748895</td>\n",
       "      <td>0.003250</td>\n",
       "      <td>0.031613</td>\n",
       "      <td>0.025346</td>\n",
       "      <td>11903</td>\n",
       "      <td>tags=73%, list=21%, signal=58%</td>\n",
       "      <td>ENSG00000203965/ENSG00000157796/ENSG0000021441...</td>\n",
       "      <td>GO:1903441-protein localization to ciliary mem...</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0001732</td>\n",
       "      <td>formation of cytoplasmic translation initiatio...</td>\n",
       "      <td>16</td>\n",
       "      <td>0.720854</td>\n",
       "      <td>1.866702</td>\n",
       "      <td>0.001029</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>14095</td>\n",
       "      <td>tags=88%, list=24%, signal=66%</td>\n",
       "      <td>ENSG00000184110/ENSG00000100353/ENSG0000020560...</td>\n",
       "      <td>GO:0001732-formation of cytoplasmic translatio...</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:2000628</td>\n",
       "      <td>regulation of miRNA metabolic process</td>\n",
       "      <td>10</td>\n",
       "      <td>0.719859</td>\n",
       "      <td>1.707898</td>\n",
       "      <td>0.001100</td>\n",
       "      <td>0.014389</td>\n",
       "      <td>0.011537</td>\n",
       "      <td>12225</td>\n",
       "      <td>tags=50%, list=21%, signal=39%</td>\n",
       "      <td>ENSG00000135870/ENSG00000164329/ENSG0000008824...</td>\n",
       "      <td>GO:2000628-regulation of miRNA metabolic process</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0008298</td>\n",
       "      <td>intracellular mRNA localization</td>\n",
       "      <td>13</td>\n",
       "      <td>0.718866</td>\n",
       "      <td>1.798306</td>\n",
       "      <td>0.001052</td>\n",
       "      <td>0.013840</td>\n",
       "      <td>0.011096</td>\n",
       "      <td>13021</td>\n",
       "      <td>tags=54%, list=22%, signal=42%</td>\n",
       "      <td>ENSG00000108349/ENSG00000130713/ENSG0000017182...</td>\n",
       "      <td>GO:0008298-intracellular mRNA localization</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0002566</td>\n",
       "      <td>somatic diversification of immune receptors vi...</td>\n",
       "      <td>15</td>\n",
       "      <td>0.715621</td>\n",
       "      <td>1.839337</td>\n",
       "      <td>0.001033</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>12107</td>\n",
       "      <td>tags=67%, list=21%, signal=53%</td>\n",
       "      <td>ENSG00000160294/ENSG00000076242/ENSG0000007624...</td>\n",
       "      <td>GO:0002566-somatic diversification of immune r...</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005852</td>\n",
       "      <td>eukaryotic translation initiation factor 3 com...</td>\n",
       "      <td>16</td>\n",
       "      <td>0.715060</td>\n",
       "      <td>1.851698</td>\n",
       "      <td>0.001029</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>14095</td>\n",
       "      <td>tags=81%, list=24%, signal=62%</td>\n",
       "      <td>ENSG00000121022/ENSG00000184110/ENSG0000010035...</td>\n",
       "      <td>GO:0005852-eukaryotic translation initiation f...</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0016446</td>\n",
       "      <td>somatic hypermutation of immunoglobulin genes</td>\n",
       "      <td>14</td>\n",
       "      <td>0.708459</td>\n",
       "      <td>1.796829</td>\n",
       "      <td>0.001043</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>12107</td>\n",
       "      <td>tags=64%, list=21%, signal=51%</td>\n",
       "      <td>ENSG00000160294/ENSG00000076242/ENSG0000007624...</td>\n",
       "      <td>GO:0016446-somatic hypermutation of immunoglob...</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0030952</td>\n",
       "      <td>establishment or maintenance of cytoskeleton p...</td>\n",
       "      <td>16</td>\n",
       "      <td>0.707121</td>\n",
       "      <td>1.831139</td>\n",
       "      <td>0.001029</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>11281</td>\n",
       "      <td>tags=50%, list=19%, signal=40%</td>\n",
       "      <td>ENSG00000142945/ENSG00000115963/ENSG0000011948...</td>\n",
       "      <td>GO:0030952-establishment or maintenance of cyt...</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949</th>\n",
       "      <td>MF</td>\n",
       "      <td>GO:0030621</td>\n",
       "      <td>U4 snRNA binding</td>\n",
       "      <td>11</td>\n",
       "      <td>0.706812</td>\n",
       "      <td>1.712315</td>\n",
       "      <td>0.004334</td>\n",
       "      <td>0.038166</td>\n",
       "      <td>0.030600</td>\n",
       "      <td>14618</td>\n",
       "      <td>tags=91%, list=25%, signal=68%</td>\n",
       "      <td>ENSG00000202337/ENSG00000075856/ENSG0000008251...</td>\n",
       "      <td>GO:0030621-U4 snRNA binding</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0002178</td>\n",
       "      <td>palmitoyltransferase complex</td>\n",
       "      <td>11</td>\n",
       "      <td>0.702940</td>\n",
       "      <td>1.702935</td>\n",
       "      <td>0.004334</td>\n",
       "      <td>0.038166</td>\n",
       "      <td>0.030600</td>\n",
       "      <td>7909</td>\n",
       "      <td>tags=55%, list=14%, signal=47%</td>\n",
       "      <td>ENSG00000123353/ENSG00000100596/ENSG0000012869...</td>\n",
       "      <td>GO:0002178-palmitoyltransferase complex</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636</th>\n",
       "      <td>MF</td>\n",
       "      <td>GO:0016251</td>\n",
       "      <td>RNA polymerase II general transcription initia...</td>\n",
       "      <td>13</td>\n",
       "      <td>0.702377</td>\n",
       "      <td>1.757057</td>\n",
       "      <td>0.001052</td>\n",
       "      <td>0.013840</td>\n",
       "      <td>0.011096</td>\n",
       "      <td>11844</td>\n",
       "      <td>tags=62%, list=20%, signal=49%</td>\n",
       "      <td>ENSG00000145736/ENSG00000111358/ENSG0000012565...</td>\n",
       "      <td>GO:0016251-RNA polymerase II general transcrip...</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0061635</td>\n",
       "      <td>regulation of protein complex stability</td>\n",
       "      <td>10</td>\n",
       "      <td>0.701015</td>\n",
       "      <td>1.663191</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.038672</td>\n",
       "      <td>0.031007</td>\n",
       "      <td>11847</td>\n",
       "      <td>tags=60%, list=20%, signal=48%</td>\n",
       "      <td>ENSG00000168538/ENSG00000126775/ENSG0000015921...</td>\n",
       "      <td>GO:0061635-regulation of protein complex stabi...</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0034770</td>\n",
       "      <td>histone H4-K20 methylation</td>\n",
       "      <td>10</td>\n",
       "      <td>0.700766</td>\n",
       "      <td>1.662601</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>0.038672</td>\n",
       "      <td>0.031007</td>\n",
       "      <td>16046</td>\n",
       "      <td>tags=70%, list=28%, signal=51%</td>\n",
       "      <td>ENSG00000136045/ENSG00000109685/ENSG0000011006...</td>\n",
       "      <td>GO:0034770-histone H4-K20 methylation</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>951</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0072321</td>\n",
       "      <td>chaperone-mediated protein transport</td>\n",
       "      <td>11</td>\n",
       "      <td>0.699857</td>\n",
       "      <td>1.695466</td>\n",
       "      <td>0.004334</td>\n",
       "      <td>0.038166</td>\n",
       "      <td>0.030600</td>\n",
       "      <td>14809</td>\n",
       "      <td>tags=73%, list=26%, signal=54%</td>\n",
       "      <td>ENSG00000162735/ENSG00000150779/ENSG0000012088...</td>\n",
       "      <td>GO:0072321-chaperone-mediated protein transport</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0006098</td>\n",
       "      <td>pentose-phosphate shunt</td>\n",
       "      <td>15</td>\n",
       "      <td>0.699838</td>\n",
       "      <td>1.798773</td>\n",
       "      <td>0.001033</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>8045</td>\n",
       "      <td>tags=53%, list=14%, signal=46%</td>\n",
       "      <td>ENSG00000235376/ENSG00000177156/ENSG0000016929...</td>\n",
       "      <td>GO:0006098-pentose-phosphate shunt</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0070993</td>\n",
       "      <td>translation preinitiation complex</td>\n",
       "      <td>18</td>\n",
       "      <td>0.699708</td>\n",
       "      <td>1.845247</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>14095</td>\n",
       "      <td>tags=83%, list=24%, signal=63%</td>\n",
       "      <td>ENSG00000184110/ENSG00000100353/ENSG0000020560...</td>\n",
       "      <td>GO:0070993-translation preinitiation complex</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>950</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0046607</td>\n",
       "      <td>positive regulation of centrosome cycle</td>\n",
       "      <td>11</td>\n",
       "      <td>0.697401</td>\n",
       "      <td>1.689516</td>\n",
       "      <td>0.004334</td>\n",
       "      <td>0.038166</td>\n",
       "      <td>0.030600</td>\n",
       "      <td>16866</td>\n",
       "      <td>tags=82%, list=29%, signal=58%</td>\n",
       "      <td>ENSG00000151849/ENSG00000168944/ENSG0000009990...</td>\n",
       "      <td>GO:0046607-positive regulation of centrosome c...</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>944</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0030950</td>\n",
       "      <td>establishment or maintenance of actin cytoskel...</td>\n",
       "      <td>12</td>\n",
       "      <td>0.691901</td>\n",
       "      <td>1.702927</td>\n",
       "      <td>0.004255</td>\n",
       "      <td>0.037634</td>\n",
       "      <td>0.030174</td>\n",
       "      <td>10547</td>\n",
       "      <td>tags=42%, list=18%, signal=34%</td>\n",
       "      <td>ENSG00000115963/ENSG00000119487/ENSG0000016000...</td>\n",
       "      <td>GO:0030950-establishment or maintenance of act...</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0032300</td>\n",
       "      <td>mismatch repair complex</td>\n",
       "      <td>10</td>\n",
       "      <td>0.690289</td>\n",
       "      <td>1.637744</td>\n",
       "      <td>0.005501</td>\n",
       "      <td>0.046071</td>\n",
       "      <td>0.036938</td>\n",
       "      <td>12133</td>\n",
       "      <td>tags=60%, list=21%, signal=47%</td>\n",
       "      <td>ENSG00000119684/ENSG00000076242/ENSG0000009500...</td>\n",
       "      <td>GO:0032300-mismatch repair complex</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>947</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0090085</td>\n",
       "      <td>regulation of protein deubiquitination</td>\n",
       "      <td>12</td>\n",
       "      <td>0.689895</td>\n",
       "      <td>1.697990</td>\n",
       "      <td>0.004255</td>\n",
       "      <td>0.037634</td>\n",
       "      <td>0.030174</td>\n",
       "      <td>15629</td>\n",
       "      <td>tags=83%, list=27%, signal=61%</td>\n",
       "      <td>ENSG00000075856/ENSG00000136560/ENSG0000010537...</td>\n",
       "      <td>GO:0090085-regulation of protein deubiquitination</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>945</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0031126</td>\n",
       "      <td>snoRNA 3'-end processing</td>\n",
       "      <td>12</td>\n",
       "      <td>0.689533</td>\n",
       "      <td>1.697099</td>\n",
       "      <td>0.004255</td>\n",
       "      <td>0.037634</td>\n",
       "      <td>0.030174</td>\n",
       "      <td>13021</td>\n",
       "      <td>tags=58%, list=22%, signal=45%</td>\n",
       "      <td>ENSG00000140694/ENSG00000188573/ENSG0000013071...</td>\n",
       "      <td>GO:0031126-snoRNA 3'-end processing</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005680</td>\n",
       "      <td>anaphase-promoting complex</td>\n",
       "      <td>21</td>\n",
       "      <td>0.689314</td>\n",
       "      <td>1.847376</td>\n",
       "      <td>0.001018</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>13011</td>\n",
       "      <td>tags=57%, list=22%, signal=44%</td>\n",
       "      <td>ENSG00000164162/ENSG00000110200/ENSG0000005390...</td>\n",
       "      <td>GO:0005680-anaphase-promoting complex</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0098644</td>\n",
       "      <td>complex of collagen trimers</td>\n",
       "      <td>19</td>\n",
       "      <td>0.688426</td>\n",
       "      <td>1.826608</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>12526</td>\n",
       "      <td>tags=63%, list=22%, signal=50%</td>\n",
       "      <td>ENSG00000164692/ENSG00000197565/ENSG0000013487...</td>\n",
       "      <td>GO:0098644-complex of collagen trimers</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0033523</td>\n",
       "      <td>histone H2B ubiquitination</td>\n",
       "      <td>12</td>\n",
       "      <td>0.688249</td>\n",
       "      <td>1.693937</td>\n",
       "      <td>0.004255</td>\n",
       "      <td>0.037634</td>\n",
       "      <td>0.030174</td>\n",
       "      <td>10031</td>\n",
       "      <td>tags=50%, list=17%, signal=41%</td>\n",
       "      <td>ENSG00000198730/ENSG00000006712/ENSG0000015582...</td>\n",
       "      <td>GO:0033523-histone H2B ubiquitination</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0031983</td>\n",
       "      <td>vesicle lumen</td>\n",
       "      <td>339</td>\n",
       "      <td>0.396306</td>\n",
       "      <td>1.270940</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>15441</td>\n",
       "      <td>tags=33%, list=27%, signal=25%</td>\n",
       "      <td>ENSG00000196230/ENSG00000100030/ENSG0000011181...</td>\n",
       "      <td>GO:0031983-vesicle lumen</td>\n",
       "      <td>999.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0060205</td>\n",
       "      <td>cytoplasmic vesicle lumen</td>\n",
       "      <td>338</td>\n",
       "      <td>0.394477</td>\n",
       "      <td>1.265232</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>15441</td>\n",
       "      <td>tags=33%, list=27%, signal=24%</td>\n",
       "      <td>ENSG00000196230/ENSG00000100030/ENSG0000011181...</td>\n",
       "      <td>GO:0060205-cytoplasmic vesicle lumen</td>\n",
       "      <td>1000.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0071383</td>\n",
       "      <td>cellular response to steroid hormone stimulus</td>\n",
       "      <td>250</td>\n",
       "      <td>0.394366</td>\n",
       "      <td>1.254128</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>0.036443</td>\n",
       "      <td>0.029220</td>\n",
       "      <td>18030</td>\n",
       "      <td>tags=36%, list=31%, signal=25%</td>\n",
       "      <td>ENSG00000004478/ENSG00000119138/ENSG0000012787...</td>\n",
       "      <td>GO:0071383-cellular response to steroid hormon...</td>\n",
       "      <td>1001.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005765</td>\n",
       "      <td>lysosomal membrane</td>\n",
       "      <td>352</td>\n",
       "      <td>0.394360</td>\n",
       "      <td>1.266250</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>16911</td>\n",
       "      <td>tags=34%, list=29%, signal=24%</td>\n",
       "      <td>ENSG00000124333/ENSG00000185896/ENSG0000016120...</td>\n",
       "      <td>GO:0005765-lysosomal membrane</td>\n",
       "      <td>1002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0001701</td>\n",
       "      <td>in utero embryonic development</td>\n",
       "      <td>373</td>\n",
       "      <td>0.394255</td>\n",
       "      <td>1.268751</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>17496</td>\n",
       "      <td>tags=35%, list=30%, signal=25%</td>\n",
       "      <td>ENSG00000013275/ENSG00000100030/ENSG0000015779...</td>\n",
       "      <td>GO:0001701-in utero embryonic development</td>\n",
       "      <td>1003.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0072659</td>\n",
       "      <td>protein localization to plasma membrane</td>\n",
       "      <td>258</td>\n",
       "      <td>0.393945</td>\n",
       "      <td>1.254067</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>0.042558</td>\n",
       "      <td>0.034122</td>\n",
       "      <td>18599</td>\n",
       "      <td>tags=40%, list=32%, signal=27%</td>\n",
       "      <td>ENSG00000124333/ENSG00000161203/ENSG0000006840...</td>\n",
       "      <td>GO:0072659-protein localization to plasma memb...</td>\n",
       "      <td>1004.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0098852</td>\n",
       "      <td>lytic vacuole membrane</td>\n",
       "      <td>353</td>\n",
       "      <td>0.393763</td>\n",
       "      <td>1.264345</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>16911</td>\n",
       "      <td>tags=34%, list=29%, signal=24%</td>\n",
       "      <td>ENSG00000124333/ENSG00000185896/ENSG0000016120...</td>\n",
       "      <td>GO:0098852-lytic vacuole membrane</td>\n",
       "      <td>1005.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005774</td>\n",
       "      <td>vacuolar membrane</td>\n",
       "      <td>409</td>\n",
       "      <td>0.393106</td>\n",
       "      <td>1.267918</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>16734</td>\n",
       "      <td>tags=33%, list=29%, signal=24%</td>\n",
       "      <td>ENSG00000124333/ENSG00000185896/ENSG0000016120...</td>\n",
       "      <td>GO:0005774-vacuolar membrane</td>\n",
       "      <td>1006.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0060070</td>\n",
       "      <td>canonical Wnt signaling pathway</td>\n",
       "      <td>330</td>\n",
       "      <td>0.388226</td>\n",
       "      <td>1.244439</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>19170</td>\n",
       "      <td>tags=40%, list=33%, signal=27%</td>\n",
       "      <td>ENSG00000199017/ENSG00000283159/ENSG0000001327...</td>\n",
       "      <td>GO:0060070-canonical Wnt signaling pathway</td>\n",
       "      <td>1007.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0051188</td>\n",
       "      <td>cofactor biosynthetic process</td>\n",
       "      <td>325</td>\n",
       "      <td>0.387934</td>\n",
       "      <td>1.242964</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>18737</td>\n",
       "      <td>tags=40%, list=32%, signal=27%</td>\n",
       "      <td>ENSG00000030066/ENSG00000135002/ENSG0000007518...</td>\n",
       "      <td>GO:0051188-cofactor biosynthetic process</td>\n",
       "      <td>1008.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0031965</td>\n",
       "      <td>nuclear membrane</td>\n",
       "      <td>296</td>\n",
       "      <td>0.387520</td>\n",
       "      <td>1.237746</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.029850</td>\n",
       "      <td>0.023933</td>\n",
       "      <td>16866</td>\n",
       "      <td>tags=36%, list=29%, signal=26%</td>\n",
       "      <td>ENSG00000174574/ENSG00000169905/ENSG0000005465...</td>\n",
       "      <td>GO:0031965-nuclear membrane</td>\n",
       "      <td>1009.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>672</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0018209</td>\n",
       "      <td>peptidyl-serine modification</td>\n",
       "      <td>321</td>\n",
       "      <td>0.386986</td>\n",
       "      <td>1.239415</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>16420</td>\n",
       "      <td>tags=34%, list=28%, signal=24%</td>\n",
       "      <td>ENSG00000144278/ENSG00000100030/ENSG0000013604...</td>\n",
       "      <td>GO:0018209-peptidyl-serine modification</td>\n",
       "      <td>1010.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0018105</td>\n",
       "      <td>peptidyl-serine phosphorylation</td>\n",
       "      <td>298</td>\n",
       "      <td>0.385427</td>\n",
       "      <td>1.230808</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>0.036443</td>\n",
       "      <td>0.029220</td>\n",
       "      <td>16420</td>\n",
       "      <td>tags=34%, list=28%, signal=25%</td>\n",
       "      <td>ENSG00000100030/ENSG00000136045/ENSG0000015418...</td>\n",
       "      <td>GO:0018105-peptidyl-serine phosphorylation</td>\n",
       "      <td>1011.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0010256</td>\n",
       "      <td>endomembrane system organization</td>\n",
       "      <td>438</td>\n",
       "      <td>0.385356</td>\n",
       "      <td>1.243962</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>18772</td>\n",
       "      <td>tags=39%, list=32%, signal=26%</td>\n",
       "      <td>ENSG00000102606/ENSG00000130340/ENSG0000010046...</td>\n",
       "      <td>GO:0010256-endomembrane system organization</td>\n",
       "      <td>1012.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0010008</td>\n",
       "      <td>endosome membrane</td>\n",
       "      <td>476</td>\n",
       "      <td>0.384863</td>\n",
       "      <td>1.245179</td>\n",
       "      <td>0.000999</td>\n",
       "      <td>0.013768</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>15491</td>\n",
       "      <td>tags=31%, list=27%, signal=23%</td>\n",
       "      <td>ENSG00000124333/ENSG00000090487/ENSG0000018589...</td>\n",
       "      <td>GO:0010008-endosome membrane</td>\n",
       "      <td>1013.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0005874</td>\n",
       "      <td>microtubule</td>\n",
       "      <td>413</td>\n",
       "      <td>0.384218</td>\n",
       "      <td>1.239570</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>19288</td>\n",
       "      <td>tags=39%, list=33%, signal=26%</td>\n",
       "      <td>ENSG00000196230/ENSG00000264364/ENSG0000017520...</td>\n",
       "      <td>GO:0005874-microtubule</td>\n",
       "      <td>1014.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0051098</td>\n",
       "      <td>regulation of binding</td>\n",
       "      <td>373</td>\n",
       "      <td>0.384008</td>\n",
       "      <td>1.235776</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>16432</td>\n",
       "      <td>tags=34%, list=28%, signal=24%</td>\n",
       "      <td>ENSG00000154813/ENSG00000162735/ENSG0000012348...</td>\n",
       "      <td>GO:0051098-regulation of binding</td>\n",
       "      <td>1015.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0001666</td>\n",
       "      <td>response to hypoxia</td>\n",
       "      <td>354</td>\n",
       "      <td>0.383594</td>\n",
       "      <td>1.231538</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>17917</td>\n",
       "      <td>tags=37%, list=31%, signal=26%</td>\n",
       "      <td>ENSG00000013275/ENSG00000129993/ENSG0000010076...</td>\n",
       "      <td>GO:0001666-response to hypoxia</td>\n",
       "      <td>1016.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0030111</td>\n",
       "      <td>regulation of Wnt signaling pathway</td>\n",
       "      <td>358</td>\n",
       "      <td>0.383383</td>\n",
       "      <td>1.231813</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>19113</td>\n",
       "      <td>tags=40%, list=33%, signal=27%</td>\n",
       "      <td>ENSG00000199017/ENSG00000283159/ENSG0000001327...</td>\n",
       "      <td>GO:0030111-regulation of Wnt signaling pathway</td>\n",
       "      <td>1017.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>MF</td>\n",
       "      <td>GO:0004842</td>\n",
       "      <td>ubiquitin-protein transferase activity</td>\n",
       "      <td>382</td>\n",
       "      <td>0.382804</td>\n",
       "      <td>1.232210</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>17188</td>\n",
       "      <td>tags=36%, list=30%, signal=25%</td>\n",
       "      <td>ENSG00000197579/ENSG00000198833/ENSG0000002284...</td>\n",
       "      <td>GO:0004842-ubiquitin-protein transferase activity</td>\n",
       "      <td>1018.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>786</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0016050</td>\n",
       "      <td>vesicle organization</td>\n",
       "      <td>322</td>\n",
       "      <td>0.382371</td>\n",
       "      <td>1.224671</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.029850</td>\n",
       "      <td>0.023933</td>\n",
       "      <td>15768</td>\n",
       "      <td>tags=34%, list=27%, signal=25%</td>\n",
       "      <td>ENSG00000124333/ENSG00000161203/ENSG0000013673...</td>\n",
       "      <td>GO:0016050-vesicle organization</td>\n",
       "      <td>1019.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>789</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0019058</td>\n",
       "      <td>viral life cycle</td>\n",
       "      <td>327</td>\n",
       "      <td>0.382240</td>\n",
       "      <td>1.224723</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.029850</td>\n",
       "      <td>0.023933</td>\n",
       "      <td>16920</td>\n",
       "      <td>tags=36%, list=29%, signal=26%</td>\n",
       "      <td>ENSG00000185896/ENSG00000030066/ENSG0000024450...</td>\n",
       "      <td>GO:0019058-viral life cycle</td>\n",
       "      <td>1020.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0043900</td>\n",
       "      <td>regulation of multi-organism process</td>\n",
       "      <td>404</td>\n",
       "      <td>0.381951</td>\n",
       "      <td>1.231538</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>17252</td>\n",
       "      <td>tags=36%, list=30%, signal=25%</td>\n",
       "      <td>ENSG00000177684/ENSG00000213246/ENSG0000010885...</td>\n",
       "      <td>GO:0043900-regulation of multi-organism process</td>\n",
       "      <td>1021.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0036293</td>\n",
       "      <td>response to decreased oxygen levels</td>\n",
       "      <td>365</td>\n",
       "      <td>0.379600</td>\n",
       "      <td>1.221273</td>\n",
       "      <td>0.002997</td>\n",
       "      <td>0.029850</td>\n",
       "      <td>0.023933</td>\n",
       "      <td>17917</td>\n",
       "      <td>tags=36%, list=31%, signal=25%</td>\n",
       "      <td>ENSG00000013275/ENSG00000129993/ENSG0000010076...</td>\n",
       "      <td>GO:0036293-response to decreased oxygen levels</td>\n",
       "      <td>1022.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0070482</td>\n",
       "      <td>response to oxygen levels</td>\n",
       "      <td>389</td>\n",
       "      <td>0.378132</td>\n",
       "      <td>1.217400</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>17945</td>\n",
       "      <td>tags=36%, list=31%, signal=25%</td>\n",
       "      <td>ENSG00000013275/ENSG00000129993/ENSG0000010076...</td>\n",
       "      <td>GO:0070482-response to oxygen levels</td>\n",
       "      <td>1023.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0062012</td>\n",
       "      <td>regulation of small molecule metabolic process</td>\n",
       "      <td>457</td>\n",
       "      <td>0.377729</td>\n",
       "      <td>1.221372</td>\n",
       "      <td>0.001998</td>\n",
       "      <td>0.022279</td>\n",
       "      <td>0.017863</td>\n",
       "      <td>15298</td>\n",
       "      <td>tags=33%, list=26%, signal=25%</td>\n",
       "      <td>ENSG00000207691/ENSG00000221263/ENSG0000015934...</td>\n",
       "      <td>GO:0062012-regulation of small molecule metabo...</td>\n",
       "      <td>1024.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0032386</td>\n",
       "      <td>regulation of intracellular transport</td>\n",
       "      <td>420</td>\n",
       "      <td>0.376271</td>\n",
       "      <td>1.214648</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>0.036443</td>\n",
       "      <td>0.029220</td>\n",
       "      <td>18599</td>\n",
       "      <td>tags=36%, list=32%, signal=25%</td>\n",
       "      <td>ENSG00000124333/ENSG00000113645/ENSG0000007284...</td>\n",
       "      <td>GO:0032386-regulation of intracellular transport</td>\n",
       "      <td>1025.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>962</th>\n",
       "      <td>CC</td>\n",
       "      <td>GO:0031252</td>\n",
       "      <td>cell leading edge</td>\n",
       "      <td>404</td>\n",
       "      <td>0.374048</td>\n",
       "      <td>1.206058</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>0.042558</td>\n",
       "      <td>0.034122</td>\n",
       "      <td>17497</td>\n",
       "      <td>tags=35%, list=30%, signal=24%</td>\n",
       "      <td>ENSG00000124333/ENSG00000113645/ENSG0000010260...</td>\n",
       "      <td>GO:0031252-cell leading edge</td>\n",
       "      <td>1026.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0016053</td>\n",
       "      <td>organic acid biosynthetic process</td>\n",
       "      <td>461</td>\n",
       "      <td>0.366857</td>\n",
       "      <td>1.186727</td>\n",
       "      <td>0.003996</td>\n",
       "      <td>0.036443</td>\n",
       "      <td>0.029220</td>\n",
       "      <td>17983</td>\n",
       "      <td>tags=36%, list=31%, signal=25%</td>\n",
       "      <td>ENSG00000207691/ENSG00000221263/ENSG0000015934...</td>\n",
       "      <td>GO:0016053-organic acid biosynthetic process</td>\n",
       "      <td>1027.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>BP</td>\n",
       "      <td>GO:0046394</td>\n",
       "      <td>carboxylic acid biosynthetic process</td>\n",
       "      <td>460</td>\n",
       "      <td>0.365890</td>\n",
       "      <td>1.183553</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>0.049269</td>\n",
       "      <td>0.039503</td>\n",
       "      <td>17983</td>\n",
       "      <td>tags=36%, list=31%, signal=25%</td>\n",
       "      <td>ENSG00000207691/ENSG00000221263/ENSG0000015934...</td>\n",
       "      <td>GO:0046394-carboxylic acid biosynthetic process</td>\n",
       "      <td>1028.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1028 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     ONTOLOGY          ID                                        Description  \\\n",
       "637        MF  GO:0017166                                   vinculin binding   \n",
       "638        CC  GO:0019774      proteasome core complex, beta-subunit complex   \n",
       "592        CC  GO:0000314                 organellar small ribosomal subunit   \n",
       "594        CC  GO:0005763              mitochondrial small ribosomal subunit   \n",
       "639        CC  GO:0000805                                       X chromosome   \n",
       "608        CC  GO:0016282               eukaryotic 43S preinitiation complex   \n",
       "861        BP  GO:1903441           protein localization to ciliary membrane   \n",
       "615        BP  GO:0001732  formation of cytoplasmic translation initiatio...   \n",
       "640        BP  GO:2000628              regulation of miRNA metabolic process   \n",
       "635        BP  GO:0008298                    intracellular mRNA localization   \n",
       "619        BP  GO:0002566  somatic diversification of immune receptors vi...   \n",
       "616        CC  GO:0005852  eukaryotic translation initiation factor 3 com...   \n",
       "628        BP  GO:0016446      somatic hypermutation of immunoglobulin genes   \n",
       "617        BP  GO:0030952  establishment or maintenance of cytoskeleton p...   \n",
       "949        MF  GO:0030621                                   U4 snRNA binding   \n",
       "948        CC  GO:0002178                       palmitoyltransferase complex   \n",
       "636        MF  GO:0016251  RNA polymerase II general transcription initia...   \n",
       "953        BP  GO:0061635            regulation of protein complex stability   \n",
       "952        BP  GO:0034770                         histone H4-K20 methylation   \n",
       "951        BP  GO:0072321               chaperone-mediated protein transport   \n",
       "620        BP  GO:0006098                            pentose-phosphate shunt   \n",
       "613        CC  GO:0070993                  translation preinitiation complex   \n",
       "950        BP  GO:0046607            positive regulation of centrosome cycle   \n",
       "944        BP  GO:0030950  establishment or maintenance of actin cytoskel...   \n",
       "1000       CC  GO:0032300                            mismatch repair complex   \n",
       "947        BP  GO:0090085             regulation of protein deubiquitination   \n",
       "945        BP  GO:0031126                           snoRNA 3'-end processing   \n",
       "609        CC  GO:0005680                         anaphase-promoting complex   \n",
       "614        CC  GO:0098644                        complex of collagen trimers   \n",
       "946        BP  GO:0033523                         histone H2B ubiquitination   \n",
       "...       ...         ...                                                ...   \n",
       "280        CC  GO:0031983                                      vesicle lumen   \n",
       "420        CC  GO:0060205                          cytoplasmic vesicle lumen   \n",
       "895        BP  GO:0071383      cellular response to steroid hormone stimulus   \n",
       "63         CC  GO:0005765                                 lysosomal membrane   \n",
       "30         BP  GO:0001701                     in utero embryonic development   \n",
       "976        BP  GO:0072659            protein localization to plasma membrane   \n",
       "477        CC  GO:0098852                             lytic vacuole membrane   \n",
       "64         CC  GO:0005774                                  vacuolar membrane   \n",
       "702        BP  GO:0060070                    canonical Wnt signaling pathway   \n",
       "701        BP  GO:0051188                      cofactor biosynthetic process   \n",
       "792        CC  GO:0031965                                   nuclear membrane   \n",
       "672        BP  GO:0018209                       peptidyl-serine modification   \n",
       "874        BP  GO:0018105                    peptidyl-serine phosphorylation   \n",
       "182        BP  GO:0010256                   endomembrane system organization   \n",
       "181        CC  GO:0010008                                  endosome membrane   \n",
       "651        CC  GO:0005874                                        microtubule   \n",
       "699        BP  GO:0051098                              regulation of binding   \n",
       "641        BP  GO:0001666                                response to hypoxia   \n",
       "678        BP  GO:0030111                regulation of Wnt signaling pathway   \n",
       "647        MF  GO:0004842             ubiquitin-protein transferase activity   \n",
       "786        BP  GO:0016050                               vesicle organization   \n",
       "789        BP  GO:0019058                                   viral life cycle   \n",
       "692        BP  GO:0043900               regulation of multi-organism process   \n",
       "800        BP  GO:0036293                response to decreased oxygen levels   \n",
       "707        BP  GO:0070482                          response to oxygen levels   \n",
       "706        BP  GO:0062012     regulation of small molecule metabolic process   \n",
       "879        BP  GO:0032386              regulation of intracellular transport   \n",
       "962        CC  GO:0031252                                  cell leading edge   \n",
       "871        BP  GO:0016053                  organic acid biosynthetic process   \n",
       "1010       BP  GO:0046394               carboxylic acid biosynthetic process   \n",
       "\n",
       "      setSize  enrichmentScore       NES    pvalue  p.adjust   qvalues   rank  \\\n",
       "637        11         0.764407  1.851843  0.001083  0.014215  0.011397   8431   \n",
       "638        11         0.756475  1.832628  0.001083  0.014215  0.011397  10764   \n",
       "592        28         0.744290  2.067121  0.001003  0.013768  0.011039  12851   \n",
       "594        28         0.744290  2.067121  0.001003  0.013768  0.011039  12851   \n",
       "639        10         0.741553  1.759370  0.001100  0.014389  0.011537  10394   \n",
       "608        17         0.723414  1.900784  0.001017  0.013768  0.011039  14095   \n",
       "861        11         0.721912  1.748895  0.003250  0.031613  0.025346  11903   \n",
       "615        16         0.720854  1.866702  0.001029  0.013768  0.011039  14095   \n",
       "640        10         0.719859  1.707898  0.001100  0.014389  0.011537  12225   \n",
       "635        13         0.718866  1.798306  0.001052  0.013840  0.011096  13021   \n",
       "619        15         0.715621  1.839337  0.001033  0.013768  0.011039  12107   \n",
       "616        16         0.715060  1.851698  0.001029  0.013768  0.011039  14095   \n",
       "628        14         0.708459  1.796829  0.001043  0.013768  0.011039  12107   \n",
       "617        16         0.707121  1.831139  0.001029  0.013768  0.011039  11281   \n",
       "949        11         0.706812  1.712315  0.004334  0.038166  0.030600  14618   \n",
       "948        11         0.702940  1.702935  0.004334  0.038166  0.030600   7909   \n",
       "636        13         0.702377  1.757057  0.001052  0.013840  0.011096  11844   \n",
       "953        10         0.701015  1.663191  0.004400  0.038672  0.031007  11847   \n",
       "952        10         0.700766  1.662601  0.004400  0.038672  0.031007  16046   \n",
       "951        11         0.699857  1.695466  0.004334  0.038166  0.030600  14809   \n",
       "620        15         0.699838  1.798773  0.001033  0.013768  0.011039   8045   \n",
       "613        18         0.699708  1.845247  0.001020  0.013768  0.011039  14095   \n",
       "950        11         0.697401  1.689516  0.004334  0.038166  0.030600  16866   \n",
       "944        12         0.691901  1.702927  0.004255  0.037634  0.030174  10547   \n",
       "1000       10         0.690289  1.637744  0.005501  0.046071  0.036938  12133   \n",
       "947        12         0.689895  1.697990  0.004255  0.037634  0.030174  15629   \n",
       "945        12         0.689533  1.697099  0.004255  0.037634  0.030174  13021   \n",
       "609        21         0.689314  1.847376  0.001018  0.013768  0.011039  13011   \n",
       "614        19         0.688426  1.826608  0.001020  0.013768  0.011039  12526   \n",
       "946        12         0.688249  1.693937  0.004255  0.037634  0.030174  10031   \n",
       "...       ...              ...       ...       ...       ...       ...    ...   \n",
       "280       339         0.396306  1.270940  0.000999  0.013768  0.011039  15441   \n",
       "420       338         0.394477  1.265232  0.000999  0.013768  0.011039  15441   \n",
       "895       250         0.394366  1.254128  0.003996  0.036443  0.029220  18030   \n",
       "63        352         0.394360  1.266250  0.000999  0.013768  0.011039  16911   \n",
       "30        373         0.394255  1.268751  0.000999  0.013768  0.011039  17496   \n",
       "976       258         0.393945  1.254067  0.004995  0.042558  0.034122  18599   \n",
       "477       353         0.393763  1.264345  0.000999  0.013768  0.011039  16911   \n",
       "64        409         0.393106  1.267918  0.000999  0.013768  0.011039  16734   \n",
       "702       330         0.388226  1.244439  0.001998  0.022279  0.017863  19170   \n",
       "701       325         0.387934  1.242964  0.001998  0.022279  0.017863  18737   \n",
       "792       296         0.387520  1.237746  0.002997  0.029850  0.023933  16866   \n",
       "672       321         0.386986  1.239415  0.001998  0.022279  0.017863  16420   \n",
       "874       298         0.385427  1.230808  0.003996  0.036443  0.029220  16420   \n",
       "182       438         0.385356  1.243962  0.000999  0.013768  0.011039  18772   \n",
       "181       476         0.384863  1.245179  0.000999  0.013768  0.011039  15491   \n",
       "651       413         0.384218  1.239570  0.001998  0.022279  0.017863  19288   \n",
       "699       373         0.384008  1.235776  0.001998  0.022279  0.017863  16432   \n",
       "641       354         0.383594  1.231538  0.001998  0.022279  0.017863  17917   \n",
       "678       358         0.383383  1.231813  0.001998  0.022279  0.017863  19113   \n",
       "647       382         0.382804  1.232210  0.001998  0.022279  0.017863  17188   \n",
       "786       322         0.382371  1.224671  0.002997  0.029850  0.023933  15768   \n",
       "789       327         0.382240  1.224723  0.002997  0.029850  0.023933  16920   \n",
       "692       404         0.381951  1.231538  0.001998  0.022279  0.017863  17252   \n",
       "800       365         0.379600  1.221273  0.002997  0.029850  0.023933  17917   \n",
       "707       389         0.378132  1.217400  0.001998  0.022279  0.017863  17945   \n",
       "706       457         0.377729  1.221372  0.001998  0.022279  0.017863  15298   \n",
       "879       420         0.376271  1.214648  0.003996  0.036443  0.029220  18599   \n",
       "962       404         0.374048  1.206058  0.004995  0.042558  0.034122  17497   \n",
       "871       461         0.366857  1.186727  0.003996  0.036443  0.029220  17983   \n",
       "1010      460         0.365890  1.183553  0.005994  0.049269  0.039503  17983   \n",
       "\n",
       "                        leading_edge  \\\n",
       "637   tags=64%, list=15%, signal=54%   \n",
       "638   tags=73%, list=19%, signal=59%   \n",
       "592   tags=75%, list=22%, signal=58%   \n",
       "594   tags=75%, list=22%, signal=58%   \n",
       "639   tags=60%, list=18%, signal=49%   \n",
       "608   tags=88%, list=24%, signal=67%   \n",
       "861   tags=73%, list=21%, signal=58%   \n",
       "615   tags=88%, list=24%, signal=66%   \n",
       "640   tags=50%, list=21%, signal=39%   \n",
       "635   tags=54%, list=22%, signal=42%   \n",
       "619   tags=67%, list=21%, signal=53%   \n",
       "616   tags=81%, list=24%, signal=62%   \n",
       "628   tags=64%, list=21%, signal=51%   \n",
       "617   tags=50%, list=19%, signal=40%   \n",
       "949   tags=91%, list=25%, signal=68%   \n",
       "948   tags=55%, list=14%, signal=47%   \n",
       "636   tags=62%, list=20%, signal=49%   \n",
       "953   tags=60%, list=20%, signal=48%   \n",
       "952   tags=70%, list=28%, signal=51%   \n",
       "951   tags=73%, list=26%, signal=54%   \n",
       "620   tags=53%, list=14%, signal=46%   \n",
       "613   tags=83%, list=24%, signal=63%   \n",
       "950   tags=82%, list=29%, signal=58%   \n",
       "944   tags=42%, list=18%, signal=34%   \n",
       "1000  tags=60%, list=21%, signal=47%   \n",
       "947   tags=83%, list=27%, signal=61%   \n",
       "945   tags=58%, list=22%, signal=45%   \n",
       "609   tags=57%, list=22%, signal=44%   \n",
       "614   tags=63%, list=22%, signal=50%   \n",
       "946   tags=50%, list=17%, signal=41%   \n",
       "...                              ...   \n",
       "280   tags=33%, list=27%, signal=25%   \n",
       "420   tags=33%, list=27%, signal=24%   \n",
       "895   tags=36%, list=31%, signal=25%   \n",
       "63    tags=34%, list=29%, signal=24%   \n",
       "30    tags=35%, list=30%, signal=25%   \n",
       "976   tags=40%, list=32%, signal=27%   \n",
       "477   tags=34%, list=29%, signal=24%   \n",
       "64    tags=33%, list=29%, signal=24%   \n",
       "702   tags=40%, list=33%, signal=27%   \n",
       "701   tags=40%, list=32%, signal=27%   \n",
       "792   tags=36%, list=29%, signal=26%   \n",
       "672   tags=34%, list=28%, signal=24%   \n",
       "874   tags=34%, list=28%, signal=25%   \n",
       "182   tags=39%, list=32%, signal=26%   \n",
       "181   tags=31%, list=27%, signal=23%   \n",
       "651   tags=39%, list=33%, signal=26%   \n",
       "699   tags=34%, list=28%, signal=24%   \n",
       "641   tags=37%, list=31%, signal=26%   \n",
       "678   tags=40%, list=33%, signal=27%   \n",
       "647   tags=36%, list=30%, signal=25%   \n",
       "786   tags=34%, list=27%, signal=25%   \n",
       "789   tags=36%, list=29%, signal=26%   \n",
       "692   tags=36%, list=30%, signal=25%   \n",
       "800   tags=36%, list=31%, signal=25%   \n",
       "707   tags=36%, list=31%, signal=25%   \n",
       "706   tags=33%, list=26%, signal=25%   \n",
       "879   tags=36%, list=32%, signal=25%   \n",
       "962   tags=35%, list=30%, signal=24%   \n",
       "871   tags=36%, list=31%, signal=25%   \n",
       "1010  tags=36%, list=31%, signal=25%   \n",
       "\n",
       "                                        core_enrichment  \\\n",
       "637   ENSG00000137076/ENSG00000173402/ENSG0000008915...   \n",
       "638   ENSG00000126067/ENSG00000159377/ENSG0000024006...   \n",
       "592   ENSG00000135972/ENSG00000132676/ENSG0000006258...   \n",
       "594   ENSG00000135972/ENSG00000132676/ENSG0000006258...   \n",
       "639   ENSG00000121931/ENSG00000132475/ENSG0000012337...   \n",
       "608   ENSG00000184110/ENSG00000100353/ENSG0000020560...   \n",
       "861   ENSG00000203965/ENSG00000157796/ENSG0000021441...   \n",
       "615   ENSG00000184110/ENSG00000100353/ENSG0000020560...   \n",
       "640   ENSG00000135870/ENSG00000164329/ENSG0000008824...   \n",
       "635   ENSG00000108349/ENSG00000130713/ENSG0000017182...   \n",
       "619   ENSG00000160294/ENSG00000076242/ENSG0000007624...   \n",
       "616   ENSG00000121022/ENSG00000184110/ENSG0000010035...   \n",
       "628   ENSG00000160294/ENSG00000076242/ENSG0000007624...   \n",
       "617   ENSG00000142945/ENSG00000115963/ENSG0000011948...   \n",
       "949   ENSG00000202337/ENSG00000075856/ENSG0000008251...   \n",
       "948   ENSG00000123353/ENSG00000100596/ENSG0000012869...   \n",
       "636   ENSG00000145736/ENSG00000111358/ENSG0000012565...   \n",
       "953   ENSG00000168538/ENSG00000126775/ENSG0000015921...   \n",
       "952   ENSG00000136045/ENSG00000109685/ENSG0000011006...   \n",
       "951   ENSG00000162735/ENSG00000150779/ENSG0000012088...   \n",
       "620   ENSG00000235376/ENSG00000177156/ENSG0000016929...   \n",
       "613   ENSG00000184110/ENSG00000100353/ENSG0000020560...   \n",
       "950   ENSG00000151849/ENSG00000168944/ENSG0000009990...   \n",
       "944   ENSG00000115963/ENSG00000119487/ENSG0000016000...   \n",
       "1000  ENSG00000119684/ENSG00000076242/ENSG0000009500...   \n",
       "947   ENSG00000075856/ENSG00000136560/ENSG0000010537...   \n",
       "945   ENSG00000140694/ENSG00000188573/ENSG0000013071...   \n",
       "609   ENSG00000164162/ENSG00000110200/ENSG0000005390...   \n",
       "614   ENSG00000164692/ENSG00000197565/ENSG0000013487...   \n",
       "946   ENSG00000198730/ENSG00000006712/ENSG0000015582...   \n",
       "...                                                 ...   \n",
       "280   ENSG00000196230/ENSG00000100030/ENSG0000011181...   \n",
       "420   ENSG00000196230/ENSG00000100030/ENSG0000011181...   \n",
       "895   ENSG00000004478/ENSG00000119138/ENSG0000012787...   \n",
       "63    ENSG00000124333/ENSG00000185896/ENSG0000016120...   \n",
       "30    ENSG00000013275/ENSG00000100030/ENSG0000015779...   \n",
       "976   ENSG00000124333/ENSG00000161203/ENSG0000006840...   \n",
       "477   ENSG00000124333/ENSG00000185896/ENSG0000016120...   \n",
       "64    ENSG00000124333/ENSG00000185896/ENSG0000016120...   \n",
       "702   ENSG00000199017/ENSG00000283159/ENSG0000001327...   \n",
       "701   ENSG00000030066/ENSG00000135002/ENSG0000007518...   \n",
       "792   ENSG00000174574/ENSG00000169905/ENSG0000005465...   \n",
       "672   ENSG00000144278/ENSG00000100030/ENSG0000013604...   \n",
       "874   ENSG00000100030/ENSG00000136045/ENSG0000015418...   \n",
       "182   ENSG00000102606/ENSG00000130340/ENSG0000010046...   \n",
       "181   ENSG00000124333/ENSG00000090487/ENSG0000018589...   \n",
       "651   ENSG00000196230/ENSG00000264364/ENSG0000017520...   \n",
       "699   ENSG00000154813/ENSG00000162735/ENSG0000012348...   \n",
       "641   ENSG00000013275/ENSG00000129993/ENSG0000010076...   \n",
       "678   ENSG00000199017/ENSG00000283159/ENSG0000001327...   \n",
       "647   ENSG00000197579/ENSG00000198833/ENSG0000002284...   \n",
       "786   ENSG00000124333/ENSG00000161203/ENSG0000013673...   \n",
       "789   ENSG00000185896/ENSG00000030066/ENSG0000024450...   \n",
       "692   ENSG00000177684/ENSG00000213246/ENSG0000010885...   \n",
       "800   ENSG00000013275/ENSG00000129993/ENSG0000010076...   \n",
       "707   ENSG00000013275/ENSG00000129993/ENSG0000010076...   \n",
       "706   ENSG00000207691/ENSG00000221263/ENSG0000015934...   \n",
       "879   ENSG00000124333/ENSG00000113645/ENSG0000007284...   \n",
       "962   ENSG00000124333/ENSG00000113645/ENSG0000010260...   \n",
       "871   ENSG00000207691/ENSG00000221263/ENSG0000015934...   \n",
       "1010  ENSG00000207691/ENSG00000221263/ENSG0000015934...   \n",
       "\n",
       "                                         ID_description  ranking  \n",
       "637                         GO:0017166-vinculin binding      1.0  \n",
       "638   GO:0019774-proteasome core complex, beta-subun...      2.0  \n",
       "592       GO:0000314-organellar small ribosomal subunit      3.5  \n",
       "594    GO:0005763-mitochondrial small ribosomal subunit      3.5  \n",
       "639                             GO:0000805-X chromosome      5.0  \n",
       "608     GO:0016282-eukaryotic 43S preinitiation complex      6.0  \n",
       "861   GO:1903441-protein localization to ciliary mem...      7.0  \n",
       "615   GO:0001732-formation of cytoplasmic translatio...      8.0  \n",
       "640    GO:2000628-regulation of miRNA metabolic process      9.0  \n",
       "635          GO:0008298-intracellular mRNA localization     10.0  \n",
       "619   GO:0002566-somatic diversification of immune r...     11.0  \n",
       "616   GO:0005852-eukaryotic translation initiation f...     12.0  \n",
       "628   GO:0016446-somatic hypermutation of immunoglob...     13.0  \n",
       "617   GO:0030952-establishment or maintenance of cyt...     14.0  \n",
       "949                         GO:0030621-U4 snRNA binding     15.0  \n",
       "948             GO:0002178-palmitoyltransferase complex     16.0  \n",
       "636   GO:0016251-RNA polymerase II general transcrip...     17.0  \n",
       "953   GO:0061635-regulation of protein complex stabi...     18.0  \n",
       "952               GO:0034770-histone H4-K20 methylation     19.0  \n",
       "951     GO:0072321-chaperone-mediated protein transport     20.0  \n",
       "620                  GO:0006098-pentose-phosphate shunt     21.0  \n",
       "613        GO:0070993-translation preinitiation complex     22.0  \n",
       "950   GO:0046607-positive regulation of centrosome c...     23.0  \n",
       "944   GO:0030950-establishment or maintenance of act...     24.0  \n",
       "1000                 GO:0032300-mismatch repair complex     25.0  \n",
       "947   GO:0090085-regulation of protein deubiquitination     26.0  \n",
       "945                 GO:0031126-snoRNA 3'-end processing     27.0  \n",
       "609               GO:0005680-anaphase-promoting complex     28.0  \n",
       "614              GO:0098644-complex of collagen trimers     29.0  \n",
       "946               GO:0033523-histone H2B ubiquitination     30.0  \n",
       "...                                                 ...      ...  \n",
       "280                            GO:0031983-vesicle lumen    999.0  \n",
       "420                GO:0060205-cytoplasmic vesicle lumen   1000.0  \n",
       "895   GO:0071383-cellular response to steroid hormon...   1001.0  \n",
       "63                        GO:0005765-lysosomal membrane   1002.0  \n",
       "30            GO:0001701-in utero embryonic development   1003.0  \n",
       "976   GO:0072659-protein localization to plasma memb...   1004.0  \n",
       "477                   GO:0098852-lytic vacuole membrane   1005.0  \n",
       "64                         GO:0005774-vacuolar membrane   1006.0  \n",
       "702          GO:0060070-canonical Wnt signaling pathway   1007.0  \n",
       "701            GO:0051188-cofactor biosynthetic process   1008.0  \n",
       "792                         GO:0031965-nuclear membrane   1009.0  \n",
       "672             GO:0018209-peptidyl-serine modification   1010.0  \n",
       "874          GO:0018105-peptidyl-serine phosphorylation   1011.0  \n",
       "182         GO:0010256-endomembrane system organization   1012.0  \n",
       "181                        GO:0010008-endosome membrane   1013.0  \n",
       "651                              GO:0005874-microtubule   1014.0  \n",
       "699                    GO:0051098-regulation of binding   1015.0  \n",
       "641                      GO:0001666-response to hypoxia   1016.0  \n",
       "678      GO:0030111-regulation of Wnt signaling pathway   1017.0  \n",
       "647   GO:0004842-ubiquitin-protein transferase activity   1018.0  \n",
       "786                     GO:0016050-vesicle organization   1019.0  \n",
       "789                         GO:0019058-viral life cycle   1020.0  \n",
       "692     GO:0043900-regulation of multi-organism process   1021.0  \n",
       "800      GO:0036293-response to decreased oxygen levels   1022.0  \n",
       "707                GO:0070482-response to oxygen levels   1023.0  \n",
       "706   GO:0062012-regulation of small molecule metabo...   1024.0  \n",
       "879    GO:0032386-regulation of intracellular transport   1025.0  \n",
       "962                        GO:0031252-cell leading edge   1026.0  \n",
       "871        GO:0016053-organic acid biosynthetic process   1027.0  \n",
       "1010    GO:0046394-carboxylic acid biosynthetic process   1028.0  \n",
       "\n",
       "[1028 rows x 14 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get ranks of template experiment\n",
    "# Rank pathways by highest enrichment score\n",
    "enriched_pathways['ranking'] = enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "enriched_pathways = enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "enriched_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GO:0017166-vinculin binding': 1.0,\n",
       " 'GO:0019774-proteasome core complex, beta-subunit complex': 2.0,\n",
       " 'GO:0000314-organellar small ribosomal subunit': 3.5,\n",
       " 'GO:0005763-mitochondrial small ribosomal subunit': 3.5,\n",
       " 'GO:0000805-X chromosome': 5.0,\n",
       " 'GO:0016282-eukaryotic 43S preinitiation complex': 6.0,\n",
       " 'GO:1903441-protein localization to ciliary membrane': 7.0,\n",
       " 'GO:0001732-formation of cytoplasmic translation initiation complex': 8.0,\n",
       " 'GO:2000628-regulation of miRNA metabolic process': 9.0,\n",
       " 'GO:0008298-intracellular mRNA localization': 10.0,\n",
       " 'GO:0002566-somatic diversification of immune receptors via somatic mutation': 11.0,\n",
       " 'GO:0005852-eukaryotic translation initiation factor 3 complex': 12.0,\n",
       " 'GO:0016446-somatic hypermutation of immunoglobulin genes': 13.0,\n",
       " 'GO:0030952-establishment or maintenance of cytoskeleton polarity': 14.0,\n",
       " 'GO:0030621-U4 snRNA binding': 15.0,\n",
       " 'GO:0002178-palmitoyltransferase complex': 16.0,\n",
       " 'GO:0016251-RNA polymerase II general transcription initiation factor activity': 17.0,\n",
       " 'GO:0061635-regulation of protein complex stability': 18.0,\n",
       " 'GO:0034770-histone H4-K20 methylation': 19.0,\n",
       " 'GO:0072321-chaperone-mediated protein transport': 20.0,\n",
       " 'GO:0006098-pentose-phosphate shunt': 21.0,\n",
       " 'GO:0070993-translation preinitiation complex': 22.0,\n",
       " 'GO:0046607-positive regulation of centrosome cycle': 23.0,\n",
       " 'GO:0030950-establishment or maintenance of actin cytoskeleton polarity': 24.0,\n",
       " 'GO:0032300-mismatch repair complex': 25.0,\n",
       " 'GO:0090085-regulation of protein deubiquitination': 26.0,\n",
       " \"GO:0031126-snoRNA 3'-end processing\": 27.0,\n",
       " 'GO:0005680-anaphase-promoting complex': 28.0,\n",
       " 'GO:0098644-complex of collagen trimers': 29.0,\n",
       " 'GO:0033523-histone H2B ubiquitination': 30.0,\n",
       " 'GO:0033290-eukaryotic 48S preinitiation complex': 31.0,\n",
       " 'GO:0090151-establishment of protein localization to mitochondrial membrane': 32.0,\n",
       " 'GO:0051095-regulation of helicase activity': 33.0,\n",
       " 'GO:0003746-translation elongation factor activity': 34.0,\n",
       " 'GO:0014745-negative regulation of muscle adaptation': 35.0,\n",
       " 'GO:1904152-regulation of retrograde protein transport, ER to cytosol': 36.0,\n",
       " 'GO:0021591-ventricular system development': 37.0,\n",
       " 'GO:0090231-regulation of spindle checkpoint': 39.0,\n",
       " 'GO:1903504-regulation of mitotic spindle checkpoint': 39.0,\n",
       " 'GO:0090266-regulation of mitotic cell cycle spindle assembly checkpoint': 39.0,\n",
       " 'GO:0017101-aminoacyl-tRNA synthetase multienzyme complex': 41.0,\n",
       " 'GO:0000940-condensed chromosome outer kinetochore': 42.0,\n",
       " 'GO:0030983-mismatched DNA binding': 43.0,\n",
       " 'GO:0019321-pentose metabolic process': 44.0,\n",
       " 'GO:0060831-smoothened signaling pathway involved in dorsal/ventral neural tube patterning': 45.0,\n",
       " 'GO:0070601-centromeric sister chromatid cohesion': 46.0,\n",
       " 'GO:0036260-RNA capping': 47.5,\n",
       " 'GO:0009452-7-methylguanosine RNA capping': 47.5,\n",
       " 'GO:0051988-regulation of attachment of spindle microtubules to kinetochore': 49.0,\n",
       " 'GO:1990173-protein localization to nucleoplasm': 50.0,\n",
       " 'GO:0006370-7-methylguanosine mRNA capping': 51.0,\n",
       " 'GO:0071459-protein localization to chromosome, centromeric region': 52.0,\n",
       " 'GO:0070986-left/right axis specification': 53.0,\n",
       " 'GO:1904355-positive regulation of telomere capping': 54.0,\n",
       " 'GO:0016074-snoRNA metabolic process': 55.0,\n",
       " 'GO:0000177-cytoplasmic exosome (RNase complex)': 56.0,\n",
       " 'GO:0033147-negative regulation of intracellular estrogen receptor signaling pathway': 57.0,\n",
       " 'GO:0000212-meiotic spindle organization': 58.0,\n",
       " 'GO:0017025-TBP-class protein binding': 59.0,\n",
       " 'GO:1903543-positive regulation of exosomal secretion': 60.0,\n",
       " 'GO:0034501-protein localization to kinetochore': 61.0,\n",
       " 'GO:0042559-pteridine-containing compound biosynthetic process': 62.0,\n",
       " 'GO:1904380-endoplasmic reticulum mannose trimming': 63.0,\n",
       " 'GO:1902236-negative regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway': 64.0,\n",
       " 'GO:0000778-condensed nuclear chromosome kinetochore': 65.0,\n",
       " 'GO:0006268-DNA unwinding involved in DNA replication': 66.0,\n",
       " 'GO:1900037-regulation of cellular response to hypoxia': 67.0,\n",
       " 'GO:0043144-snoRNA processing': 68.0,\n",
       " 'GO:0031468-nuclear envelope reassembly': 69.0,\n",
       " 'GO:0055070-copper ion homeostasis': 70.0,\n",
       " 'GO:0015037-peptide disulfide oxidoreductase activity': 71.0,\n",
       " 'GO:0007094-mitotic spindle assembly checkpoint': 73.5,\n",
       " 'GO:0071174-mitotic spindle checkpoint': 73.5,\n",
       " 'GO:0071173-spindle assembly checkpoint': 73.5,\n",
       " 'GO:0031577-spindle checkpoint': 73.5,\n",
       " 'GO:0005839-proteasome core complex': 76.0,\n",
       " 'GO:1904874-positive regulation of telomerase RNA localization to Cajal body': 77.0,\n",
       " 'GO:0032785-negative regulation of DNA-templated transcription, elongation': 78.0,\n",
       " 'GO:0098787-mRNA cleavage involved in mRNA processing': 79.0,\n",
       " 'GO:0030688-preribosome, small subunit precursor': 80.0,\n",
       " 'GO:0007100-mitotic centrosome separation': 81.0,\n",
       " 'GO:0042809-vitamin D receptor binding': 82.0,\n",
       " 'GO:0031145-anaphase-promoting complex-dependent catabolic process': 83.0,\n",
       " 'GO:1990182-exosomal secretion': 84.0,\n",
       " 'GO:0097734-extracellular exosome biogenesis': 85.0,\n",
       " 'GO:0140223-general transcription initiation factor activity': 86.0,\n",
       " 'GO:0045841-negative regulation of mitotic metaphase/anaphase transition': 87.0,\n",
       " 'GO:0032040-small-subunit processome': 88.0,\n",
       " 'GO:0033962-cytoplasmic mRNA processing body assembly': 89.0,\n",
       " 'GO:0000780-condensed nuclear chromosome, centromeric region': 90.0,\n",
       " 'GO:0004298-threonine-type endopeptidase activity': 91.5,\n",
       " 'GO:0070003-threonine-type peptidase activity': 91.5,\n",
       " 'GO:0051156-glucose 6-phosphate metabolic process': 93.0,\n",
       " 'GO:0000028-ribosomal small subunit assembly': 94.0,\n",
       " 'GO:0007020-microtubule nucleation': 95.0,\n",
       " 'GO:0043162-ubiquitin-dependent protein catabolic process via the multivesicular body sorting pathway': 96.0,\n",
       " 'GO:0019682-glyceraldehyde-3-phosphate metabolic process': 97.0,\n",
       " 'GO:0000715-nucleotide-excision repair, DNA damage recognition': 98.0,\n",
       " 'GO:1904872-regulation of telomerase RNA localization to Cajal body': 99.0,\n",
       " 'GO:0015935-small ribosomal subunit': 100.0,\n",
       " 'GO:0090671-telomerase RNA localization to Cajal body': 102.5,\n",
       " 'GO:0090672-telomerase RNA localization': 102.5,\n",
       " 'GO:0090670-RNA localization to Cajal body': 102.5,\n",
       " 'GO:0090685-RNA localization to nucleus': 102.5,\n",
       " 'GO:1902100-negative regulation of metaphase/anaphase transition of cell cycle': 105.0,\n",
       " 'GO:0022624-proteasome accessory complex': 106.0,\n",
       " 'GO:1901976-regulation of cell cycle checkpoint': 107.0,\n",
       " 'GO:0016866-intramolecular transferase activity': 108.0,\n",
       " 'GO:0001091-RNA polymerase II basal transcription factor binding': 109.0,\n",
       " 'GO:0000339-RNA cap binding': 110.0,\n",
       " 'GO:0000717-nucleotide-excision repair, DNA duplex unwinding': 111.0,\n",
       " 'GO:0043022-ribosome binding': 112.0,\n",
       " 'GO:0006622-protein targeting to lysosome': 113.0,\n",
       " 'GO:2000816-negative regulation of mitotic sister chromatid separation': 114.0,\n",
       " 'GO:1905354-exoribonuclease complex': 115.0,\n",
       " 'GO:2000104-negative regulation of DNA-dependent DNA replication': 116.0,\n",
       " 'GO:0000502-proteasome complex': 117.0,\n",
       " 'GO:1905369-endopeptidase complex': 118.0,\n",
       " 'GO:0008180-COP9 signalosome': 119.0,\n",
       " 'GO:1900118-negative regulation of execution phase of apoptosis': 120.0,\n",
       " 'GO:0070828-heterochromatin organization': 121.0,\n",
       " 'GO:1905819-negative regulation of chromosome separation': 122.0,\n",
       " 'GO:0002183-cytoplasmic translational initiation': 123.0,\n",
       " 'GO:0050434-positive regulation of viral transcription': 124.0,\n",
       " 'GO:0005697-telomerase holoenzyme complex': 125.0,\n",
       " 'GO:0004003-ATP-dependent DNA helicase activity': 127.0,\n",
       " 'GO:0008026-ATP-dependent helicase activity': 127.0,\n",
       " 'GO:0070035-purine NTP-dependent helicase activity': 127.0,\n",
       " 'GO:0006450-regulation of translational fidelity': 129.0,\n",
       " 'GO:0032806-carboxy-terminal domain protein kinase complex': 130.0,\n",
       " 'GO:0033048-negative regulation of mitotic sister chromatid segregation': 131.0,\n",
       " 'GO:0140112-extracellular vesicle biogenesis': 132.0,\n",
       " 'GO:0007549-dosage compensation': 133.0,\n",
       " 'GO:0006363-termination of RNA polymerase I transcription': 134.0,\n",
       " 'GO:0032786-positive regulation of DNA-templated transcription, elongation': 135.0,\n",
       " 'GO:0033046-negative regulation of sister chromatid segregation': 136.0,\n",
       " 'GO:0000178-exosome (RNase complex)': 137.0,\n",
       " \"GO:0070935-3'-UTR-mediated mRNA stabilization\": 138.0,\n",
       " 'GO:0008137-NADH dehydrogenase (ubiquinone) activity': 140.0,\n",
       " 'GO:0050136-NADH dehydrogenase (quinone) activity': 140.0,\n",
       " 'GO:0003954-NADH dehydrogenase activity': 140.0,\n",
       " 'GO:0006294-nucleotide-excision repair, preincision complex assembly': 142.0,\n",
       " 'GO:0008171-O-methyltransferase activity': 143.0,\n",
       " 'GO:0000462-maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)': 144.0,\n",
       " 'GO:0016556-mRNA modification': 145.0,\n",
       " 'GO:1903421-regulation of synaptic vesicle recycling': 146.0,\n",
       " 'GO:0008156-negative regulation of DNA replication': 147.0,\n",
       " 'GO:0005868-cytoplasmic dynein complex': 148.0,\n",
       " 'GO:0006760-folic acid-containing compound metabolic process': 149.0,\n",
       " 'GO:0010499-proteasomal ubiquitin-independent protein catabolic process': 150.0,\n",
       " 'GO:1902235-regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway': 151.0,\n",
       " 'GO:0051985-negative regulation of chromosome segregation': 152.0,\n",
       " 'GO:0071005-U2-type precatalytic spliceosome': 153.0,\n",
       " 'GO:0061621-canonical glycolysis': 155.0,\n",
       " 'GO:0006735-NADH regeneration': 155.0,\n",
       " 'GO:0061718-glucose catabolic process to pyruvate': 155.0,\n",
       " 'GO:0070911-global genome nucleotide-excision repair': 157.0,\n",
       " 'GO:0070125-mitochondrial translational elongation': 158.0,\n",
       " 'GO:0034661-ncRNA catabolic process': 159.0,\n",
       " 'GO:0008608-attachment of spindle microtubules to kinetochore': 160.0,\n",
       " 'GO:0036508-protein alpha-1,2-demannosylation': 161.5,\n",
       " 'GO:0036507-protein demannosylation': 161.5,\n",
       " 'GO:0061620-glycolytic process through glucose-6-phosphate': 163.5,\n",
       " 'GO:0061615-glycolytic process through fructose-6-phosphate': 163.5,\n",
       " 'GO:0045271-respiratory chain complex I': 166.0,\n",
       " 'GO:0005747-mitochondrial respiratory chain complex I': 166.0,\n",
       " 'GO:0030964-NADH dehydrogenase complex': 166.0,\n",
       " 'GO:0006361-transcription initiation from RNA polymerase I promoter': 168.0,\n",
       " 'GO:0051443-positive regulation of ubiquitin-protein transferase activity': 169.0,\n",
       " 'GO:0030071-regulation of mitotic metaphase/anaphase transition': 170.0,\n",
       " 'GO:1904666-regulation of ubiquitin protein ligase activity': 171.0,\n",
       " 'GO:0006623-protein targeting to vacuole': 172.0,\n",
       " 'GO:0070979-protein K11-linked ubiquitination': 173.0,\n",
       " 'GO:0043631-RNA polyadenylation': 174.0,\n",
       " 'GO:0009299-mRNA transcription': 175.0,\n",
       " 'GO:0071011-precatalytic spliceosome': 176.0,\n",
       " 'GO:0006414-translational elongation': 177.0,\n",
       " 'GO:0044769-ATPase activity, coupled to transmembrane movement of ions, rotational mechanism': 178.0,\n",
       " 'GO:2000765-regulation of cytoplasmic translation': 179.0,\n",
       " 'GO:0006356-regulation of transcription by RNA polymerase I': 180.0,\n",
       " 'GO:0006362-transcription elongation from RNA polymerase I promoter': 181.0,\n",
       " 'GO:0045839-negative regulation of mitotic nuclear division': 182.0,\n",
       " \"GO:0016895-exodeoxyribonuclease activity, producing 5'-phosphomonoesters\": 183.0,\n",
       " 'GO:0032981-mitochondrial respiratory chain complex I assembly': 184.5,\n",
       " 'GO:0010257-NADH dehydrogenase complex assembly': 184.5,\n",
       " 'GO:0003743-translation initiation factor activity': 186.0,\n",
       " 'GO:0021904-dorsal/ventral neural tube patterning': 187.0,\n",
       " 'GO:0046961-proton-transporting ATPase activity, rotational mechanism': 188.0,\n",
       " 'GO:0008135-translation factor activity, RNA binding': 189.0,\n",
       " 'GO:0042558-pteridine-containing compound metabolic process': 190.0,\n",
       " 'GO:0006378-mRNA polyadenylation': 191.0,\n",
       " 'GO:1903573-negative regulation of response to endoplasmic reticulum stress': 192.0,\n",
       " 'GO:0051123-RNA polymerase II preinitiation complex assembly': 193.0,\n",
       " 'GO:0015949-nucleobase-containing small molecule interconversion': 194.0,\n",
       " 'GO:0071007-U2-type catalytic step 2 spliceosome': 195.0,\n",
       " 'GO:0006379-mRNA cleavage': 196.0,\n",
       " 'GO:0000217-DNA secondary structure binding': 197.0,\n",
       " 'GO:1905368-peptidase complex': 198.0,\n",
       " 'GO:0000281-mitotic cytokinesis': 199.0,\n",
       " 'GO:0000313-organellar ribosome': 200.5,\n",
       " 'GO:0005761-mitochondrial ribosome': 200.5,\n",
       " 'GO:0008535-respiratory chain complex IV assembly': 202.0,\n",
       " 'GO:0140103-catalytic activity, acting on a glycoprotein': 203.0,\n",
       " 'GO:1902750-negative regulation of cell cycle G2/M phase transition': 204.0,\n",
       " 'GO:1900101-regulation of endoplasmic reticulum unfolded protein response': 205.0,\n",
       " 'GO:1902099-regulation of metaphase/anaphase transition of cell cycle': 206.0,\n",
       " 'GO:0046782-regulation of viral transcription': 207.0,\n",
       " 'GO:0035666-TRIF-dependent toll-like receptor signaling pathway': 208.0,\n",
       " 'GO:0006101-citrate metabolic process': 209.0,\n",
       " 'GO:0017056-structural constituent of nuclear pore': 210.0,\n",
       " 'GO:0006521-regulation of cellular amino acid metabolic process': 211.0,\n",
       " 'GO:1905897-regulation of response to endoplasmic reticulum stress': 212.0,\n",
       " 'GO:0016860-intramolecular oxidoreductase activity': 213.0,\n",
       " 'GO:0015030-Cajal body': 214.0,\n",
       " 'GO:0042788-polysomal ribosome': 215.0,\n",
       " 'GO:0009067-aspartate family amino acid biosynthetic process': 216.0,\n",
       " 'GO:0007091-metaphase/anaphase transition of mitotic cell cycle': 217.0,\n",
       " 'GO:0043021-ribonucleoprotein complex binding': 218.0,\n",
       " 'GO:0006360-transcription by RNA polymerase I': 219.0,\n",
       " 'GO:0033144-negative regulation of intracellular steroid hormone receptor signaling pathway': 220.0,\n",
       " 'GO:0003688-DNA replication origin binding': 221.0,\n",
       " 'GO:0010972-negative regulation of G2/M transition of mitotic cell cycle': 222.0,\n",
       " \"GO:0031440-regulation of mRNA 3'-end processing\": 223.0,\n",
       " 'GO:0044665-MLL1/2 complex': 224.5,\n",
       " 'GO:0071339-MLL1 complex': 224.5,\n",
       " 'GO:0036498-IRE1-mediated unfolded protein response': 226.0,\n",
       " 'GO:0060260-regulation of transcription initiation from RNA polymerase II promoter': 227.0,\n",
       " 'GO:0051438-regulation of ubiquitin-protein transferase activity': 228.0,\n",
       " 'GO:0005732-small nucleolar ribonucleoprotein complex': 229.0,\n",
       " 'GO:0061157-mRNA destabilization': 230.0,\n",
       " 'GO:0016655-oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor': 231.0,\n",
       " 'GO:1904292-regulation of ERAD pathway': 232.0,\n",
       " \"GO:0031124-mRNA 3'-end processing\": 233.0,\n",
       " 'GO:0010965-regulation of mitotic sister chromatid separation': 234.0,\n",
       " 'GO:0070126-mitochondrial translational termination': 235.0,\n",
       " 'GO:0030684-preribosome': 236.0,\n",
       " 'GO:0046540-U4/U6 x U5 tri-snRNP complex': 237.5,\n",
       " 'GO:0097526-spliceosomal tri-snRNP complex': 237.5,\n",
       " 'GO:0051306-mitotic sister chromatid separation': 239.0,\n",
       " 'GO:0006099-tricarboxylic acid cycle': 240.0,\n",
       " 'GO:0030866-cortical actin cytoskeleton organization': 241.0,\n",
       " 'GO:0051784-negative regulation of nuclear division': 242.0,\n",
       " 'GO:0016242-negative regulation of macroautophagy': 243.0,\n",
       " 'GO:0005746-mitochondrial respiratory chain': 244.0,\n",
       " \"GO:0048027-mRNA 5'-UTR binding\": 245.0,\n",
       " 'GO:0060632-regulation of microtubule-based movement': 246.0,\n",
       " 'GO:0035327-transcriptionally active chromatin': 247.0,\n",
       " 'GO:0043921-modulation by host of viral transcription': 248.5,\n",
       " 'GO:0052472-modulation by host of symbiont transcription': 248.5,\n",
       " 'GO:0031369-translation initiation factor binding': 250.0,\n",
       " 'GO:0061418-regulation of transcription from RNA polymerase II promoter in response to hypoxia': 251.0,\n",
       " 'GO:0051973-positive regulation of telomerase activity': 252.0,\n",
       " 'GO:0071539-protein localization to centrosome': 253.0,\n",
       " 'GO:0070199-establishment of protein localization to chromosome': 254.0,\n",
       " \"GO:0000175-3'-5'-exoribonuclease activity\": 255.0,\n",
       " 'GO:0042255-ribosome assembly': 256.0,\n",
       " 'GO:0072350-tricarboxylic acid metabolic process': 257.0,\n",
       " 'GO:0044784-metaphase/anaphase transition of cell cycle': 258.0,\n",
       " 'GO:0051019-mitogen-activated protein kinase binding': 259.0,\n",
       " 'GO:0060765-regulation of androgen receptor signaling pathway': 260.0,\n",
       " 'GO:0008536-Ran GTPase binding': 261.0,\n",
       " 'GO:0030990-intraciliary transport particle': 262.0,\n",
       " 'GO:0070198-protein localization to chromosome, telomeric region': 263.0,\n",
       " 'GO:0034243-regulation of transcription elongation from RNA polymerase II promoter': 264.0,\n",
       " 'GO:1902036-regulation of hematopoietic stem cell differentiation': 265.0,\n",
       " 'GO:0002479-antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent': 266.0,\n",
       " 'GO:0033108-mitochondrial respiratory chain complex assembly': 267.0,\n",
       " 'GO:0007143-female meiotic nuclear division': 268.0,\n",
       " 'GO:0070469-respiratory chain': 269.0,\n",
       " 'GO:0006415-translational termination': 270.0,\n",
       " 'GO:0031572-G2 DNA damage checkpoint': 271.0,\n",
       " 'GO:0010837-regulation of keratinocyte proliferation': 272.0,\n",
       " 'GO:1901068-guanosine-containing compound metabolic process': 273.0,\n",
       " 'GO:0030865-cortical cytoskeleton organization': 274.0,\n",
       " 'GO:0007007-inner mitochondrial membrane organization': 275.0,\n",
       " 'GO:0032212-positive regulation of telomere maintenance via telomerase': 276.0,\n",
       " 'GO:0000154-rRNA modification': 277.0,\n",
       " 'GO:2000785-regulation of autophagosome assembly': 278.0,\n",
       " 'GO:0042274-ribosomal small subunit biogenesis': 279.0,\n",
       " 'GO:0046039-GTP metabolic process': 280.0,\n",
       " 'GO:0033047-regulation of mitotic sister chromatid segregation': 281.0,\n",
       " 'GO:1905818-regulation of chromosome separation': 282.0,\n",
       " 'GO:0042795-snRNA transcription by RNA polymerase II': 283.0,\n",
       " 'GO:0019205-nucleobase-containing compound kinase activity': 284.0,\n",
       " 'GO:0050779-RNA destabilization': 285.0,\n",
       " 'GO:0019083-viral transcription': 286.0,\n",
       " 'GO:0034502-protein localization to chromosome': 287.0,\n",
       " 'GO:0097421-liver regeneration': 288.0,\n",
       " 'GO:0045005-DNA-dependent DNA replication maintenance of fidelity': 289.0,\n",
       " 'GO:0070059-intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress': 290.0,\n",
       " 'GO:0061631-ubiquitin conjugating enzyme activity': 291.0,\n",
       " 'GO:0098803-respiratory chain complex': 292.0,\n",
       " 'GO:0022625-cytosolic large ribosomal subunit': 293.0,\n",
       " 'GO:1905898-positive regulation of response to endoplasmic reticulum stress': 294.0,\n",
       " 'GO:0000132-establishment of mitotic spindle orientation': 295.0,\n",
       " 'GO:0019843-rRNA binding': 296.0,\n",
       " 'GO:0016442-RISC complex': 297.5,\n",
       " 'GO:0031332-RNAi effector complex': 297.5,\n",
       " 'GO:0006890-retrograde vesicle-mediated transport, Golgi to ER': 299.0,\n",
       " 'GO:0032784-regulation of DNA-templated transcription, elongation': 300.0,\n",
       " \"GO:0006296-nucleotide-excision repair, DNA incision, 5'-to lesion\": 301.0,\n",
       " 'GO:0034067-protein localization to Golgi apparatus': 302.0,\n",
       " 'GO:0030490-maturation of SSU-rRNA': 303.0,\n",
       " 'GO:0005876-spindle microtubule': 304.0,\n",
       " 'GO:0009301-snRNA transcription': 305.0,\n",
       " 'GO:0061014-positive regulation of mRNA catabolic process': 306.0,\n",
       " 'GO:0044088-regulation of vacuole organization': 307.0,\n",
       " 'GO:0031060-regulation of histone methylation': 308.0,\n",
       " 'GO:0006353-DNA-templated transcription, termination': 309.0,\n",
       " 'GO:1901532-regulation of hematopoietic progenitor cell differentiation': 310.0,\n",
       " 'GO:0046966-thyroid hormone receptor binding': 311.0,\n",
       " 'GO:0006446-regulation of translational initiation': 312.0,\n",
       " 'GO:0072666-establishment of protein localization to vacuole': 313.0,\n",
       " 'GO:0080008-Cul4-RING E3 ubiquitin ligase complex': 314.0,\n",
       " 'GO:0002181-cytoplasmic translation': 315.0,\n",
       " 'GO:0044818-mitotic G2/M transition checkpoint': 316.0,\n",
       " 'GO:0004812-aminoacyl-tRNA ligase activity': 317.5,\n",
       " 'GO:0016875-ligase activity, forming carbon-oxygen bonds': 317.5,\n",
       " 'GO:0005684-U2-type spliceosomal complex': 319.0,\n",
       " 'GO:0019080-viral gene expression': 320.0,\n",
       " 'GO:0006413-translational initiation': 321.0,\n",
       " 'GO:0007052-mitotic spindle organization': 322.0,\n",
       " 'GO:1900117-regulation of execution phase of apoptosis': 323.0,\n",
       " 'GO:0031062-positive regulation of histone methylation': 324.0,\n",
       " 'GO:0017069-snRNA binding': 325.0,\n",
       " 'GO:0005790-smooth endoplasmic reticulum': 326.0,\n",
       " 'GO:0033045-regulation of sister chromatid segregation': 327.0,\n",
       " \"GO:0016896-exoribonuclease activity, producing 5'-phosphomonoesters\": 328.0,\n",
       " 'GO:0061640-cytoskeleton-dependent cytokinesis': 329.0,\n",
       " 'GO:2000142-regulation of DNA-templated transcription, initiation': 330.0,\n",
       " 'GO:0032508-DNA duplex unwinding': 331.0,\n",
       " 'GO:0032506-cytokinetic process': 332.0,\n",
       " 'GO:0090307-mitotic spindle assembly': 333.0,\n",
       " 'GO:1990841-promoter-specific chromatin binding': 334.0,\n",
       " 'GO:0140053-mitochondrial gene expression': 335.0,\n",
       " 'GO:0032206-positive regulation of telomere maintenance': 336.0,\n",
       " 'GO:0042590-antigen processing and presentation of exogenous peptide antigen via MHC class I': 337.0,\n",
       " 'GO:0071431-tRNA-containing ribonucleoprotein complex export from nucleus': 338.5,\n",
       " 'GO:0006409-tRNA export from nucleus': 338.5,\n",
       " 'GO:0070498-interleukin-1-mediated signaling pathway': 340.0,\n",
       " 'GO:0000387-spliceosomal snRNP assembly': 341.0,\n",
       " 'GO:0006418-tRNA aminoacylation for protein translation': 342.0,\n",
       " 'GO:0004532-exoribonuclease activity': 343.0,\n",
       " 'GO:0006270-DNA replication initiation': 344.0,\n",
       " 'GO:1904358-positive regulation of telomere maintenance via telomere lengthening': 345.0,\n",
       " 'GO:0033146-regulation of intracellular estrogen receptor signaling pathway': 346.0,\n",
       " 'GO:0006734-NADH metabolic process': 347.0,\n",
       " 'GO:0022627-cytosolic small ribosomal subunit': 348.0,\n",
       " 'GO:0043130-ubiquitin binding': 349.0,\n",
       " 'GO:0061650-ubiquitin-like protein conjugating enzyme activity': 350.0,\n",
       " 'GO:2001243-negative regulation of intrinsic apoptotic signaling pathway': 351.0,\n",
       " 'GO:1904813-ficolin-1-rich granule lumen': 352.0,\n",
       " 'GO:0000152-nuclear ubiquitin ligase complex': 353.0,\n",
       " 'GO:0003678-DNA helicase activity': 354.0,\n",
       " 'GO:1902850-microtubule cytoskeleton organization involved in mitosis': 355.0,\n",
       " \"GO:0008408-3'-5' exonuclease activity\": 356.0,\n",
       " 'GO:0070897-transcription preinitiation complex assembly': 357.0,\n",
       " 'GO:0015036-disulfide oxidoreductase activity': 358.0,\n",
       " 'GO:0009060-aerobic respiration': 359.0,\n",
       " 'GO:0018023-peptidyl-lysine trimethylation': 360.0,\n",
       " 'GO:0044391-ribosomal subunit': 361.0,\n",
       " 'GO:0002474-antigen processing and presentation of peptide antigen via MHC class I': 362.0,\n",
       " 'GO:0006298-mismatch repair': 363.0,\n",
       " 'GO:0098781-ncRNA transcription': 364.0,\n",
       " 'GO:0051972-regulation of telomerase activity': 365.0,\n",
       " 'GO:1903231-mRNA binding involved in posttranscriptional gene silencing': 366.0,\n",
       " 'GO:0016591-RNA polymerase II, holoenzyme': 367.0,\n",
       " 'GO:0006354-DNA-templated transcription, elongation': 368.0,\n",
       " 'GO:1990391-DNA repair complex': 369.0,\n",
       " 'GO:0090503-RNA phosphodiester bond hydrolysis, exonucleolytic': 370.0,\n",
       " 'GO:0007080-mitotic metaphase plate congression': 371.0,\n",
       " 'GO:0060218-hematopoietic stem cell differentiation': 372.0,\n",
       " 'GO:0032543-mitochondrial translation': 373.0,\n",
       " 'GO:0006120-mitochondrial electron transport, NADH to ubiquinone': 374.0,\n",
       " 'GO:0043039-tRNA aminoacylation': 375.0,\n",
       " 'GO:0033683-nucleotide-excision repair, DNA incision': 376.0,\n",
       " 'GO:0051293-establishment of spindle localization': 377.0,\n",
       " \"GO:0016796-exonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5'-phosphomonoesters\": 378.0,\n",
       " 'GO:0032392-DNA geometric change': 379.0,\n",
       " 'GO:0035722-interleukin-12-mediated signaling pathway': 380.0,\n",
       " 'GO:0061512-protein localization to cilium': 381.0,\n",
       " 'GO:0000184-nuclear-transcribed mRNA catabolic process, nonsense-mediated decay': 382.0,\n",
       " 'GO:0006284-base-excision repair': 383.0,\n",
       " 'GO:0000049-tRNA binding': 384.0,\n",
       " 'GO:0032182-ubiquitin-like protein binding': 385.0,\n",
       " \"GO:0031123-RNA 3'-end processing\": 386.0,\n",
       " 'GO:0033143-regulation of intracellular steroid hormone receptor signaling pathway': 387.0,\n",
       " 'GO:0097064-ncRNA export from nucleus': 388.0,\n",
       " 'GO:0006406-mRNA export from nucleus': 389.5,\n",
       " 'GO:0071427-mRNA-containing ribonucleoprotein complex export from nucleus': 389.5,\n",
       " 'GO:0098798-mitochondrial protein complex': 391.0,\n",
       " 'GO:0051983-regulation of chromosome segregation': 392.0,\n",
       " 'GO:0005844-polysome': 393.0,\n",
       " 'GO:0019888-protein phosphatase regulator activity': 394.0,\n",
       " \"GO:0003899-DNA-directed 5'-3' RNA polymerase activity\": 395.0,\n",
       " 'GO:0006893-Golgi to plasma membrane transport': 396.0,\n",
       " 'GO:0006368-transcription elongation from RNA polymerase II promoter': 397.0,\n",
       " 'GO:0000959-mitochondrial RNA metabolic process': 398.0,\n",
       " 'GO:0043038-amino acid activation': 399.0,\n",
       " 'GO:0050681-androgen receptor binding': 400.0,\n",
       " 'GO:0098800-inner mitochondrial membrane protein complex': 401.0,\n",
       " 'GO:0071166-ribonucleoprotein complex localization': 402.0,\n",
       " 'GO:0043618-regulation of transcription from RNA polymerase II promoter in response to stress': 403.0,\n",
       " 'GO:0030521-androgen receptor signaling pathway': 404.0,\n",
       " 'GO:0022626-cytosolic ribosome': 405.0,\n",
       " 'GO:1901659-glycosyl compound biosynthetic process': 406.0,\n",
       " 'GO:1901028-regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway': 407.0,\n",
       " 'GO:1901991-negative regulation of mitotic cell cycle phase transition': 408.0,\n",
       " 'GO:0030968-endoplasmic reticulum unfolded protein response': 409.0,\n",
       " 'GO:1901988-negative regulation of cell cycle phase transition': 410.0,\n",
       " 'GO:0071013-catalytic step 2 spliceosome': 411.0,\n",
       " 'GO:0048524-positive regulation of viral process': 412.0,\n",
       " 'GO:0071426-ribonucleoprotein complex export from nucleus': 413.0,\n",
       " 'GO:0072665-protein localization to vacuole': 414.0,\n",
       " 'GO:0006402-mRNA catabolic process': 415.0,\n",
       " 'GO:1903146-regulation of autophagy of mitochondrion': 416.0,\n",
       " 'GO:0032210-regulation of telomere maintenance via telomerase': 417.0,\n",
       " 'GO:0004864-protein phosphatase inhibitor activity': 418.0,\n",
       " 'GO:0043488-regulation of mRNA stability': 419.0,\n",
       " 'GO:0043620-regulation of DNA-templated transcription in response to stress': 420.0,\n",
       " 'GO:0000070-mitotic sister chromatid segregation': 421.0,\n",
       " 'GO:0042149-cellular response to glucose starvation': 422.0,\n",
       " 'GO:0031593-polyubiquitin modification-dependent protein binding': 423.0,\n",
       " 'GO:0042775-mitochondrial ATP synthesis coupled electron transport': 424.0,\n",
       " 'GO:1901799-negative regulation of proteasomal protein catabolic process': 425.0,\n",
       " 'GO:0003684-damaged DNA binding': 426.0,\n",
       " 'GO:0043487-regulation of RNA stability': 427.0,\n",
       " 'GO:0016571-histone methylation': 428.0,\n",
       " 'GO:1902808-positive regulation of cell cycle G1/S phase transition': 429.0,\n",
       " 'GO:0034620-cellular response to unfolded protein': 430.0,\n",
       " 'GO:0009069-serine family amino acid metabolic process': 431.0,\n",
       " 'GO:0021532-neural tube patterning': 432.0,\n",
       " 'GO:0043489-RNA stabilization': 433.0,\n",
       " 'GO:0070849-response to epidermal growth factor': 434.0,\n",
       " 'GO:0051304-chromosome separation': 435.0,\n",
       " 'GO:0006405-RNA export from nucleus': 436.0,\n",
       " 'GO:0000776-kinetochore': 437.0,\n",
       " 'GO:1902369-negative regulation of RNA catabolic process': 438.0,\n",
       " 'GO:0071349-cellular response to interleukin-12': 439.0,\n",
       " 'GO:0044420-extracellular matrix component': 440.0,\n",
       " 'GO:0002478-antigen processing and presentation of exogenous peptide antigen': 441.0,\n",
       " 'GO:0000779-condensed chromosome, centromeric region': 442.0,\n",
       " 'GO:1902749-regulation of cell cycle G2/M phase transition': 443.0,\n",
       " 'GO:0022904-respiratory electron transport chain': 444.0,\n",
       " 'GO:0007004-telomere maintenance via telomerase': 445.0,\n",
       " 'GO:0044839-cell cycle G2/M phase transition': 446.0,\n",
       " 'GO:0006613-cotranslational protein targeting to membrane': 447.0,\n",
       " 'GO:0003735-structural constituent of ribosome': 448.0,\n",
       " 'GO:0070671-response to interleukin-12': 449.0,\n",
       " 'GO:0006614-SRP-dependent cotranslational protein targeting to membrane': 450.0,\n",
       " 'GO:2000573-positive regulation of DNA biosynthetic process': 451.0,\n",
       " 'GO:0042773-ATP synthesis coupled electron transport': 452.0,\n",
       " 'GO:0061013-regulation of mRNA catabolic process': 453.0,\n",
       " 'GO:2000059-negative regulation of ubiquitin-dependent protein catabolic process': 454.0,\n",
       " \"GO:0034062-5'-3' RNA polymerase activity\": 455.5,\n",
       " 'GO:0097747-RNA polymerase activity': 455.5,\n",
       " 'GO:0008023-transcription elongation factor complex': 457.0,\n",
       " 'GO:0034399-nuclear periphery': 458.0,\n",
       " 'GO:0071364-cellular response to epidermal growth factor stimulus': 459.0,\n",
       " 'GO:0000086-G2/M transition of mitotic cell cycle': 460.0,\n",
       " 'GO:0010389-regulation of G2/M transition of mitotic cell cycle': 461.0,\n",
       " 'GO:0051082-unfolded protein binding': 462.0,\n",
       " 'GO:0061912-selective autophagy': 463.0,\n",
       " 'GO:0042073-intraciliary transport': 464.0,\n",
       " 'GO:0000956-nuclear-transcribed mRNA catabolic process': 465.0,\n",
       " 'GO:0006401-RNA catabolic process': 466.0,\n",
       " 'GO:0005776-autophagosome': 467.0,\n",
       " 'GO:0009163-nucleoside biosynthetic process': 468.0,\n",
       " 'GO:0031100-animal organ regeneration': 469.0,\n",
       " 'GO:0007062-sister chromatid cohesion': 470.0,\n",
       " 'GO:0042054-histone methyltransferase activity': 471.0,\n",
       " 'GO:1903363-negative regulation of cellular protein catabolic process': 472.0,\n",
       " 'GO:1903202-negative regulation of oxidative stress-induced cell death': 473.0,\n",
       " 'GO:0042254-ribosome biogenesis': 474.0,\n",
       " 'GO:0005840-ribosome': 475.0,\n",
       " 'GO:0048255-mRNA stabilization': 476.0,\n",
       " 'GO:0035967-cellular response to topologically incorrect protein': 477.0,\n",
       " 'GO:0031970-organelle envelope lumen': 478.0,\n",
       " 'GO:0006283-transcription-coupled nucleotide-excision repair': 479.0,\n",
       " 'GO:0045296-cadherin binding': 480.0,\n",
       " 'GO:0046513-ceramide biosynthetic process': 481.0,\n",
       " 'GO:0006457-protein folding': 482.0,\n",
       " 'GO:0032456-endocytic recycling': 483.0,\n",
       " 'GO:0097345-mitochondrial outer membrane permeabilization': 484.0,\n",
       " 'GO:0006364-rRNA processing': 485.0,\n",
       " 'GO:0019884-antigen processing and presentation of exogenous antigen': 486.0,\n",
       " 'GO:0003697-single-stranded DNA binding': 487.0,\n",
       " 'GO:0075733-intracellular transport of virus': 488.0,\n",
       " 'GO:0055029-nuclear DNA-directed RNA polymerase complex': 489.0,\n",
       " 'GO:0044773-mitotic DNA damage checkpoint': 490.0,\n",
       " 'GO:0038095-Fc-epsilon receptor signaling pathway': 491.0,\n",
       " 'GO:1903051-negative regulation of proteolysis involved in cellular protein catabolic process': 492.0,\n",
       " 'GO:0033238-regulation of cellular amine metabolic process': 493.0,\n",
       " 'GO:0000428-DNA-directed RNA polymerase complex': 494.0,\n",
       " 'GO:0061726-mitochondrion disassembly': 495.5,\n",
       " 'GO:0000422-autophagy of mitochondrion': 495.5,\n",
       " 'GO:0000777-condensed chromosome kinetochore': 497.0,\n",
       " 'GO:0030880-RNA polymerase complex': 498.0,\n",
       " 'GO:0016072-rRNA metabolic process': 499.0,\n",
       " 'GO:0006998-nuclear envelope organization': 500.0,\n",
       " 'GO:0051653-spindle localization': 501.0,\n",
       " 'GO:0045333-cellular respiration': 502.0,\n",
       " 'GO:0006278-RNA-dependent DNA biosynthetic process': 503.0,\n",
       " 'GO:0006261-DNA-dependent DNA replication': 504.0,\n",
       " 'GO:0008094-DNA-dependent ATPase activity': 505.0,\n",
       " 'GO:0005758-mitochondrial intermembrane space': 506.0,\n",
       " \"GO:0003730-mRNA 3'-UTR binding\": 507.0,\n",
       " 'GO:0001736-establishment of planar polarity': 508.5,\n",
       " 'GO:0007164-establishment of tissue polarity': 508.5,\n",
       " 'GO:0019208-phosphatase regulator activity': 510.0,\n",
       " 'GO:0090329-regulation of DNA-dependent DNA replication': 511.0,\n",
       " 'GO:0015934-large ribosomal subunit': 512.0,\n",
       " 'GO:0060071-Wnt signaling pathway, planar cell polarity pathway': 513.0,\n",
       " 'GO:2001252-positive regulation of chromosome organization': 514.0,\n",
       " 'GO:0070936-protein K48-linked ubiquitination': 515.0,\n",
       " 'GO:1902373-negative regulation of mRNA catabolic process': 516.0,\n",
       " 'GO:0031146-SCF-dependent proteasomal ubiquitin-dependent protein catabolic process': 517.0,\n",
       " 'GO:0000775-chromosome, centromeric region': 518.0,\n",
       " 'GO:0007051-spindle organization': 519.0,\n",
       " 'GO:0019886-antigen processing and presentation of exogenous peptide antigen via MHC class II': 520.0,\n",
       " 'GO:0090175-regulation of establishment of planar polarity': 521.0,\n",
       " 'GO:1903312-negative regulation of mRNA metabolic process': 522.0,\n",
       " 'GO:0016073-snRNA metabolic process': 523.0,\n",
       " 'GO:0005643-nuclear pore': 524.0,\n",
       " 'GO:0061077-chaperone-mediated protein folding': 525.0,\n",
       " 'GO:1990204-oxidoreductase complex': 526.0,\n",
       " 'GO:0045182-translation regulator activity': 527.0,\n",
       " 'GO:0000819-sister chromatid segregation': 528.0,\n",
       " 'GO:0034708-methyltransferase complex': 529.0,\n",
       " 'GO:0045047-protein targeting to ER': 530.0,\n",
       " 'GO:0008213-protein alkylation': 531.5,\n",
       " 'GO:0006479-protein methylation': 531.5,\n",
       " 'GO:1902686-mitochondrial outer membrane permeabilization involved in programmed cell death': 533.0,\n",
       " 'GO:0016445-somatic diversification of immunoglobulins': 534.0,\n",
       " 'GO:0007093-mitotic cell cycle checkpoint': 535.0,\n",
       " 'GO:0044774-mitotic DNA integrity checkpoint': 536.0,\n",
       " 'GO:0008173-RNA methyltransferase activity': 537.0,\n",
       " 'GO:0034968-histone lysine methylation': 538.0,\n",
       " 'GO:1902110-positive regulation of mitochondrial membrane permeability involved in apoptotic process': 539.0,\n",
       " 'GO:0048002-antigen processing and presentation of peptide antigen': 540.0,\n",
       " 'GO:0030330-DNA damage response, signal transduction by p53 class mediator': 541.0,\n",
       " 'GO:1903749-positive regulation of establishment of protein localization to mitochondrion': 542.0,\n",
       " 'GO:0006403-RNA localization': 543.0,\n",
       " 'GO:0046794-transport of virus': 544.0,\n",
       " 'GO:0051568-histone H3-K4 methylation': 545.0,\n",
       " 'GO:0005681-spliceosomal complex': 546.0,\n",
       " 'GO:0070972-protein localization to endoplasmic reticulum': 547.0,\n",
       " 'GO:0007006-mitochondrial membrane organization': 548.0,\n",
       " 'GO:0043297-apical junction assembly': 549.0,\n",
       " 'GO:0098876-vesicle-mediated transport to the plasma membrane': 550.0,\n",
       " 'GO:1902117-positive regulation of organelle assembly': 551.0,\n",
       " 'GO:0022853-active ion transmembrane transporter activity': 552.5,\n",
       " 'GO:0042625-ATPase coupled ion transmembrane transporter activity': 552.5,\n",
       " 'GO:0010822-positive regulation of mitochondrion organization': 554.0,\n",
       " 'GO:0046902-regulation of mitochondrial membrane permeability': 555.0,\n",
       " 'GO:0034394-protein localization to cell surface': 556.0,\n",
       " 'GO:1905269-positive regulation of chromatin organization': 557.0,\n",
       " 'GO:0006611-protein export from nucleus': 558.0,\n",
       " 'GO:0006986-response to unfolded protein': 559.0,\n",
       " 'GO:2001242-regulation of intrinsic apoptotic signaling pathway': 560.0,\n",
       " 'GO:0002200-somatic diversification of immune receptors': 561.0,\n",
       " 'GO:0051310-metaphase plate congression': 562.0,\n",
       " 'GO:0006119-oxidative phosphorylation': 563.0,\n",
       " 'GO:0022613-ribonucleoprotein complex biogenesis': 564.0,\n",
       " 'GO:1901987-regulation of cell cycle phase transition': 565.0,\n",
       " 'GO:0044455-mitochondrial membrane part': 566.0,\n",
       " 'GO:0035794-positive regulation of mitochondrial membrane permeability': 567.0,\n",
       " 'GO:0000077-DNA damage checkpoint': 568.0,\n",
       " 'GO:0006892-post-Golgi vesicle-mediated transport': 569.0,\n",
       " 'GO:0006289-nucleotide-excision repair': 570.0,\n",
       " 'GO:0040033-negative regulation of translation, ncRNA-mediated': 572.0,\n",
       " 'GO:0045974-regulation of translation, ncRNA-mediated': 572.0,\n",
       " 'GO:0035278-miRNA mediated inhibition of translation': 572.0,\n",
       " 'GO:0007088-regulation of mitotic nuclear division': 574.0,\n",
       " 'GO:1903747-regulation of establishment of protein localization to mitochondrion': 575.0,\n",
       " 'GO:0001098-basal transcription machinery binding': 576.5,\n",
       " 'GO:0001099-basal RNA polymerase II transcription machinery binding': 576.5,\n",
       " 'GO:0019829-cation-transporting ATPase activity': 578.0,\n",
       " 'GO:1903050-regulation of proteolysis involved in cellular protein catabolic process': 579.0,\n",
       " 'GO:0010948-negative regulation of cell cycle process': 580.0,\n",
       " 'GO:0031058-positive regulation of histone modification': 581.0,\n",
       " 'GO:0072599-establishment of protein localization to endoplasmic reticulum': 582.0,\n",
       " 'GO:0038093-Fc receptor signaling pathway': 583.0,\n",
       " 'GO:0070585-protein localization to mitochondrion': 584.0,\n",
       " 'GO:0000075-cell cycle checkpoint': 585.0,\n",
       " 'GO:0016241-regulation of macroautophagy': 586.0,\n",
       " 'GO:0006400-tRNA modification': 587.0,\n",
       " 'GO:1902108-regulation of mitochondrial membrane permeability involved in apoptotic process': 588.0,\n",
       " 'GO:0035966-response to topologically incorrect protein': 589.0,\n",
       " 'GO:0002495-antigen processing and presentation of peptide antigen via MHC class II': 590.0,\n",
       " 'GO:1903311-regulation of mRNA metabolic process': 591.0,\n",
       " 'GO:1901990-regulation of mitotic cell cycle phase transition': 592.0,\n",
       " 'GO:0072655-establishment of protein localization to mitochondrion': 593.0,\n",
       " 'GO:0031571-mitotic G1 DNA damage checkpoint': 594.5,\n",
       " 'GO:0044819-mitotic G1/S transition checkpoint': 594.5,\n",
       " 'GO:0090559-regulation of membrane permeability': 596.0,\n",
       " 'GO:0035258-steroid hormone receptor binding': 597.0,\n",
       " 'GO:0140014-mitotic nuclear division': 598.0,\n",
       " 'GO:1903052-positive regulation of proteolysis involved in cellular protein catabolic process': 599.0,\n",
       " 'GO:0010494-cytoplasmic stress granule': 600.0,\n",
       " 'GO:0006487-protein N-linked glycosylation': 601.0,\n",
       " 'GO:0002504-antigen processing and presentation of peptide or polysaccharide antigen via MHC class II': 602.0,\n",
       " 'GO:0097194-execution phase of apoptosis': 603.0,\n",
       " 'GO:0035097-histone methyltransferase complex': 604.0,\n",
       " 'GO:1903313-positive regulation of mRNA metabolic process': 605.0,\n",
       " 'GO:0016239-positive regulation of macroautophagy': 606.0,\n",
       " 'GO:1900034-regulation of cellular response to heat': 607.0,\n",
       " 'GO:0031570-DNA integrity checkpoint': 608.0,\n",
       " 'GO:0042770-signal transduction in response to DNA damage': 609.0,\n",
       " 'GO:0016363-nuclear matrix': 610.0,\n",
       " 'GO:1901992-positive regulation of mitotic cell cycle phase transition': 611.0,\n",
       " 'GO:1902653-secondary alcohol biosynthetic process': 612.0,\n",
       " 'GO:0022618-ribonucleoprotein complex assembly': 613.0,\n",
       " 'GO:1903362-regulation of cellular protein catabolic process': 614.0,\n",
       " 'GO:0071826-ribonucleoprotein complex subunit organization': 615.0,\n",
       " 'GO:0030433-ubiquitin-dependent ERAD pathway': 616.0,\n",
       " 'GO:0044783-G1 DNA damage checkpoint': 617.0,\n",
       " 'GO:0046128-purine ribonucleoside metabolic process': 618.0,\n",
       " 'GO:0001738-morphogenesis of a polarized epithelium': 619.0,\n",
       " 'GO:0006606-protein import into nucleus': 620.0,\n",
       " 'GO:0009124-nucleoside monophosphate biosynthetic process': 621.0,\n",
       " 'GO:0045727-positive regulation of translation': 622.0,\n",
       " 'GO:0035257-nuclear hormone receptor binding': 623.0,\n",
       " 'GO:0061136-regulation of proteasomal protein catabolic process': 624.0,\n",
       " 'GO:2000736-regulation of stem cell differentiation': 625.0,\n",
       " 'GO:0009123-nucleoside monophosphate metabolic process': 626.0,\n",
       " 'GO:0006626-protein targeting to mitochondrion': 627.0,\n",
       " 'GO:2000058-regulation of ubiquitin-dependent protein catabolic process': 628.0,\n",
       " 'GO:0051168-nuclear export': 629.0,\n",
       " 'GO:0007034-vacuolar transport': 630.0,\n",
       " 'GO:0007041-lysosomal transport': 631.0,\n",
       " 'GO:0034976-response to endoplasmic reticulum stress': 632.0,\n",
       " 'GO:0032204-regulation of telomere maintenance': 633.0,\n",
       " 'GO:0009161-ribonucleoside monophosphate metabolic process': 634.0,\n",
       " 'GO:0034250-positive regulation of cellular amide metabolic process': 635.0,\n",
       " 'GO:0042278-purine nucleoside metabolic process': 636.0,\n",
       " 'GO:0006695-cholesterol biosynthetic process': 637.0,\n",
       " 'GO:0009167-purine ribonucleoside monophosphate metabolic process': 638.0,\n",
       " 'GO:0009144-purine nucleoside triphosphate metabolic process': 639.0,\n",
       " 'GO:0006260-DNA replication': 640.0,\n",
       " 'GO:0009126-purine nucleoside monophosphate metabolic process': 641.0,\n",
       " 'GO:0051225-spindle assembly': 642.0,\n",
       " 'GO:1901989-positive regulation of cell cycle phase transition': 643.0,\n",
       " 'GO:2000060-positive regulation of ubiquitin-dependent protein catabolic process': 644.0,\n",
       " 'GO:0051170-import into nucleus': 645.0,\n",
       " 'GO:0072422-signal transduction involved in DNA damage checkpoint': 646.5,\n",
       " 'GO:0072401-signal transduction involved in DNA integrity checkpoint': 646.5,\n",
       " 'GO:0000910-cytokinesis': 648.0,\n",
       " 'GO:1905710-positive regulation of membrane permeability': 649.0,\n",
       " 'GO:0051028-mRNA transport': 650.0,\n",
       " 'GO:0042273-ribosomal large subunit biogenesis': 651.0,\n",
       " 'GO:0008276-protein methyltransferase activity': 652.0,\n",
       " 'GO:0009205-purine ribonucleoside triphosphate metabolic process': 653.0,\n",
       " 'GO:0008380-RNA splicing': 654.0,\n",
       " 'GO:0002433-immune response-regulating cell surface receptor signaling pathway involved in phagocytosis': 655.5,\n",
       " 'GO:0038096-Fc-gamma receptor signaling pathway involved in phagocytosis': 655.5,\n",
       " 'GO:0009141-nucleoside triphosphate metabolic process': 657.0,\n",
       " 'GO:0140101-catalytic activity, acting on a tRNA': 658.0,\n",
       " 'GO:0033044-regulation of chromosome organization': 659.0,\n",
       " 'GO:1903008-organelle disassembly': 660.0,\n",
       " 'GO:0006672-ceramide metabolic process': 661.0,\n",
       " 'GO:0050658-RNA transport': 662.5,\n",
       " 'GO:0050657-nucleic acid transport': 662.5,\n",
       " 'GO:0072331-signal transduction by p53 class mediator': 664.0,\n",
       " 'GO:0097193-intrinsic apoptotic signaling pathway': 665.0,\n",
       " 'GO:1901657-glycosyl compound metabolic process': 666.0,\n",
       " 'GO:0042470-melanosome': 667.5,\n",
       " 'GO:0048770-pigment granule': 667.5,\n",
       " 'GO:0071230-cellular response to amino acid stimulus': 669.0,\n",
       " 'GO:1902579-multi-organism localization': 670.5,\n",
       " 'GO:0044766-multi-organism transport': 670.5,\n",
       " 'GO:0051169-nuclear transport': 672.0,\n",
       " 'GO:1900182-positive regulation of protein localization to nucleus': 673.0,\n",
       " 'GO:0009199-ribonucleoside triphosphate metabolic process': 674.0,\n",
       " 'GO:0002431-Fc receptor mediated stimulatory signaling pathway': 675.0,\n",
       " 'GO:0006399-tRNA metabolic process': 676.0,\n",
       " 'GO:0030374-nuclear receptor transcription coactivator activity': 677.0,\n",
       " 'GO:0034333-adherens junction assembly': 678.0,\n",
       " 'GO:0072395-signal transduction involved in cell cycle checkpoint': 679.0,\n",
       " 'GO:0001510-RNA methylation': 680.0,\n",
       " 'GO:0034212-peptide N-acetyltransferase activity': 681.0,\n",
       " 'GO:0034470-ncRNA processing': 682.0,\n",
       " 'GO:0007044-cell-substrate junction assembly': 683.0,\n",
       " 'GO:0000288-nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay': 684.0,\n",
       " 'GO:0042177-negative regulation of protein catabolic process': 685.0,\n",
       " 'GO:1903322-positive regulation of protein modification by small protein conjugation or removal': 686.0,\n",
       " 'GO:0006417-regulation of translation': 687.0,\n",
       " 'GO:0017038-protein import': 688.0,\n",
       " 'GO:0051236-establishment of RNA localization': 689.0,\n",
       " 'GO:0042147-retrograde transport, endosome to Golgi': 690.0,\n",
       " 'GO:0030518-intracellular steroid hormone receptor signaling pathway': 691.0,\n",
       " 'GO:0050821-protein stabilization': 692.0,\n",
       " 'GO:0010821-regulation of mitochondrion organization': 693.0,\n",
       " 'GO:0004527-exonuclease activity': 694.0,\n",
       " 'GO:0045930-negative regulation of mitotic cell cycle': 695.0,\n",
       " 'GO:0006913-nucleocytoplasmic transport': 696.0,\n",
       " 'GO:0009127-purine nucleoside monophosphate biosynthetic process': 697.5,\n",
       " 'GO:0009168-purine ribonucleoside monophosphate biosynthetic process': 697.5,\n",
       " 'GO:0046034-ATP metabolic process': 699.0,\n",
       " 'GO:0004386-helicase activity': 700.0,\n",
       " 'GO:0031503-protein-containing complex localization': 701.0,\n",
       " 'GO:0005743-mitochondrial inner membrane': 702.0,\n",
       " 'GO:0009156-ribonucleoside monophosphate biosynthetic process': 703.0,\n",
       " 'GO:0000784-nuclear chromosome, telomeric region': 704.0,\n",
       " 'GO:0045931-positive regulation of mitotic cell cycle': 705.0,\n",
       " 'GO:0009451-RNA modification': 706.0,\n",
       " 'GO:0044452-nucleolar part': 707.0,\n",
       " 'GO:0009119-ribonucleoside metabolic process': 708.0,\n",
       " 'GO:0019882-antigen processing and presentation': 709.0,\n",
       " 'GO:1903364-positive regulation of cellular protein catabolic process': 710.0,\n",
       " 'GO:0016779-nucleotidyltransferase activity': 711.0,\n",
       " 'GO:0016853-isomerase activity': 712.0,\n",
       " 'GO:0018022-peptidyl-lysine methylation': 713.0,\n",
       " 'GO:0016234-inclusion body': 714.0,\n",
       " 'GO:0006513-protein monoubiquitination': 715.0,\n",
       " 'GO:0022900-electron transport chain': 716.0,\n",
       " 'GO:0034332-adherens junction organization': 717.0,\n",
       " 'GO:0000375-RNA splicing, via transesterification reactions': 718.0,\n",
       " 'GO:2000278-regulation of DNA biosynthetic process': 719.0,\n",
       " 'GO:0005759-mitochondrial matrix': 720.0,\n",
       " 'GO:0016651-oxidoreductase activity, acting on NAD(P)H': 721.0,\n",
       " 'GO:0009116-nucleoside metabolic process': 722.0,\n",
       " 'GO:0009185-ribonucleoside diphosphate metabolic process': 723.0,\n",
       " 'GO:1901800-positive regulation of proteasomal protein catabolic process': 724.0,\n",
       " 'GO:0009206-purine ribonucleoside triphosphate biosynthetic process': 725.0,\n",
       " 'GO:0006839-mitochondrial transport': 726.0,\n",
       " 'GO:0000398-mRNA splicing, via spliceosome': 727.5,\n",
       " 'GO:0000377-RNA splicing, via transesterification reactions with bulged adenosine as nucleophile': 727.5,\n",
       " 'GO:0044786-cell cycle DNA replication': 729.0,\n",
       " 'GO:0051783-regulation of nuclear division': 730.0,\n",
       " 'GO:0006605-protein targeting': 731.0,\n",
       " 'GO:0101002-ficolin-1-rich granule': 732.0,\n",
       " 'GO:1990542-mitochondrial transmembrane transport': 733.0,\n",
       " 'GO:0038094-Fc-gamma receptor signaling pathway': 734.0,\n",
       " 'GO:0050792-regulation of viral process': 735.0,\n",
       " 'GO:0008757-S-adenosylmethionine-dependent methyltransferase activity': 736.0,\n",
       " 'GO:0010507-negative regulation of autophagy': 737.0,\n",
       " 'GO:0044445-cytosolic part': 738.0,\n",
       " 'GO:0032434-regulation of proteasomal ubiquitin-dependent protein catabolic process': 739.0,\n",
       " 'GO:0098687-chromosomal region': 740.0,\n",
       " 'GO:0005925-focal adhesion': 741.0,\n",
       " 'GO:0000794-condensed nuclear chromosome': 742.0,\n",
       " 'GO:0005924-cell-substrate adherens junction': 743.0,\n",
       " 'GO:0043161-proteasome-mediated ubiquitin-dependent protein catabolic process': 744.0,\n",
       " 'GO:0030117-membrane coat': 745.5,\n",
       " 'GO:0048475-coated membrane': 745.5,\n",
       " 'GO:0009145-purine nucleoside triphosphate biosynthetic process': 747.0,\n",
       " 'GO:0000932-P-body': 748.0,\n",
       " 'GO:0030532-small nuclear ribonucleoprotein complex': 749.0,\n",
       " 'GO:0006900-vesicle budding from membrane': 750.0,\n",
       " 'GO:0001650-fibrillar center': 751.0,\n",
       " 'GO:0120114-Sm-like protein family complex': 752.0,\n",
       " 'GO:0044389-ubiquitin-like protein ligase binding': 753.0,\n",
       " 'GO:1902806-regulation of cell cycle G1/S phase transition': 754.0,\n",
       " 'GO:0016126-sterol biosynthetic process': 755.0,\n",
       " 'GO:0009201-ribonucleoside triphosphate biosynthetic process': 756.0,\n",
       " 'GO:0006275-regulation of DNA replication': 757.0,\n",
       " 'GO:0030055-cell-substrate junction': 758.0,\n",
       " 'GO:0009132-nucleoside diphosphate metabolic process': 759.0,\n",
       " 'GO:0090501-RNA phosphodiester bond hydrolysis': 760.0,\n",
       " 'GO:0009179-purine ribonucleoside diphosphate metabolic process': 761.5,\n",
       " 'GO:0009135-purine nucleoside diphosphate metabolic process': 761.5,\n",
       " 'GO:0034504-protein localization to nucleus': 763.0,\n",
       " 'GO:0043624-cellular protein complex disassembly': 764.0,\n",
       " 'GO:0048193-Golgi vesicle transport': 765.0,\n",
       " 'GO:0010833-telomere maintenance via telomere lengthening': 766.0,\n",
       " 'GO:0031625-ubiquitin protein ligase binding': 767.0,\n",
       " 'GO:0017148-negative regulation of translation': 768.0,\n",
       " 'GO:0031056-regulation of histone modification': 769.0,\n",
       " 'GO:0009142-nucleoside triphosphate biosynthetic process': 770.0,\n",
       " 'GO:0016197-endosomal transport': 771.0,\n",
       " 'GO:0005913-cell-cell adherens junction': 772.0,\n",
       " 'GO:0140098-catalytic activity, acting on RNA': 773.0,\n",
       " 'GO:0008033-tRNA processing': 774.0,\n",
       " 'GO:0006367-transcription initiation from RNA polymerase II promoter': 775.0,\n",
       " 'GO:0031398-positive regulation of protein ubiquitination': 776.0,\n",
       " 'GO:2001251-negative regulation of chromosome organization': 777.0,\n",
       " 'GO:0034248-regulation of cellular amide metabolic process': 778.0,\n",
       " 'GO:0015931-nucleobase-containing compound transport': 779.0,\n",
       " 'GO:0071456-cellular response to hypoxia': 780.0,\n",
       " 'GO:0097711-ciliary basal body-plasma membrane docking': 781.0,\n",
       " 'GO:0004540-ribonuclease activity': 782.0,\n",
       " 'GO:0036503-ERAD pathway': 783.0,\n",
       " 'GO:0000287-magnesium ion binding': 784.0,\n",
       " 'GO:0030176-integral component of endoplasmic reticulum membrane': 785.0,\n",
       " 'GO:0044843-cell cycle G1/S phase transition': 786.0,\n",
       " 'GO:0032436-positive regulation of proteasomal ubiquitin-dependent protein catabolic process': 787.0,\n",
       " 'GO:0006754-ATP biosynthetic process': 788.0,\n",
       " 'GO:0016236-macroautophagy': 789.0,\n",
       " 'GO:0097525-spliceosomal snRNP complex': 790.0,\n",
       " 'GO:0090150-establishment of protein localization to membrane': 791.0,\n",
       " 'GO:0006612-protein targeting to membrane': 792.0,\n",
       " 'GO:0034249-negative regulation of cellular amide metabolic process': 793.0,\n",
       " 'GO:0031491-nucleosome binding': 794.0,\n",
       " 'GO:0090090-negative regulation of canonical Wnt signaling pathway': 795.0,\n",
       " 'GO:0018210-peptidyl-threonine modification': 796.0,\n",
       " 'GO:0003713-transcription coactivator activity': 797.0,\n",
       " 'GO:0010508-positive regulation of autophagy': 798.0,\n",
       " 'GO:0030496-midbody': 799.0,\n",
       " 'GO:0006165-nucleoside diphosphate phosphorylation': 800.0,\n",
       " 'GO:0043903-regulation of symbiosis, encompassing mutualism through parasitism': 801.0,\n",
       " 'GO:0090263-positive regulation of canonical Wnt signaling pathway': 802.0,\n",
       " 'GO:0043414-macromolecule methylation': 803.0,\n",
       " 'GO:0006997-nucleus organization': 804.0,\n",
       " 'GO:0019903-protein phosphatase binding': 805.0,\n",
       " 'GO:0006757-ATP generation from ADP': 806.0,\n",
       " 'GO:0009055-electron transfer activity': 807.0,\n",
       " 'GO:0047485-protein N-terminus binding': 808.0,\n",
       " 'GO:0046939-nucleotide phosphorylation': 809.0,\n",
       " 'GO:0036294-cellular response to decreased oxygen levels': 810.0,\n",
       " 'GO:0046031-ADP metabolic process': 811.0,\n",
       " 'GO:0098813-nuclear chromosome segregation': 812.0,\n",
       " 'GO:0031330-negative regulation of cellular catabolic process': 813.0,\n",
       " 'GO:0060147-regulation of posttranscriptional gene silencing': 814.5,\n",
       " 'GO:0060966-regulation of gene silencing by RNA': 814.5,\n",
       " 'GO:0016482-cytosolic transport': 816.0,\n",
       " 'GO:0006352-DNA-templated transcription, initiation': 817.0,\n",
       " 'GO:0051054-positive regulation of DNA metabolic process': 818.0,\n",
       " 'GO:0007059-chromosome segregation': 819.0,\n",
       " 'GO:0140097-catalytic activity, acting on DNA': 820.0,\n",
       " 'GO:0010498-proteasomal protein catabolic process': 821.0,\n",
       " 'GO:0019902-phosphatase binding': 822.0,\n",
       " 'GO:0006096-glycolytic process': 823.0,\n",
       " 'GO:0008168-methyltransferase activity': 824.0,\n",
       " 'GO:0008022-protein C-terminus binding': 825.0,\n",
       " 'GO:0031227-intrinsic component of endoplasmic reticulum membrane': 826.0,\n",
       " 'GO:0009411-response to UV': 827.0,\n",
       " 'GO:0030705-cytoskeleton-dependent intracellular transport': 828.0,\n",
       " 'GO:0000209-protein polyubiquitination': 829.0,\n",
       " 'GO:1900180-regulation of protein localization to nucleus': 830.0,\n",
       " 'GO:0031461-cullin-RING ubiquitin ligase complex': 831.0,\n",
       " 'GO:0032259-methylation': 832.0,\n",
       " 'GO:2000134-negative regulation of G1/S transition of mitotic cell cycle': 833.0,\n",
       " 'GO:1902275-regulation of chromatin organization': 834.0,\n",
       " 'GO:0005819-spindle': 835.0,\n",
       " 'GO:0071897-DNA biosynthetic process': 836.0,\n",
       " 'GO:2000045-regulation of G1/S transition of mitotic cell cycle': 837.0,\n",
       " 'GO:0051101-regulation of DNA binding': 838.0,\n",
       " 'GO:0003729-mRNA binding': 839.0,\n",
       " 'GO:1901796-regulation of signal transduction by p53 class mediator': 840.0,\n",
       " 'GO:0006888-ER to Golgi vesicle-mediated transport': 841.0,\n",
       " 'GO:0060968-regulation of gene silencing': 842.0,\n",
       " 'GO:0043687-post-translational protein modification': 843.0,\n",
       " 'GO:0042866-pyruvate biosynthetic process': 844.0,\n",
       " 'GO:0043484-regulation of RNA splicing': 845.0,\n",
       " 'GO:0060964-regulation of gene silencing by miRNA': 846.0,\n",
       " 'GO:0050684-regulation of mRNA processing': 847.0,\n",
       " 'GO:0000793-condensed chromosome': 848.0,\n",
       " 'GO:0016887-ATPase activity': 849.0,\n",
       " 'GO:0042176-regulation of protein catabolic process': 850.0,\n",
       " 'GO:1902807-negative regulation of cell cycle G1/S phase transition': 851.0,\n",
       " 'GO:0016741-transferase activity, transferring one-carbon groups': 852.0,\n",
       " 'GO:1902115-regulation of organelle assembly': 853.0,\n",
       " 'GO:0043200-response to amino acid': 854.0,\n",
       " 'GO:0045732-positive regulation of protein catabolic process': 855.0,\n",
       " 'GO:0000082-G1/S transition of mitotic cell cycle': 856.0,\n",
       " 'GO:0009895-negative regulation of catabolic process': 857.0,\n",
       " 'GO:0033157-regulation of intracellular protein transport': 858.0,\n",
       " 'GO:0045335-phagocytic vesicle': 859.0,\n",
       " 'GO:0061695-transferase complex, transferring phosphorus-containing groups': 860.0,\n",
       " 'GO:0000151-ubiquitin ligase complex': 861.0,\n",
       " 'GO:0032549-ribonucleoside binding': 862.0,\n",
       " 'GO:0071347-cellular response to interleukin-1': 863.0,\n",
       " 'GO:0032550-purine ribonucleoside binding': 864.0,\n",
       " 'GO:0015980-energy derivation by oxidation of organic compounds': 865.0,\n",
       " 'GO:0065004-protein-DNA complex assembly': 866.0,\n",
       " 'GO:0001883-purine nucleoside binding': 867.0,\n",
       " 'GO:0140056-organelle localization by membrane tethering': 868.0,\n",
       " 'GO:0018107-peptidyl-threonine phosphorylation': 869.0,\n",
       " 'GO:0015399-primary active transmembrane transporter activity': 870.0,\n",
       " 'GO:0043666-regulation of phosphoprotein phosphatase activity': 871.0,\n",
       " 'GO:0032984-protein-containing complex disassembly': 872.0,\n",
       " 'GO:0030178-negative regulation of Wnt signaling pathway': 873.0,\n",
       " 'GO:0001882-nucleoside binding': 874.0,\n",
       " 'GO:0070646-protein modification by small protein removal': 875.0,\n",
       " 'GO:0022406-membrane docking': 876.0,\n",
       " 'GO:0010506-regulation of autophagy': 877.0,\n",
       " 'GO:0031647-regulation of protein stability': 878.0,\n",
       " 'GO:0005525-GTP binding': 879.0,\n",
       " 'GO:0050839-cell adhesion molecule binding': 880.0,\n",
       " 'GO:0000781-chromosome, telomeric region': 881.0,\n",
       " 'GO:0043401-steroid hormone mediated signaling pathway': 882.0,\n",
       " 'GO:0048285-organelle fission': 883.0,\n",
       " 'GO:0005795-Golgi stack': 884.0,\n",
       " 'GO:0031331-positive regulation of cellular catabolic process': 885.0,\n",
       " 'GO:0009260-ribonucleotide biosynthetic process': 886.0,\n",
       " 'GO:0016579-protein deubiquitination': 887.0,\n",
       " 'GO:0002020-protease binding': 888.0,\n",
       " 'GO:0051427-hormone receptor binding': 889.0,\n",
       " 'GO:1903320-regulation of protein modification by small protein conjugation or removal': 890.0,\n",
       " 'GO:0006520-cellular amino acid metabolic process': 891.0,\n",
       " 'GO:0010970-transport along microtubule': 892.5,\n",
       " 'GO:0099111-microtubule-based transport': 892.5,\n",
       " 'GO:0046390-ribose phosphate biosynthetic process': 894.0,\n",
       " 'GO:0009152-purine ribonucleotide biosynthetic process': 895.0,\n",
       " 'GO:0007050-cell cycle arrest': 896.0,\n",
       " 'GO:0016607-nuclear speck': 897.0,\n",
       " 'GO:0034404-nucleobase-containing small molecule biosynthetic process': 898.0,\n",
       " 'GO:0002244-hematopoietic progenitor cell differentiation': 899.0,\n",
       " 'GO:0090305-nucleic acid phosphodiester bond hydrolysis': 900.0,\n",
       " 'GO:0018205-peptidyl-lysine modification': 901.0,\n",
       " 'GO:0036464-cytoplasmic ribonucleoprotein granule': 902.0,\n",
       " 'GO:0000280-nuclear division': 903.0,\n",
       " 'GO:0071453-cellular response to oxygen levels': 904.0,\n",
       " 'GO:0005741-mitochondrial outer membrane': 905.0,\n",
       " 'GO:0016573-histone acetylation': 906.0,\n",
       " 'GO:0035567-non-canonical Wnt signaling pathway': 907.0,\n",
       " 'GO:0003924-GTPase activity': 908.0,\n",
       " 'GO:0001726-ruffle': 909.0,\n",
       " 'GO:0010565-regulation of cellular ketone metabolic process': 910.0,\n",
       " 'GO:0019001-guanyl nucleotide binding': 911.5,\n",
       " 'GO:0032561-guanyl ribonucleotide binding': 911.5,\n",
       " 'GO:0019827-stem cell population maintenance': 913.0,\n",
       " 'GO:0042623-ATPase activity, coupled': 914.0,\n",
       " 'GO:0006733-oxidoreduction coenzyme metabolic process': 915.0,\n",
       " 'GO:0090068-positive regulation of cell cycle process': 916.0,\n",
       " 'GO:0006475-internal protein amino acid acetylation': 917.0,\n",
       " 'GO:0098727-maintenance of cell number': 918.0,\n",
       " 'GO:0072522-purine-containing compound biosynthetic process': 919.0,\n",
       " 'GO:0003714-transcription corepressor activity': 920.0,\n",
       " 'GO:0010921-regulation of phosphatase activity': 921.0,\n",
       " 'GO:0019362-pyridine nucleotide metabolic process': 922.5,\n",
       " 'GO:0046496-nicotinamide nucleotide metabolic process': 922.5,\n",
       " 'GO:0009896-positive regulation of catabolic process': 924.0,\n",
       " 'GO:0042579-microbody': 925.5,\n",
       " 'GO:0005777-peroxisome': 925.5,\n",
       " 'GO:0016570-histone modification': 927.0,\n",
       " 'GO:0061919-process utilizing autophagic mechanism': 928.5,\n",
       " 'GO:0006914-autophagy': 928.5,\n",
       " 'GO:0071824-protein-DNA complex subunit organization': 930.0,\n",
       " 'GO:0051052-regulation of DNA metabolic process': 931.0,\n",
       " 'GO:0070555-response to interleukin-1': 932.0,\n",
       " 'GO:0018393-internal peptidyl-lysine acetylation': 933.0,\n",
       " 'GO:0009166-nucleotide catabolic process': 934.0,\n",
       " 'GO:0032886-regulation of microtubule-based process': 935.0,\n",
       " 'GO:1905330-regulation of morphogenesis of an epithelium': 936.0,\n",
       " 'GO:0090575-RNA polymerase II transcription factor complex': 937.0,\n",
       " 'GO:0007163-establishment or maintenance of cell polarity': 938.0,\n",
       " 'GO:0030177-positive regulation of Wnt signaling pathway': 939.0,\n",
       " 'GO:0006310-DNA recombination': 940.0,\n",
       " 'GO:0000922-spindle pole': 941.0,\n",
       " 'GO:0006164-purine nucleotide biosynthetic process': 942.0,\n",
       " 'GO:0006338-chromatin remodeling': 943.0,\n",
       " 'GO:0043902-positive regulation of multi-organism process': 944.0,\n",
       " 'GO:1901292-nucleoside phosphate catabolic process': 945.0,\n",
       " 'GO:0016569-covalent chromatin modification': 946.0,\n",
       " 'GO:0045787-positive regulation of cell cycle': 947.0,\n",
       " 'GO:0072524-pyridine-containing compound metabolic process': 948.0,\n",
       " 'GO:0035770-ribonucleoprotein granule': 949.0,\n",
       " 'GO:0031396-regulation of protein ubiquitination': 950.0,\n",
       " 'GO:2001235-positive regulation of apoptotic signaling pathway': 951.0,\n",
       " 'GO:0071103-DNA conformation change': 952.0,\n",
       " 'GO:0005802-trans-Golgi network': 953.0,\n",
       " 'GO:0030522-intracellular receptor signaling pathway': 954.0,\n",
       " 'GO:0034329-cell junction assembly': 955.0,\n",
       " 'GO:0046165-alcohol biosynthetic process': 956.0,\n",
       " 'GO:0043543-protein acylation': 957.0,\n",
       " 'GO:1901293-nucleoside phosphate biosynthetic process': 958.0,\n",
       " 'GO:0006575-cellular modified amino acid metabolic process': 959.0,\n",
       " 'GO:0030674-protein binding, bridging': 960.0,\n",
       " 'GO:0005635-nuclear envelope': 961.0,\n",
       " 'GO:0009165-nucleotide biosynthetic process': 962.0,\n",
       " 'GO:0045862-positive regulation of proteolysis': 963.0,\n",
       " 'GO:0006694-steroid biosynthetic process': 964.0,\n",
       " 'GO:2001233-regulation of apoptotic signaling pathway': 965.0,\n",
       " 'GO:0038061-NIK/NF-kappaB signaling': 966.0,\n",
       " 'GO:2001020-regulation of response to DNA damage stimulus': 967.0,\n",
       " 'GO:0006473-protein acetylation': 968.0,\n",
       " 'GO:2001234-negative regulation of apoptotic signaling pathway': 969.0,\n",
       " 'GO:0007568-aging': 970.0,\n",
       " 'GO:0009408-response to heat': 971.0,\n",
       " 'GO:0009108-coenzyme biosynthetic process': 972.0,\n",
       " 'GO:0031968-organelle outer membrane': 973.0,\n",
       " 'GO:0042180-cellular ketone metabolic process': 974.0,\n",
       " 'GO:0051495-positive regulation of cytoskeleton organization': 975.0,\n",
       " 'GO:0034599-cellular response to oxidative stress': 976.0,\n",
       " 'GO:0004518-nuclease activity': 977.0,\n",
       " 'GO:0060090-molecular adaptor activity': 978.0,\n",
       " 'GO:0006732-coenzyme metabolic process': 979.0,\n",
       " 'GO:1903829-positive regulation of cellular protein localization': 980.0,\n",
       " 'GO:1990778-protein localization to cell periphery': 981.0,\n",
       " 'GO:0031334-positive regulation of protein complex assembly': 982.0,\n",
       " 'GO:0042393-histone binding': 983.0,\n",
       " 'GO:0043312-neutrophil degranulation': 984.0,\n",
       " 'GO:0016747-transferase activity, transferring acyl groups other than amino-acyl groups': 985.0,\n",
       " 'GO:0070507-regulation of microtubule cytoskeleton organization': 986.0,\n",
       " 'GO:0034774-secretory granule lumen': 987.0,\n",
       " 'GO:0060828-regulation of canonical Wnt signaling pathway': 988.0,\n",
       " 'GO:0034330-cell junction organization': 989.0,\n",
       " 'GO:0002283-neutrophil activation involved in immune response': 990.0,\n",
       " 'GO:0043254-regulation of protein complex assembly': 991.0,\n",
       " 'GO:0005788-endoplasmic reticulum lumen': 992.0,\n",
       " 'GO:0044272-sulfur compound biosynthetic process': 993.0,\n",
       " 'GO:0048863-stem cell differentiation': 994.0,\n",
       " 'GO:0042119-neutrophil activation': 995.0,\n",
       " 'GO:0002446-neutrophil mediated immunity': 996.0,\n",
       " 'GO:0010639-negative regulation of organelle organization': 997.0,\n",
       " 'GO:0006979-response to oxidative stress': 998.0,\n",
       " 'GO:0031983-vesicle lumen': 999.0,\n",
       " 'GO:0060205-cytoplasmic vesicle lumen': 1000.0,\n",
       " ...}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make dictionary {GO ID-description:rank}\n",
    "template_rank_dict = dict(zip(enriched_pathways[\"ID_description\"], \n",
    "                              enriched_pathways[\"ranking\"]\n",
    "                             ))\n",
    "template_rank_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "Again, we want to compare our ranked pathways found against what was reported in the original publication.\n",
    "\n",
    "*The DEX-responsive genes that we identified are primarily implicated in two broad classes: stress response and development (Table 1; Reimand et al. 2007). Comparison to Gene Ontology (GO) categories (Ashburner et al. 2000) revealed that the identified genes are involved in stress response (P = 6 × 10−11), organ development (P = 5 × 10−15), cell differentiation (P = 1 × 10−11), hormone secretion (P = 4 × 10−7), and apoptosis (P = 5 × 10−12).*\n",
    "\n",
    "We found pathways that are consistent with what publication found: pathways related to anatomical structure (i.e. vinculin, ) and cell differentiation (i.e. centromeric sister chromatid cohesion), hormone secretion (i.e. somatic hypermutation of immunoglobulin genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Simulated experiments**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def FullMergeDict(D1, D2):\n",
    "    for key, value in D1.items():\n",
    "        if key in D2:\n",
    "            if type(value) is dict:\n",
    "                FullMergeDict(D1[key], D2[key])\n",
    "            else:\n",
    "                if type(value) in (int, float, str):\n",
    "                    D1[key] = [value]\n",
    "                if type(D2[key]) is list:\n",
    "                    D1[key].extend(D2[key])\n",
    "                else:\n",
    "                    D1[key].append(D2[key])\n",
    "    for key, value in D2.items():\n",
    "        if key not in D1:\n",
    "            D1[key] = value\n",
    "    return(D1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get distribution of ranks for simulated experiments\n",
    "for i in range(num_runs):\n",
    "    simulated_GSEA_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"GSEA_stats\",\n",
    "        \"GSEA_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "    \n",
    "    #Read results\n",
    "    simulated_enriched_pathways = pd.read_csv(\n",
    "        simulated_GSEA_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "    \n",
    "    # zip GO ID and description to get unique key\n",
    "    simulated_enriched_pathways[\"ID_description\"] = simulated_enriched_pathways[\"ID\"]+\"-\"+simulated_enriched_pathways[\"Description\"]\n",
    "    \n",
    "    # Get ranks of simulated experiment\n",
    "    simulated_enriched_pathways['ranking'] = simulated_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "    simulated_enriched_pathways = simulated_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "    \n",
    "    if i == 0:\n",
    "        # Initiate dictionary {GO ID-description:rank}\n",
    "        simulated_rank_dict = dict(zip(simulated_enriched_pathways[\"ID_description\"], \n",
    "                                      simulated_enriched_pathways[\"ranking\"]\n",
    "                                     ))\n",
    "    else:\n",
    "        tmp_simulated_rank_dict = dict(zip(simulated_enriched_pathways[\"ID_description\"], \n",
    "                                      simulated_enriched_pathways[\"ranking\"]\n",
    "                                     ))\n",
    "        \n",
    "        simulated_rank_dict = FullMergeDict(simulated_rank_dict, tmp_simulated_rank_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GO:0000353-formation of quadruple SL/U4/U5/U6 snRNP': [2.0,\n",
       "  3.0,\n",
       "  4.0,\n",
       "  5.0,\n",
       "  3.0,\n",
       "  54.0,\n",
       "  4.0,\n",
       "  12.0,\n",
       "  29.0,\n",
       "  6.0,\n",
       "  2.0,\n",
       "  41.0,\n",
       "  3.0,\n",
       "  3.0,\n",
       "  2.0,\n",
       "  25.0,\n",
       "  52.0,\n",
       "  11.0,\n",
       "  16.0],\n",
       " 'GO:0045291-mRNA trans splicing, SL addition': [2.0,\n",
       "  3.0,\n",
       "  4.0,\n",
       "  5.0,\n",
       "  3.0,\n",
       "  54.0,\n",
       "  4.0,\n",
       "  12.0,\n",
       "  29.0,\n",
       "  6.0,\n",
       "  2.0,\n",
       "  41.0,\n",
       "  3.0,\n",
       "  3.0,\n",
       "  2.0,\n",
       "  25.0,\n",
       "  52.0,\n",
       "  11.0,\n",
       "  16.0],\n",
       " 'GO:0000365-mRNA trans splicing, via spliceosome': [2.0,\n",
       "  3.0,\n",
       "  4.0,\n",
       "  5.0,\n",
       "  3.0,\n",
       "  54.0,\n",
       "  4.0,\n",
       "  12.0,\n",
       "  29.0,\n",
       "  6.0,\n",
       "  2.0,\n",
       "  41.0,\n",
       "  3.0,\n",
       "  3.0,\n",
       "  2.0,\n",
       "  25.0,\n",
       "  52.0,\n",
       "  11.0,\n",
       "  16.0],\n",
       " 'GO:1904851-positive regulation of establishment of protein localization to telomere': [5.0,\n",
       "  2.0,\n",
       "  17.0,\n",
       "  9.0,\n",
       "  2.0],\n",
       " 'GO:1903405-protein localization to nuclear body': [5.0, 2.0, 17.0, 9.0, 2.0],\n",
       " 'GO:1904867-protein localization to Cajal body': [5.0, 2.0, 17.0, 9.0, 2.0],\n",
       " 'GO:0050786-RAGE receptor binding': [7.0, 16.0, 34.0, 12.0, 29.0, 22.0],\n",
       " 'GO:1905651-regulation of artery morphogenesis': [8.0,\n",
       "  27.0,\n",
       "  24.0,\n",
       "  57.0,\n",
       "  1.0,\n",
       "  25.0,\n",
       "  35.0,\n",
       "  4.0,\n",
       "  9.0,\n",
       "  16.0,\n",
       "  22.0,\n",
       "  1.0,\n",
       "  40.0,\n",
       "  50.0,\n",
       "  22.0],\n",
       " 'GO:1904874-positive regulation of telomerase RNA localization to Cajal body': [9.0,\n",
       "  86.0,\n",
       "  102.0,\n",
       "  8.0,\n",
       "  13.0,\n",
       "  32.0,\n",
       "  7.0,\n",
       "  73.0],\n",
       " 'GO:0042627-chylomicron': [10.0,\n",
       "  1.0,\n",
       "  8.0,\n",
       "  33.0,\n",
       "  23.0,\n",
       "  9.0,\n",
       "  26.0,\n",
       "  43.0,\n",
       "  5.0,\n",
       "  14.0,\n",
       "  8.0,\n",
       "  14.0,\n",
       "  31.0,\n",
       "  8.0,\n",
       "  21.0,\n",
       "  31.0,\n",
       "  23.0,\n",
       "  30.0,\n",
       "  18.0],\n",
       " 'GO:0072588-box H/ACA RNP complex': [11.0, 6.0, 2.0, 22.0, 19.0, 13.0],\n",
       " 'GO:0005832-chaperonin-containing T-complex': [12.0, 30.0, 51.0, 6.0, 10.0],\n",
       " 'GO:0004032-alditol:NADP+ 1-oxidoreductase activity': [13.0,\n",
       "  17.0,\n",
       "  29.0,\n",
       "  56.0,\n",
       "  21.0,\n",
       "  34.0,\n",
       "  4.0,\n",
       "  56.0,\n",
       "  38.0,\n",
       "  96.0,\n",
       "  12.0,\n",
       "  3.0,\n",
       "  1.0,\n",
       "  5.0,\n",
       "  5.0,\n",
       "  41.0,\n",
       "  38.0,\n",
       "  13.0],\n",
       " 'GO:1990173-protein localization to nucleoplasm': [14.0, 22.0],\n",
       " 'GO:0030388-fructose 1,6-bisphosphate metabolic process': [15.0,\n",
       "  18.0,\n",
       "  2.0,\n",
       "  44.0,\n",
       "  35.0,\n",
       "  20.0],\n",
       " 'GO:1904816-positive regulation of protein localization to chromosome, telomeric region': [16.0,\n",
       "  15.0,\n",
       "  67.0,\n",
       "  17.0,\n",
       "  5.0,\n",
       "  21.0,\n",
       "  48.0],\n",
       " 'GO:2000628-regulation of miRNA metabolic process': 17.0,\n",
       " 'GO:1901838-positive regulation of transcription of nucleolar large rRNA by RNA polymerase I': [18.0,\n",
       "  10.0,\n",
       "  8.0],\n",
       " \"GO:0030627-pre-mRNA 5'-splice site binding\": [19.0,\n",
       "  205.0,\n",
       "  122.0,\n",
       "  46.0,\n",
       "  251.0,\n",
       "  104.0,\n",
       "  91.0,\n",
       "  89.0,\n",
       "  173.0,\n",
       "  112.0,\n",
       "  6.0,\n",
       "  86.0,\n",
       "  16.0,\n",
       "  14.0,\n",
       "  91.0,\n",
       "  108.0,\n",
       "  287.0],\n",
       " 'GO:0070203-regulation of establishment of protein localization to telomere': [20.0,\n",
       "  5.0,\n",
       "  72.0,\n",
       "  18.0,\n",
       "  15.0],\n",
       " 'GO:0034378-chylomicron assembly': [21.0,\n",
       "  15.0,\n",
       "  16.0,\n",
       "  30.0,\n",
       "  5.0,\n",
       "  25.0,\n",
       "  33.0,\n",
       "  34.0,\n",
       "  7.0,\n",
       "  7.0,\n",
       "  13.0,\n",
       "  11.0,\n",
       "  33.0,\n",
       "  15.0,\n",
       "  6.0],\n",
       " 'GO:0034310-primary alcohol catabolic process': [22.0,\n",
       "  85.0,\n",
       "  71.0,\n",
       "  40.0,\n",
       "  102.0,\n",
       "  41.0],\n",
       " 'GO:0070202-regulation of establishment of protein localization to chromosome': [23.0,\n",
       "  7.0,\n",
       "  101.0,\n",
       "  20.0,\n",
       "  28.0],\n",
       " 'GO:0030647-aminoglycoside antibiotic metabolic process': [24.0,\n",
       "  29.0,\n",
       "  36.0,\n",
       "  14.0,\n",
       "  13.0,\n",
       "  45.0,\n",
       "  9.0,\n",
       "  11.0,\n",
       "  10.0,\n",
       "  9.0,\n",
       "  35.0,\n",
       "  32.0,\n",
       "  28.0],\n",
       " 'GO:0019911-structural constituent of myelin sheath': [25.0, 14.0],\n",
       " 'GO:0051918-negative regulation of fibrinolysis': [26.0,\n",
       "  8.0,\n",
       "  62.0,\n",
       "  28.0,\n",
       "  2.0,\n",
       "  27.0,\n",
       "  2.0,\n",
       "  7.0,\n",
       "  23.0,\n",
       "  11.0],\n",
       " 'GO:0000727-double-strand break repair via break-induced replication': [27.0,\n",
       "  6.0,\n",
       "  49.0,\n",
       "  38.0,\n",
       "  5.0,\n",
       "  17.0,\n",
       "  8.0,\n",
       "  4.0,\n",
       "  76.0,\n",
       "  3.0],\n",
       " 'GO:0034709-methylosome': [28.0,\n",
       "  26.0,\n",
       "  50.0,\n",
       "  20.0,\n",
       "  23.0,\n",
       "  32.0,\n",
       "  30.0,\n",
       "  30.0,\n",
       "  57.0],\n",
       " 'GO:1902915-negative regulation of protein polyubiquitination': [29.0, 9.0],\n",
       " 'GO:0097468-programmed cell death in response to reactive oxygen species': [30.5,\n",
       "  15.5,\n",
       "  60.5,\n",
       "  4.5],\n",
       " 'GO:0010421-hydrogen peroxide-mediated programmed cell death': [30.5,\n",
       "  15.5,\n",
       "  60.5,\n",
       "  4.5],\n",
       " 'GO:0070200-establishment of protein localization to telomere': [32.0,\n",
       "  123.0,\n",
       "  11.0,\n",
       "  24.0,\n",
       "  46.0],\n",
       " 'GO:0016327-apicolateral plasma membrane': [33.0, 171.0],\n",
       " 'GO:0004596-peptide alpha-N-acetyltransferase activity': 34.0,\n",
       " 'GO:0030621-U4 snRNA binding': [35.0, 24.0, 41.0, 53.0, 89.0, 30.0, 7.0],\n",
       " 'GO:0034370-triglyceride-rich lipoprotein particle remodeling': [36.0,\n",
       "  113.0,\n",
       "  81.0,\n",
       "  45.0,\n",
       "  52.0,\n",
       "  82.0,\n",
       "  49.0,\n",
       "  63.0,\n",
       "  68.0],\n",
       " 'GO:0002934-desmosome organization': 37.0,\n",
       " 'GO:0033700-phospholipid efflux': [38.0, 11.0, 45.0, 16.0, 8.0, 40.0, 39.0],\n",
       " 'GO:0034379-very-low-density lipoprotein particle assembly': [39.0,\n",
       "  35.0,\n",
       "  32.0,\n",
       "  31.0,\n",
       "  28.0,\n",
       "  13.0,\n",
       "  28.0,\n",
       "  16.0,\n",
       "  89.0],\n",
       " 'GO:1903351-cellular response to dopamine': [40.0, 182.0],\n",
       " 'GO:0051673-membrane disruption in other organism': [41.0,\n",
       "  42.0,\n",
       "  20.0,\n",
       "  36.0,\n",
       "  13.0,\n",
       "  45.0],\n",
       " 'GO:0090576-RNA polymerase III transcription factor complex': [42.0, 34.0],\n",
       " 'GO:0048096-chromatin-mediated maintenance of transcription': 43.0,\n",
       " 'GO:2001135-regulation of endocytic recycling': 44.0,\n",
       " 'GO:0045943-positive regulation of transcription by RNA polymerase I': [45.0,\n",
       "  41.0,\n",
       "  198.0],\n",
       " 'GO:0045176-apical protein localization': 46.0,\n",
       " 'GO:0035092-sperm chromatin condensation': 47.0,\n",
       " 'GO:0031078-histone deacetylase activity (H3-K14 specific)': [48.5, 43.5],\n",
       " 'GO:0032041-NAD-dependent histone deacetylase activity (H3-K14 specific)': [48.5,\n",
       "  43.5],\n",
       " 'GO:0005736-RNA polymerase I complex': [50.0, 45.0],\n",
       " 'GO:0016176-superoxide-generating NADPH oxidase activator activity': 51.0,\n",
       " 'GO:0016725-oxidoreductase activity, acting on CH or CH2 groups': [52.0,\n",
       "  41.0,\n",
       "  24.0,\n",
       "  29.0,\n",
       "  57.0,\n",
       "  1.0,\n",
       "  131.0,\n",
       "  50.0],\n",
       " 'GO:0034372-very-low-density lipoprotein particle remodeling': [53.0,\n",
       "  58.0,\n",
       "  59.0,\n",
       "  54.0,\n",
       "  40.0,\n",
       "  72.0,\n",
       "  49.0],\n",
       " 'GO:0007076-mitotic chromosome condensation': [54.0, 76.0, 87.0, 46.0, 115.0],\n",
       " 'GO:1904814-regulation of protein localization to chromosome, telomeric region': [55.0,\n",
       "  72.0,\n",
       "  31.0],\n",
       " 'GO:1903350-response to dopamine': 56.0,\n",
       " \"GO:0000395-mRNA 5'-splice site recognition\": [57.0,\n",
       "  129.0,\n",
       "  149.0,\n",
       "  363.0,\n",
       "  161.0,\n",
       "  209.0,\n",
       "  193.0,\n",
       "  19.0,\n",
       "  146.0,\n",
       "  124.0,\n",
       "  70.0,\n",
       "  267.0],\n",
       " 'GO:0045653-negative regulation of megakaryocyte differentiation': [58.0,\n",
       "  126.0,\n",
       "  50.0,\n",
       "  99.0,\n",
       "  51.0,\n",
       "  40.0,\n",
       "  109.0,\n",
       "  118.0,\n",
       "  23.0,\n",
       "  68.0,\n",
       "  65.0,\n",
       "  99.0,\n",
       "  169.0,\n",
       "  196.0,\n",
       "  123.0],\n",
       " 'GO:0018206-peptidyl-methionine modification': 59.0,\n",
       " 'GO:0001054-RNA polymerase I activity': [60.0, 43.0],\n",
       " 'GO:0005687-U4 snRNP': [61.0, 94.0, 59.0, 40.0],\n",
       " 'GO:0033860-regulation of NAD(P)H oxidase activity': 62.0,\n",
       " 'GO:1902713-regulation of interferon-gamma secretion': 63.0,\n",
       " 'GO:0033623-regulation of integrin activation': [64.0, 67.0, 26.0],\n",
       " 'GO:0034385-triglyceride-rich plasma lipoprotein particle': [65.5,\n",
       "  32.5,\n",
       "  59.5,\n",
       "  48.5,\n",
       "  127.5,\n",
       "  81.5,\n",
       "  85.5,\n",
       "  75.5,\n",
       "  19.5,\n",
       "  61.5,\n",
       "  56.5,\n",
       "  27.5,\n",
       "  65.5,\n",
       "  74.5,\n",
       "  112.5,\n",
       "  94.5,\n",
       "  207.5,\n",
       "  79.5],\n",
       " 'GO:0034361-very-low-density lipoprotein particle': [65.5,\n",
       "  32.5,\n",
       "  59.5,\n",
       "  48.5,\n",
       "  127.5,\n",
       "  81.5,\n",
       "  85.5,\n",
       "  75.5,\n",
       "  19.5,\n",
       "  61.5,\n",
       "  56.5,\n",
       "  27.5,\n",
       "  65.5,\n",
       "  74.5,\n",
       "  112.5,\n",
       "  94.5,\n",
       "  207.5,\n",
       "  79.5],\n",
       " 'GO:0000940-condensed chromosome outer kinetochore': [67.0,\n",
       "  6.0,\n",
       "  51.0,\n",
       "  36.0,\n",
       "  54.0,\n",
       "  65.0,\n",
       "  115.0,\n",
       "  44.0,\n",
       "  59.0,\n",
       "  35.0,\n",
       "  13.0],\n",
       " 'GO:0006068-ethanol catabolic process': [68.0, 50.0, 102.0, 75.0, 64.0],\n",
       " 'GO:0001533-cornified envelope': [69.0,\n",
       "  55.0,\n",
       "  98.0,\n",
       "  420.0,\n",
       "  120.0,\n",
       "  60.0,\n",
       "  30.0,\n",
       "  228.0,\n",
       "  15.0,\n",
       "  189.0,\n",
       "  11.0,\n",
       "  68.0,\n",
       "  256.0,\n",
       "  87.0,\n",
       "  20.0,\n",
       "  72.0,\n",
       "  32.0,\n",
       "  43.0,\n",
       "  1185.0],\n",
       " 'GO:0002716-negative regulation of natural killer cell mediated immunity': [70.0,\n",
       "  95.0,\n",
       "  178.0,\n",
       "  12.0],\n",
       " 'GO:0070628-proteasome binding': [71.0,\n",
       "  109.0,\n",
       "  102.0,\n",
       "  248.0,\n",
       "  36.0,\n",
       "  49.0,\n",
       "  84.0],\n",
       " 'GO:0018149-peptide cross-linking': [72.0,\n",
       "  128.0,\n",
       "  85.0,\n",
       "  381.0,\n",
       "  102.0,\n",
       "  57.0,\n",
       "  159.0,\n",
       "  260.0,\n",
       "  24.0,\n",
       "  155.0,\n",
       "  9.0,\n",
       "  55.0,\n",
       "  260.0,\n",
       "  91.0,\n",
       "  45.0,\n",
       "  100.0,\n",
       "  62.0,\n",
       "  125.0,\n",
       "  1003.0],\n",
       " 'GO:1900044-regulation of protein K63-linked ubiquitination': 73.0,\n",
       " 'GO:0005682-U5 snRNP': [74.0,\n",
       "  104.0,\n",
       "  67.0,\n",
       "  142.0,\n",
       "  138.0,\n",
       "  136.0,\n",
       "  165.0,\n",
       "  52.0,\n",
       "  116.0,\n",
       "  139.0,\n",
       "  42.0,\n",
       "  114.0,\n",
       "  161.0],\n",
       " 'GO:0010369-chromocenter': [75.0, 47.0, 18.0],\n",
       " 'GO:0090545-CHD-type complex': [76.5, 88.5],\n",
       " 'GO:0016581-NuRD complex': [76.5, 88.5],\n",
       " 'GO:0045953-negative regulation of natural killer cell mediated cytotoxicity': [78.0,\n",
       "  65.0,\n",
       "  194.0,\n",
       "  14.0],\n",
       " 'GO:0101031-chaperone complex': [79.0,\n",
       "  180.0,\n",
       "  73.0,\n",
       "  322.0,\n",
       "  90.0,\n",
       "  145.0,\n",
       "  163.0],\n",
       " 'GO:0072178-nephric duct morphogenesis': 80.0,\n",
       " 'GO:0070601-centromeric sister chromatid cohesion': [81.0, 65.0],\n",
       " 'GO:0031498-chromatin disassembly': [82.0, 58.0, 110.0, 274.0],\n",
       " 'GO:1904872-regulation of telomerase RNA localization to Cajal body': [83.0,\n",
       "  14.0,\n",
       "  99.0,\n",
       "  29.0],\n",
       " 'GO:0005665-RNA polymerase II, core complex': [84.0,\n",
       "  101.0,\n",
       "  128.0,\n",
       "  67.0,\n",
       "  125.0,\n",
       "  76.0,\n",
       "  68.0],\n",
       " 'GO:0031011-Ino80 complex': 85.5,\n",
       " 'GO:0033202-DNA helicase complex': 85.5,\n",
       " 'GO:1903589-positive regulation of blood vessel endothelial cell proliferation involved in sprouting angiogenesis': [87.0,\n",
       "  115.0,\n",
       "  9.0,\n",
       "  30.0,\n",
       "  37.0,\n",
       "  105.0,\n",
       "  165.0,\n",
       "  150.0,\n",
       "  49.0,\n",
       "  12.0,\n",
       "  31.0,\n",
       "  39.0,\n",
       "  93.0,\n",
       "  51.0,\n",
       "  34.0,\n",
       "  99.0,\n",
       "  35.0],\n",
       " 'GO:0002551-mast cell chemotaxis': [88.5, 34.5, 41.5],\n",
       " 'GO:0097531-mast cell migration': [88.5, 34.5, 41.5],\n",
       " 'GO:0046325-negative regulation of glucose import': [90.0, 83.0, 96.0],\n",
       " 'GO:0006241-CTP biosynthetic process': 91.5,\n",
       " 'GO:0046036-CTP metabolic process': 91.5,\n",
       " 'GO:0000054-ribosomal subunit export from nucleus': [93.5, 109.5, 147.5],\n",
       " 'GO:0033750-ribosome localization': [93.5, 109.5, 147.5],\n",
       " 'GO:0048308-organelle inheritance': [95.5, 192.5],\n",
       " 'GO:0048313-Golgi inheritance': [95.5, 192.5],\n",
       " 'GO:0005688-U6 snRNP': [97.0, 29.0, 106.0, 29.0, 9.0],\n",
       " 'GO:0006337-nucleosome disassembly': [98.0, 62.0, 113.0, 286.0],\n",
       " 'GO:0090672-telomerase RNA localization': [100.5, 18.5, 131.5, 37.5],\n",
       " 'GO:0090671-telomerase RNA localization to Cajal body': [100.5,\n",
       "  18.5,\n",
       "  131.5,\n",
       "  37.5],\n",
       " 'GO:0090670-RNA localization to Cajal body': [100.5, 18.5, 131.5, 37.5],\n",
       " 'GO:0090685-RNA localization to nucleus': [100.5, 18.5, 131.5, 37.5],\n",
       " 'GO:0097526-spliceosomal tri-snRNP complex': [103.5,\n",
       "  141.5,\n",
       "  99.5,\n",
       "  249.5,\n",
       "  253.5,\n",
       "  230.5,\n",
       "  120.5,\n",
       "  190.5,\n",
       "  235.5,\n",
       "  406.5,\n",
       "  228.5,\n",
       "  41.5,\n",
       "  60.5,\n",
       "  122.5,\n",
       "  143.5,\n",
       "  409.5,\n",
       "  47.5,\n",
       "  827.5,\n",
       "  182.5,\n",
       "  134.5],\n",
       " 'GO:0046540-U4/U6 x U5 tri-snRNP complex': [103.5,\n",
       "  141.5,\n",
       "  99.5,\n",
       "  249.5,\n",
       "  253.5,\n",
       "  230.5,\n",
       "  120.5,\n",
       "  190.5,\n",
       "  235.5,\n",
       "  406.5,\n",
       "  228.5,\n",
       "  41.5,\n",
       "  60.5,\n",
       "  122.5,\n",
       "  143.5,\n",
       "  409.5,\n",
       "  47.5,\n",
       "  827.5,\n",
       "  182.5,\n",
       "  134.5],\n",
       " 'GO:0002227-innate immune response in mucosa': [105.0,\n",
       "  127.0,\n",
       "  80.0,\n",
       "  136.0,\n",
       "  192.0,\n",
       "  261.0,\n",
       "  98.0,\n",
       "  200.0,\n",
       "  61.0,\n",
       "  32.0,\n",
       "  51.0,\n",
       "  92.0,\n",
       "  50.0,\n",
       "  41.0,\n",
       "  319.0,\n",
       "  367.0,\n",
       "  107.0],\n",
       " 'GO:0034364-high-density lipoprotein particle': [106.0,\n",
       "  22.0,\n",
       "  21.0,\n",
       "  76.0,\n",
       "  90.0,\n",
       "  123.0,\n",
       "  68.0,\n",
       "  53.0,\n",
       "  143.0,\n",
       "  37.0,\n",
       "  36.0,\n",
       "  88.0,\n",
       "  124.0,\n",
       "  18.0,\n",
       "  64.0,\n",
       "  112.0,\n",
       "  41.0,\n",
       "  53.0,\n",
       "  204.0,\n",
       "  196.0,\n",
       "  115.0],\n",
       " 'GO:0090179-planar cell polarity pathway involved in neural tube closure': [107.0,\n",
       "  28.0],\n",
       " 'GO:0030687-preribosome, large subunit precursor': [108.0,\n",
       "  151.0,\n",
       "  202.0,\n",
       "  45.0,\n",
       "  158.0,\n",
       "  168.0,\n",
       "  37.0],\n",
       " 'GO:0071428-rRNA-containing ribonucleoprotein complex export from nucleus': [109.0,\n",
       "  75.0,\n",
       "  150.0,\n",
       "  205.0],\n",
       " 'GO:0009812-flavonoid metabolic process': [110.0,\n",
       "  72.0,\n",
       "  10.0,\n",
       "  73.0,\n",
       "  87.0,\n",
       "  10.0,\n",
       "  143.0,\n",
       "  20.0,\n",
       "  14.0,\n",
       "  53.0,\n",
       "  5.0,\n",
       "  163.0,\n",
       "  2.0,\n",
       "  21.0],\n",
       " 'GO:0030280-structural constituent of epidermis': [111.0,\n",
       "  5.0,\n",
       "  68.0,\n",
       "  22.0,\n",
       "  13.0,\n",
       "  10.0,\n",
       "  15.0,\n",
       "  17.0,\n",
       "  16.0],\n",
       " 'GO:0006271-DNA strand elongation involved in DNA replication': [112.0,\n",
       "  74.0,\n",
       "  148.0,\n",
       "  53.0,\n",
       "  108.0,\n",
       "  99.0,\n",
       "  53.0,\n",
       "  164.0],\n",
       " 'GO:0070199-establishment of protein localization to chromosome': [113.0,\n",
       "  37.0,\n",
       "  113.0,\n",
       "  182.0],\n",
       " 'GO:0010985-negative regulation of lipoprotein particle clearance': [114.0,\n",
       "  57.0,\n",
       "  34.0,\n",
       "  35.0,\n",
       "  108.0,\n",
       "  62.0,\n",
       "  55.0,\n",
       "  58.0,\n",
       "  27.0,\n",
       "  51.0,\n",
       "  64.0,\n",
       "  18.0,\n",
       "  74.0,\n",
       "  40.0,\n",
       "  63.0],\n",
       " 'GO:1903587-regulation of blood vessel endothelial cell proliferation involved in sprouting angiogenesis': [115.0,\n",
       "  140.0,\n",
       "  22.0,\n",
       "  31.0,\n",
       "  47.0,\n",
       "  135.0,\n",
       "  188.0,\n",
       "  373.0,\n",
       "  60.0,\n",
       "  171.0,\n",
       "  16.0,\n",
       "  56.0,\n",
       "  10.0,\n",
       "  58.0,\n",
       "  132.0,\n",
       "  43.0,\n",
       "  63.0,\n",
       "  166.0,\n",
       "  149.0,\n",
       "  46.0,\n",
       "  148.0,\n",
       "  18.0,\n",
       "  39.0,\n",
       "  125.0,\n",
       "  27.0],\n",
       " 'GO:0045120-pronucleus': [116.0, 56.0, 59.0],\n",
       " 'GO:0017070-U6 snRNA binding': [117.0, 127.0],\n",
       " 'GO:0017136-NAD-dependent histone deacetylase activity': 118.0,\n",
       " 'GO:0034367-protein-containing complex remodeling': [119.0,\n",
       "  222.0,\n",
       "  282.0,\n",
       "  204.0,\n",
       "  96.0,\n",
       "  51.0,\n",
       "  118.0,\n",
       "  180.0,\n",
       "  121.0,\n",
       "  241.0],\n",
       " 'GO:1903726-negative regulation of phospholipid metabolic process': [120.0,\n",
       "  78.0,\n",
       "  59.0,\n",
       "  159.0],\n",
       " 'GO:2000095-regulation of Wnt signaling pathway, planar cell polarity pathway': 121.0,\n",
       " 'GO:0050711-negative regulation of interleukin-1 secretion': [122.0,\n",
       "  92.0,\n",
       "  203.0],\n",
       " 'GO:1903960-negative regulation of anion transmembrane transport': [123.0,\n",
       "  66.0,\n",
       "  41.0],\n",
       " 'GO:0005732-small nucleolar ribonucleoprotein complex': [124.0,\n",
       "  135.0,\n",
       "  186.0,\n",
       "  203.0,\n",
       "  158.0,\n",
       "  100.0,\n",
       "  112.0,\n",
       "  229.0,\n",
       "  215.0,\n",
       "  530.0],\n",
       " 'GO:0034375-high-density lipoprotein particle remodeling': [125.0,\n",
       "  75.0,\n",
       "  109.0,\n",
       "  126.0,\n",
       "  276.0,\n",
       "  29.0,\n",
       "  111.0,\n",
       "  73.0,\n",
       "  175.0],\n",
       " 'GO:2000345-regulation of hepatocyte proliferation': [126.0, 41.0, 40.0],\n",
       " 'GO:0032986-protein-DNA complex disassembly': [127.0, 79.0, 273.0],\n",
       " 'GO:0034368-protein-lipid complex remodeling': [128.5,\n",
       "  212.5,\n",
       "  291.5,\n",
       "  189.5,\n",
       "  224.5,\n",
       "  101.5,\n",
       "  206.5,\n",
       "  59.5,\n",
       "  85.5,\n",
       "  139.5,\n",
       "  180.5,\n",
       "  132.5,\n",
       "  393.5,\n",
       "  252.5],\n",
       " 'GO:0034369-plasma lipoprotein particle remodeling': [128.5,\n",
       "  212.5,\n",
       "  291.5,\n",
       "  189.5,\n",
       "  224.5,\n",
       "  101.5,\n",
       "  206.5,\n",
       "  59.5,\n",
       "  85.5,\n",
       "  139.5,\n",
       "  180.5,\n",
       "  132.5,\n",
       "  393.5,\n",
       "  252.5],\n",
       " 'GO:0050832-defense response to fungus': [130.0,\n",
       "  87.0,\n",
       "  55.0,\n",
       "  510.0,\n",
       "  163.0,\n",
       "  233.0,\n",
       "  135.0,\n",
       "  82.0,\n",
       "  240.0,\n",
       "  51.0,\n",
       "  107.0,\n",
       "  265.0,\n",
       "  255.0,\n",
       "  89.0,\n",
       "  173.0,\n",
       "  136.0,\n",
       "  509.0],\n",
       " 'GO:0099010-modification of postsynaptic structure': [131.0, 54.0, 113.0],\n",
       " 'GO:0000470-maturation of LSU-rRNA': [132.0,\n",
       "  193.0,\n",
       "  176.0,\n",
       "  168.0,\n",
       "  109.0,\n",
       "  44.0,\n",
       "  65.0],\n",
       " 'GO:1903588-negative regulation of blood vessel endothelial cell proliferation involved in sprouting angiogenesis': [133.0,\n",
       "  129.0,\n",
       "  15.0,\n",
       "  13.0,\n",
       "  111.0,\n",
       "  21.0,\n",
       "  77.0,\n",
       "  1.0,\n",
       "  18.0,\n",
       "  1.0,\n",
       "  23.0,\n",
       "  30.0,\n",
       "  12.0,\n",
       "  125.0,\n",
       "  111.0,\n",
       "  27.0,\n",
       "  14.0,\n",
       "  3.0,\n",
       "  103.0,\n",
       "  10.0],\n",
       " 'GO:0097346-INO80-type complex': [134.0, 171.0, 193.0],\n",
       " 'GO:0005685-U1 snRNP': [135.0,\n",
       "  217.0,\n",
       "  236.0,\n",
       "  141.0,\n",
       "  153.0,\n",
       "  318.0,\n",
       "  107.0,\n",
       "  151.0,\n",
       "  295.0,\n",
       "  116.0,\n",
       "  48.0,\n",
       "  78.0,\n",
       "  202.0,\n",
       "  195.0,\n",
       "  202.0,\n",
       "  133.0,\n",
       "  630.0],\n",
       " 'GO:0010984-regulation of lipoprotein particle clearance': [136.0,\n",
       "  240.0,\n",
       "  87.0,\n",
       "  94.0,\n",
       "  306.0,\n",
       "  334.0,\n",
       "  106.0,\n",
       "  201.0,\n",
       "  153.0,\n",
       "  59.0,\n",
       "  128.0,\n",
       "  176.0,\n",
       "  49.0,\n",
       "  129.0,\n",
       "  87.0,\n",
       "  129.0,\n",
       "  149.0,\n",
       "  239.0,\n",
       "  261.0,\n",
       "  94.0],\n",
       " 'GO:0090312-positive regulation of protein deacetylation': [137.0,\n",
       "  125.0,\n",
       "  158.0],\n",
       " 'GO:0008392-arachidonic acid epoxygenase activity': [138.0,\n",
       "  105.0,\n",
       "  40.0,\n",
       "  80.0,\n",
       "  59.0],\n",
       " 'GO:0046933-proton-transporting ATP synthase activity, rotational mechanism': [139.0,\n",
       "  71.0,\n",
       "  11.0,\n",
       "  15.0,\n",
       "  44.0,\n",
       "  78.0,\n",
       "  93.0],\n",
       " 'GO:0046597-negative regulation of viral entry into host cell': [140.0,\n",
       "  174.0,\n",
       "  130.0],\n",
       " 'GO:1902187-negative regulation of viral release from host cell': 141.0,\n",
       " 'GO:0099558-maintenance of synapse structure': 142.0,\n",
       " 'GO:0009209-pyrimidine ribonucleoside triphosphate biosynthetic process': 143.0,\n",
       " 'GO:0010988-regulation of low-density lipoprotein particle clearance': [144.0,\n",
       "  89.0,\n",
       "  141.0,\n",
       "  86.0,\n",
       "  191.0,\n",
       "  71.0,\n",
       "  145.0,\n",
       "  81.0],\n",
       " 'GO:0008106-alcohol dehydrogenase (NADP+) activity': [145.0,\n",
       "  96.0,\n",
       "  137.0,\n",
       "  133.0,\n",
       "  168.0,\n",
       "  370.0,\n",
       "  59.0,\n",
       "  65.0,\n",
       "  26.0,\n",
       "  35.0,\n",
       "  183.0,\n",
       "  211.0,\n",
       "  105.0],\n",
       " 'GO:0090741-pigment granule membrane': [147.0, 42.0],\n",
       " 'GO:0045009-chitosome': [147.0, 42.0],\n",
       " 'GO:0033162-melanosome membrane': [147.0, 42.0],\n",
       " 'GO:0060192-negative regulation of lipase activity': [149.0, 90.0, 228.0],\n",
       " 'GO:1903231-mRNA binding involved in posttranscriptional gene silencing': [150.0,\n",
       "  83.0,\n",
       "  17.0,\n",
       "  21.0,\n",
       "  2.0,\n",
       "  142.0,\n",
       "  160.0,\n",
       "  599.0,\n",
       "  9.0,\n",
       "  76.0,\n",
       "  7.0,\n",
       "  12.0,\n",
       "  16.0,\n",
       "  30.0,\n",
       "  26.0,\n",
       "  58.0,\n",
       "  15.0,\n",
       "  71.0,\n",
       "  146.0,\n",
       "  11.0,\n",
       "  77.0,\n",
       "  7.0,\n",
       "  17.0,\n",
       "  172.0,\n",
       "  3.0],\n",
       " 'GO:0043496-regulation of protein homodimerization activity': [151.0, 96.0],\n",
       " 'GO:0000244-spliceosomal tri-snRNP complex assembly': [152.0,\n",
       "  83.0,\n",
       "  137.0,\n",
       "  340.0,\n",
       "  103.0,\n",
       "  389.0,\n",
       "  67.0,\n",
       "  154.0,\n",
       "  194.0,\n",
       "  80.0,\n",
       "  119.0,\n",
       "  75.0,\n",
       "  313.0,\n",
       "  53.0],\n",
       " 'GO:0034979-NAD-dependent protein deacetylase activity': 153.0,\n",
       " 'GO:0015037-peptide disulfide oxidoreductase activity': [154.0,\n",
       "  97.0,\n",
       "  97.0,\n",
       "  90.0],\n",
       " 'GO:0004033-aldo-keto reductase (NADP) activity': [155.0,\n",
       "  92.0,\n",
       "  31.0,\n",
       "  85.0,\n",
       "  198.0,\n",
       "  165.0,\n",
       "  113.0,\n",
       "  221.0,\n",
       "  25.0,\n",
       "  50.0,\n",
       "  77.0,\n",
       "  22.0,\n",
       "  34.0,\n",
       "  338.0,\n",
       "  186.0,\n",
       "  109.0],\n",
       " 'GO:0072575-epithelial cell proliferation involved in liver morphogenesis': [156.5,\n",
       "  126.5,\n",
       "  242.5,\n",
       "  20.5,\n",
       "  18.5,\n",
       "  171.5,\n",
       "  110.5],\n",
       " 'GO:0072574-hepatocyte proliferation': [156.5,\n",
       "  126.5,\n",
       "  242.5,\n",
       "  20.5,\n",
       "  18.5,\n",
       "  171.5,\n",
       "  110.5],\n",
       " 'GO:0033599-regulation of mammary gland epithelial cell proliferation': [158.0,\n",
       "  105.0,\n",
       "  89.0],\n",
       " 'GO:0034377-plasma lipoprotein particle assembly': [159.0,\n",
       "  131.0,\n",
       "  89.0,\n",
       "  111.0,\n",
       "  182.0,\n",
       "  115.0,\n",
       "  92.0,\n",
       "  39.0,\n",
       "  133.0,\n",
       "  299.0,\n",
       "  130.0,\n",
       "  84.0,\n",
       "  161.0,\n",
       "  180.0,\n",
       "  363.0,\n",
       "  265.0,\n",
       "  185.0],\n",
       " 'GO:0000930-gamma-tubulin complex': [160.0, 147.0],\n",
       " 'GO:0009303-rRNA transcription': [161.0, 266.0, 218.0, 242.0, 129.0],\n",
       " 'GO:0031065-positive regulation of histone deacetylation': [162.0, 79.0],\n",
       " 'GO:0090177-establishment of planar polarity involved in neural tube closure': 163.0,\n",
       " 'GO:1905563-negative regulation of vascular endothelial cell proliferation': [164.0,\n",
       "  109.0,\n",
       "  15.0,\n",
       "  12.0,\n",
       "  20.0,\n",
       "  104.0,\n",
       "  116.0,\n",
       "  2.0,\n",
       "  123.0,\n",
       "  4.0,\n",
       "  25.0,\n",
       "  2.0,\n",
       "  14.0,\n",
       "  53.0,\n",
       "  16.0,\n",
       "  96.0,\n",
       "  106.0,\n",
       "  15.0,\n",
       "  83.0,\n",
       "  4.0,\n",
       "  14.0,\n",
       "  112.0,\n",
       "  14.0],\n",
       " 'GO:0046112-nucleobase biosynthetic process': [165.0, 237.0, 103.0, 78.0],\n",
       " 'GO:1903358-regulation of Golgi organization': 166.0,\n",
       " 'GO:0035635-entry of bacterium into host cell': [167.0, 68.0],\n",
       " 'GO:0042629-mast cell granule': [168.0, 185.0],\n",
       " 'GO:0097421-liver regeneration': [169.0,\n",
       "  172.0,\n",
       "  333.0,\n",
       "  331.0,\n",
       "  259.0,\n",
       "  199.0,\n",
       "  259.0,\n",
       "  297.0],\n",
       " 'GO:0032692-negative regulation of interleukin-1 production': [170.0,\n",
       "  254.0,\n",
       "  23.0,\n",
       "  323.0,\n",
       "  148.0,\n",
       "  473.0,\n",
       "  307.0,\n",
       "  206.0,\n",
       "  631.0,\n",
       "  193.0,\n",
       "  126.0,\n",
       "  218.0,\n",
       "  303.0,\n",
       "  334.0,\n",
       "  197.0],\n",
       " 'GO:0010586-miRNA metabolic process': [171.0, 367.0],\n",
       " 'GO:0006335-DNA replication-dependent nucleosome assembly': [172.5,\n",
       "  199.5,\n",
       "  74.5,\n",
       "  223.5,\n",
       "  312.5,\n",
       "  93.5,\n",
       "  90.5,\n",
       "  308.5,\n",
       "  582.5,\n",
       "  38.5,\n",
       "  125.5,\n",
       "  102.5,\n",
       "  87.5,\n",
       "  382.5,\n",
       "  121.5,\n",
       "  548.5,\n",
       "  196.5],\n",
       " 'GO:0034723-DNA replication-dependent nucleosome organization': [172.5,\n",
       "  199.5,\n",
       "  74.5,\n",
       "  223.5,\n",
       "  312.5,\n",
       "  93.5,\n",
       "  90.5,\n",
       "  308.5,\n",
       "  582.5,\n",
       "  38.5,\n",
       "  125.5,\n",
       "  102.5,\n",
       "  87.5,\n",
       "  382.5,\n",
       "  121.5,\n",
       "  548.5,\n",
       "  196.5],\n",
       " 'GO:0072576-liver morphogenesis': [174.0, 252.0, 25.0, 48.0, 231.0],\n",
       " 'GO:0042788-polysomal ribosome': [175.0,\n",
       "  202.0,\n",
       "  223.0,\n",
       "  326.0,\n",
       "  114.0,\n",
       "  265.0,\n",
       "  665.0,\n",
       "  168.0,\n",
       "  272.0,\n",
       "  665.0],\n",
       " 'GO:0031362-anchored component of external side of plasma membrane': 176.0,\n",
       " 'GO:0032691-negative regulation of interleukin-1 beta production': [177.0,\n",
       "  299.0,\n",
       "  20.0,\n",
       "  243.0,\n",
       "  150.0,\n",
       "  269.0,\n",
       "  259.0,\n",
       "  205.0,\n",
       "  303.0,\n",
       "  144.0,\n",
       "  113.0,\n",
       "  188.0,\n",
       "  309.0,\n",
       "  277.0,\n",
       "  142.0],\n",
       " 'GO:0044406-adhesion of symbiont to host': [178.0, 20.0],\n",
       " 'GO:0031332-RNAi effector complex': [179.5,\n",
       "  117.5,\n",
       "  18.5,\n",
       "  26.5,\n",
       "  10.5,\n",
       "  187.5,\n",
       "  215.5,\n",
       "  782.5,\n",
       "  10.5,\n",
       "  99.5,\n",
       "  14.5,\n",
       "  16.5,\n",
       "  19.5,\n",
       "  65.5,\n",
       "  32.5,\n",
       "  73.5,\n",
       "  27.5,\n",
       "  100.5,\n",
       "  178.5,\n",
       "  19.5,\n",
       "  88.5,\n",
       "  9.5,\n",
       "  35.5,\n",
       "  230.5,\n",
       "  7.5],\n",
       " 'GO:0016442-RISC complex': [179.5,\n",
       "  117.5,\n",
       "  18.5,\n",
       "  26.5,\n",
       "  10.5,\n",
       "  187.5,\n",
       "  215.5,\n",
       "  782.5,\n",
       "  10.5,\n",
       "  99.5,\n",
       "  14.5,\n",
       "  16.5,\n",
       "  19.5,\n",
       "  65.5,\n",
       "  32.5,\n",
       "  73.5,\n",
       "  27.5,\n",
       "  100.5,\n",
       "  178.5,\n",
       "  19.5,\n",
       "  88.5,\n",
       "  9.5,\n",
       "  35.5,\n",
       "  230.5,\n",
       "  7.5],\n",
       " 'GO:0062033-positive regulation of mitotic sister chromatid segregation': [181.0,\n",
       "  80.0],\n",
       " 'GO:2000291-regulation of myoblast proliferation': [182.0,\n",
       "  57.0,\n",
       "  65.0,\n",
       "  139.0,\n",
       "  140.0,\n",
       "  72.0,\n",
       "  246.0,\n",
       "  54.0],\n",
       " 'GO:0005753-mitochondrial proton-transporting ATP synthase complex': [183.0,\n",
       "  137.0,\n",
       "  127.0,\n",
       "  89.0,\n",
       "  13.0,\n",
       "  120.0,\n",
       "  20.0,\n",
       "  24.0,\n",
       "  82.0,\n",
       "  81.0,\n",
       "  210.0,\n",
       "  107.0,\n",
       "  51.0],\n",
       " 'GO:0032689-negative regulation of interferon-gamma production': [184.0,\n",
       "  272.0,\n",
       "  455.0,\n",
       "  363.0,\n",
       "  303.0,\n",
       "  141.0,\n",
       "  392.0,\n",
       "  743.0],\n",
       " 'GO:0032769-negative regulation of monooxygenase activity': [185.0,\n",
       "  143.0,\n",
       "  16.0,\n",
       "  45.0,\n",
       "  126.0,\n",
       "  134.0,\n",
       "  177.0,\n",
       "  76.0,\n",
       "  147.0,\n",
       "  129.0,\n",
       "  83.0,\n",
       "  129.0,\n",
       "  63.0,\n",
       "  58.0,\n",
       "  46.0,\n",
       "  106.0,\n",
       "  44.0,\n",
       "  60.0],\n",
       " 'GO:0002043-blood vessel endothelial cell proliferation involved in sprouting angiogenesis': [186.0,\n",
       "  152.0,\n",
       "  33.0,\n",
       "  37.0,\n",
       "  85.0,\n",
       "  184.0,\n",
       "  221.0,\n",
       "  510.0,\n",
       "  75.0,\n",
       "  324.0,\n",
       "  38.0,\n",
       "  90.0,\n",
       "  24.0,\n",
       "  91.0,\n",
       "  152.0,\n",
       "  68.0,\n",
       "  92.0,\n",
       "  262.0,\n",
       "  239.0,\n",
       "  141.0,\n",
       "  254.0,\n",
       "  25.0,\n",
       "  55.0,\n",
       "  147.0,\n",
       "  67.0],\n",
       " 'GO:0019731-antibacterial humoral response': [187.0,\n",
       "  82.0,\n",
       "  52.0,\n",
       "  238.0,\n",
       "  161.0,\n",
       "  172.0,\n",
       "  408.0,\n",
       "  348.0,\n",
       "  97.0,\n",
       "  417.0,\n",
       "  79.0,\n",
       "  46.0,\n",
       "  94.0,\n",
       "  80.0,\n",
       "  208.0,\n",
       "  110.0,\n",
       "  240.0,\n",
       "  261.0,\n",
       "  368.0,\n",
       "  267.0],\n",
       " 'GO:0006474-N-terminal protein amino acid acetylation': 188.0,\n",
       " 'GO:0010314-phosphatidylinositol-5-phosphate binding': 189.0,\n",
       " 'GO:0065005-protein-lipid complex assembly': [190.0,\n",
       "  182.0,\n",
       "  175.0,\n",
       "  139.0,\n",
       "  268.0,\n",
       "  236.0,\n",
       "  144.0,\n",
       "  227.0,\n",
       "  81.0,\n",
       "  153.0,\n",
       "  279.0,\n",
       "  147.0,\n",
       "  142.0,\n",
       "  223.0,\n",
       "  316.0,\n",
       "  377.0],\n",
       " 'GO:0001972-retinoic acid binding': [191.0, 164.0, 151.0, 52.0, 72.0, 102.0],\n",
       " 'GO:0006895-Golgi to endosome transport': 192.0,\n",
       " 'GO:0004806-triglyceride lipase activity': 193.0,\n",
       " 'GO:0033962-cytoplasmic mRNA processing body assembly': 194.0,\n",
       " 'GO:0000788-nuclear nucleosome': [195.0,\n",
       "  219.0,\n",
       "  225.0,\n",
       "  80.0,\n",
       "  154.0,\n",
       "  999.0,\n",
       "  37.0,\n",
       "  163.0,\n",
       "  260.0,\n",
       "  114.0,\n",
       "  206.0,\n",
       "  674.0,\n",
       "  434.0,\n",
       "  321.0],\n",
       " 'GO:0006544-glycine metabolic process': [196.0, 80.0],\n",
       " 'GO:0071459-protein localization to chromosome, centromeric region': [197.0,\n",
       "  62.0],\n",
       " 'GO:0060148-positive regulation of posttranscriptional gene silencing': [198.0,\n",
       "  173.0,\n",
       "  562.0,\n",
       "  143.0,\n",
       "  119.0,\n",
       "  195.0],\n",
       " 'GO:0038191-neuropilin binding': 199.0,\n",
       " 'GO:0015986-ATP synthesis coupled proton transport': [200.5,\n",
       "  154.5,\n",
       "  217.5,\n",
       "  276.5,\n",
       "  24.5,\n",
       "  159.5,\n",
       "  254.5,\n",
       "  73.5,\n",
       "  47.5,\n",
       "  95.5,\n",
       "  212.5,\n",
       "  68.5,\n",
       "  257.5,\n",
       "  138.5],\n",
       " 'GO:0015985-energy coupled proton transport, down electrochemical gradient': [200.5,\n",
       "  154.5,\n",
       "  217.5,\n",
       "  276.5,\n",
       "  24.5,\n",
       "  159.5,\n",
       "  254.5,\n",
       "  73.5,\n",
       "  47.5,\n",
       "  95.5,\n",
       "  212.5,\n",
       "  68.5,\n",
       "  257.5,\n",
       "  138.5],\n",
       " 'GO:0042307-positive regulation of protein import into nucleus': [202.0,\n",
       "  380.0,\n",
       "  444.0,\n",
       "  343.0,\n",
       "  371.0,\n",
       "  274.0,\n",
       "  307.0,\n",
       "  261.0,\n",
       "  553.0,\n",
       "  245.0,\n",
       "  224.0,\n",
       "  308.0,\n",
       "  357.0,\n",
       "  582.0,\n",
       "  209.0,\n",
       "  731.0,\n",
       "  418.0,\n",
       "  331.0],\n",
       " 'GO:0030261-chromosome condensation': [203.0,\n",
       "  306.0,\n",
       "  288.0,\n",
       "  499.0,\n",
       "  157.0,\n",
       "  885.0,\n",
       "  408.0,\n",
       "  288.0,\n",
       "  1198.0,\n",
       "  143.0,\n",
       "  279.0,\n",
       "  320.0,\n",
       "  408.0,\n",
       "  200.0,\n",
       "  274.0],\n",
       " 'GO:0010829-negative regulation of glucose transmembrane transport': [204.0,\n",
       "  90.0,\n",
       "  298.0,\n",
       "  282.0],\n",
       " 'GO:0008391-arachidonic acid monooxygenase activity': [205.0,\n",
       "  14.0,\n",
       "  56.0,\n",
       "  155.0,\n",
       "  66.0,\n",
       "  99.0,\n",
       "  81.0,\n",
       "  67.0],\n",
       " 'GO:0045259-proton-transporting ATP synthase complex': [206.0,\n",
       "  153.0,\n",
       "  145.0,\n",
       "  103.0,\n",
       "  19.0,\n",
       "  142.0,\n",
       "  26.0,\n",
       "  27.0,\n",
       "  88.0,\n",
       "  97.0,\n",
       "  233.0,\n",
       "  77.0],\n",
       " 'GO:0031935-regulation of chromatin silencing': [207.0,\n",
       "  438.0,\n",
       "  191.0,\n",
       "  238.0,\n",
       "  927.0,\n",
       "  494.0],\n",
       " 'GO:0051004-regulation of lipoprotein lipase activity': [208.0,\n",
       "  123.0,\n",
       "  36.0,\n",
       "  147.0,\n",
       "  79.0,\n",
       "  242.0,\n",
       "  89.0,\n",
       "  133.0,\n",
       "  200.0,\n",
       "  479.0,\n",
       "  80.0,\n",
       "  128.0,\n",
       "  254.0,\n",
       "  197.0,\n",
       "  91.0],\n",
       " 'GO:0005680-anaphase-promoting complex': [209.0, 331.0, 89.0],\n",
       " 'GO:1901030-positive regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway': [210.0,\n",
       "  752.0,\n",
       "  298.0,\n",
       "  275.0,\n",
       "  641.0],\n",
       " 'GO:0022624-proteasome accessory complex': [211.0,\n",
       "  286.0,\n",
       "  88.0,\n",
       "  70.0,\n",
       "  134.0,\n",
       "  74.0,\n",
       "  149.0,\n",
       "  125.0,\n",
       "  328.0,\n",
       "  90.0,\n",
       "  108.0,\n",
       "  206.0],\n",
       " 'GO:1903204-negative regulation of oxidative stress-induced neuron death': 212.0,\n",
       " 'GO:0000717-nucleotide-excision repair, DNA duplex unwinding': [213.0,\n",
       "  170.0,\n",
       "  129.0,\n",
       "  118.0,\n",
       "  190.0],\n",
       " 'GO:2000637-positive regulation of gene silencing by miRNA': [214.0,\n",
       "  147.0,\n",
       "  201.0],\n",
       " 'GO:0019585-glucuronate metabolic process': [215.5,\n",
       "  64.5,\n",
       "  12.5,\n",
       "  48.5,\n",
       "  83.5,\n",
       "  218.5,\n",
       "  26.5,\n",
       "  72.5,\n",
       "  125.5,\n",
       "  45.5,\n",
       "  27.5,\n",
       "  148.5,\n",
       "  62.5,\n",
       "  21.5,\n",
       "  28.5,\n",
       "  25.5,\n",
       "  153.5,\n",
       "  114.5,\n",
       "  85.5,\n",
       "  11.5,\n",
       "  291.5,\n",
       "  89.5,\n",
       "  48.5],\n",
       " 'GO:0006063-uronic acid metabolic process': [215.5,\n",
       "  64.5,\n",
       "  12.5,\n",
       "  48.5,\n",
       "  83.5,\n",
       "  218.5,\n",
       "  26.5,\n",
       "  72.5,\n",
       "  125.5,\n",
       "  45.5,\n",
       "  27.5,\n",
       "  148.5,\n",
       "  62.5,\n",
       "  21.5,\n",
       "  28.5,\n",
       "  25.5,\n",
       "  153.5,\n",
       "  114.5,\n",
       "  85.5,\n",
       "  11.5,\n",
       "  291.5,\n",
       "  89.5,\n",
       "  48.5],\n",
       " 'GO:1903205-regulation of hydrogen peroxide-induced cell death': [217.0,\n",
       "  221.0,\n",
       "  163.0,\n",
       "  175.0],\n",
       " 'GO:0051450-myoblast proliferation': [218.0,\n",
       "  53.0,\n",
       "  247.0,\n",
       "  175.0,\n",
       "  69.0,\n",
       "  355.0,\n",
       "  114.0,\n",
       "  174.0,\n",
       "  123.0],\n",
       " 'GO:0042730-fibrinolysis': [219.0,\n",
       "  283.0,\n",
       "  92.0,\n",
       "  108.0,\n",
       "  213.0,\n",
       "  46.0,\n",
       "  135.0,\n",
       "  33.0,\n",
       "  44.0,\n",
       "  92.0,\n",
       "  259.0,\n",
       "  51.0,\n",
       "  79.0,\n",
       "  117.0,\n",
       "  244.0],\n",
       " 'GO:2001038-regulation of cellular response to drug': [220.0, 187.0, 735.0],\n",
       " 'GO:0031100-animal organ regeneration': [221.0,\n",
       "  576.0,\n",
       "  152.0,\n",
       "  623.0,\n",
       "  639.0,\n",
       "  535.0,\n",
       "  587.0,\n",
       "  646.0,\n",
       "  666.0,\n",
       "  338.0,\n",
       "  1376.0,\n",
       "  486.0,\n",
       "  534.0,\n",
       "  591.0,\n",
       "  557.0,\n",
       "  1425.0,\n",
       "  375.0,\n",
       "  439.0],\n",
       " 'GO:0042776-mitochondrial ATP synthesis coupled proton transport': [222.0,\n",
       "  176.0,\n",
       "  191.0,\n",
       "  202.0,\n",
       "  17.0,\n",
       "  141.0,\n",
       "  37.0,\n",
       "  56.0,\n",
       "  102.0,\n",
       "  73.0,\n",
       "  268.0,\n",
       "  110.0],\n",
       " 'GO:0070129-regulation of mitochondrial translation': [223.0, 196.0, 205.0],\n",
       " 'GO:0070325-lipoprotein particle receptor binding': [224.0,\n",
       "  211.0,\n",
       "  323.0,\n",
       "  178.0,\n",
       "  164.0,\n",
       "  129.0,\n",
       "  206.0,\n",
       "  150.0,\n",
       "  157.0,\n",
       "  149.0,\n",
       "  193.0],\n",
       " 'GO:0033598-mammary gland epithelial cell proliferation': [225.0,\n",
       "  141.0,\n",
       "  203.0,\n",
       "  233.0,\n",
       "  624.0,\n",
       "  196.0,\n",
       "  272.0],\n",
       " 'GO:0006144-purine nucleobase metabolic process': [226.0, 179.0, 141.0],\n",
       " 'GO:0031936-negative regulation of chromatin silencing': [227.0, 179.0],\n",
       " 'GO:0008180-COP9 signalosome': [228.0,\n",
       "  439.0,\n",
       "  116.0,\n",
       "  253.0,\n",
       "  270.0,\n",
       "  275.0,\n",
       "  412.0,\n",
       "  116.0,\n",
       "  159.0,\n",
       "  438.0,\n",
       "  710.0,\n",
       "  146.0],\n",
       " 'GO:0002385-mucosal immune response': [229.0,\n",
       "  145.0,\n",
       "  44.0,\n",
       "  237.0,\n",
       "  214.0,\n",
       "  356.0,\n",
       "  141.0,\n",
       "  249.0,\n",
       "  194.0,\n",
       "  63.0,\n",
       "  121.0,\n",
       "  351.0,\n",
       "  95.0,\n",
       "  131.0,\n",
       "  649.0,\n",
       "  793.0,\n",
       "  270.0],\n",
       " 'GO:1901028-regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway': [230.0,\n",
       "  376.0,\n",
       "  350.0,\n",
       "  533.0,\n",
       "  713.0,\n",
       "  458.0,\n",
       "  456.0,\n",
       "  581.0,\n",
       "  518.0,\n",
       "  296.0,\n",
       "  282.0,\n",
       "  988.0],\n",
       " 'GO:0046599-regulation of centriole replication': 231.0,\n",
       " 'GO:0035198-miRNA binding': [232.0, 280.0, 285.0, 193.0, 422.0, 109.0, 386.0],\n",
       " 'GO:0005697-telomerase holoenzyme complex': [233.0, 218.0, 116.0],\n",
       " 'GO:0045974-regulation of translation, ncRNA-mediated': [235.0,\n",
       "  158.0,\n",
       "  26.0,\n",
       "  19.0,\n",
       "  7.0,\n",
       "  200.0,\n",
       "  196.0,\n",
       "  636.0,\n",
       "  28.0,\n",
       "  130.0,\n",
       "  24.0,\n",
       "  30.0,\n",
       "  41.0,\n",
       "  69.0,\n",
       "  35.0,\n",
       "  61.0,\n",
       "  38.0,\n",
       "  107.0,\n",
       "  166.0,\n",
       "  25.0,\n",
       "  101.0,\n",
       "  16.0,\n",
       "  21.0,\n",
       "  153.0,\n",
       "  25.0],\n",
       " 'GO:0040033-negative regulation of translation, ncRNA-mediated': [235.0,\n",
       "  158.0,\n",
       "  26.0,\n",
       "  19.0,\n",
       "  7.0,\n",
       "  200.0,\n",
       "  196.0,\n",
       "  636.0,\n",
       "  28.0,\n",
       "  130.0,\n",
       "  24.0,\n",
       "  30.0,\n",
       "  41.0,\n",
       "  69.0,\n",
       "  35.0,\n",
       "  61.0,\n",
       "  38.0,\n",
       "  107.0,\n",
       "  166.0,\n",
       "  25.0,\n",
       "  101.0,\n",
       "  16.0,\n",
       "  21.0,\n",
       "  153.0,\n",
       "  25.0],\n",
       " 'GO:0035278-miRNA mediated inhibition of translation': [235.0,\n",
       "  158.0,\n",
       "  26.0,\n",
       "  19.0,\n",
       "  7.0,\n",
       "  200.0,\n",
       "  196.0,\n",
       "  636.0,\n",
       "  28.0,\n",
       "  130.0,\n",
       "  24.0,\n",
       "  30.0,\n",
       "  41.0,\n",
       "  69.0,\n",
       "  35.0,\n",
       "  61.0,\n",
       "  38.0,\n",
       "  107.0,\n",
       "  166.0,\n",
       "  25.0,\n",
       "  101.0,\n",
       "  16.0,\n",
       "  21.0,\n",
       "  153.0,\n",
       "  25.0],\n",
       " 'GO:0006735-NADH regeneration': [238.0,\n",
       "  172.0,\n",
       "  111.0,\n",
       "  238.0,\n",
       "  195.0,\n",
       "  217.0,\n",
       "  203.0,\n",
       "  61.0,\n",
       "  124.0,\n",
       "  317.0,\n",
       "  57.0,\n",
       "  49.0,\n",
       "  87.0,\n",
       "  59.0,\n",
       "  499.0,\n",
       "  131.0],\n",
       " 'GO:0061718-glucose catabolic process to pyruvate': [238.0,\n",
       "  172.0,\n",
       "  111.0,\n",
       "  238.0,\n",
       "  195.0,\n",
       "  217.0,\n",
       "  203.0,\n",
       "  61.0,\n",
       "  124.0,\n",
       "  317.0,\n",
       "  57.0,\n",
       "  49.0,\n",
       "  87.0,\n",
       "  59.0,\n",
       "  499.0,\n",
       "  131.0],\n",
       " 'GO:0061621-canonical glycolysis': [238.0,\n",
       "  172.0,\n",
       "  111.0,\n",
       "  238.0,\n",
       "  195.0,\n",
       "  217.0,\n",
       "  203.0,\n",
       "  61.0,\n",
       "  124.0,\n",
       "  317.0,\n",
       "  57.0,\n",
       "  49.0,\n",
       "  87.0,\n",
       "  59.0,\n",
       "  499.0,\n",
       "  131.0],\n",
       " 'GO:0006361-transcription initiation from RNA polymerase I promoter': [240.0,\n",
       "  380.0,\n",
       "  219.0,\n",
       "  586.0,\n",
       "  251.0,\n",
       "  257.0,\n",
       "  567.0,\n",
       "  217.0],\n",
       " 'GO:0005838-proteasome regulatory particle': [241.0,\n",
       "  101.0,\n",
       "  38.0,\n",
       "  142.0,\n",
       "  75.0,\n",
       "  135.0,\n",
       "  93.0,\n",
       "  109.0],\n",
       " 'GO:0004407-histone deacetylase activity': [242.0, 418.0, 204.0],\n",
       " 'GO:1904591-positive regulation of protein import': [243.0,\n",
       "  403.0,\n",
       "  313.0,\n",
       "  494.0,\n",
       "  291.0,\n",
       "  294.0,\n",
       "  309.0,\n",
       "  617.0,\n",
       "  279.0,\n",
       "  211.0,\n",
       "  351.0,\n",
       "  378.0,\n",
       "  654.0,\n",
       "  249.0,\n",
       "  392.0],\n",
       " 'GO:0006379-mRNA cleavage': [244.0, 189.0],\n",
       " 'GO:0036474-cell death in response to hydrogen peroxide': [245.0,\n",
       "  234.0,\n",
       "  178.0,\n",
       "  203.0,\n",
       "  660.0,\n",
       "  245.0,\n",
       "  229.0],\n",
       " 'GO:1990777-lipoprotein particle': [246.5,\n",
       "  78.5,\n",
       "  150.5,\n",
       "  271.5,\n",
       "  173.5,\n",
       "  142.5,\n",
       "  450.5,\n",
       "  136.5,\n",
       "  567.5,\n",
       "  159.5,\n",
       "  109.5,\n",
       "  198.5,\n",
       "  445.5,\n",
       "  46.5,\n",
       "  163.5,\n",
       "  244.5,\n",
       "  296.5,\n",
       "  397.5],\n",
       " 'GO:0034358-plasma lipoprotein particle': [246.5,\n",
       "  78.5,\n",
       "  150.5,\n",
       "  271.5,\n",
       "  173.5,\n",
       "  142.5,\n",
       "  450.5,\n",
       "  136.5,\n",
       "  567.5,\n",
       "  159.5,\n",
       "  109.5,\n",
       "  198.5,\n",
       "  445.5,\n",
       "  46.5,\n",
       "  163.5,\n",
       "  244.5,\n",
       "  296.5,\n",
       "  397.5],\n",
       " 'GO:0071011-precatalytic spliceosome': [248.0,\n",
       "  307.0,\n",
       "  94.0,\n",
       "  347.0,\n",
       "  359.0,\n",
       "  368.0,\n",
       "  158.0,\n",
       "  161.0,\n",
       "  263.0,\n",
       "  94.0,\n",
       "  183.0,\n",
       "  183.0,\n",
       "  186.0,\n",
       "  559.0,\n",
       "  72.0,\n",
       "  26.0,\n",
       "  141.0],\n",
       " 'GO:0030502-negative regulation of bone mineralization': [249.0,\n",
       "  121.0,\n",
       "  226.0,\n",
       "  274.0,\n",
       "  244.0,\n",
       "  104.0],\n",
       " 'GO:0071005-U2-type precatalytic spliceosome': [250.0,\n",
       "  271.0,\n",
       "  95.0,\n",
       "  306.0,\n",
       "  360.0,\n",
       "  373.0,\n",
       "  162.0,\n",
       "  162.0,\n",
       "  269.0,\n",
       "  96.0,\n",
       "  167.0,\n",
       "  184.0,\n",
       "  187.0,\n",
       "  566.0,\n",
       "  69.0,\n",
       "  27.0,\n",
       "  143.0],\n",
       " 'GO:0032806-carboxy-terminal domain protein kinase complex': 251.0,\n",
       " 'GO:0070932-histone H3 deacetylation': [252.0, 158.0],\n",
       " 'GO:0097525-spliceosomal snRNP complex': [253.0,\n",
       "  335.0,\n",
       "  224.0,\n",
       "  260.0,\n",
       "  425.0,\n",
       "  460.0,\n",
       "  499.0,\n",
       "  220.0,\n",
       "  273.0,\n",
       "  852.0,\n",
       "  691.0,\n",
       "  918.0,\n",
       "  359.0,\n",
       "  102.0,\n",
       "  138.0,\n",
       "  288.0,\n",
       "  255.0,\n",
       "  539.0,\n",
       "  123.0,\n",
       "  1241.0,\n",
       "  643.0,\n",
       "  426.0],\n",
       " 'GO:1902236-negative regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway': [254.0,\n",
       "  165.0,\n",
       "  145.0],\n",
       " 'GO:0003746-translation elongation factor activity': [255.0,\n",
       "  143.0,\n",
       "  98.0,\n",
       "  197.0,\n",
       "  130.0,\n",
       "  76.0,\n",
       "  133.0],\n",
       " 'GO:2000209-regulation of anoikis': 256.0,\n",
       " 'GO:0009147-pyrimidine nucleoside triphosphate metabolic process': [257.0,\n",
       "  187.0,\n",
       "  111.0],\n",
       " 'GO:0071827-plasma lipoprotein particle organization': [258.0,\n",
       "  305.0,\n",
       "  304.0,\n",
       "  235.0,\n",
       "  324.0,\n",
       "  533.0,\n",
       "  498.0,\n",
       "  955.0,\n",
       "  146.0,\n",
       "  233.0,\n",
       "  195.0,\n",
       "  368.0,\n",
       "  475.0,\n",
       "  343.0],\n",
       " 'GO:0046716-muscle cell cellular homeostasis': [259.0, 298.0, 214.0],\n",
       " 'GO:0007549-dosage compensation': 260.0,\n",
       " 'GO:0032799-low-density lipoprotein receptor particle metabolic process': [261.0,\n",
       "  482.0],\n",
       " 'GO:0061620-glycolytic process through glucose-6-phosphate': [262.5,\n",
       "  184.5,\n",
       "  125.5,\n",
       "  257.5,\n",
       "  234.5,\n",
       "  231.5,\n",
       "  242.5,\n",
       "  72.5,\n",
       "  137.5,\n",
       "  452.5,\n",
       "  65.5,\n",
       "  160.5,\n",
       "  55.5,\n",
       "  93.5,\n",
       "  64.5,\n",
       "  162.5],\n",
       " 'GO:0061615-glycolytic process through fructose-6-phosphate': [262.5,\n",
       "  184.5,\n",
       "  125.5,\n",
       "  257.5,\n",
       "  234.5,\n",
       "  231.5,\n",
       "  242.5,\n",
       "  72.5,\n",
       "  137.5,\n",
       "  452.5,\n",
       "  65.5,\n",
       "  160.5,\n",
       "  55.5,\n",
       "  93.5,\n",
       "  64.5,\n",
       "  162.5],\n",
       " 'GO:0022625-cytosolic large ribosomal subunit': [264.0,\n",
       "  167.0,\n",
       "  463.0,\n",
       "  182.0,\n",
       "  383.0,\n",
       "  85.0,\n",
       "  181.0,\n",
       "  143.0,\n",
       "  849.0,\n",
       "  143.0,\n",
       "  175.0,\n",
       "  264.0,\n",
       "  156.0,\n",
       "  123.0,\n",
       "  269.0,\n",
       "  284.0,\n",
       "  284.0,\n",
       "  302.0,\n",
       "  346.0,\n",
       "  425.0,\n",
       "  448.0,\n",
       "  180.0],\n",
       " 'GO:0042611-MHC protein complex': [265.0,\n",
       "  25.0,\n",
       "  188.0,\n",
       "  19.0,\n",
       "  53.0,\n",
       "  85.0,\n",
       "  100.0,\n",
       "  218.0,\n",
       "  81.0,\n",
       "  42.0,\n",
       "  28.0,\n",
       "  74.0,\n",
       "  122.0,\n",
       "  50.0,\n",
       "  144.0],\n",
       " 'GO:0010592-positive regulation of lamellipodium assembly': 266.0,\n",
       " 'GO:0002438-acute inflammatory response to antigenic stimulus': [267.0,\n",
       "  168.0,\n",
       "  162.0,\n",
       "  105.0,\n",
       "  169.0,\n",
       "  275.0],\n",
       " 'GO:0008327-methyl-CpG binding': 268.0,\n",
       " 'GO:0050750-low-density lipoprotein particle receptor binding': [269.0,\n",
       "  163.0,\n",
       "  190.0,\n",
       "  218.0],\n",
       " 'GO:0070911-global genome nucleotide-excision repair': [270.0, 231.0, 177.0],\n",
       " 'GO:0000183-chromatin silencing at rDNA': [271.0,\n",
       "  218.0,\n",
       "  231.0,\n",
       "  109.0,\n",
       "  205.0,\n",
       "  896.0,\n",
       "  308.0,\n",
       "  335.0,\n",
       "  287.0],\n",
       " 'GO:0005719-nuclear euchromatin': [272.0,\n",
       "  278.0,\n",
       "  154.0,\n",
       "  279.0,\n",
       "  219.0,\n",
       "  544.0,\n",
       "  130.0,\n",
       "  325.0,\n",
       "  176.0],\n",
       " 'GO:0043486-histone exchange': [273.0,\n",
       "  453.0,\n",
       "  266.0,\n",
       "  640.0,\n",
       "  663.0,\n",
       "  856.0,\n",
       "  699.0,\n",
       "  191.0,\n",
       "  344.0,\n",
       "  849.0,\n",
       "  1320.0,\n",
       "  105.0,\n",
       "  206.0,\n",
       "  381.0,\n",
       "  268.0,\n",
       "  1235.0,\n",
       "  246.0,\n",
       "  933.0,\n",
       "  272.0],\n",
       " 'GO:0051180-vitamin transport': [274.0, 948.0],\n",
       " 'GO:0006294-nucleotide-excision repair, preincision complex assembly': [275.0,\n",
       "  269.0,\n",
       "  199.0,\n",
       "  233.0,\n",
       "  163.0,\n",
       "  207.0,\n",
       "  155.0,\n",
       "  148.0,\n",
       "  194.0],\n",
       " 'GO:0001891-phagocytic cup': 276.0,\n",
       " 'GO:0033558-protein deacetylase activity': [277.0, 241.0],\n",
       " 'GO:1900739-regulation of protein insertion into mitochondrial membrane involved in apoptotic signaling pathway': [278.5,\n",
       "  705.5],\n",
       " 'GO:1900740-positive regulation of protein insertion into mitochondrial membrane involved in apoptotic signaling pathway': [278.5,\n",
       "  705.5],\n",
       " 'GO:0042407-cristae formation': [280.0,\n",
       "  177.0,\n",
       "  182.0,\n",
       "  267.0,\n",
       "  162.0,\n",
       "  215.0,\n",
       "  188.0,\n",
       "  262.0,\n",
       "  36.0,\n",
       "  87.0,\n",
       "  202.0,\n",
       "  100.0,\n",
       "  218.0,\n",
       "  123.0,\n",
       "  495.0,\n",
       "  235.0],\n",
       " 'GO:0070268-cornification': [281.0,\n",
       "  183.0,\n",
       "  426.0,\n",
       "  604.0,\n",
       "  289.0,\n",
       "  241.0,\n",
       "  762.0,\n",
       "  206.0,\n",
       "  514.0,\n",
       "  136.0,\n",
       "  359.0,\n",
       "  804.0,\n",
       "  406.0,\n",
       "  168.0,\n",
       "  348.0,\n",
       "  630.0,\n",
       "  479.0],\n",
       " 'GO:0035066-positive regulation of histone acetylation': [282.0, 362.0],\n",
       " 'GO:0005839-proteasome core complex': [283.0,\n",
       "  150.0,\n",
       "  31.0,\n",
       "  81.0,\n",
       "  107.0,\n",
       "  130.0,\n",
       "  19.0,\n",
       "  35.0,\n",
       "  75.0,\n",
       "  161.0,\n",
       "  96.0,\n",
       "  222.0,\n",
       "  21.0,\n",
       "  108.0],\n",
       " 'GO:0046931-pore complex assembly': 284.0,\n",
       " 'GO:0010485-H4 histone acetyltransferase activity': 285.0,\n",
       " 'GO:0000466-maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)': [286.0,\n",
       "  74.0,\n",
       "  88.0],\n",
       " 'GO:0002183-cytoplasmic translational initiation': [287.0,\n",
       "  320.0,\n",
       "  138.0,\n",
       "  209.0,\n",
       "  252.0,\n",
       "  298.0,\n",
       "  504.0,\n",
       "  125.0,\n",
       "  105.0,\n",
       "  203.0,\n",
       "  230.0,\n",
       "  118.0,\n",
       "  320.0],\n",
       " 'GO:0019682-glyceraldehyde-3-phosphate metabolic process': [288.0,\n",
       "  78.0,\n",
       "  98.0,\n",
       "  134.0,\n",
       "  70.0,\n",
       "  150.0],\n",
       " 'GO:0000715-nucleotide-excision repair, DNA damage recognition': [289.0,\n",
       "  309.0,\n",
       "  542.0,\n",
       "  144.0],\n",
       " 'GO:0002861-regulation of inflammatory response to antigenic stimulus': [290.0,\n",
       "  526.0],\n",
       " 'GO:0007064-mitotic sister chromatid cohesion': [291.0, 124.0, 155.0, 91.0],\n",
       " 'GO:0090050-positive regulation of cell migration involved in sprouting angiogenesis': [292.0,\n",
       "  290.0,\n",
       "  47.0,\n",
       "  66.0,\n",
       "  155.0,\n",
       "  245.0,\n",
       "  345.0,\n",
       "  286.0,\n",
       "  144.0,\n",
       "  147.0,\n",
       "  282.0,\n",
       "  82.0,\n",
       "  251.0,\n",
       "  68.0,\n",
       "  243.0,\n",
       "  271.0],\n",
       " 'GO:0006356-regulation of transcription by RNA polymerase I': 293.0,\n",
       " 'GO:0071825-protein-lipid complex subunit organization': [294.0,\n",
       "  364.0,\n",
       "  297.0,\n",
       "  295.0,\n",
       "  442.0,\n",
       "  502.0,\n",
       "  553.0,\n",
       "  887.0,\n",
       "  164.0,\n",
       "  263.0,\n",
       "  310.0,\n",
       "  505.0,\n",
       "  856.0,\n",
       "  1027.0,\n",
       "  413.0],\n",
       " 'GO:0034724-DNA replication-independent nucleosome organization': [295.0,\n",
       "  267.0,\n",
       "  160.0,\n",
       "  270.0,\n",
       "  388.0,\n",
       "  617.0,\n",
       "  430.0,\n",
       "  228.0,\n",
       "  139.0,\n",
       "  449.0,\n",
       "  486.0,\n",
       "  88.0,\n",
       "  324.0,\n",
       "  104.0,\n",
       "  225.0,\n",
       "  205.0,\n",
       "  852.0,\n",
       "  167.0,\n",
       "  705.0,\n",
       "  332.0,\n",
       "  371.0],\n",
       " 'GO:0031639-plasminogen activation': [296.0,\n",
       "  188.0,\n",
       "  96.0,\n",
       "  117.0,\n",
       "  205.0,\n",
       "  36.0,\n",
       "  73.0,\n",
       "  47.0,\n",
       "  67.0,\n",
       "  149.0,\n",
       "  45.0,\n",
       "  115.0,\n",
       "  58.0,\n",
       "  161.0,\n",
       "  112.0],\n",
       " 'GO:0043403-skeletal muscle tissue regeneration': [297.0,\n",
       "  353.0,\n",
       "  418.0,\n",
       "  465.0,\n",
       "  400.0,\n",
       "  1184.0,\n",
       "  346.0],\n",
       " 'GO:0072595-maintenance of protein localization in organelle': [298.0,\n",
       "  481.0,\n",
       "  562.0,\n",
       "  1173.0,\n",
       "  328.0],\n",
       " 'GO:0036260-RNA capping': [299.5,\n",
       "  336.5,\n",
       "  230.5,\n",
       "  402.5,\n",
       "  271.5,\n",
       "  252.5,\n",
       "  580.5,\n",
       "  119.5,\n",
       "  237.5,\n",
       "  448.5,\n",
       "  178.5,\n",
       "  36.5,\n",
       "  218.5],\n",
       " 'GO:0009452-7-methylguanosine RNA capping': [299.5,\n",
       "  336.5,\n",
       "  230.5,\n",
       "  402.5,\n",
       "  271.5,\n",
       "  252.5,\n",
       "  580.5,\n",
       "  119.5,\n",
       "  237.5,\n",
       "  448.5,\n",
       "  178.5,\n",
       "  36.5,\n",
       "  218.5],\n",
       " 'GO:0006370-7-methylguanosine mRNA capping': [301.0,\n",
       "  347.0,\n",
       "  276.0,\n",
       "  254.0,\n",
       "  603.0,\n",
       "  123.0,\n",
       "  240.0,\n",
       "  467.0,\n",
       "  187.0,\n",
       "  150.0],\n",
       " 'GO:0070717-poly-purine tract binding': 302.0,\n",
       " 'GO:0046824-positive regulation of nucleocytoplasmic transport': [303.0,\n",
       "  726.0,\n",
       "  384.0,\n",
       "  805.0,\n",
       "  624.0,\n",
       "  1006.0,\n",
       "  508.0,\n",
       "  561.0,\n",
       "  474.0,\n",
       "  1011.0,\n",
       "  376.0,\n",
       "  487.0,\n",
       "  366.0,\n",
       "  532.0,\n",
       "  463.0,\n",
       "  1169.0,\n",
       "  296.0,\n",
       "  1285.0,\n",
       "  610.0,\n",
       "  359.0],\n",
       " \"GO:0016895-exodeoxyribonuclease activity, producing 5'-phosphomonoesters\": [304.0,\n",
       "  142.0,\n",
       "  348.0,\n",
       "  132.0,\n",
       "  93.0],\n",
       " 'GO:0048384-retinoic acid receptor signaling pathway': [305.0, 280.0, 682.0],\n",
       " 'GO:0009620-response to fungus': [306.0,\n",
       "  274.0,\n",
       "  338.0,\n",
       "  451.0,\n",
       "  426.0,\n",
       "  100.0,\n",
       "  297.0,\n",
       "  135.0,\n",
       "  176.0,\n",
       "  919.0,\n",
       "  424.0,\n",
       "  170.0,\n",
       "  413.0,\n",
       "  320.0,\n",
       "  862.0],\n",
       " 'GO:0071218-cellular response to misfolded protein': [307.0, 108.0],\n",
       " 'GO:0046596-regulation of viral entry into host cell': [308.0,\n",
       "  189.0,\n",
       "  242.0,\n",
       "  204.0,\n",
       "  357.0,\n",
       "  189.0,\n",
       "  601.0,\n",
       "  222.0,\n",
       "  199.0,\n",
       "  381.0,\n",
       "  348.0],\n",
       " 'GO:0003735-structural constituent of ribosome': [309.0,\n",
       "  198.0,\n",
       "  203.0,\n",
       "  352.0,\n",
       "  244.0,\n",
       "  628.0,\n",
       "  141.0,\n",
       "  160.0,\n",
       "  168.0,\n",
       "  844.0,\n",
       "  369.0,\n",
       "  182.0,\n",
       "  352.0,\n",
       "  250.0,\n",
       "  149.0,\n",
       "  156.0,\n",
       "  195.0,\n",
       "  318.0,\n",
       "  281.0,\n",
       "  377.0,\n",
       "  324.0,\n",
       "  169.0,\n",
       "  613.0,\n",
       "  399.0,\n",
       "  166.0],\n",
       " 'GO:0010971-positive regulation of G2/M transition of mitotic cell cycle': [310.0,\n",
       "  247.0],\n",
       " 'GO:0071280-cellular response to copper ion': [311.0,\n",
       "  244.0,\n",
       "  260.0,\n",
       "  314.0,\n",
       "  106.0,\n",
       "  182.0,\n",
       "  465.0,\n",
       "  220.0,\n",
       "  179.0,\n",
       "  159.0,\n",
       "  147.0],\n",
       " 'GO:0140142-nucleocytoplasmic carrier activity': [312.0, 394.0, 215.0],\n",
       " 'GO:0051973-positive regulation of telomerase activity': [313.0,\n",
       "  327.0,\n",
       "  191.0,\n",
       "  199.0,\n",
       "  817.0,\n",
       "  180.0],\n",
       " 'GO:0090051-negative regulation of cell migration involved in sprouting angiogenesis': [314.0,\n",
       "  285.0,\n",
       "  38.0,\n",
       "  95.0,\n",
       "  281.0,\n",
       "  423.0,\n",
       "  56.0,\n",
       "  207.0,\n",
       "  29.0,\n",
       "  68.0,\n",
       "  64.0,\n",
       "  128.0,\n",
       "  49.0,\n",
       "  69.0,\n",
       "  201.0,\n",
       "  189.0,\n",
       "  172.0,\n",
       "  28.0,\n",
       "  107.0,\n",
       "  59.0],\n",
       " 'GO:0036002-pre-mRNA binding': [315.0,\n",
       "  485.0,\n",
       "  376.0,\n",
       "  1199.0,\n",
       "  582.0,\n",
       "  1010.0,\n",
       "  278.0,\n",
       "  807.0,\n",
       "  659.0],\n",
       " 'GO:0050136-NADH dehydrogenase (quinone) activity': [317.0,\n",
       "  150.0,\n",
       "  295.0,\n",
       "  168.0,\n",
       "  557.0,\n",
       "  316.0,\n",
       "  178.0,\n",
       "  139.0,\n",
       "  176.0,\n",
       "  148.0,\n",
       "  113.0,\n",
       "  210.0,\n",
       "  138.0,\n",
       "  54.0,\n",
       "  101.0,\n",
       "  148.0,\n",
       "  292.0,\n",
       "  128.0,\n",
       "  285.0,\n",
       "  252.0,\n",
       "  39.0,\n",
       "  790.0,\n",
       "  190.0,\n",
       "  238.0],\n",
       " 'GO:0003954-NADH dehydrogenase activity': [317.0,\n",
       "  150.0,\n",
       "  295.0,\n",
       "  168.0,\n",
       "  557.0,\n",
       "  316.0,\n",
       "  178.0,\n",
       "  139.0,\n",
       "  176.0,\n",
       "  148.0,\n",
       "  113.0,\n",
       "  210.0,\n",
       "  138.0,\n",
       "  54.0,\n",
       "  101.0,\n",
       "  148.0,\n",
       "  292.0,\n",
       "  128.0,\n",
       "  285.0,\n",
       "  252.0,\n",
       "  39.0,\n",
       "  790.0,\n",
       "  190.0,\n",
       "  238.0],\n",
       " 'GO:0008137-NADH dehydrogenase (ubiquinone) activity': [317.0,\n",
       "  150.0,\n",
       "  295.0,\n",
       "  168.0,\n",
       "  557.0,\n",
       "  316.0,\n",
       "  178.0,\n",
       "  139.0,\n",
       "  176.0,\n",
       "  148.0,\n",
       "  113.0,\n",
       "  210.0,\n",
       "  138.0,\n",
       "  54.0,\n",
       "  101.0,\n",
       "  148.0,\n",
       "  292.0,\n",
       "  128.0,\n",
       "  285.0,\n",
       "  252.0,\n",
       "  39.0,\n",
       "  790.0,\n",
       "  190.0,\n",
       "  238.0],\n",
       " 'GO:0090482-vitamin transmembrane transporter activity': 319.0,\n",
       " 'GO:0006336-DNA replication-independent nucleosome assembly': [320.0,\n",
       "  268.0,\n",
       "  148.0,\n",
       "  261.0,\n",
       "  399.0,\n",
       "  628.0,\n",
       "  449.0,\n",
       "  234.0,\n",
       "  134.0,\n",
       "  404.0,\n",
       "  497.0,\n",
       "  91.0,\n",
       "  283.0,\n",
       "  106.0,\n",
       "  228.0,\n",
       "  211.0,\n",
       "  875.0,\n",
       "  175.0,\n",
       "  728.0,\n",
       "  342.0,\n",
       "  340.0],\n",
       " 'GO:0042246-tissue regeneration': [321.0,\n",
       "  803.0,\n",
       "  105.0,\n",
       "  348.0,\n",
       "  421.0,\n",
       "  453.0,\n",
       "  549.0,\n",
       "  1277.0,\n",
       "  1294.0,\n",
       "  363.0,\n",
       "  574.0,\n",
       "  1066.0,\n",
       "  375.0,\n",
       "  790.0,\n",
       "  849.0,\n",
       "  1012.0,\n",
       "  561.0,\n",
       "  1706.0,\n",
       "  763.0],\n",
       " 'GO:0004529-exodeoxyribonuclease activity': [322.0, 137.0, 95.0],\n",
       " 'GO:0050434-positive regulation of viral transcription': [323.0,\n",
       "  601.0,\n",
       "  649.0,\n",
       "  476.0,\n",
       "  262.0,\n",
       "  336.0,\n",
       "  753.0,\n",
       "  196.0,\n",
       "  302.0,\n",
       "  788.0,\n",
       "  178.0],\n",
       " 'GO:0031492-nucleosomal DNA binding': [324.0,\n",
       "  348.0,\n",
       "  130.0,\n",
       "  233.0,\n",
       "  755.0,\n",
       "  256.0,\n",
       "  291.0,\n",
       "  701.0,\n",
       "  138.0,\n",
       "  262.0,\n",
       "  377.0,\n",
       "  553.0,\n",
       "  969.0,\n",
       "  168.0,\n",
       "  362.0,\n",
       "  388.0],\n",
       " 'GO:2001024-negative regulation of response to drug': [325.0,\n",
       "  666.0,\n",
       "  135.0,\n",
       "  352.0,\n",
       "  215.0],\n",
       " 'GO:0032212-positive regulation of telomere maintenance via telomerase': [326.0,\n",
       "  396.0,\n",
       "  267.0,\n",
       "  656.0,\n",
       "  177.0,\n",
       "  312.0],\n",
       " 'GO:0006120-mitochondrial electron transport, NADH to ubiquinone': [327.0,\n",
       "  163.0,\n",
       "  199.0,\n",
       "  146.0,\n",
       "  554.0,\n",
       "  360.0,\n",
       "  164.0,\n",
       "  142.0,\n",
       "  168.0,\n",
       "  132.0,\n",
       "  97.0,\n",
       "  201.0,\n",
       "  121.0,\n",
       "  52.0,\n",
       "  92.0,\n",
       "  117.0,\n",
       "  297.0,\n",
       "  148.0,\n",
       "  292.0,\n",
       "  201.0,\n",
       "  29.0,\n",
       "  602.0,\n",
       "  157.0,\n",
       "  177.0],\n",
       " 'GO:0022626-cytosolic ribosome': [328.0,\n",
       "  228.0,\n",
       "  296.0,\n",
       "  472.0,\n",
       "  240.0,\n",
       "  405.0,\n",
       "  132.0,\n",
       "  284.0,\n",
       "  181.0,\n",
       "  1056.0,\n",
       "  537.0,\n",
       "  149.0,\n",
       "  267.0,\n",
       "  367.0,\n",
       "  217.0,\n",
       "  172.0,\n",
       "  333.0,\n",
       "  461.0,\n",
       "  360.0,\n",
       "  479.0,\n",
       "  400.0,\n",
       "  111.0,\n",
       "  460.0,\n",
       "  544.0,\n",
       "  224.0],\n",
       " 'GO:0000786-nucleosome': [329.0,\n",
       "  408.0,\n",
       "  91.0,\n",
       "  217.0,\n",
       "  842.0,\n",
       "  245.0,\n",
       "  967.0,\n",
       "  607.0,\n",
       "  399.0,\n",
       "  212.0,\n",
       "  197.0,\n",
       "  141.0,\n",
       "  458.0,\n",
       "  781.0,\n",
       "  99.0,\n",
       "  226.0,\n",
       "  315.0,\n",
       "  206.0,\n",
       "  442.0,\n",
       "  655.0,\n",
       "  212.0,\n",
       "  535.0,\n",
       "  380.0,\n",
       "  354.0],\n",
       " 'GO:0045271-respiratory chain complex I': [331.0,\n",
       "  161.0,\n",
       "  263.0,\n",
       "  172.0,\n",
       "  749.0,\n",
       "  327.0,\n",
       "  140.0,\n",
       "  125.0,\n",
       "  180.0,\n",
       "  136.0,\n",
       "  101.0,\n",
       "  252.0,\n",
       "  147.0,\n",
       "  62.0,\n",
       "  121.0,\n",
       "  115.0,\n",
       "  385.0,\n",
       "  141.0,\n",
       "  272.0,\n",
       "  269.0,\n",
       "  32.0,\n",
       "  778.0,\n",
       "  175.0,\n",
       "  220.0],\n",
       " 'GO:0030964-NADH dehydrogenase complex': [331.0,\n",
       "  161.0,\n",
       "  263.0,\n",
       "  172.0,\n",
       "  749.0,\n",
       "  327.0,\n",
       "  140.0,\n",
       "  125.0,\n",
       "  180.0,\n",
       "  136.0,\n",
       "  101.0,\n",
       "  252.0,\n",
       "  147.0,\n",
       "  62.0,\n",
       "  121.0,\n",
       "  115.0,\n",
       "  385.0,\n",
       "  141.0,\n",
       "  272.0,\n",
       "  269.0,\n",
       "  32.0,\n",
       "  778.0,\n",
       "  175.0,\n",
       "  220.0],\n",
       " 'GO:0005747-mitochondrial respiratory chain complex I': [331.0,\n",
       "  161.0,\n",
       "  263.0,\n",
       "  172.0,\n",
       "  749.0,\n",
       "  327.0,\n",
       "  140.0,\n",
       "  125.0,\n",
       "  180.0,\n",
       "  136.0,\n",
       "  101.0,\n",
       "  252.0,\n",
       "  147.0,\n",
       "  62.0,\n",
       "  121.0,\n",
       "  115.0,\n",
       "  385.0,\n",
       "  141.0,\n",
       "  272.0,\n",
       "  269.0,\n",
       "  32.0,\n",
       "  778.0,\n",
       "  175.0,\n",
       "  220.0],\n",
       " 'GO:2000727-positive regulation of cardiac muscle cell differentiation': [333.0,\n",
       "  73.0,\n",
       "  234.0,\n",
       "  331.0,\n",
       "  186.0,\n",
       "  240.0,\n",
       "  261.0,\n",
       "  31.0,\n",
       "  256.0,\n",
       "  109.0,\n",
       "  200.0],\n",
       " 'GO:0018279-protein N-linked glycosylation via asparagine': [334.0,\n",
       "  287.0,\n",
       "  134.0,\n",
       "  351.0,\n",
       "  133.0,\n",
       "  293.0,\n",
       "  275.0,\n",
       "  204.0,\n",
       "  247.0,\n",
       "  405.0,\n",
       "  163.0,\n",
       "  582.0,\n",
       "  238.0],\n",
       " 'GO:0032994-protein-lipid complex': [335.0,\n",
       "  110.0,\n",
       "  183.0,\n",
       "  326.0,\n",
       "  247.0,\n",
       "  270.0,\n",
       "  436.0,\n",
       "  147.0,\n",
       "  581.0,\n",
       "  195.0,\n",
       "  293.0,\n",
       "  619.0,\n",
       "  64.0,\n",
       "  200.0,\n",
       "  254.0,\n",
       "  385.0,\n",
       "  464.0],\n",
       " 'GO:0140104-molecular carrier activity': [336.0,\n",
       "  201.0,\n",
       "  766.0,\n",
       "  763.0,\n",
       "  843.0,\n",
       "  1193.0,\n",
       "  430.0,\n",
       "  437.0,\n",
       "  554.0,\n",
       "  1147.0],\n",
       " 'GO:0044815-DNA packaging complex': [337.0,\n",
       "  377.0,\n",
       "  96.0,\n",
       "  192.0,\n",
       "  743.0,\n",
       "  211.0,\n",
       "  829.0,\n",
       "  481.0,\n",
       "  408.0,\n",
       "  184.0,\n",
       "  249.0,\n",
       "  181.0,\n",
       "  444.0,\n",
       "  697.0,\n",
       "  84.0,\n",
       "  268.0,\n",
       "  280.0,\n",
       "  274.0,\n",
       "  353.0,\n",
       "  678.0,\n",
       "  181.0,\n",
       "  659.0,\n",
       "  328.0,\n",
       "  366.0],\n",
       " 'GO:0022627-cytosolic small ribosomal subunit': [338.0,\n",
       "  229.0,\n",
       "  308.0,\n",
       "  278.0,\n",
       "  335.0,\n",
       "  77.0,\n",
       "  248.0,\n",
       "  422.0,\n",
       "  128.0,\n",
       "  236.0,\n",
       "  277.0,\n",
       "  142.0,\n",
       "  347.0,\n",
       "  334.0,\n",
       "  296.0,\n",
       "  688.0,\n",
       "  321.0,\n",
       "  491.0,\n",
       "  157.0],\n",
       " 'GO:0030867-rough endoplasmic reticulum membrane': [339.0, 744.0],\n",
       " 'GO:0070198-protein localization to chromosome, telomeric region': [340.0,\n",
       "  169.0,\n",
       "  164.0,\n",
       "  124.0,\n",
       "  296.0],\n",
       " 'GO:0007007-inner mitochondrial membrane organization': [341.0,\n",
       "  206.0,\n",
       "  216.0,\n",
       "  504.0,\n",
       "  515.0,\n",
       "  288.0,\n",
       "  168.0,\n",
       "  257.0,\n",
       "  361.0,\n",
       "  140.0,\n",
       "  101.0,\n",
       "  280.0,\n",
       "  108.0,\n",
       "  258.0,\n",
       "  241.0,\n",
       "  508.0,\n",
       "  313.0,\n",
       "  53.0,\n",
       "  240.0],\n",
       " 'GO:0090049-regulation of cell migration involved in sprouting angiogenesis': [342.0,\n",
       "  330.0,\n",
       "  60.0,\n",
       "  82.0,\n",
       "  228.0,\n",
       "  485.0,\n",
       "  397.0,\n",
       "  697.0,\n",
       "  148.0,\n",
       "  426.0,\n",
       "  108.0,\n",
       "  118.0,\n",
       "  167.0,\n",
       "  680.0,\n",
       "  78.0,\n",
       "  225.0,\n",
       "  423.0,\n",
       "  546.0,\n",
       "  315.0,\n",
       "  74.0,\n",
       "  115.0,\n",
       "  428.0,\n",
       "  198.0],\n",
       " 'GO:0031145-anaphase-promoting complex-dependent catabolic process': [343.0,\n",
       "  289.0,\n",
       "  156.0,\n",
       "  321.0,\n",
       "  846.0,\n",
       "  374.0,\n",
       "  185.0,\n",
       "  188.0,\n",
       "  185.0,\n",
       "  264.0,\n",
       "  491.0,\n",
       "  240.0,\n",
       "  110.0,\n",
       "  513.0,\n",
       "  141.0,\n",
       "  250.0,\n",
       "  208.0,\n",
       "  595.0,\n",
       "  145.0,\n",
       "  86.0,\n",
       "  935.0,\n",
       "  211.0,\n",
       "  304.0],\n",
       " 'GO:0050687-negative regulation of defense response to virus': [344.0,\n",
       "  183.0,\n",
       "  152.0,\n",
       "  546.0,\n",
       "  267.0],\n",
       " 'GO:0033522-histone H2A ubiquitination': [345.0, 657.0, 94.0],\n",
       " 'GO:0033177-proton-transporting two-sector ATPase complex, proton-transporting domain': [346.0,\n",
       "  231.0,\n",
       "  77.0,\n",
       "  136.0,\n",
       "  97.0,\n",
       "  174.0,\n",
       "  221.0],\n",
       " 'GO:0042269-regulation of natural killer cell mediated cytotoxicity': [347.0,\n",
       "  763.0,\n",
       "  444.0,\n",
       "  290.0],\n",
       " 'GO:0043616-keratinocyte proliferation': [348.0, 689.0, 631.0, 239.0, 266.0],\n",
       " 'GO:0019400-alditol metabolic process': [349.0, 156.0, 432.0, 46.0, 339.0],\n",
       " 'GO:0010165-response to X-ray': [350.0, 342.0],\n",
       " 'GO:0015934-large ribosomal subunit': [351.0,\n",
       "  207.0,\n",
       "  373.0,\n",
       "  555.0,\n",
       "  284.0,\n",
       "  764.0,\n",
       "  287.0,\n",
       "  216.0,\n",
       "  169.0,\n",
       "  608.0,\n",
       "  545.0,\n",
       "  324.0,\n",
       "  432.0,\n",
       "  395.0,\n",
       "  181.0,\n",
       "  264.0,\n",
       "  237.0,\n",
       "  355.0,\n",
       "  319.0,\n",
       "  416.0,\n",
       "  338.0,\n",
       "  85.0,\n",
       "  861.0,\n",
       "  585.0,\n",
       "  231.0],\n",
       " 'GO:0045292-mRNA cis splicing, via spliceosome': [352.0,\n",
       "  590.0,\n",
       "  287.0,\n",
       "  652.0,\n",
       "  568.0,\n",
       "  632.0,\n",
       "  394.0,\n",
       "  190.0,\n",
       "  273.0,\n",
       "  389.0,\n",
       "  460.0,\n",
       "  189.0],\n",
       " 'GO:0044391-ribosomal subunit': [353.0,\n",
       "  252.0,\n",
       "  253.0,\n",
       "  528.0,\n",
       "  288.0,\n",
       "  777.0,\n",
       "  301.0,\n",
       "  219.0,\n",
       "  186.0,\n",
       "  642.0,\n",
       "  506.0,\n",
       "  289.0,\n",
       "  446.0,\n",
       "  361.0,\n",
       "  180.0,\n",
       "  238.0,\n",
       "  227.0,\n",
       "  380.0,\n",
       "  308.0,\n",
       "  472.0,\n",
       "  359.0,\n",
       "  76.0,\n",
       "  824.0,\n",
       "  516.0,\n",
       "  256.0],\n",
       " 'GO:1900182-positive regulation of protein localization to nucleus': [354.0,\n",
       "  617.0,\n",
       "  138.0,\n",
       "  522.0,\n",
       "  754.0,\n",
       "  617.0,\n",
       "  577.0,\n",
       "  288.0,\n",
       "  467.0,\n",
       "  470.0,\n",
       "  799.0,\n",
       "  284.0,\n",
       "  399.0,\n",
       "  257.0,\n",
       "  537.0,\n",
       "  434.0,\n",
       "  863.0,\n",
       "  138.0,\n",
       "  118.0,\n",
       "  1097.0,\n",
       "  465.0,\n",
       "  513.0],\n",
       " \"GO:0035925-mRNA 3'-UTR AU-rich region binding\": 355.0,\n",
       " 'GO:0071168-protein localization to chromatin': [356.0, 229.0, 136.0, 246.0],\n",
       " 'GO:0006415-translational termination': [357.0,\n",
       "  273.0,\n",
       "  273.0,\n",
       "  595.0,\n",
       "  459.0,\n",
       "  1256.0,\n",
       "  346.0,\n",
       "  221.0,\n",
       "  251.0,\n",
       "  187.0,\n",
       "  491.0,\n",
       "  834.0,\n",
       "  342.0,\n",
       "  97.0,\n",
       "  573.0,\n",
       "  110.0,\n",
       "  322.0,\n",
       "  282.0,\n",
       "  609.0,\n",
       "  202.0,\n",
       "  60.0,\n",
       "  1398.0,\n",
       "  309.0,\n",
       "  349.0],\n",
       " 'GO:0002715-regulation of natural killer cell mediated immunity': [358.0,\n",
       "  585.0,\n",
       "  595.0,\n",
       "  469.0,\n",
       "  330.0],\n",
       " 'GO:0000387-spliceosomal snRNP assembly': [359.0,\n",
       "  478.0,\n",
       "  162.0,\n",
       "  515.0,\n",
       "  668.0,\n",
       "  647.0,\n",
       "  253.0,\n",
       "  428.0,\n",
       "  595.0,\n",
       "  104.0,\n",
       "  223.0,\n",
       "  391.0,\n",
       "  237.0,\n",
       "  777.0,\n",
       "  110.0,\n",
       "  87.0],\n",
       " 'GO:0070126-mitochondrial translational termination': [360.0,\n",
       "  242.0,\n",
       "  278.0,\n",
       "  581.0,\n",
       "  394.0,\n",
       "  1272.0,\n",
       "  284.0,\n",
       "  163.0,\n",
       "  232.0,\n",
       "  164.0,\n",
       "  457.0,\n",
       "  705.0,\n",
       "  200.0,\n",
       "  68.0,\n",
       "  419.0,\n",
       "  101.0,\n",
       "  283.0,\n",
       "  257.0,\n",
       "  525.0,\n",
       "  158.0,\n",
       "  52.0,\n",
       "  1392.0,\n",
       "  322.0,\n",
       "  309.0],\n",
       " 'GO:0001825-blastocyst formation': [361.0,\n",
       "  169.0,\n",
       "  622.0,\n",
       "  1196.0,\n",
       "  656.0,\n",
       "  172.0],\n",
       " 'GO:0016675-oxidoreductase activity, acting on a heme group of donors': [362.0,\n",
       "  130.0,\n",
       "  107.0,\n",
       "  144.0,\n",
       "  70.0,\n",
       "  61.0,\n",
       "  37.0,\n",
       "  237.0,\n",
       "  365.0,\n",
       "  152.0,\n",
       "  234.0,\n",
       "  22.0,\n",
       "  424.0,\n",
       "  294.0,\n",
       "  78.0],\n",
       " 'GO:0005762-mitochondrial large ribosomal subunit': [363.5,\n",
       "  246.5,\n",
       "  417.5,\n",
       "  396.5,\n",
       "  1119.5,\n",
       "  445.5,\n",
       "  223.5,\n",
       "  196.5,\n",
       "  204.5,\n",
       "  794.5,\n",
       "  424.5,\n",
       "  89.5,\n",
       "  127.5,\n",
       "  441.5,\n",
       "  350.5,\n",
       "  639.5,\n",
       "  224.5,\n",
       "  56.5,\n",
       "  1329.5,\n",
       "  370.5,\n",
       "  228.5],\n",
       " 'GO:0000315-organellar large ribosomal subunit': [363.5,\n",
       "  246.5,\n",
       "  417.5,\n",
       "  396.5,\n",
       "  1119.5,\n",
       "  445.5,\n",
       "  223.5,\n",
       "  196.5,\n",
       "  204.5,\n",
       "  794.5,\n",
       "  424.5,\n",
       "  89.5,\n",
       "  127.5,\n",
       "  441.5,\n",
       "  350.5,\n",
       "  639.5,\n",
       "  224.5,\n",
       "  56.5,\n",
       "  1329.5,\n",
       "  370.5,\n",
       "  228.5],\n",
       " 'GO:0051788-response to misfolded protein': [365.0, 112.0],\n",
       " 'GO:0003729-mRNA binding': [366.0,\n",
       "  546.0,\n",
       "  261.0,\n",
       "  243.0,\n",
       "  453.0,\n",
       "  664.0,\n",
       "  909.0,\n",
       "  1115.0,\n",
       "  225.0,\n",
       "  397.0,\n",
       "  456.0,\n",
       "  287.0,\n",
       "  552.0,\n",
       "  792.0,\n",
       "  230.0,\n",
       "  600.0,\n",
       "  230.0,\n",
       "  514.0,\n",
       "  470.0,\n",
       "  572.0,\n",
       "  272.0,\n",
       "  81.0,\n",
       "  832.0,\n",
       "  517.0,\n",
       "  377.0],\n",
       " 'GO:0005763-mitochondrial small ribosomal subunit': [367.5,\n",
       "  249.5,\n",
       "  238.5,\n",
       "  255.5,\n",
       "  125.5,\n",
       "  117.5,\n",
       "  119.5,\n",
       "  77.5,\n",
       "  93.5,\n",
       "  185.5,\n",
       "  261.5,\n",
       "  487.5,\n",
       "  114.5,\n",
       "  254.5,\n",
       "  259.5],\n",
       " 'GO:0000314-organellar small ribosomal subunit': [367.5,\n",
       "  249.5,\n",
       "  238.5,\n",
       "  255.5,\n",
       "  125.5,\n",
       "  117.5,\n",
       "  119.5,\n",
       "  77.5,\n",
       "  93.5,\n",
       "  185.5,\n",
       "  261.5,\n",
       "  487.5,\n",
       "  114.5,\n",
       "  254.5,\n",
       "  259.5],\n",
       " 'GO:0018196-peptidyl-asparagine modification': [369.0,\n",
       "  328.0,\n",
       "  169.0,\n",
       "  438.0,\n",
       "  152.0,\n",
       "  340.0,\n",
       "  305.0,\n",
       "  515.0,\n",
       "  213.0,\n",
       "  715.0,\n",
       "  284.0],\n",
       " 'GO:0004129-cytochrome-c oxidase activity': [371.0,\n",
       "  134.0,\n",
       "  115.0,\n",
       "  153.0,\n",
       "  74.0,\n",
       "  69.0,\n",
       "  42.0,\n",
       "  246.0,\n",
       "  378.0,\n",
       "  126.0,\n",
       "  159.0,\n",
       "  242.0,\n",
       "  443.0,\n",
       "  87.0],\n",
       " 'GO:0015002-heme-copper terminal oxidase activity': [371.0,\n",
       "  134.0,\n",
       "  115.0,\n",
       "  153.0,\n",
       "  74.0,\n",
       "  69.0,\n",
       "  42.0,\n",
       "  246.0,\n",
       "  378.0,\n",
       "  126.0,\n",
       "  159.0,\n",
       "  242.0,\n",
       "  443.0,\n",
       "  87.0],\n",
       " 'GO:0016676-oxidoreductase activity, acting on a heme group of donors, oxygen as acceptor': [371.0,\n",
       "  134.0,\n",
       "  115.0,\n",
       "  153.0,\n",
       "  74.0,\n",
       "  69.0,\n",
       "  42.0,\n",
       "  246.0,\n",
       "  378.0,\n",
       "  126.0,\n",
       "  159.0,\n",
       "  242.0,\n",
       "  443.0,\n",
       "  87.0],\n",
       " 'GO:0033275-actin-myosin filament sliding': [373.5,\n",
       "  107.5,\n",
       "  157.5,\n",
       "  258.5,\n",
       "  509.5,\n",
       "  269.5,\n",
       "  124.5,\n",
       "  91.5,\n",
       "  111.5,\n",
       "  440.5,\n",
       "  153.5,\n",
       "  289.5,\n",
       "  618.5,\n",
       "  224.5,\n",
       "  204.5],\n",
       " 'GO:0030049-muscle filament sliding': [373.5,\n",
       "  107.5,\n",
       "  157.5,\n",
       "  258.5,\n",
       "  509.5,\n",
       "  269.5,\n",
       "  124.5,\n",
       "  91.5,\n",
       "  111.5,\n",
       "  440.5,\n",
       "  153.5,\n",
       "  289.5,\n",
       "  618.5,\n",
       "  224.5,\n",
       "  204.5],\n",
       " 'GO:0005840-ribosome': [375.0,\n",
       "  270.0,\n",
       "  305.0,\n",
       "  470.0,\n",
       "  330.0,\n",
       "  796.0,\n",
       "  344.0,\n",
       "  275.0,\n",
       "  216.0,\n",
       "  825.0,\n",
       "  567.0,\n",
       "  273.0,\n",
       "  503.0,\n",
       "  439.0,\n",
       "  169.0,\n",
       "  267.0,\n",
       "  226.0,\n",
       "  388.0,\n",
       "  341.0,\n",
       "  541.0,\n",
       "  349.0,\n",
       "  162.0,\n",
       "  851.0,\n",
       "  487.0,\n",
       "  300.0],\n",
       " 'GO:0030137-COPI-coated vesicle': [376.0, 202.0, 572.0],\n",
       " 'GO:0019433-triglyceride catabolic process': [377.0,\n",
       "  243.0,\n",
       "  186.0,\n",
       "  168.0,\n",
       "  286.0,\n",
       "  454.0,\n",
       "  210.0,\n",
       "  358.0,\n",
       "  287.0,\n",
       "  499.0],\n",
       " 'GO:0010390-histone monoubiquitination': [378.0,\n",
       "  179.0,\n",
       "  316.0,\n",
       "  622.0,\n",
       "  136.0,\n",
       "  291.0],\n",
       " 'GO:0000083-regulation of transcription involved in G1/S transition of mitotic cell cycle': [379.0,\n",
       "  207.0],\n",
       " 'GO:0034080-CENP-A containing nucleosome assembly': [380.5,\n",
       "  302.5,\n",
       "  196.5,\n",
       "  351.5,\n",
       "  422.5,\n",
       "  708.5,\n",
       "  424.5,\n",
       "  193.5,\n",
       "  186.5,\n",
       "  475.5,\n",
       "  712.5,\n",
       "  69.5,\n",
       "  366.5,\n",
       "  132.5,\n",
       "  286.5,\n",
       "  130.5,\n",
       "  964.5,\n",
       "  170.5,\n",
       "  765.5,\n",
       "  244.5],\n",
       " 'GO:0061641-CENP-A containing chromatin organization': [380.5,\n",
       "  302.5,\n",
       "  196.5,\n",
       "  351.5,\n",
       "  422.5,\n",
       "  708.5,\n",
       "  424.5,\n",
       "  193.5,\n",
       "  186.5,\n",
       "  475.5,\n",
       "  712.5,\n",
       "  69.5,\n",
       "  366.5,\n",
       "  132.5,\n",
       "  286.5,\n",
       "  130.5,\n",
       "  964.5,\n",
       "  170.5,\n",
       "  765.5,\n",
       "  244.5],\n",
       " 'GO:0042273-ribosomal large subunit biogenesis': [382.0,\n",
       "  318.0,\n",
       "  405.0,\n",
       "  469.0,\n",
       "  375.0,\n",
       "  356.0,\n",
       "  538.0,\n",
       "  473.0,\n",
       "  218.0,\n",
       "  629.0,\n",
       "  192.0,\n",
       "  505.0,\n",
       "  358.0,\n",
       "  830.0,\n",
       "  193.0,\n",
       "  95.0,\n",
       "  1156.0,\n",
       "  482.0,\n",
       "  483.0],\n",
       " 'GO:0007566-embryo implantation': [383.0,\n",
       "  396.0,\n",
       "  282.0,\n",
       "  465.0,\n",
       "  217.0,\n",
       "  502.0,\n",
       "  403.0,\n",
       "  168.0,\n",
       "  314.0,\n",
       "  1019.0,\n",
       "  295.0,\n",
       "  410.0,\n",
       "  931.0,\n",
       "  900.0,\n",
       "  634.0,\n",
       "  337.0],\n",
       " 'GO:0006346-methylation-dependent chromatin silencing': 384.0,\n",
       " 'GO:0015935-small ribosomal subunit': [385.0,\n",
       "  260.0,\n",
       "  143.0,\n",
       "  468.0,\n",
       "  280.0,\n",
       "  736.0,\n",
       "  178.0,\n",
       "  218.0,\n",
       "  221.0,\n",
       "  675.0,\n",
       "  413.0,\n",
       "  214.0,\n",
       "  390.0,\n",
       "  258.0,\n",
       "  159.0,\n",
       "  201.0,\n",
       "  203.0,\n",
       "  304.0,\n",
       "  283.0,\n",
       "  635.0,\n",
       "  302.0,\n",
       "  55.0,\n",
       "  633.0,\n",
       "  406.0,\n",
       "  210.0],\n",
       " 'GO:1903671-negative regulation of sprouting angiogenesis': [386.0,\n",
       "  282.0,\n",
       "  46.0,\n",
       "  42.0,\n",
       "  61.0,\n",
       "  242.0,\n",
       "  509.0,\n",
       "  59.0,\n",
       "  210.0,\n",
       "  50.0,\n",
       "  80.0,\n",
       "  73.0,\n",
       "  154.0,\n",
       "  187.0,\n",
       "  67.0,\n",
       "  73.0,\n",
       "  181.0,\n",
       "  383.0,\n",
       "  128.0,\n",
       "  285.0,\n",
       "  21.0,\n",
       "  92.0,\n",
       "  353.0,\n",
       "  56.0],\n",
       " 'GO:0016601-Rac protein signal transduction': 387.0,\n",
       " 'GO:1903672-positive regulation of sprouting angiogenesis': [388.0,\n",
       "  500.0,\n",
       "  61.0,\n",
       "  195.0,\n",
       "  364.0,\n",
       "  564.0,\n",
       "  432.0,\n",
       "  583.0,\n",
       "  585.0,\n",
       "  230.0,\n",
       "  353.0,\n",
       "  785.0,\n",
       "  168.0,\n",
       "  439.0,\n",
       "  382.0,\n",
       "  613.0,\n",
       "  529.0,\n",
       "  209.0,\n",
       "  477.0,\n",
       "  284.0],\n",
       " 'GO:0043276-anoikis': [389.0, 295.0, 431.0],\n",
       " 'GO:0043044-ATP-dependent chromatin remodeling': [390.0,\n",
       "  581.0,\n",
       "  310.0,\n",
       "  340.0,\n",
       "  739.0,\n",
       "  805.0,\n",
       "  1145.0,\n",
       "  787.0,\n",
       "  226.0,\n",
       "  656.0,\n",
       "  1150.0,\n",
       "  188.0,\n",
       "  903.0,\n",
       "  334.0,\n",
       "  471.0,\n",
       "  447.0,\n",
       "  1392.0,\n",
       "  288.0,\n",
       "  1197.0,\n",
       "  381.0],\n",
       " 'GO:0009651-response to salt stress': [391.0, 354.0, 632.0],\n",
       " 'GO:0070849-response to epidermal growth factor': [392.0,\n",
       "  346.0,\n",
       "  570.0,\n",
       "  718.0,\n",
       "  328.0,\n",
       "  511.0,\n",
       "  859.0,\n",
       "  379.0,\n",
       "  585.0,\n",
       "  762.0,\n",
       "  362.0,\n",
       "  762.0,\n",
       "  352.0],\n",
       " 'GO:0002102-podosome': [393.0, 229.0, 361.0, 322.0, 217.0, 338.0],\n",
       " 'GO:0070469-respiratory chain': [394.0,\n",
       "  214.0,\n",
       "  256.0,\n",
       "  228.0,\n",
       "  273.0,\n",
       "  819.0,\n",
       "  319.0,\n",
       "  176.0,\n",
       "  114.0,\n",
       "  375.0,\n",
       "  181.0,\n",
       "  106.0,\n",
       "  263.0,\n",
       "  195.0,\n",
       "  80.0,\n",
       "  133.0,\n",
       "  209.0,\n",
       "  336.0,\n",
       "  196.0,\n",
       "  279.0,\n",
       "  301.0,\n",
       "  47.0,\n",
       "  615.0,\n",
       "  206.0,\n",
       "  181.0],\n",
       " 'GO:1900101-regulation of endoplasmic reticulum unfolded protein response': [395.0,\n",
       "  694.0],\n",
       " 'GO:0070125-mitochondrial translational elongation': [396.0,\n",
       "  230.0,\n",
       "  275.0,\n",
       "  484.0,\n",
       "  312.0,\n",
       "  1219.0,\n",
       "  296.0,\n",
       "  188.0,\n",
       "  203.0,\n",
       "  174.0,\n",
       "  458.0,\n",
       "  614.0,\n",
       "  219.0,\n",
       "  72.0,\n",
       "  386.0,\n",
       "  107.0,\n",
       "  273.0,\n",
       "  248.0,\n",
       "  506.0,\n",
       "  149.0,\n",
       "  51.0,\n",
       "  1383.0,\n",
       "  333.0,\n",
       "  288.0],\n",
       " 'GO:0006958-complement activation, classical pathway': [397.0,\n",
       "  166.0,\n",
       "  81.0,\n",
       "  227.0,\n",
       "  84.0,\n",
       "  246.0,\n",
       "  202.0,\n",
       "  149.0,\n",
       "  97.0,\n",
       "  48.0,\n",
       "  119.0,\n",
       "  352.0,\n",
       "  130.0,\n",
       "  54.0,\n",
       "  66.0,\n",
       "  110.0,\n",
       "  326.0,\n",
       "  71.0,\n",
       "  268.0,\n",
       "  70.0],\n",
       " 'GO:1905369-endopeptidase complex': [398.0,\n",
       "  298.0,\n",
       "  148.0,\n",
       "  177.0,\n",
       "  575.0,\n",
       "  929.0,\n",
       "  405.0,\n",
       "  226.0,\n",
       "  192.0,\n",
       "  221.0,\n",
       "  291.0,\n",
       "  747.0,\n",
       "  270.0,\n",
       "  113.0,\n",
       "  445.0,\n",
       "  153.0,\n",
       "  347.0,\n",
       "  200.0,\n",
       "  489.0,\n",
       "  111.0,\n",
       "  84.0,\n",
       "  1262.0,\n",
       "  204.0,\n",
       "  299.0],\n",
       " 'GO:0051443-positive regulation of ubiquitin-protein transferase activity': [399.0,\n",
       "  443.0],\n",
       " 'GO:0004812-aminoacyl-tRNA ligase activity': [400.5,\n",
       "  386.5,\n",
       "  382.5,\n",
       "  304.5,\n",
       "  182.5,\n",
       "  304.5,\n",
       "  108.5,\n",
       "  139.5,\n",
       "  373.5,\n",
       "  312.5,\n",
       "  184.5,\n",
       "  209.5],\n",
       " 'GO:0016875-ligase activity, forming carbon-oxygen bonds': [400.5,\n",
       "  386.5,\n",
       "  382.5,\n",
       "  304.5,\n",
       "  182.5,\n",
       "  304.5,\n",
       "  108.5,\n",
       "  139.5,\n",
       "  373.5,\n",
       "  312.5,\n",
       "  184.5,\n",
       "  209.5],\n",
       " 'GO:1904358-positive regulation of telomere maintenance via telomere lengthening': [402.0,\n",
       "  300.0,\n",
       "  764.0,\n",
       "  204.0,\n",
       "  387.0],\n",
       " 'GO:0002251-organ or tissue specific immune response': [403.0,\n",
       "  144.0,\n",
       "  245.0,\n",
       "  371.0,\n",
       "  612.0,\n",
       "  112.0,\n",
       "  228.0,\n",
       "  83.0,\n",
       "  118.0,\n",
       "  621.0,\n",
       "  145.0,\n",
       "  167.0,\n",
       "  453.0,\n",
       "  660.0,\n",
       "  743.0,\n",
       "  258.0],\n",
       " 'GO:0002455-humoral immune response mediated by circulating immunoglobulin': [404.0,\n",
       "  139.0,\n",
       "  64.0,\n",
       "  78.0,\n",
       "  198.0,\n",
       "  132.0,\n",
       "  223.0,\n",
       "  365.0,\n",
       "  172.0,\n",
       "  310.0,\n",
       "  119.0,\n",
       "  94.0,\n",
       "  150.0,\n",
       "  257.0,\n",
       "  131.0,\n",
       "  71.0,\n",
       "  98.0,\n",
       "  136.0,\n",
       "  325.0,\n",
       "  274.0,\n",
       "  236.0,\n",
       "  104.0],\n",
       " 'GO:0000502-proteasome complex': [405.0,\n",
       "  300.0,\n",
       "  180.0,\n",
       "  590.0,\n",
       "  942.0,\n",
       "  417.0,\n",
       "  229.0,\n",
       "  195.0,\n",
       "  225.0,\n",
       "  292.0,\n",
       "  667.0,\n",
       "  236.0,\n",
       "  114.0,\n",
       "  455.0,\n",
       "  155.0,\n",
       "  354.0,\n",
       "  203.0,\n",
       "  458.0,\n",
       "  112.0,\n",
       "  89.0,\n",
       "  1159.0,\n",
       "  170.0,\n",
       "  303.0],\n",
       " 'GO:0043537-negative regulation of blood vessel endothelial cell migration': [406.0,\n",
       "  266.0,\n",
       "  92.0,\n",
       "  68.0,\n",
       "  263.0,\n",
       "  250.0,\n",
       "  450.0,\n",
       "  88.0,\n",
       "  489.0,\n",
       "  62.0,\n",
       "  126.0,\n",
       "  82.0,\n",
       "  199.0,\n",
       "  191.0,\n",
       "  84.0,\n",
       "  97.0,\n",
       "  253.0,\n",
       "  381.0,\n",
       "  219.0,\n",
       "  455.0,\n",
       "  30.0,\n",
       "  195.0,\n",
       "  489.0,\n",
       "  155.0],\n",
       " 'GO:0006977-DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest': [407.0,\n",
       "  629.0,\n",
       "  720.0,\n",
       "  290.0,\n",
       "  768.0,\n",
       "  1337.0,\n",
       "  530.0,\n",
       "  416.0,\n",
       "  571.0,\n",
       "  878.0,\n",
       "  628.0,\n",
       "  454.0],\n",
       " 'GO:0098553-lumenal side of endoplasmic reticulum membrane': [408.5,\n",
       "  190.5,\n",
       "  44.5,\n",
       "  148.5,\n",
       "  131.5,\n",
       "  532.5,\n",
       "  161.5,\n",
       "  55.5,\n",
       "  153.5,\n",
       "  140.5,\n",
       "  70.5,\n",
       "  268.5],\n",
       " 'GO:0071556-integral component of lumenal side of endoplasmic reticulum membrane': [408.5,\n",
       "  190.5,\n",
       "  44.5,\n",
       "  148.5,\n",
       "  131.5,\n",
       "  532.5,\n",
       "  161.5,\n",
       "  55.5,\n",
       "  153.5,\n",
       "  140.5,\n",
       "  70.5,\n",
       "  268.5],\n",
       " 'GO:0005746-mitochondrial respiratory chain': [410.0,\n",
       "  215.0,\n",
       "  267.0,\n",
       "  255.0,\n",
       "  257.0,\n",
       "  644.0,\n",
       "  281.0,\n",
       "  210.0,\n",
       "  117.0,\n",
       "  316.0,\n",
       "  182.0,\n",
       "  121.0,\n",
       "  274.0,\n",
       "  184.0,\n",
       "  81.0,\n",
       "  132.0,\n",
       "  185.0,\n",
       "  349.0,\n",
       "  172.0,\n",
       "  301.0,\n",
       "  316.0,\n",
       "  49.0,\n",
       "  644.0,\n",
       "  217.0,\n",
       "  209.0],\n",
       " 'GO:0005640-nuclear outer membrane': 411.0,\n",
       " 'GO:0002042-cell migration involved in sprouting angiogenesis': [412.0,\n",
       "  433.0,\n",
       "  71.0,\n",
       "  70.0,\n",
       "  275.0,\n",
       "  559.0,\n",
       "  532.0,\n",
       "  728.0,\n",
       "  204.0,\n",
       "  400.0,\n",
       "  191.0,\n",
       "  166.0,\n",
       "  285.0,\n",
       "  791.0,\n",
       "  106.0,\n",
       "  577.0,\n",
       "  432.0,\n",
       "  336.0,\n",
       "  721.0,\n",
       "  114.0,\n",
       "  194.0,\n",
       "  556.0,\n",
       "  325.0],\n",
       " 'GO:0006614-SRP-dependent cotranslational protein targeting to membrane': [413.0,\n",
       "  340.0,\n",
       "  516.0,\n",
       "  359.0,\n",
       "  661.0,\n",
       "  61.0,\n",
       "  437.0,\n",
       "  247.0,\n",
       "  1452.0,\n",
       "  655.0,\n",
       "  226.0,\n",
       "  296.0,\n",
       "  392.0,\n",
       "  252.0,\n",
       "  171.0,\n",
       "  393.0,\n",
       "  576.0,\n",
       "  419.0,\n",
       "  608.0,\n",
       "  509.0,\n",
       "  149.0,\n",
       "  462.0,\n",
       "  967.0,\n",
       "  266.0],\n",
       " 'GO:0070979-protein K11-linked ubiquitination': [414.0, 374.0],\n",
       " 'GO:0030532-small nuclear ribonucleoprotein complex': [415.0,\n",
       "  397.0,\n",
       "  265.0,\n",
       "  310.0,\n",
       "  497.0,\n",
       "  657.0,\n",
       "  659.0,\n",
       "  289.0,\n",
       "  292.0,\n",
       "  840.0,\n",
       "  767.0,\n",
       "  1100.0,\n",
       "  464.0,\n",
       "  121.0,\n",
       "  184.0,\n",
       "  406.0,\n",
       "  297.0,\n",
       "  561.0,\n",
       "  140.0,\n",
       "  1487.0,\n",
       "  782.0,\n",
       "  538.0],\n",
       " 'GO:0031055-chromatin remodeling at centromere': [416.0,\n",
       "  315.0,\n",
       "  205.0,\n",
       "  334.0,\n",
       "  279.0,\n",
       "  717.0,\n",
       "  541.0,\n",
       "  306.0,\n",
       "  174.0,\n",
       "  530.0,\n",
       "  690.0,\n",
       "  85.0,\n",
       "  313.0,\n",
       "  144.0,\n",
       "  300.0,\n",
       "  169.0,\n",
       "  871.0,\n",
       "  164.0,\n",
       "  987.0,\n",
       "  357.0],\n",
       " 'GO:0045954-positive regulation of natural killer cell mediated cytotoxicity': [417.0,\n",
       "  191.0],\n",
       " 'GO:1902751-positive regulation of cell cycle G2/M phase transition': [418.0,\n",
       "  435.0],\n",
       " 'GO:0010874-regulation of cholesterol efflux': [419.0,\n",
       "  251.0,\n",
       "  103.0,\n",
       "  221.0,\n",
       "  300.0,\n",
       "  220.0,\n",
       "  108.0,\n",
       "  382.0,\n",
       "  118.0,\n",
       "  50.0,\n",
       "  132.0,\n",
       "  547.0,\n",
       "  131.0,\n",
       "  232.0,\n",
       "  247.0,\n",
       "  553.0,\n",
       "  191.0,\n",
       "  267.0,\n",
       "  74.0],\n",
       " 'GO:0034661-ncRNA catabolic process': [420.0, 453.0, 388.0],\n",
       " 'GO:0005761-mitochondrial ribosome': [421.5,\n",
       "  275.5,\n",
       "  284.5,\n",
       "  545.5,\n",
       "  362.5,\n",
       "  1284.5,\n",
       "  386.5,\n",
       "  192.5,\n",
       "  199.5,\n",
       "  172.5,\n",
       "  494.5,\n",
       "  751.5,\n",
       "  284.5,\n",
       "  73.5,\n",
       "  589.5,\n",
       "  121.5,\n",
       "  364.5,\n",
       "  323.5,\n",
       "  591.5,\n",
       "  173.5,\n",
       "  64.5,\n",
       "  1389.5,\n",
       "  343.5,\n",
       "  291.5],\n",
       " 'GO:0000313-organellar ribosome': [421.5,\n",
       "  275.5,\n",
       "  284.5,\n",
       "  545.5,\n",
       "  362.5,\n",
       "  1284.5,\n",
       "  386.5,\n",
       "  192.5,\n",
       "  199.5,\n",
       "  172.5,\n",
       "  494.5,\n",
       "  751.5,\n",
       "  284.5,\n",
       "  73.5,\n",
       "  589.5,\n",
       "  121.5,\n",
       "  364.5,\n",
       "  323.5,\n",
       "  591.5,\n",
       "  173.5,\n",
       "  64.5,\n",
       "  1389.5,\n",
       "  343.5,\n",
       "  291.5],\n",
       " 'GO:0005689-U12-type spliceosomal complex': [423.0,\n",
       "  206.0,\n",
       "  315.0,\n",
       "  76.0,\n",
       "  202.0],\n",
       " 'GO:0000118-histone deacetylase complex': [424.0,\n",
       "  573.0,\n",
       "  600.0,\n",
       "  1657.0,\n",
       "  270.0,\n",
       "  339.0,\n",
       "  694.0,\n",
       "  343.0,\n",
       "  958.0],\n",
       " 'GO:1905562-regulation of vascular endothelial cell proliferation': [425.5,\n",
       "  264.5,\n",
       "  42.5,\n",
       "  54.5,\n",
       "  149.5,\n",
       "  213.5,\n",
       "  193.5,\n",
       "  454.5,\n",
       "  94.5,\n",
       "  66.5,\n",
       "  53.5,\n",
       "  70.5,\n",
       "  204.5,\n",
       "  205.5,\n",
       "  175.5,\n",
       "  366.5,\n",
       "  276.5,\n",
       "  198.5,\n",
       "  242.5,\n",
       "  19.5,\n",
       "  142.5,\n",
       "  227.5,\n",
       "  121.5],\n",
       " 'GO:0101023-vascular endothelial cell proliferation': [425.5,\n",
       "  264.5,\n",
       "  42.5,\n",
       "  54.5,\n",
       "  149.5,\n",
       "  213.5,\n",
       "  193.5,\n",
       "  454.5,\n",
       "  94.5,\n",
       "  66.5,\n",
       "  53.5,\n",
       "  70.5,\n",
       "  204.5,\n",
       "  205.5,\n",
       "  175.5,\n",
       "  366.5,\n",
       "  276.5,\n",
       "  198.5,\n",
       "  242.5,\n",
       "  19.5,\n",
       "  142.5,\n",
       "  227.5,\n",
       "  121.5],\n",
       " 'GO:0022616-DNA strand elongation': [427.0, 153.0, 130.0, 164.0, 120.0],\n",
       " 'GO:0071364-cellular response to epidermal growth factor stimulus': [428.0,\n",
       "  337.0,\n",
       "  498.0,\n",
       "  654.0,\n",
       "  664.0,\n",
       "  293.0,\n",
       "  571.0,\n",
       "  486.0,\n",
       "  803.0,\n",
       "  550.0,\n",
       "  772.0,\n",
       "  643.0,\n",
       "  319.0],\n",
       " 'GO:0098803-respiratory chain complex': [429.0,\n",
       "  204.0,\n",
       "  273.0,\n",
       "  224.0,\n",
       "  666.0,\n",
       "  321.0,\n",
       "  200.0,\n",
       "  113.0,\n",
       "  296.0,\n",
       "  179.0,\n",
       "  119.0,\n",
       "  228.0,\n",
       "  192.0,\n",
       "  75.0,\n",
       "  128.0,\n",
       "  169.0,\n",
       "  371.0,\n",
       "  176.0,\n",
       "  282.0,\n",
       "  308.0,\n",
       "  41.0,\n",
       "  653.0,\n",
       "  266.0,\n",
       "  199.0],\n",
       " 'GO:0002181-cytoplasmic translation': [430.0,\n",
       "  326.0,\n",
       "  281.0,\n",
       "  606.0,\n",
       "  763.0,\n",
       "  463.0,\n",
       "  264.0,\n",
       "  302.0,\n",
       "  599.0,\n",
       "  1133.0,\n",
       "  686.0,\n",
       "  195.0,\n",
       "  278.0,\n",
       "  189.0,\n",
       "  487.0,\n",
       "  416.0,\n",
       "  1007.0,\n",
       "  309.0,\n",
       "  113.0,\n",
       "  1375.0,\n",
       "  611.0,\n",
       "  373.0],\n",
       " 'GO:0051204-protein insertion into mitochondrial membrane': [431.0, 279.0],\n",
       " 'GO:0042775-mitochondrial ATP synthesis coupled electron transport': [432.0,\n",
       "  203.0,\n",
       "  227.0,\n",
       "  187.0,\n",
       "  767.0,\n",
       "  348.0,\n",
       "  182.0,\n",
       "  135.0,\n",
       "  237.0,\n",
       "  180.0,\n",
       "  120.0,\n",
       "  239.0,\n",
       "  164.0,\n",
       "  79.0,\n",
       "  104.0,\n",
       "  143.0,\n",
       "  325.0,\n",
       "  175.0,\n",
       "  339.0,\n",
       "  240.0,\n",
       "  43.0,\n",
       "  727.0,\n",
       "  193.0,\n",
       "  207.0],\n",
       " 'GO:0002228-natural killer cell mediated immunity': [433.0,\n",
       "  759.0,\n",
       "  1114.0,\n",
       "  533.0,\n",
       "  577.0,\n",
       "  484.0,\n",
       "  1582.0,\n",
       "  640.0,\n",
       "  346.0,\n",
       "  1345.0,\n",
       "  971.0,\n",
       "  1364.0,\n",
       "  504.0],\n",
       " 'GO:0002082-regulation of oxidative phosphorylation': [434.0,\n",
       "  88.0,\n",
       "  290.0,\n",
       "  192.0,\n",
       "  232.0,\n",
       "  229.0,\n",
       "  302.0,\n",
       "  172.0,\n",
       "  232.0,\n",
       "  579.0,\n",
       "  553.0,\n",
       "  191.0],\n",
       " 'GO:0000027-ribosomal large subunit assembly': [435.0, 186.0, 508.0, 222.0],\n",
       " 'GO:0006363-termination of RNA polymerase I transcription': [436.0,\n",
       "  409.0,\n",
       "  372.0,\n",
       "  623.0],\n",
       " 'GO:0070069-cytochrome complex': [437.0,\n",
       "  235.0,\n",
       "  131.0,\n",
       "  274.0,\n",
       "  355.0,\n",
       "  247.0,\n",
       "  128.0,\n",
       "  78.0,\n",
       "  231.0,\n",
       "  84.0,\n",
       "  207.0,\n",
       "  272.0,\n",
       "  40.0,\n",
       "  83.0,\n",
       "  244.0,\n",
       "  256.0,\n",
       "  154.0,\n",
       "  334.0,\n",
       "  24.0,\n",
       "  390.0,\n",
       "  242.0,\n",
       "  119.0],\n",
       " 'GO:0006515-protein quality control for misfolded or incompletely synthesized proteins': [438.0,\n",
       "  274.0],\n",
       " 'GO:0006613-cotranslational protein targeting to membrane': [439.0,\n",
       "  359.0,\n",
       "  357.0,\n",
       "  541.0,\n",
       "  368.0,\n",
       "  640.0,\n",
       "  87.0,\n",
       "  474.0,\n",
       "  229.0,\n",
       "  1427.0,\n",
       "  607.0,\n",
       "  224.0,\n",
       "  287.0,\n",
       "  383.0,\n",
       "  245.0,\n",
       "  182.0,\n",
       "  419.0,\n",
       "  645.0,\n",
       "  428.0,\n",
       "  631.0,\n",
       "  494.0,\n",
       "  144.0,\n",
       "  550.0,\n",
       "  910.0,\n",
       "  265.0],\n",
       " 'GO:0008135-translation factor activity, RNA binding': [440.0,\n",
       "  350.0,\n",
       "  268.0,\n",
       "  374.0,\n",
       "  711.0,\n",
       "  806.0,\n",
       "  488.0,\n",
       "  470.0,\n",
       "  1013.0,\n",
       "  199.0,\n",
       "  704.0,\n",
       "  235.0,\n",
       "  431.0,\n",
       "  454.0,\n",
       "  1421.0,\n",
       "  219.0,\n",
       "  131.0,\n",
       "  480.0,\n",
       "  516.0],\n",
       " 'GO:0043022-ribosome binding': [441.0,\n",
       "  391.0,\n",
       "  553.0,\n",
       "  410.0,\n",
       "  329.0,\n",
       "  757.0,\n",
       "  992.0,\n",
       "  224.0,\n",
       "  253.0,\n",
       "  472.0,\n",
       "  504.0,\n",
       "  157.0,\n",
       "  473.0,\n",
       "  477.0],\n",
       " 'GO:0050858-negative regulation of antigen receptor-mediated signaling pathway': [442.0,\n",
       "  350.0,\n",
       "  496.0,\n",
       "  501.0,\n",
       "  458.0],\n",
       " 'GO:0090169-regulation of spindle assembly': [443.0, 147.0],\n",
       " 'GO:1903955-positive regulation of protein targeting to mitochondrion': [444.0,\n",
       "  322.0,\n",
       "  379.0,\n",
       "  469.0,\n",
       "  334.0,\n",
       "  897.0,\n",
       "  233.0,\n",
       "  187.0,\n",
       "  286.0,\n",
       "  638.0,\n",
       "  216.0],\n",
       " 'GO:0050999-regulation of nitric-oxide synthase activity': [445.0,\n",
       "  525.0,\n",
       "  324.0,\n",
       "  761.0,\n",
       "  412.0,\n",
       "  586.0,\n",
       "  646.0,\n",
       "  895.0,\n",
       "  211.0,\n",
       "  243.0,\n",
       "  497.0,\n",
       "  750.0,\n",
       "  182.0,\n",
       "  621.0,\n",
       "  763.0,\n",
       "  398.0],\n",
       " 'GO:0000460-maturation of 5.8S rRNA': [446.0, 149.0, 520.0, 102.0],\n",
       " 'GO:0098800-inner mitochondrial membrane protein complex': [447.0,\n",
       "  253.0,\n",
       "  317.0,\n",
       "  383.0,\n",
       "  236.0,\n",
       "  812.0,\n",
       "  401.0,\n",
       "  213.0,\n",
       "  194.0,\n",
       "  440.0,\n",
       "  250.0,\n",
       "  227.0,\n",
       "  394.0,\n",
       "  223.0,\n",
       "  117.0,\n",
       "  176.0,\n",
       "  157.0,\n",
       "  328.0,\n",
       "  197.0,\n",
       "  374.0,\n",
       "  327.0,\n",
       "  42.0,\n",
       "  950.0,\n",
       "  359.0,\n",
       "  287.0],\n",
       " 'GO:0034728-nucleosome organization': [448.0,\n",
       "  586.0,\n",
       "  199.0,\n",
       "  261.0,\n",
       "  819.0,\n",
       "  598.0,\n",
       "  1331.0,\n",
       "  760.0,\n",
       "  419.0,\n",
       "  289.0,\n",
       "  430.0,\n",
       "  961.0,\n",
       "  1157.0,\n",
       "  170.0,\n",
       "  948.0,\n",
       "  376.0,\n",
       "  597.0,\n",
       "  668.0,\n",
       "  1068.0,\n",
       "  326.0,\n",
       "  1176.0,\n",
       "  580.0,\n",
       "  793.0],\n",
       " 'GO:0016574-histone ubiquitination': [449.0,\n",
       "  318.0,\n",
       "  520.0,\n",
       "  823.0,\n",
       "  435.0,\n",
       "  855.0,\n",
       "  360.0,\n",
       "  818.0,\n",
       "  152.0,\n",
       "  532.0],\n",
       " 'GO:0071013-catalytic step 2 spliceosome': [450.0,\n",
       "  339.0,\n",
       "  225.0,\n",
       "  439.0,\n",
       "  654.0,\n",
       "  626.0,\n",
       "  858.0,\n",
       "  303.0,\n",
       "  192.0,\n",
       "  511.0,\n",
       "  160.0,\n",
       "  841.0,\n",
       "  217.0,\n",
       "  276.0,\n",
       "  264.0,\n",
       "  724.0,\n",
       "  139.0,\n",
       "  108.0,\n",
       "  330.0,\n",
       "  822.0],\n",
       " 'GO:0043928-exonucleolytic nuclear-transcribed mRNA catabolic process involved in deadenylation-dependent decay': [451.0,\n",
       "  666.0],\n",
       " 'GO:0071007-U2-type catalytic step 2 spliceosome': [452.0,\n",
       "  232.0,\n",
       "  190.0,\n",
       "  495.0,\n",
       "  254.0,\n",
       "  414.0,\n",
       "  84.0,\n",
       "  369.0],\n",
       " 'GO:0006418-tRNA aminoacylation for protein translation': [453.0,\n",
       "  395.0,\n",
       "  391.0,\n",
       "  344.0,\n",
       "  229.0,\n",
       "  344.0,\n",
       "  98.0,\n",
       "  124.0,\n",
       "  387.0,\n",
       "  269.0,\n",
       "  143.0,\n",
       "  44.0,\n",
       "  290.0],\n",
       " 'GO:0046685-response to arsenic-containing substance': [454.0,\n",
       "  309.0,\n",
       "  454.0,\n",
       "  379.0,\n",
       "  308.0,\n",
       "  214.0,\n",
       "  411.0,\n",
       "  263.0,\n",
       "  175.0,\n",
       "  527.0],\n",
       " 'GO:0045047-protein targeting to ER': [455.0,\n",
       "  372.0,\n",
       "  368.0,\n",
       "  619.0,\n",
       "  385.0,\n",
       "  715.0,\n",
       "  94.0,\n",
       "  452.0,\n",
       "  235.0,\n",
       "  1524.0,\n",
       "  582.0,\n",
       "  284.0,\n",
       "  300.0,\n",
       "  428.0,\n",
       "  277.0,\n",
       "  190.0,\n",
       "  456.0,\n",
       "  738.0,\n",
       "  477.0,\n",
       "  749.0,\n",
       "  512.0,\n",
       "  171.0,\n",
       "  523.0,\n",
       "  995.0,\n",
       "  324.0],\n",
       " 'GO:0017091-AU-rich element binding': 456.0,\n",
       " 'GO:0032768-regulation of monooxygenase activity': [457.0,\n",
       "  556.0,\n",
       "  149.0,\n",
       "  271.0,\n",
       "  637.0,\n",
       "  614.0,\n",
       "  464.0,\n",
       "  562.0,\n",
       "  1106.0,\n",
       "  579.0,\n",
       "  986.0,\n",
       "  295.0,\n",
       "  225.0,\n",
       "  531.0,\n",
       "  226.0,\n",
       "  478.0,\n",
       "  705.0,\n",
       "  244.0,\n",
       "  494.0,\n",
       "  600.0,\n",
       "  449.0],\n",
       " 'GO:0042573-retinoic acid metabolic process': [458.0,\n",
       "  132.0,\n",
       "  45.0,\n",
       "  212.0,\n",
       "  136.0,\n",
       "  266.0,\n",
       "  384.0,\n",
       "  184.0,\n",
       "  78.0,\n",
       "  238.0,\n",
       "  366.0,\n",
       "  75.0,\n",
       "  205.0,\n",
       "  86.0,\n",
       "  311.0,\n",
       "  466.0,\n",
       "  61.0],\n",
       " 'GO:0003743-translation initiation factor activity': [459.0,\n",
       "  296.0,\n",
       "  358.0,\n",
       "  583.0,\n",
       "  675.0,\n",
       "  453.0,\n",
       "  463.0,\n",
       "  988.0,\n",
       "  206.0,\n",
       "  194.0,\n",
       "  340.0,\n",
       "  421.0,\n",
       "  207.0,\n",
       "  514.0],\n",
       " 'GO:0046464-acylglycerol catabolic process': [460.5,\n",
       "  224.5,\n",
       "  297.5,\n",
       "  208.5,\n",
       "  189.5,\n",
       "  305.5,\n",
       "  484.5,\n",
       "  273.5,\n",
       "  247.5,\n",
       "  763.5,\n",
       "  573.5],\n",
       " 'GO:0046461-neutral lipid catabolic process': [460.5,\n",
       "  224.5,\n",
       "  297.5,\n",
       "  208.5,\n",
       "  189.5,\n",
       "  305.5,\n",
       "  484.5,\n",
       "  273.5,\n",
       "  247.5,\n",
       "  763.5,\n",
       "  573.5],\n",
       " 'GO:0045940-positive regulation of steroid metabolic process': [462.0,\n",
       "  128.0,\n",
       "  241.0,\n",
       "  267.0,\n",
       "  210.0,\n",
       "  99.0,\n",
       "  549.0,\n",
       "  83.0,\n",
       "  138.0,\n",
       "  210.0,\n",
       "  224.0,\n",
       "  477.0,\n",
       "  152.0],\n",
       " 'GO:0032206-positive regulation of telomere maintenance': [463.0,\n",
       "  671.0,\n",
       "  373.0,\n",
       "  1770.0,\n",
       "  466.0,\n",
       "  382.0,\n",
       "  440.0],\n",
       " 'GO:0042773-ATP synthesis coupled electron transport': [464.0,\n",
       "  208.0,\n",
       "  251.0,\n",
       "  190.0,\n",
       "  837.0,\n",
       "  347.0,\n",
       "  208.0,\n",
       "  145.0,\n",
       "  251.0,\n",
       "  192.0,\n",
       "  125.0,\n",
       "  256.0,\n",
       "  173.0,\n",
       "  82.0,\n",
       "  111.0,\n",
       "  160.0,\n",
       "  344.0,\n",
       "  184.0,\n",
       "  359.0,\n",
       "  257.0,\n",
       "  46.0,\n",
       "  774.0,\n",
       "  203.0,\n",
       "  223.0],\n",
       " 'GO:0006414-translational elongation': [465.0,\n",
       "  316.0,\n",
       "  314.0,\n",
       "  563.0,\n",
       "  457.0,\n",
       "  1312.0,\n",
       "  521.0,\n",
       "  341.0,\n",
       "  231.0,\n",
       "  280.0,\n",
       "  483.0,\n",
       "  798.0,\n",
       "  408.0,\n",
       "  153.0,\n",
       "  348.0,\n",
       "  191.0,\n",
       "  421.0,\n",
       "  349.0,\n",
       "  718.0,\n",
       "  222.0,\n",
       "  77.0,\n",
       "  1492.0,\n",
       "  456.0,\n",
       "  312.0],\n",
       " 'GO:0017069-snRNA binding': [466.0,\n",
       "  164.0,\n",
       "  471.0,\n",
       "  439.0,\n",
       "  685.0,\n",
       "  171.0,\n",
       "  241.0,\n",
       "  489.0,\n",
       "  409.0,\n",
       "  166.0,\n",
       "  616.0],\n",
       " 'GO:0032435-negative regulation of proteasomal ubiquitin-dependent protein catabolic process': 467.0,\n",
       " 'GO:0060706-cell differentiation involved in embryonic placenta development': [468.0,\n",
       "  199.0],\n",
       " 'GO:0051205-protein insertion into membrane': [469.0, 714.0, 584.0],\n",
       " 'GO:0000791-euchromatin': [470.0, 324.0, 545.0, 197.0],\n",
       " 'GO:0000291-nuclear-transcribed mRNA catabolic process, exonucleolytic': [471.0,\n",
       "  721.0],\n",
       " 'GO:0072599-establishment of protein localization to endoplasmic reticulum': [472.0,\n",
       "  379.0,\n",
       "  385.0,\n",
       "  637.0,\n",
       "  433.0,\n",
       "  875.0,\n",
       "  121.0,\n",
       "  568.0,\n",
       "  238.0,\n",
       "  1535.0,\n",
       "  580.0,\n",
       "  303.0,\n",
       "  354.0,\n",
       "  438.0,\n",
       "  289.0,\n",
       "  231.0,\n",
       "  464.0,\n",
       "  757.0,\n",
       "  494.0,\n",
       "  757.0,\n",
       "  523.0,\n",
       "  166.0,\n",
       "  572.0,\n",
       "  1021.0,\n",
       "  357.0],\n",
       " 'GO:0002717-positive regulation of natural killer cell mediated immunity': [473.0,\n",
       "  466.0,\n",
       "  234.0],\n",
       " 'GO:0008608-attachment of spindle microtubules to kinetochore': [474.0,\n",
       "  295.0,\n",
       "  92.0],\n",
       " 'GO:0000152-nuclear ubiquitin ligase complex': [475.0,\n",
       "  412.0,\n",
       "  531.0,\n",
       "  794.0,\n",
       "  266.0,\n",
       "  375.0,\n",
       "  380.0,\n",
       "  261.0],\n",
       " 'GO:0034508-centromere complex assembly': [476.0,\n",
       "  413.0,\n",
       "  185.0,\n",
       "  447.0,\n",
       "  389.0,\n",
       "  682.0,\n",
       "  528.0,\n",
       "  323.0,\n",
       "  276.0,\n",
       "  702.0,\n",
       "  890.0,\n",
       "  123.0,\n",
       "  508.0,\n",
       "  131.0,\n",
       "  397.0,\n",
       "  271.0,\n",
       "  1053.0,\n",
       "  183.0,\n",
       "  1163.0,\n",
       "  449.0],\n",
       " 'GO:0030515-snoRNA binding': [477.0, 132.0, 458.0, 71.0, 92.0],\n",
       " 'GO:0042267-natural killer cell mediated cytotoxicity': [478.0,\n",
       "  700.0,\n",
       "  895.0,\n",
       "  634.0,\n",
       "  579.0,\n",
       "  479.0,\n",
       "  1490.0,\n",
       "  589.0,\n",
       "  343.0,\n",
       "  1006.0,\n",
       "  501.0],\n",
       " 'GO:0046966-thyroid hormone receptor binding': [479.0,\n",
       "  203.0,\n",
       "  307.0,\n",
       "  220.0,\n",
       "  252.0,\n",
       "  586.0,\n",
       "  310.0],\n",
       " 'GO:0034381-plasma lipoprotein particle clearance': [480.0,\n",
       "  375.0,\n",
       "  82.0,\n",
       "  283.0,\n",
       "  234.0,\n",
       "  518.0,\n",
       "  433.0,\n",
       "  993.0,\n",
       "  193.0,\n",
       "  410.0,\n",
       "  354.0,\n",
       "  137.0,\n",
       "  264.0,\n",
       "  333.0,\n",
       "  174.0,\n",
       "  490.0,\n",
       "  261.0,\n",
       "  235.0,\n",
       "  508.0,\n",
       "  816.0,\n",
       "  311.0,\n",
       "  325.0,\n",
       "  475.0,\n",
       "  262.0],\n",
       " 'GO:0006413-translational initiation': [481.0,\n",
       "  353.0,\n",
       "  339.0,\n",
       "  511.0,\n",
       "  568.0,\n",
       "  681.0,\n",
       "  396.0,\n",
       "  379.0,\n",
       "  339.0,\n",
       "  861.0,\n",
       "  887.0,\n",
       "  447.0,\n",
       "  991.0,\n",
       "  708.0,\n",
       "  269.0,\n",
       "  475.0,\n",
       "  356.0,\n",
       "  470.0,\n",
       "  425.0,\n",
       "  980.0,\n",
       "  438.0,\n",
       "  123.0,\n",
       "  1059.0,\n",
       "  604.0,\n",
       "  348.0],\n",
       " 'GO:0002479-antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent': [482.0,\n",
       "  263.0,\n",
       "  137.0,\n",
       "  267.0,\n",
       "  393.0,\n",
       "  1202.0,\n",
       "  376.0,\n",
       "  117.0,\n",
       "  223.0,\n",
       "  454.0,\n",
       "  267.0,\n",
       "  175.0,\n",
       "  317.0,\n",
       "  288.0,\n",
       "  134.0,\n",
       "  311.0,\n",
       "  274.0,\n",
       "  327.0,\n",
       "  145.0,\n",
       "  310.0,\n",
       "  263.0,\n",
       "  456.0,\n",
       "  194.0,\n",
       "  307.0],\n",
       " 'GO:0050779-RNA destabilization': [483.0, 422.0],\n",
       " 'GO:0016712-oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, reduced flavin or flavoprotein as one donor, and incorporation of one atom of oxygen': [484.0,\n",
       "  24.0,\n",
       "  90.0,\n",
       "  203.0,\n",
       "  303.0,\n",
       "  103.0,\n",
       "  119.0,\n",
       "  185.0,\n",
       "  265.0,\n",
       "  786.0,\n",
       "  188.0,\n",
       "  77.0,\n",
       "  257.0],\n",
       " 'GO:0000780-condensed nuclear chromosome, centromeric region': [485.0,\n",
       "  143.0,\n",
       "  201.0,\n",
       "  134.0,\n",
       "  155.0,\n",
       "  421.0],\n",
       " 'GO:0006334-nucleosome assembly': [486.0,\n",
       "  481.0,\n",
       "  180.0,\n",
       "  250.0,\n",
       "  569.0,\n",
       "  500.0,\n",
       "  1153.0,\n",
       "  737.0,\n",
       "  396.0,\n",
       "  356.0,\n",
       "  285.0,\n",
       "  617.0,\n",
       "  1224.0,\n",
       "  141.0,\n",
       "  633.0,\n",
       "  319.0,\n",
       "  580.0,\n",
       "  599.0,\n",
       "  1125.0,\n",
       "  304.0,\n",
       "  1172.0,\n",
       "  792.0,\n",
       "  569.0],\n",
       " 'GO:0051354-negative regulation of oxidoreductase activity': [487.0,\n",
       "  293.0,\n",
       "  50.0,\n",
       "  152.0,\n",
       "  367.0,\n",
       "  258.0,\n",
       "  286.0,\n",
       "  449.0,\n",
       "  169.0,\n",
       "  160.0,\n",
       "  482.0,\n",
       "  242.0,\n",
       "  303.0,\n",
       "  230.0],\n",
       " 'GO:0007062-sister chromatid cohesion': [488.0,\n",
       "  307.0,\n",
       "  618.0,\n",
       "  674.0,\n",
       "  246.0,\n",
       "  1788.0,\n",
       "  396.0,\n",
       "  726.0,\n",
       "  605.0,\n",
       "  282.0,\n",
       "  1236.0],\n",
       " 'GO:1903670-regulation of sprouting angiogenesis': [489.0,\n",
       "  417.0,\n",
       "  73.0,\n",
       "  137.0,\n",
       "  308.0,\n",
       "  499.0,\n",
       "  548.0,\n",
       "  940.0,\n",
       "  173.0,\n",
       "  526.0,\n",
       "  146.0,\n",
       "  174.0,\n",
       "  190.0,\n",
       "  774.0,\n",
       "  492.0,\n",
       "  145.0,\n",
       "  213.0,\n",
       "  473.0,\n",
       "  580.0,\n",
       "  426.0,\n",
       "  459.0,\n",
       "  71.0,\n",
       "  240.0,\n",
       "  433.0,\n",
       "  201.0],\n",
       " 'GO:0034502-protein localization to chromosome': [490.0,\n",
       "  561.0,\n",
       "  631.0,\n",
       "  873.0,\n",
       "  635.0,\n",
       "  1479.0,\n",
       "  435.0,\n",
       "  263.0,\n",
       "  1372.0,\n",
       "  234.0,\n",
       "  332.0,\n",
       "  451.0,\n",
       "  590.0,\n",
       "  1132.0,\n",
       "  196.0,\n",
       "  94.0,\n",
       "  293.0,\n",
       "  678.0],\n",
       " 'GO:0042306-regulation of protein import into nucleus': [491.0,\n",
       "  613.0,\n",
       "  427.0,\n",
       "  630.0,\n",
       "  495.0,\n",
       "  357.0,\n",
       "  338.0,\n",
       "  392.0,\n",
       "  708.0,\n",
       "  797.0,\n",
       "  400.0,\n",
       "  474.0,\n",
       "  610.0,\n",
       "  1178.0,\n",
       "  1158.0,\n",
       "  548.0,\n",
       "  364.0],\n",
       " 'GO:0042769-DNA damage response, detection of DNA damage': [492.0,\n",
       "  295.0,\n",
       "  565.0,\n",
       "  335.0,\n",
       "  316.0,\n",
       "  278.0,\n",
       "  529.0,\n",
       "  208.0,\n",
       "  261.0,\n",
       "  134.0,\n",
       "  336.0],\n",
       " 'GO:0005684-U2-type spliceosomal complex': [493.0,\n",
       "  384.0,\n",
       "  430.0,\n",
       "  691.0,\n",
       "  577.0,\n",
       "  1100.0,\n",
       "  375.0,\n",
       "  290.0,\n",
       "  570.0,\n",
       "  168.0,\n",
       "  299.0,\n",
       "  312.0,\n",
       "  318.0,\n",
       "  752.0,\n",
       "  144.0,\n",
       "  88.0,\n",
       "  375.0],\n",
       " 'GO:0006953-acute-phase response': [494.0,\n",
       "  351.0,\n",
       "  58.0,\n",
       "  328.0,\n",
       "  165.0,\n",
       "  318.0,\n",
       "  426.0,\n",
       "  628.0,\n",
       "  177.0,\n",
       "  620.0,\n",
       "  256.0,\n",
       "  145.0,\n",
       "  514.0,\n",
       "  99.0,\n",
       "  477.0,\n",
       "  210.0,\n",
       "  406.0,\n",
       "  271.0,\n",
       "  377.0,\n",
       "  327.0],\n",
       " 'GO:0019080-viral gene expression': [495.0,\n",
       "  482.0,\n",
       "  410.0,\n",
       "  557.0,\n",
       "  494.0,\n",
       "  704.0,\n",
       "  400.0,\n",
       "  431.0,\n",
       "  285.0,\n",
       "  790.0,\n",
       "  802.0,\n",
       "  506.0,\n",
       "  836.0,\n",
       "  696.0,\n",
       "  325.0,\n",
       "  560.0,\n",
       "  365.0,\n",
       "  414.0,\n",
       "  539.0,\n",
       "  957.0,\n",
       "  522.0,\n",
       "  129.0,\n",
       "  837.0,\n",
       "  692.0,\n",
       "  440.0],\n",
       " 'GO:0006734-NADH metabolic process': [496.0,\n",
       "  259.0,\n",
       "  154.0,\n",
       "  378.0,\n",
       "  261.0,\n",
       "  332.0,\n",
       "  751.0,\n",
       "  126.0,\n",
       "  247.0,\n",
       "  165.0,\n",
       "  268.0,\n",
       "  84.0,\n",
       "  652.0,\n",
       "  228.0,\n",
       "  702.0,\n",
       "  579.0,\n",
       "  273.0],\n",
       " 'GO:0006521-regulation of cellular amino acid metabolic process': [497.0,\n",
       "  324.0,\n",
       "  181.0,\n",
       "  311.0,\n",
       "  329.0,\n",
       "  150.0,\n",
       "  159.0,\n",
       "  269.0,\n",
       "  263.0,\n",
       "  430.0,\n",
       "  210.0,\n",
       "  115.0,\n",
       "  362.0,\n",
       "  166.0,\n",
       "  294.0,\n",
       "  213.0,\n",
       "  450.0,\n",
       "  128.0,\n",
       "  689.0,\n",
       "  138.0,\n",
       "  212.0],\n",
       " 'GO:0060968-regulation of gene silencing': [498.0,\n",
       "  724.0,\n",
       "  555.0,\n",
       "  475.0,\n",
       "  1140.0,\n",
       "  788.0,\n",
       "  827.0,\n",
       "  616.0,\n",
       "  455.0,\n",
       "  380.0,\n",
       "  830.0,\n",
       "  1290.0,\n",
       "  1062.0,\n",
       "  362.0,\n",
       "  902.0,\n",
       "  355.0,\n",
       "  504.0,\n",
       "  631.0,\n",
       "  1546.0,\n",
       "  293.0,\n",
       "  1615.0,\n",
       "  715.0,\n",
       "  895.0],\n",
       " 'GO:0042056-chemoattractant activity': [499.0,\n",
       "  165.0,\n",
       "  225.0,\n",
       "  365.0,\n",
       "  404.0,\n",
       "  253.0,\n",
       "  461.0,\n",
       "  226.0,\n",
       "  74.0,\n",
       "  438.0,\n",
       "  723.0,\n",
       "  108.0,\n",
       "  278.0,\n",
       "  221.0,\n",
       "  397.0,\n",
       "  519.0,\n",
       "  365.0],\n",
       " 'GO:0016655-oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor': [500.0,\n",
       "  196.0,\n",
       "  393.0,\n",
       "  216.0,\n",
       "  732.0,\n",
       "  381.0,\n",
       "  245.0,\n",
       "  164.0,\n",
       "  224.0,\n",
       "  145.0,\n",
       "  115.0,\n",
       "  233.0,\n",
       "  162.0,\n",
       "  71.0,\n",
       "  119.0,\n",
       "  178.0,\n",
       "  413.0,\n",
       "  157.0,\n",
       "  262.0,\n",
       "  341.0,\n",
       "  61.0,\n",
       "  739.0,\n",
       "  263.0,\n",
       "  282.0],\n",
       " 'GO:0032543-mitochondrial translation': [501.0,\n",
       "  383.0,\n",
       "  462.0,\n",
       "  804.0,\n",
       "  547.0,\n",
       "  1422.0,\n",
       "  529.0,\n",
       "  262.0,\n",
       "  236.0,\n",
       "  230.0,\n",
       "  699.0,\n",
       "  1187.0,\n",
       "  517.0,\n",
       "  111.0,\n",
       "  788.0,\n",
       "  162.0,\n",
       "  393.0,\n",
       "  299.0,\n",
       "  947.0,\n",
       "  165.0,\n",
       "  72.0,\n",
       "  1827.0,\n",
       "  350.0,\n",
       "  439.0],\n",
       " 'GO:0046688-response to copper ion': [502.0,\n",
       "  279.0,\n",
       "  330.0,\n",
       "  210.0,\n",
       "  403.0,\n",
       "  534.0,\n",
       "  110.0,\n",
       "  552.0,\n",
       "  299.0,\n",
       "  184.0,\n",
       "  455.0,\n",
       "  214.0,\n",
       "  267.0,\n",
       "  423.0,\n",
       "  263.0,\n",
       "  192.0,\n",
       "  187.0,\n",
       "  233.0],\n",
       " 'GO:0070229-negative regulation of lymphocyte apoptotic process': [503.0,\n",
       "  167.0,\n",
       "  279.0,\n",
       "  249.0,\n",
       "  246.0,\n",
       "  224.0,\n",
       "  58.0,\n",
       "  289.0,\n",
       "  427.0,\n",
       "  593.0],\n",
       " 'GO:0033047-regulation of mitotic sister chromatid segregation': [504.0,\n",
       "  642.0,\n",
       "  280.0,\n",
       "  1269.0,\n",
       "  894.0,\n",
       "  621.0,\n",
       "  337.0,\n",
       "  1328.0,\n",
       "  221.0,\n",
       "  295.0,\n",
       "  568.0,\n",
       "  486.0,\n",
       "  998.0,\n",
       "  205.0,\n",
       "  647.0],\n",
       " 'GO:0019083-viral transcription': [505.0,\n",
       "  505.0,\n",
       "  429.0,\n",
       "  667.0,\n",
       "  479.0,\n",
       "  741.0,\n",
       "  343.0,\n",
       "  423.0,\n",
       "  290.0,\n",
       "  883.0,\n",
       "  796.0,\n",
       "  489.0,\n",
       "  786.0,\n",
       "  736.0,\n",
       "  331.0,\n",
       "  572.0,\n",
       "  418.0,\n",
       "  395.0,\n",
       "  561.0,\n",
       "  904.0,\n",
       "  547.0,\n",
       "  134.0,\n",
       "  812.0,\n",
       "  729.0,\n",
       "  409.0],\n",
       " 'GO:0006362-transcription elongation from RNA polymerase I promoter': [506.0,\n",
       "  326.0],\n",
       " 'GO:0061980-regulatory RNA binding': [507.0, 374.0, 131.0],\n",
       " 'GO:0000803-sex chromosome': [508.0, 840.0],\n",
       " 'GO:0010837-regulation of keratinocyte proliferation': [509.0, 753.0, 253.0],\n",
       " 'GO:1905368-peptidase complex': [510.0,\n",
       "  304.0,\n",
       "  184.0,\n",
       "  215.0,\n",
       "  656.0,\n",
       "  870.0,\n",
       "  472.0,\n",
       "  301.0,\n",
       "  260.0,\n",
       "  318.0,\n",
       "  356.0,\n",
       "  1018.0,\n",
       "  391.0,\n",
       "  140.0,\n",
       "  481.0,\n",
       "  171.0,\n",
       "  427.0,\n",
       "  227.0,\n",
       "  815.0,\n",
       "  135.0,\n",
       "  91.0,\n",
       "  1608.0,\n",
       "  241.0,\n",
       "  318.0],\n",
       " 'GO:0000184-nuclear-transcribed mRNA catabolic process, nonsense-mediated decay': [511.0,\n",
       "  309.0,\n",
       "  609.0,\n",
       "  336.0,\n",
       "  504.0,\n",
       "  96.0,\n",
       "  333.0,\n",
       "  304.0,\n",
       "  1173.0,\n",
       "  260.0,\n",
       "  522.0,\n",
       "  612.0,\n",
       "  317.0,\n",
       "  318.0,\n",
       "  426.0,\n",
       "  539.0,\n",
       "  487.0,\n",
       "  1022.0,\n",
       "  557.0,\n",
       "  112.0,\n",
       "  730.0,\n",
       "  805.0,\n",
       "  390.0],\n",
       " 'GO:0072431-signal transduction involved in mitotic G1 DNA damage checkpoint': [512.5,\n",
       "  482.5,\n",
       "  596.5,\n",
       "  701.5,\n",
       "  286.5,\n",
       "  737.5,\n",
       "  1304.5,\n",
       "  402.5,\n",
       "  566.5,\n",
       "  933.5,\n",
       "  596.5,\n",
       "  450.5],\n",
       " 'GO:1902400-intracellular signal transduction involved in G1 DNA damage checkpoint': [512.5,\n",
       "  482.5,\n",
       "  596.5,\n",
       "  701.5,\n",
       "  286.5,\n",
       "  737.5,\n",
       "  1304.5,\n",
       "  402.5,\n",
       "  566.5,\n",
       "  933.5,\n",
       "  596.5,\n",
       "  450.5],\n",
       " 'GO:0071353-cellular response to interleukin-4': [514.0,\n",
       "  388.0,\n",
       "  221.0,\n",
       "  470.0,\n",
       "  134.0,\n",
       "  341.0,\n",
       "  510.0,\n",
       "  228.0,\n",
       "  208.0,\n",
       "  253.0,\n",
       "  335.0,\n",
       "  263.0,\n",
       "  271.0],\n",
       " 'GO:1903573-negative regulation of response to endoplasmic reticulum stress': [515.0,\n",
       "  727.0,\n",
       "  464.0,\n",
       "  1189.0,\n",
       "  313.0,\n",
       "  1029.0,\n",
       "  979.0,\n",
       "  414.0,\n",
       "  222.0],\n",
       " 'GO:0061157-mRNA destabilization': [516.0, 204.0, 347.0],\n",
       " 'GO:0035580-specific granule lumen': [517.0,\n",
       "  670.0,\n",
       "  111.0,\n",
       "  323.0,\n",
       "  454.0,\n",
       "  663.0,\n",
       "  152.0,\n",
       "  283.0,\n",
       "  314.0,\n",
       "  211.0,\n",
       "  730.0,\n",
       "  443.0,\n",
       "  532.0,\n",
       "  909.0],\n",
       " 'GO:0070168-negative regulation of biomineral tissue development': [518.0,\n",
       "  384.0,\n",
       "  348.0,\n",
       "  422.0,\n",
       "  297.0],\n",
       " 'GO:0019068-virion assembly': [519.0, 471.0],\n",
       " 'GO:0006323-DNA packaging': [520.0,\n",
       "  570.0,\n",
       "  277.0,\n",
       "  276.0,\n",
       "  633.0,\n",
       "  446.0,\n",
       "  1301.0,\n",
       "  794.0,\n",
       "  439.0,\n",
       "  273.0,\n",
       "  525.0,\n",
       "  771.0,\n",
       "  1400.0,\n",
       "  155.0,\n",
       "  932.0,\n",
       "  398.0,\n",
       "  610.0,\n",
       "  560.0,\n",
       "  1283.0,\n",
       "  306.0,\n",
       "  393.0,\n",
       "  1200.0,\n",
       "  557.0,\n",
       "  799.0],\n",
       " 'GO:0042826-histone deacetylase binding': [521.0,\n",
       "  1162.0,\n",
       "  418.0,\n",
       "  843.0,\n",
       "  1318.0,\n",
       "  1068.0,\n",
       "  1511.0,\n",
       "  1027.0,\n",
       "  1962.0,\n",
       "  622.0,\n",
       "  620.0,\n",
       "  1098.0,\n",
       "  833.0,\n",
       "  1781.0,\n",
       "  1382.0,\n",
       "  944.0],\n",
       " 'GO:0051306-mitotic sister chromatid separation': [522.0,\n",
       "  605.0,\n",
       "  279.0,\n",
       "  927.0,\n",
       "  627.0,\n",
       "  414.0,\n",
       "  1194.0,\n",
       "  166.0,\n",
       "  551.0,\n",
       "  292.0,\n",
       "  557.0,\n",
       "  367.0,\n",
       "  1136.0,\n",
       "  211.0,\n",
       "  351.0],\n",
       " 'GO:0050686-negative regulation of mRNA processing': [523.0,\n",
       "  155.0,\n",
       "  244.0,\n",
       "  180.0,\n",
       "  116.0,\n",
       "  254.0],\n",
       " 'GO:0006333-chromatin assembly or disassembly': [524.0,\n",
       "  472.0,\n",
       "  168.0,\n",
       "  206.0,\n",
       "  629.0,\n",
       "  525.0,\n",
       "  1267.0,\n",
       "  800.0,\n",
       "  437.0,\n",
       "  314.0,\n",
       "  420.0,\n",
       "  896.0,\n",
       "  1021.0,\n",
       "  151.0,\n",
       "  910.0,\n",
       "  433.0,\n",
       "  535.0,\n",
       "  531.0,\n",
       "  1099.0,\n",
       "  345.0,\n",
       "  1065.0,\n",
       "  664.0,\n",
       "  723.0],\n",
       " 'GO:1903749-positive regulation of establishment of protein localization to mitochondrion': [525.0,\n",
       "  461.0,\n",
       "  370.0,\n",
       "  733.0,\n",
       "  613.0,\n",
       "  707.0,\n",
       "  703.0,\n",
       "  474.0,\n",
       "  829.0,\n",
       "  958.0,\n",
       "  298.0,\n",
       "  497.0,\n",
       "  244.0,\n",
       "  362.0,\n",
       "  812.0,\n",
       "  1123.0,\n",
       "  878.0],\n",
       " 'GO:0051984-positive regulation of chromosome segregation': [526.0,\n",
       "  157.0,\n",
       "  269.0,\n",
       "  266.0,\n",
       "  104.0],\n",
       " 'GO:0046825-regulation of protein export from nucleus': 527.0,\n",
       " 'GO:0012507-ER to Golgi transport vesicle membrane': [528.0,\n",
       "  422.0,\n",
       "  518.0,\n",
       "  705.0,\n",
       "  331.0,\n",
       "  822.0,\n",
       "  266.0,\n",
       "  253.0,\n",
       "  531.0,\n",
       "  674.0,\n",
       "  891.0,\n",
       "  963.0,\n",
       "  400.0,\n",
       "  468.0],\n",
       " 'GO:0080008-Cul4-RING E3 ubiquitin ligase complex': [529.0,\n",
       "  282.0,\n",
       "  411.0,\n",
       "  255.0,\n",
       "  270.0],\n",
       " 'GO:0022904-respiratory electron transport chain': [530.0,\n",
       "  233.0,\n",
       "  343.0,\n",
       "  304.0,\n",
       "  246.0,\n",
       "  1066.0,\n",
       "  404.0,\n",
       "  323.0,\n",
       "  162.0,\n",
       "  312.0,\n",
       "  244.0,\n",
       "  178.0,\n",
       "  369.0,\n",
       "  207.0,\n",
       "  103.0,\n",
       "  148.0,\n",
       "  179.0,\n",
       "  503.0,\n",
       "  246.0,\n",
       "  491.0,\n",
       "  323.0,\n",
       "  70.0,\n",
       "  864.0,\n",
       "  335.0,\n",
       "  279.0],\n",
       " 'GO:1904680-peptide transmembrane transporter activity': [531.0, 318.0],\n",
       " 'GO:0036003-positive regulation of transcription from RNA polymerase II promoter in response to stress': 532.0,\n",
       " 'GO:0031638-zymogen activation': [533.0,\n",
       "  280.0,\n",
       "  69.0,\n",
       "  210.0,\n",
       "  314.0,\n",
       "  364.0,\n",
       "  477.0,\n",
       "  377.0,\n",
       "  133.0,\n",
       "  592.0,\n",
       "  218.0,\n",
       "  85.0,\n",
       "  108.0,\n",
       "  254.0,\n",
       "  177.0,\n",
       "  444.0,\n",
       "  420.0,\n",
       "  366.0,\n",
       "  349.0,\n",
       "  150.0,\n",
       "  249.0,\n",
       "  301.0],\n",
       " 'GO:1904589-regulation of protein import': [534.0,\n",
       "  624.0,\n",
       "  501.0,\n",
       "  667.0,\n",
       "  604.0,\n",
       "  583.0,\n",
       "  366.0,\n",
       "  319.0,\n",
       "  444.0,\n",
       "  826.0,\n",
       "  442.0,\n",
       "  511.0,\n",
       "  573.0,\n",
       "  1182.0,\n",
       "  499.0,\n",
       "  392.0],\n",
       " 'GO:0006409-tRNA export from nucleus': [535.5,\n",
       "  385.5,\n",
       "  254.5,\n",
       "  936.5,\n",
       "  228.5,\n",
       "  264.5,\n",
       "  141.5],\n",
       " 'GO:0071431-tRNA-containing ribonucleoprotein complex export from nucleus': [535.5,\n",
       "  385.5,\n",
       "  254.5,\n",
       "  936.5,\n",
       "  228.5,\n",
       "  264.5,\n",
       "  141.5],\n",
       " 'GO:0031113-regulation of microtubule polymerization': [537.0,\n",
       "  516.0,\n",
       "  537.0,\n",
       "  1696.0,\n",
       "  494.0],\n",
       " 'GO:0043489-RNA stabilization': [538.0, 674.0, 774.0, 336.0],\n",
       " 'GO:0046822-regulation of nucleocytoplasmic transport': [539.0,\n",
       "  892.0,\n",
       "  276.0,\n",
       "  654.0,\n",
       "  1025.0,\n",
       "  966.0,\n",
       "  1065.0,\n",
       "  597.0,\n",
       "  487.0,\n",
       "  732.0,\n",
       "  1109.0,\n",
       "  507.0,\n",
       "  832.0,\n",
       "  613.0,\n",
       "  698.0,\n",
       "  879.0,\n",
       "  1578.0,\n",
       "  516.0,\n",
       "  1503.0,\n",
       "  761.0,\n",
       "  489.0],\n",
       " 'GO:0070972-protein localization to endoplasmic reticulum': [540.0,\n",
       "  405.0,\n",
       "  407.0,\n",
       "  690.0,\n",
       "  462.0,\n",
       "  961.0,\n",
       "  306.0,\n",
       "  682.0,\n",
       "  263.0,\n",
       "  1351.0,\n",
       "  613.0,\n",
       "  390.0,\n",
       "  416.0,\n",
       "  525.0,\n",
       "  329.0,\n",
       "  309.0,\n",
       "  461.0,\n",
       "  888.0,\n",
       "  693.0,\n",
       "  982.0,\n",
       "  534.0,\n",
       "  182.0,\n",
       "  682.0,\n",
       "  999.0,\n",
       "  434.0],\n",
       " 'GO:0010965-regulation of mitotic sister chromatid separation': [541.0,\n",
       "  578.0,\n",
       "  270.0,\n",
       "  914.0,\n",
       "  602.0,\n",
       "  427.0,\n",
       "  1187.0,\n",
       "  162.0,\n",
       "  532.0,\n",
       "  290.0,\n",
       "  564.0,\n",
       "  364.0,\n",
       "  1116.0,\n",
       "  208.0,\n",
       "  360.0],\n",
       " 'GO:1902036-regulation of hematopoietic stem cell differentiation': [542.0,\n",
       "  519.0,\n",
       "  249.0,\n",
       "  658.0,\n",
       "  310.0,\n",
       "  207.0,\n",
       "  250.0,\n",
       "  446.0,\n",
       "  537.0,\n",
       "  431.0,\n",
       "  197.0,\n",
       "  747.0,\n",
       "  306.0,\n",
       "  586.0,\n",
       "  506.0,\n",
       "  621.0,\n",
       "  350.0,\n",
       "  1015.0,\n",
       "  326.0,\n",
       "  361.0],\n",
       " 'GO:0097006-regulation of plasma lipoprotein particle levels': [543.0,\n",
       "  463.0,\n",
       "  100.0,\n",
       "  436.0,\n",
       "  289.0,\n",
       "  531.0,\n",
       "  626.0,\n",
       "  1131.0,\n",
       "  267.0,\n",
       "  521.0,\n",
       "  463.0,\n",
       "  155.0,\n",
       "  343.0,\n",
       "  800.0,\n",
       "  212.0,\n",
       "  443.0,\n",
       "  412.0,\n",
       "  305.0,\n",
       "  629.0,\n",
       "  977.0,\n",
       "  470.0,\n",
       "  213.0,\n",
       "  558.0,\n",
       "  581.0,\n",
       "  328.0],\n",
       " 'GO:0036498-IRE1-mediated unfolded protein response': [544.0,\n",
       "  743.0,\n",
       "  408.0,\n",
       "  390.0,\n",
       "  659.0,\n",
       "  687.0,\n",
       "  1212.0,\n",
       "  198.0,\n",
       "  301.0,\n",
       "  367.0,\n",
       "  576.0,\n",
       "  554.0,\n",
       "  311.0,\n",
       "  637.0,\n",
       "  422.0,\n",
       "  911.0,\n",
       "  837.0,\n",
       "  1112.0,\n",
       "  465.0,\n",
       "  54.0,\n",
       "  813.0,\n",
       "  289.0,\n",
       "  452.0],\n",
       " 'GO:1905818-regulation of chromosome separation': [545.0,\n",
       "  597.0,\n",
       "  366.0,\n",
       "  1015.0,\n",
       "  624.0,\n",
       "  485.0,\n",
       "  1312.0,\n",
       "  225.0,\n",
       "  320.0,\n",
       "  542.0,\n",
       "  496.0,\n",
       "  1287.0,\n",
       "  232.0,\n",
       "  515.0],\n",
       " 'GO:0015020-glucuronosyltransferase activity': [546.0,\n",
       "  35.0,\n",
       "  251.0,\n",
       "  393.0,\n",
       "  146.0,\n",
       "  305.0,\n",
       "  220.0,\n",
       "  920.0,\n",
       "  106.0,\n",
       "  88.0,\n",
       "  245.0,\n",
       "  23.0,\n",
       "  672.0,\n",
       "  337.0,\n",
       "  141.0],\n",
       " 'GO:0005501-retinoid binding': [547.0,\n",
       "  297.0,\n",
       "  37.0,\n",
       "  1112.0,\n",
       "  129.0,\n",
       "  146.0,\n",
       "  179.0,\n",
       "  236.0,\n",
       "  214.0],\n",
       " 'GO:0097064-ncRNA export from nucleus': [548.0,\n",
       "  398.0,\n",
       "  252.0,\n",
       "  931.0,\n",
       "  232.0,\n",
       "  299.0,\n",
       "  137.0,\n",
       "  437.0],\n",
       " 'GO:0060218-hematopoietic stem cell differentiation': [549.0,\n",
       "  553.0,\n",
       "  257.0,\n",
       "  620.0,\n",
       "  1382.0,\n",
       "  388.0,\n",
       "  249.0,\n",
       "  276.0,\n",
       "  259.0,\n",
       "  448.0,\n",
       "  505.0,\n",
       "  405.0,\n",
       "  228.0,\n",
       "  767.0,\n",
       "  300.0,\n",
       "  593.0,\n",
       "  566.0,\n",
       "  610.0,\n",
       "  357.0,\n",
       "  1082.0,\n",
       "  299.0,\n",
       "  391.0],\n",
       " 'GO:0090224-regulation of spindle organization': [550.0,\n",
       "  536.0,\n",
       "  436.0,\n",
       "  1091.0,\n",
       "  304.0,\n",
       "  556.0,\n",
       "  444.0,\n",
       "  176.0],\n",
       " 'GO:0016860-intramolecular oxidoreductase activity': [551.0,\n",
       "  307.0,\n",
       "  250.0,\n",
       "  178.0,\n",
       "  349.0,\n",
       "  235.0,\n",
       "  339.0,\n",
       "  524.0,\n",
       "  350.0,\n",
       "  287.0,\n",
       "  399.0,\n",
       "  422.0,\n",
       "  1038.0,\n",
       "  797.0,\n",
       "  603.0,\n",
       "  290.0],\n",
       " 'GO:0045839-negative regulation of mitotic nuclear division': [552.0,\n",
       "  367.0,\n",
       "  606.0,\n",
       "  595.0,\n",
       "  1154.0,\n",
       "  213.0,\n",
       "  297.0,\n",
       "  740.0,\n",
       "  537.0,\n",
       "  1021.0,\n",
       "  277.0,\n",
       "  395.0],\n",
       " 'GO:0043021-ribonucleoprotein complex binding': [553.0,\n",
       "  595.0,\n",
       "  383.0,\n",
       "  509.0,\n",
       "  1203.0,\n",
       "  1458.0,\n",
       "  965.0,\n",
       "  685.0,\n",
       "  474.0,\n",
       "  441.0,\n",
       "  993.0,\n",
       "  1342.0,\n",
       "  1225.0,\n",
       "  336.0,\n",
       "  1262.0,\n",
       "  301.0,\n",
       "  605.0,\n",
       "  719.0,\n",
       "  1581.0,\n",
       "  215.0,\n",
       "  192.0,\n",
       "  1747.0,\n",
       "  639.0,\n",
       "  729.0],\n",
       " 'GO:0043038-amino acid activation': [554.0,\n",
       "  498.0,\n",
       "  491.0,\n",
       "  494.0,\n",
       "  277.0,\n",
       "  515.0,\n",
       "  124.0,\n",
       "  158.0,\n",
       "  538.0,\n",
       "  411.0,\n",
       "  169.0,\n",
       "  48.0,\n",
       "  469.0],\n",
       " 'GO:0051955-regulation of amino acid transport': 555.0,\n",
       " 'GO:1902235-regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway': [556.0,\n",
       "  208.0,\n",
       "  339.0,\n",
       "  250.0,\n",
       "  664.0],\n",
       " 'GO:0033144-negative regulation of intracellular steroid hormone receptor signaling pathway': [557.0,\n",
       "  278.0,\n",
       "  351.0,\n",
       "  757.0,\n",
       "  379.0],\n",
       " 'GO:0031491-nucleosome binding': [558.0,\n",
       "  532.0,\n",
       "  399.0,\n",
       "  420.0,\n",
       "  708.0,\n",
       "  332.0,\n",
       "  620.0,\n",
       "  889.0,\n",
       "  185.0,\n",
       "  408.0,\n",
       "  519.0,\n",
       "  681.0,\n",
       "  1108.0,\n",
       "  299.0,\n",
       "  128.0,\n",
       "  575.0,\n",
       "  633.0],\n",
       " 'GO:0051059-NF-kappaB binding': [559.0, 309.0, 762.0, 214.0],\n",
       " 'GO:0042590-antigen processing and presentation of exogenous peptide antigen via MHC class I': [560.0,\n",
       "  261.0,\n",
       "  147.0,\n",
       "  259.0,\n",
       "  407.0,\n",
       "  1225.0,\n",
       "  392.0,\n",
       "  124.0,\n",
       "  237.0,\n",
       "  578.0,\n",
       "  268.0,\n",
       "  183.0,\n",
       "  333.0,\n",
       "  308.0,\n",
       "  133.0,\n",
       "  340.0,\n",
       "  338.0,\n",
       "  405.0,\n",
       "  147.0,\n",
       "  322.0,\n",
       "  279.0,\n",
       "  502.0,\n",
       "  192.0,\n",
       "  346.0],\n",
       " 'GO:0002474-antigen processing and presentation of peptide antigen via MHC class I': [561.0,\n",
       "  358.0,\n",
       "  241.0,\n",
       "  407.0,\n",
       "  526.0,\n",
       "  1414.0,\n",
       "  420.0,\n",
       "  312.0,\n",
       "  261.0,\n",
       "  528.0,\n",
       "  315.0,\n",
       "  240.0,\n",
       "  453.0,\n",
       "  560.0,\n",
       "  209.0,\n",
       "  337.0,\n",
       "  359.0,\n",
       "  502.0,\n",
       "  243.0,\n",
       "  430.0,\n",
       "  377.0,\n",
       "  597.0,\n",
       "  302.0,\n",
       "  455.0],\n",
       " 'GO:0106118-regulation of sterol biosynthetic process': [562.5,\n",
       "  689.5,\n",
       "  688.5,\n",
       "  481.5,\n",
       "  1474.5,\n",
       "  239.5,\n",
       "  544.5,\n",
       "  1000.5,\n",
       "  294.5,\n",
       "  82.5,\n",
       "  1108.5,\n",
       "  552.5],\n",
       " 'GO:0045540-regulation of cholesterol biosynthetic process': [562.5,\n",
       "  689.5,\n",
       "  688.5,\n",
       "  481.5,\n",
       "  1474.5,\n",
       "  239.5,\n",
       "  544.5,\n",
       "  1000.5,\n",
       "  294.5,\n",
       "  82.5,\n",
       "  1108.5,\n",
       "  552.5],\n",
       " 'GO:0006998-nuclear envelope organization': [564.0,\n",
       "  508.0,\n",
       "  803.0,\n",
       "  664.0,\n",
       "  1398.0,\n",
       "  706.0],\n",
       " 'GO:0120114-Sm-like protein family complex': [565.0,\n",
       "  455.0,\n",
       "  344.0,\n",
       "  289.0,\n",
       "  534.0,\n",
       "  759.0,\n",
       "  799.0,\n",
       "  298.0,\n",
       "  340.0,\n",
       "  827.0,\n",
       "  819.0,\n",
       "  673.0,\n",
       "  127.0,\n",
       "  221.0,\n",
       "  508.0,\n",
       "  321.0,\n",
       "  667.0,\n",
       "  156.0,\n",
       "  1669.0,\n",
       "  750.0,\n",
       "  686.0],\n",
       " 'GO:0090314-positive regulation of protein targeting to membrane': [566.0,\n",
       "  441.0,\n",
       "  283.0,\n",
       "  272.0,\n",
       "  841.0],\n",
       " 'GO:0016592-mediator complex': 567.0,\n",
       " 'GO:2000816-negative regulation of mitotic sister chromatid separation': [568.0,\n",
       "  449.0,\n",
       "  962.0,\n",
       "  148.0,\n",
       "  271.0,\n",
       "  577.0,\n",
       "  365.0,\n",
       "  194.0,\n",
       "  292.0],\n",
       " 'GO:0043039-tRNA aminoacylation': [569.0,\n",
       "  510.0,\n",
       "  507.0,\n",
       "  502.0,\n",
       "  260.0,\n",
       "  530.0,\n",
       "  120.0,\n",
       "  161.0,\n",
       "  551.0,\n",
       "  340.0,\n",
       "  186.0,\n",
       "  50.0,\n",
       "  402.0],\n",
       " 'GO:0031497-chromatin assembly': [570.0,\n",
       "  492.0,\n",
       "  181.0,\n",
       "  235.0,\n",
       "  482.0,\n",
       "  530.0,\n",
       "  1297.0,\n",
       "  808.0,\n",
       "  371.0,\n",
       "  321.0,\n",
       "  324.0,\n",
       "  455.0,\n",
       "  648.0,\n",
       "  1188.0,\n",
       "  147.0,\n",
       "  676.0,\n",
       "  354.0,\n",
       "  626.0,\n",
       "  538.0,\n",
       "  1124.0,\n",
       "  339.0,\n",
       "  1080.0,\n",
       "  724.0,\n",
       "  632.0],\n",
       " 'GO:0060964-regulation of gene silencing by miRNA': [571.0,\n",
       "  655.0,\n",
       "  394.0,\n",
       "  1051.0,\n",
       "  887.0,\n",
       "  644.0,\n",
       "  552.0,\n",
       "  387.0,\n",
       "  403.0,\n",
       "  832.0,\n",
       "  1002.0,\n",
       "  282.0,\n",
       "  833.0,\n",
       "  288.0,\n",
       "  520.0,\n",
       "  577.0,\n",
       "  1397.0,\n",
       "  266.0,\n",
       "  1549.0,\n",
       "  502.0,\n",
       "  688.0],\n",
       " 'GO:0034383-low-density lipoprotein particle clearance': [572.0,\n",
       "  334.0,\n",
       "  222.0,\n",
       "  411.0,\n",
       "  402.0,\n",
       "  257.0,\n",
       "  323.0,\n",
       "  329.0,\n",
       "  227.0,\n",
       "  273.0,\n",
       "  272.0,\n",
       "  419.0,\n",
       "  186.0],\n",
       " 'GO:1905332-positive regulation of morphogenesis of an epithelium': [573.0,\n",
       "  476.0],\n",
       " 'GO:0071479-cellular response to ionizing radiation': [574.0,\n",
       "  457.0,\n",
       "  198.0,\n",
       "  445.0,\n",
       "  799.0,\n",
       "  830.0,\n",
       "  814.0,\n",
       "  371.0,\n",
       "  293.0,\n",
       "  957.0,\n",
       "  1991.0,\n",
       "  660.0,\n",
       "  480.0,\n",
       "  861.0,\n",
       "  1322.0,\n",
       "  1227.0,\n",
       "  578.0,\n",
       "  532.0],\n",
       " 'GO:0006007-glucose catabolic process': [575.0,\n",
       "  218.0,\n",
       "  144.0,\n",
       "  379.0,\n",
       "  356.0,\n",
       "  1058.0,\n",
       "  85.0,\n",
       "  198.0,\n",
       "  124.0,\n",
       "  627.0,\n",
       "  45.0,\n",
       "  721.0,\n",
       "  481.0,\n",
       "  153.0],\n",
       " 'GO:0032371-regulation of sterol transport': [576.5,\n",
       "  355.5,\n",
       "  497.5,\n",
       "  417.5,\n",
       "  746.5,\n",
       "  447.5,\n",
       "  208.5,\n",
       "  756.5,\n",
       "  274.5,\n",
       "  138.5,\n",
       "  243.5,\n",
       "  819.5,\n",
       "  199.5,\n",
       "  466.5,\n",
       "  521.5,\n",
       "  925.5,\n",
       "  341.5,\n",
       "  492.5,\n",
       "  217.5],\n",
       " 'GO:0032374-regulation of cholesterol transport': [576.5,\n",
       "  355.5,\n",
       "  497.5,\n",
       "  417.5,\n",
       "  746.5,\n",
       "  447.5,\n",
       "  208.5,\n",
       "  756.5,\n",
       "  274.5,\n",
       "  138.5,\n",
       "  243.5,\n",
       "  819.5,\n",
       "  199.5,\n",
       "  466.5,\n",
       "  521.5,\n",
       "  925.5,\n",
       "  341.5,\n",
       "  492.5,\n",
       "  217.5],\n",
       " 'GO:1901031-regulation of response to reactive oxygen species': [578.0,\n",
       "  972.0,\n",
       "  401.0,\n",
       "  729.0],\n",
       " 'GO:0033048-negative regulation of mitotic sister chromatid segregation': [579.0,\n",
       "  286.0,\n",
       "  516.0,\n",
       "  1049.0,\n",
       "  286.0,\n",
       "  658.0,\n",
       "  363.0,\n",
       "  822.0,\n",
       "  221.0,\n",
       "  281.0],\n",
       " 'GO:0001844-protein insertion into mitochondrial membrane involved in apoptotic signaling pathway': [580.0,\n",
       "  207.0,\n",
       "  266.0],\n",
       " 'GO:0090181-regulation of cholesterol metabolic process': [581.0,\n",
       "  838.0,\n",
       "  870.0,\n",
       "  685.0,\n",
       "  538.0,\n",
       "  824.0,\n",
       "  1869.0,\n",
       "  400.0,\n",
       "  272.0,\n",
       "  651.0,\n",
       "  1331.0,\n",
       "  468.0,\n",
       "  951.0,\n",
       "  839.0,\n",
       "  558.0],\n",
       " 'GO:0002823-negative regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains': [582.0,\n",
       "  208.0,\n",
       "  208.0,\n",
       "  596.0,\n",
       "  756.0],\n",
       " 'GO:0046794-transport of virus': [583.0,\n",
       "  644.0,\n",
       "  1299.0,\n",
       "  522.0,\n",
       "  358.0,\n",
       "  1549.0,\n",
       "  324.0,\n",
       "  426.0,\n",
       "  414.0,\n",
       "  861.0],\n",
       " 'GO:0051290-protein heterotetramerization': [584.0,\n",
       "  503.0,\n",
       "  834.0,\n",
       "  998.0,\n",
       "  469.0,\n",
       "  265.0,\n",
       "  1218.0,\n",
       "  204.0,\n",
       "  331.0,\n",
       "  805.0,\n",
       "  616.0,\n",
       "  1056.0,\n",
       "  1232.0,\n",
       "  737.0],\n",
       " 'GO:0042771-intrinsic apoptotic signaling pathway in response to DNA damage by p53 class mediator': [585.0,\n",
       "  258.0,\n",
       "  432.0,\n",
       "  575.0,\n",
       "  437.0,\n",
       "  198.0,\n",
       "  605.0,\n",
       "  1114.0,\n",
       "  269.0,\n",
       "  717.0,\n",
       "  503.0,\n",
       "  1007.0,\n",
       "  257.0],\n",
       " 'GO:2000573-positive regulation of DNA biosynthetic process': [586.0,\n",
       "  506.0,\n",
       "  489.0,\n",
       "  950.0,\n",
       "  1148.0,\n",
       "  506.0,\n",
       "  549.0,\n",
       "  1261.0,\n",
       "  522.0,\n",
       "  558.0,\n",
       "  793.0,\n",
       "  300.0,\n",
       "  525.0],\n",
       " \"GO:0006296-nucleotide-excision repair, DNA incision, 5'-to lesion\": [587.0,\n",
       "  422.0,\n",
       "  429.0,\n",
       "  336.0,\n",
       "  324.0,\n",
       "  888.0,\n",
       "  185.0],\n",
       " 'GO:0097345-mitochondrial outer membrane permeabilization': [588.0,\n",
       "  543.0,\n",
       "  826.0,\n",
       "  1014.0,\n",
       "  537.0,\n",
       "  644.0,\n",
       "  1192.0,\n",
       "  994.0],\n",
       " 'GO:0006119-oxidative phosphorylation': [589.0,\n",
       "  258.0,\n",
       "  432.0,\n",
       "  290.0,\n",
       "  242.0,\n",
       "  893.0,\n",
       "  413.0,\n",
       "  244.0,\n",
       "  175.0,\n",
       "  363.0,\n",
       "  211.0,\n",
       "  188.0,\n",
       "  385.0,\n",
       "  211.0,\n",
       "  112.0,\n",
       "  165.0,\n",
       "  198.0,\n",
       "  403.0,\n",
       "  204.0,\n",
       "  398.0,\n",
       "  364.0,\n",
       "  67.0,\n",
       "  995.0,\n",
       "  339.0,\n",
       "  263.0],\n",
       " 'GO:0033344-cholesterol efflux': [590.0,\n",
       "  349.0,\n",
       "  173.0,\n",
       "  254.0,\n",
       "  729.0,\n",
       "  361.0,\n",
       "  118.0,\n",
       "  416.0,\n",
       "  172.0,\n",
       "  75.0,\n",
       "  163.0,\n",
       "  693.0,\n",
       "  170.0,\n",
       "  360.0,\n",
       "  418.0,\n",
       "  780.0,\n",
       "  205.0,\n",
       "  543.0,\n",
       "  160.0],\n",
       " 'GO:0098798-mitochondrial protein complex': [591.0,\n",
       "  312.0,\n",
       "  441.0,\n",
       "  514.0,\n",
       "  384.0,\n",
       "  1263.0,\n",
       "  439.0,\n",
       "  314.0,\n",
       "  227.0,\n",
       "  364.0,\n",
       "  548.0,\n",
       "  408.0,\n",
       "  706.0,\n",
       "  422.0,\n",
       "  137.0,\n",
       "  415.0,\n",
       "  187.0,\n",
       "  450.0,\n",
       "  309.0,\n",
       "  669.0,\n",
       "  322.0,\n",
       "  68.0,\n",
       "  1384.0,\n",
       "  427.0,\n",
       "  412.0],\n",
       " 'GO:0043175-RNA polymerase core enzyme binding': [592.0,\n",
       "  1136.0,\n",
       "  336.0,\n",
       "  1365.0,\n",
       "  392.0,\n",
       "  230.0],\n",
       " 'GO:0042274-ribosomal small subunit biogenesis': [593.0,\n",
       "  427.0,\n",
       "  527.0,\n",
       "  1292.0,\n",
       "  625.0,\n",
       "  388.0,\n",
       "  663.0,\n",
       "  633.0,\n",
       "  328.0,\n",
       "  276.0,\n",
       "  594.0,\n",
       "  786.0,\n",
       "  1344.0,\n",
       "  540.0,\n",
       "  286.0],\n",
       " 'GO:0060236-regulation of mitotic spindle organization': [594.0,\n",
       "  442.0,\n",
       "  997.0,\n",
       "  233.0],\n",
       " 'GO:0072562-blood microparticle': [595.0,\n",
       "  147.0,\n",
       "  54.0,\n",
       "  121.0,\n",
       "  147.0,\n",
       "  239.0,\n",
       "  182.0,\n",
       "  445.0,\n",
       "  165.0,\n",
       "  686.0,\n",
       "  128.0,\n",
       "  67.0,\n",
       "  106.0,\n",
       "  245.0,\n",
       "  144.0,\n",
       "  195.0,\n",
       "  1081.0,\n",
       "  221.0,\n",
       "  194.0,\n",
       "  229.0,\n",
       "  635.0,\n",
       "  322.0,\n",
       "  422.0,\n",
       "  125.0],\n",
       " 'GO:2000725-regulation of cardiac muscle cell differentiation': [596.0,\n",
       "  514.0,\n",
       "  122.0,\n",
       "  463.0,\n",
       "  431.0,\n",
       "  642.0,\n",
       "  645.0,\n",
       "  345.0,\n",
       "  397.0,\n",
       "  503.0,\n",
       "  275.0,\n",
       "  363.0,\n",
       "  603.0,\n",
       "  532.0,\n",
       "  440.0,\n",
       "  241.0,\n",
       "  216.0,\n",
       "  751.0],\n",
       " 'GO:0051031-tRNA transport': [597.0,\n",
       "  441.0,\n",
       "  283.0,\n",
       "  1022.0,\n",
       "  250.0,\n",
       "  306.0,\n",
       "  150.0,\n",
       "  503.0],\n",
       " 'GO:0042255-ribosome assembly': [598.0,\n",
       "  360.0,\n",
       "  433.0,\n",
       "  600.0,\n",
       "  274.0,\n",
       "  251.0,\n",
       "  879.0,\n",
       "  557.0,\n",
       "  550.0,\n",
       "  321.0,\n",
       "  275.0,\n",
       "  443.0,\n",
       "  389.0,\n",
       "  712.0,\n",
       "  418.0,\n",
       "  898.0,\n",
       "  772.0,\n",
       "  344.0],\n",
       " 'GO:0035578-azurophil granule lumen': [599.0,\n",
       "  325.0,\n",
       "  95.0,\n",
       "  495.0,\n",
       "  496.0,\n",
       "  801.0,\n",
       "  453.0,\n",
       "  292.0,\n",
       "  210.0,\n",
       "  869.0,\n",
       "  326.0,\n",
       "  212.0,\n",
       "  433.0,\n",
       "  356.0,\n",
       "  189.0,\n",
       "  578.0,\n",
       "  521.0,\n",
       "  258.0,\n",
       "  356.0,\n",
       "  546.0,\n",
       "  606.0,\n",
       "  968.0,\n",
       "  523.0],\n",
       " 'GO:0031571-mitotic G1 DNA damage checkpoint': [600.5,\n",
       "  442.5,\n",
       "  707.5,\n",
       "  502.5,\n",
       "  665.5,\n",
       "  272.5,\n",
       "  657.5,\n",
       "  1267.5,\n",
       "  367.5,\n",
       "  687.5,\n",
       "  782.5,\n",
       "  1003.5,\n",
       "  500.5,\n",
       "  480.5],\n",
       " 'GO:0044819-mitotic G1/S transition checkpoint': [600.5,\n",
       "  442.5,\n",
       "  707.5,\n",
       "  502.5,\n",
       "  665.5,\n",
       "  272.5,\n",
       "  657.5,\n",
       "  1267.5,\n",
       "  367.5,\n",
       "  687.5,\n",
       "  782.5,\n",
       "  1003.5,\n",
       "  500.5,\n",
       "  480.5],\n",
       " 'GO:0072395-signal transduction involved in cell cycle checkpoint': [602.0,\n",
       "  716.0,\n",
       "  247.0,\n",
       "  490.0,\n",
       "  756.0,\n",
       "  758.0,\n",
       "  1046.0,\n",
       "  296.0,\n",
       "  305.0,\n",
       "  490.0,\n",
       "  1557.0,\n",
       "  358.0,\n",
       "  565.0,\n",
       "  303.0,\n",
       "  546.0,\n",
       "  689.0,\n",
       "  1009.0,\n",
       "  683.0,\n",
       "  102.0,\n",
       "  589.0,\n",
       "  433.0],\n",
       " 'GO:1902403-signal transduction involved in mitotic DNA integrity checkpoint': [604.0,\n",
       "  461.0,\n",
       "  705.0,\n",
       "  717.0,\n",
       "  677.0,\n",
       "  278.0,\n",
       "  698.0,\n",
       "  1285.0,\n",
       "  389.0,\n",
       "  562.0,\n",
       "  723.0,\n",
       "  918.0,\n",
       "  568.0,\n",
       "  443.0],\n",
       " 'GO:0072413-signal transduction involved in mitotic cell cycle checkpoint': [604.0,\n",
       "  461.0,\n",
       "  705.0,\n",
       "  717.0,\n",
       "  677.0,\n",
       "  278.0,\n",
       "  698.0,\n",
       "  1285.0,\n",
       "  389.0,\n",
       "  562.0,\n",
       "  723.0,\n",
       "  918.0,\n",
       "  568.0,\n",
       "  443.0],\n",
       " 'GO:1902402-signal transduction involved in mitotic DNA damage checkpoint': [604.0,\n",
       "  461.0,\n",
       "  705.0,\n",
       "  717.0,\n",
       "  677.0,\n",
       "  278.0,\n",
       "  698.0,\n",
       "  1285.0,\n",
       "  389.0,\n",
       "  562.0,\n",
       "  723.0,\n",
       "  918.0,\n",
       "  568.0,\n",
       "  443.0],\n",
       " 'GO:0030071-regulation of mitotic metaphase/anaphase transition': [606.0,\n",
       "  494.0,\n",
       "  274.0,\n",
       "  932.0,\n",
       "  517.0,\n",
       "  438.0,\n",
       "  1135.0,\n",
       "  172.0,\n",
       "  325.0,\n",
       "  669.0,\n",
       "  404.0,\n",
       "  1034.0,\n",
       "  229.0,\n",
       "  250.0],\n",
       " 'GO:0019835-cytolysis': [607.0,\n",
       "  345.0,\n",
       "  366.0,\n",
       "  401.0,\n",
       "  483.0,\n",
       "  297.0,\n",
       "  163.0,\n",
       "  316.0,\n",
       "  69.0,\n",
       "  156.0,\n",
       "  490.0,\n",
       "  119.0,\n",
       "  298.0,\n",
       "  190.0,\n",
       "  266.0,\n",
       "  641.0],\n",
       " 'GO:0060147-regulation of posttranscriptional gene silencing': [608.5,\n",
       "  686.5,\n",
       "  431.5,\n",
       "  1095.5,\n",
       "  922.5,\n",
       "  743.5,\n",
       "  553.5,\n",
       "  402.5,\n",
       "  419.5,\n",
       "  867.5,\n",
       "  1000.5,\n",
       "  260.5,\n",
       "  819.5,\n",
       "  317.5,\n",
       "  498.5,\n",
       "  558.5,\n",
       "  1427.5,\n",
       "  247.5,\n",
       "  1532.5,\n",
       "  591.5,\n",
       "  671.5],\n",
       " 'GO:0060966-regulation of gene silencing by RNA': [608.5,\n",
       "  686.5,\n",
       "  431.5,\n",
       "  1095.5,\n",
       "  922.5,\n",
       "  743.5,\n",
       "  553.5,\n",
       "  402.5,\n",
       "  419.5,\n",
       "  867.5,\n",
       "  1000.5,\n",
       "  260.5,\n",
       "  819.5,\n",
       "  317.5,\n",
       "  498.5,\n",
       "  558.5,\n",
       "  1427.5,\n",
       "  247.5,\n",
       "  1532.5,\n",
       "  591.5,\n",
       "  671.5],\n",
       " 'GO:0005844-polysome': [610.0,\n",
       "  366.0,\n",
       "  533.0,\n",
       "  530.0,\n",
       "  798.0,\n",
       "  580.0,\n",
       "  369.0,\n",
       "  627.0,\n",
       "  1008.0,\n",
       "  241.0,\n",
       "  427.0,\n",
       "  278.0,\n",
       "  541.0,\n",
       "  594.0,\n",
       "  1514.0,\n",
       "  504.0,\n",
       "  96.0,\n",
       "  1254.0,\n",
       "  560.0,\n",
       "  465.0],\n",
       " 'GO:0003417-growth plate cartilage development': [611.0, 272.0, 285.0],\n",
       " 'GO:0010591-regulation of lamellipodium assembly': [612.0, 213.0],\n",
       " 'GO:0036258-multivesicular body assembly': 613.0,\n",
       " 'GO:0005154-epidermal growth factor receptor binding': [614.0,\n",
       "  528.0,\n",
       "  995.0,\n",
       "  634.0],\n",
       " 'GO:0043457-regulation of cellular respiration': [615.0, 900.0],\n",
       " 'GO:0061650-ubiquitin-like protein conjugating enzyme activity': [616.0,\n",
       "  197.0,\n",
       "  295.0],\n",
       " 'GO:0072422-signal transduction involved in DNA damage checkpoint': [617.5,\n",
       "  734.5,\n",
       "  269.5,\n",
       "  505.5,\n",
       "  811.5,\n",
       "  773.5,\n",
       "  1061.5,\n",
       "  299.5,\n",
       "  506.5,\n",
       "  1585.5,\n",
       "  374.5,\n",
       "  576.5,\n",
       "  309.5,\n",
       "  548.5,\n",
       "  633.5,\n",
       "  1051.5,\n",
       "  105.5,\n",
       "  612.5,\n",
       "  436.5],\n",
       " 'GO:0072401-signal transduction involved in DNA integrity checkpoint': [617.5,\n",
       "  734.5,\n",
       "  269.5,\n",
       "  505.5,\n",
       "  811.5,\n",
       "  773.5,\n",
       "  1061.5,\n",
       "  299.5,\n",
       "  506.5,\n",
       "  1585.5,\n",
       "  374.5,\n",
       "  576.5,\n",
       "  309.5,\n",
       "  548.5,\n",
       "  633.5,\n",
       "  1051.5,\n",
       "  105.5,\n",
       "  612.5,\n",
       "  436.5],\n",
       " 'GO:0070670-response to interleukin-4': [619.0,\n",
       "  287.0,\n",
       "  506.0,\n",
       "  130.0,\n",
       "  606.0,\n",
       "  218.0,\n",
       "  271.0,\n",
       "  280.0,\n",
       "  372.0,\n",
       "  235.0,\n",
       "  256.0],\n",
       " 'GO:0006338-chromatin remodeling': [620.0,\n",
       "  782.0,\n",
       "  322.0,\n",
       "  373.0,\n",
       "  1078.0,\n",
       "  690.0,\n",
       "  1291.0,\n",
       "  825.0,\n",
       "  423.0,\n",
       "  641.0,\n",
       "  1368.0,\n",
       "  1623.0,\n",
       "  307.0,\n",
       "  934.0,\n",
       "  486.0,\n",
       "  625.0,\n",
       "  624.0,\n",
       "  1154.0,\n",
       "  352.0,\n",
       "  252.0,\n",
       "  1399.0,\n",
       "  730.0,\n",
       "  802.0],\n",
       " 'GO:0007091-metaphase/anaphase transition of mitotic cell cycle': [621.0,\n",
       "  619.0,\n",
       "  300.0,\n",
       "  986.0,\n",
       "  617.0,\n",
       "  444.0,\n",
       "  1358.0,\n",
       "  179.0,\n",
       "  339.0,\n",
       "  642.0,\n",
       "  402.0,\n",
       "  1189.0,\n",
       "  235.0,\n",
       "  311.0],\n",
       " 'GO:0070897-transcription preinitiation complex assembly': [622.0,\n",
       "  311.0,\n",
       "  787.0,\n",
       "  205.0,\n",
       "  575.0,\n",
       "  831.0,\n",
       "  284.0,\n",
       "  562.0],\n",
       " 'GO:0000993-RNA polymerase II complex binding': [623.0,\n",
       "  326.0,\n",
       "  1296.0,\n",
       "  364.0,\n",
       "  220.0],\n",
       " 'GO:0008630-intrinsic apoptotic signaling pathway in response to DNA damage': [624.0,\n",
       "  509.0,\n",
       "  304.0,\n",
       "  277.0,\n",
       "  735.0,\n",
       "  381.0,\n",
       "  798.0,\n",
       "  352.0,\n",
       "  788.0,\n",
       "  814.0,\n",
       "  890.0,\n",
       "  1282.0,\n",
       "  454.0,\n",
       "  321.0,\n",
       "  579.0,\n",
       "  689.0,\n",
       "  587.0,\n",
       "  1351.0,\n",
       "  725.0,\n",
       "  183.0,\n",
       "  1213.0,\n",
       "  442.0,\n",
       "  514.0],\n",
       " 'GO:1901380-negative regulation of potassium ion transmembrane transport': [625.0,\n",
       "  247.0,\n",
       "  162.0,\n",
       "  357.0,\n",
       "  478.0,\n",
       "  404.0,\n",
       "  180.0],\n",
       " 'GO:0044364-disruption of cells of other organism': [626.5,\n",
       "  209.5,\n",
       "  119.5,\n",
       "  434.5,\n",
       "  370.5,\n",
       "  475.5,\n",
       "  292.5,\n",
       "  513.5,\n",
       "  155.5,\n",
       "  338.5,\n",
       "  65.5,\n",
       "  216.5,\n",
       "  567.5,\n",
       "  280.5,\n",
       "  373.5,\n",
       "  216.5,\n",
       "  375.5,\n",
       "  441.5,\n",
       "  590.5,\n",
       "  1293.5],\n",
       " 'GO:0031640-killing of cells of other organism': [626.5,\n",
       "  209.5,\n",
       "  119.5,\n",
       "  434.5,\n",
       "  370.5,\n",
       "  475.5,\n",
       "  292.5,\n",
       "  513.5,\n",
       "  155.5,\n",
       "  338.5,\n",
       "  65.5,\n",
       "  216.5,\n",
       "  567.5,\n",
       "  280.5,\n",
       "  373.5,\n",
       "  216.5,\n",
       "  375.5,\n",
       "  441.5,\n",
       "  590.5,\n",
       "  1293.5],\n",
       " 'GO:0070317-negative regulation of G0 to G1 transition': [628.0, 317.0],\n",
       " 'GO:0006536-glutamate metabolic process': [629.0,\n",
       "  302.0,\n",
       "  111.0,\n",
       "  408.0,\n",
       "  294.0,\n",
       "  733.0,\n",
       "  411.0,\n",
       "  116.0,\n",
       "  445.0],\n",
       " 'GO:0005876-spindle microtubule': [630.0,\n",
       "  369.0,\n",
       "  483.0,\n",
       "  846.0,\n",
       "  831.0,\n",
       "  296.0,\n",
       "  453.0,\n",
       "  726.0,\n",
       "  728.0,\n",
       "  290.0,\n",
       "  697.0],\n",
       " 'GO:0097242-amyloid-beta clearance': [631.0, 1065.0, 227.0],\n",
       " 'GO:0005681-spliceosomal complex': [632.0,\n",
       "  566.0,\n",
       "  519.0,\n",
       "  817.0,\n",
       "  1011.0,\n",
       "  795.0,\n",
       "  1514.0,\n",
       "  774.0,\n",
       "  404.0,\n",
       "  790.0,\n",
       "  205.0,\n",
       "  1329.0,\n",
       "  421.0,\n",
       "  534.0,\n",
       "  525.0,\n",
       "  1123.0,\n",
       "  274.0,\n",
       "  146.0,\n",
       "  523.0,\n",
       "  1024.0],\n",
       " 'GO:0010656-negative regulation of muscle cell apoptotic process': [633.0,\n",
       "  57.0,\n",
       "  213.0,\n",
       "  390.0,\n",
       "  315.0,\n",
       "  502.0,\n",
       "  413.0,\n",
       "  281.0,\n",
       "  646.0,\n",
       "  598.0,\n",
       "  780.0],\n",
       " 'GO:0006984-ER-nucleus signaling pathway': [634.0,\n",
       "  401.0,\n",
       "  615.0,\n",
       "  591.0,\n",
       "  925.0,\n",
       "  382.0,\n",
       "  791.0,\n",
       "  518.0,\n",
       "  814.0],\n",
       " 'GO:0006353-DNA-templated transcription, termination': [635.0,\n",
       "  739.0,\n",
       "  697.0,\n",
       "  1159.0,\n",
       "  1469.0,\n",
       "  1013.0,\n",
       "  317.0,\n",
       "  1208.0,\n",
       "  293.0,\n",
       "  350.0,\n",
       "  482.0,\n",
       "  878.0,\n",
       "  1556.0,\n",
       "  237.0,\n",
       "  1165.0,\n",
       "  740.0],\n",
       " 'GO:0071824-protein-DNA complex subunit organization': [636.0,\n",
       "  634.0,\n",
       "  315.0,\n",
       "  365.0,\n",
       "  1058.0,\n",
       "  817.0,\n",
       "  1376.0,\n",
       "  784.0,\n",
       "  457.0,\n",
       "  369.0,\n",
       "  649.0,\n",
       "  1229.0,\n",
       "  1181.0,\n",
       "  243.0,\n",
       "  924.0,\n",
       "  361.0,\n",
       "  604.0,\n",
       "  650.0,\n",
       "  1165.0,\n",
       "  332.0,\n",
       "  369.0,\n",
       "  1613.0,\n",
       "  602.0,\n",
       "  689.0],\n",
       " 'GO:0016469-proton-transporting two-sector ATPase complex': [637.0,\n",
       "  338.0,\n",
       "  157.0,\n",
       "  614.0,\n",
       "  312.0,\n",
       "  590.0,\n",
       "  573.0,\n",
       "  157.0,\n",
       "  103.0,\n",
       "  256.0,\n",
       "  432.0,\n",
       "  527.0,\n",
       "  317.0,\n",
       "  712.0],\n",
       " 'GO:0042379-chemokine receptor binding': [638.0,\n",
       "  175.0,\n",
       "  72.0,\n",
       "  532.0,\n",
       "  237.0,\n",
       "  456.0,\n",
       "  262.0,\n",
       "  585.0,\n",
       "  110.0,\n",
       "  188.0,\n",
       "  34.0,\n",
       "  279.0,\n",
       "  414.0,\n",
       "  226.0,\n",
       "  136.0,\n",
       "  433.0,\n",
       "  483.0,\n",
       "  300.0,\n",
       "  327.0,\n",
       "  529.0,\n",
       "  502.0],\n",
       " 'GO:0060416-response to growth hormone': [639.0,\n",
       "  313.0,\n",
       "  135.0,\n",
       "  552.0,\n",
       "  355.0,\n",
       "  509.0,\n",
       "  270.0,\n",
       "  647.0,\n",
       "  305.0,\n",
       "  236.0],\n",
       " 'GO:0002040-sprouting angiogenesis': [640.0,\n",
       "  538.0,\n",
       "  122.0,\n",
       "  194.0,\n",
       "  468.0,\n",
       "  507.0,\n",
       "  653.0,\n",
       "  1037.0,\n",
       "  239.0,\n",
       "  647.0,\n",
       "  322.0,\n",
       "  270.0,\n",
       "  380.0,\n",
       "  903.0,\n",
       "  615.0,\n",
       "  179.0,\n",
       "  728.0,\n",
       "  492.0,\n",
       "  781.0,\n",
       "  747.0,\n",
       "  870.0,\n",
       "  132.0,\n",
       "  362.0,\n",
       "  458.0,\n",
       "  317.0],\n",
       " 'GO:0031593-polyubiquitin modification-dependent protein binding': [641.0,\n",
       "  440.0,\n",
       "  491.0,\n",
       "  498.0,\n",
       "  560.0,\n",
       "  380.0,\n",
       "  1072.0,\n",
       "  403.0],\n",
       " 'GO:1902653-secondary alcohol biosynthetic process': [642.0,\n",
       "  622.0,\n",
       "  565.0,\n",
       "  764.0,\n",
       "  754.0,\n",
       "  324.0,\n",
       "  497.0,\n",
       "  449.0,\n",
       "  697.0,\n",
       "  1134.0,\n",
       "  283.0,\n",
       "  434.0,\n",
       "  889.0,\n",
       "  854.0,\n",
       "  412.0,\n",
       "  92.0,\n",
       "  1105.0,\n",
       "  691.0,\n",
       "  472.0],\n",
       " 'GO:0071276-cellular response to cadmium ion': [643.0,\n",
       "  271.0,\n",
       "  277.0,\n",
       "  490.0,\n",
       "  476.0],\n",
       " 'GO:0042162-telomeric DNA binding': [644.0, 541.0, 241.0, 219.0, 214.0],\n",
       " 'GO:1901661-quinone metabolic process': [645.0, 522.0, 258.0],\n",
       " 'GO:1901799-negative regulation of proteasomal protein catabolic process': [646.0,\n",
       "  812.0,\n",
       "  1753.0,\n",
       "  940.0,\n",
       "  485.0],\n",
       " 'GO:0075733-intracellular transport of virus': [647.0,\n",
       "  694.0,\n",
       "  548.0,\n",
       "  344.0,\n",
       "  1409.0,\n",
       "  362.0,\n",
       "  455.0,\n",
       "  758.0],\n",
       " 'GO:0032210-regulation of telomere maintenance via telomerase': [648.0,\n",
       "  1003.0,\n",
       "  331.0],\n",
       " 'GO:0031369-translation initiation factor binding': [649.0,\n",
       "  319.0,\n",
       "  416.0,\n",
       "  174.0,\n",
       "  535.0,\n",
       "  86.0,\n",
       "  200.0,\n",
       "  210.0,\n",
       "  279.0],\n",
       " 'GO:1905819-negative regulation of chromosome separation': [650.0,\n",
       "  501.0,\n",
       "  1055.0,\n",
       "  321.0,\n",
       "  673.0,\n",
       "  395.0,\n",
       "  216.0,\n",
       "  329.0],\n",
       " 'GO:1902369-negative regulation of RNA catabolic process': [651.0,\n",
       "  871.0,\n",
       "  820.0,\n",
       "  789.0,\n",
       "  426.0],\n",
       " 'GO:0031093-platelet alpha granule lumen': [652.0,\n",
       "  223.0,\n",
       "  77.0,\n",
       "  211.0,\n",
       "  232.0,\n",
       "  461.0,\n",
       "  356.0,\n",
       "  765.0,\n",
       "  131.0,\n",
       "  594.0,\n",
       "  151.0,\n",
       "  52.0,\n",
       "  126.0,\n",
       "  415.0,\n",
       "  286.0,\n",
       "  203.0,\n",
       "  737.0,\n",
       "  303.0,\n",
       "  307.0,\n",
       "  342.0,\n",
       "  414.0,\n",
       "  459.0,\n",
       "  101.0],\n",
       " 'GO:0051972-regulation of telomerase activity': [653.0,\n",
       "  564.0,\n",
       "  227.0,\n",
       "  593.0,\n",
       "  1699.0,\n",
       "  708.0,\n",
       "  275.0],\n",
       " 'GO:0006695-cholesterol biosynthetic process': [654.0,\n",
       "  638.0,\n",
       "  580.0,\n",
       "  779.0,\n",
       "  766.0,\n",
       "  504.0,\n",
       "  455.0,\n",
       "  707.0,\n",
       "  1156.0,\n",
       "  292.0,\n",
       "  447.0,\n",
       "  894.0,\n",
       "  819.0,\n",
       "  422.0,\n",
       "  93.0,\n",
       "  1093.0,\n",
       "  708.0,\n",
       "  488.0],\n",
       " 'GO:0006890-retrograde vesicle-mediated transport, Golgi to ER': [655.0,\n",
       "  599.0,\n",
       "  519.0,\n",
       "  892.0,\n",
       "  1109.0,\n",
       "  956.0,\n",
       "  1217.0,\n",
       "  250.0,\n",
       "  222.0,\n",
       "  451.0,\n",
       "  958.0,\n",
       "  894.0,\n",
       "  290.0,\n",
       "  690.0,\n",
       "  351.0,\n",
       "  887.0,\n",
       "  737.0,\n",
       "  1417.0,\n",
       "  367.0,\n",
       "  143.0,\n",
       "  1385.0,\n",
       "  725.0,\n",
       "  712.0],\n",
       " 'GO:1902230-negative regulation of intrinsic apoptotic signaling pathway in response to DNA damage': [656.0,\n",
       "  294.0,\n",
       "  365.0,\n",
       "  156.0,\n",
       "  432.0,\n",
       "  676.0,\n",
       "  675.0,\n",
       "  647.0],\n",
       " 'GO:0043534-blood vessel endothelial cell migration': [657.0,\n",
       "  533.0,\n",
       "  129.0,\n",
       "  113.0,\n",
       "  357.0,\n",
       "  615.0,\n",
       "  588.0,\n",
       "  989.0,\n",
       "  319.0,\n",
       "  671.0,\n",
       "  318.0,\n",
       "  218.0,\n",
       "  420.0,\n",
       "  886.0,\n",
       "  692.0,\n",
       "  232.0,\n",
       "  655.0,\n",
       "  592.0,\n",
       "  892.0,\n",
       "  519.0,\n",
       "  664.0,\n",
       "  135.0,\n",
       "  448.0,\n",
       "  620.0,\n",
       "  368.0],\n",
       " 'GO:0044783-G1 DNA damage checkpoint': [658.0,\n",
       "  451.0,\n",
       "  713.0,\n",
       "  606.0,\n",
       "  796.0,\n",
       "  302.0,\n",
       "  746.0,\n",
       "  976.0,\n",
       "  1425.0,\n",
       "  424.0,\n",
       "  683.0,\n",
       "  848.0,\n",
       "  1011.0,\n",
       "  582.0,\n",
       "  506.0],\n",
       " 'GO:0061756-leukocyte adhesion to vascular endothelial cell': [659.0,\n",
       "  241.0,\n",
       "  415.0,\n",
       "  169.0,\n",
       "  161.0,\n",
       "  462.0,\n",
       "  307.0,\n",
       "  604.0,\n",
       "  217.0,\n",
       "  361.0,\n",
       "  408.0,\n",
       "  312.0,\n",
       "  452.0,\n",
       "  227.0],\n",
       " 'GO:0061418-regulation of transcription from RNA polymerase II promoter in response to hypoxia': [660.0,\n",
       "  409.0,\n",
       "  334.0,\n",
       "  464.0,\n",
       "  476.0,\n",
       "  212.0,\n",
       "  272.0,\n",
       "  616.0,\n",
       "  409.0,\n",
       "  594.0,\n",
       "  541.0,\n",
       "  272.0,\n",
       "  440.0,\n",
       "  414.0,\n",
       "  488.0,\n",
       "  374.0,\n",
       "  769.0,\n",
       "  369.0,\n",
       "  1023.0,\n",
       "  366.0,\n",
       "  298.0],\n",
       " 'GO:0045066-regulatory T cell differentiation': [661.0,\n",
       "  351.0,\n",
       "  440.0,\n",
       "  852.0,\n",
       "  279.0],\n",
       " 'GO:1903427-negative regulation of reactive oxygen species biosynthetic process': [662.0,\n",
       "  49.0,\n",
       "  376.0,\n",
       "  419.0,\n",
       "  214.0,\n",
       "  260.0,\n",
       "  892.0,\n",
       "  350.0,\n",
       "  517.0,\n",
       "  497.0,\n",
       "  261.0],\n",
       " 'GO:0002478-antigen processing and presentation of exogenous peptide antigen': [663.0,\n",
       "  361.0,\n",
       "  220.0,\n",
       "  343.0,\n",
       "  593.0,\n",
       "  1288.0,\n",
       "  668.0,\n",
       "  327.0,\n",
       "  351.0,\n",
       "  801.0,\n",
       "  368.0,\n",
       "  317.0,\n",
       "  508.0,\n",
       "  493.0,\n",
       "  154.0,\n",
       "  391.0,\n",
       "  587.0,\n",
       "  555.0,\n",
       "  335.0,\n",
       "  520.0,\n",
       "  549.0,\n",
       "  308.0,\n",
       "  696.0,\n",
       "  354.0,\n",
       "  479.0],\n",
       " 'GO:0019213-deacetylase activity': [664.0,\n",
       "  347.0,\n",
       "  492.0,\n",
       "  574.0,\n",
       "  584.0,\n",
       "  476.0,\n",
       "  1430.0,\n",
       "  434.0,\n",
       "  265.0],\n",
       " 'GO:0043130-ubiquitin binding': [665.0,\n",
       "  708.0,\n",
       "  192.0,\n",
       "  648.0,\n",
       "  785.0,\n",
       "  1316.0,\n",
       "  680.0,\n",
       "  430.0,\n",
       "  1616.0,\n",
       "  468.0,\n",
       "  704.0,\n",
       "  637.0,\n",
       "  981.0,\n",
       "  415.0,\n",
       "  115.0,\n",
       "  1418.0,\n",
       "  827.0,\n",
       "  529.0],\n",
       " 'GO:1902186-regulation of viral release from host cell': 666.0,\n",
       " 'GO:1905898-positive regulation of response to endoplasmic reticulum stress': [667.0,\n",
       "  496.0,\n",
       "  833.0],\n",
       " 'GO:0040036-regulation of fibroblast growth factor receptor signaling pathway': [668.0,\n",
       "  102.0,\n",
       "  159.0,\n",
       "  424.0,\n",
       "  217.0,\n",
       "  234.0,\n",
       "  115.0,\n",
       "  463.0,\n",
       "  181.0,\n",
       "  394.0,\n",
       "  283.0,\n",
       "  251.0,\n",
       "  331.0],\n",
       " 'GO:1902579-multi-organism localization': [669.5,\n",
       "  709.5,\n",
       "  641.5,\n",
       "  1336.5,\n",
       "  639.5,\n",
       "  327.5,\n",
       "  1280.5,\n",
       "  448.5,\n",
       "  485.5,\n",
       "  986.5,\n",
       "  463.5,\n",
       "  657.5],\n",
       " 'GO:0044766-multi-organism transport': [669.5,\n",
       "  709.5,\n",
       "  641.5,\n",
       "  1336.5,\n",
       "  639.5,\n",
       "  327.5,\n",
       "  1280.5,\n",
       "  448.5,\n",
       "  485.5,\n",
       "  986.5,\n",
       "  463.5,\n",
       "  657.5],\n",
       " 'GO:0048255-mRNA stabilization': [671.0, 361.0, 631.0, 276.0],\n",
       " 'GO:0048020-CCR chemokine receptor binding': [672.0,\n",
       "  194.0,\n",
       "  62.0,\n",
       "  154.0,\n",
       "  222.0,\n",
       "  402.0,\n",
       "  419.0,\n",
       "  43.0,\n",
       "  175.0,\n",
       "  24.0,\n",
       "  376.0,\n",
       "  421.0,\n",
       "  115.0,\n",
       "  165.0,\n",
       "  406.0,\n",
       "  226.0,\n",
       "  331.0,\n",
       "  248.0,\n",
       "  367.0],\n",
       " 'GO:0070498-interleukin-1-mediated signaling pathway': [673.0,\n",
       "  491.0,\n",
       "  647.0,\n",
       "  775.0,\n",
       "  1363.0,\n",
       "  931.0,\n",
       "  332.0,\n",
       "  306.0,\n",
       "  448.0,\n",
       "  637.0,\n",
       "  1024.0,\n",
       "  711.0,\n",
       "  423.0,\n",
       "  540.0,\n",
       "  437.0,\n",
       "  493.0,\n",
       "  490.0,\n",
       "  929.0,\n",
       "  404.0,\n",
       "  119.0,\n",
       "  1448.0,\n",
       "  497.0,\n",
       "  596.0],\n",
       " 'GO:0046782-regulation of viral transcription': [674.0,\n",
       "  786.0,\n",
       "  779.0,\n",
       "  1308.0,\n",
       "  777.0,\n",
       "  455.0,\n",
       "  987.0,\n",
       "  1309.0,\n",
       "  307.0,\n",
       "  478.0,\n",
       "  615.0,\n",
       "  1126.0,\n",
       "  556.0,\n",
       "  417.0,\n",
       "  545.0],\n",
       " 'GO:0019432-triglyceride biosynthetic process': [675.0,\n",
       "  471.0,\n",
       "  451.0,\n",
       "  415.0,\n",
       "  800.0,\n",
       "  281.0],\n",
       " 'GO:0006378-mRNA polyadenylation': [676.0,\n",
       "  607.0,\n",
       "  297.0,\n",
       "  1042.0,\n",
       "  231.0,\n",
       "  513.0,\n",
       "  162.0],\n",
       " 'GO:1905897-regulation of response to endoplasmic reticulum stress': [677.0,\n",
       "  750.0,\n",
       "  454.0,\n",
       "  927.0,\n",
       "  904.0,\n",
       "  977.0,\n",
       "  668.0,\n",
       "  631.0,\n",
       "  1081.0,\n",
       "  786.0,\n",
       "  397.0,\n",
       "  1081.0,\n",
       "  716.0,\n",
       "  952.0,\n",
       "  439.0,\n",
       "  1183.0,\n",
       "  533.0,\n",
       "  362.0],\n",
       " 'GO:0043631-RNA polyadenylation': [678.0,\n",
       "  619.0,\n",
       "  308.0,\n",
       "  1130.0,\n",
       "  246.0,\n",
       "  552.0,\n",
       "  250.0],\n",
       " 'GO:0000070-mitotic sister chromatid segregation': [679.0,\n",
       "  819.0,\n",
       "  559.0,\n",
       "  422.0,\n",
       "  962.0,\n",
       "  1143.0,\n",
       "  1120.0,\n",
       "  799.0,\n",
       "  500.0,\n",
       "  421.0,\n",
       "  826.0,\n",
       "  1237.0,\n",
       "  1222.0,\n",
       "  268.0,\n",
       "  1302.0,\n",
       "  409.0,\n",
       "  780.0,\n",
       "  556.0,\n",
       "  1533.0,\n",
       "  206.0,\n",
       "  161.0,\n",
       "  1571.0,\n",
       "  903.0,\n",
       "  861.0],\n",
       " 'GO:0070063-RNA polymerase binding': [680.0,\n",
       "  1193.0,\n",
       "  1521.0,\n",
       "  376.0,\n",
       "  1516.0,\n",
       "  446.0,\n",
       "  1221.0,\n",
       "  1379.0,\n",
       "  278.0,\n",
       "  1502.0,\n",
       "  697.0],\n",
       " 'GO:1903214-regulation of protein targeting to mitochondrion': [681.0,\n",
       "  410.0,\n",
       "  677.0,\n",
       "  724.0,\n",
       "  404.0,\n",
       "  579.0,\n",
       "  1174.0,\n",
       "  425.0,\n",
       "  685.0,\n",
       "  884.0,\n",
       "  677.0,\n",
       "  323.0],\n",
       " 'GO:0070316-regulation of G0 to G1 transition': [682.0, 496.0, 1356.0, 239.0],\n",
       " 'GO:1904752-regulation of vascular associated smooth muscle cell migration': [683.5,\n",
       "  104.5,\n",
       "  281.5,\n",
       "  161.5,\n",
       "  169.5,\n",
       "  928.5,\n",
       "  351.5,\n",
       "  278.5,\n",
       "  275.5],\n",
       " 'GO:1904738-vascular associated smooth muscle cell migration': [683.5,\n",
       "  104.5,\n",
       "  281.5,\n",
       "  161.5,\n",
       "  169.5,\n",
       "  928.5,\n",
       "  351.5,\n",
       "  278.5,\n",
       "  275.5],\n",
       " 'GO:1905710-positive regulation of membrane permeability': [685.0,\n",
       "  524.0,\n",
       "  646.0,\n",
       "  952.0,\n",
       "  660.0,\n",
       "  769.0,\n",
       "  1273.0,\n",
       "  491.0,\n",
       "  608.0,\n",
       "  1214.0,\n",
       "  961.0],\n",
       " 'GO:1900180-regulation of protein localization to nucleus': [686.0,\n",
       "  900.0,\n",
       "  238.0,\n",
       "  826.0,\n",
       "  912.0,\n",
       "  672.0,\n",
       "  397.0,\n",
       "  654.0,\n",
       "  636.0,\n",
       "  1080.0,\n",
       "  547.0,\n",
       "  630.0,\n",
       "  677.0,\n",
       "  906.0,\n",
       "  815.0,\n",
       "  1364.0,\n",
       "  446.0,\n",
       "  235.0,\n",
       "  1412.0,\n",
       "  830.0,\n",
       "  886.0],\n",
       " 'GO:0061631-ubiquitin conjugating enzyme activity': [687.0, 207.0, 308.0],\n",
       " 'GO:0019985-translesion synthesis': [688.0,\n",
       "  176.0,\n",
       "  319.0,\n",
       "  395.0,\n",
       "  183.0,\n",
       "  462.0,\n",
       "  267.0,\n",
       "  435.0,\n",
       "  412.0,\n",
       "  744.0,\n",
       "  213.0,\n",
       "  336.0],\n",
       " 'GO:0010883-regulation of lipid storage': [689.0, 464.0, 1378.0],\n",
       " 'GO:0019825-oxygen binding': [690.0,\n",
       "  138.0,\n",
       "  48.0,\n",
       "  65.0,\n",
       "  496.0,\n",
       "  49.0,\n",
       "  225.0,\n",
       "  564.0,\n",
       "  233.0,\n",
       "  185.0,\n",
       "  702.0,\n",
       "  686.0,\n",
       "  175.0],\n",
       " 'GO:0007080-mitotic metaphase plate congression': [691.0,\n",
       "  632.0,\n",
       "  532.0,\n",
       "  478.0,\n",
       "  913.0,\n",
       "  293.0,\n",
       "  784.0,\n",
       "  488.0,\n",
       "  190.0],\n",
       " 'GO:0033045-regulation of sister chromatid segregation': [692.0,\n",
       "  704.0,\n",
       "  344.0,\n",
       "  841.0,\n",
       "  1381.0,\n",
       "  1051.0,\n",
       "  693.0,\n",
       "  343.0,\n",
       "  1578.0,\n",
       "  263.0,\n",
       "  322.0,\n",
       "  559.0,\n",
       "  640.0,\n",
       "  1157.0,\n",
       "  203.0,\n",
       "  748.0],\n",
       " 'GO:0006368-transcription elongation from RNA polymerase II promoter': [693.0,\n",
       "  785.0,\n",
       "  504.0,\n",
       "  1251.0,\n",
       "  1311.0,\n",
       "  877.0,\n",
       "  353.0,\n",
       "  749.0,\n",
       "  312.0,\n",
       "  401.0,\n",
       "  817.0,\n",
       "  1464.0,\n",
       "  365.0,\n",
       "  127.0,\n",
       "  307.0,\n",
       "  625.0],\n",
       " 'GO:0032981-mitochondrial respiratory chain complex I assembly': [694.5,\n",
       "  226.5,\n",
       "  520.5,\n",
       "  264.5,\n",
       "  1068.5,\n",
       "  409.5,\n",
       "  258.5,\n",
       "  184.5,\n",
       "  365.5,\n",
       "  271.5,\n",
       "  244.5,\n",
       "  349.5,\n",
       "  224.5,\n",
       "  135.5,\n",
       "  183.5,\n",
       "  201.5,\n",
       "  417.5,\n",
       "  272.5,\n",
       "  417.5,\n",
       "  409.5,\n",
       "  58.5,\n",
       "  1034.5,\n",
       "  252.5,\n",
       "  296.5],\n",
       " 'GO:0010257-NADH dehydrogenase complex assembly': [694.5,\n",
       "  226.5,\n",
       "  520.5,\n",
       "  264.5,\n",
       "  1068.5,\n",
       "  409.5,\n",
       "  258.5,\n",
       "  184.5,\n",
       "  365.5,\n",
       "  271.5,\n",
       "  244.5,\n",
       "  349.5,\n",
       "  224.5,\n",
       "  135.5,\n",
       "  183.5,\n",
       "  201.5,\n",
       "  417.5,\n",
       "  272.5,\n",
       "  417.5,\n",
       "  409.5,\n",
       "  58.5,\n",
       "  1034.5,\n",
       "  252.5,\n",
       "  296.5],\n",
       " 'GO:0043536-positive regulation of blood vessel endothelial cell migration': [696.0,\n",
       "  508.0,\n",
       "  131.0,\n",
       "  106.0,\n",
       "  282.0,\n",
       "  784.0,\n",
       "  436.0,\n",
       "  684.0,\n",
       "  405.0,\n",
       "  260.0,\n",
       "  192.0,\n",
       "  327.0,\n",
       "  663.0,\n",
       "  307.0,\n",
       "  683.0,\n",
       "  788.0,\n",
       "  790.0,\n",
       "  556.0,\n",
       "  310.0,\n",
       "  546.0,\n",
       "  322.0],\n",
       " 'GO:0009112-nucleobase metabolic process': [697.0,\n",
       "  484.0,\n",
       "  507.0,\n",
       "  352.0,\n",
       "  718.0],\n",
       " 'GO:0044455-mitochondrial membrane part': [698.0,\n",
       "  400.0,\n",
       "  633.0,\n",
       "  663.0,\n",
       "  520.0,\n",
       "  1333.0,\n",
       "  614.0,\n",
       "  472.0,\n",
       "  281.0,\n",
       "  773.0,\n",
       "  510.0,\n",
       "  414.0,\n",
       "  787.0,\n",
       "  640.0,\n",
       "  215.0,\n",
       "  380.0,\n",
       "  277.0,\n",
       "  690.0,\n",
       "  439.0,\n",
       "  877.0,\n",
       "  483.0,\n",
       "  116.0,\n",
       "  1453.0,\n",
       "  636.0,\n",
       "  460.0],\n",
       " 'GO:0065004-protein-DNA complex assembly': [699.0,\n",
       "  656.0,\n",
       "  325.0,\n",
       "  398.0,\n",
       "  831.0,\n",
       "  876.0,\n",
       "  1277.0,\n",
       "  781.0,\n",
       "  460.0,\n",
       "  433.0,\n",
       "  588.0,\n",
       "  1130.0,\n",
       "  1243.0,\n",
       "  227.0,\n",
       "  806.0,\n",
       "  312.0,\n",
       "  639.0,\n",
       "  647.0,\n",
       "  1266.0,\n",
       "  315.0,\n",
       "  420.0,\n",
       "  1696.0,\n",
       "  670.0,\n",
       "  589.0],\n",
       " 'GO:0044769-ATPase activity, coupled to transmembrane movement of ions, rotational mechanism': [700.0,\n",
       "  331.0,\n",
       "  452.0,\n",
       "  149.0,\n",
       "  301.0,\n",
       "  738.0,\n",
       "  149.0,\n",
       "  267.0,\n",
       "  322.0,\n",
       "  391.0],\n",
       " 'GO:0006656-phosphatidylcholine biosynthetic process': [701.0, 340.0],\n",
       " 'GO:0031099-regeneration': [702.0,\n",
       "  1280.0,\n",
       "  354.0,\n",
       "  934.0,\n",
       "  665.0,\n",
       "  878.0,\n",
       "  1039.0,\n",
       "  1828.0,\n",
       "  423.0,\n",
       "  1244.0,\n",
       "  606.0,\n",
       "  652.0,\n",
       "  1366.0,\n",
       "  959.0,\n",
       "  625.0,\n",
       "  1220.0,\n",
       "  1027.0,\n",
       "  857.0,\n",
       "  1517.0,\n",
       "  924.0,\n",
       "  1388.0,\n",
       "  1346.0],\n",
       " 'GO:0038083-peptidyl-tyrosine autophosphorylation': [703.0,\n",
       "  216.0,\n",
       "  1100.0,\n",
       "  255.0,\n",
       "  256.0,\n",
       "  657.0,\n",
       "  435.0],\n",
       " 'GO:0045815-positive regulation of gene expression, epigenetic': [704.0,\n",
       "  1117.0,\n",
       "  1171.0,\n",
       "  412.0,\n",
       "  1446.0,\n",
       "  879.0],\n",
       " 'GO:0030684-preribosome': [705.0,\n",
       "  574.0,\n",
       "  1002.0,\n",
       "  778.0,\n",
       "  354.0,\n",
       "  816.0,\n",
       "  302.0,\n",
       "  243.0,\n",
       "  765.0,\n",
       "  745.0,\n",
       "  1369.0,\n",
       "  281.0,\n",
       "  169.0],\n",
       " 'GO:0008210-estrogen metabolic process': [706.0,\n",
       "  239.0,\n",
       "  938.0,\n",
       "  223.0,\n",
       "  387.0,\n",
       "  226.0],\n",
       " 'GO:0070603-SWI/SNF superfamily-type complex': [707.0,\n",
       "  672.0,\n",
       "  1345.0,\n",
       "  381.0,\n",
       "  1651.0,\n",
       "  657.0,\n",
       "  1025.0,\n",
       "  1024.0,\n",
       "  469.0,\n",
       "  905.0],\n",
       " 'GO:0006401-RNA catabolic process': [708.0,\n",
       "  673.0,\n",
       "  571.0,\n",
       "  702.0,\n",
       "  832.0,\n",
       "  1074.0,\n",
       "  957.0,\n",
       "  796.0,\n",
       "  496.0,\n",
       "  981.0,\n",
       "  1127.0,\n",
       "  1126.0,\n",
       "  1093.0,\n",
       "  461.0,\n",
       "  1021.0,\n",
       "  614.0,\n",
       "  713.0,\n",
       "  754.0,\n",
       "  1263.0,\n",
       "  623.0,\n",
       "  237.0,\n",
       "  1535.0,\n",
       "  835.0,\n",
       "  705.0],\n",
       " 'GO:0006360-transcription by RNA polymerase I': [709.0,\n",
       "  736.0,\n",
       "  555.0,\n",
       "  1377.0,\n",
       "  425.0,\n",
       "  438.0,\n",
       "  638.0,\n",
       "  1200.0,\n",
       "  351.0],\n",
       " \"GO:0003899-DNA-directed 5'-3' RNA polymerase activity\": [710.0,\n",
       "  482.0,\n",
       "  505.0,\n",
       "  509.0,\n",
       "  275.0,\n",
       "  386.0,\n",
       "  212.0,\n",
       "  459.0,\n",
       "  385.0,\n",
       "  574.0],\n",
       " 'GO:0045841-negative regulation of mitotic metaphase/anaphase transition': [711.0,\n",
       "  1075.0,\n",
       "  226.0,\n",
       "  251.0],\n",
       " 'GO:0032873-negative regulation of stress-activated MAPK cascade': [712.5,\n",
       "  587.5,\n",
       "  1151.5,\n",
       "  256.5,\n",
       "  458.5,\n",
       "  966.5,\n",
       "  746.5,\n",
       "  478.5],\n",
       " 'GO:0070303-negative regulation of stress-activated protein kinase signaling cascade': [712.5,\n",
       "  587.5,\n",
       "  1151.5,\n",
       "  256.5,\n",
       "  458.5,\n",
       "  966.5,\n",
       "  746.5,\n",
       "  478.5],\n",
       " 'GO:0051784-negative regulation of nuclear division': [714.0,\n",
       "  458.0,\n",
       "  666.0,\n",
       "  690.0,\n",
       "  1434.0,\n",
       "  335.0,\n",
       "  391.0,\n",
       "  949.0,\n",
       "  625.0,\n",
       "  1295.0,\n",
       "  314.0,\n",
       "  818.0],\n",
       " 'GO:1902099-regulation of metaphase/anaphase transition of cell cycle': [715.0,\n",
       "  529.0,\n",
       "  349.0,\n",
       "  1158.0,\n",
       "  530.0,\n",
       "  512.0,\n",
       "  1395.0,\n",
       "  214.0,\n",
       "  353.0,\n",
       "  721.0,\n",
       "  536.0,\n",
       "  1222.0,\n",
       "  236.0,\n",
       "  278.0],\n",
       " 'GO:0031304-intrinsic component of mitochondrial inner membrane': [716.5,\n",
       "  429.5,\n",
       "  358.5,\n",
       "  565.5,\n",
       "  724.5,\n",
       "  374.5,\n",
       "  173.5,\n",
       "  646.5,\n",
       "  429.5,\n",
       "  320.5],\n",
       " 'GO:0031305-integral component of mitochondrial inner membrane': [716.5,\n",
       "  429.5,\n",
       "  358.5,\n",
       "  565.5,\n",
       "  724.5,\n",
       "  374.5,\n",
       "  173.5,\n",
       "  646.5,\n",
       "  429.5,\n",
       "  320.5],\n",
       " 'GO:0000932-P-body': [718.0,\n",
       "  1033.0,\n",
       "  1419.0,\n",
       "  1420.0,\n",
       "  806.0,\n",
       "  1097.0,\n",
       "  916.0,\n",
       "  601.0,\n",
       "  1251.0,\n",
       "  995.0,\n",
       "  434.0,\n",
       "  125.0,\n",
       "  771.0,\n",
       "  541.0],\n",
       " 'GO:0033683-nucleotide-excision repair, DNA incision': [719.0,\n",
       "  433.0,\n",
       "  437.0,\n",
       "  1427.0,\n",
       "  395.0,\n",
       "  316.0,\n",
       "  184.0],\n",
       " 'GO:0019840-isoprenoid binding': [720.0,\n",
       "  342.0,\n",
       "  39.0,\n",
       "  1040.0,\n",
       "  198.0,\n",
       "  223.0,\n",
       "  289.0,\n",
       "  249.0],\n",
       " 'GO:0140053-mitochondrial gene expression': [721.0,\n",
       "  479.0,\n",
       "  578.0,\n",
       "  1065.0,\n",
       "  785.0,\n",
       "  1659.0,\n",
       "  738.0,\n",
       "  391.0,\n",
       "  312.0,\n",
       "  287.0,\n",
       "  921.0,\n",
       "  1418.0,\n",
       "  743.0,\n",
       "  161.0,\n",
       "  985.0,\n",
       "  186.0,\n",
       "  518.0,\n",
       "  482.0,\n",
       "  1196.0,\n",
       "  271.0,\n",
       "  121.0,\n",
       "  1959.0,\n",
       "  467.0,\n",
       "  592.0],\n",
       " 'GO:0043535-regulation of blood vessel endothelial cell migration': [722.0,\n",
       "  539.0,\n",
       "  121.0,\n",
       "  123.0,\n",
       "  332.0,\n",
       "  616.0,\n",
       "  501.0,\n",
       "  1093.0,\n",
       "  287.0,\n",
       "  704.0,\n",
       "  238.0,\n",
       "  170.0,\n",
       "  303.0,\n",
       "  769.0,\n",
       "  728.0,\n",
       "  192.0,\n",
       "  673.0,\n",
       "  692.0,\n",
       "  630.0,\n",
       "  546.0,\n",
       "  787.0,\n",
       "  122.0,\n",
       "  365.0,\n",
       "  598.0,\n",
       "  332.0],\n",
       " 'GO:0022618-ribonucleoprotein complex assembly': [723.0,\n",
       "  776.0,\n",
       "  485.0,\n",
       "  613.0,\n",
       "  1087.0,\n",
       "  1126.0,\n",
       "  1249.0,\n",
       "  681.0,\n",
       "  425.0,\n",
       "  961.0,\n",
       "  1167.0,\n",
       "  1511.0,\n",
       "  1110.0,\n",
       "  344.0,\n",
       "  1382.0,\n",
       "  415.0,\n",
       "  691.0,\n",
       "  564.0,\n",
       "  1352.0,\n",
       "  399.0,\n",
       "  370.0,\n",
       "  2014.0,\n",
       "  981.0,\n",
       "  789.0],\n",
       " 'GO:0002437-inflammatory response to antigenic stimulus': [724.0,\n",
       "  369.0,\n",
       "  387.0,\n",
       "  470.0,\n",
       "  581.0,\n",
       "  328.0,\n",
       "  445.0,\n",
       "  702.0,\n",
       "  464.0,\n",
       "  535.0,\n",
       "  550.0,\n",
       "  389.0],\n",
       " 'GO:0046460-neutral lipid biosynthetic process': [725.5,\n",
       "  419.5,\n",
       "  449.5,\n",
       "  496.5,\n",
       "  680.5,\n",
       "  293.5],\n",
       " 'GO:0046463-acylglycerol biosynthetic process': [725.5,\n",
       "  419.5,\n",
       "  449.5,\n",
       "  496.5,\n",
       "  680.5,\n",
       "  293.5],\n",
       " 'GO:0071826-ribonucleoprotein complex subunit organization': [727.0,\n",
       "  807.0,\n",
       "  543.0,\n",
       "  672.0,\n",
       "  1164.0,\n",
       "  1189.0,\n",
       "  1341.0,\n",
       "  726.0,\n",
       "  441.0,\n",
       "  992.0,\n",
       "  1202.0,\n",
       "  1537.0,\n",
       "  1245.0,\n",
       "  382.0,\n",
       "  1413.0,\n",
       "  445.0,\n",
       "  728.0,\n",
       "  609.0,\n",
       "  1359.0,\n",
       "  436.0,\n",
       "  402.0,\n",
       "  2052.0,\n",
       "  1010.0,\n",
       "  917.0],\n",
       " 'GO:0019886-antigen processing and presentation of exogenous peptide antigen via MHC class II': [728.0,\n",
       "  431.0,\n",
       "  236.0,\n",
       "  268.0,\n",
       "  822.0,\n",
       "  1075.0,\n",
       "  824.0,\n",
       "  392.0,\n",
       "  898.0,\n",
       "  347.0,\n",
       "  758.0,\n",
       "  565.0,\n",
       "  183.0,\n",
       "  385.0,\n",
       "  877.0,\n",
       "  682.0,\n",
       "  445.0,\n",
       "  768.0,\n",
       "  676.0,\n",
       "  896.0,\n",
       "  435.0,\n",
       "  590.0],\n",
       " 'GO:1904813-ficolin-1-rich granule lumen': [729.0,\n",
       "  415.0,\n",
       "  245.0,\n",
       "  386.0,\n",
       "  583.0,\n",
       "  838.0,\n",
       "  531.0,\n",
       "  295.0,\n",
       "  318.0,\n",
       "  1067.0,\n",
       "  475.0,\n",
       "  271.0,\n",
       "  346.0,\n",
       "  420.0,\n",
       "  192.0,\n",
       "  635.0,\n",
       "  717.0,\n",
       "  449.0,\n",
       "  311.0,\n",
       "  480.0,\n",
       "  562.0,\n",
       "  249.0,\n",
       "  808.0,\n",
       "  1199.0,\n",
       "  397.0],\n",
       " 'GO:0044784-metaphase/anaphase transition of cell cycle': [730.0,\n",
       "  664.0,\n",
       "  377.0,\n",
       "  1210.0,\n",
       "  664.0,\n",
       "  500.0,\n",
       "  1594.0,\n",
       "  222.0,\n",
       "  372.0,\n",
       "  694.0,\n",
       "  527.0,\n",
       "  245.0,\n",
       "  356.0],\n",
       " 'GO:0006342-chromatin silencing': [731.0,\n",
       "  1155.0,\n",
       "  408.0,\n",
       "  717.0,\n",
       "  721.0,\n",
       "  1946.0,\n",
       "  1501.0,\n",
       "  831.0,\n",
       "  859.0,\n",
       "  1966.0,\n",
       "  586.0,\n",
       "  1167.0,\n",
       "  654.0,\n",
       "  1213.0,\n",
       "  1432.0,\n",
       "  625.0,\n",
       "  1340.0],\n",
       " 'GO:0016126-sterol biosynthetic process': [732.0,\n",
       "  749.0,\n",
       "  650.0,\n",
       "  774.0,\n",
       "  895.0,\n",
       "  353.0,\n",
       "  640.0,\n",
       "  518.0,\n",
       "  815.0,\n",
       "  1459.0,\n",
       "  322.0,\n",
       "  527.0,\n",
       "  1000.0,\n",
       "  991.0,\n",
       "  544.0,\n",
       "  101.0,\n",
       "  1186.0,\n",
       "  913.0,\n",
       "  505.0],\n",
       " 'GO:0051983-regulation of chromosome segregation': [733.0,\n",
       "  775.0,\n",
       "  382.0,\n",
       "  1057.0,\n",
       "  1384.0,\n",
       "  1050.0,\n",
       "  789.0,\n",
       "  359.0,\n",
       "  1591.0,\n",
       "  326.0,\n",
       "  326.0,\n",
       "  595.0,\n",
       "  635.0,\n",
       "  1416.0,\n",
       "  218.0,\n",
       "  167.0,\n",
       "  887.0],\n",
       " 'GO:0070301-cellular response to hydrogen peroxide': [734.0,\n",
       "  787.0,\n",
       "  733.0,\n",
       "  489.0,\n",
       "  1101.0,\n",
       "  751.0,\n",
       "  540.0,\n",
       "  336.0,\n",
       "  687.0,\n",
       "  465.0,\n",
       "  1361.0,\n",
       "  599.0,\n",
       "  754.0,\n",
       "  980.0,\n",
       "  869.0,\n",
       "  872.0,\n",
       "  742.0,\n",
       "  899.0,\n",
       "  797.0,\n",
       "  490.0],\n",
       " 'GO:0033146-regulation of intracellular estrogen receptor signaling pathway': [735.0,\n",
       "  339.0,\n",
       "  457.0,\n",
       "  1041.0],\n",
       " 'GO:0101003-ficolin-1-rich granule membrane': [736.0,\n",
       "  485.0,\n",
       "  209.0,\n",
       "  782.0,\n",
       "  572.0,\n",
       "  724.0,\n",
       "  133.0,\n",
       "  321.0,\n",
       "  1343.0,\n",
       "  167.0,\n",
       "  796.0,\n",
       "  251.0,\n",
       "  650.0,\n",
       "  856.0],\n",
       " 'GO:1903201-regulation of oxidative stress-induced cell death': [737.0,\n",
       "  949.0,\n",
       "  1044.0,\n",
       "  995.0],\n",
       " 'GO:0043620-regulation of DNA-templated transcription in response to stress': [738.0,\n",
       "  717.0,\n",
       "  384.0,\n",
       "  564.0,\n",
       "  653.0,\n",
       "  1396.0,\n",
       "  740.0,\n",
       "  370.0,\n",
       "  374.0,\n",
       "  527.0,\n",
       "  614.0,\n",
       "  806.0,\n",
       "  911.0,\n",
       "  446.0,\n",
       "  773.0,\n",
       "  406.0,\n",
       "  663.0,\n",
       "  492.0,\n",
       "  1129.0,\n",
       "  449.0,\n",
       "  196.0,\n",
       "  1249.0,\n",
       "  539.0,\n",
       "  384.0],\n",
       " 'GO:0050704-regulation of interleukin-1 secretion': [739.0,\n",
       "  284.0,\n",
       "  68.0,\n",
       "  466.0,\n",
       "  347.0,\n",
       "  484.0,\n",
       "  201.0,\n",
       "  311.0,\n",
       "  1098.0,\n",
       "  402.0,\n",
       "  391.0,\n",
       "  363.0,\n",
       "  503.0,\n",
       "  696.0],\n",
       " 'GO:0030968-endoplasmic reticulum unfolded protein response': [740.0,\n",
       "  1304.0,\n",
       "  549.0,\n",
       "  755.0,\n",
       "  1253.0,\n",
       "  1080.0,\n",
       "  1182.0,\n",
       "  618.0,\n",
       "  793.0,\n",
       "  1239.0,\n",
       "  1373.0,\n",
       "  562.0,\n",
       "  741.0,\n",
       "  484.0,\n",
       "  1345.0,\n",
       "  847.0,\n",
       "  1521.0,\n",
       "  598.0,\n",
       "  140.0,\n",
       "  1421.0,\n",
       "  571.0,\n",
       "  619.0],\n",
       " 'GO:0043618-regulation of transcription from RNA polymerase II promoter in response to stress': [741.0,\n",
       "  703.0,\n",
       "  409.0,\n",
       "  579.0,\n",
       "  619.0,\n",
       "  1378.0,\n",
       "  725.0,\n",
       "  336.0,\n",
       "  393.0,\n",
       "  580.0,\n",
       "  605.0,\n",
       "  864.0,\n",
       "  973.0,\n",
       "  448.0,\n",
       "  802.0,\n",
       "  436.0,\n",
       "  681.0,\n",
       "  515.0,\n",
       "  1167.0,\n",
       "  445.0,\n",
       "  177.0,\n",
       "  1321.0,\n",
       "  550.0,\n",
       "  386.0],\n",
       " 'GO:0048002-antigen processing and presentation of peptide antigen': [742.0,\n",
       "  406.0,\n",
       "  283.0,\n",
       "  413.0,\n",
       "  651.0,\n",
       "  1348.0,\n",
       "  684.0,\n",
       "  361.0,\n",
       "  386.0,\n",
       "  895.0,\n",
       "  410.0,\n",
       "  352.0,\n",
       "  618.0,\n",
       "  669.0,\n",
       "  203.0,\n",
       "  433.0,\n",
       "  625.0,\n",
       "  634.0,\n",
       "  356.0,\n",
       "  584.0,\n",
       "  606.0,\n",
       "  794.0,\n",
       "  463.0,\n",
       "  539.0],\n",
       " 'GO:0038066-p38MAPK cascade': [743.0,\n",
       "  293.0,\n",
       "  936.0,\n",
       "  951.0,\n",
       "  1027.0,\n",
       "  197.0,\n",
       "  570.0,\n",
       "  999.0,\n",
       "  338.0],\n",
       " 'GO:0046329-negative regulation of JNK cascade': 744.0,\n",
       " 'GO:1903312-negative regulation of mRNA metabolic process': [745.0,\n",
       "  845.0,\n",
       "  326.0,\n",
       "  760.0,\n",
       "  1444.0,\n",
       "  1263.0,\n",
       "  901.0,\n",
       "  1932.0,\n",
       "  584.0,\n",
       "  587.0,\n",
       "  283.0,\n",
       "  1702.0],\n",
       " 'GO:0006446-regulation of translational initiation': [746.0,\n",
       "  633.0,\n",
       "  500.0,\n",
       "  1015.0,\n",
       "  1261.0,\n",
       "  721.0,\n",
       "  469.0,\n",
       "  1210.0,\n",
       "  239.0,\n",
       "  248.0,\n",
       "  711.0,\n",
       "  568.0,\n",
       "  286.0,\n",
       "  841.0,\n",
       "  654.0],\n",
       " 'GO:0002832-negative regulation of response to biotic stimulus': [747.0,\n",
       "  441.0,\n",
       "  300.0,\n",
       "  325.0,\n",
       "  424.0,\n",
       "  364.0,\n",
       "  317.0,\n",
       "  516.0,\n",
       "  575.0,\n",
       "  482.0],\n",
       " 'GO:1902229-regulation of intrinsic apoptotic signaling pathway in response to DNA damage': [748.0,\n",
       "  395.0,\n",
       "  355.0,\n",
       "  231.0,\n",
       "  590.0,\n",
       "  771.0,\n",
       "  286.0,\n",
       "  456.0,\n",
       "  617.0,\n",
       "  869.0],\n",
       " 'GO:0000731-DNA synthesis involved in DNA repair': [749.0,\n",
       "  314.0,\n",
       "  540.0,\n",
       "  696.0,\n",
       "  383.0,\n",
       "  495.0,\n",
       "  571.0,\n",
       "  490.0,\n",
       "  579.0,\n",
       "  1210.0,\n",
       "  224.0],\n",
       " 'GO:0051310-metaphase plate congression': [750.0,\n",
       "  873.0,\n",
       "  630.0,\n",
       "  570.0,\n",
       "  963.0,\n",
       "  246.0,\n",
       "  381.0,\n",
       "  985.0,\n",
       "  550.0,\n",
       "  1330.0,\n",
       "  188.0],\n",
       " 'GO:2000059-negative regulation of ubiquitin-dependent protein catabolic process': [751.0,\n",
       "  653.0,\n",
       "  859.0,\n",
       "  1331.0,\n",
       "  476.0,\n",
       "  760.0,\n",
       "  353.0],\n",
       " 'GO:0000956-nuclear-transcribed mRNA catabolic process': [752.0,\n",
       "  690.0,\n",
       "  1042.0,\n",
       "  776.0,\n",
       "  976.0,\n",
       "  897.0,\n",
       "  899.0,\n",
       "  516.0,\n",
       "  1277.0,\n",
       "  1242.0,\n",
       "  1369.0,\n",
       "  627.0,\n",
       "  702.0,\n",
       "  715.0,\n",
       "  897.0,\n",
       "  976.0,\n",
       "  1354.0,\n",
       "  784.0,\n",
       "  234.0,\n",
       "  1741.0,\n",
       "  1220.0,\n",
       "  685.0],\n",
       " 'GO:1903533-regulation of protein targeting': [753.0,\n",
       "  630.0,\n",
       "  712.0,\n",
       "  923.0,\n",
       "  775.0,\n",
       "  681.0,\n",
       "  420.0,\n",
       "  270.0,\n",
       "  717.0,\n",
       "  603.0,\n",
       "  1265.0,\n",
       "  324.0,\n",
       "  609.0,\n",
       "  671.0,\n",
       "  593.0,\n",
       "  1226.0,\n",
       "  595.0,\n",
       "  124.0,\n",
       "  1030.0,\n",
       "  917.0,\n",
       "  478.0],\n",
       " 'GO:0048770-pigment granule': [754.5,\n",
       "  447.5,\n",
       "  359.5,\n",
       "  594.5,\n",
       "  1061.5,\n",
       "  523.5,\n",
       "  477.5,\n",
       "  320.5,\n",
       "  638.5,\n",
       "  387.5,\n",
       "  519.5,\n",
       "  767.5,\n",
       "  273.5,\n",
       "  338.5,\n",
       "  554.5,\n",
       "  665.5,\n",
       "  475.5,\n",
       "  791.5,\n",
       "  596.5,\n",
       "  172.5,\n",
       "  657.5,\n",
       "  1013.5,\n",
       "  701.5],\n",
       " 'GO:0042470-melanosome': [754.5,\n",
       "  447.5,\n",
       "  359.5,\n",
       "  594.5,\n",
       "  1061.5,\n",
       "  523.5,\n",
       "  477.5,\n",
       "  320.5,\n",
       "  638.5,\n",
       "  387.5,\n",
       "  519.5,\n",
       "  767.5,\n",
       "  273.5,\n",
       "  338.5,\n",
       "  554.5,\n",
       "  665.5,\n",
       "  475.5,\n",
       "  791.5,\n",
       "  596.5,\n",
       "  172.5,\n",
       "  657.5,\n",
       "  1013.5,\n",
       "  701.5],\n",
       " 'GO:0006402-mRNA catabolic process': [756.0,\n",
       "  680.0,\n",
       "  629.0,\n",
       "  694.0,\n",
       "  800.0,\n",
       "  1152.0,\n",
       "  985.0,\n",
       "  805.0,\n",
       "  475.0,\n",
       "  997.0,\n",
       "  1133.0,\n",
       "  1153.0,\n",
       "  1094.0,\n",
       "  508.0,\n",
       "  1070.0,\n",
       "  594.0,\n",
       "  773.0,\n",
       "  830.0,\n",
       "  1179.0,\n",
       "  643.0,\n",
       "  265.0,\n",
       "  1602.0,\n",
       "  900.0,\n",
       "  709.0],\n",
       " 'GO:0034774-secretory granule lumen': [757.0,\n",
       "  451.0,\n",
       "  146.0,\n",
       "  502.0,\n",
       "  544.0,\n",
       "  766.0,\n",
       "  503.0,\n",
       "  670.0,\n",
       "  271.0,\n",
       "  1125.0,\n",
       "  391.0,\n",
       "  210.0,\n",
       "  360.0,\n",
       "  491.0,\n",
       "  223.0,\n",
       "  525.0,\n",
       "  935.0,\n",
       "  480.0,\n",
       "  328.0,\n",
       "  461.0,\n",
       "  719.0,\n",
       "  329.0,\n",
       "  724.0,\n",
       "  1121.0,\n",
       "  473.0],\n",
       " 'GO:0030490-maturation of SSU-rRNA': [758.0,\n",
       "  524.0,\n",
       "  613.0,\n",
       "  912.0,\n",
       "  345.0,\n",
       "  743.0,\n",
       "  168.0],\n",
       " 'GO:0032204-regulation of telomere maintenance': [759.0,\n",
       "  1497.0,\n",
       "  1043.0,\n",
       "  2059.0,\n",
       "  640.0,\n",
       "  1215.0,\n",
       "  553.0,\n",
       "  1463.0],\n",
       " 'GO:0015030-Cajal body': [760.0,\n",
       "  723.0,\n",
       "  379.0,\n",
       "  871.0,\n",
       "  611.0,\n",
       "  1387.0,\n",
       "  926.0,\n",
       "  391.0,\n",
       "  584.0,\n",
       "  882.0,\n",
       "  182.0,\n",
       "  249.0,\n",
       "  436.0,\n",
       "  450.0,\n",
       "  1163.0,\n",
       "  195.0,\n",
       "  659.0,\n",
       "  600.0],\n",
       " 'GO:0019884-antigen processing and presentation of exogenous antigen': [761.0,\n",
       "  389.0,\n",
       "  258.0,\n",
       "  392.0,\n",
       "  659.0,\n",
       "  1344.0,\n",
       "  742.0,\n",
       "  368.0,\n",
       "  395.0,\n",
       "  896.0,\n",
       "  423.0,\n",
       "  353.0,\n",
       "  612.0,\n",
       "  596.0,\n",
       "  178.0,\n",
       "  476.0,\n",
       "  647.0,\n",
       "  624.0,\n",
       "  370.0,\n",
       "  626.0,\n",
       "  618.0,\n",
       "  328.0,\n",
       "  770.0,\n",
       "  398.0,\n",
       "  579.0],\n",
       " 'GO:0000462-maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)': [762.0,\n",
       "  343.0,\n",
       "  1137.0,\n",
       "  311.0,\n",
       "  133.0],\n",
       " 'GO:0006354-DNA-templated transcription, elongation': [763.0,\n",
       "  855.0,\n",
       "  788.0,\n",
       "  1179.0,\n",
       "  1603.0,\n",
       "  970.0,\n",
       "  388.0,\n",
       "  969.0,\n",
       "  366.0,\n",
       "  429.0,\n",
       "  709.0,\n",
       "  1224.0,\n",
       "  1715.0,\n",
       "  393.0,\n",
       "  186.0,\n",
       "  615.0,\n",
       "  838.0],\n",
       " 'GO:1903051-negative regulation of proteolysis involved in cellular protein catabolic process': [764.0,\n",
       "  822.0,\n",
       "  1047.0,\n",
       "  1776.0,\n",
       "  539.0,\n",
       "  984.0,\n",
       "  489.0,\n",
       "  765.0],\n",
       " 'GO:0090199-regulation of release of cytochrome c from mitochondria': [765.0,\n",
       "  547.0,\n",
       "  332.0,\n",
       "  657.0,\n",
       "  429.0,\n",
       "  584.0,\n",
       "  907.0,\n",
       "  163.0,\n",
       "  277.0,\n",
       "  220.0,\n",
       "  830.0,\n",
       "  459.0,\n",
       "  639.0,\n",
       "  504.0],\n",
       " 'GO:0031970-organelle envelope lumen': [766.0,\n",
       "  711.0,\n",
       "  614.0,\n",
       "  617.0,\n",
       "  745.0,\n",
       "  440.0,\n",
       "  350.0,\n",
       "  910.0,\n",
       "  616.0,\n",
       "  962.0,\n",
       "  989.0,\n",
       "  308.0,\n",
       "  500.0,\n",
       "  777.0,\n",
       "  763.0,\n",
       "  570.0,\n",
       "  659.0,\n",
       "  586.0,\n",
       "  1472.0,\n",
       "  498.0,\n",
       "  598.0],\n",
       " 'GO:2001243-negative regulation of intrinsic apoptotic signaling pathway': [767.0,\n",
       "  577.0,\n",
       "  255.0,\n",
       "  602.0,\n",
       "  1030.0,\n",
       "  786.0,\n",
       "  996.0,\n",
       "  405.0,\n",
       "  672.0,\n",
       "  881.0,\n",
       "  1300.0,\n",
       "  388.0,\n",
       "  559.0,\n",
       "  578.0,\n",
       "  909.0,\n",
       "  614.0,\n",
       "  1083.0,\n",
       "  535.0,\n",
       "  199.0,\n",
       "  1169.0,\n",
       "  587.0,\n",
       "  602.0],\n",
       " 'GO:0032182-ubiquitin-like protein binding': [768.0,\n",
       "  972.0,\n",
       "  240.0,\n",
       "  834.0,\n",
       "  951.0,\n",
       "  1141.0,\n",
       "  844.0,\n",
       "  572.0,\n",
       "  1743.0,\n",
       "  559.0,\n",
       "  851.0,\n",
       "  877.0,\n",
       "  1072.0,\n",
       "  467.0,\n",
       "  148.0,\n",
       "  1011.0,\n",
       "  786.0],\n",
       " 'GO:0045333-cellular respiration': [769.0,\n",
       "  432.0,\n",
       "  524.0,\n",
       "  477.0,\n",
       "  636.0,\n",
       "  1687.0,\n",
       "  543.0,\n",
       "  634.0,\n",
       "  254.0,\n",
       "  601.0,\n",
       "  569.0,\n",
       "  361.0,\n",
       "  700.0,\n",
       "  594.0,\n",
       "  176.0,\n",
       "  303.0,\n",
       "  323.0,\n",
       "  955.0,\n",
       "  417.0,\n",
       "  803.0,\n",
       "  405.0,\n",
       "  104.0,\n",
       "  1341.0,\n",
       "  520.0,\n",
       "  484.0],\n",
       " 'GO:0045727-positive regulation of translation': [770.0,\n",
       "  720.0,\n",
       "  735.0,\n",
       "  852.0,\n",
       "  1131.0,\n",
       "  1383.0,\n",
       "  900.0,\n",
       "  1094.0,\n",
       "  1676.0,\n",
       "  458.0,\n",
       "  823.0,\n",
       "  507.0,\n",
       "  772.0,\n",
       "  796.0,\n",
       "  1576.0,\n",
       "  403.0,\n",
       "  233.0,\n",
       "  1912.0,\n",
       "  663.0,\n",
       "  730.0],\n",
       " 'GO:0090316-positive regulation of intracellular protein transport': [771.0,\n",
       "  824.0,\n",
       "  379.0,\n",
       "  822.0,\n",
       "  1125.0,\n",
       "  1103.0,\n",
       "  1023.0,\n",
       "  709.0,\n",
       "  453.0,\n",
       "  853.0,\n",
       "  878.0,\n",
       "  892.0,\n",
       "  1393.0,\n",
       "  452.0,\n",
       "  785.0,\n",
       "  552.0,\n",
       "  892.0,\n",
       "  665.0,\n",
       "  1204.0,\n",
       "  687.0,\n",
       "  240.0,\n",
       "  1218.0,\n",
       "  1080.0,\n",
       "  715.0],\n",
       " 'GO:0050706-regulation of interleukin-1 beta secretion': [772.0,\n",
       "  255.0,\n",
       "  66.0,\n",
       "  429.0,\n",
       "  402.0,\n",
       "  507.0,\n",
       "  171.0,\n",
       "  270.0,\n",
       "  984.0,\n",
       "  314.0,\n",
       "  461.0,\n",
       "  331.0,\n",
       "  379.0,\n",
       "  719.0],\n",
       " 'GO:0043029-T cell homeostasis': [773.0, 189.0, 207.0, 374.0, 862.0],\n",
       " 'GO:0043966-histone H3 acetylation': [774.0,\n",
       "  621.0,\n",
       "  142.0,\n",
       "  429.0,\n",
       "  493.0,\n",
       "  1054.0,\n",
       "  625.0,\n",
       "  832.0,\n",
       "  1250.0,\n",
       "  349.0,\n",
       "  487.0,\n",
       "  991.0,\n",
       "  372.0,\n",
       "  817.0],\n",
       " 'GO:0070888-E-box binding': [775.0, 593.0, 683.0],\n",
       " 'GO:0044409-entry into host': [777.5,\n",
       "  474.5,\n",
       "  194.5,\n",
       "  524.5,\n",
       "  789.5,\n",
       "  649.5,\n",
       "  560.5,\n",
       "  416.5,\n",
       "  364.5,\n",
       "  1210.5,\n",
       "  710.5,\n",
       "  410.5,\n",
       "  516.5,\n",
       "  950.5,\n",
       "  314.5,\n",
       "  726.5,\n",
       "  723.5,\n",
       "  620.5,\n",
       "  619.5,\n",
       "  825.5,\n",
       "  659.5,\n",
       "  712.5,\n",
       "  740.5,\n",
       "  581.5],\n",
       " 'GO:0030260-entry into host cell': [777.5,\n",
       "  474.5,\n",
       "  194.5,\n",
       "  524.5,\n",
       "  789.5,\n",
       "  649.5,\n",
       "  560.5,\n",
       "  416.5,\n",
       "  364.5,\n",
       "  1210.5,\n",
       "  710.5,\n",
       "  410.5,\n",
       "  516.5,\n",
       "  950.5,\n",
       "  314.5,\n",
       "  726.5,\n",
       "  723.5,\n",
       "  620.5,\n",
       "  619.5,\n",
       "  825.5,\n",
       "  659.5,\n",
       "  712.5,\n",
       "  740.5,\n",
       "  581.5],\n",
       " 'GO:0051806-entry into cell of other organism involved in symbiotic interaction': [777.5,\n",
       "  474.5,\n",
       "  194.5,\n",
       "  524.5,\n",
       "  789.5,\n",
       "  649.5,\n",
       "  560.5,\n",
       "  416.5,\n",
       "  364.5,\n",
       "  1210.5,\n",
       "  710.5,\n",
       "  410.5,\n",
       "  516.5,\n",
       "  950.5,\n",
       "  314.5,\n",
       "  726.5,\n",
       "  723.5,\n",
       "  620.5,\n",
       "  619.5,\n",
       "  825.5,\n",
       "  659.5,\n",
       "  712.5,\n",
       "  740.5,\n",
       "  581.5],\n",
       " 'GO:0051828-entry into other organism involved in symbiotic interaction': [777.5,\n",
       "  474.5,\n",
       "  194.5,\n",
       "  524.5,\n",
       "  789.5,\n",
       "  649.5,\n",
       "  560.5,\n",
       "  416.5,\n",
       "  364.5,\n",
       "  1210.5,\n",
       "  710.5,\n",
       "  410.5,\n",
       "  516.5,\n",
       "  950.5,\n",
       "  314.5,\n",
       "  726.5,\n",
       "  723.5,\n",
       "  620.5,\n",
       "  619.5,\n",
       "  825.5,\n",
       "  659.5,\n",
       "  712.5,\n",
       "  740.5,\n",
       "  581.5],\n",
       " 'GO:1902373-negative regulation of mRNA catabolic process': [780.0,\n",
       "  421.0,\n",
       "  758.0,\n",
       "  371.0],\n",
       " 'GO:1900026-positive regulation of substrate adhesion-dependent cell spreading': [781.0,\n",
       "  344.0,\n",
       "  891.0,\n",
       "  252.0,\n",
       "  341.0],\n",
       " 'GO:0032369-negative regulation of lipid transport': [782.0,\n",
       "  286.0,\n",
       "  61.0,\n",
       "  293.0,\n",
       "  363.0,\n",
       "  295.0,\n",
       "  115.0,\n",
       "  239.0,\n",
       "  208.0,\n",
       "  45.0,\n",
       "  166.0,\n",
       "  295.0,\n",
       "  270.0,\n",
       "  156.0,\n",
       "  573.0,\n",
       "  327.0,\n",
       "  389.0,\n",
       "  164.0,\n",
       "  116.0],\n",
       " 'GO:0045023-G0 to G1 transition': [783.0, 518.0, 745.0, 1525.0, 842.0, 288.0],\n",
       " 'GO:0032784-regulation of DNA-templated transcription, elongation': [784.0,\n",
       "  504.0,\n",
       "  268.0,\n",
       "  1169.0,\n",
       "  443.0,\n",
       "  256.0,\n",
       "  890.0],\n",
       " 'GO:0090207-regulation of triglyceride metabolic process': [785.0,\n",
       "  179.0,\n",
       "  550.0,\n",
       "  534.0,\n",
       "  607.0,\n",
       "  524.0,\n",
       "  806.0,\n",
       "  213.0],\n",
       " 'GO:1905268-negative regulation of chromatin organization': [786.0,\n",
       "  646.0,\n",
       "  1118.0,\n",
       "  901.0,\n",
       "  787.0,\n",
       "  988.0],\n",
       " 'GO:0043267-negative regulation of potassium ion transport': [787.0,\n",
       "  460.0,\n",
       "  921.0,\n",
       "  469.0,\n",
       "  281.0],\n",
       " 'GO:0050856-regulation of T cell receptor signaling pathway': [788.0,\n",
       "  712.0,\n",
       "  482.0,\n",
       "  395.0,\n",
       "  766.0,\n",
       "  219.0,\n",
       "  765.0,\n",
       "  552.0],\n",
       " 'GO:0051341-regulation of oxidoreductase activity': [789.0,\n",
       "  808.0,\n",
       "  259.0,\n",
       "  596.0,\n",
       "  1355.0,\n",
       "  925.0,\n",
       "  946.0,\n",
       "  1553.0,\n",
       "  911.0,\n",
       "  1724.0,\n",
       "  522.0,\n",
       "  441.0,\n",
       "  993.0,\n",
       "  1017.0,\n",
       "  1061.0,\n",
       "  1418.0,\n",
       "  646.0,\n",
       "  889.0,\n",
       "  1205.0,\n",
       "  932.0],\n",
       " 'GO:0032506-cytokinetic process': [790.0, 535.0, 450.0, 996.0, 392.0],\n",
       " 'GO:0045814-negative regulation of gene expression, epigenetic': [791.0,\n",
       "  1191.0,\n",
       "  517.0,\n",
       "  862.0,\n",
       "  945.0,\n",
       "  2014.0,\n",
       "  1567.0,\n",
       "  891.0,\n",
       "  2035.0,\n",
       "  600.0,\n",
       "  1314.0,\n",
       "  696.0,\n",
       "  1131.0,\n",
       "  1483.0,\n",
       "  686.0,\n",
       "  1487.0],\n",
       " 'GO:0006641-triglyceride metabolic process': [792.0,\n",
       "  744.0,\n",
       "  250.0,\n",
       "  991.0,\n",
       "  715.0,\n",
       "  1308.0,\n",
       "  864.0,\n",
       "  1753.0,\n",
       "  335.0,\n",
       "  1539.0,\n",
       "  583.0,\n",
       "  279.0,\n",
       "  698.0,\n",
       "  1418.0,\n",
       "  830.0,\n",
       "  798.0,\n",
       "  925.0,\n",
       "  1375.0,\n",
       "  860.0,\n",
       "  1346.0,\n",
       "  420.0],\n",
       " 'GO:0031983-vesicle lumen': [793.0,\n",
       "  438.0,\n",
       "  144.0,\n",
       "  471.0,\n",
       "  545.0,\n",
       "  734.0,\n",
       "  498.0,\n",
       "  644.0,\n",
       "  256.0,\n",
       "  1152.0,\n",
       "  401.0,\n",
       "  194.0,\n",
       "  336.0,\n",
       "  488.0,\n",
       "  210.0,\n",
       "  497.0,\n",
       "  1000.0,\n",
       "  468.0,\n",
       "  315.0,\n",
       "  420.0,\n",
       "  766.0,\n",
       "  339.0,\n",
       "  687.0,\n",
       "  1093.0,\n",
       "  526.0],\n",
       " 'GO:0101002-ficolin-1-rich granule': [794.0,\n",
       "  434.0,\n",
       "  211.0,\n",
       "  333.0,\n",
       "  672.0,\n",
       "  921.0,\n",
       "  606.0,\n",
       "  520.0,\n",
       "  349.0,\n",
       "  1410.0,\n",
       "  577.0,\n",
       "  219.0,\n",
       "  347.0,\n",
       "  699.0,\n",
       "  201.0,\n",
       "  656.0,\n",
       "  943.0,\n",
       "  652.0,\n",
       "  336.0,\n",
       "  565.0,\n",
       "  669.0,\n",
       "  281.0,\n",
       "  867.0,\n",
       "  1404.0,\n",
       "  498.0],\n",
       " 'GO:0060205-cytoplasmic vesicle lumen': [795.0,\n",
       "  439.0,\n",
       "  145.0,\n",
       "  476.0,\n",
       "  532.0,\n",
       "  719.0,\n",
       "  504.0,\n",
       "  648.0,\n",
       "  258.0,\n",
       "  1154.0,\n",
       "  402.0,\n",
       "  196.0,\n",
       "  337.0,\n",
       "  475.0,\n",
       "  207.0,\n",
       "  499.0,\n",
       "  991.0,\n",
       "  463.0,\n",
       "  302.0,\n",
       "  424.0,\n",
       "  759.0,\n",
       "  330.0,\n",
       "  691.0,\n",
       "  1057.0,\n",
       "  509.0],\n",
       " 'GO:0006405-RNA export from nucleus': [796.0,\n",
       "  951.0,\n",
       "  684.0,\n",
       "  1183.0,\n",
       "  1667.0,\n",
       "  935.0,\n",
       "  362.0,\n",
       "  1132.0,\n",
       "  439.0,\n",
       "  341.0,\n",
       "  615.0,\n",
       "  984.0,\n",
       "  1977.0,\n",
       "  333.0,\n",
       "  674.0,\n",
       "  913.0],\n",
       " 'GO:0044088-regulation of vacuole organization': [797.0,\n",
       "  578.0,\n",
       "  1253.0,\n",
       "  348.0,\n",
       "  534.0],\n",
       " 'GO:0031091-platelet alpha granule': [798.0,\n",
       "  329.0,\n",
       "  89.0,\n",
       "  335.0,\n",
       "  296.0,\n",
       "  728.0,\n",
       "  550.0,\n",
       "  1257.0,\n",
       "  176.0,\n",
       "  641.0,\n",
       "  201.0,\n",
       "  87.0,\n",
       "  290.0,\n",
       "  687.0,\n",
       "  402.0,\n",
       "  425.0,\n",
       "  378.0,\n",
       "  410.0,\n",
       "  361.0,\n",
       "  683.0,\n",
       "  1047.0,\n",
       "  129.0],\n",
       " 'GO:0006693-prostaglandin metabolic process': [799.5,\n",
       "  370.5,\n",
       "  479.5,\n",
       "  485.5,\n",
       "  366.5,\n",
       "  597.5,\n",
       "  289.5,\n",
       "  613.5,\n",
       "  795.5,\n",
       "  623.5,\n",
       "  576.5],\n",
       " 'GO:0006692-prostanoid metabolic process': [799.5,\n",
       "  370.5,\n",
       "  479.5,\n",
       "  485.5,\n",
       "  366.5,\n",
       "  597.5,\n",
       "  289.5,\n",
       "  613.5,\n",
       "  795.5,\n",
       "  623.5,\n",
       "  576.5],\n",
       " 'GO:0050830-defense response to Gram-positive bacterium': [801.0,\n",
       "  418.0,\n",
       "  264.0,\n",
       "  977.0,\n",
       "  900.0,\n",
       "  643.0,\n",
       "  1220.0,\n",
       "  798.0,\n",
       "  311.0,\n",
       "  377.0,\n",
       "  169.0,\n",
       "  529.0,\n",
       "  1162.0,\n",
       "  619.0,\n",
       "  176.0,\n",
       "  377.0,\n",
       "  829.0,\n",
       "  1267.0,\n",
       "  1424.0,\n",
       "  790.0],\n",
       " 'GO:0002576-platelet degranulation': [802.0,\n",
       "  301.0,\n",
       "  99.0,\n",
       "  319.0,\n",
       "  256.0,\n",
       "  529.0,\n",
       "  525.0,\n",
       "  753.0,\n",
       "  174.0,\n",
       "  665.0,\n",
       "  190.0,\n",
       "  107.0,\n",
       "  227.0,\n",
       "  658.0,\n",
       "  236.0,\n",
       "  332.0,\n",
       "  901.0,\n",
       "  323.0,\n",
       "  320.0,\n",
       "  323.0,\n",
       "  699.0,\n",
       "  570.0,\n",
       "  684.0,\n",
       "  159.0],\n",
       " 'GO:0030224-monocyte differentiation': [803.5,\n",
       "  310.5,\n",
       "  307.5,\n",
       "  558.5,\n",
       "  403.5,\n",
       "  139.5],\n",
       " 'GO:1903131-mononuclear cell differentiation': [803.5,\n",
       "  310.5,\n",
       "  307.5,\n",
       "  558.5,\n",
       "  403.5,\n",
       "  139.5],\n",
       " 'GO:0030947-regulation of vascular endothelial growth factor receptor signaling pathway': [805.0,\n",
       "  274.0,\n",
       "  352.0],\n",
       " 'GO:1903146-regulation of autophagy of mitochondrion': [806.0,\n",
       "  672.0,\n",
       "  344.0,\n",
       "  353.0],\n",
       " 'GO:1902100-negative regulation of metaphase/anaphase transition of cell cycle': [807.0,\n",
       "  558.0,\n",
       "  260.0,\n",
       "  277.0],\n",
       " 'GO:0008206-bile acid metabolic process': [808.0, 549.0],\n",
       " 'GO:0071174-mitotic spindle checkpoint': [810.5, 314.5],\n",
       " 'GO:0071173-spindle assembly checkpoint': [810.5, 314.5],\n",
       " 'GO:0031577-spindle checkpoint': [810.5, 314.5],\n",
       " 'GO:0007094-mitotic spindle assembly checkpoint': [810.5, 314.5],\n",
       " 'GO:0006298-mismatch repair': [813.0, 335.0, 251.0, 338.0, 191.0, 212.0],\n",
       " 'GO:0033046-negative regulation of sister chromatid segregation': [814.0,\n",
       "  322.0,\n",
       "  546.0,\n",
       "  1244.0,\n",
       "  417.0,\n",
       "  633.0,\n",
       "  464.0,\n",
       "  881.0,\n",
       "  231.0,\n",
       "  432.0],\n",
       " 'GO:0042743-hydrogen peroxide metabolic process': [815.0,\n",
       "  367.0,\n",
       "  171.0,\n",
       "  987.0,\n",
       "  571.0,\n",
       "  797.0,\n",
       "  131.0,\n",
       "  450.0,\n",
       "  557.0,\n",
       "  229.0,\n",
       "  674.0,\n",
       "  393.0,\n",
       "  340.0,\n",
       "  584.0,\n",
       "  786.0],\n",
       " 'GO:0016591-RNA polymerase II, holoenzyme': [816.0,\n",
       "  873.0,\n",
       "  507.0,\n",
       "  1530.0,\n",
       "  690.0,\n",
       "  479.0,\n",
       "  1435.0,\n",
       "  367.0,\n",
       "  264.0,\n",
       "  856.0,\n",
       "  1315.0,\n",
       "  379.0,\n",
       "  966.0,\n",
       "  663.0],\n",
       " 'GO:0002820-negative regulation of adaptive immune response': [817.0,\n",
       "  227.0,\n",
       "  610.0,\n",
       "  282.0,\n",
       "  632.0,\n",
       "  860.0],\n",
       " 'GO:0000289-nuclear-transcribed mRNA poly(A) tail shortening': 818.0,\n",
       " 'GO:0030134-COPII-coated ER to Golgi transport vesicle': [819.0,\n",
       "  628.0,\n",
       "  865.0,\n",
       "  952.0,\n",
       "  710.0,\n",
       "  1004.0,\n",
       "  1386.0,\n",
       "  460.0,\n",
       "  878.0,\n",
       "  354.0,\n",
       "  898.0,\n",
       "  1109.0,\n",
       "  1100.0,\n",
       "  1256.0,\n",
       "  1092.0,\n",
       "  646.0,\n",
       "  683.0],\n",
       " 'GO:0008009-chemokine activity': [820.0,\n",
       "  122.0,\n",
       "  59.0,\n",
       "  130.0,\n",
       "  540.0,\n",
       "  187.0,\n",
       "  93.0,\n",
       "  170.0,\n",
       "  23.0,\n",
       "  172.0,\n",
       "  244.0,\n",
       "  180.0,\n",
       "  343.0,\n",
       "  427.0,\n",
       "  299.0,\n",
       "  353.0,\n",
       "  699.0,\n",
       "  360.0],\n",
       " 'GO:0060251-regulation of glial cell proliferation': [821.0,\n",
       "  166.0,\n",
       "  367.0,\n",
       "  200.0,\n",
       "  795.0,\n",
       "  526.0],\n",
       " 'GO:0005048-signal sequence binding': 822.0,\n",
       " 'GO:0000819-sister chromatid segregation': [823.0,\n",
       "  897.0,\n",
       "  531.0,\n",
       "  467.0,\n",
       "  1085.0,\n",
       "  1368.0,\n",
       "  1477.0,\n",
       "  734.0,\n",
       "  562.0,\n",
       "  431.0,\n",
       "  935.0,\n",
       "  1391.0,\n",
       "  1720.0,\n",
       "  337.0,\n",
       "  1347.0,\n",
       "  451.0,\n",
       "  908.0,\n",
       "  705.0,\n",
       "  1749.0,\n",
       "  264.0,\n",
       "  191.0,\n",
       "  1760.0,\n",
       "  1193.0,\n",
       "  969.0],\n",
       " 'GO:0044445-cytosolic part': [824.0,\n",
       "  490.0,\n",
       "  553.0,\n",
       "  964.0,\n",
       "  585.0,\n",
       "  1299.0,\n",
       "  511.0,\n",
       "  764.0,\n",
       "  337.0,\n",
       "  1373.0,\n",
       "  869.0,\n",
       "  413.0,\n",
       "  754.0,\n",
       "  830.0,\n",
       "  482.0,\n",
       "  504.0,\n",
       "  703.0,\n",
       "  1057.0,\n",
       "  760.0,\n",
       "  987.0,\n",
       "  791.0,\n",
       "  203.0,\n",
       "  1045.0,\n",
       "  1197.0,\n",
       "  571.0],\n",
       " 'GO:1904356-regulation of telomere maintenance via telomere lengthening': [825.0,\n",
       "  2136.0,\n",
       "  384.0],\n",
       " 'GO:1901532-regulation of hematopoietic progenitor cell differentiation': [826.0,\n",
       "  780.0,\n",
       "  336.0,\n",
       "  941.0,\n",
       "  427.0,\n",
       "  338.0,\n",
       "  386.0,\n",
       "  639.0,\n",
       "  921.0,\n",
       "  718.0,\n",
       "  354.0,\n",
       "  992.0,\n",
       "  549.0,\n",
       "  822.0,\n",
       "  739.0,\n",
       "  835.0,\n",
       "  373.0,\n",
       "  1373.0,\n",
       "  558.0,\n",
       "  411.0],\n",
       " 'GO:0097352-autophagosome maturation': [827.0, 390.0, 1341.0],\n",
       " 'GO:0061844-antimicrobial humoral immune response mediated by antimicrobial peptide': [828.0,\n",
       "  148.0,\n",
       "  104.0,\n",
       "  481.0,\n",
       "  305.0,\n",
       "  465.0,\n",
       "  407.0,\n",
       "  529.0,\n",
       "  121.0,\n",
       "  176.0,\n",
       "  55.0,\n",
       "  185.0,\n",
       "  357.0,\n",
       "  303.0,\n",
       "  302.0,\n",
       "  220.0,\n",
       "  396.0,\n",
       "  337.0,\n",
       "  540.0,\n",
       "  1244.0,\n",
       "  417.0],\n",
       " 'GO:0045296-cadherin binding': [829.0,\n",
       "  768.0,\n",
       "  522.0,\n",
       "  790.0,\n",
       "  837.0,\n",
       "  1450.0,\n",
       "  1134.0,\n",
       "  1031.0,\n",
       "  492.0,\n",
       "  606.0,\n",
       "  742.0,\n",
       "  1104.0,\n",
       "  1260.0,\n",
       "  465.0,\n",
       "  1040.0,\n",
       "  808.0,\n",
       "  1041.0,\n",
       "  859.0,\n",
       "  1561.0,\n",
       "  708.0,\n",
       "  323.0,\n",
       "  1280.0,\n",
       "  880.0,\n",
       "  725.0],\n",
       " 'GO:1903363-negative regulation of cellular protein catabolic process': [830.0,\n",
       "  927.0,\n",
       "  1125.0,\n",
       "  1515.0,\n",
       "  695.0,\n",
       "  954.0,\n",
       "  1877.0,\n",
       "  627.0,\n",
       "  1141.0,\n",
       "  956.0,\n",
       "  560.0,\n",
       "  618.0,\n",
       "  670.0],\n",
       " 'GO:0090313-regulation of protein targeting to membrane': [831.0,\n",
       "  540.0,\n",
       "  434.0,\n",
       "  365.0,\n",
       "  923.0],\n",
       " 'GO:0071166-ribonucleoprotein complex localization': [832.0,\n",
       "  870.0,\n",
       "  742.0,\n",
       "  1248.0,\n",
       "  1725.0,\n",
       "  759.0,\n",
       "  355.0,\n",
       "  1175.0,\n",
       "  406.0,\n",
       "  344.0,\n",
       "  598.0,\n",
       "  1034.0,\n",
       "  347.0,\n",
       "  681.0,\n",
       "  967.0],\n",
       " 'GO:0006997-nucleus organization': [833.0,\n",
       "  1274.0,\n",
       "  455.0,\n",
       "  775.0,\n",
       "  1714.0,\n",
       "  1687.0,\n",
       "  1145.0,\n",
       "  732.0,\n",
       "  1883.0,\n",
       "  583.0,\n",
       "  1313.0,\n",
       "  638.0,\n",
       "  1040.0,\n",
       "  1119.0,\n",
       "  1938.0,\n",
       "  576.0,\n",
       "  210.0,\n",
       "  1795.0,\n",
       "  1032.0,\n",
       "  981.0],\n",
       " \"GO:0003730-mRNA 3'-UTR binding\": [834.0,\n",
       "  967.0,\n",
       "  540.0,\n",
       "  1048.0,\n",
       "  1168.0,\n",
       "  1435.0,\n",
       "  1380.0,\n",
       "  977.0,\n",
       "  2174.0,\n",
       "  492.0,\n",
       "  1007.0,\n",
       "  581.0,\n",
       "  1511.0,\n",
       "  573.0],\n",
       " 'GO:1904724-tertiary granule lumen': [835.0,\n",
       "  582.0,\n",
       "  456.0,\n",
       "  424.0,\n",
       "  584.0,\n",
       "  352.0,\n",
       "  456.0,\n",
       "  185.0,\n",
       "  189.0,\n",
       "  293.0,\n",
       "  494.0,\n",
       "  675.0,\n",
       "  372.0,\n",
       "  378.0,\n",
       "  645.0,\n",
       "  427.0],\n",
       " 'GO:0006301-postreplication repair': [836.0,\n",
       "  395.0,\n",
       "  455.0,\n",
       "  632.0,\n",
       "  785.0,\n",
       "  406.0,\n",
       "  577.0,\n",
       "  721.0,\n",
       "  281.0,\n",
       "  465.0,\n",
       "  472.0,\n",
       "  676.0,\n",
       "  996.0,\n",
       "  495.0,\n",
       "  283.0],\n",
       " 'GO:0007004-telomere maintenance via telomerase': [837.0,\n",
       "  1108.0,\n",
       "  872.0,\n",
       "  886.0,\n",
       "  1414.0,\n",
       "  499.0,\n",
       "  875.0,\n",
       "  458.0,\n",
       "  853.0],\n",
       " 'GO:0002504-antigen processing and presentation of peptide or polysaccharide antigen via MHC class II': [838.0,\n",
       "  440.0,\n",
       "  260.0,\n",
       "  309.0,\n",
       "  781.0,\n",
       "  1052.0,\n",
       "  857.0,\n",
       "  409.0,\n",
       "  1012.0,\n",
       "  366.0,\n",
       "  738.0,\n",
       "  566.0,\n",
       "  200.0,\n",
       "  392.0,\n",
       "  940.0,\n",
       "  712.0,\n",
       "  462.0,\n",
       "  770.0,\n",
       "  713.0,\n",
       "  901.0,\n",
       "  447.0,\n",
       "  577.0],\n",
       " 'GO:0051701-interaction with host': [839.0,\n",
       "  700.0,\n",
       "  365.0,\n",
       "  643.0,\n",
       "  947.0,\n",
       "  1102.0,\n",
       "  851.0,\n",
       "  767.0,\n",
       "  456.0,\n",
       "  964.0,\n",
       "  866.0,\n",
       "  701.0,\n",
       "  1216.0,\n",
       "  520.0,\n",
       "  1047.0,\n",
       "  701.0,\n",
       "  735.0,\n",
       "  769.0,\n",
       "  1150.0,\n",
       "  685.0,\n",
       "  333.0,\n",
       "  948.0,\n",
       "  1065.0,\n",
       "  747.0],\n",
       " 'GO:0005766-primary lysosome': [840.5,\n",
       "  435.5,\n",
       "  182.5,\n",
       "  552.5,\n",
       "  764.5,\n",
       "  998.5,\n",
       "  575.5,\n",
       "  587.5,\n",
       "  300.5,\n",
       "  1123.5,\n",
       "  452.5,\n",
       "  254.5,\n",
       "  597.5,\n",
       "  739.5,\n",
       "  333.5,\n",
       "  707.5,\n",
       "  1062.5,\n",
       "  696.5,\n",
       "  414.5,\n",
       "  431.5,\n",
       "  649.5,\n",
       "  619.5,\n",
       "  1160.5,\n",
       "  806.5],\n",
       " 'GO:0042582-azurophil granule': [840.5,\n",
       "  435.5,\n",
       "  182.5,\n",
       "  552.5,\n",
       "  764.5,\n",
       "  998.5,\n",
       "  575.5,\n",
       "  587.5,\n",
       "  300.5,\n",
       "  1123.5,\n",
       "  452.5,\n",
       "  254.5,\n",
       "  597.5,\n",
       "  739.5,\n",
       "  333.5,\n",
       "  707.5,\n",
       "  1062.5,\n",
       "  696.5,\n",
       "  414.5,\n",
       "  431.5,\n",
       "  649.5,\n",
       "  619.5,\n",
       "  1160.5,\n",
       "  806.5],\n",
       " 'GO:0009295-nucleoid': [842.5,\n",
       "  488.5,\n",
       "  459.5,\n",
       "  240.5,\n",
       "  521.5,\n",
       "  136.5,\n",
       "  341.5,\n",
       "  511.5,\n",
       "  126.5,\n",
       "  425.5],\n",
       " 'GO:0042645-mitochondrial nucleoid': [842.5,\n",
       "  488.5,\n",
       "  459.5,\n",
       "  240.5,\n",
       "  521.5,\n",
       "  136.5,\n",
       "  341.5,\n",
       "  511.5,\n",
       "  126.5,\n",
       "  425.5],\n",
       " 'GO:0034620-cellular response to unfolded protein': [844.0,\n",
       "  1267.0,\n",
       "  590.0,\n",
       "  796.0,\n",
       "  1289.0,\n",
       "  1124.0,\n",
       "  1120.0,\n",
       "  673.0,\n",
       "  807.0,\n",
       "  1145.0,\n",
       "  1528.0,\n",
       "  557.0,\n",
       "  870.0,\n",
       "  526.0,\n",
       "  1347.0,\n",
       "  841.0,\n",
       "  1523.0,\n",
       "  554.0,\n",
       "  147.0,\n",
       "  1345.0,\n",
       "  531.0,\n",
       "  512.0],\n",
       " 'GO:0071426-ribonucleoprotein complex export from nucleus': [845.0,\n",
       "  889.0,\n",
       "  753.0,\n",
       "  1266.0,\n",
       "  1738.0,\n",
       "  769.0,\n",
       "  360.0,\n",
       "  1200.0,\n",
       "  412.0,\n",
       "  349.0,\n",
       "  612.0,\n",
       "  1041.0,\n",
       "  354.0,\n",
       "  698.0,\n",
       "  973.0],\n",
       " 'GO:0006283-transcription-coupled nucleotide-excision repair': [846.0,\n",
       "  650.0,\n",
       "  594.0,\n",
       "  655.0,\n",
       "  987.0,\n",
       "  818.0,\n",
       "  488.0,\n",
       "  508.0,\n",
       "  1003.0,\n",
       "  318.0,\n",
       "  664.0,\n",
       "  343.0,\n",
       "  445.0,\n",
       "  695.0,\n",
       "  893.0,\n",
       "  370.0,\n",
       "  420.0,\n",
       "  507.0],\n",
       " 'GO:1905953-negative regulation of lipid localization': [847.0,\n",
       "  403.0,\n",
       "  202.0,\n",
       "  360.0,\n",
       "  473.0,\n",
       "  513.0,\n",
       "  179.0,\n",
       "  649.0,\n",
       "  355.0,\n",
       "  132.0,\n",
       "  316.0,\n",
       "  765.0,\n",
       "  437.0,\n",
       "  472.0,\n",
       "  702.0,\n",
       "  540.0,\n",
       "  497.0,\n",
       "  438.0,\n",
       "  882.0,\n",
       "  243.0],\n",
       " 'GO:0008631-intrinsic apoptotic signaling pathway in response to oxidative stress': [848.0,\n",
       "  492.0,\n",
       "  439.0,\n",
       "  559.0],\n",
       " 'GO:0043487-regulation of RNA stability': [849.0,\n",
       "  692.0,\n",
       "  623.0,\n",
       "  448.0,\n",
       "  987.0,\n",
       "  1356.0,\n",
       "  1342.0,\n",
       "  879.0,\n",
       "  538.0,\n",
       "  534.0,\n",
       "  1039.0,\n",
       "  1287.0,\n",
       "  1143.0,\n",
       "  421.0,\n",
       "  1350.0,\n",
       "  616.0,\n",
       "  729.0,\n",
       "  773.0,\n",
       "  1187.0,\n",
       "  559.0,\n",
       "  341.0,\n",
       "  1770.0,\n",
       "  654.0,\n",
       "  762.0],\n",
       " 'GO:0022613-ribonucleoprotein complex biogenesis': [850.0,\n",
       "  815.0,\n",
       "  738.0,\n",
       "  874.0,\n",
       "  1194.0,\n",
       "  1571.0,\n",
       "  1298.0,\n",
       "  992.0,\n",
       "  589.0,\n",
       "  919.0,\n",
       "  1260.0,\n",
       "  1088.0,\n",
       "  419.0,\n",
       "  1488.0,\n",
       "  471.0,\n",
       "  871.0,\n",
       "  811.0,\n",
       "  1515.0,\n",
       "  466.0,\n",
       "  398.0,\n",
       "  2121.0,\n",
       "  852.0,\n",
       "  1091.0],\n",
       " 'GO:1903747-regulation of establishment of protein localization to mitochondrion': [851.0,\n",
       "  501.0,\n",
       "  607.0,\n",
       "  738.0,\n",
       "  1002.0,\n",
       "  1024.0,\n",
       "  686.0,\n",
       "  678.0,\n",
       "  848.0,\n",
       "  1172.0,\n",
       "  368.0,\n",
       "  656.0,\n",
       "  574.0,\n",
       "  543.0,\n",
       "  869.0,\n",
       "  639.0,\n",
       "  1447.0,\n",
       "  1056.0],\n",
       " 'GO:0005080-protein kinase C binding': [852.0,\n",
       "  623.0,\n",
       "  577.0,\n",
       "  884.0,\n",
       "  341.0,\n",
       "  585.0,\n",
       "  1397.0,\n",
       "  872.0],\n",
       " 'GO:0006986-response to unfolded protein': [853.0,\n",
       "  1032.0,\n",
       "  624.0,\n",
       "  608.0,\n",
       "  846.0,\n",
       "  1237.0,\n",
       "  1149.0,\n",
       "  1134.0,\n",
       "  442.0,\n",
       "  667.0,\n",
       "  840.0,\n",
       "  1115.0,\n",
       "  1336.0,\n",
       "  552.0,\n",
       "  866.0,\n",
       "  508.0,\n",
       "  1225.0,\n",
       "  715.0,\n",
       "  1575.0,\n",
       "  575.0,\n",
       "  152.0,\n",
       "  1324.0,\n",
       "  507.0,\n",
       "  613.0],\n",
       " 'GO:1902807-negative regulation of cell cycle G1/S phase transition': [854.0,\n",
       "  829.0,\n",
       "  366.0,\n",
       "  385.0,\n",
       "  761.0,\n",
       "  620.0,\n",
       "  729.0,\n",
       "  923.0,\n",
       "  385.0,\n",
       "  1062.0,\n",
       "  638.0,\n",
       "  915.0,\n",
       "  1517.0,\n",
       "  649.0,\n",
       "  659.0,\n",
       "  592.0,\n",
       "  1024.0,\n",
       "  776.0,\n",
       "  1041.0,\n",
       "  803.0,\n",
       "  185.0,\n",
       "  1405.0,\n",
       "  735.0,\n",
       "  681.0],\n",
       " 'GO:0016925-protein sumoylation': [855.0,\n",
       "  760.0,\n",
       "  633.0,\n",
       "  1411.0,\n",
       "  1159.0,\n",
       "  399.0,\n",
       "  1270.0,\n",
       "  348.0,\n",
       "  442.0,\n",
       "  648.0,\n",
       "  481.0,\n",
       "  506.0,\n",
       "  587.0],\n",
       " 'GO:0002495-antigen processing and presentation of peptide antigen via MHC class II': [856.0,\n",
       "  444.0,\n",
       "  272.0,\n",
       "  321.0,\n",
       "  792.0,\n",
       "  1104.0,\n",
       "  889.0,\n",
       "  411.0,\n",
       "  1032.0,\n",
       "  378.0,\n",
       "  760.0,\n",
       "  585.0,\n",
       "  196.0,\n",
       "  403.0,\n",
       "  900.0,\n",
       "  719.0,\n",
       "  469.0,\n",
       "  784.0,\n",
       "  720.0,\n",
       "  906.0,\n",
       "  454.0,\n",
       "  584.0],\n",
       " 'GO:0006611-protein export from nucleus': [857.0,\n",
       "  1056.0,\n",
       "  762.0,\n",
       "  1391.0,\n",
       "  1362.0,\n",
       "  1717.0,\n",
       "  908.0,\n",
       "  556.0,\n",
       "  1652.0,\n",
       "  481.0,\n",
       "  496.0,\n",
       "  650.0,\n",
       "  1106.0,\n",
       "  1982.0,\n",
       "  447.0,\n",
       "  2027.0,\n",
       "  940.0,\n",
       "  936.0],\n",
       " 'GO:0050853-B cell receptor signaling pathway': [858.0,\n",
       "  38.0,\n",
       "  306.0,\n",
       "  414.0,\n",
       "  140.0,\n",
       "  380.0,\n",
       "  489.0,\n",
       "  327.0,\n",
       "  440.0,\n",
       "  487.0,\n",
       "  165.0,\n",
       "  383.0,\n",
       "  333.0,\n",
       "  347.0,\n",
       "  522.0,\n",
       "  817.0,\n",
       "  733.0,\n",
       "  471.0],\n",
       " 'GO:0035794-positive regulation of mitochondrial membrane permeability': [859.0,\n",
       "  551.0,\n",
       "  1085.0,\n",
       "  704.0,\n",
       "  903.0,\n",
       "  1353.0,\n",
       "  550.0,\n",
       "  695.0,\n",
       "  1320.0,\n",
       "  1066.0],\n",
       " 'GO:0042542-response to hydrogen peroxide': [860.0,\n",
       "  851.0,\n",
       "  457.0,\n",
       "  671.0,\n",
       "  596.0,\n",
       "  1290.0,\n",
       "  848.0,\n",
       "  1022.0,\n",
       "  444.0,\n",
       "  1045.0,\n",
       "  929.0,\n",
       "  550.0,\n",
       "  1211.0,\n",
       "  648.0,\n",
       "  705.0,\n",
       "  1023.0,\n",
       "  1326.0,\n",
       "  1113.0,\n",
       "  900.0,\n",
       "  855.0,\n",
       "  897.0,\n",
       "  922.0,\n",
       "  638.0],\n",
       " 'GO:0006638-neutral lipid metabolic process': [861.0,\n",
       "  850.0,\n",
       "  380.0,\n",
       "  1130.0,\n",
       "  823.0,\n",
       "  1265.0,\n",
       "  911.0,\n",
       "  1927.0,\n",
       "  424.0,\n",
       "  780.0,\n",
       "  363.0,\n",
       "  924.0,\n",
       "  1628.0,\n",
       "  1042.0,\n",
       "  844.0,\n",
       "  948.0,\n",
       "  1589.0,\n",
       "  982.0,\n",
       "  1671.0,\n",
       "  595.0],\n",
       " 'GO:0097028-dendritic cell differentiation': [862.0,\n",
       "  385.0,\n",
       "  41.0,\n",
       "  508.0,\n",
       "  177.0,\n",
       "  342.0,\n",
       "  372.0,\n",
       "  297.0,\n",
       "  940.0,\n",
       "  237.0,\n",
       "  630.0,\n",
       "  446.0,\n",
       "  353.0,\n",
       "  732.0,\n",
       "  468.0],\n",
       " 'GO:0022900-electron transport chain': [863.0,\n",
       "  288.0,\n",
       "  387.0,\n",
       "  452.0,\n",
       "  442.0,\n",
       "  1036.0,\n",
       "  500.0,\n",
       "  326.0,\n",
       "  187.0,\n",
       "  573.0,\n",
       "  313.0,\n",
       "  195.0,\n",
       "  428.0,\n",
       "  404.0,\n",
       "  175.0,\n",
       "  258.0,\n",
       "  407.0,\n",
       "  655.0,\n",
       "  332.0,\n",
       "  535.0,\n",
       "  433.0,\n",
       "  78.0,\n",
       "  902.0,\n",
       "  446.0,\n",
       "  365.0],\n",
       " 'GO:0052126-movement in host environment': [866.0, 409.0],\n",
       " 'GO:0052192-movement in environment of other organism involved in symbiotic interaction': [866.0,\n",
       "  409.0],\n",
       " 'GO:0019076-viral release from host cell': [866.0, 409.0],\n",
       " 'GO:0035890-exit from host': [866.0, 409.0],\n",
       " 'GO:0035891-exit from host cell': [866.0, 409.0],\n",
       " 'GO:0098761-cellular response to interleukin-7': [869.5,\n",
       "  571.5,\n",
       "  329.5,\n",
       "  703.5,\n",
       "  303.5,\n",
       "  159.5],\n",
       " 'GO:0098760-response to interleukin-7': [869.5,\n",
       "  571.5,\n",
       "  329.5,\n",
       "  703.5,\n",
       "  303.5,\n",
       "  159.5],\n",
       " 'GO:0035196-production of miRNAs involved in gene silencing by miRNA': [871.0,\n",
       "  623.0,\n",
       "  508.0,\n",
       "  227.0],\n",
       " 'GO:2000378-negative regulation of reactive oxygen species metabolic process': [872.0,\n",
       "  681.0,\n",
       "  124.0,\n",
       "  569.0,\n",
       "  948.0,\n",
       "  857.0,\n",
       "  645.0,\n",
       "  1125.0,\n",
       "  603.0,\n",
       "  1111.0,\n",
       "  718.0,\n",
       "  1023.0,\n",
       "  667.0,\n",
       "  1055.0,\n",
       "  461.0],\n",
       " 'GO:0019843-rRNA binding': [873.0,\n",
       "  583.0,\n",
       "  580.0,\n",
       "  422.0,\n",
       "  594.0,\n",
       "  853.0,\n",
       "  496.0,\n",
       "  473.0,\n",
       "  616.0,\n",
       "  802.0,\n",
       "  492.0,\n",
       "  1229.0,\n",
       "  448.0],\n",
       " 'GO:0055092-sterol homeostasis': [874.0,\n",
       "  437.0,\n",
       "  90.0,\n",
       "  512.0,\n",
       "  343.0,\n",
       "  673.0,\n",
       "  600.0,\n",
       "  1409.0,\n",
       "  264.0,\n",
       "  888.0,\n",
       "  374.0,\n",
       "  159.0,\n",
       "  660.0,\n",
       "  1447.0,\n",
       "  306.0,\n",
       "  388.0,\n",
       "  474.0,\n",
       "  330.0,\n",
       "  589.0,\n",
       "  1257.0,\n",
       "  363.0,\n",
       "  117.0,\n",
       "  753.0,\n",
       "  781.0,\n",
       "  286.0],\n",
       " 'GO:0042254-ribosome biogenesis': [875.0,\n",
       "  721.0,\n",
       "  845.0,\n",
       "  1108.0,\n",
       "  1190.0,\n",
       "  1956.0,\n",
       "  1256.0,\n",
       "  1050.0,\n",
       "  644.0,\n",
       "  803.0,\n",
       "  935.0,\n",
       "  471.0,\n",
       "  1479.0,\n",
       "  467.0,\n",
       "  1019.0,\n",
       "  941.0,\n",
       "  1552.0,\n",
       "  517.0,\n",
       "  338.0,\n",
       "  2069.0,\n",
       "  682.0,\n",
       "  1141.0],\n",
       " 'GO:0043488-regulation of mRNA stability': [876.0,\n",
       "  653.0,\n",
       "  630.0,\n",
       "  402.0,\n",
       "  906.0,\n",
       "  1383.0,\n",
       "  1295.0,\n",
       "  931.0,\n",
       "  512.0,\n",
       "  544.0,\n",
       "  978.0,\n",
       "  1265.0,\n",
       "  1034.0,\n",
       "  401.0,\n",
       "  1351.0,\n",
       "  612.0,\n",
       "  736.0,\n",
       "  792.0,\n",
       "  1104.0,\n",
       "  550.0,\n",
       "  352.0,\n",
       "  1777.0,\n",
       "  609.0,\n",
       "  696.0],\n",
       " 'GO:0046718-viral entry into host cell': [877.0,\n",
       "  424.0,\n",
       "  214.0,\n",
       "  530.0,\n",
       "  768.0,\n",
       "  781.0,\n",
       "  569.0,\n",
       "  447.0,\n",
       "  361.0,\n",
       "  1234.0,\n",
       "  715.0,\n",
       "  372.0,\n",
       "  521.0,\n",
       "  975.0,\n",
       "  304.0,\n",
       "  730.0,\n",
       "  727.0,\n",
       "  629.0,\n",
       "  656.0,\n",
       "  921.0,\n",
       "  689.0,\n",
       "  676.0,\n",
       "  832.0,\n",
       "  475.0],\n",
       " 'GO:0016651-oxidoreductase activity, acting on NAD(P)H': [878.0,\n",
       "  354.0,\n",
       "  604.0,\n",
       "  671.0,\n",
       "  1011.0,\n",
       "  592.0,\n",
       "  384.0,\n",
       "  274.0,\n",
       "  539.0,\n",
       "  341.0,\n",
       "  241.0,\n",
       "  451.0,\n",
       "  437.0,\n",
       "  177.0,\n",
       "  328.0,\n",
       "  346.0,\n",
       "  824.0,\n",
       "  301.0,\n",
       "  587.0,\n",
       "  543.0,\n",
       "  107.0,\n",
       "  1022.0,\n",
       "  651.0,\n",
       "  464.0],\n",
       " 'GO:0005758-mitochondrial intermembrane space': [879.0,\n",
       "  742.0,\n",
       "  680.0,\n",
       "  678.0,\n",
       "  877.0,\n",
       "  521.0,\n",
       "  839.0,\n",
       "  1236.0,\n",
       "  377.0,\n",
       "  382.0,\n",
       "  580.0,\n",
       "  854.0,\n",
       "  685.0,\n",
       "  868.0,\n",
       "  521.0,\n",
       "  537.0,\n",
       "  645.0],\n",
       " 'GO:0042632-cholesterol homeostasis': [880.0,\n",
       "  460.0,\n",
       "  94.0,\n",
       "  539.0,\n",
       "  327.0,\n",
       "  695.0,\n",
       "  610.0,\n",
       "  1424.0,\n",
       "  253.0,\n",
       "  791.0,\n",
       "  365.0,\n",
       "  161.0,\n",
       "  606.0,\n",
       "  1472.0,\n",
       "  327.0,\n",
       "  384.0,\n",
       "  498.0,\n",
       "  345.0,\n",
       "  602.0,\n",
       "  1311.0,\n",
       "  330.0,\n",
       "  120.0,\n",
       "  726.0,\n",
       "  671.0,\n",
       "  289.0],\n",
       " 'GO:1900117-regulation of execution phase of apoptosis': [881.0,\n",
       "  283.0,\n",
       "  282.0,\n",
       "  198.0,\n",
       "  34.0,\n",
       "  248.0],\n",
       " 'GO:0051168-nuclear export': [882.0,\n",
       "  1154.0,\n",
       "  756.0,\n",
       "  801.0,\n",
       "  1428.0,\n",
       "  1404.0,\n",
       "  1789.0,\n",
       "  1160.0,\n",
       "  541.0,\n",
       "  1746.0,\n",
       "  518.0,\n",
       "  524.0,\n",
       "  693.0,\n",
       "  1088.0,\n",
       "  1947.0,\n",
       "  476.0,\n",
       "  2087.0,\n",
       "  1037.0,\n",
       "  867.0],\n",
       " 'GO:0072583-clathrin-dependent endocytosis': [883.0, 518.0],\n",
       " 'GO:2000107-negative regulation of leukocyte apoptotic process': [884.0,\n",
       "  407.0,\n",
       "  317.0,\n",
       "  350.0,\n",
       "  865.0,\n",
       "  317.0,\n",
       "  396.0,\n",
       "  467.0,\n",
       "  800.0,\n",
       "  484.0,\n",
       "  751.0,\n",
       "  776.0],\n",
       " 'GO:0072686-mitotic spindle': [885.0,\n",
       "  995.0,\n",
       "  867.0,\n",
       "  1509.0,\n",
       "  1963.0,\n",
       "  821.0,\n",
       "  934.0,\n",
       "  1504.0,\n",
       "  493.0,\n",
       "  604.0,\n",
       "  869.0,\n",
       "  889.0,\n",
       "  582.0,\n",
       "  1274.0,\n",
       "  1085.0],\n",
       " 'GO:0006901-vesicle coating': [886.0,\n",
       "  513.0,\n",
       "  542.0,\n",
       "  1067.0,\n",
       "  1002.0,\n",
       "  262.0,\n",
       "  460.0,\n",
       "  512.0,\n",
       "  709.0,\n",
       "  345.0,\n",
       "  458.0,\n",
       "  513.0,\n",
       "  972.0,\n",
       "  1079.0,\n",
       "  499.0,\n",
       "  661.0,\n",
       "  445.0],\n",
       " 'GO:1902110-positive regulation of mitochondrial membrane permeability involved in apoptotic process': [887.0,\n",
       "  567.0,\n",
       "  1046.0,\n",
       "  779.0,\n",
       "  747.0,\n",
       "  1278.0,\n",
       "  503.0,\n",
       "  781.0,\n",
       "  1274.0,\n",
       "  891.0],\n",
       " 'GO:0030520-intracellular estrogen receptor signaling pathway': [888.0,\n",
       "  245.0,\n",
       "  892.0,\n",
       "  761.0,\n",
       "  838.0,\n",
       "  624.0,\n",
       "  953.0,\n",
       "  299.0,\n",
       "  1247.0],\n",
       " 'GO:0008395-steroid hydroxylase activity': [889.0,\n",
       "  40.0,\n",
       "  576.0,\n",
       "  150.0,\n",
       "  154.0,\n",
       "  153.0],\n",
       " 'GO:0051287-NAD binding': [890.0,\n",
       "  416.0,\n",
       "  513.0,\n",
       "  377.0,\n",
       "  335.0,\n",
       "  197.0,\n",
       "  637.0,\n",
       "  590.0,\n",
       "  288.0,\n",
       "  453.0,\n",
       "  337.0,\n",
       "  762.0,\n",
       "  222.0,\n",
       "  903.0,\n",
       "  239.0,\n",
       "  957.0,\n",
       "  295.0,\n",
       "  342.0],\n",
       " 'GO:0000779-condensed chromosome, centromeric region': [891.0,\n",
       "  813.0,\n",
       "  389.0,\n",
       "  804.0,\n",
       "  1042.0,\n",
       "  1332.0,\n",
       "  790.0,\n",
       "  415.0,\n",
       "  795.0,\n",
       "  666.0,\n",
       "  1137.0,\n",
       "  1535.0,\n",
       "  244.0,\n",
       "  839.0,\n",
       "  427.0,\n",
       "  853.0,\n",
       "  524.0,\n",
       "  292.0,\n",
       "  145.0,\n",
       "  1420.0,\n",
       "  723.0,\n",
       "  630.0],\n",
       " 'GO:0051438-regulation of ubiquitin-protein transferase activity': [892.0,\n",
       "  860.0,\n",
       "  441.0,\n",
       "  894.0,\n",
       "  307.0,\n",
       "  1004.0,\n",
       "  400.0],\n",
       " 'GO:0017001-antibiotic catabolic process': [893.0,\n",
       "  515.0,\n",
       "  149.0,\n",
       "  695.0,\n",
       "  957.0,\n",
       "  164.0,\n",
       "  371.0,\n",
       "  122.0,\n",
       "  306.0,\n",
       "  313.0,\n",
       "  715.0,\n",
       "  474.0,\n",
       "  307.0,\n",
       "  545.0,\n",
       "  375.0],\n",
       " 'GO:0048199-vesicle targeting, to, from or within Golgi': [894.0,\n",
       "  609.0,\n",
       "  571.0,\n",
       "  1102.0,\n",
       "  971.0,\n",
       "  278.0,\n",
       "  468.0,\n",
       "  574.0,\n",
       "  748.0,\n",
       "  357.0,\n",
       "  579.0,\n",
       "  518.0,\n",
       "  1010.0,\n",
       "  1048.0,\n",
       "  496.0,\n",
       "  662.0,\n",
       "  469.0],\n",
       " 'GO:0006403-RNA localization': [895.0,\n",
       "  1201.0,\n",
       "  667.0,\n",
       "  905.0,\n",
       "  1440.0,\n",
       "  1415.0,\n",
       "  1482.0,\n",
       "  874.0,\n",
       "  465.0,\n",
       "  1362.0,\n",
       "  536.0,\n",
       "  1489.0,\n",
       "  470.0,\n",
       "  792.0,\n",
       "  1039.0,\n",
       "  1853.0,\n",
       "  411.0,\n",
       "  375.0,\n",
       "  2107.0,\n",
       "  810.0,\n",
       "  919.0],\n",
       " 'GO:0071103-DNA conformation change': [896.0,\n",
       "  714.0,\n",
       "  424.0,\n",
       "  457.0,\n",
       "  1021.0,\n",
       "  898.0,\n",
       "  1665.0,\n",
       "  1067.0,\n",
       "  544.0,\n",
       "  377.0,\n",
       "  692.0,\n",
       "  1283.0,\n",
       "  1807.0,\n",
       "  242.0,\n",
       "  1148.0,\n",
       "  440.0,\n",
       "  794.0,\n",
       "  794.0,\n",
       "  1670.0,\n",
       "  328.0,\n",
       "  345.0,\n",
       "  1739.0,\n",
       "  660.0,\n",
       "  1034.0],\n",
       " 'GO:0006513-protein monoubiquitination': [897.0,\n",
       "  494.0,\n",
       "  780.0,\n",
       "  1258.0,\n",
       "  1070.0,\n",
       "  947.0,\n",
       "  1402.0,\n",
       "  427.0,\n",
       "  494.0,\n",
       "  1066.0,\n",
       "  1289.0,\n",
       "  401.0,\n",
       "  1308.0,\n",
       "  486.0],\n",
       " 'GO:0006612-protein targeting to membrane': [898.0,\n",
       "  705.0,\n",
       "  635.0,\n",
       "  1195.0,\n",
       "  921.0,\n",
       "  1357.0,\n",
       "  739.0,\n",
       "  865.0,\n",
       "  411.0,\n",
       "  1638.0,\n",
       "  968.0,\n",
       "  772.0,\n",
       "  880.0,\n",
       "  690.0,\n",
       "  584.0,\n",
       "  738.0,\n",
       "  1272.0,\n",
       "  1068.0,\n",
       "  1383.0,\n",
       "  783.0,\n",
       "  392.0,\n",
       "  1161.0,\n",
       "  1843.0,\n",
       "  889.0],\n",
       " 'GO:2000134-negative regulation of G1/S transition of mitotic cell cycle': [899.0,\n",
       "  926.0,\n",
       "  400.0,\n",
       "  406.0,\n",
       "  746.0,\n",
       "  645.0,\n",
       "  716.0,\n",
       "  987.0,\n",
       "  1121.0,\n",
       "  684.0,\n",
       "  967.0,\n",
       "  1615.0,\n",
       "  687.0,\n",
       "  733.0,\n",
       "  645.0,\n",
       "  1069.0,\n",
       "  836.0,\n",
       "  1036.0,\n",
       "  822.0,\n",
       "  176.0,\n",
       "  1470.0,\n",
       "  778.0,\n",
       "  731.0],\n",
       " 'GO:0035967-cellular response to topologically incorrect protein': [900.0,\n",
       "  1255.0,\n",
       "  585.0,\n",
       "  883.0,\n",
       "  1342.0,\n",
       "  1169.0,\n",
       "  1319.0,\n",
       "  482.0,\n",
       "  583.0,\n",
       "  884.0,\n",
       "  1151.0,\n",
       "  1497.0,\n",
       "  571.0,\n",
       "  1078.0,\n",
       "  541.0,\n",
       "  1416.0,\n",
       "  931.0,\n",
       "  1693.0,\n",
       "  561.0,\n",
       "  160.0,\n",
       "  1443.0,\n",
       "  653.0,\n",
       "  540.0],\n",
       " 'GO:0051985-negative regulation of chromosome segregation': [901.0,\n",
       "  362.0,\n",
       "  612.0,\n",
       "  1387.0,\n",
       "  733.0,\n",
       "  507.0,\n",
       "  955.0,\n",
       "  262.0,\n",
       "  470.0],\n",
       " 'GO:0046627-negative regulation of insulin receptor signaling pathway': 902.0,\n",
       " 'GO:0010633-negative regulation of epithelial cell migration': [903.0,\n",
       "  618.0,\n",
       "  220.0,\n",
       "  401.0,\n",
       "  441.0,\n",
       "  1078.0,\n",
       "  224.0,\n",
       "  1302.0,\n",
       "  171.0,\n",
       "  266.0,\n",
       "  281.0,\n",
       "  1576.0,\n",
       "  178.0,\n",
       "  792.0,\n",
       "  965.0,\n",
       "  710.0,\n",
       "  620.0,\n",
       "  178.0,\n",
       "  695.0,\n",
       "  936.0,\n",
       "  334.0],\n",
       " 'GO:1900407-regulation of cellular response to oxidative stress': [904.0,\n",
       "  1123.0,\n",
       "  1166.0,\n",
       "  632.0,\n",
       "  963.0,\n",
       "  1594.0,\n",
       "  1759.0],\n",
       " 'GO:0034113-heterotypic cell-cell adhesion': [905.0,\n",
       "  710.0,\n",
       "  539.0,\n",
       "  382.0,\n",
       "  1028.0,\n",
       "  667.0,\n",
       "  865.0,\n",
       "  585.0],\n",
       " 'GO:0050854-regulation of antigen receptor-mediated signaling pathway': [906.0,\n",
       "  140.0,\n",
       "  717.0,\n",
       "  608.0,\n",
       "  722.0,\n",
       "  567.0,\n",
       "  459.0,\n",
       "  626.0,\n",
       "  823.0,\n",
       "  299.0,\n",
       "  756.0,\n",
       "  667.0,\n",
       "  905.0,\n",
       "  873.0,\n",
       "  793.0],\n",
       " 'GO:0035872-nucleotide-binding domain, leucine rich repeat containing receptor signaling pathway': [907.0,\n",
       "  569.0,\n",
       "  384.0],\n",
       " 'GO:0017148-negative regulation of translation': [908.0,\n",
       "  600.0,\n",
       "  371.0,\n",
       "  329.0,\n",
       "  470.0,\n",
       "  797.0,\n",
       "  941.0,\n",
       "  1006.0,\n",
       "  249.0,\n",
       "  458.0,\n",
       "  482.0,\n",
       "  367.0,\n",
       "  690.0,\n",
       "  1074.0,\n",
       "  424.0,\n",
       "  685.0,\n",
       "  375.0,\n",
       "  678.0,\n",
       "  670.0,\n",
       "  523.0,\n",
       "  416.0,\n",
       "  90.0,\n",
       "  887.0,\n",
       "  623.0,\n",
       "  466.0],\n",
       " 'GO:0043531-ADP binding': [909.0,\n",
       "  485.0,\n",
       "  394.0,\n",
       "  459.0,\n",
       "  780.0,\n",
       "  369.0,\n",
       "  325.0,\n",
       "  280.0],\n",
       " 'GO:0006639-acylglycerol metabolic process': [910.0,\n",
       "  890.0,\n",
       "  390.0,\n",
       "  1148.0,\n",
       "  797.0,\n",
       "  1313.0,\n",
       "  953.0,\n",
       "  1942.0,\n",
       "  418.0,\n",
       "  798.0,\n",
       "  375.0,\n",
       "  947.0,\n",
       "  1687.0,\n",
       "  1081.0,\n",
       "  882.0,\n",
       "  981.0,\n",
       "  1612.0,\n",
       "  1005.0,\n",
       "  1691.0,\n",
       "  616.0],\n",
       " 'GO:0051881-regulation of mitochondrial membrane potential': [911.0,\n",
       "  565.0,\n",
       "  543.0,\n",
       "  816.0,\n",
       "  790.0,\n",
       "  516.0,\n",
       "  1000.0,\n",
       "  870.0,\n",
       "  1105.0,\n",
       "  679.0,\n",
       "  454.0,\n",
       "  677.0,\n",
       "  1135.0,\n",
       "  713.0],\n",
       " 'GO:0003009-skeletal muscle contraction': [912.0, 341.0, 117.0, 496.0],\n",
       " 'GO:0036473-cell death in response to oxidative stress': [913.0,\n",
       "  948.0,\n",
       "  1184.0,\n",
       "  933.0,\n",
       "  1020.0,\n",
       "  811.0,\n",
       "  885.0,\n",
       "  1083.0,\n",
       "  1584.0,\n",
       "  1284.0],\n",
       " 'GO:0010662-regulation of striated muscle cell apoptotic process': [914.0,\n",
       "  233.0,\n",
       "  466.0,\n",
       "  754.0,\n",
       "  517.0,\n",
       "  1010.0],\n",
       " 'GO:0055090-acylglycerol homeostasis': [915.5,\n",
       "  310.5,\n",
       "  355.5,\n",
       "  291.5,\n",
       "  467.5,\n",
       "  465.5,\n",
       "  517.5,\n",
       "  92.5,\n",
       "  283.5,\n",
       "  356.5,\n",
       "  399.5,\n",
       "  153.5,\n",
       "  172.5],\n",
       " 'GO:0070328-triglyceride homeostasis': [915.5,\n",
       "  310.5,\n",
       "  355.5,\n",
       "  291.5,\n",
       "  467.5,\n",
       "  465.5,\n",
       "  517.5,\n",
       "  92.5,\n",
       "  283.5,\n",
       "  356.5,\n",
       "  399.5,\n",
       "  153.5,\n",
       "  172.5],\n",
       " 'GO:0050798-activated T cell proliferation': [917.0,\n",
       "  408.0,\n",
       "  467.0,\n",
       "  613.0,\n",
       "  407.0,\n",
       "  387.0,\n",
       "  1315.0,\n",
       "  262.0,\n",
       "  346.0,\n",
       "  387.0,\n",
       "  429.0,\n",
       "  269.0],\n",
       " 'GO:0004867-serine-type endopeptidase inhibitor activity': [918.0,\n",
       "  294.0,\n",
       "  102.0,\n",
       "  292.0,\n",
       "  194.0,\n",
       "  536.0,\n",
       "  424.0,\n",
       "  887.0,\n",
       "  119.0,\n",
       "  739.0,\n",
       "  167.0,\n",
       "  86.0,\n",
       "  250.0,\n",
       "  321.0,\n",
       "  291.0,\n",
       "  166.0,\n",
       "  353.0,\n",
       "  291.0,\n",
       "  225.0,\n",
       "  158.0,\n",
       "  700.0,\n",
       "  550.0],\n",
       " 'GO:0001889-liver development': [919.0,\n",
       "  1122.0,\n",
       "  539.0,\n",
       "  1459.0,\n",
       "  1012.0,\n",
       "  1357.0,\n",
       "  1285.0,\n",
       "  1201.0,\n",
       "  2179.0,\n",
       "  1279.0,\n",
       "  1025.0,\n",
       "  1388.0,\n",
       "  1663.0,\n",
       "  2055.0],\n",
       " \"GO:0034062-5'-3' RNA polymerase activity\": [920.5,\n",
       "  618.5,\n",
       "  574.5,\n",
       "  514.5,\n",
       "  607.5,\n",
       "  383.5,\n",
       "  483.5,\n",
       "  551.5],\n",
       " 'GO:0097747-RNA polymerase activity': [920.5,\n",
       "  618.5,\n",
       "  574.5,\n",
       "  514.5,\n",
       "  607.5,\n",
       "  383.5,\n",
       "  483.5,\n",
       "  551.5],\n",
       " 'GO:0010761-fibroblast migration': [922.0, 916.0],\n",
       " 'GO:0015036-disulfide oxidoreductase activity': [923.0,\n",
       "  557.0,\n",
       "  564.0,\n",
       "  239.0,\n",
       "  412.0,\n",
       "  462.0,\n",
       "  294.0,\n",
       "  784.0,\n",
       "  285.0],\n",
       " 'GO:0007006-mitochondrial membrane organization': [924.0,\n",
       "  535.0,\n",
       "  778.0,\n",
       "  914.0,\n",
       "  1024.0,\n",
       "  901.0,\n",
       "  589.0,\n",
       "  758.0,\n",
       "  941.0,\n",
       "  1119.0,\n",
       "  956.0,\n",
       "  403.0,\n",
       "  978.0,\n",
       "  358.0,\n",
       "  725.0,\n",
       "  690.0,\n",
       "  1057.0,\n",
       "  651.0,\n",
       "  126.0,\n",
       "  1727.0,\n",
       "  746.0,\n",
       "  941.0],\n",
       " 'GO:1902750-negative regulation of cell cycle G2/M phase transition': [925.0,\n",
       "  443.0,\n",
       "  397.0,\n",
       "  681.0,\n",
       "  1203.0,\n",
       "  1366.0,\n",
       "  378.0,\n",
       "  392.0,\n",
       "  361.0,\n",
       "  589.0,\n",
       "  999.0,\n",
       "  985.0,\n",
       "  193.0,\n",
       "  869.0,\n",
       "  377.0,\n",
       "  806.0,\n",
       "  479.0,\n",
       "  853.0,\n",
       "  310.0,\n",
       "  142.0,\n",
       "  1684.0,\n",
       "  394.0,\n",
       "  527.0],\n",
       " 'GO:0038095-Fc-epsilon receptor signaling pathway': [926.0,\n",
       "  756.0,\n",
       "  349.0,\n",
       "  409.0,\n",
       "  862.0,\n",
       "  1307.0,\n",
       "  1151.0,\n",
       "  557.0,\n",
       "  391.0,\n",
       "  748.0,\n",
       "  647.0,\n",
       "  1096.0,\n",
       "  966.0,\n",
       "  395.0,\n",
       "  842.0,\n",
       "  649.0,\n",
       "  606.0,\n",
       "  491.0,\n",
       "  910.0,\n",
       "  389.0,\n",
       "  1427.0,\n",
       "  859.0,\n",
       "  423.0],\n",
       " 'GO:0000288-nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay': 927.0,\n",
       " 'GO:0035966-response to topologically incorrect protein': [928.0,\n",
       "  1078.0,\n",
       "  684.0,\n",
       "  618.0,\n",
       "  932.0,\n",
       "  1322.0,\n",
       "  1201.0,\n",
       "  1314.0,\n",
       "  443.0,\n",
       "  607.0,\n",
       "  914.0,\n",
       "  1139.0,\n",
       "  1357.0,\n",
       "  572.0,\n",
       "  1054.0,\n",
       "  532.0,\n",
       "  1301.0,\n",
       "  808.0,\n",
       "  1731.0,\n",
       "  604.0,\n",
       "  163.0,\n",
       "  1426.0,\n",
       "  614.0,\n",
       "  610.0],\n",
       " 'GO:0006778-porphyrin-containing compound metabolic process': [929.0,\n",
       "  475.0,\n",
       "  416.0,\n",
       "  626.0,\n",
       "  779.0,\n",
       "  329.0,\n",
       "  809.0],\n",
       " 'GO:0097327-response to antineoplastic agent': [930.0,\n",
       "  908.0,\n",
       "  856.0,\n",
       "  1478.0,\n",
       "  1204.0,\n",
       "  750.0,\n",
       "  1203.0,\n",
       "  2262.0,\n",
       "  1069.0,\n",
       "  921.0,\n",
       "  1253.0,\n",
       "  1052.0,\n",
       "  1490.0,\n",
       "  1295.0,\n",
       "  1425.0,\n",
       "  682.0],\n",
       " 'GO:0099086-synaptonemal structure': [931.5, 305.5],\n",
       " 'GO:0000795-synaptonemal complex': [931.5, 305.5],\n",
       " 'GO:0005925-focal adhesion': [933.0,\n",
       "  752.0,\n",
       "  419.0,\n",
       "  669.0,\n",
       "  644.0,\n",
       "  1234.0,\n",
       "  674.0,\n",
       "  1081.0,\n",
       "  421.0,\n",
       "  973.0,\n",
       "  611.0,\n",
       "  450.0,\n",
       "  632.0,\n",
       "  808.0,\n",
       "  638.0,\n",
       "  735.0,\n",
       "  1053.0,\n",
       "  1532.0,\n",
       "  1180.0,\n",
       "  1183.0,\n",
       "  940.0,\n",
       "  344.0,\n",
       "  819.0,\n",
       "  1134.0,\n",
       "  623.0],\n",
       " 'GO:0046364-monosaccharide biosynthetic process': [934.0,\n",
       "  489.0,\n",
       "  208.0,\n",
       "  325.0,\n",
       "  542.0,\n",
       "  636.0,\n",
       "  947.0,\n",
       "  613.0,\n",
       "  327.0,\n",
       "  477.0,\n",
       "  673.0,\n",
       "  232.0,\n",
       "  627.0,\n",
       "  885.0,\n",
       "  360.0,\n",
       "  517.0,\n",
       "  500.0,\n",
       "  507.0,\n",
       "  545.0,\n",
       "  1219.0,\n",
       "  358.0,\n",
       "  150.0,\n",
       "  1064.0,\n",
       "  1076.0,\n",
       "  463.0],\n",
       " 'GO:2000403-positive regulation of lymphocyte migration': [935.0,\n",
       "  230.0,\n",
       "  277.0,\n",
       "  248.0,\n",
       "  127.0,\n",
       "  325.0,\n",
       "  108.0,\n",
       "  280.0,\n",
       "  294.0,\n",
       "  289.0,\n",
       "  604.0,\n",
       "  298.0],\n",
       " 'GO:0019882-antigen processing and presentation': [936.0,\n",
       "  446.0,\n",
       "  316.0,\n",
       "  438.0,\n",
       "  750.0,\n",
       "  1350.0,\n",
       "  732.0,\n",
       "  442.0,\n",
       "  480.0,\n",
       "  979.0,\n",
       "  461.0,\n",
       "  374.0,\n",
       "  744.0,\n",
       "  770.0,\n",
       "  270.0,\n",
       "  535.0,\n",
       "  768.0,\n",
       "  746.0,\n",
       "  426.0,\n",
       "  709.0,\n",
       "  695.0,\n",
       "  376.0,\n",
       "  825.0,\n",
       "  536.0,\n",
       "  673.0],\n",
       " 'GO:0005924-cell-substrate adherens junction': [937.0,\n",
       "  751.0,\n",
       "  422.0,\n",
       "  660.0,\n",
       "  642.0,\n",
       "  1243.0,\n",
       "  685.0,\n",
       "  1083.0,\n",
       "  420.0,\n",
       "  976.0,\n",
       "  610.0,\n",
       "  452.0,\n",
       "  631.0,\n",
       "  810.0,\n",
       "  642.0,\n",
       "  736.0,\n",
       "  1050.0,\n",
       "  1536.0,\n",
       "  1196.0,\n",
       "  1181.0,\n",
       "  942.0,\n",
       "  346.0,\n",
       "  844.0,\n",
       "  1133.0,\n",
       "  622.0],\n",
       " 'GO:0030330-DNA damage response, signal transduction by p53 class mediator': [938.0,\n",
       "  1223.0,\n",
       "  842.0,\n",
       "  832.0,\n",
       "  1346.0,\n",
       "  496.0,\n",
       "  1042.0,\n",
       "  2277.0,\n",
       "  976.0,\n",
       "  748.0,\n",
       "  1323.0,\n",
       "  1386.0,\n",
       "  1189.0,\n",
       "  929.0],\n",
       " 'GO:1900408-negative regulation of cellular response to oxidative stress': 939.0,\n",
       " 'GO:0002260-lymphocyte homeostasis': [940.0,\n",
       "  219.0,\n",
       "  679.0,\n",
       "  569.0,\n",
       "  703.0,\n",
       "  722.0,\n",
       "  411.0,\n",
       "  357.0,\n",
       "  837.0,\n",
       "  413.0,\n",
       "  331.0,\n",
       "  611.0,\n",
       "  449.0,\n",
       "  638.0,\n",
       "  547.0,\n",
       "  488.0,\n",
       "  594.0],\n",
       " 'GO:1902686-mitochondrial outer membrane permeabilization involved in programmed cell death': [941.0,\n",
       "  540.0,\n",
       "  1089.0,\n",
       "  702.0,\n",
       "  727.0,\n",
       "  1329.0,\n",
       "  525.0,\n",
       "  703.0,\n",
       "  1302.0,\n",
       "  897.0],\n",
       " 'GO:0010494-cytoplasmic stress granule': [942.0,\n",
       "  1303.0,\n",
       "  535.0,\n",
       "  596.0,\n",
       "  657.0,\n",
       "  594.0,\n",
       "  883.0,\n",
       "  588.0],\n",
       " 'GO:0001098-basal transcription machinery binding': [943.5,\n",
       "  486.5,\n",
       "  339.5,\n",
       "  1290.5,\n",
       "  573.5,\n",
       "  1419.5,\n",
       "  355.5],\n",
       " 'GO:0001099-basal RNA polymerase II transcription machinery binding': [943.5,\n",
       "  486.5,\n",
       "  339.5,\n",
       "  1290.5,\n",
       "  573.5,\n",
       "  1419.5,\n",
       "  355.5],\n",
       " 'GO:0031227-intrinsic component of endoplasmic reticulum membrane': [945.0,\n",
       "  791.0,\n",
       "  689.0,\n",
       "  1078.0,\n",
       "  1148.0,\n",
       "  911.0,\n",
       "  1425.0,\n",
       "  1246.0,\n",
       "  1129.0,\n",
       "  776.0,\n",
       "  488.0,\n",
       "  1167.0,\n",
       "  1556.0,\n",
       "  656.0,\n",
       "  719.0,\n",
       "  797.0,\n",
       "  1616.0,\n",
       "  874.0,\n",
       "  1720.0,\n",
       "  851.0,\n",
       "  291.0,\n",
       "  1127.0,\n",
       "  901.0,\n",
       "  909.0],\n",
       " 'GO:0009060-aerobic respiration': [946.0,\n",
       "  425.0,\n",
       "  357.0,\n",
       "  652.0,\n",
       "  267.0,\n",
       "  470.0,\n",
       "  211.0,\n",
       "  484.0,\n",
       "  532.0,\n",
       "  764.0,\n",
       "  398.0,\n",
       "  158.0,\n",
       "  372.0,\n",
       "  387.0,\n",
       "  815.0,\n",
       "  278.0,\n",
       "  547.0,\n",
       "  303.0,\n",
       "  80.0,\n",
       "  1331.0,\n",
       "  508.0,\n",
       "  358.0],\n",
       " 'GO:0000784-nuclear chromosome, telomeric region': [947.0,\n",
       "  981.0,\n",
       "  508.0,\n",
       "  622.0,\n",
       "  1257.0,\n",
       "  807.0,\n",
       "  1771.0,\n",
       "  920.0,\n",
       "  701.0,\n",
       "  925.0,\n",
       "  1650.0,\n",
       "  369.0,\n",
       "  551.0,\n",
       "  950.0,\n",
       "  1090.0,\n",
       "  1718.0,\n",
       "  342.0,\n",
       "  605.0,\n",
       "  1078.0],\n",
       " 'GO:0045840-positive regulation of mitotic nuclear division': [948.0,\n",
       "  653.0,\n",
       "  875.0,\n",
       "  680.0,\n",
       "  595.0],\n",
       " 'GO:2000351-regulation of endothelial cell apoptotic process': [949.0,\n",
       "  452.0,\n",
       "  371.0,\n",
       "  192.0,\n",
       "  579.0,\n",
       "  757.0,\n",
       "  792.0,\n",
       "  220.0,\n",
       "  304.0,\n",
       "  140.0,\n",
       "  372.0,\n",
       "  593.0,\n",
       "  396.0,\n",
       "  700.0,\n",
       "  238.0,\n",
       "  480.0,\n",
       "  584.0,\n",
       "  274.0],\n",
       " 'GO:0032200-telomere organization': [950.0,\n",
       "  955.0,\n",
       "  732.0,\n",
       "  1388.0,\n",
       "  1174.0,\n",
       "  1540.0,\n",
       "  1287.0,\n",
       "  395.0,\n",
       "  1620.0,\n",
       "  447.0,\n",
       "  463.0,\n",
       "  929.0,\n",
       "  993.0,\n",
       "  2072.0,\n",
       "  485.0,\n",
       "  996.0,\n",
       "  1195.0],\n",
       " 'GO:0032508-DNA duplex unwinding': [951.0,\n",
       "  903.0,\n",
       "  741.0,\n",
       "  1474.0,\n",
       "  2008.0,\n",
       "  1065.0,\n",
       "  461.0,\n",
       "  893.0,\n",
       "  2172.0,\n",
       "  483.0,\n",
       "  465.0,\n",
       "  754.0,\n",
       "  801.0,\n",
       "  340.0,\n",
       "  155.0,\n",
       "  794.0,\n",
       "  543.0],\n",
       " 'GO:0009301-snRNA transcription': [952.0,\n",
       "  1333.0,\n",
       "  1270.0,\n",
       "  378.0,\n",
       "  1697.0,\n",
       "  483.0,\n",
       "  1026.0,\n",
       "  1557.0,\n",
       "  435.0],\n",
       " 'GO:0042036-negative regulation of cytokine biosynthetic process': [953.0,\n",
       "  56.0,\n",
       "  299.0,\n",
       "  299.0,\n",
       "  815.0,\n",
       "  419.0,\n",
       "  198.0,\n",
       "  596.0,\n",
       "  411.0],\n",
       " 'GO:0032392-DNA geometric change': [954.0,\n",
       "  899.0,\n",
       "  720.0,\n",
       "  1583.0,\n",
       "  1991.0,\n",
       "  1009.0,\n",
       "  451.0,\n",
       "  836.0,\n",
       "  2047.0,\n",
       "  469.0,\n",
       "  1285.0,\n",
       "  481.0,\n",
       "  751.0,\n",
       "  861.0,\n",
       "  337.0,\n",
       "  175.0,\n",
       "  690.0,\n",
       "  561.0],\n",
       " 'GO:0006417-regulation of translation': [955.0,\n",
       "  693.0,\n",
       "  450.0,\n",
       "  518.0,\n",
       "  969.0,\n",
       "  1233.0,\n",
       "  1164.0,\n",
       "  1027.0,\n",
       "  409.0,\n",
       "  784.0,\n",
       "  918.0,\n",
       "  1317.0,\n",
       "  1600.0,\n",
       "  479.0,\n",
       "  854.0,\n",
       "  501.0,\n",
       "  801.0,\n",
       "  862.0,\n",
       "  1356.0,\n",
       "  480.0,\n",
       "  156.0,\n",
       "  1486.0,\n",
       "  825.0,\n",
       "  679.0],\n",
       " 'GO:0051353-positive regulation of oxidoreductase activity': [956.0,\n",
       "  738.0,\n",
       "  1010.0,\n",
       "  533.0,\n",
       "  999.0,\n",
       "  1208.0,\n",
       "  376.0,\n",
       "  1002.0,\n",
       "  1031.0],\n",
       " 'GO:1903556-negative regulation of tumor necrosis factor superfamily cytokine production': [957.0,\n",
       "  392.0,\n",
       "  126.0,\n",
       "  638.0,\n",
       "  537.0,\n",
       "  262.0,\n",
       "  1130.0,\n",
       "  831.0,\n",
       "  370.0,\n",
       "  418.0,\n",
       "  1168.0,\n",
       "  294.0,\n",
       "  189.0,\n",
       "  702.0,\n",
       "  613.0,\n",
       "  1042.0,\n",
       "  444.0,\n",
       "  614.0],\n",
       " 'GO:0000777-condensed chromosome kinetochore': [958.0,\n",
       "  881.0,\n",
       "  380.0,\n",
       "  725.0,\n",
       "  1105.0,\n",
       "  1269.0,\n",
       "  944.0,\n",
       "  381.0,\n",
       "  824.0,\n",
       "  682.0,\n",
       "  1202.0,\n",
       "  1693.0,\n",
       "  235.0,\n",
       "  762.0,\n",
       "  404.0,\n",
       "  876.0,\n",
       "  523.0,\n",
       "  321.0,\n",
       "  136.0,\n",
       "  1473.0,\n",
       "  738.0,\n",
       "  675.0],\n",
       " 'GO:0060071-Wnt signaling pathway, planar cell polarity pathway': [959.0,\n",
       "  765.0,\n",
       "  838.0,\n",
       "  751.0,\n",
       "  1125.0,\n",
       "  720.0,\n",
       "  412.0,\n",
       "  603.0,\n",
       "  888.0,\n",
       "  1117.0,\n",
       "  994.0,\n",
       "  1087.0,\n",
       "  569.0,\n",
       "  1292.0,\n",
       "  541.0,\n",
       "  1538.0,\n",
       "  614.0,\n",
       "  1513.0,\n",
       "  595.0,\n",
       "  1089.0],\n",
       " 'GO:0030055-cell-substrate junction': [960.0,\n",
       "  795.0,\n",
       "  436.0,\n",
       "  689.0,\n",
       "  663.0,\n",
       "  1278.0,\n",
       "  686.0,\n",
       "  1106.0,\n",
       "  435.0,\n",
       "  1007.0,\n",
       "  618.0,\n",
       "  459.0,\n",
       "  629.0,\n",
       "  834.0,\n",
       "  659.0,\n",
       "  734.0,\n",
       "  1073.0,\n",
       "  1553.0,\n",
       "  1202.0,\n",
       "  1193.0,\n",
       "  956.0,\n",
       "  360.0,\n",
       "  838.0,\n",
       "  1168.0,\n",
       "  647.0],\n",
       " 'GO:0043392-negative regulation of DNA binding': [961.0,\n",
       "  631.0,\n",
       "  1084.0,\n",
       "  1476.0,\n",
       "  964.0],\n",
       " 'GO:0030574-collagen catabolic process': [962.0, 385.0, 105.0, 1491.0],\n",
       " 'GO:0033238-regulation of cellular amine metabolic process': [963.0,\n",
       "  522.0,\n",
       "  341.0,\n",
       "  552.0,\n",
       "  429.0,\n",
       "  214.0,\n",
       "  326.0,\n",
       "  650.0,\n",
       "  440.0,\n",
       "  635.0,\n",
       "  347.0,\n",
       "  198.0,\n",
       "  681.0,\n",
       "  265.0,\n",
       "  583.0,\n",
       "  337.0,\n",
       "  754.0,\n",
       "  234.0,\n",
       "  984.0,\n",
       "  318.0,\n",
       "  374.0],\n",
       " 'GO:0019218-regulation of steroid metabolic process': [964.0,\n",
       "  778.0,\n",
       "  171.0,\n",
       "  824.0,\n",
       "  974.0,\n",
       "  1475.0,\n",
       "  908.0,\n",
       "  1116.0,\n",
       "  998.0,\n",
       "  528.0,\n",
       "  823.0,\n",
       "  2014.0,\n",
       "  521.0,\n",
       "  528.0,\n",
       "  586.0,\n",
       "  1085.0,\n",
       "  1550.0,\n",
       "  730.0,\n",
       "  256.0,\n",
       "  931.0,\n",
       "  777.0,\n",
       "  850.0],\n",
       " 'GO:0048208-COPII vesicle coating': [965.5,\n",
       "  558.5,\n",
       "  534.5,\n",
       "  1007.5,\n",
       "  861.5,\n",
       "  523.5,\n",
       "  772.5,\n",
       "  380.5,\n",
       "  421.5,\n",
       "  556.5,\n",
       "  915.5,\n",
       "  1086.5,\n",
       "  528.5,\n",
       "  668.5,\n",
       "  456.5],\n",
       " 'GO:0048207-vesicle targeting, rough ER to cis-Golgi': [965.5,\n",
       "  558.5,\n",
       "  534.5,\n",
       "  1007.5,\n",
       "  861.5,\n",
       "  523.5,\n",
       "  772.5,\n",
       "  380.5,\n",
       "  421.5,\n",
       "  556.5,\n",
       "  915.5,\n",
       "  1086.5,\n",
       "  528.5,\n",
       "  668.5,\n",
       "  456.5],\n",
       " 'GO:0010596-negative regulation of endothelial cell migration': [967.0,\n",
       "  411.0,\n",
       "  168.0,\n",
       "  328.0,\n",
       "  452.0,\n",
       "  782.0,\n",
       "  107.0,\n",
       "  1301.0,\n",
       "  98.0,\n",
       "  179.0,\n",
       "  180.0,\n",
       "  742.0,\n",
       "  152.0,\n",
       "  268.0,\n",
       "  479.0,\n",
       "  555.0,\n",
       "  465.0,\n",
       "  73.0,\n",
       "  358.0,\n",
       "  842.0,\n",
       "  264.0],\n",
       " 'GO:0046626-regulation of insulin receptor signaling pathway': [968.0,\n",
       "  1133.0,\n",
       "  1948.0,\n",
       "  429.0,\n",
       "  1156.0,\n",
       "  479.0,\n",
       "  1094.0,\n",
       "  969.0],\n",
       " 'GO:0032720-negative regulation of tumor necrosis factor production': [969.0,\n",
       "  374.0,\n",
       "  134.0,\n",
       "  597.0,\n",
       "  548.0,\n",
       "  248.0,\n",
       "  1179.0,\n",
       "  913.0,\n",
       "  389.0,\n",
       "  439.0,\n",
       "  1271.0,\n",
       "  261.0,\n",
       "  802.0,\n",
       "  148.0,\n",
       "  611.0,\n",
       "  589.0,\n",
       "  937.0,\n",
       "  483.0,\n",
       "  626.0],\n",
       " 'GO:0061008-hepaticobiliary system development': [970.0,\n",
       "  1189.0,\n",
       "  1499.0,\n",
       "  1184.0,\n",
       "  1420.0,\n",
       "  1326.0,\n",
       "  1275.0,\n",
       "  2339.0,\n",
       "  1059.0,\n",
       "  1456.0,\n",
       "  1766.0,\n",
       "  2102.0],\n",
       " 'GO:0090307-mitotic spindle assembly': [971.0,\n",
       "  428.0,\n",
       "  888.0,\n",
       "  659.0,\n",
       "  1038.0,\n",
       "  435.0,\n",
       "  1062.0,\n",
       "  291.0],\n",
       " 'GO:0030301-cholesterol transport': [972.0,\n",
       "  832.0,\n",
       "  948.0,\n",
       "  618.0,\n",
       "  1019.0,\n",
       "  1584.0,\n",
       "  384.0,\n",
       "  865.0,\n",
       "  536.0,\n",
       "  299.0,\n",
       "  679.0,\n",
       "  1570.0,\n",
       "  538.0,\n",
       "  865.0,\n",
       "  1187.0,\n",
       "  1253.0,\n",
       "  719.0,\n",
       "  1203.0,\n",
       "  555.0],\n",
       " 'GO:0098781-ncRNA transcription': [973.0,\n",
       "  1196.0,\n",
       "  876.0,\n",
       "  1330.0,\n",
       "  1214.0,\n",
       "  393.0,\n",
       "  1780.0,\n",
       "  510.0,\n",
       "  648.0,\n",
       "  1037.0,\n",
       "  715.0,\n",
       "  157.0,\n",
       "  1791.0,\n",
       "  1042.0],\n",
       " 'GO:1900077-negative regulation of cellular response to insulin stimulus': 974.0,\n",
       " 'GO:0030176-integral component of endoplasmic reticulum membrane': [975.0,\n",
       "  747.0,\n",
       "  1054.0,\n",
       "  1073.0,\n",
       "  899.0,\n",
       "  1377.0,\n",
       "  1180.0,\n",
       "  1088.0,\n",
       "  678.0,\n",
       "  464.0,\n",
       "  1196.0,\n",
       "  1407.0,\n",
       "  669.0,\n",
       "  751.0,\n",
       "  807.0,\n",
       "  1733.0,\n",
       "  856.0,\n",
       "  1662.0,\n",
       "  848.0,\n",
       "  220.0,\n",
       "  1155.0,\n",
       "  893.0,\n",
       "  898.0],\n",
       " 'GO:0008380-RNA splicing': [976.0,\n",
       "  924.0,\n",
       "  645.0,\n",
       "  887.0,\n",
       "  1201.0,\n",
       "  1212.0,\n",
       "  1749.0,\n",
       "  1153.0,\n",
       "  688.0,\n",
       "  714.0,\n",
       "  1221.0,\n",
       "  1563.0,\n",
       "  1349.0,\n",
       "  389.0,\n",
       "  1504.0,\n",
       "  512.0,\n",
       "  829.0,\n",
       "  839.0,\n",
       "  1527.0,\n",
       "  477.0,\n",
       "  354.0,\n",
       "  2176.0,\n",
       "  1118.0,\n",
       "  1064.0],\n",
       " 'GO:1900076-regulation of cellular response to insulin stimulus': [977.0,\n",
       "  206.0,\n",
       "  1324.0,\n",
       "  1165.0,\n",
       "  972.0,\n",
       "  1817.0,\n",
       "  786.0,\n",
       "  1195.0,\n",
       "  1251.0,\n",
       "  513.0,\n",
       "  1194.0,\n",
       "  868.0],\n",
       " 'GO:0042795-snRNA transcription by RNA polymerase II': [978.0,\n",
       "  1368.0,\n",
       "  1043.0,\n",
       "  387.0,\n",
       "  1507.0,\n",
       "  455.0,\n",
       "  1032.0,\n",
       "  1445.0,\n",
       "  438.0],\n",
       " 'GO:2000257-regulation of protein activation cascade': [979.0,\n",
       "  237.0,\n",
       "  67.0,\n",
       "  293.0,\n",
       "  349.0,\n",
       "  244.0,\n",
       "  398.0,\n",
       "  389.0,\n",
       "  154.0,\n",
       "  830.0,\n",
       "  200.0,\n",
       "  76.0,\n",
       "  218.0,\n",
       "  545.0,\n",
       "  109.0,\n",
       "  155.0,\n",
       "  150.0,\n",
       "  321.0,\n",
       "  295.0,\n",
       "  545.0,\n",
       "  124.0],\n",
       " 'GO:0005743-mitochondrial inner membrane': [980.0,\n",
       "  480.0,\n",
       "  724.0,\n",
       "  944.0,\n",
       "  736.0,\n",
       "  1604.0,\n",
       "  746.0,\n",
       "  480.0,\n",
       "  342.0,\n",
       "  767.0,\n",
       "  768.0,\n",
       "  551.0,\n",
       "  1146.0,\n",
       "  805.0,\n",
       "  248.0,\n",
       "  512.0,\n",
       "  327.0,\n",
       "  850.0,\n",
       "  530.0,\n",
       "  1325.0,\n",
       "  599.0,\n",
       "  130.0,\n",
       "  1641.0,\n",
       "  637.0,\n",
       "  650.0],\n",
       " 'GO:1903202-negative regulation of oxidative stress-induced cell death': 981.0,\n",
       " 'GO:0035307-positive regulation of protein dephosphorylation': [982.0,\n",
       "  625.0,\n",
       "  803.0,\n",
       "  381.0,\n",
       "  514.0,\n",
       "  884.0,\n",
       "  569.0,\n",
       "  652.0,\n",
       "  476.0],\n",
       " 'GO:0044215-other organism': [984.0,\n",
       "  789.0,\n",
       "  1395.0,\n",
       "  771.0,\n",
       "  446.0,\n",
       "  1451.0,\n",
       "  883.0,\n",
       "  600.0,\n",
       "  1554.0,\n",
       "  1564.0,\n",
       "  907.0],\n",
       " 'GO:0044216-other organism cell': [984.0,\n",
       "  789.0,\n",
       "  1395.0,\n",
       "  771.0,\n",
       "  446.0,\n",
       "  1451.0,\n",
       "  883.0,\n",
       "  600.0,\n",
       "  1554.0,\n",
       "  1564.0,\n",
       "  907.0],\n",
       " 'GO:0044217-other organism part': [984.0,\n",
       "  789.0,\n",
       "  1395.0,\n",
       "  771.0,\n",
       "  446.0,\n",
       "  1451.0,\n",
       "  883.0,\n",
       "  600.0,\n",
       "  1554.0,\n",
       "  1564.0,\n",
       "  907.0],\n",
       " 'GO:2001252-positive regulation of chromosome organization': [986.0,\n",
       "  978.0,\n",
       "  680.0,\n",
       "  965.0,\n",
       "  1127.0,\n",
       "  2037.0,\n",
       "  1369.0,\n",
       "  609.0,\n",
       "  2178.0,\n",
       "  668.0,\n",
       "  692.0,\n",
       "  1115.0,\n",
       "  1354.0,\n",
       "  1750.0,\n",
       "  602.0,\n",
       "  1100.0,\n",
       "  1229.0],\n",
       " 'GO:0047485-protein N-terminus binding': [987.0,\n",
       "  1161.0,\n",
       "  1147.0,\n",
       "  1029.0,\n",
       "  1983.0,\n",
       "  1370.0,\n",
       "  1017.0,\n",
       "  1828.0,\n",
       "  519.0,\n",
       "  788.0,\n",
       "  1242.0,\n",
       "  960.0,\n",
       "  1811.0,\n",
       "  790.0,\n",
       "  1495.0,\n",
       "  884.0],\n",
       " 'GO:0042770-signal transduction in response to DNA damage': [988.0,\n",
       "  974.0,\n",
       "  520.0,\n",
       "  729.0,\n",
       "  1219.0,\n",
       "  1147.0,\n",
       "  1381.0,\n",
       "  473.0,\n",
       "  918.0,\n",
       "  1012.0,\n",
       "  2154.0,\n",
       "  647.0,\n",
       "  968.0,\n",
       "  644.0,\n",
       "  1167.0,\n",
       "  1172.0,\n",
       "  1459.0,\n",
       "  779.0,\n",
       "  1040.0,\n",
       "  910.0],\n",
       " 'GO:0006903-vesicle targeting': [989.0,\n",
       "  725.0,\n",
       "  573.0,\n",
       "  1318.0,\n",
       "  961.0,\n",
       "  369.0,\n",
       "  668.0,\n",
       "  724.0,\n",
       "  1186.0,\n",
       "  433.0,\n",
       "  692.0,\n",
       "  615.0,\n",
       "  1055.0,\n",
       "  1104.0,\n",
       "  1101.0,\n",
       "  542.0,\n",
       "  1656.0,\n",
       "  803.0,\n",
       "  419.0],\n",
       " 'GO:0019915-lipid storage': [990.0,\n",
       "  716.0,\n",
       "  974.0,\n",
       "  1023.0,\n",
       "  507.0,\n",
       "  1108.0,\n",
       "  640.0,\n",
       "  1184.0,\n",
       "  1168.0,\n",
       "  940.0],\n",
       " 'GO:1902117-positive regulation of organelle assembly': [991.0,\n",
       "  699.0,\n",
       "  636.0,\n",
       "  1097.0,\n",
       "  1073.0,\n",
       "  944.0,\n",
       "  1782.0,\n",
       "  576.0,\n",
       "  767.0,\n",
       "  910.0,\n",
       "  1461.0,\n",
       "  552.0,\n",
       "  1450.0,\n",
       "  1150.0,\n",
       "  687.0],\n",
       " 'GO:0046686-response to cadmium ion': [992.0,\n",
       "  646.0,\n",
       "  132.0,\n",
       "  955.0,\n",
       "  586.0,\n",
       "  211.0,\n",
       "  937.0,\n",
       "  331.0,\n",
       "  1039.0,\n",
       "  448.0,\n",
       "  447.0,\n",
       "  510.0,\n",
       "  740.0,\n",
       "  383.0,\n",
       "  721.0,\n",
       "  608.0],\n",
       " 'GO:0019730-antimicrobial humoral response': [993.0,\n",
       "  201.0,\n",
       "  106.0,\n",
       "  781.0,\n",
       "  467.0,\n",
       "  682.0,\n",
       "  554.0,\n",
       "  620.0,\n",
       "  166.0,\n",
       "  278.0,\n",
       "  64.0,\n",
       "  268.0,\n",
       "  639.0,\n",
       "  323.0,\n",
       "  395.0,\n",
       "  301.0,\n",
       "  438.0,\n",
       "  375.0,\n",
       "  581.0,\n",
       "  1179.0,\n",
       "  624.0],\n",
       " 'GO:0030660-Golgi-associated vesicle membrane': [994.0,\n",
       "  729.0,\n",
       "  1091.0,\n",
       "  1354.0,\n",
       "  1330.0,\n",
       "  1266.0,\n",
       "  593.0,\n",
       "  1242.0,\n",
       "  533.0,\n",
       "  684.0,\n",
       "  767.0,\n",
       "  1303.0,\n",
       "  1096.0,\n",
       "  1424.0,\n",
       "  812.0,\n",
       "  1515.0,\n",
       "  1251.0],\n",
       " 'GO:0003416-endochondral bone growth': [995.0,\n",
       "  181.0,\n",
       "  594.0,\n",
       "  390.0,\n",
       "  456.0,\n",
       "  825.0,\n",
       "  387.0,\n",
       "  915.0],\n",
       " 'GO:0031625-ubiquitin protein ligase binding': [996.0,\n",
       "  818.0,\n",
       "  438.0,\n",
       "  776.0,\n",
       "  1136.0,\n",
       "  1514.0,\n",
       "  1459.0,\n",
       "  1049.0,\n",
       "  539.0,\n",
       "  904.0,\n",
       "  970.0,\n",
       "  1234.0,\n",
       "  1694.0,\n",
       "  477.0,\n",
       "  937.0,\n",
       "  747.0,\n",
       "  927.0,\n",
       "  988.0,\n",
       "  1145.0,\n",
       "  613.0,\n",
       "  287.0,\n",
       "  1534.0,\n",
       "  1170.0,\n",
       "  990.0],\n",
       " 'GO:0051187-cofactor catabolic process': [997.0,\n",
       "  291.0,\n",
       "  253.0,\n",
       "  1182.0,\n",
       "  289.0,\n",
       "  1229.0,\n",
       "  148.0,\n",
       "  474.0,\n",
       "  747.0,\n",
       "  173.0,\n",
       "  512.0,\n",
       "  231.0,\n",
       "  992.0,\n",
       "  890.0,\n",
       "  1410.0,\n",
       "  406.0],\n",
       " 'GO:0032388-positive regulation of intracellular transport': [998.0,\n",
       "  1035.0,\n",
       "  444.0,\n",
       "  894.0,\n",
       "  1226.0,\n",
       "  1402.0,\n",
       "  1380.0,\n",
       "  842.0,\n",
       "  537.0,\n",
       "  1181.0,\n",
       "  948.0,\n",
       "  1189.0,\n",
       "  1583.0,\n",
       "  517.0,\n",
       "  1011.0,\n",
       "  782.0,\n",
       "  1090.0,\n",
       "  809.0,\n",
       "  1450.0,\n",
       "  761.0,\n",
       "  408.0,\n",
       "  1498.0,\n",
       "  1368.0,\n",
       "  959.0],\n",
       " 'GO:0000792-heterochromatin': [999.0,\n",
       "  428.0,\n",
       "  751.0,\n",
       "  2358.0,\n",
       "  708.0,\n",
       "  1008.0,\n",
       "  1601.0,\n",
       "  655.0],\n",
       " 'GO:0016877-ligase activity, forming carbon-sulfur bonds': 1000.0,\n",
       " ...}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "simulated_rank_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Manually examine enriched pathways\n",
    "Examine the enriched pathways identified from template experiment -- How are these enriched pathways ranked in the simulated experiments?\n",
    "\n",
    "Recall that there were 25 simulated experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GO:0019774-proteasome core complex, beta-subunit complex\n",
      "Template rank:  2.0\n",
      "Median simulated rank:  1.0\n",
      "GO:0000314-organellar small ribosomal subunit\n",
      "Template rank:  3.5\n",
      "Median simulated rank:  238.5\n",
      "GO:0005763-mitochondrial small ribosomal subunit\n",
      "Template rank:  3.5\n",
      "Median simulated rank:  238.5\n",
      "GO:2000628-regulation of miRNA metabolic process\n",
      "Template rank:  9.0\n",
      "Median simulated rank:  17.0\n",
      "GO:0002566-somatic diversification of immune receptors via somatic mutation\n",
      "Template rank:  11.0\n",
      "Median simulated rank:  57.0\n",
      "GO:0016446-somatic hypermutation of immunoglobulin genes\n",
      "Template rank:  13.0\n",
      "Median simulated rank:  69.0\n",
      "GO:0030952-establishment or maintenance of cytoskeleton polarity\n",
      "Template rank:  14.0\n",
      "Median simulated rank:  20.5\n",
      "GO:0030621-U4 snRNA binding\n",
      "Template rank:  15.0\n",
      "Median simulated rank:  35.0\n",
      "GO:0072321-chaperone-mediated protein transport\n",
      "Template rank:  20.0\n",
      "Median simulated rank:  57.0\n",
      "GO:0030950-establishment or maintenance of actin cytoskeleton polarity\n",
      "Template rank:  24.0\n",
      "Median simulated rank:  60.0\n",
      "GO:0005680-anaphase-promoting complex\n",
      "Template rank:  28.0\n",
      "Median simulated rank:  209.0\n",
      "GO:0098644-complex of collagen trimers\n",
      "Template rank:  29.0\n",
      "Median simulated rank:  74.0\n",
      "GO:0090151-establishment of protein localization to mitochondrial membrane\n",
      "Template rank:  32.0\n",
      "Median simulated rank:  33.0\n",
      "GO:0051095-regulation of helicase activity\n",
      "Template rank:  33.0\n",
      "Median simulated rank:  35.0\n",
      "GO:0003746-translation elongation factor activity\n",
      "Template rank:  34.0\n",
      "Median simulated rank:  133.0\n",
      "GO:0017101-aminoacyl-tRNA synthetase multienzyme complex\n",
      "Template rank:  41.0\n",
      "Median simulated rank:  51.0\n",
      "GO:0000940-condensed chromosome outer kinetochore\n",
      "Template rank:  42.0\n",
      "Median simulated rank:  51.0\n",
      "GO:0030983-mismatched DNA binding\n",
      "Template rank:  43.0\n",
      "Median simulated rank:  78.0\n",
      "GO:0070601-centromeric sister chromatid cohesion\n",
      "Template rank:  46.0\n",
      "Median simulated rank:  73.0\n",
      "GO:0036260-RNA capping\n",
      "Template rank:  47.5\n",
      "Median simulated rank:  252.5\n",
      "GO:0009452-7-methylguanosine RNA capping\n",
      "Template rank:  47.5\n",
      "Median simulated rank:  252.5\n",
      "GO:0006370-7-methylguanosine mRNA capping\n",
      "Template rank:  51.0\n",
      "Median simulated rank:  265.0\n",
      "GO:0071459-protein localization to chromosome, centromeric region\n",
      "Template rank:  52.0\n",
      "Median simulated rank:  129.5\n",
      "GO:0016074-snoRNA metabolic process\n",
      "Template rank:  55.0\n",
      "Median simulated rank:  90.0\n",
      "GO:0000177-cytoplasmic exosome (RNase complex)\n",
      "Template rank:  56.0\n",
      "Median simulated rank:  71.0\n",
      "GO:0000212-meiotic spindle organization\n",
      "Template rank:  58.0\n",
      "Median simulated rank:  78.0\n",
      "GO:0034501-protein localization to kinetochore\n",
      "Template rank:  61.0\n",
      "Median simulated rank:  89.5\n",
      "GO:1902236-negative regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway\n",
      "Template rank:  64.0\n",
      "Median simulated rank:  165.0\n",
      "GO:0000778-condensed nuclear chromosome kinetochore\n",
      "Template rank:  65.0\n",
      "Median simulated rank:  71.5\n",
      "GO:0043144-snoRNA processing\n",
      "Template rank:  68.0\n",
      "Median simulated rank:  82.0\n",
      "GO:0015037-peptide disulfide oxidoreductase activity\n",
      "Template rank:  71.0\n",
      "Median simulated rank:  97.0\n",
      "GO:0007094-mitotic spindle assembly checkpoint\n",
      "Template rank:  73.5\n",
      "Median simulated rank:  562.5\n",
      "GO:0071174-mitotic spindle checkpoint\n",
      "Template rank:  73.5\n",
      "Median simulated rank:  562.5\n",
      "GO:0071173-spindle assembly checkpoint\n",
      "Template rank:  73.5\n",
      "Median simulated rank:  562.5\n",
      "GO:0031577-spindle checkpoint\n",
      "Template rank:  73.5\n",
      "Median simulated rank:  562.5\n",
      "GO:0005839-proteasome core complex\n",
      "Template rank:  76.0\n",
      "Median simulated rank:  101.5\n",
      "GO:0030688-preribosome, small subunit precursor\n",
      "Template rank:  80.0\n",
      "Median simulated rank:  87.0\n",
      "GO:0031145-anaphase-promoting complex-dependent catabolic process\n",
      "Template rank:  83.0\n",
      "Median simulated rank:  250.0\n",
      "GO:0045841-negative regulation of mitotic metaphase/anaphase transition\n",
      "Template rank:  87.0\n",
      "Median simulated rank:  481.0\n",
      "GO:0032040-small-subunit processome\n",
      "Template rank:  88.0\n",
      "Median simulated rank:  302.0\n",
      "GO:0033962-cytoplasmic mRNA processing body assembly\n",
      "Template rank:  89.0\n",
      "Median simulated rank:  194.0\n",
      "GO:0000780-condensed nuclear chromosome, centromeric region\n",
      "Template rank:  90.0\n",
      "Median simulated rank:  178.0\n",
      "GO:0004298-threonine-type endopeptidase activity\n",
      "Template rank:  91.5\n",
      "Median simulated rank:  97.5\n",
      "GO:0070003-threonine-type peptidase activity\n",
      "Template rank:  91.5\n",
      "Median simulated rank:  97.5\n",
      "GO:0051156-glucose 6-phosphate metabolic process\n",
      "Template rank:  93.0\n",
      "Median simulated rank:  133.5\n",
      "GO:0000028-ribosomal small subunit assembly\n",
      "Template rank:  94.0\n",
      "Median simulated rank:  113.0\n",
      "GO:0019682-glyceraldehyde-3-phosphate metabolic process\n",
      "Template rank:  97.0\n",
      "Median simulated rank:  116.0\n",
      "GO:0000715-nucleotide-excision repair, DNA damage recognition\n",
      "Template rank:  98.0\n",
      "Median simulated rank:  299.0\n",
      "GO:0015935-small ribosomal subunit\n",
      "Template rank:  100.0\n",
      "Median simulated rank:  280.0\n",
      "GO:1902100-negative regulation of metaphase/anaphase transition of cell cycle\n",
      "Template rank:  105.0\n",
      "Median simulated rank:  417.5\n",
      "GO:0022624-proteasome accessory complex\n",
      "Template rank:  106.0\n",
      "Median simulated rank:  129.5\n",
      "GO:0000339-RNA cap binding\n",
      "Template rank:  110.0\n",
      "Median simulated rank:  124.0\n",
      "GO:0000717-nucleotide-excision repair, DNA duplex unwinding\n",
      "Template rank:  111.0\n",
      "Median simulated rank:  170.0\n",
      "GO:0043022-ribosome binding\n",
      "Template rank:  112.0\n",
      "Median simulated rank:  456.5\n",
      "GO:2000816-negative regulation of mitotic sister chromatid separation\n",
      "Template rank:  114.0\n",
      "Median simulated rank:  365.0\n",
      "GO:0000502-proteasome complex\n",
      "Template rank:  117.0\n",
      "Median simulated rank:  292.0\n",
      "GO:1905369-endopeptidase complex\n",
      "Template rank:  118.0\n",
      "Median simulated rank:  280.5\n",
      "GO:0008180-COP9 signalosome\n",
      "Template rank:  119.0\n",
      "Median simulated rank:  261.5\n",
      "GO:1900118-negative regulation of execution phase of apoptosis\n",
      "Template rank:  120.0\n",
      "Median simulated rank:  117.0\n",
      "GO:1905819-negative regulation of chromosome separation\n",
      "Template rank:  122.0\n",
      "Median simulated rank:  448.0\n",
      "GO:0002183-cytoplasmic translational initiation\n",
      "Template rank:  123.0\n",
      "Median simulated rank:  230.0\n",
      "GO:0050434-positive regulation of viral transcription\n",
      "Template rank:  124.0\n",
      "Median simulated rank:  336.0\n",
      "GO:0005697-telomerase holoenzyme complex\n",
      "Template rank:  125.0\n",
      "Median simulated rank:  218.0\n",
      "GO:0032806-carboxy-terminal domain protein kinase complex\n",
      "Template rank:  130.0\n",
      "Median simulated rank:  251.0\n",
      "GO:0033048-negative regulation of mitotic sister chromatid segregation\n",
      "Template rank:  131.0\n",
      "Median simulated rank:  439.5\n",
      "GO:0007549-dosage compensation\n",
      "Template rank:  133.0\n",
      "Median simulated rank:  260.0\n",
      "GO:0006363-termination of RNA polymerase I transcription\n",
      "Template rank:  134.0\n",
      "Median simulated rank:  422.5\n",
      "GO:0032786-positive regulation of DNA-templated transcription, elongation\n",
      "Template rank:  135.0\n",
      "Median simulated rank:  315.0\n",
      "GO:0033046-negative regulation of sister chromatid segregation\n",
      "Template rank:  136.0\n",
      "Median simulated rank:  505.0\n",
      "GO:0008137-NADH dehydrogenase (ubiquinone) activity\n",
      "Template rank:  140.0\n",
      "Median simulated rank:  177.0\n",
      "GO:0050136-NADH dehydrogenase (quinone) activity\n",
      "Template rank:  140.0\n",
      "Median simulated rank:  177.0\n",
      "GO:0003954-NADH dehydrogenase activity\n",
      "Template rank:  140.0\n",
      "Median simulated rank:  177.0\n",
      "GO:0006294-nucleotide-excision repair, preincision complex assembly\n",
      "Template rank:  142.0\n",
      "Median simulated rank:  199.0\n",
      "GO:0000462-maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\n",
      "Template rank:  144.0\n",
      "Median simulated rank:  343.0\n",
      "GO:1903421-regulation of synaptic vesicle recycling\n",
      "Template rank:  146.0\n",
      "Median simulated rank:  215.0\n",
      "GO:0010499-proteasomal ubiquitin-independent protein catabolic process\n",
      "Template rank:  150.0\n",
      "Median simulated rank:  157.0\n",
      "GO:1902235-regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway\n",
      "Template rank:  151.0\n",
      "Median simulated rank:  339.0\n",
      "GO:0051985-negative regulation of chromosome segregation\n",
      "Template rank:  152.0\n",
      "Median simulated rank:  612.0\n",
      "GO:0071005-U2-type precatalytic spliceosome\n",
      "Template rank:  153.0\n",
      "Median simulated rank:  184.0\n",
      "GO:0061621-canonical glycolysis\n",
      "Template rank:  155.0\n",
      "Median simulated rank:  151.5\n",
      "GO:0006735-NADH regeneration\n",
      "Template rank:  155.0\n",
      "Median simulated rank:  151.5\n",
      "GO:0061718-glucose catabolic process to pyruvate\n",
      "Template rank:  155.0\n",
      "Median simulated rank:  151.5\n",
      "GO:0070911-global genome nucleotide-excision repair\n",
      "Template rank:  157.0\n",
      "Median simulated rank:  231.0\n",
      "GO:0070125-mitochondrial translational elongation\n",
      "Template rank:  158.0\n",
      "Median simulated rank:  281.5\n",
      "GO:0034661-ncRNA catabolic process\n",
      "Template rank:  159.0\n",
      "Median simulated rank:  420.0\n",
      "GO:0008608-attachment of spindle microtubules to kinetochore\n",
      "Template rank:  160.0\n",
      "Median simulated rank:  295.0\n",
      "GO:0061620-glycolytic process through glucose-6-phosphate\n",
      "Template rank:  163.5\n",
      "Median simulated rank:  161.5\n",
      "GO:0061615-glycolytic process through fructose-6-phosphate\n",
      "Template rank:  163.5\n",
      "Median simulated rank:  161.5\n",
      "GO:0045271-respiratory chain complex I\n",
      "Template rank:  166.0\n",
      "Median simulated rank:  173.5\n",
      "GO:0005747-mitochondrial respiratory chain complex I\n",
      "Template rank:  166.0\n",
      "Median simulated rank:  173.5\n",
      "GO:0030964-NADH dehydrogenase complex\n",
      "Template rank:  166.0\n",
      "Median simulated rank:  173.5\n",
      "GO:0006361-transcription initiation from RNA polymerase I promoter\n",
      "Template rank:  168.0\n",
      "Median simulated rank:  254.0\n",
      "GO:0051443-positive regulation of ubiquitin-protein transferase activity\n",
      "Template rank:  169.0\n",
      "Median simulated rank:  421.0\n",
      "GO:0030071-regulation of mitotic metaphase/anaphase transition\n",
      "Template rank:  170.0\n",
      "Median simulated rank:  466.0\n",
      "GO:1904666-regulation of ubiquitin protein ligase activity\n",
      "Template rank:  171.0\n",
      "Median simulated rank:  401.0\n",
      "GO:0070979-protein K11-linked ubiquitination\n",
      "Template rank:  173.0\n",
      "Median simulated rank:  394.0\n",
      "GO:0043631-RNA polyadenylation\n",
      "Template rank:  174.0\n",
      "Median simulated rank:  552.0\n",
      "GO:0009299-mRNA transcription\n",
      "Template rank:  175.0\n",
      "Median simulated rank:  456.0\n",
      "GO:0071011-precatalytic spliceosome\n",
      "Template rank:  176.0\n",
      "Median simulated rank:  183.0\n",
      "GO:0006414-translational elongation\n",
      "Template rank:  177.0\n",
      "Median simulated rank:  378.5\n",
      "GO:0044769-ATPase activity, coupled to transmembrane movement of ions, rotational mechanism\n",
      "Template rank:  178.0\n",
      "Median simulated rank:  326.5\n",
      "GO:0006356-regulation of transcription by RNA polymerase I\n",
      "Template rank:  180.0\n",
      "Median simulated rank:  293.0\n",
      "GO:0006362-transcription elongation from RNA polymerase I promoter\n",
      "Template rank:  181.0\n",
      "Median simulated rank:  416.0\n",
      "GO:0045839-negative regulation of mitotic nuclear division\n",
      "Template rank:  182.0\n",
      "Median simulated rank:  544.5\n",
      "GO:0032981-mitochondrial respiratory chain complex I assembly\n",
      "Template rank:  184.5\n",
      "Median simulated rank:  272.0\n",
      "GO:0010257-NADH dehydrogenase complex assembly\n",
      "Template rank:  184.5\n",
      "Median simulated rank:  272.0\n",
      "GO:0003743-translation initiation factor activity\n",
      "Template rank:  186.0\n",
      "Median simulated rank:  437.0\n",
      "GO:0008135-translation factor activity, RNA binding\n",
      "Template rank:  189.0\n",
      "Median simulated rank:  454.0\n",
      "GO:0042558-pteridine-containing compound metabolic process\n",
      "Template rank:  190.0\n",
      "Median simulated rank:  191.5\n",
      "GO:0006378-mRNA polyadenylation\n",
      "Template rank:  191.0\n",
      "Median simulated rank:  513.0\n",
      "GO:1903573-negative regulation of response to endoplasmic reticulum stress\n",
      "Template rank:  192.0\n",
      "Median simulated rank:  515.0\n",
      "GO:0051123-RNA polymerase II preinitiation complex assembly\n",
      "Template rank:  193.0\n",
      "Median simulated rank:  320.0\n",
      "GO:0015949-nucleobase-containing small molecule interconversion\n",
      "Template rank:  194.0\n",
      "Median simulated rank:  357.5\n",
      "GO:0071007-U2-type catalytic step 2 spliceosome\n",
      "Template rank:  195.0\n",
      "Median simulated rank:  311.5\n",
      "GO:0006379-mRNA cleavage\n",
      "Template rank:  196.0\n",
      "Median simulated rank:  216.5\n",
      "GO:1905368-peptidase complex\n",
      "Template rank:  198.0\n",
      "Median simulated rank:  318.0\n",
      "GO:0000281-mitotic cytokinesis\n",
      "Template rank:  199.0\n",
      "Median simulated rank:  840.5\n",
      "GO:0000313-organellar ribosome\n",
      "Template rank:  200.5\n",
      "Median simulated rank:  333.5\n",
      "GO:0005761-mitochondrial ribosome\n",
      "Template rank:  200.5\n",
      "Median simulated rank:  333.5\n",
      "GO:1902750-negative regulation of cell cycle G2/M phase transition\n",
      "Template rank:  204.0\n",
      "Median simulated rank:  527.0\n",
      "GO:1900101-regulation of endoplasmic reticulum unfolded protein response\n",
      "Template rank:  205.0\n",
      "Median simulated rank:  544.5\n",
      "GO:1902099-regulation of metaphase/anaphase transition of cell cycle\n",
      "Template rank:  206.0\n",
      "Median simulated rank:  529.5\n",
      "GO:0046782-regulation of viral transcription\n",
      "Template rank:  207.0\n",
      "Median simulated rank:  674.0\n",
      "GO:0035666-TRIF-dependent toll-like receptor signaling pathway\n",
      "Template rank:  208.0\n",
      "Median simulated rank:  549.0\n",
      "GO:0006101-citrate metabolic process\n",
      "Template rank:  209.0\n",
      "Median simulated rank:  349.0\n",
      "GO:0006521-regulation of cellular amino acid metabolic process\n",
      "Template rank:  211.0\n",
      "Median simulated rank:  263.0\n",
      "GO:1905897-regulation of response to endoplasmic reticulum stress\n",
      "Template rank:  212.0\n",
      "Median simulated rank:  733.0\n",
      "GO:0016860-intramolecular oxidoreductase activity\n",
      "Template rank:  213.0\n",
      "Median simulated rank:  349.5\n",
      "GO:0015030-Cajal body\n",
      "Template rank:  214.0\n",
      "Median simulated rank:  605.5\n",
      "GO:0042788-polysomal ribosome\n",
      "Template rank:  215.0\n",
      "Median simulated rank:  244.0\n",
      "GO:0007091-metaphase/anaphase transition of mitotic cell cycle\n",
      "Template rank:  217.0\n",
      "Median simulated rank:  530.5\n",
      "GO:0043021-ribonucleoprotein complex binding\n",
      "Template rank:  218.0\n",
      "Median simulated rank:  662.0\n",
      "GO:0006360-transcription by RNA polymerase I\n",
      "Template rank:  219.0\n",
      "Median simulated rank:  638.0\n",
      "GO:0033144-negative regulation of intracellular steroid hormone receptor signaling pathway\n",
      "Template rank:  220.0\n",
      "Median simulated rank:  379.0\n",
      "GO:0010972-negative regulation of G2/M transition of mitotic cell cycle\n",
      "Template rank:  222.0\n",
      "Median simulated rank:  554.0\n",
      "GO:0031440-regulation of mRNA 3'-end processing\n",
      "Template rank:  223.0\n",
      "Median simulated rank:  223.5\n",
      "GO:0044665-MLL1/2 complex\n",
      "Template rank:  224.5\n",
      "Median simulated rank:  307.5\n",
      "GO:0071339-MLL1 complex\n",
      "Template rank:  224.5\n",
      "Median simulated rank:  307.5\n",
      "GO:0036498-IRE1-mediated unfolded protein response\n",
      "Template rank:  226.0\n",
      "Median simulated rank:  544.0\n",
      "GO:0051438-regulation of ubiquitin-protein transferase activity\n",
      "Template rank:  228.0\n",
      "Median simulated rank:  860.0\n",
      "GO:0061157-mRNA destabilization\n",
      "Template rank:  230.0\n",
      "Median simulated rank:  347.0\n",
      "GO:0016655-oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor\n",
      "Template rank:  231.0\n",
      "Median simulated rank:  228.5\n",
      "GO:1904292-regulation of ERAD pathway\n",
      "Template rank:  232.0\n",
      "Median simulated rank:  255.0\n",
      "GO:0031124-mRNA 3'-end processing\n",
      "Template rank:  233.0\n",
      "Median simulated rank:  1098.5\n",
      "GO:0010965-regulation of mitotic sister chromatid separation\n",
      "Template rank:  234.0\n",
      "Median simulated rank:  532.0\n",
      "GO:0070126-mitochondrial translational termination\n",
      "Template rank:  235.0\n",
      "Median simulated rank:  283.5\n",
      "GO:0030684-preribosome\n",
      "Template rank:  236.0\n",
      "Median simulated rank:  705.0\n",
      "GO:0051306-mitotic sister chromatid separation\n",
      "Template rank:  239.0\n",
      "Median simulated rank:  522.0\n",
      "GO:0006099-tricarboxylic acid cycle\n",
      "Template rank:  240.0\n",
      "Median simulated rank:  370.0\n",
      "GO:0030866-cortical actin cytoskeleton organization\n",
      "Template rank:  241.0\n",
      "Median simulated rank:  871.0\n",
      "GO:0051784-negative regulation of nuclear division\n",
      "Template rank:  242.0\n",
      "Median simulated rank:  678.0\n",
      "GO:0016242-negative regulation of macroautophagy\n",
      "Template rank:  243.0\n",
      "Median simulated rank:  280.0\n",
      "GO:0048027-mRNA 5'-UTR binding\n",
      "Template rank:  245.0\n",
      "Median simulated rank:  763.0\n",
      "GO:0043921-modulation by host of viral transcription\n",
      "Template rank:  248.5\n",
      "Median simulated rank:  303.5\n",
      "GO:0052472-modulation by host of symbiont transcription\n",
      "Template rank:  248.5\n",
      "Median simulated rank:  303.5\n",
      "GO:0031369-translation initiation factor binding\n",
      "Template rank:  250.0\n",
      "Median simulated rank:  279.0\n",
      "GO:0061418-regulation of transcription from RNA polymerase II promoter in response to hypoxia\n",
      "Template rank:  251.0\n",
      "Median simulated rank:  414.0\n",
      "GO:0051973-positive regulation of telomerase activity\n",
      "Template rank:  252.0\n",
      "Median simulated rank:  256.0\n",
      "GO:0042255-ribosome assembly\n",
      "Template rank:  256.0\n",
      "Median simulated rank:  438.0\n",
      "GO:0072350-tricarboxylic acid metabolic process\n",
      "Template rank:  257.0\n",
      "Median simulated rank:  385.0\n",
      "GO:0044784-metaphase/anaphase transition of cell cycle\n",
      "Template rank:  258.0\n",
      "Median simulated rank:  527.0\n",
      "GO:0051019-mitogen-activated protein kinase binding\n",
      "Template rank:  259.0\n",
      "Median simulated rank:  261.5\n",
      "GO:0060765-regulation of androgen receptor signaling pathway\n",
      "Template rank:  260.0\n",
      "Median simulated rank:  391.0\n",
      "GO:0034243-regulation of transcription elongation from RNA polymerase II promoter\n",
      "Template rank:  264.0\n",
      "Median simulated rank:  284.0\n",
      "GO:1902036-regulation of hematopoietic stem cell differentiation\n",
      "Template rank:  265.0\n",
      "Median simulated rank:  438.5\n",
      "GO:0002479-antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent\n",
      "Template rank:  266.0\n",
      "Median simulated rank:  281.0\n",
      "GO:0033108-mitochondrial respiratory chain complex assembly\n",
      "Template rank:  267.0\n",
      "Median simulated rank:  494.0\n",
      "GO:0006415-translational termination\n",
      "Template rank:  270.0\n",
      "Median simulated rank:  332.0\n",
      "GO:0031572-G2 DNA damage checkpoint\n",
      "Template rank:  271.0\n",
      "Median simulated rank:  371.0\n",
      "GO:0010837-regulation of keratinocyte proliferation\n",
      "Template rank:  272.0\n",
      "Median simulated rank:  509.0\n",
      "GO:1901068-guanosine-containing compound metabolic process\n",
      "Template rank:  273.0\n",
      "Median simulated rank:  1021.0\n",
      "GO:0030865-cortical cytoskeleton organization\n",
      "Template rank:  274.0\n",
      "Median simulated rank:  1077.5\n",
      "GO:0032212-positive regulation of telomere maintenance via telomerase\n",
      "Template rank:  276.0\n",
      "Median simulated rank:  319.0\n",
      "GO:0000154-rRNA modification\n",
      "Template rank:  277.0\n",
      "Median simulated rank:  755.0\n",
      "GO:2000785-regulation of autophagosome assembly\n",
      "Template rank:  278.0\n",
      "Median simulated rank:  543.5\n",
      "GO:0042274-ribosomal small subunit biogenesis\n",
      "Template rank:  279.0\n",
      "Median simulated rank:  593.0\n",
      "GO:0033047-regulation of mitotic sister chromatid segregation\n",
      "Template rank:  281.0\n",
      "Median simulated rank:  568.0\n",
      "GO:1905818-regulation of chromosome separation\n",
      "Template rank:  282.0\n",
      "Median simulated rank:  528.5\n",
      "GO:0042795-snRNA transcription by RNA polymerase II\n",
      "Template rank:  283.0\n",
      "Median simulated rank:  1032.0\n",
      "GO:0019205-nucleobase-containing compound kinase activity\n",
      "Template rank:  284.0\n",
      "Median simulated rank:  1523.0\n",
      "GO:0050779-RNA destabilization\n",
      "Template rank:  285.0\n",
      "Median simulated rank:  452.5\n",
      "GO:0019083-viral transcription\n",
      "Template rank:  286.0\n",
      "Median simulated rank:  505.0\n",
      "GO:0034502-protein localization to chromosome\n",
      "Template rank:  287.0\n",
      "Median simulated rank:  525.5\n",
      "GO:0070059-intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress\n",
      "Template rank:  290.0\n",
      "Median simulated rank:  1086.5\n",
      "GO:0061631-ubiquitin conjugating enzyme activity\n",
      "Template rank:  291.0\n",
      "Median simulated rank:  308.0\n",
      "GO:1905898-positive regulation of response to endoplasmic reticulum stress\n",
      "Template rank:  294.0\n",
      "Median simulated rank:  667.0\n",
      "GO:0019843-rRNA binding\n",
      "Template rank:  296.0\n",
      "Median simulated rank:  583.0\n",
      "GO:0006890-retrograde vesicle-mediated transport, Golgi to ER\n",
      "Template rank:  299.0\n",
      "Median simulated rank:  712.0\n",
      "GO:0032784-regulation of DNA-templated transcription, elongation\n",
      "Template rank:  300.0\n",
      "Median simulated rank:  504.0\n",
      "GO:0006296-nucleotide-excision repair, DNA incision, 5'-to lesion\n",
      "Template rank:  301.0\n",
      "Median simulated rank:  422.0\n",
      "GO:0030490-maturation of SSU-rRNA\n",
      "Template rank:  303.0\n",
      "Median simulated rank:  613.0\n",
      "GO:0005876-spindle microtubule\n",
      "Template rank:  304.0\n",
      "Median simulated rank:  630.0\n",
      "GO:0009301-snRNA transcription\n",
      "Template rank:  305.0\n",
      "Median simulated rank:  1026.0\n",
      "GO:0061014-positive regulation of mRNA catabolic process\n",
      "Template rank:  306.0\n",
      "Median simulated rank:  962.0\n",
      "GO:0044088-regulation of vacuole organization\n",
      "Template rank:  307.0\n",
      "Median simulated rank:  578.0\n",
      "GO:0031060-regulation of histone methylation\n",
      "Template rank:  308.0\n",
      "Median simulated rank:  621.0\n",
      "GO:0006353-DNA-templated transcription, termination\n",
      "Template rank:  309.0\n",
      "Median simulated rank:  739.5\n",
      "GO:1901532-regulation of hematopoietic progenitor cell differentiation\n",
      "Template rank:  310.0\n",
      "Median simulated rank:  678.5\n",
      "GO:0046966-thyroid hormone receptor binding\n",
      "Template rank:  311.0\n",
      "Median simulated rank:  307.0\n",
      "GO:0006446-regulation of translational initiation\n",
      "Template rank:  312.0\n",
      "Median simulated rank:  654.0\n",
      "GO:0002181-cytoplasmic translation\n",
      "Template rank:  315.0\n",
      "Median simulated rank:  423.0\n",
      "GO:0005684-U2-type spliceosomal complex\n",
      "Template rank:  319.0\n",
      "Median simulated rank:  375.0\n",
      "GO:0019080-viral gene expression\n",
      "Template rank:  320.0\n",
      "Median simulated rank:  506.0\n",
      "GO:0006413-translational initiation\n",
      "Template rank:  321.0\n",
      "Median simulated rank:  470.0\n",
      "GO:0007052-mitotic spindle organization\n",
      "Template rank:  322.0\n",
      "Median simulated rank:  985.0\n",
      "GO:0031062-positive regulation of histone methylation\n",
      "Template rank:  324.0\n",
      "Median simulated rank:  434.0\n",
      "GO:0017069-snRNA binding\n",
      "Template rank:  325.0\n",
      "Median simulated rank:  439.0\n",
      "GO:0005790-smooth endoplasmic reticulum\n",
      "Template rank:  326.0\n",
      "Median simulated rank:  423.0\n",
      "GO:0033045-regulation of sister chromatid segregation\n",
      "Template rank:  327.0\n",
      "Median simulated rank:  692.5\n",
      "GO:0061640-cytoskeleton-dependent cytokinesis\n",
      "Template rank:  329.0\n",
      "Median simulated rank:  1086.5\n",
      "GO:0032508-DNA duplex unwinding\n",
      "Template rank:  331.0\n",
      "Median simulated rank:  794.0\n",
      "GO:0032506-cytokinetic process\n",
      "Template rank:  332.0\n",
      "Median simulated rank:  535.0\n",
      "GO:0090307-mitotic spindle assembly\n",
      "Template rank:  333.0\n",
      "Median simulated rank:  773.5\n",
      "GO:1990841-promoter-specific chromatin binding\n",
      "Template rank:  334.0\n",
      "Median simulated rank:  1189.0\n",
      "GO:0140053-mitochondrial gene expression\n",
      "Template rank:  335.0\n",
      "Median simulated rank:  585.0\n",
      "GO:0032206-positive regulation of telomere maintenance\n",
      "Template rank:  336.0\n",
      "Median simulated rank:  463.0\n",
      "GO:0070498-interleukin-1-mediated signaling pathway\n",
      "Template rank:  340.0\n",
      "Median simulated rank:  540.0\n",
      "GO:0000387-spliceosomal snRNP assembly\n",
      "Template rank:  341.0\n",
      "Median simulated rank:  375.0\n",
      "GO:1904358-positive regulation of telomere maintenance via telomere lengthening\n",
      "Template rank:  345.0\n",
      "Median simulated rank:  387.0\n",
      "GO:0033146-regulation of intracellular estrogen receptor signaling pathway\n",
      "Template rank:  346.0\n",
      "Median simulated rank:  596.0\n",
      "GO:0043130-ubiquitin binding\n",
      "Template rank:  349.0\n",
      "Median simulated rank:  672.5\n",
      "GO:2001243-negative regulation of intrinsic apoptotic signaling pathway\n",
      "Template rank:  351.0\n",
      "Median simulated rank:  608.0\n",
      "GO:1904813-ficolin-1-rich granule lumen\n",
      "Template rank:  352.0\n",
      "Median simulated rank:  449.0\n",
      "GO:0000152-nuclear ubiquitin ligase complex\n",
      "Template rank:  353.0\n",
      "Median simulated rank:  396.0\n",
      "GO:0003678-DNA helicase activity\n",
      "Template rank:  354.0\n",
      "Median simulated rank:  1046.0\n",
      "GO:1902850-microtubule cytoskeleton organization involved in mitosis\n",
      "Template rank:  355.0\n",
      "Median simulated rank:  1219.0\n",
      "GO:0008408-3'-5' exonuclease activity\n",
      "Template rank:  356.0\n",
      "Median simulated rank:  1257.5\n",
      "GO:0070897-transcription preinitiation complex assembly\n",
      "Template rank:  357.0\n",
      "Median simulated rank:  568.5\n",
      "GO:0015036-disulfide oxidoreductase activity\n",
      "Template rank:  358.0\n",
      "Median simulated rank:  462.0\n",
      "GO:0009060-aerobic respiration\n",
      "Template rank:  359.0\n",
      "Median simulated rank:  411.5\n",
      "GO:0002474-antigen processing and presentation of peptide antigen via MHC class I\n",
      "Template rank:  362.0\n",
      "Median simulated rank:  392.0\n",
      "GO:0098781-ncRNA transcription\n",
      "Template rank:  364.0\n",
      "Median simulated rank:  1005.0\n",
      "GO:0051972-regulation of telomerase activity\n",
      "Template rank:  365.0\n",
      "Median simulated rank:  593.0\n",
      "GO:0016591-RNA polymerase II, holoenzyme\n",
      "Template rank:  367.0\n",
      "Median simulated rank:  753.0\n",
      "GO:0006354-DNA-templated transcription, elongation\n",
      "Template rank:  368.0\n",
      "Median simulated rank:  788.0\n",
      "GO:0007080-mitotic metaphase plate congression\n",
      "Template rank:  371.0\n",
      "Median simulated rank:  532.0\n",
      "GO:0060218-hematopoietic stem cell differentiation\n",
      "Template rank:  372.0\n",
      "Median simulated rank:  426.5\n",
      "GO:0032543-mitochondrial translation\n",
      "Template rank:  373.0\n",
      "Median simulated rank:  450.5\n",
      "GO:0043039-tRNA aminoacylation\n",
      "Template rank:  375.0\n",
      "Median simulated rank:  402.0\n",
      "GO:0033683-nucleotide-excision repair, DNA incision\n",
      "Template rank:  376.0\n",
      "Median simulated rank:  433.0\n",
      "GO:0016796-exonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5'-phosphomonoesters\n",
      "Template rank:  378.0\n",
      "Median simulated rank:  1412.0\n",
      "GO:0032392-DNA geometric change\n",
      "Template rank:  379.0\n",
      "Median simulated rank:  793.5\n",
      "GO:0035722-interleukin-12-mediated signaling pathway\n",
      "Template rank:  380.0\n",
      "Median simulated rank:  522.0\n",
      "GO:0000184-nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "Template rank:  382.0\n",
      "Median simulated rank:  487.0\n",
      "GO:0006284-base-excision repair\n",
      "Template rank:  383.0\n",
      "Median simulated rank:  390.0\n",
      "GO:0000049-tRNA binding\n",
      "Template rank:  384.0\n",
      "Median simulated rank:  1201.0\n",
      "GO:0032182-ubiquitin-like protein binding\n",
      "Template rank:  385.0\n",
      "Median simulated rank:  844.0\n",
      "GO:0031123-RNA 3'-end processing\n",
      "Template rank:  386.0\n",
      "Median simulated rank:  1237.0\n",
      "GO:0033143-regulation of intracellular steroid hormone receptor signaling pathway\n",
      "Template rank:  387.0\n",
      "Median simulated rank:  1055.0\n",
      "GO:0006406-mRNA export from nucleus\n",
      "Template rank:  389.5\n",
      "Median simulated rank:  962.5\n",
      "GO:0071427-mRNA-containing ribonucleoprotein complex export from nucleus\n",
      "Template rank:  389.5\n",
      "Median simulated rank:  962.5\n",
      "GO:0098798-mitochondrial protein complex\n",
      "Template rank:  391.0\n",
      "Median simulated rank:  415.0\n",
      "GO:0051983-regulation of chromosome segregation\n",
      "Template rank:  392.0\n",
      "Median simulated rank:  733.0\n",
      "GO:0005844-polysome\n",
      "Template rank:  393.0\n",
      "Median simulated rank:  537.0\n",
      "GO:0003899-DNA-directed 5'-3' RNA polymerase activity\n",
      "Template rank:  395.0\n",
      "Median simulated rank:  470.5\n",
      "GO:0006893-Golgi to plasma membrane transport\n",
      "Template rank:  396.0\n",
      "Median simulated rank:  1718.0\n",
      "GO:0006368-transcription elongation from RNA polymerase II promoter\n",
      "Template rank:  397.0\n",
      "Median simulated rank:  659.0\n",
      "GO:0000959-mitochondrial RNA metabolic process\n",
      "Template rank:  398.0\n",
      "Median simulated rank:  677.0\n",
      "GO:0043038-amino acid activation\n",
      "Template rank:  399.0\n",
      "Median simulated rank:  469.0\n",
      "GO:0050681-androgen receptor binding\n",
      "Template rank:  400.0\n",
      "Median simulated rank:  656.0\n",
      "GO:0071166-ribonucleoprotein complex localization\n",
      "Template rank:  402.0\n",
      "Median simulated rank:  759.0\n",
      "GO:0043618-regulation of transcription from RNA polymerase II promoter in response to stress\n",
      "Template rank:  403.0\n",
      "Median simulated rank:  592.5\n",
      "GO:0030521-androgen receptor signaling pathway\n",
      "Template rank:  404.0\n",
      "Median simulated rank:  662.0\n",
      "GO:1901028-regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway\n",
      "Template rank:  407.0\n",
      "Median simulated rank:  457.0\n",
      "GO:1901991-negative regulation of mitotic cell cycle phase transition\n",
      "Template rank:  408.0\n",
      "Median simulated rank:  804.0\n",
      "GO:0030968-endoplasmic reticulum unfolded protein response\n",
      "Template rank:  409.0\n",
      "Median simulated rank:  774.0\n",
      "GO:1901988-negative regulation of cell cycle phase transition\n",
      "Template rank:  410.0\n",
      "Median simulated rank:  797.0\n",
      "GO:0048524-positive regulation of viral process\n",
      "Template rank:  412.0\n",
      "Median simulated rank:  973.0\n",
      "GO:0071426-ribonucleoprotein complex export from nucleus\n",
      "Template rank:  413.0\n",
      "Median simulated rank:  769.0\n",
      "GO:0006402-mRNA catabolic process\n",
      "Template rank:  415.0\n",
      "Median simulated rank:  802.5\n",
      "GO:1903146-regulation of autophagy of mitochondrion\n",
      "Template rank:  416.0\n",
      "Median simulated rank:  512.5\n",
      "GO:0032210-regulation of telomere maintenance via telomerase\n",
      "Template rank:  417.0\n",
      "Median simulated rank:  648.0\n",
      "GO:0043488-regulation of mRNA stability\n",
      "Template rank:  419.0\n",
      "Median simulated rank:  764.0\n",
      "GO:0043620-regulation of DNA-templated transcription in response to stress\n",
      "Template rank:  420.0\n",
      "Median simulated rank:  589.0\n",
      "GO:0000070-mitotic sister chromatid segregation\n",
      "Template rank:  421.0\n",
      "Median simulated rank:  809.0\n",
      "GO:0031593-polyubiquitin modification-dependent protein binding\n",
      "Template rank:  423.0\n",
      "Median simulated rank:  494.5\n",
      "GO:1901799-negative regulation of proteasomal protein catabolic process\n",
      "Template rank:  425.0\n",
      "Median simulated rank:  812.0\n",
      "GO:0003684-damaged DNA binding\n",
      "Template rank:  426.0\n",
      "Median simulated rank:  608.0\n",
      "GO:0043487-regulation of RNA stability\n",
      "Template rank:  427.0\n",
      "Median simulated rank:  767.5\n",
      "GO:0016571-histone methylation\n",
      "Template rank:  428.0\n",
      "Median simulated rank:  1821.0\n",
      "GO:1902808-positive regulation of cell cycle G1/S phase transition\n",
      "Template rank:  429.0\n",
      "Median simulated rank:  810.0\n",
      "GO:0034620-cellular response to unfolded protein\n",
      "Template rank:  430.0\n",
      "Median simulated rank:  842.5\n",
      "GO:0009069-serine family amino acid metabolic process\n",
      "Template rank:  431.0\n",
      "Median simulated rank:  520.0\n",
      "GO:0043489-RNA stabilization\n",
      "Template rank:  433.0\n",
      "Median simulated rank:  606.0\n",
      "GO:0070849-response to epidermal growth factor\n",
      "Template rank:  434.0\n",
      "Median simulated rank:  511.0\n",
      "GO:0051304-chromosome separation\n",
      "Template rank:  435.0\n",
      "Median simulated rank:  772.5\n",
      "GO:0006405-RNA export from nucleus\n",
      "Template rank:  436.0\n",
      "Median simulated rank:  854.5\n",
      "GO:0000776-kinetochore\n",
      "Template rank:  437.0\n",
      "Median simulated rank:  922.5\n",
      "GO:1902369-negative regulation of RNA catabolic process\n",
      "Template rank:  438.0\n",
      "Median simulated rank:  789.0\n",
      "GO:0071349-cellular response to interleukin-12\n",
      "Template rank:  439.0\n",
      "Median simulated rank:  610.0\n",
      "GO:0002478-antigen processing and presentation of exogenous peptide antigen\n",
      "Template rank:  441.0\n",
      "Median simulated rank:  479.0\n",
      "GO:0000779-condensed chromosome, centromeric region\n",
      "Template rank:  442.0\n",
      "Median simulated rank:  792.5\n",
      "GO:1902749-regulation of cell cycle G2/M phase transition\n",
      "Template rank:  443.0\n",
      "Median simulated rank:  866.0\n",
      "GO:0007004-telomere maintenance via telomerase\n",
      "Template rank:  445.0\n",
      "Median simulated rank:  872.0\n",
      "GO:0044839-cell cycle G2/M phase transition\n",
      "Template rank:  446.0\n",
      "Median simulated rank:  1045.5\n",
      "GO:0070671-response to interleukin-12\n",
      "Template rank:  449.0\n",
      "Median simulated rank:  647.0\n",
      "GO:2000573-positive regulation of DNA biosynthetic process\n",
      "Template rank:  451.0\n",
      "Median simulated rank:  549.0\n",
      "GO:0061013-regulation of mRNA catabolic process\n",
      "Template rank:  453.0\n",
      "Median simulated rank:  909.0\n",
      "GO:2000059-negative regulation of ubiquitin-dependent protein catabolic process\n",
      "Template rank:  454.0\n",
      "Median simulated rank:  751.0\n",
      "GO:0034062-5'-3' RNA polymerase activity\n",
      "Template rank:  455.5\n",
      "Median simulated rank:  563.0\n",
      "GO:0097747-RNA polymerase activity\n",
      "Template rank:  455.5\n",
      "Median simulated rank:  563.0\n",
      "GO:0008023-transcription elongation factor complex\n",
      "Template rank:  457.0\n",
      "Median simulated rank:  1375.5\n",
      "GO:0034399-nuclear periphery\n",
      "Template rank:  458.0\n",
      "Median simulated rank:  1199.0\n",
      "GO:0071364-cellular response to epidermal growth factor stimulus\n",
      "Template rank:  459.0\n",
      "Median simulated rank:  550.0\n",
      "GO:0000086-G2/M transition of mitotic cell cycle\n",
      "Template rank:  460.0\n",
      "Median simulated rank:  1095.0\n",
      "GO:0010389-regulation of G2/M transition of mitotic cell cycle\n",
      "Template rank:  461.0\n",
      "Median simulated rank:  870.5\n",
      "GO:0051082-unfolded protein binding\n",
      "Template rank:  462.0\n",
      "Median simulated rank:  862.0\n",
      "GO:0000956-nuclear-transcribed mRNA catabolic process\n",
      "Template rank:  465.0\n",
      "Median simulated rank:  897.0\n",
      "GO:0006401-RNA catabolic process\n",
      "Template rank:  466.0\n",
      "Median simulated rank:  775.0\n",
      "GO:0005776-autophagosome\n",
      "Template rank:  467.0\n",
      "Median simulated rank:  1674.0\n",
      "GO:0009163-nucleoside biosynthetic process\n",
      "Template rank:  468.0\n",
      "Median simulated rank:  902.5\n",
      "GO:0031100-animal organ regeneration\n",
      "Template rank:  469.0\n",
      "Median simulated rank:  566.5\n",
      "GO:0007062-sister chromatid cohesion\n",
      "Template rank:  470.0\n",
      "Median simulated rank:  605.0\n",
      "GO:1903363-negative regulation of cellular protein catabolic process\n",
      "Template rank:  472.0\n",
      "Median simulated rank:  927.0\n",
      "GO:1903202-negative regulation of oxidative stress-induced cell death\n",
      "Template rank:  473.0\n",
      "Median simulated rank:  981.0\n",
      "GO:0042254-ribosome biogenesis\n",
      "Template rank:  474.0\n",
      "Median simulated rank:  938.0\n",
      "GO:0048255-mRNA stabilization\n",
      "Template rank:  476.0\n",
      "Median simulated rank:  496.0\n",
      "GO:0035967-cellular response to topologically incorrect protein\n",
      "Template rank:  477.0\n",
      "Median simulated rank:  900.0\n",
      "GO:0031970-organelle envelope lumen\n",
      "Template rank:  478.0\n",
      "Median simulated rank:  617.0\n",
      "GO:0006283-transcription-coupled nucleotide-excision repair\n",
      "Template rank:  479.0\n",
      "Median simulated rank:  622.0\n",
      "GO:0045296-cadherin binding\n",
      "Template rank:  480.0\n",
      "Median simulated rank:  833.0\n",
      "GO:0006457-protein folding\n",
      "Template rank:  482.0\n",
      "Median simulated rank:  1072.0\n",
      "GO:0097345-mitochondrial outer membrane permeabilization\n",
      "Template rank:  484.0\n",
      "Median simulated rank:  735.0\n",
      "GO:0006364-rRNA processing\n",
      "Template rank:  485.0\n",
      "Median simulated rank:  1062.0\n",
      "GO:0019884-antigen processing and presentation of exogenous antigen\n",
      "Template rank:  486.0\n",
      "Median simulated rank:  579.0\n",
      "GO:0003697-single-stranded DNA binding\n",
      "Template rank:  487.0\n",
      "Median simulated rank:  710.0\n",
      "GO:0075733-intracellular transport of virus\n",
      "Template rank:  488.0\n",
      "Median simulated rank:  597.5\n",
      "GO:0055029-nuclear DNA-directed RNA polymerase complex\n",
      "Template rank:  489.0\n",
      "Median simulated rank:  845.5\n",
      "GO:0044773-mitotic DNA damage checkpoint\n",
      "Template rank:  490.0\n",
      "Median simulated rank:  1016.0\n",
      "GO:0038095-Fc-epsilon receptor signaling pathway\n",
      "Template rank:  491.0\n",
      "Median simulated rank:  748.0\n",
      "GO:1903051-negative regulation of proteolysis involved in cellular protein catabolic process\n",
      "Template rank:  492.0\n",
      "Median simulated rank:  793.5\n",
      "GO:0000428-DNA-directed RNA polymerase complex\n",
      "Template rank:  494.0\n",
      "Median simulated rank:  873.0\n",
      "GO:0061726-mitochondrion disassembly\n",
      "Template rank:  495.5\n",
      "Median simulated rank:  979.5\n",
      "GO:0000422-autophagy of mitochondrion\n",
      "Template rank:  495.5\n",
      "Median simulated rank:  979.5\n",
      "GO:0000777-condensed chromosome kinetochore\n",
      "Template rank:  497.0\n",
      "Median simulated rank:  750.0\n",
      "GO:0030880-RNA polymerase complex\n",
      "Template rank:  498.0\n",
      "Median simulated rank:  800.0\n",
      "GO:0016072-rRNA metabolic process\n",
      "Template rank:  499.0\n",
      "Median simulated rank:  1059.0\n",
      "GO:0006998-nuclear envelope organization\n",
      "Template rank:  500.0\n",
      "Median simulated rank:  685.0\n",
      "GO:0045333-cellular respiration\n",
      "Template rank:  502.0\n",
      "Median simulated rank:  524.0\n",
      "GO:0006278-RNA-dependent DNA biosynthetic process\n",
      "Template rank:  503.0\n",
      "Median simulated rank:  900.0\n",
      "GO:0006261-DNA-dependent DNA replication\n",
      "Template rank:  504.0\n",
      "Median simulated rank:  1203.0\n",
      "GO:0008094-DNA-dependent ATPase activity\n",
      "Template rank:  505.0\n",
      "Median simulated rank:  533.0\n",
      "GO:0005758-mitochondrial intermembrane space\n",
      "Template rank:  506.0\n",
      "Median simulated rank:  680.0\n",
      "GO:0003730-mRNA 3'-UTR binding\n",
      "Template rank:  507.0\n",
      "Median simulated rank:  992.0\n",
      "GO:0001736-establishment of planar polarity\n",
      "Template rank:  508.5\n",
      "Median simulated rank:  1088.5\n",
      "GO:0007164-establishment of tissue polarity\n",
      "Template rank:  508.5\n",
      "Median simulated rank:  1088.5\n",
      "GO:0090329-regulation of DNA-dependent DNA replication\n",
      "Template rank:  511.0\n",
      "Median simulated rank:  719.0\n",
      "GO:0060071-Wnt signaling pathway, planar cell polarity pathway\n",
      "Template rank:  513.0\n",
      "Median simulated rank:  863.0\n",
      "GO:2001252-positive regulation of chromosome organization\n",
      "Template rank:  514.0\n",
      "Median simulated rank:  1100.0\n",
      "GO:0070936-protein K48-linked ubiquitination\n",
      "Template rank:  515.0\n",
      "Median simulated rank:  971.0\n",
      "GO:1902373-negative regulation of mRNA catabolic process\n",
      "Template rank:  516.0\n",
      "Median simulated rank:  589.5\n",
      "GO:0031146-SCF-dependent proteasomal ubiquitin-dependent protein catabolic process\n",
      "Template rank:  517.0\n",
      "Median simulated rank:  1369.0\n",
      "GO:0000775-chromosome, centromeric region\n",
      "Template rank:  518.0\n",
      "Median simulated rank:  1029.0\n",
      "GO:0007051-spindle organization\n",
      "Template rank:  519.0\n",
      "Median simulated rank:  1202.0\n",
      "GO:0019886-antigen processing and presentation of exogenous peptide antigen via MHC class II\n",
      "Template rank:  520.0\n",
      "Median simulated rank:  633.0\n",
      "GO:0090175-regulation of establishment of planar polarity\n",
      "Template rank:  521.0\n",
      "Median simulated rank:  990.5\n",
      "GO:1903312-negative regulation of mRNA metabolic process\n",
      "Template rank:  522.0\n",
      "Median simulated rank:  802.5\n",
      "GO:0016073-snRNA metabolic process\n",
      "Template rank:  523.0\n",
      "Median simulated rank:  1156.0\n",
      "GO:0005643-nuclear pore\n",
      "Template rank:  524.0\n",
      "Median simulated rank:  1336.0\n",
      "GO:0061077-chaperone-mediated protein folding\n",
      "Template rank:  525.0\n",
      "Median simulated rank:  1159.0\n",
      "GO:0000819-sister chromatid segregation\n",
      "Template rank:  528.0\n",
      "Median simulated rank:  902.5\n",
      "GO:0034708-methyltransferase complex\n",
      "Template rank:  529.0\n",
      "Median simulated rank:  984.0\n",
      "GO:0008213-protein alkylation\n",
      "Template rank:  531.5\n",
      "Median simulated rank:  1869.5\n",
      "GO:0006479-protein methylation\n",
      "Template rank:  531.5\n",
      "Median simulated rank:  1869.5\n",
      "GO:1902686-mitochondrial outer membrane permeabilization involved in programmed cell death\n",
      "Template rank:  533.0\n",
      "Median simulated rank:  812.0\n",
      "GO:0016445-somatic diversification of immunoglobulins\n",
      "Template rank:  534.0\n",
      "Median simulated rank:  972.5\n",
      "GO:0007093-mitotic cell cycle checkpoint\n",
      "Template rank:  535.0\n",
      "Median simulated rank:  1010.0\n",
      "GO:0044774-mitotic DNA integrity checkpoint\n",
      "Template rank:  536.0\n",
      "Median simulated rank:  996.0\n",
      "GO:0034968-histone lysine methylation\n",
      "Template rank:  538.0\n",
      "Median simulated rank:  1863.0\n",
      "GO:1902110-positive regulation of mitochondrial membrane permeability involved in apoptotic process\n",
      "Template rank:  539.0\n",
      "Median simulated rank:  834.0\n",
      "GO:0048002-antigen processing and presentation of peptide antigen\n",
      "Template rank:  540.0\n",
      "Median simulated rank:  561.5\n",
      "GO:0030330-DNA damage response, signal transduction by p53 class mediator\n",
      "Template rank:  541.0\n",
      "Median simulated rank:  1009.0\n",
      "GO:1903749-positive regulation of establishment of protein localization to mitochondrion\n",
      "Template rank:  542.0\n",
      "Median simulated rank:  613.0\n",
      "GO:0006403-RNA localization\n",
      "Template rank:  543.0\n",
      "Median simulated rank:  905.0\n",
      "GO:0046794-transport of virus\n",
      "Template rank:  544.0\n",
      "Median simulated rank:  552.5\n",
      "GO:0005681-spliceosomal complex\n",
      "Template rank:  546.0\n",
      "Median simulated rank:  599.0\n",
      "GO:0007006-mitochondrial membrane organization\n",
      "Template rank:  548.0\n",
      "Median simulated rank:  839.5\n",
      "GO:0043297-apical junction assembly\n",
      "Template rank:  549.0\n",
      "Median simulated rank:  1024.0\n",
      "GO:0098876-vesicle-mediated transport to the plasma membrane\n",
      "Template rank:  550.0\n",
      "Median simulated rank:  1474.0\n",
      "GO:1902117-positive regulation of organelle assembly\n",
      "Template rank:  551.0\n",
      "Median simulated rank:  944.0\n",
      "GO:0022853-active ion transmembrane transporter activity\n",
      "Template rank:  552.5\n",
      "Median simulated rank:  738.0\n",
      "GO:0042625-ATPase coupled ion transmembrane transporter activity\n",
      "Template rank:  552.5\n",
      "Median simulated rank:  738.0\n",
      "GO:0010822-positive regulation of mitochondrion organization\n",
      "Template rank:  554.0\n",
      "Median simulated rank:  966.0\n",
      "GO:0046902-regulation of mitochondrial membrane permeability\n",
      "Template rank:  555.0\n",
      "Median simulated rank:  876.0\n",
      "GO:0034394-protein localization to cell surface\n",
      "Template rank:  556.0\n",
      "Median simulated rank:  1434.0\n",
      "GO:1905269-positive regulation of chromatin organization\n",
      "Template rank:  557.0\n",
      "Median simulated rank:  1080.0\n",
      "GO:0006611-protein export from nucleus\n",
      "Template rank:  558.0\n",
      "Median simulated rank:  938.0\n",
      "GO:0006986-response to unfolded protein\n",
      "Template rank:  559.0\n",
      "Median simulated rank:  843.0\n",
      "GO:2001242-regulation of intrinsic apoptotic signaling pathway\n",
      "Template rank:  560.0\n",
      "Median simulated rank:  941.0\n",
      "GO:0002200-somatic diversification of immune receptors\n",
      "Template rank:  561.0\n",
      "Median simulated rank:  1283.0\n",
      "GO:0051310-metaphase plate congression\n",
      "Template rank:  562.0\n",
      "Median simulated rank:  630.0\n",
      "GO:0022613-ribonucleoprotein complex biogenesis\n",
      "Template rank:  564.0\n",
      "Median simulated rank:  874.0\n",
      "GO:1901987-regulation of cell cycle phase transition\n",
      "Template rank:  565.0\n",
      "Median simulated rank:  1049.5\n",
      "GO:0035794-positive regulation of mitochondrial membrane permeability\n",
      "Template rank:  567.0\n",
      "Median simulated rank:  881.0\n",
      "GO:0000077-DNA damage checkpoint\n",
      "Template rank:  568.0\n",
      "Median simulated rank:  1009.5\n",
      "GO:0006892-post-Golgi vesicle-mediated transport\n",
      "Template rank:  569.0\n",
      "Median simulated rank:  1425.5\n",
      "GO:0006289-nucleotide-excision repair\n",
      "Template rank:  570.0\n",
      "Median simulated rank:  907.0\n",
      "GO:0007088-regulation of mitotic nuclear division\n",
      "Template rank:  574.0\n",
      "Median simulated rank:  968.0\n",
      "GO:1903747-regulation of establishment of protein localization to mitochondrion\n",
      "Template rank:  575.0\n",
      "Median simulated rank:  712.0\n",
      "GO:0001098-basal transcription machinery binding\n",
      "Template rank:  576.5\n",
      "Median simulated rank:  573.5\n",
      "GO:0001099-basal RNA polymerase II transcription machinery binding\n",
      "Template rank:  576.5\n",
      "Median simulated rank:  573.5\n",
      "GO:0019829-cation-transporting ATPase activity\n",
      "Template rank:  578.0\n",
      "Median simulated rank:  820.0\n",
      "GO:1903050-regulation of proteolysis involved in cellular protein catabolic process\n",
      "Template rank:  579.0\n",
      "Median simulated rank:  1296.0\n",
      "GO:0010948-negative regulation of cell cycle process\n",
      "Template rank:  580.0\n",
      "Median simulated rank:  999.0\n",
      "GO:0031058-positive regulation of histone modification\n",
      "Template rank:  581.0\n",
      "Median simulated rank:  871.0\n",
      "GO:0038093-Fc receptor signaling pathway\n",
      "Template rank:  583.0\n",
      "Median simulated rank:  984.0\n",
      "GO:0070585-protein localization to mitochondrion\n",
      "Template rank:  584.0\n",
      "Median simulated rank:  712.0\n",
      "GO:0000075-cell cycle checkpoint\n",
      "Template rank:  585.0\n",
      "Median simulated rank:  1041.5\n",
      "GO:0016241-regulation of macroautophagy\n",
      "Template rank:  586.0\n",
      "Median simulated rank:  1499.5\n",
      "GO:1902108-regulation of mitochondrial membrane permeability involved in apoptotic process\n",
      "Template rank:  588.0\n",
      "Median simulated rank:  1008.0\n",
      "GO:0035966-response to topologically incorrect protein\n",
      "Template rank:  589.0\n",
      "Median simulated rank:  921.0\n",
      "GO:0002495-antigen processing and presentation of peptide antigen via MHC class II\n",
      "Template rank:  590.0\n",
      "Median simulated rank:  652.0\n",
      "GO:1903311-regulation of mRNA metabolic process\n",
      "Template rank:  591.0\n",
      "Median simulated rank:  1238.0\n",
      "GO:1901990-regulation of mitotic cell cycle phase transition\n",
      "Template rank:  592.0\n",
      "Median simulated rank:  1020.0\n",
      "GO:0072655-establishment of protein localization to mitochondrion\n",
      "Template rank:  593.0\n",
      "Median simulated rank:  713.0\n",
      "GO:0031571-mitotic G1 DNA damage checkpoint\n",
      "Template rank:  594.5\n",
      "Median simulated rank:  629.0\n",
      "GO:0044819-mitotic G1/S transition checkpoint\n",
      "Template rank:  594.5\n",
      "Median simulated rank:  629.0\n",
      "GO:0090559-regulation of membrane permeability\n",
      "Template rank:  596.0\n",
      "Median simulated rank:  928.0\n",
      "GO:0035258-steroid hormone receptor binding\n",
      "Template rank:  597.0\n",
      "Median simulated rank:  821.0\n",
      "GO:0140014-mitotic nuclear division\n",
      "Template rank:  598.0\n",
      "Median simulated rank:  1001.0\n",
      "GO:1903052-positive regulation of proteolysis involved in cellular protein catabolic process\n",
      "Template rank:  599.0\n",
      "Median simulated rank:  1456.5\n",
      "GO:0010494-cytoplasmic stress granule\n",
      "Template rank:  600.0\n",
      "Median simulated rank:  626.5\n",
      "GO:0006487-protein N-linked glycosylation\n",
      "Template rank:  601.0\n",
      "Median simulated rank:  1352.0\n",
      "GO:0002504-antigen processing and presentation of peptide or polysaccharide antigen via MHC class II\n",
      "Template rank:  602.0\n",
      "Median simulated rank:  644.5\n",
      "GO:0097194-execution phase of apoptosis\n",
      "Template rank:  603.0\n",
      "Median simulated rank:  783.0\n",
      "GO:0035097-histone methyltransferase complex\n",
      "Template rank:  604.0\n",
      "Median simulated rank:  934.0\n",
      "GO:1903313-positive regulation of mRNA metabolic process\n",
      "Template rank:  605.0\n",
      "Median simulated rank:  1425.0\n",
      "GO:0016239-positive regulation of macroautophagy\n",
      "Template rank:  606.0\n",
      "Median simulated rank:  1236.0\n",
      "GO:1900034-regulation of cellular response to heat\n",
      "Template rank:  607.0\n",
      "Median simulated rank:  760.0\n",
      "GO:0031570-DNA integrity checkpoint\n",
      "Template rank:  608.0\n",
      "Median simulated rank:  943.0\n",
      "GO:0042770-signal transduction in response to DNA damage\n",
      "Template rank:  609.0\n",
      "Median simulated rank:  981.0\n",
      "GO:0016363-nuclear matrix\n",
      "Template rank:  610.0\n",
      "Median simulated rank:  1233.0\n",
      "GO:1901992-positive regulation of mitotic cell cycle phase transition\n",
      "Template rank:  611.0\n",
      "Median simulated rank:  1160.0\n",
      "GO:1902653-secondary alcohol biosynthetic process\n",
      "Template rank:  612.0\n",
      "Median simulated rank:  622.0\n",
      "GO:0022618-ribonucleoprotein complex assembly\n",
      "Template rank:  613.0\n",
      "Median simulated rank:  782.5\n",
      "GO:1903362-regulation of cellular protein catabolic process\n",
      "Template rank:  614.0\n",
      "Median simulated rank:  1307.0\n",
      "GO:0071826-ribonucleoprotein complex subunit organization\n",
      "Template rank:  615.0\n",
      "Median simulated rank:  862.0\n",
      "GO:0030433-ubiquitin-dependent ERAD pathway\n",
      "Template rank:  616.0\n",
      "Median simulated rank:  822.0\n",
      "GO:0044783-G1 DNA damage checkpoint\n",
      "Template rank:  617.0\n",
      "Median simulated rank:  683.0\n",
      "GO:0046128-purine ribonucleoside metabolic process\n",
      "Template rank:  618.0\n",
      "Median simulated rank:  1100.5\n",
      "GO:0001738-morphogenesis of a polarized epithelium\n",
      "Template rank:  619.0\n",
      "Median simulated rank:  1197.0\n",
      "GO:0006606-protein import into nucleus\n",
      "Template rank:  620.0\n",
      "Median simulated rank:  1162.0\n",
      "GO:0009124-nucleoside monophosphate biosynthetic process\n",
      "Template rank:  621.0\n",
      "Median simulated rank:  1211.0\n",
      "GO:0045727-positive regulation of translation\n",
      "Template rank:  622.0\n",
      "Median simulated rank:  784.0\n",
      "GO:0035257-nuclear hormone receptor binding\n",
      "Template rank:  623.0\n",
      "Median simulated rank:  860.0\n",
      "GO:0061136-regulation of proteasomal protein catabolic process\n",
      "Template rank:  624.0\n",
      "Median simulated rank:  1181.0\n",
      "GO:2000736-regulation of stem cell differentiation\n",
      "Template rank:  625.0\n",
      "Median simulated rank:  986.0\n",
      "GO:0009123-nucleoside monophosphate metabolic process\n",
      "Template rank:  626.0\n",
      "Median simulated rank:  779.0\n",
      "GO:0006626-protein targeting to mitochondrion\n",
      "Template rank:  627.0\n",
      "Median simulated rank:  751.5\n",
      "GO:2000058-regulation of ubiquitin-dependent protein catabolic process\n",
      "Template rank:  628.0\n",
      "Median simulated rank:  1379.0\n",
      "GO:0051168-nuclear export\n",
      "Template rank:  629.0\n",
      "Median simulated rank:  1037.0\n",
      "GO:0007034-vacuolar transport\n",
      "Template rank:  630.0\n",
      "Median simulated rank:  1778.0\n",
      "GO:0007041-lysosomal transport\n",
      "Template rank:  631.0\n",
      "Median simulated rank:  1606.0\n",
      "GO:0034976-response to endoplasmic reticulum stress\n",
      "Template rank:  632.0\n",
      "Median simulated rank:  1174.0\n",
      "GO:0032204-regulation of telomere maintenance\n",
      "Template rank:  633.0\n",
      "Median simulated rank:  1129.0\n",
      "GO:0009161-ribonucleoside monophosphate metabolic process\n",
      "Template rank:  634.0\n",
      "Median simulated rank:  730.0\n",
      "GO:0034250-positive regulation of cellular amide metabolic process\n",
      "Template rank:  635.0\n",
      "Median simulated rank:  963.5\n",
      "GO:0042278-purine nucleoside metabolic process\n",
      "Template rank:  636.0\n",
      "Median simulated rank:  1126.5\n",
      "GO:0006695-cholesterol biosynthetic process\n",
      "Template rank:  637.0\n",
      "Median simulated rank:  646.0\n",
      "GO:0009167-purine ribonucleoside monophosphate metabolic process\n",
      "Template rank:  638.0\n",
      "Median simulated rank:  738.0\n",
      "GO:0009144-purine nucleoside triphosphate metabolic process\n",
      "Template rank:  639.0\n",
      "Median simulated rank:  754.0\n",
      "GO:0006260-DNA replication\n",
      "Template rank:  640.0\n",
      "Median simulated rank:  1455.0\n",
      "GO:0009126-purine nucleoside monophosphate metabolic process\n",
      "Template rank:  641.0\n",
      "Median simulated rank:  754.0\n",
      "GO:0051225-spindle assembly\n",
      "Template rank:  642.0\n",
      "Median simulated rank:  1179.5\n",
      "GO:1901989-positive regulation of cell cycle phase transition\n",
      "Template rank:  643.0\n",
      "Median simulated rank:  1289.0\n",
      "GO:2000060-positive regulation of ubiquitin-dependent protein catabolic process\n",
      "Template rank:  644.0\n",
      "Median simulated rank:  1687.5\n",
      "GO:0051170-import into nucleus\n",
      "Template rank:  645.0\n",
      "Median simulated rank:  1153.5\n",
      "GO:0000910-cytokinesis\n",
      "Template rank:  648.0\n",
      "Median simulated rank:  1579.0\n",
      "GO:1905710-positive regulation of membrane permeability\n",
      "Template rank:  649.0\n",
      "Median simulated rank:  685.0\n",
      "GO:0051028-mRNA transport\n",
      "Template rank:  650.0\n",
      "Median simulated rank:  988.0\n",
      "GO:0009205-purine ribonucleoside triphosphate metabolic process\n",
      "Template rank:  653.0\n",
      "Median simulated rank:  734.0\n",
      "GO:0008380-RNA splicing\n",
      "Template rank:  654.0\n",
      "Median simulated rank:  1020.0\n",
      "GO:0002433-immune response-regulating cell surface receptor signaling pathway involved in phagocytosis\n",
      "Template rank:  655.5\n",
      "Median simulated rank:  1235.5\n",
      "GO:0038096-Fc-gamma receptor signaling pathway involved in phagocytosis\n",
      "Template rank:  655.5\n",
      "Median simulated rank:  1235.5\n",
      "GO:0009141-nucleoside triphosphate metabolic process\n",
      "Template rank:  657.0\n",
      "Median simulated rank:  788.0\n",
      "GO:0140101-catalytic activity, acting on a tRNA\n",
      "Template rank:  658.0\n",
      "Median simulated rank:  1918.0\n",
      "GO:0033044-regulation of chromosome organization\n",
      "Template rank:  659.0\n",
      "Median simulated rank:  1283.5\n",
      "GO:1903008-organelle disassembly\n",
      "Template rank:  660.0\n",
      "Median simulated rank:  1326.0\n",
      "GO:0050658-RNA transport\n",
      "Template rank:  662.5\n",
      "Median simulated rank:  936.5\n",
      "GO:0050657-nucleic acid transport\n",
      "Template rank:  662.5\n",
      "Median simulated rank:  936.5\n",
      "GO:0072331-signal transduction by p53 class mediator\n",
      "Template rank:  664.0\n",
      "Median simulated rank:  1359.0\n",
      "GO:0097193-intrinsic apoptotic signaling pathway\n",
      "Template rank:  665.0\n",
      "Median simulated rank:  1006.0\n",
      "GO:1901657-glycosyl compound metabolic process\n",
      "Template rank:  666.0\n",
      "Median simulated rank:  1059.0\n",
      "GO:0071230-cellular response to amino acid stimulus\n",
      "Template rank:  669.0\n",
      "Median simulated rank:  1487.0\n",
      "GO:0051169-nuclear transport\n",
      "Template rank:  672.0\n",
      "Median simulated rank:  1149.0\n",
      "GO:0009199-ribonucleoside triphosphate metabolic process\n",
      "Template rank:  674.0\n",
      "Median simulated rank:  775.0\n",
      "GO:0002431-Fc receptor mediated stimulatory signaling pathway\n",
      "Template rank:  675.0\n",
      "Median simulated rank:  1132.0\n",
      "GO:0006399-tRNA metabolic process\n",
      "Template rank:  676.0\n",
      "Median simulated rank:  1750.0\n",
      "GO:0030374-nuclear receptor transcription coactivator activity\n",
      "Template rank:  677.0\n",
      "Median simulated rank:  718.5\n",
      "GO:0034333-adherens junction assembly\n",
      "Template rank:  678.0\n",
      "Median simulated rank:  1080.5\n",
      "GO:0034212-peptide N-acetyltransferase activity\n",
      "Template rank:  681.0\n",
      "Median simulated rank:  950.5\n",
      "GO:0034470-ncRNA processing\n",
      "Template rank:  682.0\n",
      "Median simulated rank:  1535.5\n",
      "GO:0007044-cell-substrate junction assembly\n",
      "Template rank:  683.0\n",
      "Median simulated rank:  741.0\n",
      "GO:0000288-nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay\n",
      "Template rank:  684.0\n",
      "Median simulated rank:  927.0\n",
      "GO:0042177-negative regulation of protein catabolic process\n",
      "Template rank:  685.0\n",
      "Median simulated rank:  1647.5\n",
      "GO:1903322-positive regulation of protein modification by small protein conjugation or removal\n",
      "Template rank:  686.0\n",
      "Median simulated rank:  1866.0\n",
      "GO:0006417-regulation of translation\n",
      "Template rank:  687.0\n",
      "Median simulated rank:  839.5\n",
      "GO:0017038-protein import\n",
      "Template rank:  688.0\n",
      "Median simulated rank:  1316.5\n",
      "GO:0051236-establishment of RNA localization\n",
      "Template rank:  689.0\n",
      "Median simulated rank:  970.0\n",
      "GO:0042147-retrograde transport, endosome to Golgi\n",
      "Template rank:  690.0\n",
      "Median simulated rank:  1848.0\n",
      "GO:0030518-intracellular steroid hormone receptor signaling pathway\n",
      "Template rank:  691.0\n",
      "Median simulated rank:  1299.0\n",
      "GO:0050821-protein stabilization\n",
      "Template rank:  692.0\n",
      "Median simulated rank:  1019.5\n",
      "GO:0010821-regulation of mitochondrion organization\n",
      "Template rank:  693.0\n",
      "Median simulated rank:  1237.0\n",
      "GO:0004527-exonuclease activity\n",
      "Template rank:  694.0\n",
      "Median simulated rank:  1132.0\n",
      "GO:0045930-negative regulation of mitotic cell cycle\n",
      "Template rank:  695.0\n",
      "Median simulated rank:  1095.5\n",
      "GO:0006913-nucleocytoplasmic transport\n",
      "Template rank:  696.0\n",
      "Median simulated rank:  1164.0\n",
      "GO:0009127-purine nucleoside monophosphate biosynthetic process\n",
      "Template rank:  697.5\n",
      "Median simulated rank:  1186.5\n",
      "GO:0009168-purine ribonucleoside monophosphate biosynthetic process\n",
      "Template rank:  697.5\n",
      "Median simulated rank:  1186.5\n",
      "GO:0046034-ATP metabolic process\n",
      "Template rank:  699.0\n",
      "Median simulated rank:  695.0\n",
      "GO:0004386-helicase activity\n",
      "Template rank:  700.0\n",
      "Median simulated rank:  1368.5\n",
      "GO:0031503-protein-containing complex localization\n",
      "Template rank:  701.0\n",
      "Median simulated rank:  1538.0\n",
      "GO:0005743-mitochondrial inner membrane\n",
      "Template rank:  702.0\n",
      "Median simulated rank:  724.0\n",
      "GO:0009156-ribonucleoside monophosphate biosynthetic process\n",
      "Template rank:  703.0\n",
      "Median simulated rank:  1176.0\n",
      "GO:0000784-nuclear chromosome, telomeric region\n",
      "Template rank:  704.0\n",
      "Median simulated rank:  925.0\n",
      "GO:0045931-positive regulation of mitotic cell cycle\n",
      "Template rank:  705.0\n",
      "Median simulated rank:  1208.0\n",
      "GO:0009451-RNA modification\n",
      "Template rank:  706.0\n",
      "Median simulated rank:  2266.0\n",
      "GO:0044452-nucleolar part\n",
      "Template rank:  707.0\n",
      "Median simulated rank:  1436.0\n",
      "GO:0009119-ribonucleoside metabolic process\n",
      "Template rank:  708.0\n",
      "Median simulated rank:  1119.0\n",
      "GO:1903364-positive regulation of cellular protein catabolic process\n",
      "Template rank:  710.0\n",
      "Median simulated rank:  1398.0\n",
      "GO:0016779-nucleotidyltransferase activity\n",
      "Template rank:  711.0\n",
      "Median simulated rank:  1533.0\n",
      "GO:0016853-isomerase activity\n",
      "Template rank:  712.0\n",
      "Median simulated rank:  1032.0\n",
      "GO:0018022-peptidyl-lysine methylation\n",
      "Template rank:  713.0\n",
      "Median simulated rank:  1484.0\n",
      "GO:0016234-inclusion body\n",
      "Template rank:  714.0\n",
      "Median simulated rank:  1532.5\n",
      "GO:0006513-protein monoubiquitination\n",
      "Template rank:  715.0\n",
      "Median simulated rank:  922.0\n",
      "GO:0034332-adherens junction organization\n",
      "Template rank:  717.0\n",
      "Median simulated rank:  1331.0\n",
      "GO:0000375-RNA splicing, via transesterification reactions\n",
      "Template rank:  718.0\n",
      "Median simulated rank:  998.0\n",
      "GO:2000278-regulation of DNA biosynthetic process\n",
      "Template rank:  719.0\n",
      "Median simulated rank:  1003.0\n",
      "GO:0005759-mitochondrial matrix\n",
      "Template rank:  720.0\n",
      "Median simulated rank:  927.0\n",
      "GO:0009116-nucleoside metabolic process\n",
      "Template rank:  722.0\n",
      "Median simulated rank:  1230.5\n",
      "GO:0009185-ribonucleoside diphosphate metabolic process\n",
      "Template rank:  723.0\n",
      "Median simulated rank:  1334.0\n",
      "GO:1901800-positive regulation of proteasomal protein catabolic process\n",
      "Template rank:  724.0\n",
      "Median simulated rank:  1196.0\n",
      "GO:0009206-purine ribonucleoside triphosphate biosynthetic process\n",
      "Template rank:  725.0\n",
      "Median simulated rank:  1206.0\n",
      "GO:0006839-mitochondrial transport\n",
      "Template rank:  726.0\n",
      "Median simulated rank:  996.0\n",
      "GO:0000398-mRNA splicing, via spliceosome\n",
      "Template rank:  727.5\n",
      "Median simulated rank:  990.5\n",
      "GO:0000377-RNA splicing, via transesterification reactions with bulged adenosine as nucleophile\n",
      "Template rank:  727.5\n",
      "Median simulated rank:  990.5\n",
      "GO:0051783-regulation of nuclear division\n",
      "Template rank:  730.0\n",
      "Median simulated rank:  1233.0\n",
      "GO:0006605-protein targeting\n",
      "Template rank:  731.0\n",
      "Median simulated rank:  1098.5\n",
      "GO:1990542-mitochondrial transmembrane transport\n",
      "Template rank:  733.0\n",
      "Median simulated rank:  896.5\n",
      "GO:0038094-Fc-gamma receptor signaling pathway\n",
      "Template rank:  734.0\n",
      "Median simulated rank:  1185.0\n",
      "GO:0050792-regulation of viral process\n",
      "Template rank:  735.0\n",
      "Median simulated rank:  1023.5\n",
      "GO:0008757-S-adenosylmethionine-dependent methyltransferase activity\n",
      "Template rank:  736.0\n",
      "Median simulated rank:  1987.0\n",
      "GO:0010507-negative regulation of autophagy\n",
      "Template rank:  737.0\n",
      "Median simulated rank:  1027.5\n",
      "GO:0044445-cytosolic part\n",
      "Template rank:  738.0\n",
      "Median simulated rank:  760.0\n",
      "GO:0032434-regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "Template rank:  739.0\n",
      "Median simulated rank:  1173.0\n",
      "GO:0098687-chromosomal region\n",
      "Template rank:  740.0\n",
      "Median simulated rank:  1133.5\n",
      "GO:0005925-focal adhesion\n",
      "Template rank:  741.0\n",
      "Median simulated rank:  752.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GO:0000794-condensed nuclear chromosome\n",
      "Template rank:  742.0\n",
      "Median simulated rank:  956.0\n",
      "GO:0005924-cell-substrate adherens junction\n",
      "Template rank:  743.0\n",
      "Median simulated rank:  751.0\n",
      "GO:0043161-proteasome-mediated ubiquitin-dependent protein catabolic process\n",
      "Template rank:  744.0\n",
      "Median simulated rank:  1392.0\n",
      "GO:0030117-membrane coat\n",
      "Template rank:  745.5\n",
      "Median simulated rank:  1881.5\n",
      "GO:0048475-coated membrane\n",
      "Template rank:  745.5\n",
      "Median simulated rank:  1881.5\n",
      "GO:0009145-purine nucleoside triphosphate biosynthetic process\n",
      "Template rank:  747.0\n",
      "Median simulated rank:  1192.0\n",
      "GO:0000932-P-body\n",
      "Template rank:  748.0\n",
      "Median simulated rank:  861.0\n",
      "GO:0006900-vesicle budding from membrane\n",
      "Template rank:  750.0\n",
      "Median simulated rank:  916.0\n",
      "GO:0001650-fibrillar center\n",
      "Template rank:  751.0\n",
      "Median simulated rank:  1557.0\n",
      "GO:0044389-ubiquitin-like protein ligase binding\n",
      "Template rank:  753.0\n",
      "Median simulated rank:  962.0\n",
      "GO:1902806-regulation of cell cycle G1/S phase transition\n",
      "Template rank:  754.0\n",
      "Median simulated rank:  905.5\n",
      "GO:0009201-ribonucleoside triphosphate biosynthetic process\n",
      "Template rank:  756.0\n",
      "Median simulated rank:  1147.5\n",
      "GO:0006275-regulation of DNA replication\n",
      "Template rank:  757.0\n",
      "Median simulated rank:  1720.0\n",
      "GO:0030055-cell-substrate junction\n",
      "Template rank:  758.0\n",
      "Median simulated rank:  795.0\n",
      "GO:0009132-nucleoside diphosphate metabolic process\n",
      "Template rank:  759.0\n",
      "Median simulated rank:  1626.0\n",
      "GO:0090501-RNA phosphodiester bond hydrolysis\n",
      "Template rank:  760.0\n",
      "Median simulated rank:  1413.0\n",
      "GO:0009179-purine ribonucleoside diphosphate metabolic process\n",
      "Template rank:  761.5\n",
      "Median simulated rank:  1353.5\n",
      "GO:0009135-purine nucleoside diphosphate metabolic process\n",
      "Template rank:  761.5\n",
      "Median simulated rank:  1353.5\n",
      "GO:0034504-protein localization to nucleus\n",
      "Template rank:  763.0\n",
      "Median simulated rank:  1327.5\n",
      "GO:0043624-cellular protein complex disassembly\n",
      "Template rank:  764.0\n",
      "Median simulated rank:  873.5\n",
      "GO:0048193-Golgi vesicle transport\n",
      "Template rank:  765.0\n",
      "Median simulated rank:  1457.5\n",
      "GO:0010833-telomere maintenance via telomere lengthening\n",
      "Template rank:  766.0\n",
      "Median simulated rank:  1122.0\n",
      "GO:0031625-ubiquitin protein ligase binding\n",
      "Template rank:  767.0\n",
      "Median simulated rank:  979.0\n",
      "GO:0031056-regulation of histone modification\n",
      "Template rank:  769.0\n",
      "Median simulated rank:  1353.0\n",
      "GO:0009142-nucleoside triphosphate biosynthetic process\n",
      "Template rank:  770.0\n",
      "Median simulated rank:  1241.0\n",
      "GO:0016197-endosomal transport\n",
      "Template rank:  771.0\n",
      "Median simulated rank:  1520.0\n",
      "GO:0005913-cell-cell adherens junction\n",
      "Template rank:  772.0\n",
      "Median simulated rank:  1742.0\n",
      "GO:0140098-catalytic activity, acting on RNA\n",
      "Template rank:  773.0\n",
      "Median simulated rank:  1619.0\n",
      "GO:0008033-tRNA processing\n",
      "Template rank:  774.0\n",
      "Median simulated rank:  1506.0\n",
      "GO:0006367-transcription initiation from RNA polymerase II promoter\n",
      "Template rank:  775.0\n",
      "Median simulated rank:  1603.5\n",
      "GO:0031398-positive regulation of protein ubiquitination\n",
      "Template rank:  776.0\n",
      "Median simulated rank:  1958.5\n",
      "GO:2001251-negative regulation of chromosome organization\n",
      "Template rank:  777.0\n",
      "Median simulated rank:  1136.0\n",
      "GO:0034248-regulation of cellular amide metabolic process\n",
      "Template rank:  778.0\n",
      "Median simulated rank:  926.5\n",
      "GO:0015931-nucleobase-containing compound transport\n",
      "Template rank:  779.0\n",
      "Median simulated rank:  1342.0\n",
      "GO:0071456-cellular response to hypoxia\n",
      "Template rank:  780.0\n",
      "Median simulated rank:  816.5\n",
      "GO:0097711-ciliary basal body-plasma membrane docking\n",
      "Template rank:  781.0\n",
      "Median simulated rank:  1607.0\n",
      "GO:0004540-ribonuclease activity\n",
      "Template rank:  782.0\n",
      "Median simulated rank:  1532.5\n",
      "GO:0036503-ERAD pathway\n",
      "Template rank:  783.0\n",
      "Median simulated rank:  1321.0\n",
      "GO:0000287-magnesium ion binding\n",
      "Template rank:  784.0\n",
      "Median simulated rank:  1992.0\n",
      "GO:0030176-integral component of endoplasmic reticulum membrane\n",
      "Template rank:  785.0\n",
      "Median simulated rank:  899.0\n",
      "GO:0044843-cell cycle G1/S phase transition\n",
      "Template rank:  786.0\n",
      "Median simulated rank:  969.5\n",
      "GO:0032436-positive regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "Template rank:  787.0\n",
      "Median simulated rank:  1361.0\n",
      "GO:0006754-ATP biosynthetic process\n",
      "Template rank:  788.0\n",
      "Median simulated rank:  1168.0\n",
      "GO:0016236-macroautophagy\n",
      "Template rank:  789.0\n",
      "Median simulated rank:  1549.0\n",
      "GO:0090150-establishment of protein localization to membrane\n",
      "Template rank:  791.0\n",
      "Median simulated rank:  1176.0\n",
      "GO:0006612-protein targeting to membrane\n",
      "Template rank:  792.0\n",
      "Median simulated rank:  884.5\n",
      "GO:0090090-negative regulation of canonical Wnt signaling pathway\n",
      "Template rank:  795.0\n",
      "Median simulated rank:  1354.0\n",
      "GO:0018210-peptidyl-threonine modification\n",
      "Template rank:  796.0\n",
      "Median simulated rank:  2049.0\n",
      "GO:0003713-transcription coactivator activity\n",
      "Template rank:  797.0\n",
      "Median simulated rank:  1403.5\n",
      "GO:0010508-positive regulation of autophagy\n",
      "Template rank:  798.0\n",
      "Median simulated rank:  1686.5\n",
      "GO:0030496-midbody\n",
      "Template rank:  799.0\n",
      "Median simulated rank:  1328.0\n",
      "GO:0006165-nucleoside diphosphate phosphorylation\n",
      "Template rank:  800.0\n",
      "Median simulated rank:  1526.0\n",
      "GO:0043903-regulation of symbiosis, encompassing mutualism through parasitism\n",
      "Template rank:  801.0\n",
      "Median simulated rank:  907.0\n",
      "GO:0090263-positive regulation of canonical Wnt signaling pathway\n",
      "Template rank:  802.0\n",
      "Median simulated rank:  1407.5\n",
      "GO:0043414-macromolecule methylation\n",
      "Template rank:  803.0\n",
      "Median simulated rank:  2161.0\n",
      "GO:0006997-nucleus organization\n",
      "Template rank:  804.0\n",
      "Median simulated rank:  1036.0\n",
      "GO:0019903-protein phosphatase binding\n",
      "Template rank:  805.0\n",
      "Median simulated rank:  1522.5\n",
      "GO:0006757-ATP generation from ADP\n",
      "Template rank:  806.0\n",
      "Median simulated rank:  1304.0\n",
      "GO:0047485-protein N-terminus binding\n",
      "Template rank:  808.0\n",
      "Median simulated rank:  1088.0\n",
      "GO:0046939-nucleotide phosphorylation\n",
      "Template rank:  809.0\n",
      "Median simulated rank:  1465.0\n",
      "GO:0036294-cellular response to decreased oxygen levels\n",
      "Template rank:  810.0\n",
      "Median simulated rank:  802.5\n",
      "GO:0046031-ADP metabolic process\n",
      "Template rank:  811.0\n",
      "Median simulated rank:  1506.0\n",
      "GO:0098813-nuclear chromosome segregation\n",
      "Template rank:  812.0\n",
      "Median simulated rank:  1342.0\n",
      "GO:0031330-negative regulation of cellular catabolic process\n",
      "Template rank:  813.0\n",
      "Median simulated rank:  1412.5\n",
      "GO:0016482-cytosolic transport\n",
      "Template rank:  816.0\n",
      "Median simulated rank:  1617.0\n",
      "GO:0006352-DNA-templated transcription, initiation\n",
      "Template rank:  817.0\n",
      "Median simulated rank:  1484.0\n",
      "GO:0051054-positive regulation of DNA metabolic process\n",
      "Template rank:  818.0\n",
      "Median simulated rank:  1399.5\n",
      "GO:0007059-chromosome segregation\n",
      "Template rank:  819.0\n",
      "Median simulated rank:  1228.0\n",
      "GO:0140097-catalytic activity, acting on DNA\n",
      "Template rank:  820.0\n",
      "Median simulated rank:  1293.0\n",
      "GO:0010498-proteasomal protein catabolic process\n",
      "Template rank:  821.0\n",
      "Median simulated rank:  1401.5\n",
      "GO:0019902-phosphatase binding\n",
      "Template rank:  822.0\n",
      "Median simulated rank:  1785.5\n",
      "GO:0006096-glycolytic process\n",
      "Template rank:  823.0\n",
      "Median simulated rank:  1216.0\n",
      "GO:0008168-methyltransferase activity\n",
      "Template rank:  824.0\n",
      "Median simulated rank:  2342.5\n",
      "GO:0008022-protein C-terminus binding\n",
      "Template rank:  825.0\n",
      "Median simulated rank:  1197.0\n",
      "GO:0031227-intrinsic component of endoplasmic reticulum membrane\n",
      "Template rank:  826.0\n",
      "Median simulated rank:  910.0\n",
      "GO:0009411-response to UV\n",
      "Template rank:  827.0\n",
      "Median simulated rank:  1470.0\n",
      "GO:0030705-cytoskeleton-dependent intracellular transport\n",
      "Template rank:  828.0\n",
      "Median simulated rank:  1828.0\n",
      "GO:0000209-protein polyubiquitination\n",
      "Template rank:  829.0\n",
      "Median simulated rank:  1620.0\n",
      "GO:0031461-cullin-RING ubiquitin ligase complex\n",
      "Template rank:  831.0\n",
      "Median simulated rank:  2049.0\n",
      "GO:0032259-methylation\n",
      "Template rank:  832.0\n",
      "Median simulated rank:  2043.0\n",
      "GO:1902275-regulation of chromatin organization\n",
      "Template rank:  834.0\n",
      "Median simulated rank:  1216.0\n",
      "GO:0005819-spindle\n",
      "Template rank:  835.0\n",
      "Median simulated rank:  1405.5\n",
      "GO:0071897-DNA biosynthetic process\n",
      "Template rank:  836.0\n",
      "Median simulated rank:  1193.5\n",
      "GO:2000045-regulation of G1/S transition of mitotic cell cycle\n",
      "Template rank:  837.0\n",
      "Median simulated rank:  889.0\n",
      "GO:0051101-regulation of DNA binding\n",
      "Template rank:  838.0\n",
      "Median simulated rank:  1656.0\n",
      "GO:1901796-regulation of signal transduction by p53 class mediator\n",
      "Template rank:  840.0\n",
      "Median simulated rank:  1530.0\n",
      "GO:0006888-ER to Golgi vesicle-mediated transport\n",
      "Template rank:  841.0\n",
      "Median simulated rank:  1399.0\n",
      "GO:0043687-post-translational protein modification\n",
      "Template rank:  843.0\n",
      "Median simulated rank:  1265.0\n",
      "GO:0042866-pyruvate biosynthetic process\n",
      "Template rank:  844.0\n",
      "Median simulated rank:  1239.0\n",
      "GO:0043484-regulation of RNA splicing\n",
      "Template rank:  845.0\n",
      "Median simulated rank:  1229.0\n",
      "GO:0050684-regulation of mRNA processing\n",
      "Template rank:  847.0\n",
      "Median simulated rank:  1260.0\n",
      "GO:0000793-condensed chromosome\n",
      "Template rank:  848.0\n",
      "Median simulated rank:  1134.5\n",
      "GO:0016887-ATPase activity\n",
      "Template rank:  849.0\n",
      "Median simulated rank:  1582.0\n",
      "GO:0042176-regulation of protein catabolic process\n",
      "Template rank:  850.0\n",
      "Median simulated rank:  1377.0\n",
      "GO:0016741-transferase activity, transferring one-carbon groups\n",
      "Template rank:  852.0\n",
      "Median simulated rank:  2178.0\n",
      "GO:1902115-regulation of organelle assembly\n",
      "Template rank:  853.0\n",
      "Median simulated rank:  1588.0\n",
      "GO:0043200-response to amino acid\n",
      "Template rank:  854.0\n",
      "Median simulated rank:  1382.0\n",
      "GO:0045732-positive regulation of protein catabolic process\n",
      "Template rank:  855.0\n",
      "Median simulated rank:  1405.0\n",
      "GO:0000082-G1/S transition of mitotic cell cycle\n",
      "Template rank:  856.0\n",
      "Median simulated rank:  1003.5\n",
      "GO:0009895-negative regulation of catabolic process\n",
      "Template rank:  857.0\n",
      "Median simulated rank:  1477.0\n",
      "GO:0033157-regulation of intracellular protein transport\n",
      "Template rank:  858.0\n",
      "Median simulated rank:  932.5\n",
      "GO:0045335-phagocytic vesicle\n",
      "Template rank:  859.0\n",
      "Median simulated rank:  949.0\n",
      "GO:0061695-transferase complex, transferring phosphorus-containing groups\n",
      "Template rank:  860.0\n",
      "Median simulated rank:  1623.0\n",
      "GO:0000151-ubiquitin ligase complex\n",
      "Template rank:  861.0\n",
      "Median simulated rank:  1727.0\n",
      "GO:0032549-ribonucleoside binding\n",
      "Template rank:  862.0\n",
      "Median simulated rank:  1632.0\n",
      "GO:0032550-purine ribonucleoside binding\n",
      "Template rank:  864.0\n",
      "Median simulated rank:  1611.0\n",
      "GO:0015980-energy derivation by oxidation of organic compounds\n",
      "Template rank:  865.0\n",
      "Median simulated rank:  926.0\n",
      "GO:0001883-purine nucleoside binding\n",
      "Template rank:  867.0\n",
      "Median simulated rank:  1599.0\n",
      "GO:0140056-organelle localization by membrane tethering\n",
      "Template rank:  868.0\n",
      "Median simulated rank:  1840.0\n",
      "GO:0018107-peptidyl-threonine phosphorylation\n",
      "Template rank:  869.0\n",
      "Median simulated rank:  1919.0\n",
      "GO:0015399-primary active transmembrane transporter activity\n",
      "Template rank:  870.0\n",
      "Median simulated rank:  1116.0\n",
      "GO:0043666-regulation of phosphoprotein phosphatase activity\n",
      "Template rank:  871.0\n",
      "Median simulated rank:  2381.0\n",
      "GO:0032984-protein-containing complex disassembly\n",
      "Template rank:  872.0\n",
      "Median simulated rank:  1006.5\n",
      "GO:0030178-negative regulation of Wnt signaling pathway\n",
      "Template rank:  873.0\n",
      "Median simulated rank:  1401.0\n",
      "GO:0001882-nucleoside binding\n",
      "Template rank:  874.0\n",
      "Median simulated rank:  1633.0\n",
      "GO:0070646-protein modification by small protein removal\n",
      "Template rank:  875.0\n",
      "Median simulated rank:  1667.5\n",
      "GO:0022406-membrane docking\n",
      "Template rank:  876.0\n",
      "Median simulated rank:  1668.0\n",
      "GO:0010506-regulation of autophagy\n",
      "Template rank:  877.0\n",
      "Median simulated rank:  1443.0\n",
      "GO:0031647-regulation of protein stability\n",
      "Template rank:  878.0\n",
      "Median simulated rank:  1378.0\n",
      "GO:0005525-GTP binding\n",
      "Template rank:  879.0\n",
      "Median simulated rank:  1618.0\n",
      "GO:0050839-cell adhesion molecule binding\n",
      "Template rank:  880.0\n",
      "Median simulated rank:  998.0\n",
      "GO:0000781-chromosome, telomeric region\n",
      "Template rank:  881.0\n",
      "Median simulated rank:  1175.5\n",
      "GO:0043401-steroid hormone mediated signaling pathway\n",
      "Template rank:  882.0\n",
      "Median simulated rank:  1458.0\n",
      "GO:0048285-organelle fission\n",
      "Template rank:  883.0\n",
      "Median simulated rank:  1483.5\n",
      "GO:0031331-positive regulation of cellular catabolic process\n",
      "Template rank:  885.0\n",
      "Median simulated rank:  1342.5\n",
      "GO:0009260-ribonucleotide biosynthetic process\n",
      "Template rank:  886.0\n",
      "Median simulated rank:  1470.0\n",
      "GO:0016579-protein deubiquitination\n",
      "Template rank:  887.0\n",
      "Median simulated rank:  1727.0\n",
      "GO:0002020-protease binding\n",
      "Template rank:  888.0\n",
      "Median simulated rank:  1004.5\n",
      "GO:0051427-hormone receptor binding\n",
      "Template rank:  889.0\n",
      "Median simulated rank:  979.0\n",
      "GO:1903320-regulation of protein modification by small protein conjugation or removal\n",
      "Template rank:  890.0\n",
      "Median simulated rank:  1557.0\n",
      "GO:0006520-cellular amino acid metabolic process\n",
      "Template rank:  891.0\n",
      "Median simulated rank:  1146.5\n",
      "GO:0010970-transport along microtubule\n",
      "Template rank:  892.5\n",
      "Median simulated rank:  1767.5\n",
      "GO:0099111-microtubule-based transport\n",
      "Template rank:  892.5\n",
      "Median simulated rank:  1767.5\n",
      "GO:0046390-ribose phosphate biosynthetic process\n",
      "Template rank:  894.0\n",
      "Median simulated rank:  1450.0\n",
      "GO:0009152-purine ribonucleotide biosynthetic process\n",
      "Template rank:  895.0\n",
      "Median simulated rank:  1454.0\n",
      "GO:0007050-cell cycle arrest\n",
      "Template rank:  896.0\n",
      "Median simulated rank:  1809.0\n",
      "GO:0016607-nuclear speck\n",
      "Template rank:  897.0\n",
      "Median simulated rank:  1494.5\n",
      "GO:0034404-nucleobase-containing small molecule biosynthetic process\n",
      "Template rank:  898.0\n",
      "Median simulated rank:  1501.0\n",
      "GO:0002244-hematopoietic progenitor cell differentiation\n",
      "Template rank:  899.0\n",
      "Median simulated rank:  1066.5\n",
      "GO:0090305-nucleic acid phosphodiester bond hydrolysis\n",
      "Template rank:  900.0\n",
      "Median simulated rank:  1515.5\n",
      "GO:0018205-peptidyl-lysine modification\n",
      "Template rank:  901.0\n",
      "Median simulated rank:  1556.0\n",
      "GO:0036464-cytoplasmic ribonucleoprotein granule\n",
      "Template rank:  902.0\n",
      "Median simulated rank:  1549.0\n",
      "GO:0000280-nuclear division\n",
      "Template rank:  903.0\n",
      "Median simulated rank:  1496.5\n",
      "GO:0005741-mitochondrial outer membrane\n",
      "Template rank:  905.0\n",
      "Median simulated rank:  1156.0\n",
      "GO:0016573-histone acetylation\n",
      "Template rank:  906.0\n",
      "Median simulated rank:  1520.5\n",
      "GO:0035567-non-canonical Wnt signaling pathway\n",
      "Template rank:  907.0\n",
      "Median simulated rank:  1528.0\n",
      "GO:0003924-GTPase activity\n",
      "Template rank:  908.0\n",
      "Median simulated rank:  1631.0\n",
      "GO:0001726-ruffle\n",
      "Template rank:  909.0\n",
      "Median simulated rank:  1334.0\n",
      "GO:0010565-regulation of cellular ketone metabolic process\n",
      "Template rank:  910.0\n",
      "Median simulated rank:  935.0\n",
      "GO:0019001-guanyl nucleotide binding\n",
      "Template rank:  911.5\n",
      "Median simulated rank:  1679.5\n",
      "GO:0032561-guanyl ribonucleotide binding\n",
      "Template rank:  911.5\n",
      "Median simulated rank:  1679.5\n",
      "GO:0019827-stem cell population maintenance\n",
      "Template rank:  913.0\n",
      "Median simulated rank:  1105.0\n",
      "GO:0042623-ATPase activity, coupled\n",
      "Template rank:  914.0\n",
      "Median simulated rank:  1514.0\n",
      "GO:0006733-oxidoreduction coenzyme metabolic process\n",
      "Template rank:  915.0\n",
      "Median simulated rank:  1280.0\n",
      "GO:0090068-positive regulation of cell cycle process\n",
      "Template rank:  916.0\n",
      "Median simulated rank:  1515.0\n",
      "GO:0006475-internal protein amino acid acetylation\n",
      "Template rank:  917.0\n",
      "Median simulated rank:  1466.5\n",
      "GO:0098727-maintenance of cell number\n",
      "Template rank:  918.0\n",
      "Median simulated rank:  1106.5\n",
      "GO:0072522-purine-containing compound biosynthetic process\n",
      "Template rank:  919.0\n",
      "Median simulated rank:  1435.0\n",
      "GO:0003714-transcription corepressor activity\n",
      "Template rank:  920.0\n",
      "Median simulated rank:  1568.5\n",
      "GO:0010921-regulation of phosphatase activity\n",
      "Template rank:  921.0\n",
      "Median simulated rank:  2016.5\n",
      "GO:0019362-pyridine nucleotide metabolic process\n",
      "Template rank:  922.5\n",
      "Median simulated rank:  1156.0\n",
      "GO:0046496-nicotinamide nucleotide metabolic process\n",
      "Template rank:  922.5\n",
      "Median simulated rank:  1156.0\n",
      "GO:0009896-positive regulation of catabolic process\n",
      "Template rank:  924.0\n",
      "Median simulated rank:  1340.0\n",
      "GO:0042579-microbody\n",
      "Template rank:  925.5\n",
      "Median simulated rank:  1500.5\n",
      "GO:0005777-peroxisome\n",
      "Template rank:  925.5\n",
      "Median simulated rank:  1500.5\n",
      "GO:0016570-histone modification\n",
      "Template rank:  927.0\n",
      "Median simulated rank:  1592.0\n",
      "GO:0061919-process utilizing autophagic mechanism\n",
      "Template rank:  928.5\n",
      "Median simulated rank:  1486.0\n",
      "GO:0006914-autophagy\n",
      "Template rank:  928.5\n",
      "Median simulated rank:  1486.0\n",
      "GO:0051052-regulation of DNA metabolic process\n",
      "Template rank:  931.0\n",
      "Median simulated rank:  1562.0\n",
      "GO:0018393-internal peptidyl-lysine acetylation\n",
      "Template rank:  933.0\n",
      "Median simulated rank:  1475.5\n",
      "GO:0009166-nucleotide catabolic process\n",
      "Template rank:  934.0\n",
      "Median simulated rank:  1617.0\n",
      "GO:0032886-regulation of microtubule-based process\n",
      "Template rank:  935.0\n",
      "Median simulated rank:  1900.5\n",
      "GO:1905330-regulation of morphogenesis of an epithelium\n",
      "Template rank:  936.0\n",
      "Median simulated rank:  1331.0\n",
      "GO:0090575-RNA polymerase II transcription factor complex\n",
      "Template rank:  937.0\n",
      "Median simulated rank:  1500.0\n",
      "GO:0007163-establishment or maintenance of cell polarity\n",
      "Template rank:  938.0\n",
      "Median simulated rank:  1624.0\n",
      "GO:0030177-positive regulation of Wnt signaling pathway\n",
      "Template rank:  939.0\n",
      "Median simulated rank:  1599.0\n",
      "GO:0006310-DNA recombination\n",
      "Template rank:  940.0\n",
      "Median simulated rank:  1452.0\n",
      "GO:0000922-spindle pole\n",
      "Template rank:  941.0\n",
      "Median simulated rank:  1413.0\n",
      "GO:0006164-purine nucleotide biosynthetic process\n",
      "Template rank:  942.0\n",
      "Median simulated rank:  1460.0\n",
      "GO:0043902-positive regulation of multi-organism process\n",
      "Template rank:  944.0\n",
      "Median simulated rank:  1522.0\n",
      "GO:1901292-nucleoside phosphate catabolic process\n",
      "Template rank:  945.0\n",
      "Median simulated rank:  1537.0\n",
      "GO:0016569-covalent chromatin modification\n",
      "Template rank:  946.0\n",
      "Median simulated rank:  1662.0\n",
      "GO:0045787-positive regulation of cell cycle\n",
      "Template rank:  947.0\n",
      "Median simulated rank:  1532.0\n",
      "GO:0072524-pyridine-containing compound metabolic process\n",
      "Template rank:  948.0\n",
      "Median simulated rank:  1095.0\n",
      "GO:0035770-ribonucleoprotein granule\n",
      "Template rank:  949.0\n",
      "Median simulated rank:  1531.5\n",
      "GO:0031396-regulation of protein ubiquitination\n",
      "Template rank:  950.0\n",
      "Median simulated rank:  1866.0\n",
      "GO:2001235-positive regulation of apoptotic signaling pathway\n",
      "Template rank:  951.0\n",
      "Median simulated rank:  1141.5\n",
      "GO:0005802-trans-Golgi network\n",
      "Template rank:  953.0\n",
      "Median simulated rank:  1791.0\n",
      "GO:0030522-intracellular receptor signaling pathway\n",
      "Template rank:  954.0\n",
      "Median simulated rank:  1477.0\n",
      "GO:0034329-cell junction assembly\n",
      "Template rank:  955.0\n",
      "Median simulated rank:  1377.0\n",
      "GO:0046165-alcohol biosynthetic process\n",
      "Template rank:  956.0\n",
      "Median simulated rank:  1571.0\n",
      "GO:0043543-protein acylation\n",
      "Template rank:  957.0\n",
      "Median simulated rank:  1662.5\n",
      "GO:1901293-nucleoside phosphate biosynthetic process\n",
      "Template rank:  958.0\n",
      "Median simulated rank:  1448.5\n",
      "GO:0006575-cellular modified amino acid metabolic process\n",
      "Template rank:  959.0\n",
      "Median simulated rank:  1097.0\n",
      "GO:0030674-protein binding, bridging\n",
      "Template rank:  960.0\n",
      "Median simulated rank:  1564.5\n",
      "GO:0005635-nuclear envelope\n",
      "Template rank:  961.0\n",
      "Median simulated rank:  1609.0\n",
      "GO:0009165-nucleotide biosynthetic process\n",
      "Template rank:  962.0\n",
      "Median simulated rank:  1451.0\n",
      "GO:0045862-positive regulation of proteolysis\n",
      "Template rank:  963.0\n",
      "Median simulated rank:  1417.5\n",
      "GO:0006694-steroid biosynthetic process\n",
      "Template rank:  964.0\n",
      "Median simulated rank:  1280.5\n",
      "GO:2001233-regulation of apoptotic signaling pathway\n",
      "Template rank:  965.0\n",
      "Median simulated rank:  1157.0\n",
      "GO:0038061-NIK/NF-kappaB signaling\n",
      "Template rank:  966.0\n",
      "Median simulated rank:  990.0\n",
      "GO:2001020-regulation of response to DNA damage stimulus\n",
      "Template rank:  967.0\n",
      "Median simulated rank:  1520.0\n",
      "GO:0006473-protein acetylation\n",
      "Template rank:  968.0\n",
      "Median simulated rank:  1488.0\n",
      "GO:2001234-negative regulation of apoptotic signaling pathway\n",
      "Template rank:  969.0\n",
      "Median simulated rank:  1095.0\n",
      "GO:0007568-aging\n",
      "Template rank:  970.0\n",
      "Median simulated rank:  1054.0\n",
      "GO:0009408-response to heat\n",
      "Template rank:  971.0\n",
      "Median simulated rank:  1392.0\n",
      "GO:0009108-coenzyme biosynthetic process\n",
      "Template rank:  972.0\n",
      "Median simulated rank:  1494.0\n",
      "GO:0031968-organelle outer membrane\n",
      "Template rank:  973.0\n",
      "Median simulated rank:  1311.0\n",
      "GO:0042180-cellular ketone metabolic process\n",
      "Template rank:  974.0\n",
      "Median simulated rank:  1064.0\n",
      "GO:0051495-positive regulation of cytoskeleton organization\n",
      "Template rank:  975.0\n",
      "Median simulated rank:  1701.0\n",
      "GO:0034599-cellular response to oxidative stress\n",
      "Template rank:  976.0\n",
      "Median simulated rank:  1089.0\n",
      "GO:0004518-nuclease activity\n",
      "Template rank:  977.0\n",
      "Median simulated rank:  1914.0\n",
      "GO:0060090-molecular adaptor activity\n",
      "Template rank:  978.0\n",
      "Median simulated rank:  1455.0\n",
      "GO:0006732-coenzyme metabolic process\n",
      "Template rank:  979.0\n",
      "Median simulated rank:  1302.5\n",
      "GO:1903829-positive regulation of cellular protein localization\n",
      "Template rank:  980.0\n",
      "Median simulated rank:  1240.5\n",
      "GO:1990778-protein localization to cell periphery\n",
      "Template rank:  981.0\n",
      "Median simulated rank:  2062.5\n",
      "GO:0031334-positive regulation of protein complex assembly\n",
      "Template rank:  982.0\n",
      "Median simulated rank:  1408.0\n",
      "GO:0042393-histone binding\n",
      "Template rank:  983.0\n",
      "Median simulated rank:  1642.0\n",
      "GO:0016747-transferase activity, transferring acyl groups other than amino-acyl groups\n",
      "Template rank:  985.0\n",
      "Median simulated rank:  1889.0\n",
      "GO:0070507-regulation of microtubule cytoskeleton organization\n",
      "Template rank:  986.0\n",
      "Median simulated rank:  1556.0\n",
      "GO:0060828-regulation of canonical Wnt signaling pathway\n",
      "Template rank:  988.0\n",
      "Median simulated rank:  2018.0\n",
      "GO:0034330-cell junction organization\n",
      "Template rank:  989.0\n",
      "Median simulated rank:  1419.5\n",
      "GO:0043254-regulation of protein complex assembly\n",
      "Template rank:  991.0\n",
      "Median simulated rank:  1577.0\n",
      "GO:0005788-endoplasmic reticulum lumen\n",
      "Template rank:  992.0\n",
      "Median simulated rank:  1110.0\n",
      "GO:0044272-sulfur compound biosynthetic process\n",
      "Template rank:  993.0\n",
      "Median simulated rank:  1433.0\n",
      "GO:0048863-stem cell differentiation\n",
      "Template rank:  994.0\n",
      "Median simulated rank:  1727.5\n",
      "GO:0010639-negative regulation of organelle organization\n",
      "Template rank:  997.0\n",
      "Median simulated rank:  1582.0\n",
      "GO:0006979-response to oxidative stress\n",
      "Template rank:  998.0\n",
      "Median simulated rank:  1171.0\n",
      "GO:0071383-cellular response to steroid hormone stimulus\n",
      "Template rank:  1001.0\n",
      "Median simulated rank:  1383.0\n",
      "GO:0005765-lysosomal membrane\n",
      "Template rank:  1002.0\n",
      "Median simulated rank:  1542.0\n",
      "GO:0001701-in utero embryonic development\n",
      "Template rank:  1003.0\n",
      "Median simulated rank:  1553.5\n",
      "GO:0072659-protein localization to plasma membrane\n",
      "Template rank:  1004.0\n",
      "Median simulated rank:  2017.0\n",
      "GO:0098852-lytic vacuole membrane\n",
      "Template rank:  1005.0\n",
      "Median simulated rank:  1546.0\n",
      "GO:0005774-vacuolar membrane\n",
      "Template rank:  1006.0\n",
      "Median simulated rank:  1537.0\n",
      "GO:0060070-canonical Wnt signaling pathway\n",
      "Template rank:  1007.0\n",
      "Median simulated rank:  2167.0\n",
      "GO:0051188-cofactor biosynthetic process\n",
      "Template rank:  1008.0\n",
      "Median simulated rank:  1513.0\n",
      "GO:0031965-nuclear membrane\n",
      "Template rank:  1009.0\n",
      "Median simulated rank:  1864.0\n",
      "GO:0018209-peptidyl-serine modification\n",
      "Template rank:  1010.0\n",
      "Median simulated rank:  2160.0\n",
      "GO:0018105-peptidyl-serine phosphorylation\n",
      "Template rank:  1011.0\n",
      "Median simulated rank:  2141.0\n",
      "GO:0010256-endomembrane system organization\n",
      "Template rank:  1012.0\n",
      "Median simulated rank:  1771.5\n",
      "GO:0010008-endosome membrane\n",
      "Template rank:  1013.0\n",
      "Median simulated rank:  1524.5\n",
      "GO:0005874-microtubule\n",
      "Template rank:  1014.0\n",
      "Median simulated rank:  1991.5\n",
      "GO:0051098-regulation of binding\n",
      "Template rank:  1015.0\n",
      "Median simulated rank:  1536.0\n",
      "GO:0001666-response to hypoxia\n",
      "Template rank:  1016.0\n",
      "Median simulated rank:  1024.0\n",
      "GO:0030111-regulation of Wnt signaling pathway\n",
      "Template rank:  1017.0\n",
      "Median simulated rank:  2000.0\n",
      "GO:0004842-ubiquitin-protein transferase activity\n",
      "Template rank:  1018.0\n",
      "Median simulated rank:  2133.5\n",
      "GO:0016050-vesicle organization\n",
      "Template rank:  1019.0\n",
      "Median simulated rank:  1443.0\n",
      "GO:0019058-viral life cycle\n",
      "Template rank:  1020.0\n",
      "Median simulated rank:  1132.5\n",
      "GO:0043900-regulation of multi-organism process\n",
      "Template rank:  1021.0\n",
      "Median simulated rank:  1230.0\n",
      "GO:0036293-response to decreased oxygen levels\n",
      "Template rank:  1022.0\n",
      "Median simulated rank:  1018.0\n",
      "GO:0070482-response to oxygen levels\n",
      "Template rank:  1023.0\n",
      "Median simulated rank:  1071.0\n",
      "GO:0062012-regulation of small molecule metabolic process\n",
      "Template rank:  1024.0\n",
      "Median simulated rank:  1401.5\n",
      "GO:0032386-regulation of intracellular transport\n",
      "Template rank:  1025.0\n",
      "Median simulated rank:  1365.0\n",
      "GO:0031252-cell leading edge\n",
      "Template rank:  1026.0\n",
      "Median simulated rank:  1464.0\n",
      "GO:0016053-organic acid biosynthetic process\n",
      "Template rank:  1027.0\n",
      "Median simulated rank:  1369.0\n",
      "GO:0046394-carboxylic acid biosynthetic process\n",
      "Template rank:  1028.0\n",
      "Median simulated rank:  1361.0\n"
     ]
    }
   ],
   "source": [
    "# Compare template rank vs median simulated rank \n",
    "# Print those pathways where template rank < simulated rank (template is higher ranked than simulated)\n",
    "# Some of these are pathways that are potentially template-specific\n",
    "\n",
    "for key, val in template_rank_dict.items():\n",
    "    if key in simulated_rank_dict.keys():\n",
    "        diff_rank = val - np.median(simulated_rank_dict[key])\n",
    "        if diff_rank < 10:\n",
    "            print(key)\n",
    "            print(\"Template rank: \", val)\n",
    "            print(\"Median simulated rank: \", np.median(simulated_rank_dict[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GO:0019774-proteasome core complex, beta-subunit complex\n",
      "Template rank:  2.0\n",
      "Median simulated rank:  1.0\n",
      "GO:2000628-regulation of miRNA metabolic process\n",
      "Template rank:  9.0\n",
      "Median simulated rank:  17.0\n",
      "GO:0030952-establishment or maintenance of cytoskeleton polarity\n",
      "Template rank:  14.0\n",
      "Median simulated rank:  20.5\n",
      "GO:0090151-establishment of protein localization to mitochondrial membrane\n",
      "Template rank:  32.0\n",
      "Median simulated rank:  33.0\n",
      "GO:0051095-regulation of helicase activity\n",
      "Template rank:  33.0\n",
      "Median simulated rank:  35.0\n",
      "GO:0000940-condensed chromosome outer kinetochore\n",
      "Template rank:  42.0\n",
      "Median simulated rank:  51.0\n"
     ]
    }
   ],
   "source": [
    "# Compare template rank vs median simulated rank \n",
    "# Print those pathways that ranked high in template experiment AND have similar ranks in simulated experiment\n",
    "for key, val in template_rank_dict.items():\n",
    "    if key in simulated_rank_dict.keys():\n",
    "        diff_rank = val - np.median(simulated_rank_dict[key])\n",
    "        if (val < 50) and (abs(diff_rank) < 10):\n",
    "            print(key)\n",
    "            print(\"Template rank: \", val)\n",
    "            print(\"Median simulated rank: \", np.median(simulated_rank_dict[key]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are a few immune pathways that are template-enriched but there are others that look to be involved in cellular maintenancem, which I wouldn't expect to be specific to these experiment.\n",
    "\n",
    "For GSEA I am currently using the p-values from the DE analysis to rank genes, should I try using avg gene expression instead..? Maybe p-values don't cluster genes as much..?\n",
    "\n",
    "Note: this notebook takes ~1hr to run\n",
    "\n",
    "**Check:**\n",
    "* Are there template-specific pathways (i.e. pathways specific to Dexamethasone treatment) that we can use as a positive?\n",
    "* Are there agnostic pathways that we can use as a negative control?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:ranked_pathways] *",
   "language": "python",
   "name": "conda-env-ranked_pathways-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
