{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gene set enrichment analysis\n",
    "\n",
    "**Goal:** To detect modest but coordinated changes in prespecified sets of related genes (i.e. those genes in the same pathway or share the same GO term).\n",
    "\n",
    "1. Ranks all genes based using DE association statistics. In this case we used the p-value scores to rank genes. logFC returned error -- need to look into this.\n",
    "2. An enrichment score (ES) is defined as the maximum distance from the middle of the ranked list. Thus, the enrichment score indicates whether the genes contained in a gene set are clustered towards the beginning or the end of the ranked list (indicating a correlation with change in expression). \n",
    "3. Estimate the statistical significance of the ES by a phenotypic-based permutation test in order to produce a null distribution for the ES( i.e. scores based on permuted phenotype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: \n",
      "examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.\n",
      "  \"found relative to the 'datapath' directory.\".format(key))\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%load_ext rpy2.ipython\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import seaborn as sns\n",
    "import rpy2.robjects\n",
    "from rpy2.robjects.packages import importr\n",
    "from rpy2.robjects import pandas2ri\n",
    "pandas2ri.activate()\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "\n",
    "\n",
    "def fxn():\n",
    "    warnings.warn(\"deprecated\", DeprecationWarning)\n",
    "\n",
    "with warnings.catch_warnings():\n",
    "    warnings.simplefilter(\"ignore\")\n",
    "    fxn()\n",
    "\n",
    "sys.path.append(\"../\")\n",
    "from functions import utils\n",
    "\n",
    "from numpy.random import seed\n",
    "randomState = 123\n",
    "seed(randomState)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in config variables\n",
    "base_dir = os.path.abspath(os.path.join(os.getcwd(),\"../\"))\n",
    "\n",
    "config_file = os.path.abspath(os.path.join(base_dir,\n",
    "                                           \"Rank_pathways\",\n",
    "                                           \"init_config.tsv\"))\n",
    "params = utils.read_config(config_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load params\n",
    "local_dir = params[\"local_dir\"]\n",
    "dataset_name = params['dataset_name']\n",
    "num_runs = params['num_simulated']\n",
    "project_id = params['project_id']\n",
    "\n",
    "rerun_template = True\n",
    "rerun_simulated = False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install R libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Select 59\n",
    "# Run one time\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "#BiocManager::install('clusterProfiler')\n",
    "#BiocManager::install(\"biomaRt\")\n",
    "#install.packages(\"devtools\")\n",
    "#BiocManager::install(\"GSA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Registered S3 method overwritten by 'enrichplot':\n",
      "  method               from\n",
      "  fortify.enrichResult DOSE\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: clusterProfiler v3.14.3  For help: https://guangchuangyu.github.io/software/clusterProfiler\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: AnnotationDbi\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: stats4\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: BiocGenerics\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: parallel\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: Biobase\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: IRanges\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: S4Vectors\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: DOSE v3.12.0  For help: https://guangchuangyu.github.io/software/DOSE\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Loading required package: Rcpp\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "suppressWarnings(library(\"clusterProfiler\"))\n",
    "suppressWarnings(library(\"org.Hs.eg.db\"))\n",
    "suppressWarnings(library(\"DOSE\"))\n",
    "suppressWarnings(library(\"biomaRt\"))\n",
    "suppressWarnings(library(\"fgsea\"))\n",
    "suppressWarnings(library(\"GSA\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get pathway enrichment for template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load Hallmark pathways database used by Powers et. al.\n",
    "# https://github.com/CostelloLab/GSEA-InContext/blob/master/data/gene_sets/hallmarks.gmt\n",
    "hallmark_DB_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"hallmark_DB.gmt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_real.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=>-----------------------------]   6% eta: 13s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [==>----------------------------]  11% eta: 10s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [====>--------------------------]  17% eta:  9s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [======>------------------------]  22% eta:  8s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [========>----------------------]  28% eta:  7s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=========>---------------------]  33% eta:  6s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [===========>-------------------]  39% eta:  6s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=============>-----------------]  44% eta:  5s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [===============>---------------]  50% eta:  5s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [================>--------------]  56% eta:  4s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [==================>------------]  61% eta:  4s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [====================>----------]  67% eta:  3s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=====================>---------]  72% eta:  2s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=======================>-------]  78% eta:  2s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [=========================>-----]  83% eta:  1s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [===========================>---]  89% eta:  1s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Batch submitting query [============================>--]  94% eta:  0s\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "                                                                      ages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: \n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/robjects/pandas2ri.py:191: FutureWarning: from_items is deprecated. Please use DataFrame.from_dict(dict(items), ...) instead. DataFrame.from_dict(OrderedDict(items)) may be used to preserve the key order.\n",
      "  res = PandasDataFrame.from_items(items)\n"
     ]
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file -o gene_id_mapping\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "gene_id_mapping <- get_ensembl_symbol_mapping(template_DE_stats_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(17811, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>C1orf112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000001617</th>\n",
       "      <td>SEMA3F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000004776</th>\n",
       "      <td>HSPB6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000004799</th>\n",
       "      <td>PDK4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000006638</th>\n",
       "      <td>TBXA2R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                hgnc_symbol\n",
       "ensembl_gene_id            \n",
       "ENSG00000000460    C1orf112\n",
       "ENSG00000001617      SEMA3F\n",
       "ENSG00000004776       HSPB6\n",
       "ENSG00000004799        PDK4\n",
       "ENSG00000006638      TBXA2R"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Set ensembl id as index\n",
    "gene_id_mapping.set_index(\"ensembl_gene_id\", inplace=True)\n",
    "print(gene_id_mapping.shape)\n",
    "gene_id_mapping.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace ensembl ids with gene symbols\n",
    "utils.replace_ensembl_ids(template_DE_stats_file,\n",
    "                          gene_id_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Shuffle ranking\n",
    "# Create 5 shuffled datasets\n",
    "for r in range(5):\n",
    "    template_DE_stats = pd.read_csv(\n",
    "    template_DE_stats_file,\n",
    "    header=0,\n",
    "    sep='\\t',\n",
    "    index_col=0)\n",
    "    \n",
    "    shuffled_template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_shuffled_\"+str(r)+\".txt\")\n",
    "    \n",
    "    shuffled_template_DE_stats = template_DE_stats.iloc[np.random.permutation(len(template_DE_stats))]\n",
    "    shuffled_template_DE_stats.index = template_DE_stats.index\n",
    "    \n",
    "    # Save \n",
    "    shuffled_template_DE_stats.to_csv(shuffled_template_DE_stats_file, float_format='%.5f', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Test shuffled files\n",
    "r = 2\n",
    "shuffled_template_DE_stats_file = os.path.join(\n",
    "    local_dir,\n",
    "    \"DE_stats\",\n",
    "    \"DE_stats_template_data_\"+project_id+\"_shuffled_\"+str(r)+\".txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%R\n",
    "#wpgmtfile <- system.file(\"extdata/wikipathways-20180810-gmt-Homo_sapiens.gmt\", package=\"clusterProfiler\")\n",
    "#wp2gene <- read.gmt(wpgmtfile)\n",
    "#wpid2gene <- wp2gene %>% dplyr::select(wpid, gene) #TERM2GENE\n",
    "#wp2gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       " SNRK-AS1     ITIH5     HYOU1     GATA6  KIAA0040     PXDC1 \n",
       "-21.02853 -20.76592  17.94308 -17.44069 -17.14382 -17.10286 \n",
       "12345678910111213141516171819202122232425262728293031323334353637383940414243444546474849501\n",
       "2\n",
       "3\n",
       "4\n",
       "5\n",
       "6\n",
       "7\n",
       "8\n",
       "9\n",
       "10\n",
       "11\n",
       "12\n",
       "13\n",
       "14\n",
       "15\n",
       "16\n",
       "17\n",
       "18\n",
       "19\n",
       "20\n",
       "21\n",
       "22\n",
       "23\n",
       "24\n",
       "25\n",
       "26\n",
       "27\n",
       "28\n",
       "29\n",
       "30\n",
       "31\n",
       "32\n",
       "33\n",
       "34\n",
       "35\n",
       "36\n",
       "37\n",
       "38\n",
       "39\n",
       "40\n",
       "41\n",
       "42\n",
       "43\n",
       "44\n",
       "45\n",
       "46\n",
       "47\n",
       "48\n",
       "49\n",
       "                                       pathway         pval         padj\n",
       " 1:                      HALLMARK_ADIPOGENESIS 3.609267e-01 0.4196821913\n",
       " 2:               HALLMARK_ALLOGRAFT_REJECTION 2.328042e-03 0.0044770045\n",
       " 3:                 HALLMARK_ANDROGEN_RESPONSE 5.101362e-01 0.5426980646\n",
       " 4:                      HALLMARK_ANGIOGENESIS 1.813292e-02 0.0283326826\n",
       " 5:                   HALLMARK_APICAL_JUNCTION 7.055669e-05 0.0004161811\n",
       " 6:                    HALLMARK_APICAL_SURFACE 1.693193e-02 0.0273095704\n",
       " 7:                         HALLMARK_APOPTOSIS 7.243752e-05 0.0004161811\n",
       " 8:              HALLMARK_BILE_ACID_METABOLISM 1.528750e-02 0.0254791603\n",
       " 9:           HALLMARK_CHOLESTEROL_HOMEOSTASIS 5.036845e-02 0.0699561818\n",
       "10:                       HALLMARK_COAGULATION 3.707549e-04 0.0009268871\n",
       "11:                        HALLMARK_COMPLEMENT 7.028395e-05 0.0004161811\n",
       "12:                        HALLMARK_DNA_REPAIR 4.662729e-04 0.0011101736\n",
       "13:                       HALLMARK_E2F_TARGETS 1.709986e-04 0.0004831758\n",
       "14: HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION 1.406371e-04 0.0004831758\n",
       "15:           HALLMARK_ESTROGEN_RESPONSE_EARLY 1.128907e-03 0.0023518897\n",
       "16:            HALLMARK_ESTROGEN_RESPONSE_LATE 2.817329e-01 0.3353962766\n",
       "17:             HALLMARK_FATTY_ACID_METABOLISM 7.717582e-01 0.7854022102\n",
       "18:                    HALLMARK_G2M_CHECKPOINT 1.709110e-04 0.0004831758\n",
       "19:                        HALLMARK_GLYCOLYSIS 1.739433e-04 0.0004831758\n",
       "20:                HALLMARK_HEDGEHOG_SIGNALING 5.121528e-03 0.0094843107\n",
       "21:                   HALLMARK_HEME_METABOLISM 9.805996e-02 0.1325134658\n",
       "22:                           HALLMARK_HYPOXIA 1.130422e-02 0.0194900430\n",
       "23:               HALLMARK_IL2_STAT5_SIGNALING 7.031854e-05 0.0004161811\n",
       "24:           HALLMARK_IL6_JAK_STAT3_SIGNALING 7.880221e-05 0.0004161811\n",
       "25:             HALLMARK_INFLAMMATORY_RESPONSE 7.020007e-05 0.0004161811\n",
       "26:         HALLMARK_INTERFERON_ALPHA_RESPONSE 3.830799e-01 0.4256443578\n",
       "27:         HALLMARK_INTERFERON_GAMMA_RESPONSE 1.340388e-03 0.0026807760\n",
       "28:                 HALLMARK_KRAS_SIGNALING_DN 4.258867e-01 0.4629202991\n",
       "29:                 HALLMARK_KRAS_SIGNALING_UP 3.519392e-04 0.0009261557\n",
       "30:                   HALLMARK_MITOTIC_SPINDLE 2.352534e-02 0.0356444570\n",
       "31:                  HALLMARK_MTORC1_SIGNALING 1.716149e-04 0.0004831758\n",
       "32:                    HALLMARK_MYC_TARGETS_V1 1.690046e-04 0.0004831758\n",
       "33:                    HALLMARK_MYC_TARGETS_V2 9.225325e-03 0.0164737954\n",
       "34:                        HALLMARK_MYOGENESIS 7.028395e-05 0.0004161811\n",
       "35:                   HALLMARK_NOTCH_SIGNALING 1.625395e-01 0.2029429823\n",
       "36:         HALLMARK_OXIDATIVE_PHOSPHORYLATION 1.017122e-03 0.0022111338\n",
       "37:                       HALLMARK_P53_PATHWAY 1.360141e-01 0.1743770633\n",
       "38:               HALLMARK_PANCREAS_BETA_CELLS 7.854022e-01 0.7854022102\n",
       "39:                        HALLMARK_PEROXISOME 3.782125e-01 0.4256443578\n",
       "40:           HALLMARK_PI3K_AKT_MTOR_SIGNALING 5.505394e-01 0.5734785880\n",
       "41:                 HALLMARK_PROTEIN_SECRETION 1.409841e-04 0.0004831758\n",
       "42:   HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY 1.664132e-01 0.2029429823\n",
       "43:                   HALLMARK_SPERMATOGENESIS 3.067766e-02 0.0451141995\n",
       "44:                HALLMARK_TGF_BETA_SIGNALING 8.323622e-05 0.0004161811\n",
       "45:           HALLMARK_TNFA_SIGNALING_VIA_NFKB 7.020007e-05 0.0004161811\n",
       "46:         HALLMARK_UNFOLDED_PROTEIN_RESPONSE 1.445504e-04 0.0004831758\n",
       "47:                    HALLMARK_UV_RESPONSE_DN 7.383888e-05 0.0004161811\n",
       "48:                    HALLMARK_UV_RESPONSE_UP 4.351636e-02 0.0621662216\n",
       "49:        HALLMARK_WNT_BETA_CATENIN_SIGNALING 9.323614e-04 0.0021190032\n",
       "50:             HALLMARK_XENOBIOTIC_METABOLISM 1.021328e-01 0.1343851993\n",
       "                                       pathway         pval         padj\n",
       "            ES        NES nMoreExtreme size\n",
       " 1: -0.2356845 -1.0437155         5109  187\n",
       " 2: -0.3421751 -1.5170022           32  189\n",
       " 3: -0.2410511 -0.9723133         6592   95\n",
       " 4: -0.4721293 -1.5760351          208   35\n",
       " 5: -0.4172858 -1.8520944            0  191\n",
       " 6: -0.4480587 -1.5651077          199   43\n",
       " 7: -0.4042894 -1.7491664            0  156\n",
       " 8: -0.3525330 -1.4525806          200  110\n",
       " 9: -0.3627479 -1.3804518          621   67\n",
       "10: -0.3938375 -1.6692022            4  133\n",
       "11: -0.4258069 -1.8953096            0  195\n",
       "12:  0.3613435  1.6845343            2  138\n",
       "13:  0.5854966  2.8462232            0  185\n",
       "14: -0.3847806 -1.7105320            1  193\n",
       "15: -0.3518621 -1.5617160           15  191\n",
       "16: -0.2443892 -1.0847048         3992  191\n",
       "17: -0.2008310 -0.8637314        10569  150\n",
       "18:  0.5146512  2.4994745            0  184\n",
       "19:  0.4912689  2.4129760            0  196\n",
       "20: -0.5177539 -1.7181588           58   34\n",
       "21: -0.2748717 -1.2186188         1389  189\n",
       "22: -0.3211350 -1.4225851          159  188\n",
       "23: -0.4334384 -1.9268387            0  193\n",
       "24: -0.4903961 -1.9431835            0   85\n",
       "25: -0.5108042 -2.2769369            0  197\n",
       "26: -0.2598825 -1.0394826         4912   90\n",
       "27: -0.3512781 -1.5573593           18  189\n",
       "28: -0.2291065 -1.0133644         6027  185\n",
       "29: -0.3596439 -1.5978861            4  192\n",
       "30:  0.2664407  1.3047797          135  193\n",
       "31:  0.4171365  2.0362104            0  189\n",
       "32:  0.4594538  2.2262328            0  180\n",
       "33:  0.4031808  1.5941314           72   55\n",
       "34: -0.5396169 -2.4018897            0  195\n",
       "35: -0.3813346 -1.2455289         1850   32\n",
       "36:  0.3281309  1.5904738            5  181\n",
       "37: -0.2662767 -1.1805135         1927  189\n",
       "38: -0.2345319 -0.8019745         9167   39\n",
       "39:  0.2336956  1.0322795         2648   99\n",
       "40: -0.2334821 -0.9534724         7194  103\n",
       "41:  0.4186311  1.8281775            0   93\n",
       "42: -0.3481273 -1.2276875         1969   45\n",
       "43:  0.2880959  1.3325842          200  130\n",
       "44: -0.5820704 -2.1260683            0   54\n",
       "45: -0.6593546 -2.9391082            0  197\n",
       "46:  0.5007446  2.2328746            0  104\n",
       "47: -0.4566539 -1.9423319            0  137\n",
       "48: -0.3066427 -1.3188052          595  150\n",
       "49: -0.5319370 -1.8491153           10   42\n",
       "50: -0.2731517 -1.2136041         1450  192\n",
       "            ES        NES nMoreExtreme size\n",
       "                                       leadingEdge\n",
       " 1:          ITIH5,PPARG,LPL,HSPB8,ANGPT1,STOM,...\n",
       " 2:            APBB1,ETS1,KLRD1,NCR1,CFP,HLA-E,...\n",
       " 3:    AKAP12,SRF,PTPN21,ADAMTS1,TSC22D1,ZMIZ1,...\n",
       " 4:         THBD,LPL,S100A4,MSX1,KCNJ8,SLCO2A1,...\n",
       " 5:     ARHGEF6,ZYX,NFASC,MYH10,PECAM1,EPB41L2,...\n",
       " 6:          MAL,SCUBE1,NTNG1,MDGA1,CD160,SRPX,...\n",
       " 7:          FEZ1,RARA,NEDD9,SPTAN1,ANXA1,CAV1,...\n",
       " 8:           ABCA8,ABCA3,AQP9,RXRA,PECR,ABCA6,...\n",
       " 9:            PPARG,LPL,SEMA3B,LSS,PNRC1,LDLR,...\n",
       "10:         THBD,PECAM1,A2M,SERPING1,ANXA1,F10,...\n",
       "11:       SPOCK2,EHD1,GNAI2,DOCK4,SERPING1,F10,...\n",
       "12:            FEN1,ZWINT,TMED2,TP53,DDB1,NME4,...\n",
       "13:       TK1,CHEK1,CDCA3,ASF1B,RAD51AP1,MYBL2,...\n",
       "14:         ID2,ABI3BP,FBLN5,SLIT2,PMP22,SLIT3,...\n",
       "15:        SH3BP5,KLF4,CBFA2T3,CLIC3,DLC1,RARA,...\n",
       "16:          ID2,KLF4,CLIC3,ABCA3,SEMA3B,HSPB8,...\n",
       "17:      AOC3,GABARAPL1,CA4,SERINC1,INMT,ACADL,...\n",
       "18:          CCNF,CHEK1,INCENP,MYBL2,DKC1,PLK1,...\n",
       "19:       KDELR3,SLC37A4,IDH1,COG2,HSPA5,SPAG4,...\n",
       "20:         DPYSL2,CNTFR,ETS2,RTN1,PTCH1,CRMP1,...\n",
       "21:          GYPC,TNS1,SNCA,SDCBP,RANBP10,RCL1,...\n",
       "22:         AKAP12,DUSP1,S100A4,GPC3,KLF6,CAV1,...\n",
       "23: ITIH5,GLIPR2,AHNAK,GABARAPL1,DENND5A,PTH1R,...\n",
       "24:       MYD88,A2M,IL17RA,MAP3K8,CNTFR,ACVRL1,...\n",
       "25:            MARCO,EMP3,MEFV,GPC3,KLF6,ITGA5,...\n",
       "26:     GBP4,LAMP3,OASL,CCRL2,SLC25A28,TMEM140,...\n",
       "27:     LATS2,MYD88,P2RY14,SERPING1,VAMP5,GBP4,...\n",
       "28:         SYNPO,MEFV,RYR2,CNTFR,MTHFR,TCF7L1,...\n",
       "29:         ID2,KLF4,GYPC,TSPAN7,AKAP12,PECAM1,...\n",
       "30:         CKAP5,KNTC1,INCENP,PLK1,CDK1,CCNB2,...\n",
       "31:       SLC1A4,SHMT2,SLC37A4,GAPDH,CCNF,IDH1,...\n",
       "32:          SRPK1,CANX,PRDX3,CSTF2,PRDX4,TYMS,...\n",
       "33:        PRMT3,PLK1,DCTPP1,MCM4,EXOSC5,HSPD1,...\n",
       "34:         MEF2D,MYL3,TEAD4,FHL1,FXYD1,SPTAN1,...\n",
       "35:           DLL1,NOTCH1,ARRB1,HEYL,JAG1,WNT2,...\n",
       "36:            IDH1,AIFM1,MTX2,GPI,PRDX3,VDAC1,...\n",
       "37:   KLF4,S100A4,RXRA,TSC22D1,NOTCH1,PPP1R15A,...\n",
       "38:          AKT3,LMO2,FOXO1,PAX6,ABCC8,STXBP1,...\n",
       "39:       CRABP2,IDH1,IDE,PEX11B,SMARCC1,TOP2A,...\n",
       "40:  MYD88,DUSP3,ARHGDIA,MKNK2,CAB39L,TNFRSF1A,...\n",
       "41:         SEC24D,COG2,CLTC,TMED2,ARCN1,COPB1,...\n",
       "42:           PDLIM1,JUNB,CAT,CDKN2D,GPX3,MSRA,...\n",
       "43:            IDE,CDK1,CCNB2,NCAPH,BUB1,KIF2C,...\n",
       "44:        ID2,SPTBN1,ID1,JUNB,PPP1R15A,FKBP1A,...\n",
       "45:           ID2,KLF4,SIK1,EHD1,DENND5A,DUSP1,...\n",
       "46:   HYOU1,ALDH18A1,SLC1A4,KDELR3,PDIA6,SRPRB,...\n",
       "47:         FBLN5,GRK5,PTPRM,DLC1,DUSP1,TGFBR2,...\n",
       "48:          NR4A1,CDO1,JUNB,BTG2,NFKBIA,FGF18,...\n",
       "49:       NKD1,DLL1,NOTCH1,NOTCH4,FRAT1,CSNK1E,...\n",
       "50:           ID2,GABARAPL1,CDO1,F10,RBP4,AQP9,...\n",
       "                                       leadingEdge\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file -i hallmark_DB_file\n",
    "# Read in data\n",
    "DE_stats_data <- read.table(template_DE_stats_file, sep=\"\\t\", header=TRUE, row.names=NULL)\n",
    "\n",
    "# Sort genes by feature 1\n",
    "\n",
    "# feature 1: numeric vector\n",
    "# 5: p-values\n",
    "# 6: adjusted p-values\n",
    "# 2: logFC\n",
    "rank_genes <- as.numeric(as.character(DE_stats_data[,4]))\n",
    "\n",
    "#print(head(rank_genes))\n",
    "\n",
    "# feature 2: named vector of gene ids\n",
    "# Remove version from gene id\n",
    "DE_stats_data[,1] <- gsub(\"\\\\..*\",\"\", DE_stats_data[,1])\n",
    "\n",
    "names(rank_genes) <- as.character(DE_stats_data[,1])\n",
    "\n",
    "print(head(rank_genes))\n",
    "\n",
    "## feature 3: decreasing order\n",
    "rank_genes = sort(rank_genes, decreasing = TRUE)\n",
    "\n",
    "#pathway_DB_data <- read.gmt(hallmark_DB_file)\n",
    "pathway_DB_data <- GSA.read.gmt(hallmark_DB_file)\n",
    "pathway_parsed <- {}\n",
    "for (i in 1:length(pathway_DB_data$genesets)){\n",
    "pathway_parsed[pathway_DB_data$geneset.name[i]] <- as.list(pathway_DB_data$genesets[i])\n",
    "}\n",
    "#print(head(pathway_DB_data))\n",
    "# GSEA is a generic gene set enrichment function\n",
    "# Different backend methods can be applied depending on the \n",
    "# type of annotations\n",
    "# Here we will use fgsea\n",
    "#enrich_pathways <- GSEA(geneList=rank_genes, \n",
    "#                        TERM2GENE=pathway_DB_data,\n",
    "#                        nPerm=100000,\n",
    "#                        by='fgsea',\n",
    "#                        verbose=T)\n",
    "enrich_pathways <- fgsea(pathways=pathway_parsed,\n",
    "                         stats=rank_genes,\n",
    "                         nperm=20000)\n",
    "print(enrich_pathways)\n",
    "#plotEnrichment(pathway_parsed[[\"HALLMARK_P53_PATHWAY\"]], stats=rank_genes, gseaParam = 1, ticksSize = 0.2)\n",
    "#barplot(sort(rank_genes, decreasing = T))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*fgsea* \n",
    "\n",
    "For each of the input pathways, an ES value is calculated. Next, a number\n",
    "of random gene sets of the same size are generated, and for each of them\n",
    "an ES value is calculated. Then a P-value is estimated as the number of\n",
    "random gene sets with the same or more extreme ES value divided by the\n",
    "total number of generated gene sets\n",
    "\n",
    "What is the ES value that is used?\n",
    "Source code for *fgsea* is available [here](https://rdrr.io/bioc/fgsea/src/R/fgsea.R). This package was based on the method published by [Korotkevich et. al.](https://www.biorxiv.org/content/10.1101/060012v2.full.pdf).\n",
    "Blog about calulcation is [here](https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Read 50 records\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/alexandra/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Read 7424 items\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "12345678910111213141516171819202122232425262728293031323334353637383940414243444546474849501\n",
       "2\n",
       "3\n",
       "4\n",
       "5\n",
       "6\n",
       "7\n",
       "8\n",
       "9\n",
       "10\n",
       "11\n",
       "12\n",
       "13\n",
       "14\n",
       "15\n",
       "16\n",
       "17\n",
       "18\n",
       "19\n",
       "20\n",
       "21\n",
       "22\n",
       "23\n",
       "24\n",
       "25\n",
       "26\n",
       "27\n",
       "28\n",
       "29\n",
       "30\n",
       "31\n",
       "32\n",
       "33\n",
       "34\n",
       "35\n",
       "36\n",
       "37\n",
       "38\n",
       "39\n",
       "40\n",
       "41\n",
       "42\n",
       "43\n",
       "44\n",
       "45\n",
       "46\n",
       "47\n",
       "48\n",
       "49\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i template_DE_stats_file -i hallmark_DB_file -o template_enriched_pathways\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "template_enriched_pathways <- find_enriched_pathways(template_DE_stats_file, hallmark_DB_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(50, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pathway</th>\n",
       "      <th>pval</th>\n",
       "      <th>padj</th>\n",
       "      <th>ES</th>\n",
       "      <th>NES</th>\n",
       "      <th>nMoreExtreme</th>\n",
       "      <th>size</th>\n",
       "      <th>leadingEdge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>0.265973</td>\n",
       "      <td>0.550847</td>\n",
       "      <td>-0.544872</td>\n",
       "      <td>-1.121209</td>\n",
       "      <td>178.0</td>\n",
       "      <td>187</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HALLMARK_ALLOGRAFT_REJECTION</td>\n",
       "      <td>0.172107</td>\n",
       "      <td>0.537834</td>\n",
       "      <td>-0.583642</td>\n",
       "      <td>-1.200379</td>\n",
       "      <td>115.0</td>\n",
       "      <td>189</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HALLMARK_ANDROGEN_RESPONSE</td>\n",
       "      <td>0.724409</td>\n",
       "      <td>0.787402</td>\n",
       "      <td>-0.415369</td>\n",
       "      <td>-0.814069</td>\n",
       "      <td>459.0</td>\n",
       "      <td>95</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HALLMARK_ANGIOGENESIS</td>\n",
       "      <td>0.096471</td>\n",
       "      <td>0.537834</td>\n",
       "      <td>0.759664</td>\n",
       "      <td>1.478738</td>\n",
       "      <td>40.0</td>\n",
       "      <td>35</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HALLMARK_APICAL_JUNCTION</td>\n",
       "      <td>0.158754</td>\n",
       "      <td>0.537834</td>\n",
       "      <td>-0.590486</td>\n",
       "      <td>-1.214244</td>\n",
       "      <td>106.0</td>\n",
       "      <td>191</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        pathway      pval      padj        ES       NES  \\\n",
       "0         HALLMARK_ADIPOGENESIS  0.265973  0.550847 -0.544872 -1.121209   \n",
       "1  HALLMARK_ALLOGRAFT_REJECTION  0.172107  0.537834 -0.583642 -1.200379   \n",
       "2    HALLMARK_ANDROGEN_RESPONSE  0.724409  0.787402 -0.415369 -0.814069   \n",
       "3         HALLMARK_ANGIOGENESIS  0.096471  0.537834  0.759664  1.478738   \n",
       "4      HALLMARK_APICAL_JUNCTION  0.158754  0.537834 -0.590486 -1.214244   \n",
       "\n",
       "   nMoreExtreme  size leadingEdge  \n",
       "0         178.0   187        None  \n",
       "1         115.0   189        None  \n",
       "2         459.0    95        None  \n",
       "3          40.0    35        None  \n",
       "4         106.0   191        None  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get pathway enrichment for simulated experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace ensembl ids with gene symbols\n",
    "if rerun_simulated:\n",
    "    for i in range(num_runs):\n",
    "        simulated_DE_stats_file = os.path.join(\n",
    "            local_dir, \n",
    "            \"DE_stats\",\n",
    "            \"DE_stats_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "\n",
    "        utils.replace_ensembl_ids(simulated_DE_stats_file,\n",
    "                                  gene_id_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i project_id -i local_dir -i hallmark_DB_file -i num_runs -i rerun_simulated\n",
    "\n",
    "source('../functions/GSEA_analysis.R')\n",
    "\n",
    "for (i in 0:(num_runs-1)){\n",
    "    simulated_DE_stats_file <- paste(local_dir, \n",
    "                                 \"DE_stats/DE_stats_simulated_data_\", \n",
    "                                 project_id,\n",
    "                                 \"_\", \n",
    "                                 i,\n",
    "                                 \".txt\",\n",
    "                                 sep=\"\")\n",
    "    \n",
    "    out_file = paste(local_dir, \n",
    "                     \"GSEA_stats/GSEA_simulated_data_\",\n",
    "                     project_id,\n",
    "                     \"_\",\n",
    "                     i,\n",
    "                     \".txt\", \n",
    "                     sep=\"\")\n",
    "    \n",
    "    if (rerun_simulated){\n",
    "        enriched_pathways <- find_enriched_pathways(simulated_DE_stats_file, hallmark_DB_file) \n",
    "        #print(head(enriched_pathways))\n",
    "    \n",
    "        write.table(enriched_pathways, file = out_file, row.names = T, sep = \"\\t\")\n",
    "        }\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check**\n",
    "\n",
    "Again, we want to compare our ranked pathways found against what was reported in the original publication."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get statistics for enriched pathways\n",
    "Examine the enriched pathways identified from template experiment -- How are these enriched pathways ranked in the simulated experiments?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Template experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_to_rank = 'enrichmentScore'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'enrichmentScore'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3077\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3078\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3079\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'enrichmentScore'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-20-8f3e419206ff>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Get ranks of template experiment\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;31m# Rank pathways by highest enrichment score\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mtemplate_enriched_pathways\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ranking'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtemplate_enriched_pathways\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcol_to_rank\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrank\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mascending\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mtemplate_enriched_pathways\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtemplate_enriched_pathways\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mby\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcol_to_rank\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mascending\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2686\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2687\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2688\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2690\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2693\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2694\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2695\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2696\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2697\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionality\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   2487\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2488\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2489\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2490\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2491\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/internals.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   4113\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4114\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4115\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4116\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4117\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ranked_pathways/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3078\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3079\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3080\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3081\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3082\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtolerance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtolerance\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'enrichmentScore'"
     ]
    }
   ],
   "source": [
    "# Get ranks of template experiment\n",
    "# Rank pathways by highest enrichment score\n",
    "template_enriched_pathways['ranking'] = template_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "template_enriched_pathways = template_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "\n",
    "# Set index to GO ID\n",
    "template_enriched_pathways.set_index(\"ID\", inplace=True)\n",
    "print(template_enriched_pathways.shape)\n",
    "template_enriched_pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check that GO IDs are unique\n",
    "template_enriched_pathways.index.nunique() == len(template_enriched_pathways)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simulated experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Concatenate simulated experiments\n",
    "simulated_enriched_pathways_all = pd.DataFrame()\n",
    "for i in range(num_runs):\n",
    "    simulated_GSEA_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"GSEA_stats\",\n",
    "        \"GSEA_simulated_data_\"+project_id+\"_\"+str(i)+\".txt\")\n",
    "    \n",
    "    #Read results\n",
    "    simulated_enriched_pathways = pd.read_csv(\n",
    "        simulated_GSEA_file,\n",
    "        header=0,\n",
    "        sep='\\t',\n",
    "        index_col=0)\n",
    "    \n",
    "    # Add ranks of simulated experiment\n",
    "    simulated_enriched_pathways['ranking'] = simulated_enriched_pathways[col_to_rank].rank(ascending = 0) \n",
    "    simulated_enriched_pathways = simulated_enriched_pathways.sort_values(by=col_to_rank, ascending=False)\n",
    "    \n",
    "    # Concatenate df\n",
    "    simulated_enriched_pathways_all = pd.concat([simulated_enriched_pathways_all,\n",
    "                                       simulated_enriched_pathways])\n",
    "    \n",
    "print(simulated_enriched_pathways_all.shape)\n",
    "simulated_enriched_pathways_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "simulated_enriched_pathways_stats = simulated_enriched_pathways_all.groupby(['ID'])[['enrichmentScore', 'pvalue', 'ranking']].agg({\n",
    "    col_to_rank:['mean', 'std','count'],\n",
    "    'pvalue':['median'],\n",
    "    'ranking':['median']\n",
    "})\n",
    "\n",
    "simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge template statistics with simulated statistics\n",
    "template_simulated_enriched_pathways_stats = template_enriched_pathways.merge(simulated_enriched_pathways_stats, \n",
    "                                                                              how='outer',\n",
    "                                                                              left_index=True,\n",
    "                                                                              right_index=True)\n",
    "template_simulated_enriched_pathways_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse columns\n",
    "median_pval_simulated = template_simulated_enriched_pathways_stats[('pvalue','median')]\n",
    "median_rank_simulated = template_simulated_enriched_pathways_stats[('ranking','median')]\n",
    "mean_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'mean')]\n",
    "std_test_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'std')]\n",
    "count_simulated = template_simulated_enriched_pathways_stats[(col_to_rank,'count')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculations for summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary = pd.DataFrame(data={'Pathway': template_simulated_enriched_pathways_stats.index,\n",
    "                             'P-value (Real)': template_simulated_enriched_pathways_stats['pvalue'],\n",
    "                             'Rank (Real)': template_simulated_enriched_pathways_stats['ranking'],\n",
    "                             'Test statistic (Real)': template_enriched_pathways[col_to_rank],\n",
    "                             'Median p-value (simulated)': median_pval_simulated ,\n",
    "                             'Median rank (simulated)': median_rank_simulated ,\n",
    "                             'Mean test statistic (simulated)': mean_test_simulated ,\n",
    "                             'Std deviation (simulated)': std_test_simulated,\n",
    "                             'Number of experiments (simulated)': count_simulated\n",
    "                            }\n",
    "                      )\n",
    "summary['Z score'] = (summary['Test statistic (Real)'] - summary['Mean test statistic (simulated)'])/summary['Std deviation (simulated)']\n",
    "summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save file\n",
    "summary_file = os.path.join(\n",
    "        local_dir, \n",
    "        \"pathway_summary_table.tsv\")\n",
    "\n",
    "summary.to_csv(summary_file, float_format='%.5f', sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:ranked_pathways] *",
   "language": "python",
   "name": "conda-env-ranked_pathways-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
